<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />

<meta name="date" content="2021-12-18" />

<title>Atrial fibrillation - Heart Left Ventricle</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Atrial fibrillation - Heart Left Ventricle</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-12-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree87fee8b56a4f94471388b3fc74b464d3b6efdd41targetblank87fee8ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/87fee8b56a4f94471388b3fc74b464d3b6efdd41" target="_blank">87fee8b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree87fee8b56a4f94471388b3fc74b464d3b6efdd41targetblank87fee8ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/87fee8b56a4f94471388b3fc74b464d3b6efdd41" target="_blank">87fee8b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd</code>) and HTML (<code>docs/Atrial_Fibrillation_Heart_Left_Ventricle.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f797004ce2d323bcc76bd17517b83b86e304fa44/analysis/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd" target="_blank">f797004</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/762fd7f08c8fdf8e702d763cace4f0cde9daad1d/analysis/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd" target="_blank">762fd7f</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/1f785cffd36f1af589f47f1a7373a30191b0dd6b/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">1f785cf</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/82c68fdd84120d702cc7b5233cbad9411174dfe9/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">82c68fd</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/9cf0e70ad824b1554fe1f2598c64b5fcd4f11146/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">9cf0e70</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/6b46de7ec1a695719b465266eedd482afc78cd71/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e22d0c906642196ab22d358cad154fab6f31a91d/docs/Atrial_Fibrillation_Heart_Left_Ventricle.html" target="_blank">e22d0c9</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a6c7890ef95af27ea685d302cd9ff9e62b9dcd36/analysis/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd" target="_blank">a6c7890</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
<td>
Start my new project
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre><code>[1] 10150</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1003  711  570  412  451  607  495  367  384  411  592  556  186  332  344  494 
  17   18   19   20   21   22 
 639  150  783  300  110  253 </code></pre>
<pre><code>[1] 6644</code></pre>
<pre><code>[1] 0.6545813</code></pre>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-4-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>        gene          snp 
0.0196935664 0.0002083061 </code></pre>
<pre><code>    gene      snp 
9.374271 8.577991 </code></pre>
<pre><code>[1] 1030836</code></pre>
<pre><code>[1]   10150 6839050</code></pre>
<pre><code>       gene         snp 
0.001817768 0.011854791 </code></pre>
<pre><code>[1] 0.01188393 0.13624409</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-5-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>       genename region_tag susie_pip        mu2          PVE         z num_eqtl
3169      PRRX1       1_84 0.9999999  153.79916 1.491985e-04 14.848088        2
3147      WIPF1      2_105 0.9993101 2124.21550 2.059251e-03  8.987351        2
2431    DNAJC12      10_44 0.9967458  850.85693 8.227187e-04 -5.328244        1
1214      GMCL1       2_46 0.9944249   60.60721 5.846645e-05 -8.434995        2
9840     NKX2-5      5_103 0.9942183   61.83714 5.964054e-05 -9.616279        2
10979     MSRB1       16_2 0.9667967   26.84161 2.517411e-05  5.929412        1
9417       PLEC       8_94 0.9666163   43.61991 4.090245e-05  3.093844        2
6663       JAM2       21_9 0.9633616   22.37976 2.091487e-05  4.569598        2
7794      KDM1B       6_14 0.9581444   64.33734 5.980046e-05 -9.000000        1
2987       GNB4      3_110 0.9537748   32.01351 2.962031e-05 -5.869048        1
11279      DPF3      14_34 0.9246749   30.95765 2.776946e-05  6.100000        1
2960       PCCB       3_84 0.9229236  849.82984 7.608659e-04 -5.382353        1
9960        PGP       16_2 0.9051693   28.48915 2.501611e-05  5.943820        1
1551    CCDC134      22_17 0.8683170   26.15261 2.202945e-05 -5.180556        1
12376 LINC01314      15_37 0.8300211   20.71969 1.668334e-05 -4.379164        4
4891        NPL       1_90 0.8251030   18.75569 1.501245e-05 -3.825175        1
5042      PSMB7       9_64 0.8211676   25.19921 2.007378e-05 -4.820896        1
8847    WASHC2C      10_31 0.8186904   19.80281 1.572740e-05 -3.972152        1
12133 LINC00964       8_82 0.8180590   22.20129 1.761867e-05  4.554265        2
3725       SSPN      12_18 0.7969230   42.42202 3.279579e-05 -6.791667        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-6-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>        genename region_tag    susie_pip       mu2          PVE         z
3147       WIPF1      2_105 0.9993100519 2124.2155 2.059251e-03  8.987351
929        BAZ2A      12_35 0.0000000000 1327.6344 0.000000e+00 -5.942857
10471     ZNF292       6_59 0.0000000000 1098.8945 0.000000e+00 -5.216216
4819         CGA       6_59 0.0000000000  912.8138 0.000000e+00  4.780822
7442      GPR155      2_105 0.0000000000  887.5060 0.000000e+00 -4.929575
2431     DNAJC12      10_44 0.9967457594  850.8569 8.227187e-04 -5.328244
2960        PCCB       3_84 0.9229235511  849.8298 7.608659e-04 -5.382353
6231       HERC4      10_44 0.0519716850  784.1117 3.953258e-05 -5.368895
7690     SLC35A1       6_59 0.0000000000  731.3839 0.000000e+00 -4.906053
4816        ORC3       6_59 0.0000000000  575.4343 0.000000e+00 -4.305556
11646 AC010894.3      2_105 0.0000000000  567.5177 0.000000e+00  2.807292
2445        WNT3      17_28 0.0012204211  558.1877 6.608462e-07 -5.270588
10151     ARL17A      17_28 0.0006810977  537.9069 3.554078e-07 -4.159800
10265       MAPT      17_28 0.0246154818  498.1750 1.189599e-05  5.606451
5236        MYPN      10_44 0.0557487640  467.4182 2.527850e-05  4.754237
11600    MKRN2OS        3_9 0.0000000000  326.2847 0.000000e+00 -2.927621
1450       SIRT1      10_44 0.0073685535  237.7219 1.699268e-06 -2.641975
856      PPP2R3A       3_84 0.0000000000  231.0518 0.000000e+00 -2.690141
8296        PDHB       3_40 0.0160173130  201.8814 3.136869e-06  2.296671
11694     ARL17B      17_28 0.0028320351  185.4496 5.094893e-07  3.095544
      num_eqtl
3147         2
929          1
10471        1
4819         1
7442         3
2431         1
2960         1
6231         2
7690         2
4816         1
11646        2
2445         1
10151        2
10265        2
5236         1
11600        2
1450         1
856          1
8296         2
11694        2</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>      genename region_tag  susie_pip        mu2          PVE          z
3147     WIPF1      2_105 0.99931005 2124.21550 2.059251e-03   8.987351
2431   DNAJC12      10_44 0.99674576  850.85693 8.227187e-04  -5.328244
2960      PCCB       3_84 0.92292355  849.82984 7.608659e-04  -5.382353
3169     PRRX1       1_84 0.99999992  153.79916 1.491985e-04  14.848088
11458   ZSWIM8      10_49 0.72951933  109.78862 7.769705e-05 -11.216495
7794     KDM1B       6_14 0.95814438   64.33734 5.980046e-05  -9.000000
9840    NKX2-5      5_103 0.99421828   61.83714 5.964054e-05  -9.616279
1214     GMCL1       2_46 0.99442495   60.60721 5.846645e-05  -8.434995
9417      PLEC       8_94 0.96661634   43.61991 4.090245e-05   3.093844
6231     HERC4      10_44 0.05197169  784.11168 3.953258e-05  -5.368895
2927     NR3C1       5_84 0.79565655   45.98656 3.549498e-05  -8.392405
3725      SSPN      12_18 0.79692298   42.42202 3.279579e-05  -6.791667
2987      GNB4      3_110 0.95377479   32.01351 2.962031e-05  -5.869048
11279     DPF3      14_34 0.92467491   30.95765 2.776946e-05   6.100000
3396     CASQ2       1_72 0.64365572   44.46248 2.776244e-05   6.684932
2401     PITX3      10_65 0.70424642   39.87615 2.724258e-05  -6.793786
5236      MYPN      10_44 0.05574876  467.41822 2.527850e-05   4.754237
10979    MSRB1       16_2 0.96679672   26.84161 2.517411e-05   5.929412
9960       PGP       16_2 0.90516927   28.48915 2.501611e-05   5.943820
1551   CCDC134      22_17 0.86831704   26.15261 2.202945e-05  -5.180556
      num_eqtl
3147         2
2431         1
2960         1
3169         2
11458        1
7794         1
9840         2
1214         2
9417         2
6231         2
2927         1
3725         1
2987         1
11279        1
3396         1
2401         2
5236         1
10979        1
9960         1
1551         1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>      genename region_tag    susie_pip        mu2          PVE          z
3169     PRRX1       1_84 9.999999e-01  153.79916 1.491985e-04  14.848088
7976   SYNPO2L      10_48 1.466304e-01  134.47936 1.912890e-05 -11.782609
11458   ZSWIM8      10_49 7.295193e-01  109.78862 7.769705e-05 -11.216495
7134    ZBTB7B       1_76 1.008001e-04  120.87046 1.181930e-08  10.638889
504      SYNE2      14_29 4.741777e-02   83.62062 3.846492e-06  10.029851
9840    NKX2-5      5_103 9.942183e-01   61.83714 5.964054e-05  -9.616279
7794     KDM1B       6_14 9.581444e-01   64.33734 5.980046e-05  -9.000000
3147     WIPF1      2_105 9.993101e-01 2124.21550 2.059251e-03   8.987351
3876       DEK       6_14 9.860627e-02   62.21941 5.951697e-06  -8.651764
1214     GMCL1       2_46 9.944249e-01   60.60721 5.846645e-05  -8.434995
7445     DCST2       1_76 6.958988e-04   82.23437 5.551494e-08  -8.408993
2927     NR3C1       5_84 7.956566e-01   45.98656 3.549498e-05  -8.392405
9371     MYOZ1      10_48 4.594225e-02   72.94692 3.251095e-06   8.282302
5279      HCN4      15_35 1.027118e-03  142.65541 1.421409e-07   7.582677
554       MXD1       2_46 5.931507e-02   49.69584 2.859536e-06   7.537313
2039     ASAH1       8_21 2.248119e-05   67.20598 1.465675e-09  -7.443420
2401     PITX3      10_65 7.042464e-01   39.87615 2.724258e-05  -6.793786
3725      SSPN      12_18 7.969230e-01   42.42202 3.279579e-05  -6.791667
7823      CFL2       14_9 2.408814e-02  112.05711 2.618504e-06  -6.728039
5803    ADAM15       1_76 7.026909e-04   51.73422 3.526571e-08  -6.717647
      num_eqtl
3169         2
7976         1
11458        1
7134         1
504          1
9840         2
7794         1
3147         2
3876         3
1214         2
7445         2
2927         1
9371         2
5279         1
554          1
2039         2
2401         2
3725         1
7823         2
5803         1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-9-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-9-2.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.007783251</code></pre>
<pre><code>      genename region_tag    susie_pip        mu2          PVE          z
3169     PRRX1       1_84 9.999999e-01  153.79916 1.491985e-04  14.848088
7976   SYNPO2L      10_48 1.466304e-01  134.47936 1.912890e-05 -11.782609
11458   ZSWIM8      10_49 7.295193e-01  109.78862 7.769705e-05 -11.216495
7134    ZBTB7B       1_76 1.008001e-04  120.87046 1.181930e-08  10.638889
504      SYNE2      14_29 4.741777e-02   83.62062 3.846492e-06  10.029851
9840    NKX2-5      5_103 9.942183e-01   61.83714 5.964054e-05  -9.616279
7794     KDM1B       6_14 9.581444e-01   64.33734 5.980046e-05  -9.000000
3147     WIPF1      2_105 9.993101e-01 2124.21550 2.059251e-03   8.987351
3876       DEK       6_14 9.860627e-02   62.21941 5.951697e-06  -8.651764
1214     GMCL1       2_46 9.944249e-01   60.60721 5.846645e-05  -8.434995
7445     DCST2       1_76 6.958988e-04   82.23437 5.551494e-08  -8.408993
2927     NR3C1       5_84 7.956566e-01   45.98656 3.549498e-05  -8.392405
9371     MYOZ1      10_48 4.594225e-02   72.94692 3.251095e-06   8.282302
5279      HCN4      15_35 1.027118e-03  142.65541 1.421409e-07   7.582677
554       MXD1       2_46 5.931507e-02   49.69584 2.859536e-06   7.537313
2039     ASAH1       8_21 2.248119e-05   67.20598 1.465675e-09  -7.443420
2401     PITX3      10_65 7.042464e-01   39.87615 2.724258e-05  -6.793786
3725      SSPN      12_18 7.969230e-01   42.42202 3.279579e-05  -6.791667
7823      CFL2       14_9 2.408814e-02  112.05711 2.618504e-06  -6.728039
5803    ADAM15       1_76 7.026909e-04   51.73422 3.526571e-08  -6.717647
      num_eqtl
3169         2
7976         1
11458        1
7134         1
504          1
9840         2
7794         1
3147         2
3876         3
1214         2
7445         2
2927         1
9371         2
5279         1
554          1
2039         2
2401         2
3725         1
7823         2
5803         1</code></pre>
</div>
<div id="snps-with-highest-pips" class="section level2">
<h2>SNPs with highest PIPs</h2>
<pre><code>                id region_tag susie_pip        mu2          PVE           z
28637   rs12404927       1_75 1.0000000 1404.55102 1.362536e-03  -1.5517241
28638    rs7536152       1_75 1.0000000 1439.64649 1.396582e-03  -6.4626866
28731   rs34515871       1_76 1.0000000  296.98405 2.881002e-04  17.0000000
32608   rs12142529       1_83 1.0000000  145.38857 1.410395e-04  12.0396040
32633  rs112797273       1_83 1.0000000  101.40197 9.836867e-05   9.4903846
183948   rs1906615       4_72 1.0000000 3299.17745 3.200487e-03  45.1604938
183953   rs7440714       4_72 1.0000000 1498.40640 1.453584e-03  -9.2183406
272791 rs111990232       6_59 1.0000000 1672.21316 1.622191e-03   0.5392670
472755  rs17287293      12_18 1.0000000  150.43066 1.459307e-04 -10.2947368
556238 rs140798119      15_35 1.0000000  282.62226 2.741680e-04   3.9950739
556241  rs74022964      15_35 1.0000000  340.33003 3.301495e-04  12.5777778
698783   rs4972703      2_105 1.0000000 3592.88497 3.485409e-03  -0.7842670
703489 rs114415025        3_9 1.0000000 2723.92507 2.642443e-03   1.2192513
728978 rs201862093       7_48 1.0000000 1114.36488 1.081030e-03  -0.8625000
765251  rs76106073      10_44 1.0000000  836.85538 8.118220e-04   0.6572238
768375  rs61937778      12_35 1.0000000 1898.73730 1.841939e-03  -1.3056995
28735    rs2878412       1_76 1.0000000  104.66388 1.015330e-04   5.4250000
423415  rs60469668      10_66 1.0000000  171.39264 1.662657e-04  16.9647059
28733   rs11576820       1_76 1.0000000  131.66128 1.277228e-04   8.0243902
592256  rs62056842      17_28 1.0000000  764.07813 7.412218e-04   1.1269036
733889  rs62643683       8_21 1.0000000  262.98617 2.551193e-04   0.8007519
183979 rs186546224       4_72 1.0000000  278.28019 2.699558e-04  -6.8398876
117614   rs7373492       3_28 1.0000000   57.12050 5.541183e-05  -7.6417323
323213  rs62469005       7_70 1.0000000  248.90606 2.414604e-04  -3.8636364
323204  rs28557111       7_70 0.9999995  268.50949 2.604773e-04  13.6818182
299063  rs12112152       7_15 0.9999995   45.05583 4.370803e-05   6.9358974
28738   rs10908445       1_76 0.9999993  192.70836 1.869436e-04 -13.1470588
576584   rs6499606      16_39 0.9999989  103.66865 1.005674e-04  12.6376812
785055   rs8005417       14_9 0.9999977  418.17010 4.056602e-04   0.9863946
323219  rs12669209       7_70 0.9999965  107.24435 1.040359e-04  -4.2937500
423413   rs7094488      10_66 0.9999961  105.12014 1.019752e-04  14.1060606
96493    rs1975584      2_118 0.9999837   71.18500 6.905447e-05   1.2311828
804323 rs140185678       16_2 0.9999825   44.60953 4.327434e-05   7.6100917
280210  rs11756438       6_79 0.9999688   47.48500 4.606312e-05  -8.2089552
117485 rs116202356       3_27 0.9999532   36.40320 3.531260e-05   6.0813559
176633   rs1458038       4_54 0.9998799   34.94748 3.389801e-05   6.0277778
661464    rs464901       22_3 0.9997751   44.32757 4.299190e-05  -7.0555556
121514  rs12330500       3_40 0.9996970  506.36533 4.910693e-04   0.5641026
397657   rs1886296       9_73 0.9995450   34.53850 3.349008e-05   5.9863014
734589      rs7508       8_21 0.9993753   79.71601 7.728312e-05   9.4800000
725018   rs3176326       6_29 0.9991706   50.71157 4.915381e-05  -7.3647059
117627  rs10428132       3_28 0.9991198   74.57949 7.228487e-05   9.2058824
586504  rs72811292      17_11 0.9982520   39.85061 3.859096e-05  -6.5233645
39384   rs12353975      1_102 0.9981456   35.67154 3.454031e-05  -4.3000000
184535   rs7700110       4_73 0.9978791   45.09924 4.365737e-05   7.0666667
117601   rs9824157       3_28 0.9976315   29.97394 2.900844e-05   3.8732394
734231    rs208757       8_21 0.9970887  239.59396 2.317502e-04   7.6629213
232614 rs199992924       5_68 0.9967799 1308.36140 1.265137e-03  -0.6309524
663223    rs133902       22_7 0.9963536   31.00376 2.996666e-05   6.1617647
232434    rs338623       5_68 0.9953355   66.77245 6.447290e-05  -7.9855072
40060    rs4951023      1_104 0.9951712   32.47858 3.135489e-05   5.5970149
39383    rs6427989      1_102 0.9941339   35.76701 3.449355e-05  -4.4142857
592472  rs75230966      17_28 0.9928717   43.15503 4.156569e-05   6.0582524
576592    rs876727      16_39 0.9925728   48.76988 4.695961e-05 -10.0000000
275859   rs9496567       6_67 0.9919760   26.48986 2.549126e-05   5.1125000
801039  rs12908004      15_37 0.9873706   57.56113 5.513405e-05   8.1333333
183968   rs4631108       4_72 0.9858853  278.72914 2.665748e-04 -17.6376812
576597  rs60602157      16_39 0.9858538   38.73318 3.704299e-05  10.4415584
526368  rs74968516       14_8 0.9857359   34.86384 3.333851e-05  -2.0924370
657485   rs7282237      21_16 0.9765780   24.87091 2.356183e-05  -4.7605634
534174   rs2738413      14_29 0.9761605  111.24337 1.053430e-04 -11.6119403
366949   rs7460121       8_88 0.9755516   24.08834 2.279647e-05   4.7500000
323211   rs4730742       7_70 0.9739413  148.25445 1.400719e-04  -5.3181818
576589   rs4788691      16_39 0.9737540   39.90042 3.769096e-05  -1.6900000
331163  rs35760656       7_93 0.9721535   35.73611 3.370176e-05  -6.3536585
284152    rs958747       6_89 0.9716551   24.32337 2.292695e-05  -4.9710145
2727      rs284278        1_7 0.9709039   35.05628 3.301813e-05  -6.0857143
280167  rs77435894       6_78 0.9646884   28.76469 2.691889e-05   6.1111111
785061   rs8011559       14_9 0.9644048  426.28574 3.988142e-04   5.6913580
561807  rs12898337      15_48 0.9627997   35.30067 3.297079e-05  -6.0579710
58545    rs7578482       2_15 0.9603813   32.19728 2.999669e-05   5.7435897
555608    rs745636      15_33 0.9584917   25.47962 2.369146e-05  -5.0740741
25981    rs4839174       1_69 0.9562825   33.83277 3.138588e-05   5.9459459
99263   rs35880620      2_125 0.9544729   39.72780 3.678482e-05  -6.2168675
610855  rs17794590      18_24 0.9544647   24.57014 2.274981e-05  -4.8242424
96468   rs11889306      2_118 0.9532842   45.60961 4.217831e-05   6.7391304
728771  rs74910854       7_48 0.9504108   30.92661 2.851373e-05   5.4878049
705484   rs9830653       3_84 0.9497164  834.68211 7.689985e-04  -0.9344262
784984  rs73241997       14_9 0.9463663  271.39025 2.491517e-04   7.8817204
183920   rs1823291       4_72 0.9443849  275.38267 2.522877e-04 -19.4520548
32655    rs7522387       1_83 0.9380199   59.33695 5.399427e-05   6.6375000
28693     rs906280       1_75 0.9352012   59.46409 5.394737e-05   7.9111111
492885  rs12425471      12_69 0.9315032   41.65186 3.763823e-05  -5.2500000
506242   rs1326122      13_21 0.9280182   24.16457 2.175434e-05   4.7543103
729019 rs112661041       7_48 0.9246584 1098.87113 9.856858e-04   4.6532258
280117     rs89107       6_78 0.9200337   47.40943 4.231350e-05  -9.5606061
372218   rs1594768        9_9 0.9191704   24.06779 2.146064e-05  -4.6911765
218017 rs114414434       5_30 0.9100084   25.52863 2.253634e-05  -4.9718310
416198  rs60632610      10_48 0.9078072  142.32178 1.253359e-04 -12.2395833
5627    rs10917072       1_15 0.9035741   33.32457 2.921048e-05   5.8235294
666612  rs11705586      22_15 0.9032885   25.71531 2.253350e-05   4.8988764
698793   rs1367220      2_105 0.9027873 3624.47361 3.174247e-03   7.2207792
239095   rs6894302       5_84 0.8962639   34.59893 3.008215e-05  -7.2933333
232607   rs4073838       5_68 0.8907151 1336.89003 1.155167e-03  -5.9545455
272788    rs371814       6_59 0.8887549 1684.18630 1.452053e-03   6.1911765
252811  rs73724866       6_13 0.8854526   93.92663 8.067974e-05 -10.4141414
556230    rs519946      15_34 0.8802637   34.42175 2.939383e-05   5.9242424
497493   rs6560886      12_82 0.8786081   28.83992 2.458101e-05   5.6666667
769560  rs71454237      12_43 0.8731783   45.03633 3.814840e-05  -7.3809524
28729   rs12128882       1_76 0.8660530   31.94687 2.684004e-05   1.0731707
469758  rs12821447      12_12 0.8604052   24.28342 2.026858e-05   4.5128205
622090  rs73919353       19_6 0.8573709   29.15094 2.424554e-05   5.4076433
455650    rs565449      11_54 0.8568684   28.25040 2.348276e-05  -5.2535211
67841     rs243080       2_40 0.8464105   23.94966 1.966486e-05  -4.5522388
245279  rs62377226      5_100 0.8419746   33.84803 2.764667e-05   5.8493151
583304  rs75329315       17_2 0.8321222   26.35764 2.127669e-05  -5.2460317
121517      rs6924       3_40 0.8238581  510.02479 4.076187e-04   4.2133333
498623   rs7321083       13_3 0.8169737   24.61195 1.950584e-05   4.8292683
304963 rs145593380       7_26 0.8132918   29.70683 2.343760e-05   5.3333333
151266    rs720389      3_114 0.8068588   23.55072 1.843368e-05   4.2898551</code></pre>
</div>
<div id="snps-with-largest-effect-sizes" class="section level2">
<h2>SNPs with largest effect sizes</h2>
<pre><code>                id region_tag    susie_pip      mu2          PVE         z
698793   rs1367220      2_105 9.027873e-01 3624.474 3.174247e-03  7.220779
698794   rs1367219      2_105 8.971134e-02 3615.477 3.146469e-04  7.103896
698789  rs10168156      2_105 2.265398e-01 3615.000 7.944439e-04  7.233766
698787   rs6713018      2_105 5.859132e-02 3613.822 2.054047e-04  7.168831
698778   rs1864453      2_105 1.251599e-01 3612.780 4.386490e-04  7.207792
698770   rs2033315      2_105 5.844195e-02 3609.583 2.046408e-04  7.064935
698781   rs6707162      2_105 1.730083e-02 3603.368 6.047640e-05  7.090909
698782   rs6735680      2_105 1.150635e-02 3602.875 4.021584e-05  7.077922
698771   rs2033314      2_105 1.165286e-02 3602.717 4.072611e-05  7.155844
698773  rs28485554      2_105 2.399129e-03 3600.749 8.380248e-06  7.000000
698783   rs4972703      2_105 1.000000e+00 3592.885 3.485409e-03 -0.784267
698804  rs10803884      2_105 3.412607e-05 3550.073 1.175260e-07  7.285714
698812   rs6738901      2_105 6.033312e-05 3535.783 2.069435e-07 -7.272727
698797  rs12466643      2_105 1.495366e-05 3532.253 5.124007e-08  7.272727
698828  rs35368253      2_105 1.910799e-05 3515.570 6.516601e-08 -7.256410
698814  rs10197521      2_105 2.167801e-08 3505.736 7.372403e-11 -6.547619
698816  rs34661753      2_105 1.912728e-06 3466.123 6.431433e-09 -7.076923
698767   rs7590328      2_105 1.204450e-07 3454.423 4.036218e-10  6.784810
183948   rs1906615       4_72 1.000000e+00 3299.177 3.200487e-03 45.160494
698765   rs6433497      2_105 2.090339e-11 2907.446 5.895746e-14  6.345679
698766   rs2115874      2_105 2.801248e-11 2905.297 7.895008e-14  6.419753
698758   rs1430185      2_105 1.876388e-12 2842.067 5.173296e-15  6.185185
698745   rs1991601      2_105 5.335732e-13 2821.002 1.460185e-15  5.950617
703494  rs56204325        3_9 7.189264e-01 2763.521 1.927337e-03  7.865672
703475   rs1969154        3_9 3.789176e-01 2761.449 1.015061e-03  7.850746
703474   rs6799179        3_9 3.766987e-01 2761.418 1.009106e-03  7.850746
703484  rs12714880        3_9 1.905204e-01 2760.752 5.102457e-04  7.820896
703442   rs6763619        3_9 6.794745e-03 2742.502 1.807717e-05  7.776119
703441  rs13320486        3_9 6.774412e-03 2742.480 1.802293e-05  7.776119
703443   rs9883561        3_9 5.623830e-04 2740.038 1.494855e-06  7.676471
703469   rs9821066        3_9 3.249060e-05 2729.436 8.602823e-08  7.588235
703464   rs9872103        3_9 1.785730e-04 2728.262 4.726201e-07  7.686567
703465   rs9871991        3_9 3.800932e-04 2727.319 1.005626e-06  7.716418
703470   rs9877165        3_9 4.469283e-05 2727.313 1.182451e-07  7.626866
703476   rs1985428        3_9 3.173254e-05 2727.176 8.395150e-08  7.611940
703489 rs114415025        3_9 1.000000e+00 2723.925 2.642443e-03  1.219251
703461   rs6791647        3_9 2.233448e-05 2720.842 5.895080e-08  7.611940
703456   rs4447735        3_9 3.120521e-05 2720.726 8.236113e-08  7.626866
703462   rs6767504        3_9 7.609822e-06 2717.065 2.005788e-08  7.597015
703459   rs4321514        3_9 7.424475e-06 2716.792 1.956737e-08  7.597015
703457   rs4299468        3_9 7.118141e-06 2716.475 1.875783e-08  7.597015
703452   rs3901665        3_9 1.258296e-05 2715.352 3.314511e-08  7.626866
703488  rs56082700        3_9 0.000000e+00 2702.262 0.000000e+00  2.696970
703471   rs9822382        3_9 0.000000e+00 2678.591 0.000000e+00  4.277228
703447   rs9870070        3_9 0.000000e+00 2676.505 0.000000e+00  4.516129
703445   rs9850214        3_9 0.000000e+00 2676.317 0.000000e+00  4.483871
703448   rs9850227        3_9 0.000000e+00 2675.871 0.000000e+00  4.397849
703558   rs3732675        3_9 1.173251e-08 2674.016 3.043446e-11  7.735294
703446   rs9850480        3_9 0.000000e+00 2664.144 0.000000e+00  4.505376
703492  rs11706068        3_9 2.610084e-06 2600.801 6.585245e-09  7.970588</code></pre>
</div>
<div id="snps-with-highest-pve" class="section level2">
<h2>SNPs with highest PVE</h2>
<pre><code>                id region_tag  susie_pip       mu2          PVE           z
698783   rs4972703      2_105 1.00000000 3592.8850 0.0034854089  -0.7842670
183948   rs1906615       4_72 1.00000000 3299.1775 0.0032004872  45.1604938
698793   rs1367220      2_105 0.90278725 3624.4736 0.0031742475   7.2207792
703489 rs114415025        3_9 1.00000000 2723.9251 0.0026424427   1.2192513
703494  rs56204325        3_9 0.71892638 2763.5212 0.0019273369   7.8656716
768375  rs61937778      12_35 1.00000000 1898.7373 0.0018419393  -1.3056995
272791 rs111990232       6_59 1.00000000 1672.2132 0.0016221913   0.5392670
183953   rs7440714       4_72 1.00000000 1498.4064 0.0014535837  -9.2183406
272788    rs371814       6_59 0.88875486 1684.1863 0.0014520532   6.1911765
28638    rs7536152       1_75 1.00000000 1439.6465 0.0013965815  -6.4626866
703515   rs4642101        3_9 0.71504872 1969.7326 0.0013663229  10.2318841
28637   rs12404927       1_75 1.00000000 1404.5510 0.0013625359  -1.5517241
768315   rs2860482      12_35 0.72039239 1945.5816 0.0013596558  -7.1052632
232614 rs199992924       5_68 0.99677991 1308.3614 0.0012651366  -0.6309524
232607   rs4073838       5_68 0.89071507 1336.8900 0.0011551674  -5.9545455
768370   rs7313074      12_35 0.59547601 1953.4234 0.0011284208  -6.9868421
728978 rs201862093       7_48 1.00000000 1114.3649 0.0010810302  -0.8625000
703475   rs1969154        3_9 0.37891760 2761.4487 0.0010150611   7.8507463
703474   rs6799179        3_9 0.37669868 2761.4183 0.0010091059   7.8507463
729019 rs112661041       7_48 0.92465841 1098.8711 0.0009856858   4.6532258
765251  rs76106073      10_44 1.00000000  836.8554 0.0008118220   0.6572238
698789  rs10168156      2_105 0.22653976 3615.0005 0.0007944439   7.2337662
705484   rs9830653       3_84 0.94971640  834.6821 0.0007689985  -0.9344262
768373   rs4759256      12_35 0.40562812 1952.8889 0.0007684507  -6.9736842
592256  rs62056842      17_28 1.00000000  764.0781 0.0007412218   1.1269036
272792    rs384318       6_59 0.37973072 1682.6223 0.0006198303   6.1029412
728888  rs12154789       7_48 0.51476523 1098.6922 0.0005486504   4.4923077
703514   rs7650482        3_9 0.28023466 1973.5020 0.0005365002  10.1571429
703484  rs12714880        3_9 0.19052042 2760.7520 0.0005102457   7.8208955
121514  rs12330500       3_40 0.99969702  506.3653 0.0004910693   0.5641026
704796 rs199808025       3_84 0.56221627  819.6038 0.0004470106   0.3170732
698778   rs1864453      2_105 0.12515989 3612.7802 0.0004386490   7.2077922
272793   rs1145714       6_59 0.26668394 1679.0994 0.0004343939   6.1029412
121517      rs6924       3_40 0.82385812  510.0248 0.0004076187   4.2133333
785055   rs8005417       14_9 0.99999767  418.1701 0.0004056602   0.9863946
785061   rs8011559       14_9 0.96440477  426.2857 0.0003988142   5.6913580
232606   rs4235764       5_68 0.29404637 1335.1711 0.0003808581  -5.8787879
768310   rs7978685      12_35 0.19327209 1942.9081 0.0003642770  -7.0657895
556241  rs74022964      15_35 1.00000000  340.3300 0.0003301495  12.5777778
698794   rs1367219      2_105 0.08971134 3615.4774 0.0003146469   7.1038961
232609   rs4235768       5_68 0.24023863 1335.2967 0.0003111939  -5.8030303
272750   rs9444476       6_59 0.19310251 1660.7317 0.0003110984  -6.3088235
28731   rs34515871       1_76 1.00000000  296.9840 0.0002881002  17.0000000
272775   rs9444488       6_59 0.17207131 1665.5818 0.0002780256  -6.2647059
556238 rs140798119      15_35 1.00000000  282.6223 0.0002741680   3.9950739
183979 rs186546224       4_72 0.99999999  278.2802 0.0002699558  -6.8398876
183968   rs4631108       4_72 0.98588526  278.7291 0.0002665748 -17.6376812
323204  rs28557111       7_70 0.99999953  268.5095 0.0002604773  13.6818182
733889  rs62643683       8_21 0.99999999  262.9862 0.0002551193   0.8007519
183920   rs1823291       4_72 0.94438487  275.3827 0.0002522877 -19.4520548</code></pre>
</div>
<div id="snps-with-largest-z-scores" class="section level2">
<h2>SNPs with largest z scores</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-13-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                id region_tag    susie_pip       mu2          PVE         z
183948   rs1906615       4_72 1.000000e+00 3299.1775 3.200487e-03  45.16049
183949  rs75725917       4_72 0.000000e+00 1501.2534 0.000000e+00  42.33663
183947   rs1906611       4_72 0.000000e+00 1386.7531 0.000000e+00  40.97143
183946  rs28521134       4_72 0.000000e+00 1323.8641 0.000000e+00  40.22772
183942  rs10019689       4_72 0.000000e+00 1314.7532 0.000000e+00  40.11000
183943  rs76013973       4_72 0.000000e+00 1309.8604 0.000000e+00  40.04902
183945  rs12639820       4_72 0.000000e+00 1308.0495 0.000000e+00  40.03922
183941  rs12647393       4_72 0.000000e+00 1303.9357 0.000000e+00  39.98020
183944  rs74496596       4_72 0.000000e+00 1284.8290 0.000000e+00  39.75728
183940  rs12647316       4_72 0.000000e+00 1271.3853 0.000000e+00  39.55882
183939   rs4529121       4_72 0.000000e+00 1275.2573 0.000000e+00  39.55340
183933  rs12650829       4_72 0.000000e+00  605.3700 0.000000e+00  31.54444
183956   rs3866831       4_72 0.000000e+00 1596.6468 0.000000e+00 -30.10000
183955   rs6533530       4_72 0.000000e+00 1580.2990 0.000000e+00 -29.91429
183938  rs12644107       4_72 0.000000e+00  465.5759 0.000000e+00  25.98851
183936   rs2723318       4_72 0.000000e+00  813.2565 0.000000e+00 -25.45833
183931   rs2218698       4_72 0.000000e+00  771.3648 0.000000e+00 -25.09589
183934   rs2197814       4_72 0.000000e+00  772.0146 0.000000e+00  24.59722
183932   rs1448799       4_72 0.000000e+00  770.9787 0.000000e+00  24.56944
183930 rs112927894       4_72 0.000000e+00  741.1342 0.000000e+00  24.50685
183920   rs1823291       4_72 9.443849e-01  275.3827 2.522877e-04 -19.45205
183924   rs2723296       4_72 5.561070e-02  266.6973 1.438757e-05 -19.02740
183929  rs11724067       4_72 0.000000e+00  654.4966 0.000000e+00 -18.94059
183923   rs2044674       4_72 4.429420e-06  242.9736 1.044038e-09  18.58108
183968   rs4631108       4_72 9.858853e-01  278.7291 2.665748e-04 -17.63768
183967   rs1906613       4_72 1.411474e-02  268.6140 3.678003e-06  17.34783
28731   rs34515871       1_76 1.000000e+00  296.9840 2.881002e-04  17.00000
423415  rs60469668      10_66 1.000000e+00  171.3926 1.662657e-04  16.96471
183928  rs13111704       4_72 0.000000e+00  700.2351 0.000000e+00 -16.91743
28734   rs12058931       1_76 2.023295e-04  220.3706 4.325369e-08  16.54054
183952   rs4124159       4_72 0.000000e+00 1664.4211 0.000000e+00 -16.14414
183950   rs1906606       4_72 0.000000e+00 1659.0494 0.000000e+00 -16.02703
183957   rs4032974       4_72 0.000000e+00 1649.6039 0.000000e+00 -15.86726
183977  rs17513625       4_72 0.000000e+00  196.2481 0.000000e+00  15.71130
183954  rs10006881       4_72 0.000000e+00 1641.6919 0.000000e+00 -15.68750
423411  rs12572965      10_66 2.787622e-04  132.8475 3.592507e-08  15.27059
423410  rs56965730      10_66 2.796755e-04  131.7056 3.573296e-08  15.21176
183922  rs12642151       4_72 7.894574e-12  227.0161 1.738585e-15  14.14286
423413   rs7094488      10_66 9.999961e-01  105.1201 1.019752e-04  14.10606
672306    rs577676       1_84 4.329934e-04  124.1225 5.213653e-08 -13.77612
323204  rs28557111       7_70 9.999995e-01  268.5095 2.604773e-04  13.68182
672415    rs680084       1_84 6.024752e-04  136.4064 7.972312e-08 -13.53731
672386    rs608930       1_84 1.253762e-03  138.0822 1.679436e-07 -13.52239
672422    rs629234       1_84 5.624740e-04  135.5991 7.398942e-08 -13.52239
672340    rs598993       1_84 9.849966e-04  137.8588 1.317285e-07 -13.51515
672343    rs541557       1_84 9.796191e-04  137.8182 1.309707e-07 -13.51515
672429    rs525489       1_84 1.190958e-03  137.5496 1.589156e-07 -13.49254
672430    rs503706       1_84 1.177181e-03  137.4686 1.569846e-07 -13.49254
672350    rs539045       1_84 8.984718e-04  136.7722 1.192100e-07 -13.48485
672356    rs580487       1_84 8.877917e-04  136.6820 1.177153e-07 -13.48485</code></pre>
</div>
<div id="gene-set-enrichment-for-genes-with-pip0.8" class="section level2">
<h2>Gene set enrichment for genes with PIP&gt;0.8</h2>
<pre><code>Welcome to enrichR
Checking connection ... </code></pre>
<pre><code>Enrichr ... Connection is Live!
FlyEnrichr ... Connection is available!
WormEnrichr ... Connection is available!
YeastEnrichr ... Connection is available!
FishEnrichr ... Connection is available!</code></pre>
<pre><code>[1] 60</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-14-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-14-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-14-2.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                     Term Overlap Adjusted.P.value          Genes
1 sarcolemma (GO:0042383)    3/52       0.04598333 SSPN;BVES;PLEC
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre><code>MRPS31 gene(s) from the input list not found in DisGeNET CURATEDLINC01314 gene(s) from the input list not found in DisGeNET CURATEDGCN1 gene(s) from the input list not found in DisGeNET CURATEDFAM46C gene(s) from the input list not found in DisGeNET CURATEDMICU2 gene(s) from the input list not found in DisGeNET CURATEDTOMM5 gene(s) from the input list not found in DisGeNET CURATEDCMTM5 gene(s) from the input list not found in DisGeNET CURATEDGPRIN3 gene(s) from the input list not found in DisGeNET CURATEDSPNS1 gene(s) from the input list not found in DisGeNET CURATEDZSWIM8 gene(s) from the input list not found in DisGeNET CURATEDVPS37B gene(s) from the input list not found in DisGeNET CURATEDSRL gene(s) from the input list not found in DisGeNET CURATEDC9orf43 gene(s) from the input list not found in DisGeNET CURATEDCTNNAL1 gene(s) from the input list not found in DisGeNET CURATEDSSSCA1 gene(s) from the input list not found in DisGeNET CURATEDGMCL1 gene(s) from the input list not found in DisGeNET CURATEDLINC00964 gene(s) from the input list not found in DisGeNET CURATEDRAB1A gene(s) from the input list not found in DisGeNET CURATEDSTK11IP gene(s) from the input list not found in DisGeNET CURATEDJAM2 gene(s) from the input list not found in DisGeNET CURATEDC8orf48 gene(s) from the input list not found in DisGeNET CURATEDWASHC2C gene(s) from the input list not found in DisGeNET CURATEDPWP1 gene(s) from the input list not found in DisGeNET CURATEDPAFAH1B2 gene(s) from the input list not found in DisGeNET CURATEDADSS gene(s) from the input list not found in DisGeNET CURATEDSEC23IP gene(s) from the input list not found in DisGeNET CURATED</code></pre>
<pre><code>                                 Description          FDR Ratio  BgRatio
89            Paroxysmal atrial fibrillation 1.518444e-13 13/34 156/9703
174           Persistent atrial fibrillation 1.518444e-13 13/34 156/9703
192             familial atrial fibrillation 1.518444e-13 13/34 156/9703
6                        Atrial Fibrillation 1.590688e-13 13/34 160/9703
62                  Congenital retrognathism 2.170801e-02  1/34   1/9703
91              Female Pseudohermaphroditism 2.170801e-02  1/34   1/9703
98               Toxic effect of heavy metal 2.170801e-02  1/34   1/9703
118 Epidermolysis bullosa simplex, Ogna type 2.170801e-02  1/34   1/9703
144             Glucose Metabolism Disorders 2.170801e-02  1/34   1/9703
158       Glucocorticoid Receptor Deficiency 2.170801e-02  1/34   1/9703</code></pre>
<pre><code>******************************************

*                                        *

*          Welcome to WebGestaltR !      *

*                                        *

******************************************</code></pre>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-14-3.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ tidyr   1.1.4     ✔ dplyr   1.0.7
✔ readr   2.1.1     ✔ stringr 1.4.0
✔ purrr   0.3.4     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ tidyr::extract() masks disgenet2r::extract()
✖ dplyr::filter()  masks stats::filter()
✖ dplyr::lag()     masks stats::lag()</code></pre>
<pre><code>Warning: ggrepel: 27 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-15-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-15-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-plots---1_84" class="section level2">
<h2>Locus Plots - 1_84</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-16-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-16-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-plots---2_105" class="section level2">
<h2>Locus Plots - 2_105</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-17-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-17-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-plots---10_44" class="section level2">
<h2>Locus Plots - 10_44</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-18-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/41069190a9ee6622855efc159d8c38d2ba0e54f1/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-18-1.png" target="_blank">4106919</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-plots---2_46" class="section level2">
<h2>Locus Plots - 2_46</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-19-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-plots---5_103" class="section level2">
<h2>Locus Plots - 5_103</h2>
<p><img src="figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/b79a1eb5d6a640cec966b2db2b19a26afc70185d/docs/figure/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd/unnamed-chunk-20-1.png" target="_blank">b79a1eb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7       purrr_0.3.4      
 [5] readr_2.1.1       tidyr_1.1.4       tidyverse_1.3.1   tibble_3.1.6     
 [9] WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0       cowplot_1.0.0    
[13] ggplot2_3.3.5     workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       doParallel_1.0.16
 [5] httr_1.4.2        rprojroot_2.0.2   tools_3.6.1       backports_1.4.1  
 [9] doRNG_1.8.2       utf8_1.2.2        R6_2.5.1          vipor_0.4.5      
[13] DBI_1.1.1         colorspace_2.0-2  withr_2.4.3       ggrastr_1.0.1    
[17] tidyselect_1.1.1  bit_4.0.4         curl_4.3.2        compiler_3.6.1   
[21] git2r_0.26.1      cli_3.1.0         rvest_1.0.2       Cairo_1.5-12.2   
[25] xml2_1.3.3        labeling_0.4.2    scales_1.1.1      apcluster_1.4.8  
[29] digest_0.6.29     rmarkdown_2.11    svglite_1.2.2     pkgconfig_2.0.3  
[33] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[37] rlang_0.4.12      readxl_1.3.1      rstudioapi_0.13   RSQLite_2.2.8    
[41] jquerylib_0.1.4   farver_2.1.0      generics_0.1.1    jsonlite_1.7.2   
[45] vroom_1.5.7       magrittr_2.0.1    Matrix_1.2-18     ggbeeswarm_0.6.0 
[49] Rcpp_1.0.7        munsell_0.5.0     fansi_0.5.0       gdtools_0.1.9    
[53] lifecycle_1.0.1   stringi_1.7.6     whisker_0.3-2     yaml_2.2.1       
[57] plyr_1.8.6        grid_3.6.1        blob_1.2.2        ggrepel_0.9.1    
[61] parallel_3.6.1    promises_1.0.1    crayon_1.4.2      lattice_0.20-38  
[65] haven_2.4.3       hms_1.1.1         knitr_1.36        pillar_1.6.4     
[69] igraph_1.2.10     rjson_0.2.20      rngtools_1.5.2    reshape2_1.4.4   
[73] codetools_0.2-16  reprex_2.0.1      glue_1.5.1        evaluate_0.14    
[77] data.table_1.14.2 modelr_0.1.8      vctrs_0.3.8       tzdb_0.2.0       
[81] httpuv_1.5.1      foreach_1.5.1     cellranger_1.1.0  gtable_0.3.0     
[85] assertthat_0.2.1  cachem_1.0.6      xfun_0.29         broom_0.7.10     
[89] later_0.8.0       iterators_1.0.13  beeswarm_0.2.3    memoise_2.0.1    
[93] ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
