<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />

<meta name="date" content="2021-12-18" />

<title>Atrial fibrillation - Heart Atrial Appendage</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Atrial fibrillation - Heart Atrial Appendage</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-12-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree762fd7f08c8fdf8e702d763cace4f0cde9daad1dtargetblank762fd7fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/762fd7f08c8fdf8e702d763cace4f0cde9daad1d" target="_blank">762fd7f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree762fd7f08c8fdf8e702d763cace4f0cde9daad1dtargetblank762fd7fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/762fd7f08c8fdf8e702d763cace4f0cde9daad1d" target="_blank">762fd7f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  code/ctwas_config.R
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  data/AF/
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/summary_known_genes_annotations.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd</code>) and HTML (<code>docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/d35d1217f13a1398163dd89b57321be499bea132/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">d35d121</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
<td>
update analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/1f785cffd36f1af589f47f1a7373a30191b0dd6b/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">1f785cf</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/7784da678a5acc5342afe166f04fc825ea538346/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">7784da6</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/82c68fdd84120d702cc7b5233cbad9411174dfe9/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">82c68fd</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/7e05bbc394287384daeb7802a9ccc009a246ad6e/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">7e05bbc</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/9cf0e70ad824b1554fe1f2598c64b5fcd4f11146/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">9cf0e70</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/fbe71b7701a9a151008b0395bbbf55faccd8ef33/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">fbe71b7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/c22d2e96e20bd8d6296adbbea206b20ca6e6e9f6/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">c22d2e9</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/6b46de7ec1a695719b465266eedd482afc78cd71/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/374327b9f3cd04a21f02954ba5afd43d06db1a2b/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">374327b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e22d0c906642196ab22d358cad154fab6f31a91d/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">e22d0c9</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/a8dbae0bab7e7b0c861cbac1b67e3a297f8fc5ad/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">a8dbae0</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/37c4593e91dfad4804698e9a368a80e146e1bedc/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">37c4593</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
<td>
publish files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e080e7ba33161d30413d5ba3032e65f38318b6b2/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">e080e7b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/0b4a88679cbec1c7cd24ff781349c804e77646bb/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">0b4a886</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
<td>
Start my new project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/9b1bb6e6cdaebd6e1e1f12fb74a2d0502b9ea07b/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">9b1bb6e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/773dcb5bf8c0a4fe714e6d391de27b8193563da4/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">773dcb5</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/3052d49e1cacb26e4aa48a8de07f1dcc650c51d3/docs/Atrial_Fibrillation_Heart_Atrial_Appendage.html" target="_blank">3052d49</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/12069cf8a6562261faa1763ca38cc7949257f7f4/analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd" target="_blank">12069cf</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
<td>
Publish the files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>qclist_all &lt;- list()

qc_files &lt;- paste0(results_dir, &quot;/&quot;, list.files(results_dir, pattern=&quot;exprqc.Rd&quot;))

for (i in 1:length(qc_files)){
  load(qc_files[i])
  chr &lt;- unlist(strsplit(rev(unlist(strsplit(qc_files[i], &quot;_&quot;)))[1], &quot;[.]&quot;))[1]
  qclist_all[[chr]] &lt;- cbind(do.call(rbind, lapply(qclist,unlist)), as.numeric(substring(chr,4)))
}

qclist_all &lt;- data.frame(do.call(rbind, qclist_all))
colnames(qclist_all)[ncol(qclist_all)] &lt;- &quot;chr&quot;

rm(qclist, wgtlist, z_gene_chr)

#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 10862</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1103  776  625  427  501  624  518  377  399  423  648  611  203  368  356  495 
  17   18   19   20   21   22 
 676  177  840  318  126  271 </code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.6548518</code></pre>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<pre class="r"><code>library(ggplot2)
library(cowplot)</code></pre>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<pre class="r"><code>load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s2.susieIrssres.Rd&quot;))
  
df &lt;- data.frame(niter = rep(1:ncol(group_prior_rec), 2),
                 value = c(group_prior_rec[1,], group_prior_rec[2,]),
                 group = rep(c(&quot;Gene&quot;, &quot;SNP&quot;), each = ncol(group_prior_rec)))
df$group &lt;- as.factor(df$group)

df$value[df$group==&quot;SNP&quot;] &lt;- df$value[df$group==&quot;SNP&quot;]*thin #adjust parameter to account for thin argument

p_pi &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(pi)) +
  ggtitle(&quot;Prior mean&quot;) +
  theme_cowplot()

df &lt;- data.frame(niter = rep(1:ncol(group_prior_var_rec), 2),
                 value = c(group_prior_var_rec[1,], group_prior_var_rec[2,]),
                 group = rep(c(&quot;Gene&quot;, &quot;SNP&quot;), each = ncol(group_prior_var_rec)))
df$group &lt;- as.factor(df$group)
p_sigma2 &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(sigma^2)) +
  ggtitle(&quot;Prior variance&quot;) +
  theme_cowplot()

plot_grid(p_pi, p_sigma2)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-4-1.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6b46de7ec1a695719b465266eedd482afc78cd71/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-4-1.png" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a8dbae0bab7e7b0c861cbac1b67e3a297f8fc5ad/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-4-1.png" target="_blank">a8dbae0</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e080e7ba33161d30413d5ba3032e65f38318b6b2/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-4-1.png" target="_blank">e080e7b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3052d49e1cacb26e4aa48a8de07f1dcc650c51d3/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-4-1.png" target="_blank">3052d49</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>        gene          snp 
0.0208487453 0.0001927855 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code>    gene      snp 
9.646456 8.679052 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 1030836</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   10862 6839050</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>      gene        snp 
0.00211918 0.01110076 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.01225055 0.12523867</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<pre class="r"><code>#distribution of PIPs
hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main=&quot;Distribution of Gene PIPs&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-5-1.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6b46de7ec1a695719b465266eedd482afc78cd71/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-5-1.png" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a8dbae0bab7e7b0c861cbac1b67e3a297f8fc5ad/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-5-1.png" target="_blank">a8dbae0</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e080e7ba33161d30413d5ba3032e65f38318b6b2/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-5-1.png" target="_blank">e080e7b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3052d49e1cacb26e4aa48a8de07f1dcc650c51d3/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-5-1.png" target="_blank">3052d49</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#genes with PIP&gt;0.8 or 20 highest PIPs
head(ctwas_gene_res[order(-ctwas_gene_res$susie_pip),report_cols], max(sum(ctwas_gene_res$susie_pip&gt;0.8), 20))</code></pre>
<pre><code>          genename region_tag susie_pip        mu2          PVE         z
2293          CAV1       7_70 1.0000000  622.84815 6.042165e-04 15.567870
3275         PRRX1       1_84 0.9998808  119.34663 1.157627e-04 14.667578
4009           DEK       6_14 0.9925129   63.47784 6.111795e-05 -9.000000
3527         CCND2       12_4 0.9906174   27.01637 2.596231e-05 -5.283784
1310           PXN      12_75 0.9826498   29.47170 2.809405e-05 -5.328302
12836 RP11-325L7.2       5_82 0.9826020 1993.73789 1.900449e-03 12.356322
9257         AGAP5      10_49 0.9779202   48.91516 4.640420e-05 11.518590
3523         KLF12      13_36 0.9770829   26.00017 2.464439e-05 -5.072464
6621         AKAP6       14_8 0.9721549   76.81846 7.244551e-05 -9.197368
6914          JAM2       21_9 0.9640543   22.19279 2.075505e-05  4.563232
9639         DLEU1      13_21 0.9586468   23.59091 2.193885e-05  4.697095
11818         DPF3      14_34 0.9574312   33.35512 3.097993e-05  6.264960
10521       FAM43A      3_120 0.9569176   29.69898 2.756935e-05 -5.487179
2444       SEC23IP      10_74 0.9517809   22.38862 2.067163e-05 -4.565228
13075    LINC01629      14_36 0.9471613   31.88580 2.929757e-05 -5.695652
2138           AES       19_4 0.9403219   20.20438 1.843030e-05  4.182804
4658        POPDC3       6_70 0.9252511   25.19514 2.261449e-05 -4.758170
10290       NKX2-5      5_103 0.9228745   61.55975 5.511247e-05 -9.391892
7515       TNFSF13       17_7 0.9135188   34.26969 3.036953e-05 -5.883117
5185          GYPC       2_74 0.9068050   38.95864 3.427111e-05 -6.380531
8248         CMTM5       14_3 0.9067300   30.09803 2.647442e-05 -5.472727
10548       SCN10A       3_28 0.8867624   77.72022 6.685774e-05 -8.814286
13967 RP5-890E16.5      17_28 0.8660551   23.11200 1.941751e-05 -4.761194
712          SP100      2_135 0.8658723   18.73540 1.573719e-05 -3.671335
9012         MTSS1       8_82 0.8594108   20.87861 1.740655e-05  4.402634
8992          MURC       9_50 0.8478869   23.34736 1.920376e-05  4.911964
5223         PSMB7       9_64 0.8430652   25.30057 2.069197e-05 -4.820896
6114       STK11IP      2_130 0.8406381   18.96825 1.546845e-05 -3.868022
10416          PGP       16_2 0.8298586   28.51218 2.295329e-05  5.943820
9691           BOK      2_144 0.8255975   19.18259 1.536335e-05  3.910125
8420          MARS      12_36 0.8180336   17.81958 1.414097e-05 -3.366197
3088          GNB4      3_110 0.8097696   30.34628 2.383842e-05 -5.583333
      num_eqtl
2293         3
3275         2
4009         1
3527         1
1310         1
12836        1
9257         2
3523         1
6621         1
6914         2
9639         1
11818        3
10521        1
2444         2
13075        1
2138         3
4658         1
10290        1
7515         1
5185         1
8248         1
10548        1
13967        1
712          2
9012         2
8992         2
5223         1
6114         2
10416        1
9691         3
8420         1
3088         1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<pre class="r"><code>#plot PIP vs effect size
plot(ctwas_gene_res$susie_pip, ctwas_gene_res$mu2, xlab=&quot;PIP&quot;, ylab=&quot;mu^2&quot;, main=&quot;Gene PIPs vs Effect Size&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-6-1.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6b46de7ec1a695719b465266eedd482afc78cd71/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-6-1.png" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a8dbae0bab7e7b0c861cbac1b67e3a297f8fc5ad/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-6-1.png" target="_blank">a8dbae0</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e080e7ba33161d30413d5ba3032e65f38318b6b2/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-6-1.png" target="_blank">e080e7b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3052d49e1cacb26e4aa48a8de07f1dcc650c51d3/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-6-1.png" target="_blank">3052d49</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#genes with 20 largest effect sizes
head(ctwas_gene_res[order(-ctwas_gene_res$mu2),report_cols],20)</code></pre>
<pre><code>          genename region_tag    susie_pip       mu2          PVE           z
3250         WIPF1      2_105 0.000000e+00 2852.2491 0.000000e+00   8.4415584
12836 RP11-325L7.2       5_82 9.826020e-01 1993.7379 1.900449e-03  12.3563218
1487         SIRT1      10_44 3.337242e-01 1299.2406 4.206179e-04  -5.0538462
7413          IL6R       1_75 5.862422e-12 1273.1807 7.240649e-15  -4.9784946
6444         HERC4      10_44 1.013394e-01 1220.9395 1.200281e-04  -5.3595598
11106         NACA      12_35 5.127731e-03 1179.0569 5.865032e-06  -6.2408092
960          BAZ2A      12_35 2.047295e-03 1162.5562 2.308898e-06  -5.9428571
7986       SLC35A1       6_59 0.000000e+00  804.8763 0.000000e+00  -5.0724638
2507          WNT3      17_27 3.707228e-05  797.4584 2.867925e-08  -4.3894024
2292          CAV2       7_70 0.000000e+00  689.1121 0.000000e+00  14.5349429
2293          CAV1       7_70 1.000000e+00  622.8482 6.042165e-04  15.5678701
4999          ORC3       6_59 0.000000e+00  597.6005 0.000000e+00   4.1666667
10616       ARL17A      17_27 7.713881e-06  488.6853 3.656896e-09  -0.5399498
3687         KDM3B       5_82 5.275983e-06  436.2575 2.232835e-09  -6.6287277
7728        GPR155      2_105 0.000000e+00  371.9331 0.000000e+00  -1.4546586
7276      ARHGAP27      17_27 5.875984e-06  189.4420 1.079860e-09   0.1576482
10739         MAPT      17_27 2.791224e-04  166.5127 4.508711e-08   3.8724621
7729          PMVK       1_76 9.693335e-05  161.1464 1.515319e-08 -12.1029412
7730        PBXIP1       1_76 9.335873e-05  156.1414 1.414111e-08 -11.8676471
341         FAM13B       5_82 1.826415e-01  153.9825 2.728233e-05   7.6009242
      num_eqtl
3250         1
12836        1
1487         1
7413         1
6444         2
11106        2
960          1
7986         1
2507         2
2292         2
2293         3
4999         1
10616        2
3687         2
7728         2
7276         2
10739        2
7729         1
7730         1
341          2</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre class="r"><code>#genes with 20 highest pve
head(ctwas_gene_res[order(-ctwas_gene_res$PVE),report_cols],20)</code></pre>
<pre><code>          genename region_tag susie_pip        mu2          PVE          z
12836 RP11-325L7.2       5_82 0.9826020 1993.73789 1.900449e-03  12.356322
2293          CAV1       7_70 1.0000000  622.84815 6.042165e-04  15.567870
1487         SIRT1      10_44 0.3337242 1299.24064 4.206179e-04  -5.053846
6444         HERC4      10_44 0.1013394 1220.93950 1.200281e-04  -5.359560
3275         PRRX1       1_84 0.9998808  119.34663 1.157627e-04  14.667578
6621         AKAP6       14_8 0.9721549   76.81846 7.244551e-05  -9.197368
10548       SCN10A       3_28 0.8867624   77.72022 6.685774e-05  -8.814286
12013       ZSWIM8      10_49 0.7522384   87.25963 6.367652e-05 -11.216495
4009           DEK       6_14 0.9925129   63.47784 6.111795e-05  -9.000000
8298       SYNPO2L      10_49 0.6265676   99.72467 6.061512e-05 -11.945652
10290       NKX2-5      5_103 0.9228745   61.55975 5.511247e-05  -9.391892
9257         AGAP5      10_49 0.9779202   48.91516 4.640420e-05  11.518590
5185          GYPC       2_74 0.9068050   38.95864 3.427111e-05  -6.380531
3665         KCNJ5      11_80 0.5037532   67.73277 3.309993e-05  -8.748092
11818         DPF3      14_34 0.9574312   33.35512 3.097993e-05   6.264960
10515      C5orf47      5_104 0.7661297   41.61108 3.092586e-05   6.680556
7515       TNFSF13       17_7 0.9135188   34.26969 3.036953e-05  -5.883117
13075    LINC01629      14_36 0.9471613   31.88580 2.929757e-05  -5.695652
1310           PXN      12_75 0.9826498   29.47170 2.809405e-05  -5.328302
10521       FAM43A      3_120 0.9569176   29.69898 2.756935e-05  -5.487179
      num_eqtl
12836        1
2293         3
1487         1
6444         2
3275         2
6621         1
10548        1
12013        1
4009         1
8298         1
10290        1
9257         2
5185         1
3665         1
11818        3
10515        1
7515         1
13075        1
1310         1
10521        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre class="r"><code>#genes with 20 largest z scores
head(ctwas_gene_res[order(-abs(ctwas_gene_res$z)),report_cols],20)</code></pre>
<pre><code>          genename region_tag    susie_pip        mu2          PVE          z
2293          CAV1       7_70 1.000000e+00  622.84815 6.042165e-04  15.567870
3275         PRRX1       1_84 9.998808e-01  119.34663 1.157627e-04  14.667578
2292          CAV2       7_70 0.000000e+00  689.11215 0.000000e+00  14.534943
12836 RP11-325L7.2       5_82 9.826020e-01 1993.73789 1.900449e-03  12.356322
7729          PMVK       1_76 9.693335e-05  161.14638 1.515319e-08 -12.102941
8298       SYNPO2L      10_49 6.265676e-01   99.72467 6.061512e-05 -11.945652
7730        PBXIP1       1_76 9.335873e-05  156.14141 1.414111e-08 -11.867647
9805         MYOZ1      10_49 3.497772e-03   88.98387 3.019349e-07  11.759348
9257         AGAP5      10_49 9.779202e-01   48.91516 4.640420e-05  11.518590
12013       ZSWIM8      10_49 7.522384e-01   87.25963 6.367652e-05 -11.216495
6395        C9orf3       9_48 4.919856e-02   94.74228 4.521751e-06  10.671642
7407        ZBTB7B       1_76 9.636563e-05  122.09475 1.141378e-08  10.638889
13556   RP1-79C4.4       1_84 8.254209e-03   58.95140 4.720414e-07   9.586404
10290       NKX2-5      5_103 9.228745e-01   61.55975 5.511247e-05  -9.391892
9719        SEC24C      10_49 9.597200e-04   46.26092 4.306945e-08   9.271945
6621         AKAP6       14_8 9.721549e-01   76.81846 7.244551e-05  -9.197368
4009           DEK       6_14 9.925129e-01   63.47784 6.111795e-05  -9.000000
10548       SCN10A       3_28 8.867624e-01   77.72022 6.685774e-05  -8.814286
3665         KCNJ5      11_80 5.037532e-01   67.73277 3.309993e-05  -8.748092
7733         DCST2       1_76 2.613168e-04   89.76014 2.275418e-08  -8.718310
      num_eqtl
2293         3
3275         2
2292         2
12836        1
7729         1
8298         1
7730         1
9805         2
9257         2
12013        1
6395         1
7407         1
13556        2
10290        1
9719         2
6621         1
4009         1
10548        1
3665         1
7733         1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<pre class="r"><code>#set nominal signifiance threshold for z scores
alpha &lt;- 0.05

#bonferroni adjusted threshold for z scores
sig_thresh &lt;- qnorm(1-(alpha/nrow(ctwas_gene_res)/2), lower=T)

#Q-Q plot for z scores
obs_z &lt;- ctwas_gene_res$z[order(ctwas_gene_res$z)]
exp_z &lt;- qnorm((1:nrow(ctwas_gene_res))/nrow(ctwas_gene_res))

plot(exp_z, obs_z, xlab=&quot;Expected z&quot;, ylab=&quot;Observed z&quot;, main=&quot;Gene z score Q-Q plot&quot;)
abline(a=0,b=1)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-1.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6b46de7ec1a695719b465266eedd482afc78cd71/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-1.png" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a8dbae0bab7e7b0c861cbac1b67e3a297f8fc5ad/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-1.png" target="_blank">a8dbae0</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e080e7ba33161d30413d5ba3032e65f38318b6b2/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-1.png" target="_blank">e080e7b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3052d49e1cacb26e4aa48a8de07f1dcc650c51d3/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-1.png" target="_blank">3052d49</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#plot z score vs PIP
plot(abs(ctwas_gene_res$z), ctwas_gene_res$susie_pip, xlab=&quot;abs(z)&quot;, ylab=&quot;PIP&quot;)
abline(v=sig_thresh, col=&quot;red&quot;, lty=2)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-2.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6b46de7ec1a695719b465266eedd482afc78cd71/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-2.png" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a8dbae0bab7e7b0c861cbac1b67e3a297f8fc5ad/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-2.png" target="_blank">a8dbae0</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e080e7ba33161d30413d5ba3032e65f38318b6b2/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-2.png" target="_blank">e080e7b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3052d49e1cacb26e4aa48a8de07f1dcc650c51d3/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-9-2.png" target="_blank">3052d49</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#proportion of significant z scores
mean(abs(ctwas_gene_res$z) &gt; sig_thresh)</code></pre>
<pre><code>[1] 0.008654023</code></pre>
<pre class="r"><code>#genes with most significant z scores
head(ctwas_gene_res[order(-abs(ctwas_gene_res$z)),report_cols],20)</code></pre>
<pre><code>          genename region_tag    susie_pip        mu2          PVE          z
2293          CAV1       7_70 1.000000e+00  622.84815 6.042165e-04  15.567870
3275         PRRX1       1_84 9.998808e-01  119.34663 1.157627e-04  14.667578
2292          CAV2       7_70 0.000000e+00  689.11215 0.000000e+00  14.534943
12836 RP11-325L7.2       5_82 9.826020e-01 1993.73789 1.900449e-03  12.356322
7729          PMVK       1_76 9.693335e-05  161.14638 1.515319e-08 -12.102941
8298       SYNPO2L      10_49 6.265676e-01   99.72467 6.061512e-05 -11.945652
7730        PBXIP1       1_76 9.335873e-05  156.14141 1.414111e-08 -11.867647
9805         MYOZ1      10_49 3.497772e-03   88.98387 3.019349e-07  11.759348
9257         AGAP5      10_49 9.779202e-01   48.91516 4.640420e-05  11.518590
12013       ZSWIM8      10_49 7.522384e-01   87.25963 6.367652e-05 -11.216495
6395        C9orf3       9_48 4.919856e-02   94.74228 4.521751e-06  10.671642
7407        ZBTB7B       1_76 9.636563e-05  122.09475 1.141378e-08  10.638889
13556   RP1-79C4.4       1_84 8.254209e-03   58.95140 4.720414e-07   9.586404
10290       NKX2-5      5_103 9.228745e-01   61.55975 5.511247e-05  -9.391892
9719        SEC24C      10_49 9.597200e-04   46.26092 4.306945e-08   9.271945
6621         AKAP6       14_8 9.721549e-01   76.81846 7.244551e-05  -9.197368
4009           DEK       6_14 9.925129e-01   63.47784 6.111795e-05  -9.000000
10548       SCN10A       3_28 8.867624e-01   77.72022 6.685774e-05  -8.814286
3665         KCNJ5      11_80 5.037532e-01   67.73277 3.309993e-05  -8.748092
7733         DCST2       1_76 2.613168e-04   89.76014 2.275418e-08  -8.718310
      num_eqtl
2293         3
3275         2
2292         2
12836        1
7729         1
8298         1
7730         1
9805         2
9257         2
12013        1
6395         1
7407         1
13556        2
10290        1
9719         2
6621         1
4009         1
10548        1
3665         1
7733         1</code></pre>
</div>
<div id="locus-plots-for-genes-and-snps" class="section level2">
<h2>Locus plots for genes and SNPs</h2>
<pre class="r"><code>ctwas_gene_res_sortz &lt;- ctwas_gene_res[order(-abs(ctwas_gene_res$z)),]
report_cols_region &lt;- report_cols[!(report_cols %in% c(&quot;num_eqtl&quot;))]

n_plots &lt;- 5
for (region_tag_plot in head(unique(ctwas_gene_res_sortz$region_tag), n_plots)){
  ctwas_res_region &lt;-  ctwas_res[ctwas_res$region_tag==region_tag_plot,]
  start &lt;- min(ctwas_res_region$pos)
  end &lt;- max(ctwas_res_region$pos)
  
  ctwas_res_region &lt;- ctwas_res_region[order(ctwas_res_region$pos),]
  ctwas_res_region_gene &lt;- ctwas_res_region[ctwas_res_region$type==&quot;gene&quot;,]
  ctwas_res_region_snp &lt;- ctwas_res_region[ctwas_res_region$type==&quot;SNP&quot;,]
  
  #region name
  print(paste0(&quot;Region: &quot;, region_tag_plot))
  
  #table of genes in region
  print(ctwas_res_region_gene[,report_cols_region])
  
  par(mfrow=c(4,1))
  
  #gene z scores
  plot(ctwas_res_region_gene$pos, abs(ctwas_res_region_gene$z), xlab=&quot;Position&quot;, ylab=&quot;abs(gene_z)&quot;, xlim=c(start,end),
   ylim=c(0,max(sig_thresh, abs(ctwas_res_region_gene$z))),
   main=paste0(&quot;Region: &quot;, region_tag_plot))
  abline(h=sig_thresh,col=&quot;red&quot;,lty=2)
  
  #significance threshold for SNPs
  alpha_snp &lt;- 5*10^(-8)
  sig_thresh_snp &lt;- qnorm(1-alpha_snp/2, lower=T)
  
  #snp z scores
  plot(ctwas_res_region_snp$pos, abs(ctwas_res_region_snp$z), xlab=&quot;Position&quot;, ylab=&quot;abs(snp_z)&quot;,xlim=c(start,end),
   ylim=c(0,max(sig_thresh_snp, max(abs(ctwas_res_region_snp$z)))))
  abline(h=sig_thresh_snp,col=&quot;purple&quot;,lty=2)
  
  #gene pips
  plot(ctwas_res_region_gene$pos, ctwas_res_region_gene$susie_pip, xlab=&quot;Position&quot;, ylab=&quot;Gene PIP&quot;, xlim=c(start,end), ylim=c(0,1))
  abline(h=0.8,col=&quot;blue&quot;,lty=2)
  
  #snp pips
  plot(ctwas_res_region_snp$pos, ctwas_res_region_snp$susie_pip, xlab=&quot;Position&quot;, ylab=&quot;SNP PIP&quot;, xlim=c(start,end), ylim=c(0,1))
  abline(h=0.8,col=&quot;blue&quot;,lty=2)
}</code></pre>
<pre><code>[1] &quot;Region: 7_70&quot;
       genename region_tag susie_pip        mu2          PVE          z
9316      LRRN3       7_70         0   4.798139 0.0000000000 -0.2554113
10486    IMMP2L       7_70         0   9.726840 0.0000000000 -1.1251408
10096    LSMEM1       7_70         0   4.653263 0.0000000000  0.1730349
6325    TMEM168       7_70         0  13.129092 0.0000000000  1.4473363
8015      GPR85       7_70         0  59.638319 0.0000000000  3.6640335
8014       BMT2       7_70         0  14.144825 0.0000000000  1.5304348
12439    HRAT17       7_70         0  10.070641 0.0000000000 -1.1617647
11985 LINC00998       7_70         0   9.138407 0.0000000000 -1.0595238
6892    PPP1R3A       7_70         0  11.619381 0.0000000000 -1.3141517
4991      MDFIC       7_70         0   9.711018 0.0000000000  1.1940299
4990        TES       7_70         0   5.365475 0.0000000000  0.2279955
2292       CAV2       7_70         0 689.112146 0.0000000000 14.5349429
2293       CAV1       7_70         1 622.848152 0.0006042165 15.5678701
11541    CAPZA2       7_70         0   4.817249 0.0000000000 -0.5522388</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-1.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6b46de7ec1a695719b465266eedd482afc78cd71/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-1.png" target="_blank">6b46de7</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a8dbae0bab7e7b0c861cbac1b67e3a297f8fc5ad/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-1.png" target="_blank">a8dbae0</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e080e7ba33161d30413d5ba3032e65f38318b6b2/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-1.png" target="_blank">e080e7b</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3052d49e1cacb26e4aa48a8de07f1dcc650c51d3/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-1.png" target="_blank">3052d49</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Region: 1_84&quot;
        genename region_tag   susie_pip        mu2          PVE          z
3275       PRRX1       1_84 0.999880773 119.346630 1.157627e-04 14.6675780
13556 RP1-79C4.4       1_84 0.008254209  58.951404 4.720414e-07  9.5864041
132         FMO3       1_84 0.007235866   6.979886 4.899472e-08  0.7058824
1450        FMO2       1_84 0.005819057   4.561901 2.575187e-08 -0.1257130
964         FMO4       1_84 0.006067492   5.455974 3.211382e-08  0.8619618
364         MYOC       1_84 0.005790338   4.650929 2.612486e-08  0.2537313
3427       VAMP4       1_84 0.024556213  14.423816 3.435991e-07 -1.8006622
11344       DNM3       1_84 0.028756639  18.635294 5.198581e-07  1.9131944
5075        PIGC       1_84 0.009407288   9.222146 8.416022e-08 -1.1709099
10093   C1orf105       1_84 0.006032874   4.956744 2.900889e-08 -0.5632184
1451        SUCO       1_84 0.010489256   9.463884 9.629961e-08  0.9829060</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-2.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Region: 5_82&quot;
          genename region_tag    susie_pip         mu2          PVE           z
1029        PKD2L2       5_82 1.074133e-06   44.303563 4.616439e-11 -3.01646091
12836 RP11-325L7.2       5_82 9.826020e-01 1993.737895 1.900449e-03 12.35632184
341         FAM13B       5_82 1.826415e-01  153.982522 2.728233e-05  7.60092424
2972          NME5       5_82 8.035979e-08    7.216916 5.626015e-13 -0.78899083
3687         KDM3B       5_82 5.275983e-06  436.257478 2.232835e-09 -6.62872770
4678         REEP2       5_82 3.273136e-07   34.258854 1.087795e-11 -2.33668342
3688          EGR1       5_82 1.582507e-07   28.468689 4.370424e-12 -0.46666667
2975         HSPA9       5_82 1.191309e-07   15.195222 1.756070e-12 -1.47659574
425         CTNNA1       5_82 8.323910e-08   55.376623 4.471614e-12 -0.68493151
6256        LRRTM2       5_82 8.644342e-08   53.238157 4.464423e-12 -0.57534247
8964       SLC23A1       5_82 7.435045e-08    9.459617 6.822878e-13 -0.45945946
12284        PROB1       5_82 7.443824e-08    7.340816 5.300915e-13 -0.37662338
8961       SPATA24       5_82 7.971111e-08   12.023543 9.297404e-13  0.74647887
8959       DNAJC18       5_82 7.430306e-08    6.843115 4.932544e-13  0.07692308
10457      TMEM173       5_82 1.052563e-07    9.743512 9.948880e-13  0.52710454</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-3">
Past versions of unnamed-chunk-10-3.png
</button>
</p>
<div id="fig-unnamed-chunk-10-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-3.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Region: 1_76&quot;
      genename region_tag    susie_pip        mu2          PVE           z
7729      PMVK       1_76 9.693335e-05 161.146376 1.515319e-08 -12.1029412
7730    PBXIP1       1_76 9.335873e-05 156.141409 1.414111e-08 -11.8676471
7409      SHC1       1_76 1.029606e-04  12.706941 1.269178e-09  -0.5945946
7407    ZBTB7B       1_76 9.636563e-05 122.094754 1.141378e-08  10.6388889
6001    ADAM15       1_76 1.054076e-04  19.807533 2.025408e-09   3.0000000
7733     DCST2       1_76 2.613168e-04  89.760135 2.275418e-08  -8.7183099
6011     EFNA3       1_76 1.089596e-04  11.240179 1.188089e-09  -2.2724138
8787     EFNA1       1_76 1.606785e-04  11.329113 1.765892e-09  -1.8067354
8786   SLC50A1       1_76 1.003646e-04   6.652241 6.476776e-10   0.8410502
10571     MUC1       1_76 2.518113e-04  20.984333 5.126026e-09  -0.9716981
9323      MTX1       1_76 1.791435e-04  16.586200 2.882427e-09   0.5555556
9782       GBA       1_76 1.026910e-04   7.311893 7.284044e-10  -0.5299145
3310    SCAMP3       1_76 2.912632e-04  23.792281 6.722520e-09  -3.3289474
7735    YY1AP1       1_76 2.425271e-04  16.618716 3.909923e-09  -1.9970399
4695      DAP3       1_76 2.664936e-04  29.575646 7.645948e-09  -3.5890411
3313     GON4L       1_76 2.496430e-03  41.586291 1.007117e-07  -3.9452055
4703     SYT11       1_76 2.496430e-03  41.586291 1.007117e-07   3.9452055
6015      RIT1       1_76 2.304738e-04  20.358641 4.551775e-09   2.5915493
4696  KIAA0907       1_76 1.475526e-04   7.222145 1.033769e-09   0.5050776
3314   ARHGEF2       1_76 2.545536e-03  38.509404 9.509474e-08   3.1323529
7755      SSR2       1_76 1.009922e-04  10.918250 1.069673e-09   1.8503401
3315   LAMTOR2       1_76 2.403669e-04  11.662362 2.719390e-09   0.7234637
11033   SEMA4A       1_76 3.065448e-04  12.239053 3.639588e-09  -0.4358974
7419  SLC25A44       1_76 1.133154e-04   6.396866 7.031801e-10   0.9130609
12648    BGLAP       1_76 1.086835e-04  12.509241 1.318879e-09   2.3348238
7418     PAQR6       1_76 1.717449e-04  10.265336 1.710281e-09   0.2561399
7754    TMEM79       1_76 9.629572e-05   9.278082 8.667136e-10   2.2625000
11561     SMG5       1_76 9.629572e-05   9.278082 8.667136e-10  -2.2625000</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-4">
Past versions of unnamed-chunk-10-4.png
</button>
</p>
<div id="fig-unnamed-chunk-10-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-4.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;Region: 10_49&quot;
            genename region_tag    susie_pip       mu2          PVE           z
2448           PALD1      10_49 6.930061e-05 10.464471 7.035011e-10   1.2023810
5427        ADAMTS14      10_49 4.041175e-05  5.577877 2.186689e-10  -0.5051546
8279           SGPL1      10_49 4.383695e-05  6.314746 2.685386e-10   0.6594096
8282           PCBD1      10_49 3.637634e-05  4.624703 1.631974e-10  -0.1515152
2449           UNC5B      10_49 4.079184e-05  5.662476 2.240733e-10   0.5252525
2450           CDH23      10_49 5.372325e-05  8.157613 4.251437e-10  -0.9400479
2451            VSIR      10_49 4.235290e-05  5.884211 2.417585e-10  -0.4533295
12014      C10orf105      10_49 3.693800e-05  4.763447 1.706889e-10   0.2383178
11301           PSAP      10_49 4.143656e-05  5.328503 2.141901e-10   0.6926829
3840           CHST3      10_49 3.920994e-05  6.023333 2.291097e-10  -1.2205882
2452          SPOCK2      10_49 3.803373e-05  5.050485 1.863427e-10   0.5000000
8296         ANAPC16      10_49 7.630441e-05 11.415357 8.449861e-10   1.5096021
5424           ASCC1      10_49 3.691548e-05  5.519486 1.976595e-10  -0.3622163
8632           DDIT4      10_49 4.114673e-05  6.155464 2.457008e-10   0.4788732
6453         DNAJB12      10_49 3.871607e-05  7.240038 2.719209e-10  -1.2362964
7004             MCU      10_49 3.659951e-05  5.173114 1.836698e-10   0.6527778
5426            OIT3      10_49 4.946253e-05  8.617426 4.134894e-10  -0.6470588
13645  RP11-344N10.5      10_49 3.675546e-05  4.856516 1.731638e-10   0.8848495
8299          NUDT13      10_49 4.246247e-05 16.333612 6.728184e-10  -4.1574570
10190         MRPS16      10_49 8.192817e-05 15.875112 1.261713e-09   2.4854195
11923         DNAJC9      10_49 1.447809e-04 21.802775 3.062200e-09  -2.9593023
7006          CFAP70      10_49 4.129670e-05 16.906306 6.772897e-10  -4.3161765
5422           ANXA7      10_49 4.129111e-05 16.129161 6.460689e-10  -4.2137405
13668  RP11-464F9.22      10_49 5.378631e-05 19.460835 1.015415e-09   4.8622754
9805           MYOZ1      10_49 3.497772e-03 88.983867 3.019349e-07  11.7593478
9257           AGAP5      10_49 9.779202e-01 48.915157 4.640420e-05  11.5185898
8298         SYNPO2L      10_49 6.265676e-01 99.724672 6.061512e-05 -11.9456522
13589 RP11-574K11.29      10_49 5.552979e-05 15.390089 8.290441e-10   1.2828283
9719          SEC24C      10_49 9.597200e-04 46.260923 4.306945e-08   9.2719446
11176          FUT11      10_49 5.375832e-04 21.866281 1.140331e-08  -6.5522388
12013         ZSWIM8      10_49 7.522384e-01 87.259631 6.367652e-05 -11.2164948
3838            PLAU      10_49 2.066160e-04 15.107410 3.028060e-09   3.5310286
10503          AP3M1      10_49 3.334117e-04 23.844396 7.712187e-09  -2.0202020
7013             ADK      10_49 3.465243e-04 24.183065 8.129342e-09   2.0408163
1049          DUSP13      10_49 5.692456e-05  8.684736 4.795862e-10  -1.7916667
8167          COMTD1      10_49 1.542145e-04 10.638192 1.591489e-09   2.1111111
13481 RP11-399K21.11      10_49 2.039823e-04 18.379908 3.637025e-09   0.6792247
6445        C10orf11      10_49 4.685159e-05  6.813717 3.096841e-10  -1.0581482
13654 RP11-399K21.14      10_49 3.636678e-05  4.614057 1.627790e-10  -0.1324675</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-5">
Past versions of unnamed-chunk-10-5.png
</button>
</p>
<div id="fig-unnamed-chunk-10-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-10-5.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="snps-with-highest-pips" class="section level2">
<h2>SNPs with highest PIPs</h2>
<pre class="r"><code>#snps with PIP&gt;0.8 or 20 highest PIPs
head(ctwas_snp_res[order(-ctwas_snp_res$susie_pip),report_cols_snps],
max(sum(ctwas_snp_res$susie_pip&gt;0.8), 20))</code></pre>
<pre><code>                id region_tag susie_pip        mu2          PVE           z
28439   rs12404927       1_75 1.0000000 1437.87698 1.394865e-03  -1.5517241
28440    rs7536152       1_75 1.0000000 1473.01903 1.428956e-03  -6.4626866
28533   rs34515871       1_76 1.0000000  299.28422 2.903316e-04  17.0000000
32410   rs12142529       1_83 1.0000000  145.61337 1.412576e-04  12.0396040
32435  rs112797273       1_83 1.0000000  101.67077 9.862944e-05   9.4903846
183533   rs1906615       4_72 1.0000000 1756.73093 1.704181e-03  45.1604938
183538   rs7440714       4_72 1.0000000  992.10110 9.624238e-04  -9.2183406
272612 rs111990232       6_59 1.0000000 1710.90898 1.659730e-03   0.5392670
414968  rs76106073      10_44 1.0000000 1273.72173 1.235620e-03   0.6572238
550561 rs140798119      15_35 1.0000000  285.26104 2.767279e-04   3.9950739
550564  rs74022964      15_35 1.0000000  342.92230 3.326643e-04  12.5777778
681435   rs4972703      2_105 1.0000000 4299.78452 4.171163e-03  -0.7842670
739005   rs9770220       7_70 1.0000000  835.76918 8.107683e-04  -5.2814815
788780  rs61937778      12_35 1.0000000 1668.54978 1.618637e-03  -1.3056995
28537    rs2878412       1_76 1.0000000  105.57502 1.024169e-04   5.4250000
586354  rs62056842      17_27 1.0000000 1130.21900 1.096410e-03   1.1269036
28535   rs11576820       1_76 1.0000000  133.71597 1.297160e-04   8.0243902
422177  rs60469668      10_66 1.0000000  172.10824 1.669599e-04  16.9647059
110779   rs9852222        3_9 1.0000000   83.84308 8.133503e-05   8.3823529
339255    rs373983       8_21 0.9999998   67.64329 6.561982e-05   8.5588235
298378  rs12112152       7_15 0.9999995   45.13406 4.378392e-05   6.9358974
422175   rs7094488      10_66 0.9999992  110.01432 1.067233e-04  14.1060606
28540   rs10908445       1_76 0.9999991  193.56990 1.877794e-04 -13.1470588
570907   rs6499606      16_39 0.9999990  103.94144 1.008321e-04  12.6376812
826958   rs8005417       14_9 0.9999978  427.56249 4.147716e-04   0.9863946
852634 rs140185678       16_2 0.9999910   45.86064 4.448839e-05   7.6100917
97513    rs1975584      2_118 0.9999847   71.76048 6.961280e-05   1.2311828
580602  rs72811292      17_11 0.9999789   40.04445 3.884576e-05  -6.5233645
279525  rs11756438       6_79 0.9999723   47.69440 4.626641e-05  -8.2089552
117531 rs116202356       3_27 0.9999305   36.10604 3.502355e-05   6.0813559
176218   rs1458038       4_54 0.9998791   34.99562 3.394467e-05   6.0277778
121102  rs12330500       3_40 0.9997290  519.11865 5.034535e-04   0.5641026
396844   rs1886296       9_73 0.9995279   34.54923 3.349992e-05   5.9863014
39581    rs6427989      1_102 0.9993189   36.39724 3.528442e-05  -4.4142857
653873    rs464901       22_4 0.9992684   43.84562 4.250292e-05  -7.0555556
708945 rs574293775       5_82 0.9987918 1853.69252 1.796069e-03   6.2411765
39582   rs12353975      1_102 0.9982647   35.87479 3.474126e-05  -4.3000000
184120   rs7700110       4_73 0.9979235   45.19125 4.374839e-05   7.0666667
232199 rs199992924       5_68 0.9969497 1340.40151 1.296339e-03  -0.6309524
655632    rs133902       22_7 0.9963501   31.05883 3.001978e-05   6.1617647
232019    rs338623       5_68 0.9953592   67.12738 6.481716e-05  -7.9855072
414975  rs12360521      10_44 0.9953399 1283.59990 1.239400e-03   5.3740458
696421   rs7374540       3_28 0.9939021   45.86079 4.421764e-05   4.7794118
570915    rs876727      16_39 0.9926586   48.82838 4.702000e-05 -10.0000000
275680   rs9496567       6_67 0.9918710   26.51232 2.551017e-05   5.1125000
586570  rs75230966      17_27 0.9917637   49.43748 4.756362e-05   6.0582524
183553   rs4631108       4_72 0.9915393  306.03072 2.943645e-04 -17.6376812
40258    rs4951023      1_104 0.9906227   31.13453 2.991996e-05   5.5970149
238444   rs6894302       5_84 0.9877503   34.35180 3.291601e-05  -7.2933333
666039   rs2885697       1_25 0.9874950   44.28194 4.242013e-05  -6.2714286
570920  rs60602157      16_39 0.9854801   38.72262 3.701885e-05  10.4415584
339226    rs403894       8_19 0.9793925   56.27052 5.346236e-05   6.8529412
650809   rs7282237      21_16 0.9782270   25.00237 2.372636e-05  -4.7605634
529120   rs2738413      14_29 0.9782089  112.52732 1.067825e-04 -11.6119403
570912   rs4788691      16_39 0.9747904   40.13308 3.795108e-05  -1.6900000
329588  rs35760656       7_94 0.9746629   35.32804 3.340291e-05  -6.3536585
470281  rs17380837      12_18 0.9737999   44.42786 4.196967e-05  -6.9583333
366399   rs7460121       8_88 0.9725824   24.19453 2.282727e-05   4.7500000
2727      rs284278        1_7 0.9713320   35.00958 3.298869e-05  -6.0857143
826964   rs8011559       14_9 0.9693907  435.54854 4.095867e-04   5.6913580
283467    rs958747       6_89 0.9688123   24.37662 2.290992e-05  -4.9710145
556130  rs12898337      15_48 0.9643924   35.44339 3.315885e-05  -6.0579710
279482  rs77435894       6_78 0.9642414   28.80796 2.694690e-05   6.1111111
549931    rs745636      15_33 0.9571831   25.61925 2.378876e-05  -5.0740741
25783    rs4839174       1_69 0.9569309   33.97079 3.153528e-05   5.9459459
183505   rs1823291       4_72 0.9568146  286.06204 2.655207e-04 -19.4520548
604563  rs17794590      18_24 0.9550117   24.59348 2.278448e-05  -4.8242424
100283  rs35880620      2_125 0.9544571   39.77143 3.682460e-05  -6.2168675
58743    rs7578482       2_16 0.9542037   32.14647 2.975670e-05   5.7435897
738958   rs1633714       7_70 0.9499686  778.22440 7.171740e-04  11.7536232
97488   rs11889306      2_118 0.9487414   45.59873 4.196730e-05   6.7391304
826887  rs73241997       14_9 0.9472521  276.10117 2.537139e-04   7.8817204
681445   rs1367220      2_105 0.9467808 4303.78606 3.952852e-03   7.2207792
784225   rs2291437      12_17 0.9462388   66.11326 6.068757e-05   9.1826923
32457    rs7522387       1_83 0.9460866   59.86066 5.493926e-05   6.6375000
696586   rs7373065       3_28 0.9355464   65.79596 5.971384e-05  -8.0637450
28495     rs906280       1_75 0.9353945   59.57170 5.405616e-05   7.9111111
490521  rs12425471      12_69 0.9303264   41.49244 3.744680e-05  -5.2500000
614935   rs7256735       19_3 0.9241700   23.45925 2.103180e-05   4.4952381
279432     rs89107       6_78 0.9189049   47.36791 4.222457e-05  -9.5606061
372071   rs1594768        9_9 0.9139999   24.21822 2.147330e-05  -4.6911765
5627    rs10917072       1_15 0.9043759   33.53922 2.942472e-05   5.8235294
217602 rs114414434       5_31 0.8994692   26.16276 2.282866e-05  -4.9718310
232192   rs4073838       5_68 0.8923424 1368.92743 1.185011e-03  -5.9545455
272609    rs371814       6_59 0.8922509 1722.76723 1.491159e-03   6.1911765
252160  rs73724866       6_13 0.8855608   94.00847 8.075990e-05 -10.4141414
479761    rs776211      12_43 0.8831880   28.12821 2.409937e-05  -6.3424658
550553    rs519946      15_34 0.8819650   34.47055 2.949239e-05   5.9242424
494561   rs6560886      12_82 0.8817026   28.94415 2.475673e-05   5.6666667
28531   rs12128882       1_76 0.8779576   32.56646 2.773668e-05   1.0731707
68039     rs243080       2_40 0.8749115   24.16683 2.051135e-05  -4.5522388
467254  rs12821447      12_12 0.8716947   24.08112 2.036346e-05   4.5128205
615414  rs73919353       19_5 0.8614285   28.38347 2.371894e-05   5.4076433
453904    rs565449      11_55 0.8575434   28.38859 2.361622e-05  -5.2535211
123427   rs1091584       3_45 0.8532454   30.27502 2.505929e-05   5.5882353
244628  rs62377226      5_100 0.8422335   33.87890 2.768039e-05   5.8493151
113948  rs73032373       3_18 0.8375231   33.29814 2.705373e-05  -5.9178082
369034  rs72693377       8_94 0.8312985   24.72583 1.993968e-05   4.2762431
659021  rs11705586      22_15 0.8290134   26.32683 2.117242e-05   4.8988764
121105      rs6924       3_40 0.8277296  522.73041 4.197364e-04   4.2133333
304278 rs145593380       7_26 0.8170209   29.74581 2.357596e-05   5.3333333
495691   rs7321083       13_3 0.8149736   24.65878 1.949510e-05   4.8292683</code></pre>
</div>
<div id="snps-with-largest-effect-sizes" class="section level2">
<h2>SNPs with largest effect sizes</h2>
<pre class="r"><code>#plot PIP vs effect size
#plot(ctwas_snp_res$susie_pip, ctwas_snp_res$mu2, xlab=&quot;PIP&quot;, ylab=&quot;mu^2&quot;, main=&quot;SNP PIPs vs Effect Size&quot;)

#SNPs with 50 largest effect sizes
head(ctwas_snp_res[order(-ctwas_snp_res$mu2),report_cols_snps],50)</code></pre>
<pre><code>                id region_tag    susie_pip      mu2          PVE         z
681445   rs1367220      2_105 9.467808e-01 4303.786 3.952852e-03  7.220779
681435   rs4972703      2_105 1.000000e+00 4299.785 4.171163e-03 -0.784267
681446   rs1367219      2_105 2.069799e-01 4294.507 8.622874e-04  7.103896
681441  rs10168156      2_105 3.312313e-01 4292.416 1.379252e-03  7.233766
681439   rs6713018      2_105 9.687858e-02 4291.538 4.033213e-04  7.168831
681430   rs1864453      2_105 1.876497e-01 4289.961 7.809293e-04  7.207792
681422   rs2033315      2_105 1.145049e-01 4287.502 4.762543e-04  7.064935
681433   rs6707162      2_105 3.496928e-02 4279.920 1.451887e-04  7.090909
681434   rs6735680      2_105 2.455791e-02 4279.461 1.019509e-04  7.077922
681423   rs2033314      2_105 2.432678e-02 4278.387 1.009660e-04  7.155844
681425  rs28485554      2_105 8.747105e-03 4277.770 3.629879e-05  7.000000
681456  rs10803884      2_105 7.083440e-03 4217.525 2.898093e-05  7.285714
681464   rs6738901      2_105 7.743316e-03 4201.159 3.155778e-05 -7.272727
681449  rs12466643      2_105 3.135672e-04 4196.140 1.276412e-06  7.272727
681480  rs35368253      2_105 1.998443e-04 4178.444 8.100592e-07 -7.256410
681466  rs10197521      2_105 1.763290e-11 4171.648 7.135785e-14 -6.547619
681468  rs34661753      2_105 3.194389e-08 4119.395 1.276532e-10 -7.076923
681419   rs7590328      2_105 2.272449e-11 4103.428 9.045890e-14  6.784810
681417   rs6433497      2_105 0.000000e+00 3465.041 0.000000e+00  6.345679
681418   rs2115874      2_105 0.000000e+00 3462.201 0.000000e+00  6.419753
681410   rs1430185      2_105 0.000000e+00 3387.075 0.000000e+00  6.185185
681397   rs1991601      2_105 0.000000e+00 3363.058 0.000000e+00  5.950617
681395   rs6759870      2_105 0.000000e+00 2978.209 0.000000e+00  5.987342
681486  rs35215597      2_105 0.000000e+00 2793.497 0.000000e+00 -8.441558
681506  rs56181519      2_105 0.000000e+00 2703.083 0.000000e+00 -8.597403
681308  rs13032076      2_105 0.000000e+00 2402.122 0.000000e+00 -5.250000
681408  rs13024657      2_105 0.000000e+00 2366.705 0.000000e+00 -5.258065
681509   rs2358891      2_105 0.000000e+00 2265.703 0.000000e+00 -7.500000
681526  rs35808589      2_105 0.000000e+00 2030.164 0.000000e+00 -5.229885
681558   rs1376875      2_105 0.000000e+00 2019.770 0.000000e+00 -5.000000
681541  rs35444726      2_105 0.000000e+00 2016.551 0.000000e+00 -5.172414
681550  rs13420492      2_105 0.000000e+00 1989.902 0.000000e+00 -4.908046
681552  rs71417497      2_105 0.000000e+00 1980.143 0.000000e+00 -4.772727
708976   rs2040862       5_82 1.932759e-01 1961.536 3.677770e-04 12.459770
708860  rs17171711       5_82 5.061870e-02 1958.969 9.619419e-05 12.482759
708884  rs13355516       5_82 4.398123e-03 1956.728 8.348495e-06 12.379310
708835   rs9327807       5_82 2.294241e-03 1953.970 4.348779e-06 12.367816
708869  rs77915370       5_82 3.425950e-04 1952.684 6.489682e-07 12.356322
708740  rs78081438       5_82 6.691465e-04 1951.053 1.266487e-06 12.390805
708691  rs73300168       5_82 3.565820e-04 1950.786 6.748068e-07 12.298851
708817  rs11959181       5_82 9.541911e-05 1949.634 1.804674e-07 12.241379
708671  rs10076361       5_82 2.676249e-04 1949.238 5.060598e-07 12.298851
708836 rs113331339       5_82 8.891833e-05 1949.202 1.681352e-07 12.241379
708789  rs73299268       5_82 8.169862e-05 1949.085 1.544742e-07 12.241379
708892 rs148378888       5_82 1.992837e-04 1948.655 3.767186e-07 12.344828
708714  rs73299210       5_82 5.888915e-05 1946.246 1.111843e-07 12.287356
708717  rs73299219       5_82 6.204555e-05 1945.825 1.171183e-07 12.287356
708737  rs13362264       5_82 2.545517e-05 1945.354 4.803802e-08 12.264368
681335   rs2303891      2_105 0.000000e+00 1921.675 0.000000e+00 -4.185185
708899  rs10479176       5_82 6.731608e-01 1866.560 1.218908e-03  6.931818</code></pre>
</div>
<div id="snps-with-highest-pve" class="section level2">
<h2>SNPs with highest PVE</h2>
<pre class="r"><code>#SNPs with 50 highest pve
head(ctwas_snp_res[order(-ctwas_snp_res$PVE),report_cols_snps],50)</code></pre>
<pre><code>                id region_tag  susie_pip       mu2          PVE           z
681435   rs4972703      2_105 1.00000000 4299.7845 0.0041711625  -0.7842670
681445   rs1367220      2_105 0.94678083 4303.7861 0.0039528520   7.2207792
708945 rs574293775       5_82 0.99879179 1853.6925 0.0017960693   6.2411765
183533   rs1906615       4_72 1.00000000 1756.7309 0.0017041808  45.1604938
272612 rs111990232       6_59 1.00000000 1710.9090 0.0016597296   0.5392670
788780  rs61937778      12_35 1.00000000 1668.5498 0.0016186375  -1.3056995
272609    rs371814       6_59 0.89225093 1722.7672 0.0014911593   6.1911765
28440    rs7536152       1_75 1.00000000 1473.0190 0.0014289557  -6.4626866
28439   rs12404927       1_75 1.00000000 1437.8770 0.0013948649  -1.5517241
681441  rs10168156      2_105 0.33123129 4292.4155 0.0013792517   7.2337662
232199 rs199992924       5_68 0.99694972 1340.4015 0.0012963390  -0.6309524
414975  rs12360521      10_44 0.99533986 1283.5999 0.0012394000   5.3740458
414968  rs76106073      10_44 1.00000000 1273.7217 0.0012356201   0.6572238
708899  rs10479176       5_82 0.67316075 1866.5597 0.0012189085   6.9318182
232192   rs4073838       5_68 0.89234244 1368.9274 0.0011850110  -5.9545455
586354  rs62056842      17_27 1.00000000 1130.2190 0.0010964101   1.1269036
788720   rs2860482      12_35 0.64493512 1697.2111 0.0010618479  -7.1052632
788775   rs7313074      12_35 0.58220672 1705.1065 0.0009630285  -6.9868421
183538   rs7440714       4_72 1.00000000  992.1011 0.0009624238  -9.2183406
681446   rs1367219      2_105 0.20697994 4294.5073 0.0008622874   7.1038961
739005   rs9770220       7_70 1.00000000  835.7692 0.0008107683  -5.2814815
681430   rs1864453      2_105 0.18764974 4289.9609 0.0007809293   7.2077922
738958   rs1633714       7_70 0.94996864  778.2244 0.0007171740  11.7536232
788778   rs4759256      12_35 0.41979796 1704.6254 0.0006941921  -6.9736842
272613    rs384318       6_59 0.37976257 1721.2083 0.0006340975   6.1029412
738995  rs10255816       7_70 0.78914390  709.0714 0.0005428210  -9.7123288
121102  rs12330500       3_40 0.99972902  519.1187 0.0005034535   0.5641026
681422   rs2033315      2_105 0.11450492 4287.5023 0.0004762543   7.0649351
272614   rs1145714       6_59 0.26478699 1717.6032 0.0004411943   6.1029412
121105      rs6924       3_40 0.82772961  522.7304 0.0004197364   4.2133333
826958   rs8005417       14_9 0.99999777  427.5625 0.0004147716   0.9863946
826964   rs8011559       14_9 0.96939072  435.5485 0.0004095867   5.6913580
681439   rs6713018      2_105 0.09687858 4291.5379 0.0004033213   7.1688312
232191   rs4235764       5_68 0.29136909 1367.2078 0.0003864457  -5.8787879
708976   rs2040862       5_82 0.19327595 1961.5362 0.0003677770  12.4597701
788715   rs7978685      12_35 0.21731446 1694.8625 0.0003573004  -7.0657895
550564  rs74022964      15_35 1.00000000  342.9223 0.0003326643  12.5777778
232194   rs4235768       5_68 0.24121476 1367.3620 0.0003199616  -5.8030303
272571   rs9444476       6_59 0.19008883 1698.6999 0.0003132447  -6.3088235
183553   rs4631108       4_72 0.99153932  306.0307 0.0002943645 -17.6376812
28533   rs34515871       1_76 1.00000000  299.2842 0.0002903316  17.0000000
272596   rs9444488       6_59 0.16974071 1703.6881 0.0002805347  -6.2647059
550561 rs140798119      15_35 1.00000000  285.2610 0.0002767279   3.9950739
183505   rs1823291       4_72 0.95681461  286.0620 0.0002655207 -19.4520548
826887  rs73241997       14_9 0.94725212  276.1012 0.0002537139   7.8817204
708944 rs191830974       5_82 0.12101272 1865.7774 0.0002190288   6.9206349
121104   rs1554125       3_40 0.38957721  520.5125 0.0001967139   4.1333333
28540   rs10908445       1_76 0.99999905  193.5699 0.0001877794 -13.1470588
272583   rs9444481       6_59 0.10849613 1701.3375 0.0001790668  -6.2500000
422177  rs60469668      10_66 1.00000000  172.1082 0.0001669599  16.9647059</code></pre>
</div>
<div id="snps-with-largest-z-scores" class="section level2">
<h2>SNPs with largest z scores</h2>
<pre class="r"><code>#histogram of (abs) SNP z scores
hist(abs(ctwas_snp_res$z))</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-14-1.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#SNPs with 50 largest z scores
head(ctwas_snp_res[order(-abs(ctwas_snp_res$z)),report_cols_snps],50)</code></pre>
<pre><code>                id region_tag    susie_pip       mu2          PVE         z
183533   rs1906615       4_72 1.000000e+00 1756.7309 1.704181e-03  45.16049
183534  rs75725917       4_72 0.000000e+00  588.7458 0.000000e+00  42.33663
183532   rs1906611       4_72 0.000000e+00  526.9126 0.000000e+00  40.97143
183531  rs28521134       4_72 0.000000e+00  512.6248 0.000000e+00  40.22772
183527  rs10019689       4_72 0.000000e+00  508.5039 0.000000e+00  40.11000
183528  rs76013973       4_72 0.000000e+00  502.3963 0.000000e+00  40.04902
183530  rs12639820       4_72 0.000000e+00  499.7947 0.000000e+00  40.03922
183526  rs12647393       4_72 0.000000e+00  500.8094 0.000000e+00  39.98020
183529  rs74496596       4_72 0.000000e+00  484.7331 0.000000e+00  39.75728
183525  rs12647316       4_72 0.000000e+00  476.7021 0.000000e+00  39.55882
183524   rs4529121       4_72 0.000000e+00  480.5177 0.000000e+00  39.55340
183518  rs12650829       4_72 0.000000e+00  450.0533 0.000000e+00  31.54444
183541   rs3866831       4_72 0.000000e+00  919.4634 0.000000e+00 -30.10000
183540   rs6533530       4_72 0.000000e+00  904.8879 0.000000e+00 -29.91429
183523  rs12644107       4_72 0.000000e+00  153.0712 0.000000e+00  25.98851
183521   rs2723318       4_72 0.000000e+00  365.6985 0.000000e+00 -25.45833
183516   rs2218698       4_72 0.000000e+00  337.5178 0.000000e+00 -25.09589
183519   rs2197814       4_72 0.000000e+00  339.8970 0.000000e+00  24.59722
183517   rs1448799       4_72 0.000000e+00  339.2379 0.000000e+00  24.56944
183515 rs112927894       4_72 0.000000e+00  320.7713 0.000000e+00  24.50685
183505   rs1823291       4_72 9.568146e-01  286.0620 2.655207e-04 -19.45205
183509   rs2723296       4_72 4.318269e-02  278.0630 1.164832e-05 -19.02740
183514  rs11724067       4_72 0.000000e+00  321.1887 0.000000e+00 -18.94059
183508   rs2044674       4_72 2.641186e-06  253.4848 6.494734e-10  18.58108
183553   rs4631108       4_72 9.915393e-01  306.0307 2.943645e-04 -17.63768
183552   rs1906613       4_72 8.460684e-03  295.1509 2.422479e-06  17.34783
28533   rs34515871       1_76 1.000000e+00  299.2842 2.903316e-04  17.00000
422177  rs60469668      10_66 1.000000e+00  172.1082 1.669599e-04  16.96471
183513  rs13111704       4_72 0.000000e+00  393.8288 0.000000e+00 -16.91743
28536   rs12058931       1_76 1.643627e-04  221.4339 3.530676e-08  16.54054
183537   rs4124159       4_72 0.000000e+00 1159.1777 0.000000e+00 -16.14414
183535   rs1906606       4_72 0.000000e+00 1154.4323 0.000000e+00 -16.02703
183542   rs4032974       4_72 0.000000e+00 1146.0129 0.000000e+00 -15.86726
739346  rs11773845       7_70 0.000000e+00  555.7324 0.000000e+00  15.73134
183562  rs17513625       4_72 0.000000e+00  176.0291 0.000000e+00  15.71130
183539  rs10006881       4_72 0.000000e+00 1136.3170 0.000000e+00 -15.68750
739366   rs1997571       7_70 0.000000e+00  553.9040 0.000000e+00  15.68657
739329   rs3807989       7_70 0.000000e+00  551.5677 0.000000e+00  15.62687
422173  rs12572965      10_66 2.273800e-04  132.7631 2.928466e-08  15.27059
422172  rs56965730      10_66 2.300323e-04  131.7918 2.940950e-08  15.21176
739367   rs1997572       7_70 0.000000e+00  525.1335 0.000000e+00  14.50685
739223   rs2270188       7_70 0.000000e+00  711.6787 0.000000e+00  14.43939
739224   rs2270189       7_70 0.000000e+00  711.6066 0.000000e+00  14.43939
739272   rs2109514       7_70 0.000000e+00  687.7806 0.000000e+00  14.42424
739276  rs55883210       7_70 0.000000e+00  687.5295 0.000000e+00  14.42424
739239  rs10271007       7_70 0.000000e+00  707.4203 0.000000e+00  14.39394
739256   rs6466579       7_70 0.000000e+00  703.1231 0.000000e+00  14.33333
739267   rs7795510       7_70 0.000000e+00  696.7170 0.000000e+00  14.22727
183507  rs12642151       4_72 2.282285e-12  237.6299 5.261159e-16  14.14286
422175   rs7094488      10_66 9.999992e-01  110.0143 1.067233e-04  14.10606</code></pre>
</div>
<div id="gene-set-enrichment-for-genes-with-pip0.8" class="section level2">
<h2>Gene set enrichment for genes with PIP&gt;0.8</h2>
<pre class="r"><code>#GO enrichment analysis
library(enrichR)</code></pre>
<pre><code>Welcome to enrichR
Checking connection ... </code></pre>
<pre><code>Enrichr ... Connection is Live!
FlyEnrichr ... Connection is available!
WormEnrichr ... Connection is available!
YeastEnrichr ... Connection is available!
FishEnrichr ... Connection is available!</code></pre>
<pre class="r"><code>dbs &lt;- c(&quot;GO_Biological_Process_2021&quot;, &quot;GO_Cellular_Component_2021&quot;, &quot;GO_Molecular_Function_2021&quot;)
genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.8]

#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 32</code></pre>
<pre class="r"><code>if (length(genes)&gt;0){
  GO_enrichment &lt;- enrichr(genes, dbs)

  for (db in dbs){
    print(db)
    df &lt;- GO_enrichment[[db]]
    print(plotEnrich(GO_enrichment[[db]]))
    df &lt;- df[df$Adjusted.P.value&lt;0.05,c(&quot;Term&quot;, &quot;Overlap&quot;, &quot;Adjusted.P.value&quot;, &quot;Genes&quot;)]
    print(df)
  }
  
  #DisGeNET enrichment
  
  # devtools::install_bitbucket(&quot;ibi_group/disgenet2r&quot;)
  library(disgenet2r)
  
  disgenet_api_key &lt;- get_disgenet_api_key(
                    email = &quot;wesleycrouse@gmail.com&quot;, 
                    password = &quot;uchicago1&quot; )
  
  Sys.setenv(DISGENET_API_KEY= disgenet_api_key)
  
  res_enrich &lt;-disease_enrichment(entities=genes, vocabulary = &quot;HGNC&quot;,
                               database = &quot;CURATED&quot; )
  
  df &lt;- res_enrich@qresult[1:10, c(&quot;Description&quot;, &quot;FDR&quot;, &quot;Ratio&quot;,  &quot;BgRatio&quot;)]
  print(df)
  
  #WebGestalt enrichment
  library(WebGestaltR)
  
  background &lt;- ctwas_gene_res$genename
  
  #listGeneSet()
  databases &lt;- c(&quot;pathway_KEGG&quot;, &quot;disease_GLAD4U&quot;, &quot;disease_OMIM&quot;)
  
  enrichResult &lt;- WebGestaltR(enrichMethod=&quot;ORA&quot;, organism=&quot;hsapiens&quot;,
                              interestGene=genes, referenceGene=background,
                              enrichDatabase=databases, interestGeneType=&quot;genesymbol&quot;,
                              referenceGeneType=&quot;genesymbol&quot;, isOutput=F)
  print(enrichResult[,c(&quot;description&quot;, &quot;size&quot;, &quot;overlap&quot;, &quot;FDR&quot;, &quot;database&quot;, &quot;userId&quot;)])
}</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-15-1.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-15-2.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre><code>BOK gene(s) from the input list not found in DisGeNET CURATEDDLEU1 gene(s) from the input list not found in DisGeNET CURATEDRP5-890E16.5 gene(s) from the input list not found in DisGeNET CURATEDJAM2 gene(s) from the input list not found in DisGeNET CURATEDSEC23IP gene(s) from the input list not found in DisGeNET CURATEDCMTM5 gene(s) from the input list not found in DisGeNET CURATEDMURC gene(s) from the input list not found in DisGeNET CURATEDAGAP5 gene(s) from the input list not found in DisGeNET CURATEDMARS gene(s) from the input list not found in DisGeNET CURATEDPOPDC3 gene(s) from the input list not found in DisGeNET CURATEDSP100 gene(s) from the input list not found in DisGeNET CURATEDAES gene(s) from the input list not found in DisGeNET CURATEDSTK11IP gene(s) from the input list not found in DisGeNET CURATEDRP11-325L7.2 gene(s) from the input list not found in DisGeNET CURATEDLINC01629 gene(s) from the input list not found in DisGeNET CURATED</code></pre>
<pre><code>                                                                   Description
5                                                          Atrial Fibrillation
85                                              Paroxysmal atrial fibrillation
156                                             Persistent atrial fibrillation
171                                               familial atrial fibrillation
37                                                                Cardiomegaly
130                                                        Cardiac Hypertrophy
58                                                    Congenital retrognathism
157                      HYPOTHYROIDISM, CONGENITAL, NONGOITROUS, 5 (disorder)
158                              Lipodystrophy, Congenital Generalized, Type 3
167 ATRIAL SEPTAL DEFECT 7 WITH OR WITHOUT ATRIOVENTRICULAR CONDUCTION DEFECTS
             FDR Ratio  BgRatio
5   7.915974e-11  9/17 160/9703
85  7.915974e-11  9/17 156/9703
156 7.915974e-11  9/17 156/9703
171 7.915974e-11  9/17 156/9703
37  1.217805e-02  3/17  82/9703
130 1.217805e-02  3/17  82/9703
58  1.467481e-02  1/17   1/9703
157 1.467481e-02  1/17   1/9703
158 1.467481e-02  1/17   1/9703
167 1.467481e-02  1/17   1/9703</code></pre>
<pre><code>******************************************

*                                        *

*          Welcome to WebGestaltR !      *

*                                        *

******************************************</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-3">
Past versions of unnamed-chunk-15-3.png
</button>
</p>
<div id="fig-unnamed-chunk-15-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/77de9fb222c2ed3d38a025cd3fdab8f49d2f1073/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-15-3.png" target="_blank">77de9fb</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...
                 description size overlap         FDR       database
1            Isaacs Syndrome   55       5 0.001525439 disease_GLAD4U
2        Atrial Fibrillation   52       4 0.022003754 disease_GLAD4U
3 Atrioventricular block NOS   22       3 0.031360061 disease_GLAD4U
                          userId
1 SCN10A;GNB4;NKX2-5;CCND2;KLF12
2       PRRX1;SCN10A;NKX2-5;CAV1
3           SCN10A;NKX2-5;POPDC3</code></pre>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>library(&quot;readxl&quot;)

known_annotations &lt;- read_xlsx(&quot;data/summary_known_genes_annotations.xlsx&quot;, sheet=&quot;LDL&quot;)</code></pre>
<pre><code>New names:
* `` -&gt; ...4
* `` -&gt; ...5</code></pre>
<pre class="r"><code>known_annotations &lt;- unique(known_annotations$`Gene Symbol`)

unrelated_genes &lt;- ctwas_gene_res$genename[!(ctwas_gene_res$genename %in% known_annotations)]

#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 69</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 44</code></pre>
<pre class="r"><code>#assign ctwas, TWAS, and bystander genes
ctwas_genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res$genename[abs(ctwas_gene_res$z)&gt;sig_thresh]
novel_genes &lt;- ctwas_genes[!(ctwas_genes %in% twas_genes)]

#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.582104</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 32</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 94</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>     genename region_tag susie_pip      mu2          PVE         z num_eqtl
6114  STK11IP      2_130 0.8406381 18.96825 1.546845e-05 -3.868022        2
712     SP100      2_135 0.8658723 18.73540 1.573719e-05 -3.671335        2
9691      BOK      2_144 0.8255975 19.18259 1.536335e-05  3.910125        3
9012    MTSS1       8_82 0.8594108 20.87861 1.740655e-05  4.402634        2
2444  SEC23IP      10_74 0.9517809 22.38862 2.067163e-05 -4.565228        2
8420     MARS      12_36 0.8180336 17.81958 1.414097e-05 -3.366197        1
2138      AES       19_4 0.9403219 20.20438 1.843030e-05  4.182804        3
6914     JAM2       21_9 0.9640543 22.19279 2.075505e-05  4.563232        2</code></pre>
<pre class="r"><code>#sensitivity / recall
sensitivity &lt;- rep(NA,2)
names(sensitivity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
sensitivity[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
sensitivity</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#specificity
specificity &lt;- rep(NA,2)
names(specificity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
specificity[&quot;ctwas&quot;] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity[&quot;TWAS&quot;] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
specificity</code></pre>
<pre><code>    ctwas      TWAS 
0.9970420 0.9913108 </code></pre>
<pre class="r"><code>#precision / PPV
precision &lt;- rep(NA,2)
names(precision) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
precision[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(twas_genes)
precision</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#ROC curves

pip_range &lt;- (0:1000)/1000
sensitivity &lt;- rep(NA, length(pip_range))
specificity &lt;- rep(NA, length(pip_range))

for (index in 1:length(pip_range)){
  pip &lt;- pip_range[index]
  ctwas_genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;=pip]
  sensitivity[index] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
}

plot(1-specificity, sensitivity, type=&quot;l&quot;, xlim=c(0,1), ylim=c(0,1))

sig_thresh_range &lt;- seq(from=0, to=max(abs(ctwas_gene_res$z)), length.out=length(pip_range))

for (index in 1:length(sig_thresh_range)){
  sig_thresh_plot &lt;- sig_thresh_range[index]
  twas_genes &lt;- ctwas_gene_res$genename[abs(ctwas_gene_res$z)&gt;=sig_thresh_plot]
  sensitivity[index] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
}

lines(1-specificity, sensitivity, xlim=c(0,1), ylim=c(0,1), col=&quot;red&quot;, lty=2)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/9cf0e70ad824b1554fe1f2598c64b5fcd4f11146/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-16-1.png" target="_blank">9cf0e70</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes---bystanders-only" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes - bystanders only</h2>
<p>This section first uses imputed silver standard genes to identify bystander genes within 1Mb. The bystander gene list is then subset to only genes with imputed expression in this analysis. Then, the ctwas and TWAS gene lists from this analysis are subset to only genes that are in the (subset) silver standard and bystander genes. These gene lists are then used to compute sensitivity, specificity and precision for ctwas and TWAS.</p>
<pre class="r"><code># library(biomaRt)
# library(GenomicRanges)
# 
# ensembl &lt;- useEnsembl(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
# G_list &lt;- getBM(filters= &quot;chromosome_name&quot;, attributes= c(&quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;end_position&quot;,&quot;gene_biotype&quot;), values=1:22, mart=ensembl)
# G_list &lt;- G_list[G_list$hgnc_symbol!=&quot;&quot;,]
# G_list &lt;- G_list[G_list$gene_biotype %in% c(&quot;protein_coding&quot;,&quot;lncRNA&quot;),]
# G_list$start &lt;- G_list$start_position
# G_list$end &lt;- G_list$end_position
# G_list_granges &lt;- makeGRangesFromDataFrame(G_list, keep.extra.columns=T)
# 
# #remove genes without imputed expression from gene lists
# known_annotations &lt;- known_annotations[known_annotations %in% ctwas_gene_res$genename]
# 
# known_annotations_positions &lt;- G_list[G_list$hgnc_symbol %in% known_annotations,]
# half_window &lt;- 1000000
# known_annotations_positions$start &lt;- known_annotations_positions$start_position - half_window
# known_annotations_positions$end &lt;- known_annotations_positions$end_position + half_window
# known_annotations_positions$start[known_annotations_positions$start&lt;1] &lt;- 1
# known_annotations_granges &lt;- makeGRangesFromDataFrame(known_annotations_positions, keep.extra.columns=T)
# 
# bystanders &lt;- findOverlaps(known_annotations_granges,G_list_granges)
# bystanders &lt;- unique(subjectHits(bystanders))
# bystanders &lt;- G_list$hgnc_symbol[bystanders]
# bystanders &lt;- unique(bystanders[!(bystanders %in% known_annotations)])
# unrelated_genes &lt;- bystanders
# 
# #save gene lists
# save(known_annotations, file=paste0(results_dir, &quot;/known_annotations.Rd&quot;))
# save(unrelated_genes, file=paste0(results_dir, &quot;/bystanders.Rd&quot;))


load(paste0(results_dir, &quot;/known_annotations.Rd&quot;))
load(paste0(results_dir, &quot;/bystanders.Rd&quot;))


#remove genes without imputed expression from bystander list
unrelated_genes &lt;- unrelated_genes[unrelated_genes %in% ctwas_gene_res$genename]

#number of genes in known annotations (with imputed expression)
print(length(known_annotations))</code></pre>
<pre><code>[1] 44</code></pre>
<pre class="r"><code>#number of bystander genes (with imputed expression)
print(length(unrelated_genes))</code></pre>
<pre><code>[1] 605</code></pre>
<pre class="r"><code>#subset results to genes in known annotations or bystanders
ctwas_gene_res_subset &lt;- ctwas_gene_res[ctwas_gene_res$genename %in% c(known_annotations, unrelated_genes),]

#assign ctwas and TWAS genes
ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;sig_thresh]

#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.582104</code></pre>
<pre class="r"><code>#number of ctwas genes (in known annotations or bystanders)
length(ctwas_genes)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#number of TWAS genes (in known annotations or bystanders)
length(twas_genes)</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code>#sensitivity / recall
sensitivity &lt;- rep(NA,2)
names(sensitivity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
sensitivity[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
sensitivity</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#specificity / (1 - False Positive Rate)
specificity &lt;- rep(NA,2)
names(specificity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
specificity[&quot;ctwas&quot;] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity[&quot;TWAS&quot;] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
specificity</code></pre>
<pre><code>    ctwas      TWAS 
0.9983471 0.9966942 </code></pre>
<pre class="r"><code>#precision / PPV / (1 - False Discovery Rate)
precision &lt;- rep(NA,2)
names(precision) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
precision[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(twas_genes)
precision</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#store sensitivity and specificity calculations for plots
sensitivity_plot &lt;- sensitivity
specificity_plot &lt;- specificity

#precision / PPV by PIP bin
pip_range &lt;- c(0.2, 0.4, 0.6, 0.8, 1)
precision_range &lt;- rep(NA, length(pip_range))

for (i in 1:length(pip_range)){
  pip_upper &lt;- pip_range[i]

  if (i==1){
    pip_lower &lt;- 0
  } else {
    pip_lower &lt;- pip_range[i-1]
  }
  
  #assign ctwas genes in PIP bin
  ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;=pip_lower &amp; ctwas_gene_res_subset$susie_pip&lt;pip_upper]
  
  precision_range[i] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
}

names(precision_range) &lt;- paste(c(0, pip_range[-length(pip_range)]), pip_range,sep=&quot; - &quot;)

barplot(precision_range, ylim=c(0,1), main=&quot;Precision by PIP Range&quot;, xlab=&quot;PIP Range&quot;, ylab=&quot;Precision&quot;)
abline(h=0.2, lty=2)
abline(h=0.4, lty=2)
abline(h=0.6, lty=2)
abline(h=0.8, lty=2)
barplot(precision_range, add=T, col=&quot;darkgrey&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/82c68fdd84120d702cc7b5233cbad9411174dfe9/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-1.png" target="_blank">82c68fd</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#precision / PPV by PIP threshold
#pip_range &lt;- c(0.2, 0.4, 0.6, 0.8, 1)
pip_range &lt;- c(0.5, 0.8, 1)
precision_range &lt;- rep(NA, length(pip_range))
number_detected &lt;- rep(NA, length(pip_range))

for (i in 1:length(pip_range)){
  pip_upper &lt;- pip_range[i]

  if (i==1){
    pip_lower &lt;- 0
  } else {
    pip_lower &lt;- pip_range[i-1]
  }
  
  #assign ctwas genes using PIP threshold
  ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;=pip_lower]
  
  number_detected[i] &lt;- length(ctwas_genes)
  precision_range[i] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
}

names(precision_range) &lt;- paste0(&quot;&gt;= &quot;, c(0, pip_range[-length(pip_range)]))

precision_range &lt;- precision_range*100

precision_range &lt;- c(precision_range, precision[&quot;TWAS&quot;]*100)
names(precision_range)[4] &lt;- &quot;TWAS Bonferroni&quot;
number_detected &lt;- c(number_detected, length(twas_genes))

barplot(precision_range, ylim=c(0,100), main=&quot;Precision for Distinguishing Silver Standard and Bystander Genes&quot;, xlab=&quot;PIP Threshold for Detection&quot;, ylab=&quot;% of Detected Genes in Silver Standard&quot;)
abline(h=20, lty=2)
abline(h=40, lty=2)
abline(h=60, lty=2)
abline(h=80, lty=2)
xx &lt;- barplot(precision_range, add=T, col=c(rep(&quot;darkgrey&quot;,3), &quot;white&quot;))
text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, &quot; detected&quot;), pos = 3, cex=0.8)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-2">
Past versions of unnamed-chunk-17-2.png
</button>
</p>
<div id="fig-unnamed-chunk-17-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/82c68fdd84120d702cc7b5233cbad9411174dfe9/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-2.png" target="_blank">82c68fd</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#text(x = xx, y = precision_range, label = paste0(round(precision_range,1), &quot;%&quot;), pos = 3, cex=0.8, offset = 1.5)

#false discovery rate by PIP threshold

barplot(100-precision_range, ylim=c(0,100), main=&quot;False Discovery Rate for Distinguishing Silver Standard and Bystander Genes&quot;, xlab=&quot;PIP Threshold for Detection&quot;, ylab=&quot;% Bystanders in Detected Genes&quot;)
abline(h=20, lty=2)
abline(h=40, lty=2)
abline(h=60, lty=2)
abline(h=80, lty=2)
xx &lt;- barplot(100-precision_range, add=T, col=c(rep(&quot;darkgrey&quot;,3), &quot;white&quot;))
text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, &quot; detected&quot;), pos = 3, cex=0.8)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-3">
Past versions of unnamed-chunk-17-3.png
</button>
</p>
<div id="fig-unnamed-chunk-17-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/82c68fdd84120d702cc7b5233cbad9411174dfe9/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-3.png" target="_blank">82c68fd</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#text(x = xx, y = precision_range, label = paste0(round(precision_range,1), &quot;%&quot;), pos = 3, cex=0.8, offset = 1.5)

#ROC curves

pip_range &lt;- (0:1000)/1000
sensitivity &lt;- rep(NA, length(pip_range))
specificity &lt;- rep(NA, length(pip_range))

for (index in 1:length(pip_range)){
  pip &lt;- pip_range[index]
  ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;=pip]
  sensitivity[index] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
}

plot(1-specificity, sensitivity, type=&quot;l&quot;, xlim=c(0,1), ylim=c(0,1), main=&quot;&quot;, xlab=&quot;1 - Specificity&quot;, ylab=&quot;Sensitivity&quot;)
title(expression(&quot;ROC Curve for cTWAS (black) and TWAS (&quot; * phantom(&quot;red&quot;) * &quot;)&quot;))
title(expression(phantom(&quot;ROC Curve for cTWAS (black) and TWAS (&quot;) * &quot;red&quot; * phantom(&quot;)&quot;)), col.main=&quot;red&quot;)

sig_thresh_range &lt;- seq(from=0, to=max(abs(ctwas_gene_res_subset$z)), length.out=length(pip_range))

for (index in 1:length(sig_thresh_range)){
  sig_thresh_plot &lt;- sig_thresh_range[index]
  twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;=sig_thresh_plot]
  sensitivity[index] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
}

lines(1-specificity, sensitivity, xlim=c(0,1), ylim=c(0,1), col=&quot;red&quot;, lty=1)

abline(a=0,b=1,lty=3)

#add previously computed points from the analysis
ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;sig_thresh]

points(1-specificity_plot[&quot;ctwas&quot;], sensitivity_plot[&quot;ctwas&quot;], pch=21, bg=&quot;black&quot;)
points(1-specificity_plot[&quot;TWAS&quot;], sensitivity_plot[&quot;TWAS&quot;], pch=21, bg=&quot;red&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-4">
Past versions of unnamed-chunk-17-4.png
</button>
</p>
<div id="fig-unnamed-chunk-17-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/82c68fdd84120d702cc7b5233cbad9411174dfe9/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-17-4.png" target="_blank">82c68fd</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre class="r"><code>library(tibble)
library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ tidyr   1.1.4     ✔ dplyr   1.0.7
✔ readr   2.1.1     ✔ stringr 1.4.0
✔ purrr   0.3.4     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ tidyr::extract() masks disgenet2r::extract()
✖ dplyr::filter()  masks stats::filter()
✖ dplyr::lag()     masks stats::lag()</code></pre>
<pre class="r"><code>full.gene.pip.summary &lt;- data.frame(gene_name = ctwas_gene_res$genename, 
                                    gene_pip = ctwas_gene_res$susie_pip, 
                                    gene_id = ctwas_gene_res$id, 
                                    chr = as.integer(ctwas_gene_res$chrom),
                                    start = ctwas_gene_res$pos / 1e3,
                                    is_highlight = F, stringsAsFactors = F) %&gt;% as_tibble()
full.gene.pip.summary$is_highlight &lt;- full.gene.pip.summary$gene_pip &gt; 0.80

don &lt;- full.gene.pip.summary %&gt;% 
  
  # Compute chromosome size
  group_by(chr) %&gt;% 
  summarise(chr_len=max(start)) %&gt;% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %&gt;%
  dplyr::select(-chr_len) %&gt;%
  
  # Add this info to the initial dataset
  left_join(full.gene.pip.summary, ., by=c(&quot;chr&quot;=&quot;chr&quot;)) %&gt;%
  
  # Add a cumulative position of each SNP
  arrange(chr, start) %&gt;%
  mutate( BPcum=start+tot)

axisdf &lt;- don %&gt;% group_by(chr) %&gt;% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

x_axis_labels &lt;- axisdf$chr
x_axis_labels[seq(1,21,2)] &lt;- &quot;&quot;

ggplot(don, aes(x=BPcum, y=gene_pip)) +
  
  # Show all points
  ggrastr::geom_point_rast(aes(color=as.factor(chr)), size=2) +
  scale_color_manual(values = rep(c(&quot;grey&quot;, &quot;skyblue&quot;), 22 )) +
  
  # custom X axis:
  # scale_x_continuous(label = axisdf$chr, 
  #                    breaks= axisdf$center,
  #                    guide = guide_axis(n.dodge = 2)) +
  scale_x_continuous(label = x_axis_labels,
                     breaks = axisdf$center) +
  
  scale_y_continuous(expand = c(0, 0), limits = c(0,1.25), breaks=(1:5)*0.2, minor_breaks=(1:10)*0.1) + # remove space between plot area and x axis
  
  # Add highlighted points
  ggrastr::geom_point_rast(data=subset(don, is_highlight==T), color=&quot;orange&quot;, size=2) +
  
  # Add label using ggrepel to avoid overlapping
  ggrepel::geom_label_repel(data=subset(don, is_highlight==T), 
                            aes(label=gene_name), 
                            size=4,
                            min.segment.length = 0, 
                            label.size = NA,
                            fill = alpha(c(&quot;white&quot;),0)) +
  
  # Custom the theme:
  theme_bw() +
  theme( 
    text = element_text(size = 14),
    legend.position=&quot;none&quot;,
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  ) +
  xlab(&quot;Chromosome&quot;) + 
  ylab(&quot;cTWAS PIP&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1f785cffd36f1af589f47f1a7373a30191b0dd6b/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-18-1.png" target="_blank">1f785cf</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-plots---5_45" class="section level2">
<h2>Locus Plots - 5_45</h2>
<pre class="r"><code>library(ctwas)</code></pre>
<pre><code>
Attaching package: &#39;ctwas&#39;</code></pre>
<pre><code>The following object is masked _by_ &#39;.GlobalEnv&#39;:

    z_snp</code></pre>
<pre class="r"><code>locus_plot &lt;- function(region_tag, rerun_ctwas = F, plot_eqtl = T, label=&quot;cTWAS&quot;){
  region_tag1 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[1]
  region_tag2 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[2]
  
  a &lt;- ctwas_res[ctwas_res$region_tag==region_tag,]
  
  regionlist &lt;- readRDS(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.regionlist.RDS&quot;))
  region &lt;- regionlist[[as.numeric(region_tag1)]][[region_tag2]]
  
  R_snp_info &lt;- do.call(rbind, lapply(region$regRDS, function(x){data.table::fread(paste0(tools::file_path_sans_ext(x), &quot;.Rvar&quot;))}))
  
  if (isTRUE(rerun_ctwas)){
    ld_exprfs &lt;- paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, 1:22, &quot;.expr.gz&quot;)
    temp_reg &lt;- data.frame(&quot;chr&quot; = paste0(&quot;chr&quot;,region_tag1), &quot;start&quot; = region$start, &quot;stop&quot; = region$stop)
  
    write.table(temp_reg, 
                #file= paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.temp.reg.txt&quot;) , 
                file= &quot;temp_reg.txt&quot;,
                row.names=F, col.names=T, sep=&quot;\t&quot;, quote = F)
  
    load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_snp.Rd&quot;))
  
    z_gene_temp &lt;-  z_gene[z_gene$id %in% a$id[a$type==&quot;gene&quot;],]
    z_snp_temp &lt;-  z_snp[z_snp$id %in% R_snp_info$id,]
  
    ctwas_rss(z_gene_temp, z_snp_temp, ld_exprfs, ld_pgenfs = NULL, 
              ld_R_dir = dirname(region$regRDS)[1],
              ld_regions_custom = &quot;temp_reg.txt&quot;, thin = 1, 
              outputdir = &quot;.&quot;, outname = &quot;temp&quot;, ncore = 1, ncore.rerun = 1, prob_single = 0,
              group_prior = estimated_group_prior, group_prior_var = estimated_group_prior_var,
              estimate_group_prior = F, estimate_group_prior_var = F)
            
            
    a &lt;- data.table::fread(&quot;temp.susieIrss.txt&quot;, header = T)
    
    rownames(z_snp_temp) &lt;- z_snp_temp$id
    z_snp_temp &lt;- z_snp_temp[a$id[a$type==&quot;SNP&quot;],]
    z_gene_temp &lt;- z_gene_temp[a$id[a$type==&quot;gene&quot;],]
    
    a$z &lt;- NA
    a$z[a$type==&quot;SNP&quot;] &lt;- z_snp_temp$z
    a$z[a$type==&quot;gene&quot;] &lt;- z_gene_temp$z
  }
  
  a$ifcausal &lt;- 0
  focus &lt;- a$id[a$type==&quot;gene&quot;][which.max(abs(a$z[a$type==&quot;gene&quot;]))]
  a$ifcausal &lt;- as.numeric(a$id==focus)
    
  a$PVALUE &lt;- (-log(2) - pnorm(abs(a$z), lower.tail=F, log.p=T))/log(10)
  
  R_gene &lt;- readRDS(region$R_g_file)
  R_snp_gene &lt;- readRDS(region$R_sg_file)
  R_snp &lt;- as.matrix(Matrix::bdiag(lapply(region$regRDS, readRDS)))
  
  rownames(R_gene) &lt;- region$gid
  colnames(R_gene) &lt;- region$gid
  rownames(R_snp_gene) &lt;- R_snp_info$id
  colnames(R_snp_gene) &lt;- region$gid
  rownames(R_snp) &lt;- R_snp_info$id
  colnames(R_snp) &lt;- R_snp_info$id
  
  a$r2max &lt;- NA
  
  a$r2max[a$type==&quot;gene&quot;] &lt;- R_gene[focus,a$id[a$type==&quot;gene&quot;]]
  a$r2max[a$type==&quot;SNP&quot;] &lt;- R_snp_gene[a$id[a$type==&quot;SNP&quot;],focus]
  
  r2cut &lt;- 0.4
  colorsall &lt;- c(&quot;#7fc97f&quot;, &quot;#beaed4&quot;, &quot;#fdc086&quot;)
  
  layout(matrix(1:2, ncol = 1), widths = 1, heights = c(1.5,1.5), respect = FALSE)
  par(mar = c(0, 4.1, 4.1, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 19, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;),frame.plot=FALSE, col = &quot;white&quot;, ylim= c(-0.1,1.1), ylab = &quot;cTWAS PIP&quot;, xaxt = &#39;n&#39;)
  
  grid()
  points(a$pos[a$type==&quot;SNP&quot;], a$susie_pip[a$type == &quot;SNP&quot;], pch = 21, xlab=&quot;Genomic position&quot;, bg = colorsall[1])
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;SNP&quot;  &amp; a$r2max &gt;r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$susie_pip[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  
  if (isTRUE(plot_eqtl)){
    for (cgene in a[a$type==&quot;gene&quot; &amp; a$ifcausal == 1, ]$id){
      load(paste0(results_dir, &quot;/&quot;,analysis_id, &quot;_expr_chr&quot;, region_tag1, &quot;.exprqc.Rd&quot;))
      eqtls &lt;- rownames(wgtlist[[cgene]])
      points(a[a$id %in% eqtls,]$pos, rep( -0.15, nrow(a[a$id %in% eqtls,])), pch = &quot;|&quot;, col = &quot;salmon&quot;, cex = 1.5)
    }
  }
  
  legend(min(a$pos), y= 1.1 ,c(&quot;Gene&quot;, &quot;SNP&quot;), pch = c(22,21), title=&quot;Shape Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  legend(min(a$pos), y= 0.7 ,c(&quot;Lead TWAS Gene&quot;, &quot;R2 &gt; 0.4&quot;, &quot;R2 &lt;= 0.4&quot;), pch = 19, col = c(&quot;salmon&quot;, &quot;purple&quot;, colorsall[1]), title=&quot;Color Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  
  if (label==&quot;cTWAS&quot;){
    text(a$pos[a$id==focus], a$susie_pip[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
  
  par(mar = c(4.1, 4.1, 0.5, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 21, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;), frame.plot=FALSE, bg = colorsall[1], ylab = &quot;TWAS -log10(p value)&quot;, panel.first = grid(), ylim =c(0, max(a$PVALUE)*1.2))
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;SNP&quot;  &amp; a$r2max &gt; r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$PVALUE[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  abline(h=-log10(alpha/nrow(ctwas_gene_res)), col =&quot;red&quot;, lty = 2)
  
  if (label==&quot;TWAS&quot;){
    text(a$pos[a$id==focus], a$PVALUE[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
}

locus_plot(&quot;5_45&quot;, label=&quot;TWAS&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-19-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>locus_plot4 &lt;- function(region_tag, rerun_ctwas = F, plot_eqtl = T, label=&quot;cTWAS&quot;){
  region_tag1 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[1]
  region_tag2 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[2]
  
  a &lt;- ctwas_res[ctwas_res$region_tag==region_tag,]
  
  regionlist &lt;- readRDS(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.regionlist.RDS&quot;))
  region &lt;- regionlist[[as.numeric(region_tag1)]][[region_tag2]]
  
  R_snp_info &lt;- do.call(rbind, lapply(region$regRDS, function(x){data.table::fread(paste0(tools::file_path_sans_ext(x), &quot;.Rvar&quot;))}))
  
  if (isTRUE(rerun_ctwas)){
    ld_exprfs &lt;- paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, 1:22, &quot;.expr.gz&quot;)
    temp_reg &lt;- data.frame(&quot;chr&quot; = paste0(&quot;chr&quot;,region_tag1), &quot;start&quot; = region$start, &quot;stop&quot; = region$stop)
  
    write.table(temp_reg, 
                #file= paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.temp.reg.txt&quot;) , 
                file= &quot;temp_reg.txt&quot;,
                row.names=F, col.names=T, sep=&quot;\t&quot;, quote = F)
  
    load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_snp.Rd&quot;))
  
    z_gene_temp &lt;-  z_gene[z_gene$id %in% a$id[a$type==&quot;gene&quot;],]
    z_snp_temp &lt;-  z_snp[z_snp$id %in% R_snp_info$id,]
  
    ctwas_rss(z_gene_temp, z_snp_temp, ld_exprfs, ld_pgenfs = NULL, 
              ld_R_dir = dirname(region$regRDS)[1],
              ld_regions_custom = &quot;temp_reg.txt&quot;, thin = 1, 
              outputdir = &quot;.&quot;, outname = &quot;temp&quot;, ncore = 1, ncore.rerun = 1, prob_single = 0,
              group_prior = estimated_group_prior, group_prior_var = estimated_group_prior_var,
              estimate_group_prior = F, estimate_group_prior_var = F)
            
            
    a &lt;- data.table::fread(&quot;temp.susieIrss.txt&quot;, header = T)
    
    rownames(z_snp_temp) &lt;- z_snp_temp$id
    z_snp_temp &lt;- z_snp_temp[a$id[a$type==&quot;SNP&quot;],]
    z_gene_temp &lt;- z_gene_temp[a$id[a$type==&quot;gene&quot;],]
    
    a$z &lt;- NA
    a$z[a$type==&quot;SNP&quot;] &lt;- z_snp_temp$z
    a$z[a$type==&quot;gene&quot;] &lt;- z_gene_temp$z
  }
  
  a$ifcausal &lt;- 0
  focus &lt;- a$id[a$type==&quot;gene&quot;][which.max(abs(a$z[a$type==&quot;gene&quot;]))]
  a$ifcausal &lt;- as.numeric(a$id==focus)
    
  a$PVALUE &lt;- (-log(2) - pnorm(abs(a$z), lower.tail=F, log.p=T))/log(10)
  
  R_gene &lt;- readRDS(region$R_g_file)
  R_snp_gene &lt;- readRDS(region$R_sg_file)
  R_snp &lt;- as.matrix(Matrix::bdiag(lapply(region$regRDS, readRDS)))
  
  rownames(R_gene) &lt;- region$gid
  colnames(R_gene) &lt;- region$gid
  rownames(R_snp_gene) &lt;- R_snp_info$id
  colnames(R_snp_gene) &lt;- region$gid
  rownames(R_snp) &lt;- R_snp_info$id
  colnames(R_snp) &lt;- R_snp_info$id
  
  a$r2max &lt;- NA
  
  a$r2max[a$type==&quot;gene&quot;] &lt;- R_gene[focus,a$id[a$type==&quot;gene&quot;]]
  a$r2max[a$type==&quot;SNP&quot;] &lt;- R_snp_gene[a$id[a$type==&quot;SNP&quot;],focus]
  
  r2cut &lt;- 0.4
  colorsall &lt;- c(&quot;#7fc97f&quot;, &quot;#beaed4&quot;, &quot;#fdc086&quot;)
  
  layout(matrix(1:2, ncol = 1), widths = 1, heights = c(1.5,1.5), respect = FALSE)
  par(mar = c(0, 4.1, 4.1, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 19, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;),frame.plot=FALSE, col = &quot;white&quot;, ylim= c(-0.1,1.1), ylab = &quot;cTWAS PIP&quot;, xaxt = &#39;n&#39;)
  
  grid()
  points(a$pos[a$type==&quot;SNP&quot;], a$susie_pip[a$type == &quot;SNP&quot;], pch = 21, xlab=&quot;Genomic position&quot;, bg = colorsall[1])
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;SNP&quot;  &amp; a$r2max &gt;r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$susie_pip[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  
  if (isTRUE(plot_eqtl)){
    for (cgene in a[a$type==&quot;gene&quot; &amp; a$ifcausal == 1, ]$id){
      load(paste0(results_dir, &quot;/&quot;,analysis_id, &quot;_expr_chr&quot;, region_tag1, &quot;.exprqc.Rd&quot;))
      eqtls &lt;- rownames(wgtlist[[cgene]])
      points(a[a$id %in% eqtls,]$pos, rep( -0.15, nrow(a[a$id %in% eqtls,])), pch = &quot;|&quot;, col = &quot;salmon&quot;, cex = 1.5)
    }
  }
  
  #legend(min(a$pos), y= 1.1 ,c(&quot;Gene&quot;, &quot;SNP&quot;), pch = c(22,21), title=&quot;Shape Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  #legend(min(a$pos), y= 0.7 ,c(&quot;Lead TWAS Gene&quot;, &quot;R2 &gt; 0.4&quot;, &quot;R2 &lt;= 0.4&quot;), pch = 19, col = c(&quot;salmon&quot;, &quot;purple&quot;, colorsall[1]), title=&quot;Color Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  
  legend(max(a$pos)-0.2*(max(a$pos)-min(a$pos)), y= 1.1 ,c(&quot;Gene&quot;, &quot;SNP&quot;), pch = c(22,21), title=&quot;Shape Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  legend(max(a$pos)-0.2*(max(a$pos)-min(a$pos)), y= 0.7 ,c(&quot;Lead TWAS Gene&quot;, &quot;R2 &gt; 0.4&quot;, &quot;R2 &lt;= 0.4&quot;), pch = 19, col = c(&quot;salmon&quot;, &quot;purple&quot;, colorsall[1]), title=&quot;Color Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  
  if (label==&quot;cTWAS&quot;){
    text(a$pos[a$id==focus], a$susie_pip[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
  
  par(mar = c(4.1, 4.1, 0.5, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 21, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;), frame.plot=FALSE, bg = colorsall[1], ylab = &quot;TWAS -log10(p value)&quot;, panel.first = grid(), ylim =c(0, max(a$PVALUE)*1.2))
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;SNP&quot;  &amp; a$r2max &gt; r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$PVALUE[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  abline(h=-log10(alpha/nrow(ctwas_gene_res)), col =&quot;red&quot;, lty = 2)
  
  if (label==&quot;TWAS&quot;){
    text(a$pos[a$id==focus], a$PVALUE[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
}

locus_plot4(&quot;8_12&quot;, label=&quot;cTWAS&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-20-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>locus_plot5 &lt;- function(region_tag, rerun_ctwas = F, plot_eqtl = T, label=&quot;cTWAS&quot;, focus){
  region_tag1 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[1]
  region_tag2 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[2]
  
  a &lt;- ctwas_res[ctwas_res$region_tag==region_tag,]
  
  regionlist &lt;- readRDS(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.regionlist.RDS&quot;))
  region &lt;- regionlist[[as.numeric(region_tag1)]][[region_tag2]]
  
  R_snp_info &lt;- do.call(rbind, lapply(region$regRDS, function(x){data.table::fread(paste0(tools::file_path_sans_ext(x), &quot;.Rvar&quot;))}))
  
  if (isTRUE(rerun_ctwas)){
    ld_exprfs &lt;- paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, 1:22, &quot;.expr.gz&quot;)
    temp_reg &lt;- data.frame(&quot;chr&quot; = paste0(&quot;chr&quot;,region_tag1), &quot;start&quot; = region$start, &quot;stop&quot; = region$stop)
  
    write.table(temp_reg, 
                #file= paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.temp.reg.txt&quot;) , 
                file= &quot;temp_reg.txt&quot;,
                row.names=F, col.names=T, sep=&quot;\t&quot;, quote = F)
  
    load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_snp.Rd&quot;))
  
    z_gene_temp &lt;-  z_gene[z_gene$id %in% a$id[a$type==&quot;gene&quot;],]
    z_snp_temp &lt;-  z_snp[z_snp$id %in% R_snp_info$id,]
  
    ctwas_rss(z_gene_temp, z_snp_temp, ld_exprfs, ld_pgenfs = NULL, 
              ld_R_dir = dirname(region$regRDS)[1],
              ld_regions_custom = &quot;temp_reg.txt&quot;, thin = 1, 
              outputdir = &quot;.&quot;, outname = &quot;temp&quot;, ncore = 1, ncore.rerun = 1, prob_single = 0,
              group_prior = estimated_group_prior, group_prior_var = estimated_group_prior_var,
              estimate_group_prior = F, estimate_group_prior_var = F)
            
            
    a &lt;- data.table::fread(&quot;temp.susieIrss.txt&quot;, header = T)
    
    rownames(z_snp_temp) &lt;- z_snp_temp$id
    z_snp_temp &lt;- z_snp_temp[a$id[a$type==&quot;SNP&quot;],]
    z_gene_temp &lt;- z_gene_temp[a$id[a$type==&quot;gene&quot;],]
    
    a$z &lt;- NA
    a$z[a$type==&quot;SNP&quot;] &lt;- z_snp_temp$z
    a$z[a$type==&quot;gene&quot;] &lt;- z_gene_temp$z
  }
  
  a$ifcausal &lt;- 0
  focus &lt;- a$id[which(a$genename==focus)]
  a$ifcausal &lt;- as.numeric(a$id==focus)

  a$PVALUE &lt;- (-log(2) - pnorm(abs(a$z), lower.tail=F, log.p=T))/log(10)
  
  R_gene &lt;- readRDS(region$R_g_file)
  R_snp_gene &lt;- readRDS(region$R_sg_file)
  R_snp &lt;- as.matrix(Matrix::bdiag(lapply(region$regRDS, readRDS)))
  
  rownames(R_gene) &lt;- region$gid
  colnames(R_gene) &lt;- region$gid
  rownames(R_snp_gene) &lt;- R_snp_info$id
  colnames(R_snp_gene) &lt;- region$gid
  rownames(R_snp) &lt;- R_snp_info$id
  colnames(R_snp) &lt;- R_snp_info$id
  
  a$r2max &lt;- NA
  
  a$r2max[a$type==&quot;gene&quot;] &lt;- R_gene[focus,a$id[a$type==&quot;gene&quot;]]
  a$r2max[a$type==&quot;SNP&quot;] &lt;- R_snp_gene[a$id[a$type==&quot;SNP&quot;],focus]
  
  r2cut &lt;- 0.4
  colorsall &lt;- c(&quot;#7fc97f&quot;, &quot;#beaed4&quot;, &quot;#fdc086&quot;)
  
  layout(matrix(1:2, ncol = 1), widths = 1, heights = c(1.5,1.5), respect = FALSE)
  par(mar = c(0, 4.1, 4.1, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 19, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;),frame.plot=FALSE, col = &quot;white&quot;, ylim= c(-0.1,1.1), ylab = &quot;cTWAS PIP&quot;, xaxt = &#39;n&#39;)
  
  grid()
  points(a$pos[a$type==&quot;SNP&quot;], a$susie_pip[a$type == &quot;SNP&quot;], pch = 21, xlab=&quot;Genomic position&quot;, bg = colorsall[1])
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;SNP&quot;  &amp; a$r2max &gt;r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$susie_pip[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  
  if (isTRUE(plot_eqtl)){
    for (cgene in a[a$type==&quot;gene&quot; &amp; a$ifcausal == 1, ]$id){
      load(paste0(results_dir, &quot;/&quot;,analysis_id, &quot;_expr_chr&quot;, region_tag1, &quot;.exprqc.Rd&quot;))
      eqtls &lt;- rownames(wgtlist[[cgene]])
      points(a[a$id %in% eqtls,]$pos, rep( -0.15, nrow(a[a$id %in% eqtls,])), pch = &quot;|&quot;, col = &quot;salmon&quot;, cex = 1.5)
    }
  }
  
  legend(max(a$pos)-0.2*(max(a$pos)-min(a$pos)), y= 1.1 ,c(&quot;Gene&quot;, &quot;SNP&quot;), pch = c(22,21), title=&quot;Shape Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  legend(max(a$pos)-0.2*(max(a$pos)-min(a$pos)), y= 0.7 ,c(&quot;Focal Gene&quot;, &quot;R2 &gt; 0.4&quot;, &quot;R2 &lt;= 0.4&quot;), pch = 19, col = c(&quot;salmon&quot;, &quot;purple&quot;, colorsall[1]), title=&quot;Color Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  
  if (label==&quot;cTWAS&quot;){
    text(a$pos[a$id==focus], a$susie_pip[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
  
  par(mar = c(4.1, 4.1, 0.5, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 21, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;), frame.plot=FALSE, bg = colorsall[1], ylab = &quot;TWAS -log10(p value)&quot;, panel.first = grid(), ylim =c(0, max(a$PVALUE)*1.2))
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;SNP&quot;  &amp; a$r2max &gt; r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$PVALUE[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  abline(h=-log10(alpha/nrow(ctwas_gene_res)), col =&quot;red&quot;, lty = 2)
  
  if (label==&quot;TWAS&quot;){
    text(a$pos[a$id==focus], a$PVALUE[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
}

locus_plot5(&quot;19_33&quot;, focus=&quot;PRKD2&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-21-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>locus_plot3 &lt;- function(region_tag, rerun_ctwas = F, plot_eqtl = T, label=&quot;cTWAS&quot;, focus){
  region_tag1 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[1]
  region_tag2 &lt;- unlist(strsplit(region_tag, &quot;_&quot;))[2]
  
  a &lt;- ctwas_res[ctwas_res$region_tag==region_tag,]
  
  regionlist &lt;- readRDS(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.regionlist.RDS&quot;))
  region &lt;- regionlist[[as.numeric(region_tag1)]][[region_tag2]]
  
  R_snp_info &lt;- do.call(rbind, lapply(region$regRDS, function(x){data.table::fread(paste0(tools::file_path_sans_ext(x), &quot;.Rvar&quot;))}))
  
  if (isTRUE(rerun_ctwas)){
    ld_exprfs &lt;- paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, 1:22, &quot;.expr.gz&quot;)
    temp_reg &lt;- data.frame(&quot;chr&quot; = paste0(&quot;chr&quot;,region_tag1), &quot;start&quot; = region$start, &quot;stop&quot; = region$stop)
  
    write.table(temp_reg, 
                #file= paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.temp.reg.txt&quot;) , 
                file= &quot;temp_reg.txt&quot;,
                row.names=F, col.names=T, sep=&quot;\t&quot;, quote = F)
  
    load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_snp.Rd&quot;))
  
    z_gene_temp &lt;-  z_gene[z_gene$id %in% a$id[a$type==&quot;gene&quot;],]
    z_snp_temp &lt;-  z_snp[z_snp$id %in% R_snp_info$id,]
  
    ctwas_rss(z_gene_temp, z_snp_temp, ld_exprfs, ld_pgenfs = NULL, 
              ld_R_dir = dirname(region$regRDS)[1],
              ld_regions_custom = &quot;temp_reg.txt&quot;, thin = 1, 
              outputdir = &quot;.&quot;, outname = &quot;temp&quot;, ncore = 1, ncore.rerun = 1, prob_single = 0,
              group_prior = estimated_group_prior, group_prior_var = estimated_group_prior_var,
              estimate_group_prior = F, estimate_group_prior_var = F)
            
            
    a &lt;- data.table::fread(&quot;temp.susieIrss.txt&quot;, header = T)
    
    rownames(z_snp_temp) &lt;- z_snp_temp$id
    z_snp_temp &lt;- z_snp_temp[a$id[a$type==&quot;SNP&quot;],]
    z_gene_temp &lt;- z_gene_temp[a$id[a$type==&quot;gene&quot;],]
    
    a$z &lt;- NA
    a$z[a$type==&quot;SNP&quot;] &lt;- z_snp_temp$z
    a$z[a$type==&quot;gene&quot;] &lt;- z_gene_temp$z
  }
  
  a$ifcausal &lt;- 0
  focus &lt;- a$id[which(a$genename==focus)]
  a$ifcausal &lt;- as.numeric(a$id==focus)

  a$PVALUE &lt;- (-log(2) - pnorm(abs(a$z), lower.tail=F, log.p=T))/log(10)
  
  R_gene &lt;- readRDS(region$R_g_file)
  R_snp_gene &lt;- readRDS(region$R_sg_file)
  R_snp &lt;- as.matrix(Matrix::bdiag(lapply(region$regRDS, readRDS)))
  
  rownames(R_gene) &lt;- region$gid
  colnames(R_gene) &lt;- region$gid
  rownames(R_snp_gene) &lt;- R_snp_info$id
  colnames(R_snp_gene) &lt;- region$gid
  rownames(R_snp) &lt;- R_snp_info$id
  colnames(R_snp) &lt;- R_snp_info$id
  
  a$r2max &lt;- NA
  
  a$r2max[a$type==&quot;gene&quot;] &lt;- R_gene[focus,a$id[a$type==&quot;gene&quot;]]
  a$r2max[a$type==&quot;SNP&quot;] &lt;- R_snp_gene[a$id[a$type==&quot;SNP&quot;],focus]
  
  r2cut &lt;- 0.4
  colorsall &lt;- c(&quot;#7fc97f&quot;, &quot;#beaed4&quot;, &quot;#fdc086&quot;)
  
  layout(matrix(1:2, ncol = 1), widths = 1, heights = c(1.5,1.5), respect = FALSE)
  par(mar = c(0, 4.1, 4.1, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 19, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;),frame.plot=FALSE, col = &quot;white&quot;, ylim= c(-0.1,1.1), ylab = &quot;cTWAS PIP&quot;, xaxt = &#39;n&#39;)
  
  grid()
  points(a$pos[a$type==&quot;SNP&quot;], a$susie_pip[a$type == &quot;SNP&quot;], pch = 21, xlab=&quot;Genomic position&quot;, bg = colorsall[1])
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;SNP&quot;  &amp; a$r2max &gt;r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$susie_pip[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$susie_pip[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$susie_pip[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  
  if (isTRUE(plot_eqtl)){
    for (cgene in a[a$type==&quot;gene&quot; &amp; a$ifcausal == 1, ]$id){
      load(paste0(results_dir, &quot;/&quot;,analysis_id, &quot;_expr_chr&quot;, region_tag1, &quot;.exprqc.Rd&quot;))
      eqtls &lt;- rownames(wgtlist[[cgene]])
      points(a[a$id %in% eqtls,]$pos, rep( -0.15, nrow(a[a$id %in% eqtls,])), pch = &quot;|&quot;, col = &quot;salmon&quot;, cex = 1.5)
    }
  }
  
  legend(min(a$pos), y= 1.1 ,c(&quot;Gene&quot;, &quot;SNP&quot;), pch = c(22,21), title=&quot;Shape Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  legend(min(a$pos), y= 0.7 ,c(&quot;Lead TWAS Gene&quot;, &quot;R2 &gt; 0.4&quot;, &quot;R2 &lt;= 0.4&quot;), pch = 19, col = c(&quot;salmon&quot;, &quot;purple&quot;, colorsall[1]), title=&quot;Color Legend&quot;, bty =&#39;n&#39;, cex=0.6, title.adj = 0)
  
  if (label==&quot;cTWAS&quot;){
    text(a$pos[a$id==focus], a$susie_pip[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
  
  par(mar = c(4.1, 4.1, 0.5, 2.1))
  plot(a$pos[a$type==&quot;SNP&quot;], a$PVALUE[a$type == &quot;SNP&quot;], pch = 21, xlab=paste0(&quot;Chromosome &quot;, region_tag1, &quot; Position&quot;), frame.plot=FALSE, bg = colorsall[1], ylab = &quot;TWAS -log10(p value)&quot;, panel.first = grid(), ylim =c(0, max(a$PVALUE)*1.2))
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;SNP&quot;  &amp; a$r2max &gt; r2cut], pch = 21, bg = &quot;purple&quot;)
  points(a$pos[a$type==&quot;SNP&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;SNP&quot; &amp; a$ifcausal == 1], pch = 21, bg = &quot;salmon&quot;)
  points(a$pos[a$type==&quot;gene&quot;], a$PVALUE[a$type == &quot;gene&quot;], pch = 22, bg = colorsall[1], cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$r2max &gt; r2cut], a$PVALUE[a$type == &quot;gene&quot;  &amp; a$r2max &gt; r2cut], pch = 22, bg = &quot;purple&quot;, cex = 2)
  points(a$pos[a$type==&quot;gene&quot; &amp; a$ifcausal == 1], a$PVALUE[a$type == &quot;gene&quot; &amp; a$ifcausal == 1], pch = 22, bg = &quot;salmon&quot;, cex = 2)
  abline(h=-log10(alpha/nrow(ctwas_gene_res)), col =&quot;red&quot;, lty = 2)
  
  if (label==&quot;TWAS&quot;){
    text(a$pos[a$id==focus], a$PVALUE[a$id==focus], labels=ctwas_gene_res$genename[ctwas_gene_res$id==focus], pos=3, cex=0.6)
  }
}</code></pre>
<pre class="r"><code>load(paste0(results_dir, &quot;/known_annotations.Rd&quot;))
load(paste0(results_dir, &quot;/bystanders.Rd&quot;))

for (i in 1:length(known_annotations)){
  focus &lt;- known_annotations[i]
  region_tag &lt;- ctwas_res$region_tag[which(ctwas_res$genename==focus)]

  locus_plot3(region_tag, focus=focus)
  mtext(text=region_tag)

  print(focus)
  print(region_tag)
  print(ctwas_gene_res[ctwas_gene_res$region_tag==region_tag,report_cols,])
}</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOA2&quot;
[1] &quot;1_79&quot;
       genename region_tag  susie_pip       mu2          PVE           z
259      ATP1A2       1_79 0.02232677  4.551601 9.858264e-08  0.07777778
690        CD84       1_79 0.02445465  5.384296 1.277323e-07  0.51964897
1094     DUSP12       1_79 0.04437271 10.856957 4.673416e-07  1.41714286
1185      IGSF9       1_79 0.03513173  8.706494 2.967244e-07 -1.11342462
3375     SLAMF1       1_79 0.12883011 20.843702 2.604970e-06 -2.22875817
3376       CD48       1_79 0.02421329  5.293529 1.243396e-07 -0.45491803
3468       ATF6       1_79 0.02562464  5.812063 1.444769e-07  0.53097345
3790       COPA       1_79 0.04219359 10.392503 4.253800e-07  1.32350564
3791      CD244       1_79 0.02496083  5.571796 1.349164e-07  0.59471366
4628      FCRLA       1_79 0.05630893 13.060553 7.134266e-07 -1.77669903
4697     ATP1A4       1_79 0.10166656 18.589602 1.833406e-06  2.06586644
4702      DCAF8       1_79 0.04828663 11.637584 5.451300e-07 -1.43939394
5946       PPOX       1_79 0.02289066  4.779682 1.061372e-07  0.15573770
5947     FCGR2A       1_79 0.12905796 20.860563 2.611688e-06  2.48000000
5949       SDHC       1_79 0.11265004 19.562650 2.137812e-06 -2.08724832
5950      NR1I3       1_79 0.08553689 16.961091 1.407400e-06 -2.02857143
5954       PIGM       1_79 0.04527462 11.042659 4.849969e-07 -1.32314410
7220      ITLN2       1_79 0.02248328  4.615475 1.006669e-07  0.18987342
7221       F11R       1_79 0.02325707  4.924920 1.111129e-07  0.28619272
7223       NIT1       1_79 0.03060573  7.440007 2.208953e-07 -0.93879364
7233     NDUFS2       1_79 0.02802655  6.632779 1.803332e-07 -0.78307586
7235     FCER1G       1_79 0.02524152  5.674157 1.389400e-07  0.52631579
7236      APOA2       1_79 0.03370659  8.325963 2.722449e-07  1.21739130
7237    TOMM40L       1_79 0.07088884 15.202522 1.045452e-06 -1.91304348
7238        MPZ       1_79 0.02941405  7.075746 2.019005e-07 -0.75154614
7626     SLAMF9       1_79 0.02239659  4.580152 9.951128e-08  0.03448276
7627      IGSF8       1_79 0.05691677 13.160142 7.266265e-07 -1.52830189
7629      PEX19       1_79 0.02629840  6.049720 1.543388e-07  0.63280467
7630      NCSTN       1_79 0.03447255  8.532417 2.853356e-07 -0.99781384
7632      FCRLB       1_79 0.02491546  5.555148 1.342687e-07 -0.58151404
7634     KLHDC9       1_79 0.03356796  8.288108 2.698925e-07  1.06849315
9128    SPATA46       1_79 0.04523183 11.033926 4.841552e-07  1.23750000
9314      HSPA6       1_79 0.06806905 14.823938 9.788670e-07  1.82882883
9999      ITLN1       1_79 0.05095282 12.134618 5.997976e-07  1.54929577
11553    NOS1AP       1_79 0.02828404  6.716600 1.842898e-07  0.74025974
12062     TSTD1       1_79 0.02232309  4.550095 9.853378e-08  0.15944626
12145 LINC01133       1_79 0.05226939 12.370747 6.272690e-07 -1.48306018
12583  C1orf226       1_79 0.02229566  4.538854 9.816960e-08  0.02857143
12782    PCP4L1       1_79 0.03158946  7.730320 2.368918e-07  1.05063291
      num_eqtl
259          1
690          2
1094         1
1185         2
3375         1
3376         1
3468         1
3790         2
3791         1
4628         1
4697         2
4702         1
5946         1
5947         1
5949         1
5950         1
5954         1
7220         1
7221         2
7223         2
7233         3
7235         1
7236         1
7237         1
7238         2
7626         1
7627         1
7629         2
7630         2
7632         2
7634         1
9128         1
9314         1
9999         1
11553        1
12062        2
12145        3
12583        1
12782        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-2">
Past versions of unnamed-chunk-23-2.png
</button>
</p>
<div id="fig-unnamed-chunk-23-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-2.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ITIH4&quot;
[1] &quot;3_36&quot;
      genename region_tag  susie_pip       mu2          PVE          z num_eqtl
176     SEMA3G       3_36 0.03892813  5.882125 2.221305e-07 -0.3605769        1
177      NISCH       3_36 0.03501379  4.907373 1.666858e-07  0.2403846        1
178      STAB1       3_36 0.05267797  8.674289 4.432751e-07 -1.2993631        1
248       CHDH       3_36 0.03442462  4.751404 1.586725e-07 -0.2307692        1
251     GLT8D1       3_36 0.03500186  4.904243 1.665228e-07  0.2814371        1
292      ALAS1       3_36 0.05805748  9.575996 5.393275e-07 -1.0765027        1
531      ITIH4       3_36 0.03840350  5.757342 2.144882e-07  0.8173077        1
538     IL17RB       3_36 0.04383722  6.976700 2.966904e-07 -0.7272727        1
3050   SELENOK       3_36 0.05845472  9.639338 5.466095e-07 -1.1715552        2
3051     ACTR8       3_36 0.07702387 12.210050 9.123326e-07  1.5751679        2
3113      RRP9       3_36 0.05886224  9.703827 5.541027e-07 -1.0801527        1
3120     DNAH1       3_36 0.03448260  4.766850 1.594564e-07 -0.3181818        1
3123     TNNC1       3_36 0.08797256 13.456208 1.148366e-06 -1.7987805        1
3128      NEK4       3_36 0.06425446 10.518637 6.556517e-07  1.8823529        1
7112   CACNA1D       3_36 0.07625317 12.115995 8.962464e-07  1.8985507        1
7543     RPL29       3_36 0.07191249 11.568177 8.070114e-07  1.4583333        1
7544     ITIH3       3_36 0.04958308  8.114356 3.902995e-07 -1.5362319        1
7867       TKT       3_36 0.10852581 15.440173 1.625532e-06  2.1739130        1
7868     PRKCD       3_36 0.03655681  5.303862 1.880923e-07 -0.4230769        1
7869      RFT1       3_36 0.05539321  9.140198 4.911595e-07  1.7910448        1
7871      GNL3       3_36 0.05767993  9.515554 5.324382e-07  1.7925028        2
7912     POC1A       3_36 0.03705553  5.428529 1.951397e-07  0.4108527        1
7913     PPM1M       3_36 0.03670298  5.340552 1.901507e-07 -0.3214261        2
8637    GLYCTK       3_36 0.03444132  4.755838 1.588976e-07  0.1515152        1
8642    NT5DC2       3_36 0.04435247  7.084555 3.048181e-07 -1.3761055        2
8643     SMIM4       3_36 0.05073643  8.327070 4.098478e-07  1.5074627        1
11922  TMEM110       3_36 0.04187176  6.553733 2.662076e-07 -1.6043526        2
12579     TLR9       3_36 0.16509570 19.471131 3.118440e-06 -2.2559524        1
12702     ACY1       3_36 0.03545778  5.023196 1.727834e-07  0.4567901        1
12767     TWF2       3_36 0.07691229 12.196522 9.100016e-07 -1.6206897        1
13640   MUSTN1       3_36 0.04590077  7.401293 3.295627e-07 -1.5443038        1
13678    DCP1A       3_36 0.04258930  6.710355 2.772403e-07  0.8337349        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-3">
Past versions of unnamed-chunk-23-3.png
</button>
</p>
<div id="fig-unnamed-chunk-23-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-3.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;GHR&quot;
[1] &quot;5_28&quot;
      genename region_tag  susie_pip       mu2          PVE           z
2969       GHR       5_28 0.19125588 17.236901 3.198044e-06 -2.68604651
2970    HMGCS1       5_28 0.05652290  5.649690 3.097844e-07  0.64482759
2973       NNT       5_28 0.05054509  4.614259 2.262514e-07 -0.05662818
6681     FBXO4       5_28 0.05180791  4.842651 2.433827e-07 -0.73297467
6682   TMEM267       5_28 0.06372503  6.763756 4.181272e-07  0.83582090
9758     NIM1K       5_28 0.06765561  7.321140 4.804995e-07  0.98373058
11530  CCDC152       5_28 0.15418984 15.129397 2.263017e-06 -2.48172469
12854  SELENOP       5_28 0.06422306  6.836265 4.259125e-07  0.90099010
      num_eqtl
2969         1
2970         1
2973         2
6681         2
6682         1
9758         2
11530        2
12854        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-4">
Past versions of unnamed-chunk-23-4.png
</button>
</p>
<div id="fig-unnamed-chunk-23-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-4.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;EPHX2&quot;
[1] &quot;8_27&quot;
      genename region_tag  susie_pip       mu2          PVE           z
1435    DPYSL2       8_27 0.02485388  4.784257 1.153504e-07 -0.27748691
2097    BNIP3L       8_27 0.03748736  8.555083 3.111140e-07 -1.08571429
3705    ADRA1A       8_27 0.22411517 25.514834 5.547208e-06  2.94751053
12086     EBF2       8_27 0.02652548  5.380226 1.384441e-07  0.48439655
12602    PNMA2       8_27 0.02451811  4.659772 1.108312e-07 -0.07042254
2037    TRIM35       8_27 0.02423569  4.553755 1.070620e-07 -0.08536585
3702       CLU       8_27 0.02460261  4.691260 1.119647e-07  0.38023585
3704     PTK2B       8_27 0.02737181  5.667939 1.505009e-07  0.51339769
3708     EPHX2       8_27 0.02649535  5.369815 1.380191e-07  0.48402855
6343    CCDC25       8_27 0.02499550  4.836267 1.172688e-07 -0.25224323
8618    SCARA3       8_27 0.03280532  7.328798 2.332316e-07  1.00883393
9082     ESCO2       8_27 0.03536059  8.017939 2.750380e-07 -1.10606061
      num_eqtl
1435         1
2097         1
3705         2
12086        2
12602        1
2037         1
3702         2
3704         3
3708         3
6343         2
8618         2
9082         1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-5">
Past versions of unnamed-chunk-23-5.png
</button>
</p>
<div id="fig-unnamed-chunk-23-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-5.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;LPL&quot;
[1] &quot;8_21&quot;
       genename region_tag  susie_pip       mu2          PVE          z
2096      ASAH1       8_21 0.06087832 22.207900 1.311537e-06 -3.0530181
6342 CSGALNACT1       8_21 0.03028582 10.539089 3.096370e-07 -1.2215401
7002       NAT2       8_21 0.11610082 26.464882 2.980682e-06  3.6206897
7003       PSD3       8_21 0.01596183  4.798613 7.430341e-08 -0.2647059
2077     INTS10       8_21 0.02759994  9.793581 2.622165e-07 -1.1484375
9575        LPL       8_21 0.04832215 14.815748 6.945128e-07  1.7500000
     num_eqtl
2096        2
6342        2
7002        1
7003        1
2077        1
9575        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-6">
Past versions of unnamed-chunk-23-6.png
</button>
</p>
<div id="fig-unnamed-chunk-23-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-6.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;MTTP&quot;
[1] &quot;4_66&quot;
           genename region_tag  susie_pip       mu2          PVE           z
5518           MTTP       4_66 0.07672006  7.882237 5.866361e-07 -1.04046243
6184        TRMT10A       4_66 0.05371170  4.563792 2.377963e-07  0.02083333
6617          EIF4E       4_66 0.11727722 11.886921 1.352364e-06 -1.81333333
8714         TSPAN5       4_66 0.06040562  5.653403 3.312819e-07  0.44252874
9300           ADH6       4_66 0.17858509 15.949040 2.763059e-06 -2.40716823
11121         ADH1B       4_66 0.08911404  9.287286 8.028703e-07  1.29411765
11329          ADH5       4_66 0.23991541 18.891767 4.396845e-06  2.65454545
12774         ADH1C       4_66 0.08529751  8.875806 7.344370e-07  1.23802332
12794 RP11-766F14.2       4_66 0.06015217  5.614326 3.276117e-07 -0.44844283
13664 RP11-571L19.8       4_66 0.16995144 15.463789 2.549477e-06 -2.35541164
      num_eqtl
5518         1
6184         1
6617         1
8714         1
9300         2
11121        1
11329        1
12774        2
12794        2
13664        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-7">
Past versions of unnamed-chunk-23-7.png
</button>
</p>
<div id="fig-unnamed-chunk-23-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-7.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;DHCR7&quot;
[1] &quot;11_40&quot;
      genename region_tag  susie_pip       mu2          PVE          z num_eqtl
2685     FOLR1      11_40 0.06258367  5.213023 3.164908e-07 -0.4785714        1
5304    IL18BP      11_40 0.10890933 10.413126 1.100162e-06  1.4642857        1
5312    RNF121      11_40 0.06163017  5.070277 3.031346e-07  0.3772455        1
7534      CLPB      11_40 0.13114123 12.185398 1.550206e-06  1.7218935        1
8137     FOLR2      11_40 0.08772321  8.369647 7.122494e-07  1.0524345        1
9291   NADSYN1      11_40 0.06402732  5.425169 3.369683e-07  0.5577167        2
9292     DHCR7      11_40 0.05900796  4.666336 2.671142e-07 -0.1466667        1
10407   LRTOMT      11_40 0.06061223  4.915490 2.890264e-07 -0.3013699        1
12963 KRTAP5-9      11_40 0.08504535  8.078318 6.664721e-07 -1.0649351        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-8">
Past versions of unnamed-chunk-23-8.png
</button>
</p>
<div id="fig-unnamed-chunk-23-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-8.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;LIPA&quot;
[1] &quot;10_57&quot;
         genename region_tag  susie_pip       mu2          PVE           z
2455       BMPR1A      10_57 0.03519791  9.444573 3.224850e-07  1.04477612
3796         OPN4      10_57 0.11736877 20.673933 2.353890e-06 -2.45405716
3797       FAM35A      10_57 0.02198949  5.133217 1.095003e-07 -0.43750000
9339         SNCG      10_57 0.02087998  4.660016 9.439041e-08  0.22058824
9340        MMRN2      10_57 0.02149415  4.924955 1.026911e-07  0.43283582
10489      NUTM2A      10_57 0.02122421  4.809446 9.902322e-08  0.27432909
12160   LINC00863      10_57 0.02558729  6.519488 1.618260e-07  0.71187876
313           FAS      10_57 0.05925597 14.254753 8.194118e-07 -1.71641791
2456       MINPP1      10_57 0.02525645  6.400349 1.568145e-07  0.69432314
2457        ACTA2      10_57 0.02061418  4.542955 9.084789e-08 -0.03797468
2458         LIPA      10_57 0.02967295  7.876891 2.267389e-07 -0.94788048
5408        ATAD1      10_57 0.02561701  6.530118 1.622781e-07 -0.73134328
6757      ANKRD22      10_57 0.05908532 14.227973 8.155171e-07  1.64516129
10211        LIPF      10_57 0.03066253  8.177807 2.432513e-07 -0.95108696
10468        RNLS      10_57 0.02058621  4.530549 9.047687e-08 -0.03707098
11472      PAPSS2      10_57 0.02479869  6.232931 1.499449e-07  0.70178462
11607        LIPN      10_57 0.02064596  4.557032 9.126993e-08 -0.08219178
12241        KLLN      10_57 0.02916591  7.718844 2.183928e-07 -0.90551181
5415       KIF20B      10_57 0.02786753  7.301416 1.973858e-07 -0.90599016
6759        IFIT5      10_57 0.03836548 10.237178 3.810055e-07 -1.19736842
6760     SLC16A12      10_57 0.18461536 25.055997 4.487350e-06 -2.58551947
12374   LINC00865      10_57 0.06301921 14.826994 9.064346e-07  1.73137107
12460 RP11-80H5.9      10_57 0.04484944 11.676348 5.080125e-07  1.53051643
12612 RP11-80H5.7      10_57 0.04842283 12.384402 5.817490e-07  1.59433962
      num_eqtl
2455         1
3796         2
3797         1
9339         1
9340         1
10489        2
12160        2
313          1
2456         1
2457         1
2458         2
5408         1
6757         1
10211        1
10468        2
11472        2
11607        1
12241        1
5415         2
6759         1
6760         2
12374        2
12460        1
12612        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-9">
Past versions of unnamed-chunk-23-9.png
</button>
</p>
<div id="fig-unnamed-chunk-23-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-9.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;LDLRAP1&quot;
[1] &quot;1_18&quot;
          genename region_tag  susie_pip       mu2          PVE          z
575           PIGV       1_18 0.06374105  6.967278 4.308170e-07  0.9159664
3435          SYF2       1_18 0.14346972 14.633673 2.036686e-06 -1.6825397
3438        MTFR1L       1_18 0.04980418  4.678812 2.260538e-07 -0.3536585
3440       RPS6KA1       1_18 0.07102014  7.974975 5.494413e-07  1.0059524
3441         DHDDS       1_18 0.05380369  5.393824 2.815265e-07 -0.6725101
4211         AUNIP       1_18 0.09201181 10.402720 9.285406e-07 -1.4179104
4603         NR0B2       1_18 0.04979733  4.677539 2.259612e-07 -0.1588282
5888      SH3BGRL3       1_18 0.05022431  4.756530 2.317473e-07 -0.4166667
5889        CNKSR1       1_18 0.05166707  5.018596 2.515396e-07 -0.3717949
5897          GPN2       1_18 0.05919837  6.280058 3.606483e-07 -0.7983193
7154       LDLRAP1       1_18 0.04901100  4.530361 2.153956e-07 -0.1515152
7157        PAFAH2       1_18 0.04935011  4.594120 2.199383e-07  0.2173913
7158         EXTL1       1_18 0.05268064  5.198438 2.656650e-07 -0.6067637
7159       SLC30A2       1_18 0.05322706  5.294011 2.733555e-07  0.5277778
7161        TRIM63       1_18 0.09101490 10.300110 9.094206e-07 -1.5035039
7166        UBXN11       1_18 0.07037800  7.890028 5.386739e-07  0.5286808
7567       SELENON       1_18 0.05390663  5.411522 2.829906e-07 -0.2835821
8813          CD52       1_18 0.05640361  5.831306 3.190679e-07  0.9574677
9605          KDF1       1_18 0.06375737  6.969660 4.310746e-07  0.9159664
9632        ZNF683       1_18 0.11491686 12.507060 1.394278e-06  2.1467890
10250        PAQR7       1_18 0.05138529  4.967951 2.476433e-07 -0.3214286
10367      TMEM50A       1_18 0.18574567 17.150421 3.090323e-06  2.2714286
10923         RHCE       1_18 0.08253391  9.380747 7.510697e-07 -1.4973672
11222      FAM110D       1_18 0.06128334  6.601642 3.924685e-07  0.6959054
11515        HMGN2       1_18 0.05094986  4.889203 2.416526e-07 -0.3648649
11621      ZDHHC18       1_18 0.11413453 12.442379 1.377625e-06 -1.5652174
12514 RP11-96L14.7       1_18 0.07056985  7.915638 5.418955e-07 -1.3544304
      num_eqtl
575          1
3435         1
3438         1
3440         1
3441         2
4211         1
4603         2
5888         1
5889         1
5897         1
7154         1
7157         1
7158         3
7159         1
7161         2
7166         2
7567         1
8813         3
9605         1
9632         1
10250        1
10367        1
10923        2
11222        3
11515        1
11621        1
12514        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-10">
Past versions of unnamed-chunk-23-10.png
</button>
</p>
<div id="fig-unnamed-chunk-23-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-10.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOB&quot;
[1] &quot;2_13&quot;
        genename region_tag  susie_pip      mu2          PVE         z num_eqtl
1162        APOB       2_13 0.05171173 6.022032 3.020943e-07 0.8315789        1
12398 AC067959.1       2_13 0.06126714 7.594636 4.513828e-07 1.0606061        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-11">
Past versions of unnamed-chunk-23-11.png
</button>
</p>
<div id="fig-unnamed-chunk-23-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-11.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOE&quot;
[1] &quot;19_31&quot;
      genename region_tag  susie_pip       mu2          PVE           z
204      PLAUR      19_31 0.02849177  4.808054 1.328921e-07  0.31958763
867      XRCC1      19_31 0.02829552  4.744703 1.302378e-07  0.25581395
2099     KCNN4      19_31 0.04351501  8.701255 3.673089e-07 -1.06930693
2266     ETHE1      19_31 0.03414140  6.468101 2.142242e-07  0.68748771
2267      SMG9      19_31 0.03758500  7.351636 2.680458e-07 -0.89043174
3971     ZNF45      19_31 0.05900086 11.518291 6.592601e-07 -1.37024221
4540    ZNF227      19_31 0.03640418  7.057995 2.492545e-07 -0.91280315
7319    ZNF230      19_31 0.03395243  6.417107 2.113589e-07 -0.73630137
7323    ZNF221      19_31 0.03974818  7.866755 3.033355e-07 -0.95714286
8475      IRGQ      19_31 0.02777114  4.573267 1.232057e-07  0.11217949
8476    ZNF226      19_31 0.02765363  4.534412 1.216420e-07  0.06122449
8519    ZNF283      19_31 0.03058882  5.459326 1.619989e-07 -0.46268657
9651    ZNF404      19_31 0.03087040  5.543399 1.660079e-07 -0.46268657
9865    ZNF223      19_31 0.03028714  5.368403 1.577298e-07  0.44285714
10644   ZNF284      19_31 0.02770775  4.552327 1.223616e-07  0.10101010
11740   ZNF155      19_31 0.02962992  5.167181 1.485233e-07  0.47048028
12434   PINLYP      19_31 0.02765847  4.536014 1.217063e-07 -0.06795808
13018   ZNF225      19_31 0.04727306  9.465932 4.340977e-07 -1.17391304
13291   ZNF234      19_31 0.02770092  4.550066 1.222707e-07  0.11594203
13411   ZNF224      19_31 0.03783347  7.412266 2.720430e-07  0.89772727
112      MARK4      19_31 0.03067690  5.485707 1.632505e-07 -0.47282609
119   TRAPPC6A      19_31 0.06385391 12.253005 7.589978e-07 -1.43410853
214      ERCC1      19_31 0.06209460 11.993145 7.224326e-07  1.43750000
593     ZNF112      19_31 0.02823966  4.726588 1.294844e-07  0.15555945
866        PVR      19_31 0.02786954  4.605680 1.245186e-07 -0.16666667
2111    CLPTM1      19_31 0.04952683  9.896424 4.754767e-07  1.22413793
2113   PPP1R37      19_31 0.02764689  4.532178 1.215524e-07  0.02325315
2115       CKM      19_31 0.02947524  5.119184 1.463756e-07  0.42066734
2117  PPP1R13L      19_31 0.03936284  7.777046 2.969693e-07  0.94933333
3451    CD3EAP      19_31 0.02904211  4.983442 1.404003e-07  0.38478177
4079      FOSB      19_31 0.02784195  4.596603 1.241501e-07  0.11538462
4080      OPA3      19_31 0.04559245  9.131621 4.038790e-07  1.09859155
4082      RTN2      19_31 0.03633139  7.039587 2.481074e-07  0.82935506
4412   NECTIN2      19_31 0.03594458  6.941161 2.420338e-07  0.91447368
4413      APOE      19_31 0.06880641 12.948792 8.643080e-07  1.75000000
4414    TOMM40      19_31 0.03077329  5.514488 1.646226e-07 -0.59268930
4415     APOC1      19_31 0.16958180 21.526348 3.541278e-06  2.46376812
5860    GEMIN7      19_31 0.04929982  9.853941 4.712656e-07 -1.23208778
7324    ZNF233      19_31 0.14629720 20.091293 2.851375e-06  2.44329897
7325    ZNF235      19_31 0.06927352 13.011865 8.744143e-07  1.88899386
8477    ZNF180      19_31 0.03262520  6.050867 1.915055e-07  0.70149254
8993    ZNF296      19_31 0.02805998  4.668090 1.270682e-07  0.19444444
10708 CEACAM19      19_31 0.03280796  6.102161 1.942108e-07  0.66278513
10978  BLOC1S3      19_31 0.02976706  5.209531 1.504337e-07 -0.41492537
11951    PPM1N      19_31 0.02966612  5.178378 1.490270e-07 -0.46241261
12455    APOC2      19_31 0.03222310  5.937003 1.855859e-07  0.52968037
13399   ZNF285      19_31 0.02878369  4.901500 1.368630e-07 -0.35172414
      num_eqtl
204          1
867          1
2099         1
2266         2
2267         2
3971         1
4540         2
7319         1
7323         1
8475         1
8476         1
8519         1
9651         1
9865         1
10644        1
11740        2
12434        3
13018        1
13291        1
13411        1
112          1
119          1
214          1
593          2
866          1
2111         1
2113         2
2115         2
2117         1
3451         2
4079         1
4080         1
4082         2
4412         1
4413         1
4414         1
4415         1
5860         3
7324         1
7325         2
8477         1
8993         1
10708        2
10978        1
11951        2
12455        1
13399        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-12">
Past versions of unnamed-chunk-23-12.png
</button>
</p>
<div id="fig-unnamed-chunk-23-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-12.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;SOAT1&quot;
[1] &quot;1_89&quot;
           genename region_tag  susie_pip       mu2          PVE           z
540           SOAT1       1_89 0.05268186  8.479530 4.333545e-07  1.01176471
3288         FAM20B       1_89 0.04168309  6.316295 2.554070e-07  0.72222222
3297          QSOX1       1_89 0.03814560  5.499501 2.035064e-07 -0.56944444
3738           LHX4       1_89 0.03450814  4.577904 1.532494e-07 -0.22123894
5072       KIAA1614       1_89 0.09426456 13.911566 1.272140e-06  1.55555556
5073         CEP350       1_89 0.05369001  8.655103 4.507919e-07  1.19655172
5957           ABL2       1_89 0.03712037  5.248817 1.890097e-07 -0.65194565
5958           XPR1       1_89 0.25759467 23.712071 5.925388e-06 -2.84545455
5961        FAM163A       1_89 0.03564314  4.875355 1.685748e-07 -0.27882629
10672         TOR3A       1_89 0.03445531  4.563826 1.525442e-07  0.05797101
12311         ACBD6       1_89 0.15917424 18.920478 2.921563e-06  2.67647059
13170 RP11-533E19.5       1_89 0.05369001  8.655103 4.507919e-07  1.19655172
      num_eqtl
540          1
3288         1
3297         1
3738         1
5072         1
5073         1
5957         2
5958         1
5961         2
10672        1
12311        1
13170        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-13">
Past versions of unnamed-chunk-23-13.png
</button>
</p>
<div id="fig-unnamed-chunk-23-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-13.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;MYLIP&quot;
[1] &quot;6_13&quot;
          genename region_tag  susie_pip       mu2          PVE         z
133          MYLIP       6_13 0.03540141  8.989224 3.087118e-07 1.0444444
437         DTNBP1       6_13 0.02834756  6.944155 1.909613e-07 0.8356641
5264          GMPR       6_13 0.02157356  5.014321 1.049408e-07 0.7214247
13562 RP11-560J1.2       6_13 0.05478702 13.476172 7.162336e-07 2.0738691
13609 RP1-151F17.2       6_13 0.04426768 10.793664 4.635175e-07 1.1410021
      num_eqtl
133          1
437          2
5264         2
13562        2
13609        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-14">
Past versions of unnamed-chunk-23-14.png
</button>
</p>
<div id="fig-unnamed-chunk-23-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-14.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;SCARB1&quot;
[1] &quot;12_76&quot;
     genename region_tag  susie_pip       mu2          PVE            z
868    SCARB1      12_76 0.62884142 20.201894 1.232377e-05  3.511961722
1095     AACS      12_76 0.06933066  7.380080 4.963600e-07  0.971014493
5568 TMEM132B      12_76 0.05107164  4.534029 2.246335e-07 -0.002253938
6595    DHX37      12_76 0.11417682 12.222112 1.353738e-06 -1.600000000
6596      UBC      12_76 0.13556009 13.784176 1.812688e-06  1.712707182
     num_eqtl
868         1
1095        1
5568        3
6595        1
6596        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-15">
Past versions of unnamed-chunk-23-15.png
</button>
</p>
<div id="fig-unnamed-chunk-23-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-15.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ABCB11&quot;
[1] &quot;2_102&quot;
     genename region_tag  susie_pip      mu2          PVE           z num_eqtl
885    ABCB11      2_102 0.04874027 7.916105 3.742914e-07 -1.05981763        2
886     DHRS9      2_102 0.03386763 4.561817 1.498763e-07 -0.09510815        2
6708    SPC25      2_102 0.03743384 5.482276 1.990837e-07  0.88512935        3
9208    CERS6      2_102 0.04146862 6.424799 2.584578e-07 -0.67662405        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-16">
Past versions of unnamed-chunk-23-16.png
</button>
</p>
<div id="fig-unnamed-chunk-23-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-16.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;CETP&quot;
[1] &quot;16_30&quot;
          genename region_tag  susie_pip       mu2          PVE           z
55         CIAPIN1      16_30 0.04845288  6.302228 2.962266e-07 -0.73183714
87          CX3CL1      16_30 0.07153896  9.920655 6.884833e-07 -1.27536232
476        HERPUD1      16_30 0.05680504  7.774791 4.284361e-07 -0.96998736
1234          CETP      16_30 0.07838425 10.775033 8.193281e-07 -1.40545710
1237         GNAO1      16_30 0.04013178  4.563708 1.776710e-07 -0.08828906
1274          COQ9      16_30 0.11314960 14.237664 1.562796e-06 -1.79032091
1904         NUP93      16_30 0.04463449  5.544105 2.400559e-07 -0.56595745
1913        POLR2C      16_30 0.04887837  6.383052 3.026603e-07 -0.74626866
4021          BBS2      16_30 0.04248260  5.088312 2.096984e-07 -0.45344441
4023          MT2A      16_30 0.04970500  6.538125 3.152562e-07 -0.78160920
4026          DOK4      16_30 0.04048193  4.643731 1.823638e-07 -0.13131313
5060      CCDC102A      16_30 0.04419186  5.452138 2.337327e-07 -0.50704225
5718         CPNE2      16_30 0.04730006  6.079700 2.789679e-07 -0.65178571
5719         NLRC5      16_30 0.15157292 17.042580 2.505921e-06 -1.96932515
7285         CES5A      16_30 0.04847434  6.306322 2.965503e-07 -0.74264706
7293          AMFR      16_30 0.04732496  6.084562 2.793380e-07  0.69115648
7296        RSPRY1      16_30 0.07009822  9.730778 6.617059e-07 -1.25362319
8424        NUDT21      16_30 0.04005713  4.546559 1.766742e-07 -0.06870005
8846          MT1E      16_30 0.09262666 12.343192 1.109108e-06  1.51612903
9275       FAM192A      16_30 0.05394953  7.296643 3.818750e-07 -0.97354497
10774         MT1X      16_30 0.04139045  4.848234 1.946678e-07 -0.26865672
11419         MT1F      16_30 0.04985897  6.566730 3.176164e-07 -0.70526316
11792         MT1H      16_30 0.04000831  4.535327 1.760229e-07 -0.01470588
11793         MT1A      16_30 0.08943528 12.013025 1.042249e-06  1.47590361
11795         MT1M      16_30 0.04720573  6.061256 2.775669e-07 -0.68013825
12749 RP11-461O7.1      16_30 0.04242036  5.074794 2.088350e-07 -0.36774954
      num_eqtl
55           2
87           1
476          2
1234         2
1237         2
1274         2
1904         1
1913         1
4021         2
4023         1
4026         1
5060         1
5718         1
5719         1
7285         1
7293         2
7296         1
8424         2
8846         1
9275         1
10774        1
11419        1
11792        1
11793        1
11795        2
12749        3</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-17">
Past versions of unnamed-chunk-23-17.png
</button>
</p>
<div id="fig-unnamed-chunk-23-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-17.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOH&quot;
[1] &quot;17_38&quot;
     genename region_tag  susie_pip       mu2          PVE          z num_eqtl
1394     APOH      17_38 0.13334553 14.802053 1.914744e-06 -2.0358974        1
6876    PRKCA      17_38 0.40925888 26.254174 1.042334e-05 -3.6842105        1
6878   CEP112      17_38 0.04898550  5.398308 2.565285e-07  0.9939394        1
8696    AXIN2      17_38 0.05966651  7.225824 4.182427e-07  0.8732394        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-18">
Past versions of unnamed-chunk-23-18.png
</button>
</p>
<div id="fig-unnamed-chunk-23-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-18.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;TSPO&quot;
[1] &quot;22_18&quot;
      genename region_tag  susie_pip       mu2          PVE          z num_eqtl
1614    CYB5R3      22_18 0.04708721  5.352272 2.444847e-07 -0.4895833        1
1619   PACSIN2      22_18 0.04695762  5.326821 2.426524e-07 -0.4969661        2
1620     TTLL1      22_18 0.06193207  7.891516 4.741180e-07 -1.0573210        2
1627      MCAT      22_18 0.05771660  7.236749 4.051862e-07 -0.9702970        1
1631      TSPO      22_18 0.04336523  4.592180 1.931839e-07 -0.2083333        1
1633    TTLL12      22_18 0.33762365 24.457289 8.010352e-06  2.9906361        3
4272    A4GALT      22_18 0.08909122 11.290982 9.758364e-07  1.4286502        2
10343   SERHL2      22_18 0.04385792  4.696390 1.998125e-07  0.2220141        4
10997    RRP7A      22_18 0.04472207  4.876429 2.115603e-07 -0.1760735        6
12647  ARFGAP3      22_18 0.08577514 10.934347 9.098393e-07  1.3790111        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-19">
Past versions of unnamed-chunk-23-19.png
</button>
</p>
<div id="fig-unnamed-chunk-23-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-19.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;PLTP&quot;
[1] &quot;20_28&quot;
           genename region_tag  susie_pip       mu2          PVE           z
310          TOMM34      20_28 0.02468007  5.500228 1.316853e-07  0.48837209
621           WISP2      20_28 0.02479161  5.579655 1.341907e-07  0.60139860
629            CTSA      20_28 0.04075101 10.074828 3.982782e-07  1.52238806
1755           PLTP      20_28 0.06905741 14.928154 1.000062e-06  1.86250000
1756          PCIF1      20_28 0.08025242 16.355427 1.273299e-06 -2.00769231
1758           MMP9      20_28 0.03504881  8.742433 2.972460e-07  1.12113489
1766           CD40      20_28 0.04733968 11.437378 5.252453e-07  1.71428571
1774          RIMS4      20_28 0.02387507  5.232015 1.211781e-07  0.53846154
1841          WFDC2      20_28 0.02508320  5.676817 1.381333e-07 -0.57003938
1845        DNTTIP1      20_28 0.02364407  5.121869 1.174793e-07  0.56338028
1848          ACOT8      20_28 0.07003223 15.127497 1.027722e-06  1.73584906
3926          SNX21      20_28 0.02213841  4.531737 9.732435e-08  0.21865506
3927           SLPI      20_28 0.02223462  4.568065 9.853087e-08  0.21872230
3928          WFDC3      20_28 0.02218200  4.552138 9.795499e-08  0.06067048
3930        SLC12A5      20_28 0.10205195 18.636907 1.845039e-06  2.08823529
3931           SDC4      20_28 0.02234897  4.618843 1.001385e-07 -0.18571429
3933          MATN4      20_28 0.04658125 11.387780 5.145891e-07 -1.45454545
3934          NCOA5      20_28 0.17522231 23.619903 4.014930e-06  2.74698795
3951          RBPJL      20_28 0.05823843 13.358664 7.547152e-07  1.61333333
3953         KCNK15      20_28 0.06597254 14.580137 9.331151e-07 -1.94169918
3954        TP53TG5      20_28 0.03004858  7.345751 2.141266e-07 -0.80103890
3957         NEURL2      20_28 0.06397941 14.235770 8.835510e-07  1.79219427
4713          OSER1      20_28 0.02316326  4.898168 1.100636e-07 -0.11842105
4714        SERINC3      20_28 0.02440935  5.415560 1.282360e-07 -0.41420118
6528           JPH2      20_28 0.02894562  7.140966 2.005166e-07 -0.97014925
6532        SPATA25      20_28 0.04316533 10.600331 4.438794e-07  1.42105582
8404          YWHAB      20_28 0.03229352  7.979207 2.499686e-07 -1.06664098
8706           PKIG      20_28 0.02360497  5.135441 1.175957e-07 -0.48837209
11156           ADA      20_28 0.02412992  5.322283 1.245846e-07  0.58282209
11231         FITM2      20_28 0.34466420 30.888719 1.032777e-05  3.47633136
11609          SYS1      20_28 0.02494313  5.624333 1.360919e-07 -0.52857143
12140     OSER1-AS1      20_28 0.02660360  6.086123 1.570694e-07 -0.42101203
12711        DBNDD2      20_28 0.02943472  7.141745 2.039270e-07 -0.75153374
14021 RP11-445H22.3      20_28 0.08667650 17.159906 1.442868e-06 -2.02222222
      num_eqtl
310          1
621          1
629          1
1755         1
1756         1
1758         2
1766         1
1774         1
1841         2
1845         1
1848         1
3926         2
3927         2
3928         2
3930         1
3931         1
3933         1
3934         1
3951         1
3953         2
3954         2
3957         2
4713         1
4714         1
6528         1
6532         2
8404         2
8706         1
11156        1
11231        1
11609        1
12140        5
12711        1
14021        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-20">
Past versions of unnamed-chunk-23-20.png
</button>
</p>
<div id="fig-unnamed-chunk-23-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-20.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;VAPA&quot;
[1] &quot;18_7&quot;
        genename region_tag  susie_pip       mu2          PVE           z
1854        VAPA       18_7 0.02250502  5.532316 1.207805e-07 -0.50724638
1865     ANKRD12       18_7 0.02062323  4.734186 9.471361e-08  0.11574364
4309       TWSG1       18_7 0.02194395  5.301492 1.128557e-07  0.44247788
4865        NAPG       18_7 0.16226954 23.979882 3.774805e-06 -2.71301314
6927      PPP4R1       18_7 0.26857459 29.010443 7.558397e-06 -3.31948106
6928      APCDD1       18_7 0.04546274 11.988948 5.287460e-07 -1.30950974
8673       RAB31       18_7 0.02143633  5.087559 1.057962e-07 -0.42878143
8681       MTCL1       18_7 0.02088305  4.848590 9.822450e-08  0.25022734
9842      NDUFV2       18_7 0.06264118 14.958272 9.089746e-07  1.77821164
11845      RAB12       18_7 0.02018144  4.536357 8.881162e-08  0.05155737
13303 PPP4R1-AS1       18_7 0.02416486  6.183233 1.449474e-07  0.96153846
      num_eqtl
1854         1
1865         2
4309         1
4865         2
6927         2
6928         2
8673         2
8681         2
9842         2
11845        2
13303        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-21">
Past versions of unnamed-chunk-23-21.png
</button>
</p>
<div id="fig-unnamed-chunk-23-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-21.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;KPNB1&quot;
[1] &quot;17_27&quot;
          genename region_tag    susie_pip        mu2          PVE          z
45           CDC27      17_27 8.718574e-06   7.921742 6.700027e-11  1.3689320
889          TBX21      17_27 4.779319e-06   6.009998 2.786447e-11  0.6217949
891            NSF      17_27 4.482510e-06 107.169511 4.660182e-10  0.3333333
2507          WNT3      17_27 3.707228e-05 797.458403 2.867925e-08 -4.3894024
2515         KPNB1      17_27 4.505073e-06   5.881217 2.570274e-11  0.4344828
2516         GOSR2      17_27 4.092394e-06  26.523101 1.052961e-10 -0.4405248
3637        KANSL1      17_27 1.284852e-05  26.280094 3.275598e-10  1.2652096
5155          NMT1      17_27 5.146064e-06  34.645366 1.729541e-10 -0.2835821
7240         WNT9B      17_27 6.962174e-06  23.610209 1.594612e-10 -1.4179104
7276      ARHGAP27      17_27 5.875984e-06 189.442021 1.079860e-09  0.1576482
7497         PLCD3      17_27 4.090509e-06  16.256175 6.450690e-11 -0.3418803
9304         DCAKD      17_27 4.151521e-06  46.023255 1.853510e-10 -0.7214662
9905       EFCAB13      17_27 5.098684e-06   6.464700 3.197547e-11 -0.7481167
10133        ACBD4      17_27 1.493108e-05  36.452467 5.279935e-10  2.0149254
10488        FMNL1      17_27 4.442251e-06  50.069692 2.157687e-10  0.9729730
10616       ARL17A      17_27 7.713881e-06 488.685319 3.656896e-09 -0.5399498
10733       HEXIM1      17_27 2.623803e-05  51.799366 1.318457e-09 -2.4155844
10739         MAPT      17_27 2.791224e-04 166.512662 4.508711e-08  3.8724621
11409         MYL4      17_27 2.747494e-05  11.509995 3.067766e-10 -2.1660197
11555       TBKBP1      17_27 4.296711e-05  33.693899 1.404423e-09  2.3676471
12286       ARL17B      17_27 1.618098e-05 117.096052 1.838050e-09  2.5357143
13108        ITGB3      17_27 1.319177e-05  10.880709 1.392421e-10 -1.5791058
13371 RP11-798G7.6      17_27 9.150494e-06 115.994088 1.029653e-09  2.1204819
13879   AC142472.6      17_27 4.310133e-05  48.132994 2.012538e-09  2.6212121
      num_eqtl
45           1
889          1
891          1
2507         2
2515         1
2516         2
3637         3
5155         1
7240         1
7276         2
7497         1
9304         2
9905         3
10133        1
10488        1
10616        2
10733        1
10739        2
11409        2
11555        1
12286        1
13108        4
13371        1
13879        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-22.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-22">
Past versions of unnamed-chunk-23-22.png
</button>
</p>
<div id="fig-unnamed-chunk-23-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-22.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOA4&quot;
[1] &quot;11_70&quot;
      genename region_tag  susie_pip       mu2          PVE           z
2690     APOA4      11_70 0.08597876 10.093678 8.418816e-07 -0.37037037
2691    CEP164      11_70 0.09111581 11.335944 1.001987e-06  1.44776119
3461     APOA1      11_70 0.06263300  7.448341 4.525569e-07  0.16909746
5336     FXYD2      11_70 0.05788562  7.468529 4.193882e-07  0.86178768
6527     TAGLN      11_70 0.04582118  9.456397 4.203416e-07  2.39185905
7399      SIK3      11_70 0.09457604 10.724944 9.839808e-07  0.86597938
7404     PCSK7      11_70 0.09065674 16.195458 1.424308e-06 -3.41666667
8458    RNF214      11_70 0.04229603  4.536722 1.861453e-07  0.01470588
8620  PAFAH1B2      11_70 0.76294098 21.209914 1.569785e-05  4.16485221
9733   DSCAML1      11_70 0.04229107  4.532488 1.859498e-07  0.05504602
10677    BACE1      11_70 0.06261943  8.003026 4.861539e-07  1.08724832
      num_eqtl
2690         1
2691         1
3461         3
5336         2
6527         2
7399         1
7404         1
8458         1
8620         2
9733         3
10677        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-23.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-23">
Past versions of unnamed-chunk-23-23.png
</button>
</p>
<div id="fig-unnamed-chunk-23-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-23.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ALDH2&quot;
[1] &quot;12_67&quot;
          genename region_tag  susie_pip       mu2          PVE          z
1296      MAPKAPK5      12_67 0.08012151 11.945585 9.284680e-07 -1.1733333
1317         ERP29      12_67 0.03855432  5.176368 1.936015e-07 -0.6422018
2770         ARPC3      12_67 0.03582291  4.673904 1.624243e-07 -0.1810345
2771          GPN3      12_67 0.03614353  4.760320 1.669080e-07 -0.2307692
2772         VPS29      12_67 0.03751501  8.210948 2.988194e-07 -1.9631441
2773          MYL2      12_67 0.13743467 17.251484 2.300028e-06 -1.9552239
2775         SH2B3      12_67 0.09091118 12.677368 1.118039e-06  1.7037037
2780        ACAD10      12_67 0.07176932 10.502546 7.312129e-07  1.5186529
2781         ALDH2      12_67 0.03922845  5.349209 2.035641e-07 -0.7294118
2784         NAA25      12_67 0.03553311  4.621570 1.593064e-07  0.2922582
3851         IFT81      12_67 0.04218807  8.332797 3.410287e-07  2.0042599
5575          GIT2      12_67 0.70103238 22.545466 1.533231e-05 -4.5737419
5576          TCHP      12_67 0.18384103 18.805427 3.353792e-06 -4.1063830
6611         RAD9B      12_67 0.06529046 13.668112 8.657025e-07  2.6250000
9310        HECTD4      12_67 0.14007693 16.328968 2.218890e-06 -2.0573653
9463      C12orf76      12_67 0.18931921 21.116844 3.878235e-06  2.6123077
10674       PPP1CC      12_67 0.03640971  4.784495 1.689910e-07 -0.3119266
11102       ANAPC7      12_67 0.03609400  4.717581 1.651828e-07 -0.2200000
11158        PPTC7      12_67 0.03689056  4.955877 1.773561e-07  0.3162393
11403      TMEM116      12_67 0.04026398  5.515291 2.154247e-07  0.7459176
11408      FAM109A      12_67 0.03821478  5.682622 2.106641e-07  0.6309694
11730        ATXN2      12_67 0.03999521  5.624313 2.182167e-07  0.6532258
12444 MAPKAPK5-AS1      12_67 0.03971005  5.457536 2.102362e-07  0.7674419
13052  RP3-473L9.4      12_67 0.03646776  4.764151 1.685408e-07  0.2820513
      num_eqtl
1296         1
1317         1
2770         1
2771         1
2772         2
2773         1
2775         1
2780         2
2781         1
2784         2
3851         2
5575         2
5576         1
6611         1
9310         2
9463         1
10674        1
11102        1
11158        1
11403        3
11408        2
11730        1
12444        1
13052        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-24.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-24">
Past versions of unnamed-chunk-23-24.png
</button>
</p>
<div id="fig-unnamed-chunk-23-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-24.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOA1&quot;
[1] &quot;11_70&quot;
      genename region_tag  susie_pip       mu2          PVE           z
2690     APOA4      11_70 0.08597876 10.093678 8.418816e-07 -0.37037037
2691    CEP164      11_70 0.09111581 11.335944 1.001987e-06  1.44776119
3461     APOA1      11_70 0.06263300  7.448341 4.525569e-07  0.16909746
5336     FXYD2      11_70 0.05788562  7.468529 4.193882e-07  0.86178768
6527     TAGLN      11_70 0.04582118  9.456397 4.203416e-07  2.39185905
7399      SIK3      11_70 0.09457604 10.724944 9.839808e-07  0.86597938
7404     PCSK7      11_70 0.09065674 16.195458 1.424308e-06 -3.41666667
8458    RNF214      11_70 0.04229603  4.536722 1.861453e-07  0.01470588
8620  PAFAH1B2      11_70 0.76294098 21.209914 1.569785e-05  4.16485221
9733   DSCAML1      11_70 0.04229107  4.532488 1.859498e-07  0.05504602
10677    BACE1      11_70 0.06261943  8.003026 4.861539e-07  1.08724832
      num_eqtl
2690         1
2691         1
3461         3
5336         2
6527         2
7399         1
7404         1
8458         1
8620         2
9733         3
10677        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-25.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-25">
Past versions of unnamed-chunk-23-25.png
</button>
</p>
<div id="fig-unnamed-chunk-23-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-25.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;NPC2&quot;
[1] &quot;14_34&quot;
           genename region_tag  susie_pip       mu2          PVE          z
1073          PSEN1      14_34 0.21393548 32.401216 6.724416e-06  4.2307692
1727          PAPLN      14_34 0.02452860  9.765600 2.323711e-07 -1.2104233
3570          DCAF4      14_34 0.01385379 10.020297 1.346665e-07  3.0495367
3571          PROX2      14_34 0.02186940  9.092144 1.928918e-07  1.2272727
3574          BBOF1      14_34 0.01815829  7.624654 1.343091e-07 -1.3641061
3575           NEK9      14_34 0.01663519  6.192502 9.993193e-08  0.5909091
3576          ACYP1      14_34 0.01654069  6.142174 9.855672e-08 -0.5757576
3577          IFT43      14_34 0.03605355 13.283572 4.645937e-07 -1.4666667
3578           NPC2      14_34 0.01468276  5.106894 7.274027e-08  0.3777174
3581          ACOT2      14_34 0.03507351 16.030096 5.454134e-07 -2.5454490
3582          LTBP2      14_34 0.01402540  4.715639 6.416026e-08 -0.2238095
3584           MLH3      14_34 0.01515096  5.401685 7.939254e-08  0.4878049
3586         FLVCR2      14_34 0.01393610  4.662053 6.302732e-08 -0.1908397
3587          ABCD4      14_34 0.01637664  6.476112 1.028844e-07 -1.0000000
3588           DLST      14_34 0.03450396 13.063234 4.372502e-07  1.5571429
3596         EIF2B2      14_34 0.01544762  5.536585 8.296865e-08  0.4477612
3597           COQ6      14_34 0.01522450  5.397768 7.972008e-08  0.4558824
3598         ZNF410      14_34 0.02141551  8.312245 1.726860e-07 -0.9333063
4815        C14orf1      14_34 0.04197844 14.622577 5.954710e-07 -1.6000000
5639          PTGR2      14_34 0.01457521  5.222525 7.384237e-08 -0.7412556
8206         ZFYVE1      14_34 0.03286027 11.810349 3.764821e-07 -1.1335878
9705          PNMA1      14_34 0.01376090  4.595818 6.135078e-08 -0.1502769
9763          ACOT4      14_34 0.01823760  6.751362 1.194454e-07  0.6428571
10421         ACOT1      14_34 0.01379707  4.685384 6.271085e-08  0.3316324
10758        ENTPD5      14_34 0.03987186 14.269894 5.519473e-07 -1.6660365
10760        HEATR4      14_34 0.02479427  8.888447 2.137902e-07  0.3698630
11394       RPS6KL1      14_34 0.01390024  4.615402 6.223608e-08  0.0620155
11818          DPF3      14_34 0.95743121 33.355115 3.097993e-05  6.2649595
13091 RP11-270M14.5      14_34 0.01394386  4.652603 6.293458e-08  0.1590909
13097 CTD-2207P18.2      14_34 0.01389708  4.648876 6.267323e-08  0.2686567
13101   RP3-449M8.6      14_34 0.01900602  7.539356 1.390067e-07  1.0179851
13131     LINC01220      14_34 0.01535249  5.598671 8.338236e-08 -0.5555556
13472   RP3-449M8.9      14_34 0.01385642  4.597187 6.179506e-08  0.2204239
      num_eqtl
1073         1
1727         2
3570         2
3571         1
3574         2
3575         1
3576         1
3577         1
3578         1
3581         3
3582         1
3584         1
3586         1
3587         1
3588         1
3596         1
3597         1
3598         2
4815         1
5639         4
8206         1
9705         3
9763         1
10421        3
10758        3
10760        1
11394        1
11818        3
13091        1
13097        1
13101        2
13131        1
13472        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-26.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-26">
Past versions of unnamed-chunk-23-26.png
</button>
</p>
<div id="fig-unnamed-chunk-23-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-26.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;VAPB&quot;
[1] &quot;20_34&quot;
          genename region_tag  susie_pip       mu2          PVE          z
1251       PHACTR3      20_34 0.04769367  5.922148 2.739999e-07  0.6041667
1785        NELFCD      20_34 0.04117991  4.567122 1.824477e-07 -0.3370275
1786          CTSZ      20_34 0.04268982  4.899060 2.028839e-07  0.3513970
3935          VAPB      20_34 0.04166560  4.675176 1.889670e-07  0.1826668
3941        ZNF831      20_34 0.08074517 10.818934 8.474448e-07  1.5252525
3943        RAB22A      20_34 0.05066719  6.481360 3.185689e-07 -0.7815028
11494      APCDD1L      20_34 0.10370004 13.175833 1.325462e-06  1.5859808
12054       NPEPL1      20_34 0.06110568  8.218286 4.871618e-07 -1.0426678
13435 RP4-806M20.4      20_34 0.11006771 13.741049 1.467203e-06 -1.6701031
      num_eqtl
1251         1
1785         2
1786         2
3935         3
3941         1
3943         2
11494        2
12054        2
13435        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-27.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-27">
Past versions of unnamed-chunk-23-27.png
</button>
</p>
<div id="fig-unnamed-chunk-23-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-27.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOC1&quot;
[1] &quot;19_31&quot;
      genename region_tag  susie_pip       mu2          PVE           z
204      PLAUR      19_31 0.02849177  4.808054 1.328921e-07  0.31958763
867      XRCC1      19_31 0.02829552  4.744703 1.302378e-07  0.25581395
2099     KCNN4      19_31 0.04351501  8.701255 3.673089e-07 -1.06930693
2266     ETHE1      19_31 0.03414140  6.468101 2.142242e-07  0.68748771
2267      SMG9      19_31 0.03758500  7.351636 2.680458e-07 -0.89043174
3971     ZNF45      19_31 0.05900086 11.518291 6.592601e-07 -1.37024221
4540    ZNF227      19_31 0.03640418  7.057995 2.492545e-07 -0.91280315
7319    ZNF230      19_31 0.03395243  6.417107 2.113589e-07 -0.73630137
7323    ZNF221      19_31 0.03974818  7.866755 3.033355e-07 -0.95714286
8475      IRGQ      19_31 0.02777114  4.573267 1.232057e-07  0.11217949
8476    ZNF226      19_31 0.02765363  4.534412 1.216420e-07  0.06122449
8519    ZNF283      19_31 0.03058882  5.459326 1.619989e-07 -0.46268657
9651    ZNF404      19_31 0.03087040  5.543399 1.660079e-07 -0.46268657
9865    ZNF223      19_31 0.03028714  5.368403 1.577298e-07  0.44285714
10644   ZNF284      19_31 0.02770775  4.552327 1.223616e-07  0.10101010
11740   ZNF155      19_31 0.02962992  5.167181 1.485233e-07  0.47048028
12434   PINLYP      19_31 0.02765847  4.536014 1.217063e-07 -0.06795808
13018   ZNF225      19_31 0.04727306  9.465932 4.340977e-07 -1.17391304
13291   ZNF234      19_31 0.02770092  4.550066 1.222707e-07  0.11594203
13411   ZNF224      19_31 0.03783347  7.412266 2.720430e-07  0.89772727
112      MARK4      19_31 0.03067690  5.485707 1.632505e-07 -0.47282609
119   TRAPPC6A      19_31 0.06385391 12.253005 7.589978e-07 -1.43410853
214      ERCC1      19_31 0.06209460 11.993145 7.224326e-07  1.43750000
593     ZNF112      19_31 0.02823966  4.726588 1.294844e-07  0.15555945
866        PVR      19_31 0.02786954  4.605680 1.245186e-07 -0.16666667
2111    CLPTM1      19_31 0.04952683  9.896424 4.754767e-07  1.22413793
2113   PPP1R37      19_31 0.02764689  4.532178 1.215524e-07  0.02325315
2115       CKM      19_31 0.02947524  5.119184 1.463756e-07  0.42066734
2117  PPP1R13L      19_31 0.03936284  7.777046 2.969693e-07  0.94933333
3451    CD3EAP      19_31 0.02904211  4.983442 1.404003e-07  0.38478177
4079      FOSB      19_31 0.02784195  4.596603 1.241501e-07  0.11538462
4080      OPA3      19_31 0.04559245  9.131621 4.038790e-07  1.09859155
4082      RTN2      19_31 0.03633139  7.039587 2.481074e-07  0.82935506
4412   NECTIN2      19_31 0.03594458  6.941161 2.420338e-07  0.91447368
4413      APOE      19_31 0.06880641 12.948792 8.643080e-07  1.75000000
4414    TOMM40      19_31 0.03077329  5.514488 1.646226e-07 -0.59268930
4415     APOC1      19_31 0.16958180 21.526348 3.541278e-06  2.46376812
5860    GEMIN7      19_31 0.04929982  9.853941 4.712656e-07 -1.23208778
7324    ZNF233      19_31 0.14629720 20.091293 2.851375e-06  2.44329897
7325    ZNF235      19_31 0.06927352 13.011865 8.744143e-07  1.88899386
8477    ZNF180      19_31 0.03262520  6.050867 1.915055e-07  0.70149254
8993    ZNF296      19_31 0.02805998  4.668090 1.270682e-07  0.19444444
10708 CEACAM19      19_31 0.03280796  6.102161 1.942108e-07  0.66278513
10978  BLOC1S3      19_31 0.02976706  5.209531 1.504337e-07 -0.41492537
11951    PPM1N      19_31 0.02966612  5.178378 1.490270e-07 -0.46241261
12455    APOC2      19_31 0.03222310  5.937003 1.855859e-07  0.52968037
13399   ZNF285      19_31 0.02878369  4.901500 1.368630e-07 -0.35172414
      num_eqtl
204          1
867          1
2099         1
2266         2
2267         2
3971         1
4540         2
7319         1
7323         1
8475         1
8476         1
8519         1
9651         1
9865         1
10644        1
11740        2
12434        3
13018        1
13291        1
13411        1
112          1
119          1
214          1
593          2
866          1
2111         1
2113         2
2115         2
2117         1
3451         2
4079         1
4080         1
4082         2
4412         1
4413         1
4414         1
4415         1
5860         3
7324         1
7325         2
8477         1
8993         1
10708        2
10978        1
11951        2
12455        1
13399        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-28.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-28">
Past versions of unnamed-chunk-23-28.png
</button>
</p>
<div id="fig-unnamed-chunk-23-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-28.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;LPIN3&quot;
[1] &quot;20_25&quot;
      genename region_tag  susie_pip      mu2          PVE          z num_eqtl
3937      CHD6      20_25 0.04149349 4.887035 1.967142e-07 -0.4739866        2
3938     PLCG1      20_25 0.04372960 5.371009 2.278462e-07  0.5237893        2
4710     LPIN3      20_25 0.05168602 6.915591 3.467471e-07  0.8249187        2
10378  EMILIN3      20_25 0.05127277 6.841276 3.402783e-07  1.0852760        2
11542     TOP1      20_25 0.04073443 4.716900 1.863926e-07 -0.2518045        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-29.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-29">
Past versions of unnamed-chunk-23-29.png
</button>
</p>
<div id="fig-unnamed-chunk-23-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-29.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;SORT1&quot;
[1] &quot;1_69&quot;
          genename region_tag   susie_pip       mu2          PVE           z
4854          VAV3       1_69 0.007360794  5.792493 4.136191e-08 -0.61577136
7607         NTNG1       1_69 0.013475947 11.242393 1.469699e-07 -1.38461538
11538        PRMT6       1_69 0.051956490 23.381155 1.178464e-06 -2.41666667
344           SARS       1_69 0.006986679  5.334827 3.615776e-08  0.44800000
1179         WDR47       1_69 0.007170521  5.573058 3.876633e-08  0.50000000
1182      SLC25A24       1_69 0.006787076  5.068679 3.337244e-08  0.38461538
3298        STXBP3       1_69 0.006407795  4.551223 2.829092e-08  0.05000000
3773         CLCC1       1_69 0.010017917  8.601256 8.358911e-08 -1.01639344
3774         GPSM2       1_69 0.006485344  4.661177 2.932507e-08  0.16239316
4849         GSTM1       1_69 0.006633141  4.864554 3.130204e-08  0.28399653
4850       PRPF38B       1_69 0.010568522  9.064166 9.292928e-08 -1.11560694
4852         GSTM5       1_69 0.009068114  7.679306 6.755374e-08 -0.93675660
4853         GSTM3       1_69 0.006394558  4.531569 2.811057e-08  0.02985075
4855         PSRC1       1_69 0.006462700  4.633937 2.905190e-08 -0.08557694
4856         SORT1       1_69 0.006451700  4.612983 2.887131e-08 -0.13513514
5918         SYPL2       1_69 0.008165993  6.732360 5.333187e-08 -0.78993535
5923         PSMA5       1_69 0.008126179  6.706339 5.286671e-08  0.73457603
7609        HENMT1       1_69 0.018928170 14.363043 2.637336e-07 -1.60447761
8710         GSTM4       1_69 0.006964859  5.299935 3.580909e-08  0.46882593
9435      CYB561D1       1_69 0.013011187 10.987508 1.386841e-07  1.25874126
9996      C1orf194       1_69 0.015938161 12.812202 1.980945e-07 -1.45832274
10150       AMIGO1       1_69 0.006809047  5.089012 3.361478e-08 -0.44117647
11311        TAF13       1_69 0.007127584  5.512816 3.811766e-08  0.50839722
11907        GSTM2       1_69 0.011424171  9.805167 1.086651e-07 -1.13618389
12058     TMEM167B       1_69 0.008933496  7.560141 6.551817e-08 -0.88157895
12108       MYBPHL       1_69 0.006414246  4.562159 2.838745e-08  0.03282884
12227 RP11-356N1.2       1_69 0.010409893  8.942052 9.030127e-08  1.07246377
4861         CEPT1       1_69 0.016470252 13.065713 2.087583e-07 -1.54255319
5922        STRIP1       1_69 0.006749723  5.023231 3.289119e-08  0.34317369
5925         PROK1       1_69 0.010705486  9.208717 9.563480e-08 -1.08002080
7017         DRAM2       1_69 0.009975593  8.525865 8.250639e-08 -1.08219178
7639       DENND2D       1_69 0.009558310  8.147190 7.554390e-08 -1.00808455
8701       SLC16A4       1_69 0.007391127  5.844495 4.190521e-08 -0.57575758
8704        AHCYL1       1_69 0.006941342  5.278333 3.554272e-08 -0.42045455
9747         KCNA2       1_69 0.088489011 28.369070 2.435257e-06 -2.62500000
10435         CSF1       1_69 0.007023761  5.382626 3.667536e-08 -0.46236559
11198      SLC6A17       1_69 0.009108989  7.738553 6.838177e-08 -0.90243902
13139 RP11-284N8.3       1_69 0.035959924 20.121611 7.019270e-07 -2.11254860
123           ST7L       1_69 0.009530354  8.119170 7.506389e-08  0.92424242
1181         OVGP1       1_69 0.010417281  8.707643 8.799650e-08  0.86567164
3304         WDR77       1_69 0.016278415 12.636548 1.995497e-07 -1.25373134
3305        ATP5F1       1_69 0.013148012 10.630536 1.355894e-07 -1.02702703
3306         RAP1A       1_69 0.006394824  4.563040 2.830696e-08  0.06410256
3308        CAPZA1       1_69 0.009530771  8.119460 7.506987e-08  0.92424242
3772        TMIGD3       1_69 0.011619766 10.037766 1.131475e-07  1.20588235
5924      C1orf162       1_69 0.006997775  5.365705 3.642480e-08  0.45238095
6959         MOV10       1_69 0.011760283  9.969968 1.137423e-07 -1.19389720
6960          RHOC       1_69 0.017219897 13.435675 2.244401e-07  1.51851852
6961         PPM1J       1_69 0.006395630  4.543890 2.819172e-08 -0.10000000
9086         KCND3       1_69 0.009322585  8.111148 7.335489e-08 -1.18238994
9416          PIFO       1_69 0.010801681  8.982942 9.412833e-08 -0.91082852
11320      FAM212B       1_69 0.006426704  5.149213 3.210256e-08  0.86764706
12341  RP5-965F6.2       1_69 0.006442001  4.589414 2.868062e-08  0.07352941
12345    LINC01160       1_69 0.010806169  8.727229 9.148683e-08  0.79310345
12348    LINC01750       1_69 0.007941604  6.110965 4.707913e-08  0.06666667
      num_eqtl
4854         2
7607         1
11538        1
344          1
1179         1
1182         1
3298         1
3773         1
3774         1
4849         3
4850         1
4852         2
4853         1
4855         2
4856         1
5918         2
5923         2
7609         1
8710         2
9435         1
9996         2
10150        1
11311        2
11907        4
12058        1
12108        3
12227        1
4861         1
5922         2
5925         2
7017         1
7639         2
8701         1
8704         1
9747         1
10435        1
11198        1
13139        2
123          1
1181         1
3304         1
3305         1
3306         1
3308         1
3772         1
5924         1
6959         2
6960         1
6961         1
9086         1
9416         2
11320        1
12341        1
12345        1
12348        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-30.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-30">
Past versions of unnamed-chunk-23-30.png
</button>
</p>
<div id="fig-unnamed-chunk-23-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-30.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;FADS2&quot;
[1] &quot;11_34&quot;
           genename region_tag  susie_pip       mu2          PVE           z
2664           DTX4      11_34 0.03099303  6.991372 2.102020e-07  0.85897436
2679         CCDC86      11_34 0.02551854  5.210056 1.289759e-07  0.37220667
2680         PRPF19      11_34 0.02554661  5.220123 1.293673e-07  0.44117647
2681        TMEM109      11_34 0.02481824  4.955340 1.193040e-07  0.35294118
2708            CD5      11_34 0.03596171  8.357137 2.915468e-07  1.04347826
4016           MYRF      11_34 0.03186538  7.246121 2.239934e-07 -0.84466019
4927          DAGLA      11_34 0.02480277  4.949630 1.190922e-07 -0.34650349
4932          FADS2      11_34 0.07656635 15.358491 1.140767e-06  1.99325920
4933        TMEM258      11_34 0.04188552  9.760925 3.966114e-07  1.33463162
4934           TCN1      11_34 0.05049878 11.487781 5.627655e-07  1.30687831
6505           TKFC      11_34 0.08736548 16.595898 1.406537e-06 -1.88215488
6507        TMEM138      11_34 0.08387976 16.213411 1.319295e-06  1.96202532
6511         INCENP      11_34 0.16897569 22.902963 3.754277e-06  2.45901639
6512            ZP1      11_34 0.03411467  7.872500 2.605339e-07 -0.91980700
6515          MS4A2      11_34 0.02457572  4.865466 1.159955e-07 -0.33823529
7536       CYB561A3      11_34 0.08387976 16.213411 1.319295e-06  1.96202532
7537        PPP1R32      11_34 0.02769822  5.960772 1.601640e-07  0.60736196
8379        FAM111A      11_34 0.02482961  4.959529 1.194595e-07  0.37903226
8398          PATL1      11_34 0.02477812  4.940530 1.187551e-07 -0.29903537
8400           STX3      11_34 0.02380707  4.574707 1.056525e-07  0.02186908
8595         VPS37C      11_34 0.02512710  5.068543 1.235481e-07 -0.33636364
8597          BEST1      11_34 0.04076498  9.511029 3.761189e-07  1.25321316
8598           FTH1      11_34 0.05332834 11.992459 6.204071e-07  1.65217391
8680           FEN1      11_34 0.08312665 16.128761 1.300624e-06  2.04225352
10297        PTGDR2      11_34 0.02750529  5.896723 1.573393e-07  0.68811881
10755       TMEM216      11_34 0.03799363  8.862739 3.266549e-07 -1.13978495
10971       FAM111B      11_34 0.03225465  7.357582 2.302172e-07 -0.85853659
11292         MPEG1      11_34 0.03351433  7.709398 2.506464e-07 -0.96609070
12022        MS4A4E      11_34 0.03198641  7.280922 2.259240e-07  0.97058824
12736    AP001258.4      11_34 0.02549331  5.201000 1.286244e-07 -0.38235294
12965    AP000442.4      11_34 0.04194449  9.773896 3.976977e-07 -1.14210526
13022 RP11-794G24.1      11_34 0.02916128  6.432586 1.819712e-07 -0.70680628
13026 RP11-286N22.8      11_34 0.04123878  9.617506 3.847500e-07 -1.22580645
      num_eqtl
2664         1
2679         2
2680         1
2681         1
2708         1
4016         1
4927         2
4932         2
4933         2
4934         1
6505         1
6507         1
6511         1
6512         2
6515         1
7536         1
7537         1
8379         1
8398         1
8400         2
8595         1
8597         2
8598         1
8680         1
10297        1
10755        1
10971        1
11292        2
12022        1
12736        1
12965        1
13022        1
13026        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-31.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-31">
Past versions of unnamed-chunk-23-31.png
</button>
</p>
<div id="fig-unnamed-chunk-23-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-31.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;CD36&quot;
[1] &quot;7_51&quot;
     genename region_tag  susie_pip      mu2          PVE          z num_eqtl
925    SEMA3C       7_51 0.04540722 4.669443 2.056840e-07 -0.3409091        1
4984     CD36       7_51 0.05163038 5.756439 2.883166e-07  0.8130841        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-32.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-32">
Past versions of unnamed-chunk-23-32.png
</button>
</p>
<div id="fig-unnamed-chunk-23-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-32.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;CYP27A1&quot;
[1] &quot;2_129&quot;
         genename region_tag  susie_pip       mu2          PVE          z
256       SLC11A1      2_129 0.02818779  5.128755 1.402437e-07 -0.3776224
507         PTPRN      2_129 0.40887978 31.055978 1.231831e-05 -3.6739927
838         ASIC4      2_129 0.04403253  9.229360 3.942354e-07  1.3373494
839          SPEG      2_129 0.07029966 13.568034 9.252957e-07 -1.5628415
905         BCS1L      2_129 0.11201778 17.952033 1.950792e-06  2.5752688
1042         TNS1      2_129 0.30024256 27.707559 8.070138e-06  2.9756098
3208        PLCD4      2_129 0.11181367 17.934589 1.945346e-06  2.5089459
3210       ZNF142      2_129 0.11293908 18.029919 1.975370e-06  2.5806452
3221       CNPPD1      2_129 0.05823445 11.816018 6.675158e-07  1.8364237
3223        ABCB6      2_129 0.02673374  4.643488 1.204244e-07  0.2678571
3916        DNPEP      2_129 0.03051430  5.855960 1.733453e-07  0.5665584
3917         INHA      2_129 0.03262286  6.469243 2.047321e-07 -0.7761194
3919        OBSL1      2_129 0.04426631  9.278222 3.984267e-07 -1.2058824
4236       TUBA4A      2_129 0.02682524  4.674795 1.216513e-07 -0.3905109
4237         AAMP      2_129 0.04191628  8.775105 3.568169e-07  1.0000000
4238         PNKD      2_129 0.04191628  8.775105 3.568169e-07  1.0000000
5082        TTLL4      2_129 0.02996833  5.690352 1.654292e-07 -0.5362319
5083        USP37      2_129 0.07467321 14.131302 1.023664e-06 -2.1171493
5087       TMBIM1      2_129 0.04591592  9.615892 4.283150e-07  1.1327420
5088      CYP27A1      2_129 0.03026815  5.781614 1.697639e-07  1.0593803
6113        CNOT9      2_129 0.03060940  5.884493 1.747328e-07  0.9891156
6115        GMPPA      2_129 0.05084309 10.558123 5.207498e-07  1.2318841
7205      ZFAND2B      2_129 0.03664767  7.538405 2.680009e-07 -1.5142857
7750        CXCR1      2_129 0.02773211  4.979390 1.339583e-07  0.3382353
7751        ARPC2      2_129 0.02743903  4.882049 1.299515e-07  0.2049660
7756        RNF25      2_129 0.08636921 15.495056 1.298262e-06 -2.3920455
7760          IHH      2_129 0.03415260  6.890097 2.282756e-07 -0.8875000
7762       ANKZF1      2_129 0.03382441  6.801437 2.231729e-07  0.9333333
7765        GLB1L      2_129 0.15705396 21.197410 3.229551e-06 -2.7954545
9526          DES      2_129 0.03228830  6.374523 1.996656e-07 -0.8028169
10001      GPBAR1      2_129 0.02762010  4.942315 1.324238e-07  0.2603852
10076       CXCR2      2_129 0.02824113  5.146079 1.409837e-07  0.3409091
10817       NHEJ1      2_129 0.02913239  5.430957 1.534839e-07  0.8700191
10935     TMEM198      2_129 0.19590827 23.365013 4.440473e-06  2.6865672
11552       ATG9A      2_129 0.04992114 10.388856 5.031096e-07 -1.5736388
11952     SLC23A3      2_129 0.15922753 21.331066 3.294892e-06  2.8769191
12352       DIRC3      2_129 0.03927978  8.176621 3.115684e-07 -0.9728261
13648 RP11-33O4.1      2_129 0.02690161  4.700872 1.226781e-07  0.6661952
      num_eqtl
256          1
507          1
838          1
839          1
905          1
1042         1
3208         2
3210         1
3221         2
3223         1
3916         2
3917         1
3919         1
4236         1
4237         1
4238         1
5082         1
5083         2
5087         2
5088         2
6113         2
6115         1
7205         1
7750         1
7751         2
7756         1
7760         1
7762         1
7765         1
9526         1
10001        2
10076        1
10817        2
10935        1
11552        2
11952        2
12352        1
13648        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-33.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-33">
Past versions of unnamed-chunk-23-33.png
</button>
</p>
<div id="fig-unnamed-chunk-23-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-33.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;NPC1&quot;
[1] &quot;18_12&quot;
           genename region_tag  susie_pip       mu2          PVE           z
1870          RIOK3      18_12 0.04551269  5.955563 2.629455e-07 -0.56481481
4896        TMEM241      18_12 0.11515422 14.634959 1.634865e-06  2.77611940
5782        OSBPL1A      18_12 0.04608661  6.071346 2.714377e-07 -0.67164179
5784        C18orf8      18_12 0.30565619 24.310650 7.208422e-06  3.57683496
5786           NPC1      18_12 0.15283418 17.351284 2.572542e-06  3.10871482
6853          CABYR      18_12 0.04808927  6.464307 3.015648e-07  0.70028818
6855        ANKRD29      18_12 0.04084802  4.958308 1.964785e-07  0.08415842
8636         TTC39C      18_12 0.04065429  4.914584 1.938222e-07 -0.27631579
13315     LINC01894      18_12 0.06742822  9.600360 6.279711e-07  1.20930233
13326     LINC01915      18_12 0.04587287  6.028430 2.682690e-07  0.68750000
13331 RP11-799B12.4      18_12 0.07023556  9.980795 6.800372e-07  1.27536232
13332 RP11-403A21.1      18_12 0.06089290  8.651258 5.110417e-07 -0.84375000
13348  RP11-449D8.5      18_12 0.06742822  9.600360 6.279711e-07  1.20930233
13729  RP11-621L6.3      18_12 0.05970007  8.467863 4.904097e-07 -1.06784079
      num_eqtl
1870         1
4896         1
5782         1
5784         3
5786         2
6853         1
6855         1
8636         1
13315        1
13326        1
13331        1
13332        1
13348        1
13729        3</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-34.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-34">
Past versions of unnamed-chunk-23-34.png
</button>
</p>
<div id="fig-unnamed-chunk-23-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-34.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;NCEH1&quot;
[1] &quot;3_106&quot;
     genename region_tag  susie_pip      mu2          PVE          z num_eqtl
6158    NCEH1      3_106 0.04148114 4.572903 1.840149e-07 0.09859155        1
8855    NLGN1      3_106 0.04435164 5.189937 2.232966e-07 0.40472815        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-35.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-35">
Past versions of unnamed-chunk-23-35.png
</button>
</p>
<div id="fig-unnamed-chunk-23-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-35.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;STAR&quot;
[1] &quot;8_34&quot;
           genename region_tag  susie_pip       mu2          PVE          z
269          ADGRA2       8_34 0.06354746  7.933602 4.890791e-07  1.0142857
1002          FGFR1       8_34 0.05699148  6.922049 3.826969e-07  1.0519481
1194          DDHD2       8_34 0.16865141 17.197954 2.813696e-06 -2.0444444
6349           STAR       8_34 0.04716142  5.169443 2.365054e-07 -0.3458647
6350          PROSC       8_34 0.19803812 18.778491 3.607613e-06 -2.1703057
6353          TACC1       8_34 0.04521276  4.779775 2.096423e-07  0.2803738
6356           NSD3       8_34 0.06498656  8.142002 5.132928e-07  1.0840336
8093          LETM2       8_34 0.08107619 10.208263 8.028892e-07  1.4736842
8695          ADAM9       8_34 0.08283038 10.409025 8.363924e-07 -1.2732558
8815          TM2D2       8_34 0.04505410  4.747326 2.074884e-07 -0.2027542
12893     LINC01605       8_34 0.05084619  5.864974 2.892910e-07 -0.5764706
13583 RP11-350N15.5       8_34 0.08860578 11.042216 9.491367e-07  1.5707641
      num_eqtl
269          1
1002         1
1194         1
6349         1
6350         1
6353         1
6356         1
8093         1
8695         1
8815         2
12893        1
13583        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-36.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-36">
Past versions of unnamed-chunk-23-36.png
</button>
</p>
<div id="fig-unnamed-chunk-23-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-36.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;LRPAP1&quot;
[1] &quot;4_4&quot;
        genename region_tag  susie_pip       mu2          PVE           z
1240       NOP14        4_4 0.04978278  4.530374 2.187881e-07  0.04285714
1241        ADD1        4_4 0.05001056  4.572595 2.218375e-07  0.18740108
2638      MFSD10        4_4 0.08232703  9.212764 7.357713e-07 -1.17237308
2641       HGFAC        4_4 0.05011092  4.591138 2.231840e-07 -0.16296339
4040        GRK4        4_4 0.04978278  4.530374 2.187881e-07  0.04285714
7315       RGS12        4_4 0.07504018  8.344860 6.074679e-07 -1.14954438
7875      LRPAP1        4_4 0.05036700  4.638291 2.266285e-07 -0.16304348
9622        DOK7        4_4 0.24873833 19.929428 4.808925e-06  2.35949040
10409     ADRA2C        4_4 0.05244587  5.012614 2.550269e-07 -0.41000000
10960    MSANTD1        4_4 0.05081558  4.720325 2.326908e-07  0.30000000
11252        HTT        4_4 0.05202175  4.937439 2.491708e-07 -0.31884058
12864 AC141928.1        4_4 0.13383145 13.819728 1.794189e-06 -1.88888889
      num_eqtl
1240         1
1241         2
2638         3
2641         2
4040         1
7315         2
7875         1
9622         2
10409        1
10960        1
11252        1
12864        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-37.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-37">
Past versions of unnamed-chunk-23-37.png
</button>
</p>
<div id="fig-unnamed-chunk-23-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-37.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;LIPC&quot;
[1] &quot;15_26&quot;
         genename region_tag  susie_pip       mu2          PVE           z
5343         SLTM      15_26 0.09283266 12.511027 1.126689e-06  2.07058824
5362       ADAM10      15_26 0.04403073  5.565549 2.377247e-07 -0.97769990
7117       RNF111      15_26 0.14666590 16.870906 2.400369e-06  2.41176471
7118       FAM81A      15_26 0.03958787  4.585133 1.760858e-07  0.19172932
8241         LIPC      15_26 0.05373014  7.405989 3.860215e-07  0.96987827
9173      LDHAL6B      15_26 0.04033631  4.757634 1.861648e-07 -0.18000000
12744 RP11-30K9.6      15_26 0.03956020  4.578692 1.757156e-07 -0.05298013
      num_eqtl
5343         1
5362         2
7117         1
7118         1
8241         3
9173         1
12744        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-38.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-38">
Past versions of unnamed-chunk-23-38.png
</button>
</p>
<div id="fig-unnamed-chunk-23-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-38.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ANGPTL4&quot;
[1] &quot;19_8&quot;
          genename region_tag  susie_pip       mu2          PVE            z
347         PNPLA6       19_8 0.01959833  4.533508 8.619141e-08  0.066666667
681         ELAVL1       19_8 0.10125862 19.724509 1.937531e-06 -2.192771084
976        CAMSAP3       19_8 0.01960340  4.535871 8.625865e-08 -0.018115359
978           XAB2       19_8 0.02081469  5.083575 1.026478e-07 -0.428571429
979         STXBP2       19_8 0.02115783  5.232992 1.074068e-07 -0.471428571
1362         CD209       19_8 0.08022301 17.532325 1.364423e-06 -1.897833916
1363         CERS4       19_8 0.01959274  4.530901 8.611724e-08 -0.014285714
1364        MCOLN1       19_8 0.02250985  5.799227 1.266348e-07 -0.594594595
2116      ARHGEF18       19_8 0.02268853  5.871529 1.292314e-07 -0.597222222
2118        PEX11G       19_8 0.03326015  9.376453 3.025333e-07  1.111137463
2144        SNAPC2       19_8 0.02056093  4.971497 9.916086e-08  0.481481481
2146        TIMM44       19_8 0.03367665  9.490760 3.100561e-07 -1.164609629
4074            C3       19_8 0.02914260  8.163690 2.307944e-07  0.958904110
4075        SH2D3A       19_8 0.01973486  4.596910 8.800562e-08  0.145525293
4076        TRIP10       19_8 0.02727433  7.556445 1.999319e-07  0.879629630
4077        GPR108       19_8 0.02173865  5.480518 1.155752e-07  0.488174923
4461        ZNF557       19_8 0.27538955 29.534361 7.890154e-06 -3.045178789
5871         EVI5L       19_8 0.02032747  4.867171 9.597771e-08  0.413707888
9033        LRRC8E       19_8 0.03511530  9.875152 3.363958e-07  1.222507877
9044          INSR       19_8 0.02270928  5.879887 1.295337e-07 -0.618631151
9497          PCP2       19_8 0.02066267  5.016596 1.005555e-07 -0.400000000
9877         CTXN1       19_8 0.03450845  9.714932 3.252188e-07  1.243750000
10099      TRAPPC5       19_8 0.01959164  4.530388 8.610267e-08  0.007936508
10234       CLEC4G       19_8 0.03115928  8.777440 2.653174e-07 -1.050050849
10280       MCEMP1       19_8 0.02490636  6.724756 1.624790e-07 -0.730263158
10306        PRR36       19_8 0.07391205 16.765541 1.202107e-06  1.983333333
10748        KANK3       19_8 0.02370047  6.270631 1.441712e-07 -0.609205878
11508       ZNF358       19_8 0.04965731 13.069327 6.295741e-07  1.478873239
13421 CTD-2325M2.1       19_8 0.03469176  9.763620 3.285849e-07 -1.212669683
1518        HNRNPM       19_8 0.03725889 10.420035 3.766253e-07  1.227678571
1519        MARCH2       19_8 0.02016181  4.792419 9.373349e-08  0.211379694
4760         PRAM1       19_8 0.04925542 12.994183 6.208882e-07  1.579512358
4762        ZNF414       19_8 0.04819059 12.792139 5.980202e-07  1.588888889
5862      ADAMTS10       19_8 0.07019384 16.283577 1.108815e-06 -1.829268293
5866         MYO1F       19_8 0.05061982 13.246908 6.504974e-07  1.616000000
8557       ANGPTL4       19_8 0.06458429 15.507418 9.715760e-07  1.808823529
8558         CD320       19_8 0.06840312 16.042554 1.064535e-06 -1.862381374
8561        ZNF558       19_8 0.02431398  6.504462 1.534186e-07  0.675308883
12420        RPS28       19_8 0.05413168 13.867957 7.282398e-07  1.592592593
13418       NDUFA7       19_8 0.04347266 11.840848 4.993550e-07  1.401346519
      num_eqtl
347          1
681          1
976          4
978          1
979          1
1362         2
1363         1
1364         1
2116         1
2118         2
2144         1
2146         2
4074         1
4075         2
4076         1
4077         2
4461         2
5871         2
9033         2
9044         2
9497         1
9877         1
10099        1
10234        2
10280        1
10306        1
10748        2
11508        1
13421        1
1518         1
1519         2
4760         2
4762         1
5862         1
5866         1
8557         1
8558         3
8561         3
12420        1
13418        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-39.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-39">
Past versions of unnamed-chunk-23-39.png
</button>
</p>
<div id="fig-unnamed-chunk-23-39" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-39.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;SOAT2&quot;
[1] &quot;12_33&quot;
            genename region_tag  susie_pip       mu2          PVE           z
600            EIF4B      12_33 0.03817402  5.376694 1.991103e-07 -0.36193441
1449            CBX5      12_33 0.32038709 25.855612 8.036006e-06 -2.88421053
2756           KRT18      12_33 0.07743131 11.933995 8.964227e-07 -1.45555556
2758            TNS2      12_33 0.03514490  4.616299 1.573862e-07 -0.02479339
3884           SMUG1      12_33 0.03693231  5.072471 1.817341e-07 -0.52941176
5022           ESPL1      12_33 0.06304153 10.016040 6.125383e-07  1.66869034
5588           GPR84      12_33 0.04280238  6.431030 2.670293e-07 -0.49253731
5594         MAP3K12      12_33 0.41148394 28.593077 1.141364e-05 -3.72857143
5597            CSAD      12_33 0.04082426  5.994798 2.374123e-07 -0.51615347
5603          ZNF740      12_33 0.04755341  7.402510 3.414845e-07  0.75757576
8554            KRT1      12_33 0.03565415  4.748568 1.642416e-07 -0.34883721
8559          SPRYD3      12_33 0.04756694  7.405195 3.417056e-07  0.81437126
8560           SOAT2      12_33 0.11036988 15.274751 1.635442e-06  1.80000000
8950           KRT78      12_33 0.14404157 17.823332 2.490504e-06  2.19672131
8963            KRT4      12_33 0.03796041  5.325081 1.960955e-07 -0.53191489
8989            ATF7      12_33 0.11008551 15.250319 1.628619e-06  2.28235294
9278           HOXC5      12_33 0.07150595 11.189819 7.762036e-07 -1.52174831
10584            SP1      12_33 0.05406712  8.590487 4.505691e-07 -1.39534884
11410          HOXC4      12_33 0.04343837  6.566996 2.767265e-07  0.75401070
11790          PRR13      12_33 0.06770936 10.681133 7.015788e-07  1.63953488
12777       FLJ12825      12_33 0.03564947  4.747353 1.641780e-07  0.41791045
12855  RP11-834C11.4      12_33 0.03564947  4.747353 1.641780e-07  0.41791045
13181 RP11-834C11.11      12_33 0.05044089  7.947453 3.888850e-07  1.07894737
13183          CISTR      12_33 0.04794054  7.477562 3.477550e-07 -1.08403361
      num_eqtl
600          2
1449         1
2756         1
2758         1
3884         1
5022         2
5588         1
5594         1
5597         2
5603         1
8554         1
8559         1
8560         1
8950         1
8963         1
8989         1
9278         2
10584        1
11410        1
11790        1
12777        1
12855        1
13181        1
13183        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-40.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-40">
Past versions of unnamed-chunk-23-40.png
</button>
</p>
<div id="fig-unnamed-chunk-23-40" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-40.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;TNKS&quot;
[1] &quot;8_12&quot;
     genename region_tag  susie_pip      mu2          PVE          z num_eqtl
9342     TNKS       8_12 0.04205266 5.937176 2.422054e-07 -0.6212121        1</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-41.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-41">
Past versions of unnamed-chunk-23-41.png
</button>
</p>
<div id="fig-unnamed-chunk-23-41" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-41.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;ADH1B&quot;
[1] &quot;4_66&quot;
           genename region_tag  susie_pip       mu2          PVE           z
5518           MTTP       4_66 0.07672006  7.882237 5.866361e-07 -1.04046243
6184        TRMT10A       4_66 0.05371170  4.563792 2.377963e-07  0.02083333
6617          EIF4E       4_66 0.11727722 11.886921 1.352364e-06 -1.81333333
8714         TSPAN5       4_66 0.06040562  5.653403 3.312819e-07  0.44252874
9300           ADH6       4_66 0.17858509 15.949040 2.763059e-06 -2.40716823
11121         ADH1B       4_66 0.08911404  9.287286 8.028703e-07  1.29411765
11329          ADH5       4_66 0.23991541 18.891767 4.396845e-06  2.65454545
12774         ADH1C       4_66 0.08529751  8.875806 7.344370e-07  1.23802332
12794 RP11-766F14.2       4_66 0.06015217  5.614326 3.276117e-07 -0.44844283
13664 RP11-571L19.8       4_66 0.16995144 15.463789 2.549477e-06 -2.35541164
      num_eqtl
5518         1
6184         1
6617         1
8714         1
9300         2
11121        1
11329        1
12774        2
12794        2
13664        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-42.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-42">
Past versions of unnamed-chunk-23-42.png
</button>
</p>
<div id="fig-unnamed-chunk-23-42" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-42.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;LCAT&quot;
[1] &quot;16_36&quot;
          genename region_tag  susie_pip       mu2          PVE          z
391           EDC4      16_36 0.03400473  4.660952 1.537533e-07 -0.2045455
394           CDH1      16_36 0.03389962  4.632534 1.523435e-07 -0.1683168
397         FAM65A      16_36 0.03782273  5.639383 2.069164e-07  0.4771242
591           CDH3      16_36 0.03436924  4.758944 1.586686e-07 -0.3938948
731           CBFB      16_36 0.05411423  8.946107 4.696302e-07  1.5744681
853         NFATC3      16_36 0.03891414  5.901233 2.227720e-07  0.9370463
1324         CMTM1      16_36 0.05948939  9.824574 5.669747e-07 -1.5940594
1901         ELMO3      16_36 0.07278104 11.702651 8.262529e-07 -1.6954497
1903         NUTF2      16_36 0.04148319  6.489996 2.611722e-07  1.3725490
1905      TSNAXIP1      16_36 0.03437529  4.760525 1.587492e-07 -0.1460674
1912           ACD      16_36 0.10212577 14.885295 1.474698e-06  2.3179191
1914        PARD6A      16_36 0.10212577 14.885295 1.474698e-06  2.3179191
1929        SLC7A6      16_36 0.03413838  4.697017 1.555520e-07 -0.2000000
1931         ESRP2      16_36 0.03677567  5.381101 1.919739e-07  0.4476190
3921       SLC12A4      16_36 0.03422702  4.720811 1.567459e-07 -0.1379310
3922         ENKD1      16_36 0.04705703  7.653340 3.493703e-07  0.8580858
4020        LRRC29      16_36 0.03649905  5.311661 1.880712e-07  0.6692913
4024      C16orf70      16_36 0.05887384  9.728014 5.555933e-07  1.3144944
4682         PRMT7      16_36 0.04359627  6.948192 2.938540e-07  0.7735476
5058      DYNC1LI2      16_36 0.03512533  4.958930 1.689736e-07 -0.6000000
5059         FHOD1      16_36 0.03456573  4.811325 1.613321e-07  0.0608060
5061        SLC9A5      16_36 0.03644022  5.296826 1.872437e-07  0.6577274
7299          NAE1      16_36 0.05372475  8.879170 4.627615e-07  1.4328358
7308        LRRC36      16_36 0.04904254  8.035048 3.822714e-07 -1.6036585
7309         TPPP3      16_36 0.03430976  4.743029 1.578642e-07  0.3612335
7310      ATP6V0D1      16_36 0.03728246  5.507007 1.991731e-07  1.0800000
7312      C16orf86      16_36 0.05196721  8.570843 4.320792e-07  1.7131399
8349         BEAN1      16_36 0.05319507  8.787406 4.534636e-07 -1.2727273
8350           TK2      16_36 0.08137123 12.746689 1.006187e-06 -1.8119658
9285          PDP2      16_36 0.04895400  8.018556 3.807981e-07 -1.1764706
9928       EXOC3L1      16_36 0.03809024  5.704243 2.107765e-07 -0.7656250
10366        CMTM4      16_36 0.07785828 12.333128 9.315121e-07 -1.7164179
10492        ZFP90      16_36 0.03414356  4.698411 1.556217e-07 -0.2000000
10854        NRN1L      16_36 0.04746783  7.733646 3.561182e-07 -0.9029126
11019    KIAA0895L      16_36 0.03506720  4.943707 1.681761e-07  0.5526279
11027      PLEKHG4      16_36 0.06966926 11.294435 7.633367e-07 -1.9274194
11778       PSMB10      16_36 0.07275881 11.699098 8.257497e-07  2.0465116
11781         E2F4      16_36 0.09616595 14.317207 1.335642e-06 -2.2233846
11910         LCAT      16_36 0.03381462  4.609450 1.512043e-07 -0.2272727
12523       B3GNT9      16_36 0.06757969 11.010839 7.218501e-07 -1.7058824
12756    LINC00920      16_36 0.04325337  6.875356 2.884865e-07 -0.8559874
13829 RP11-615I2.6      16_36 0.09808115 14.504259 1.380040e-06 -1.7346386
      num_eqtl
391          1
394          1
397          1
591          2
731          1
853          2
1324         1
1901         2
1903         1
1905         1
1912         1
1914         1
1929         1
1931         1
3921         1
3922         1
4020         1
4024         2
4682         2
5058         1
5059         2
5061         2
7299         1
7308         1
7309         1
7310         1
7312         2
8349         1
8350         1
9285         1
9928         1
10366        1
10492        1
10854        1
11019        2
11027        1
11778        1
11781        2
11910        1
12523        1
12756        2
13829        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-43.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-43">
Past versions of unnamed-chunk-23-43.png
</button>
</p>
<div id="fig-unnamed-chunk-23-43" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-43.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;VDAC1&quot;
[1] &quot;5_80&quot;
        genename region_tag  susie_pip       mu2          PVE           z
111        CDKL3       5_80 0.04198163  4.564652 1.858992e-07 -0.08130081
756        PITX1       5_80 0.06532111  8.658644 5.486733e-07  1.07027027
844         AFF4       5_80 0.04940146  6.067178 2.907615e-07  0.68807339
1084        TCF7       5_80 0.09434263 12.101939 1.107576e-06  1.51470588
3022      PPP2CA       5_80 0.04771665  5.746423 2.659978e-07  0.53676471
3024     C5orf15       5_80 0.04421073  5.041777 2.162329e-07  0.30293910
3531       UBE2B       5_80 0.05122915  6.403247 3.182203e-07 -0.72514620
4679       PCBD2       5_80 0.11029947 13.580855 1.453152e-06 -1.75470345
7982       SEPT8       5_80 0.05110468  6.380735 3.163310e-07  0.77333333
7983     SHROOM1       5_80 0.07188192  9.551008 6.660077e-07 -1.21875000
7984        GDF9       5_80 0.04514514  5.234798 2.292564e-07  0.46268657
7985       UQCRQ       5_80 0.04572256  5.352124 2.373926e-07 -0.52293578
8018       CAMLG       5_80 0.07276961  9.665632 6.823241e-07  1.24770642
11925      VDAC1       5_80 0.04307558  4.801835 2.006544e-07  0.22752809
12524 CDKN2AIPNL       5_80 0.10703776 13.295919 1.380593e-06  1.59019991
12868  LINC01843       5_80 0.04220565  4.613711 1.888997e-07  0.11428851
      num_eqtl
111          1
756          1
844          1
1084         1
3022         1
3024         2
3531         1
4679         2
7982         1
7983         1
7984         1
7985         1
8018         1
11925        1
12524        2
12868        2</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-44.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-44">
Past versions of unnamed-chunk-23-44.png
</button>
</p>
<div id="fig-unnamed-chunk-23-44" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-23-44.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;APOC2&quot;
[1] &quot;19_31&quot;
      genename region_tag  susie_pip       mu2          PVE           z
204      PLAUR      19_31 0.02849177  4.808054 1.328921e-07  0.31958763
867      XRCC1      19_31 0.02829552  4.744703 1.302378e-07  0.25581395
2099     KCNN4      19_31 0.04351501  8.701255 3.673089e-07 -1.06930693
2266     ETHE1      19_31 0.03414140  6.468101 2.142242e-07  0.68748771
2267      SMG9      19_31 0.03758500  7.351636 2.680458e-07 -0.89043174
3971     ZNF45      19_31 0.05900086 11.518291 6.592601e-07 -1.37024221
4540    ZNF227      19_31 0.03640418  7.057995 2.492545e-07 -0.91280315
7319    ZNF230      19_31 0.03395243  6.417107 2.113589e-07 -0.73630137
7323    ZNF221      19_31 0.03974818  7.866755 3.033355e-07 -0.95714286
8475      IRGQ      19_31 0.02777114  4.573267 1.232057e-07  0.11217949
8476    ZNF226      19_31 0.02765363  4.534412 1.216420e-07  0.06122449
8519    ZNF283      19_31 0.03058882  5.459326 1.619989e-07 -0.46268657
9651    ZNF404      19_31 0.03087040  5.543399 1.660079e-07 -0.46268657
9865    ZNF223      19_31 0.03028714  5.368403 1.577298e-07  0.44285714
10644   ZNF284      19_31 0.02770775  4.552327 1.223616e-07  0.10101010
11740   ZNF155      19_31 0.02962992  5.167181 1.485233e-07  0.47048028
12434   PINLYP      19_31 0.02765847  4.536014 1.217063e-07 -0.06795808
13018   ZNF225      19_31 0.04727306  9.465932 4.340977e-07 -1.17391304
13291   ZNF234      19_31 0.02770092  4.550066 1.222707e-07  0.11594203
13411   ZNF224      19_31 0.03783347  7.412266 2.720430e-07  0.89772727
112      MARK4      19_31 0.03067690  5.485707 1.632505e-07 -0.47282609
119   TRAPPC6A      19_31 0.06385391 12.253005 7.589978e-07 -1.43410853
214      ERCC1      19_31 0.06209460 11.993145 7.224326e-07  1.43750000
593     ZNF112      19_31 0.02823966  4.726588 1.294844e-07  0.15555945
866        PVR      19_31 0.02786954  4.605680 1.245186e-07 -0.16666667
2111    CLPTM1      19_31 0.04952683  9.896424 4.754767e-07  1.22413793
2113   PPP1R37      19_31 0.02764689  4.532178 1.215524e-07  0.02325315
2115       CKM      19_31 0.02947524  5.119184 1.463756e-07  0.42066734
2117  PPP1R13L      19_31 0.03936284  7.777046 2.969693e-07  0.94933333
3451    CD3EAP      19_31 0.02904211  4.983442 1.404003e-07  0.38478177
4079      FOSB      19_31 0.02784195  4.596603 1.241501e-07  0.11538462
4080      OPA3      19_31 0.04559245  9.131621 4.038790e-07  1.09859155
4082      RTN2      19_31 0.03633139  7.039587 2.481074e-07  0.82935506
4412   NECTIN2      19_31 0.03594458  6.941161 2.420338e-07  0.91447368
4413      APOE      19_31 0.06880641 12.948792 8.643080e-07  1.75000000
4414    TOMM40      19_31 0.03077329  5.514488 1.646226e-07 -0.59268930
4415     APOC1      19_31 0.16958180 21.526348 3.541278e-06  2.46376812
5860    GEMIN7      19_31 0.04929982  9.853941 4.712656e-07 -1.23208778
7324    ZNF233      19_31 0.14629720 20.091293 2.851375e-06  2.44329897
7325    ZNF235      19_31 0.06927352 13.011865 8.744143e-07  1.88899386
8477    ZNF180      19_31 0.03262520  6.050867 1.915055e-07  0.70149254
8993    ZNF296      19_31 0.02805998  4.668090 1.270682e-07  0.19444444
10708 CEACAM19      19_31 0.03280796  6.102161 1.942108e-07  0.66278513
10978  BLOC1S3      19_31 0.02976706  5.209531 1.504337e-07 -0.41492537
11951    PPM1N      19_31 0.02966612  5.178378 1.490270e-07 -0.46241261
12455    APOC2      19_31 0.03222310  5.937003 1.855859e-07  0.52968037
13399   ZNF285      19_31 0.02878369  4.901500 1.368630e-07 -0.35172414
      num_eqtl
204          1
867          1
2099         1
2266         2
2267         2
3971         1
4540         2
7319         1
7323         1
8475         1
8476         1
8519         1
9651         1
9865         1
10644        1
11740        2
12434        3
13018        1
13291        1
13411        1
112          1
119          1
214          1
593          2
866          1
2111         1
2113         2
2115         2
2117         1
3451         2
4079         1
4080         1
4082         2
4412         1
4413         1
4414         1
4415         1
5860         3
7324         1
7325         2
8477         1
8993         1
10708        2
10978        1
11951        2
12455        1
13399        1</code></pre>
<pre class="r"><code>#run APOE locus again using full SNPs
# focus &lt;- &quot;APOE&quot;
# region_tag &lt;- ctwas_res$region_tag[which(ctwas_res$genename==focus)]
# 
# locus_plot(region_tag, label=&quot;TWAS&quot;, rerun_ctwas = T)
# 
# mtext(text=region_tag)
# 
# print(focus)
# print(region_tag)
# print(ctwas_gene_res[ctwas_gene_res$region_tag==region_tag,report_cols,])</code></pre>
</div>
<div id="locus-plots---false-positives" class="section level2">
<h2>Locus Plots - False positives</h2>
<p>This section produces locus plots for all bystander genes with PIP&gt;0.8 (false positives). The highlighted gene at each region is the false positive gene.</p>
<pre class="r"><code>false_positives &lt;- ctwas_gene_res$genename[ctwas_gene_res$genename %in% unrelated_genes &amp; ctwas_gene_res$susie_pip&gt;0.8]

for (i in 1:length(false_positives)){
  focus &lt;- false_positives[i]
  region_tag &lt;- ctwas_res$region_tag[which(ctwas_res$genename==focus)]

  locus_plot3(region_tag, focus=focus)
  mtext(text=region_tag)

  print(focus)
  print(region_tag)
  print(ctwas_gene_res[ctwas_gene_res$region_tag==region_tag,report_cols,])
  
  #genes at this locus that are in known annotations
  ctwas_gene_res$genename[ctwas_gene_res$region_tag==region_tag][ctwas_gene_res$genename[ctwas_gene_res$region_tag==region_tag] %in% known_annotations]
}</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-24-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] &quot;STK11IP&quot;
[1] &quot;2_130&quot;
           genename region_tag  susie_pip       mu2          PVE          z
3272          EPHA4      2_130 0.03760385  6.393523 2.332292e-07 -0.6912247
6114        STK11IP      2_130 0.84063808 18.968247 1.546845e-05 -3.8680218
13420 RP11-256I23.3      2_130 0.03070736  4.584763 1.365745e-07  0.2298851
      num_eqtl
3272         2
6114         2
13420        1</code></pre>
</div>
<div id="genes-with-many-eqtl" class="section level2">
<h2>Genes with many eQTL</h2>
<pre class="r"><code>#distribution of number of eQTL for all imputed genes (after dropping ambiguous variants)
table(ctwas_gene_res$num_eqtl)</code></pre>
<pre><code>
   1    2    3    4    5    6 
7076 3067  605   98   13    3 </code></pre>
<pre class="r"><code>#all genes with 4+ eQTL
ctwas_gene_res[ctwas_gene_res$num_eqtl&gt;3,]</code></pre>
<pre><code>      chrom                 id       pos type region_tag1 region_tag2 cs_index
8829      1 ENSG00000169598.15   3856947 gene           1           3        0
7228      1  ENSG00000158825.5  20587958 gene           1          14        0
10362     1  ENSG00000183682.7  39491437 gene           1          24        0
11907     1 ENSG00000213366.12 109668049 gene           1          69        0
3657      1 ENSG00000120332.15 175068086 gene           1          86        0
13141     1  ENSG00000259865.1 247164089 gene           1         131        0
12651     2  ENSG00000242282.6   3525259 gene           2           2        0
13152     2  ENSG00000260077.1  10011416 gene           2           6        0
6061      2 ENSG00000144026.11  95125518 gene           2          57        0
3534      2  ENSG00000119147.9 106041343 gene           2          63        0
837       2 ENSG00000072163.19 127617901 gene           2          75        0
11022     2 ENSG00000196141.13 200299060 gene           2         118        0
5431      2 ENSG00000138363.14 215075224 gene           2         127        0
5865      2 ENSG00000142330.19 240586128 gene           2         143        0
11971     3 ENSG00000214021.15   9809721 gene           3           8        0
4655      4 ENSG00000132406.11   4247756 gene           4           5        0
12866     4  ENSG00000251129.1  31994228 gene           4          27        0
12748     4  ENSG00000246375.2  88284457 gene           4          59        0
7927      4 ENSG00000164124.10 158173634 gene           4         102        0
12863     4  ENSG00000250971.1 186813865 gene           4         120        0
12845     5  ENSG00000250490.1   6363492 gene           5           6        0
7951      5  ENSG00000164237.8  10308070 gene           5           9        0
4716      5  ENSG00000132840.9  79069112 gene           5          46        0
8063      5 ENSG00000164904.17 126544897 gene           5          77        0
12810     5  ENSG00000249306.5 174323562 gene           5         105        0
6251      6  ENSG00000145949.9   2680498 gene           6           3        0
6252      6 ENSG00000145979.17  13279232 gene           6          12        0
11690     6  ENSG00000204516.9  31494471 gene           6          27        0
11695     6 ENSG00000204531.17  31179417 gene           6          27        0
11944     6 ENSG00000213780.10  30904497 gene           6          27        0
11647     6  ENSG00000204301.6  32202656 gene           6          27        0
11633     6  ENSG00000204228.3  33204859 gene           6          27        0
9023      6  ENSG00000170915.8  52335661 gene           6          39        0
2358      7  ENSG00000106524.8  16599990 gene           7          16        0
6768      7 ENSG00000152926.14  64998115 gene           7          43        0
12433     7  ENSG00000234444.9  64307784 gene           7          43        0
11275     7 ENSG00000197558.11 149756134 gene           7          92        0
10781     7 ENSG00000187260.15 151404454 gene           7          94        0
2296      7 ENSG00000105983.20 156806908 gene           7          98        0
10215     8  ENSG00000182372.9   1292383 gene           8           3        0
12980     8  ENSG00000255394.4  11758186 gene           8          15        0
5316      8 ENSG00000137563.11  63029281 gene           8          48        0
11294     9  ENSG00000197646.7   5483361 gene           9           6        0
2416      9 ENSG00000107186.16  13106599 gene           9          12        0
5242      9 ENSG00000137074.18  33024919 gene           9          25        0
11958     9 ENSG00000213930.11  34637693 gene           9          27        0
6423      9 ENSG00000148357.16 130254358 gene           9          67        0
8177      9 ENSG00000165689.16 136406987 gene           9          73        0
5186     10 ENSG00000136738.14  17641079 gene          10          14        0
4791     10 ENSG00000133661.15  79979389 gene          10          51        0
5404     10 ENSG00000138119.16  93323282 gene          10          59        0
8952     10  ENSG00000170430.9 129467281 gene          10          81        0
9724     11 ENSG00000177042.14    688091 gene          11           1        0
3726     11 ENSG00000121236.20   5590335 gene          11           4        0
2696     11  ENSG00000110328.5  11410495 gene          11           9        0
764      12 ENSG00000069493.14   9664279 gene          12           9        0
40       12 ENSG00000004700.15  21501331 gene          12          16        0
11169    12  ENSG00000196935.8  63844067 gene          12          39        0
2808     12  ENSG00000111581.9  68686129 gene          12          42        0
10495    12  ENSG00000184967.6 132083372 gene          12          81        0
13025    12  ENSG00000256576.2 132189571 gene          12          81        0
2018     15 ENSG00000103966.10  41923122 gene          15          15        0
10624    15 ENSG00000185880.12  44727667 gene          15          17        0
1214     16 ENSG00000086504.15    345757 gene          16           1        0
1938     16 ENSG00000103148.15    137677 gene          16           1        0
1980     16 ENSG00000103381.11  12803538 gene          16          13        0
10341    16 ENSG00000183549.10  20409610 gene          16          19        0
6544     16 ENSG00000149922.10  30091070 gene          16          24        0
5732     16 ENSG00000140955.10  84184959 gene          16          48        0
5743     16 ENSG00000141013.16  90018731 gene          16          54        0
4344     17 ENSG00000129204.16   5116034 gene          17           5        0
10120    17  ENSG00000181291.6  34581017 gene          17          21        0
13108    17  ENSG00000259207.7  47221042 gene          17          27        0
13351    17  ENSG00000266714.6  75588005 gene          17          42        0
13289    17  ENSG00000262877.4  81388422 gene          17          46        0
5796     17 ENSG00000141526.16  82228706 gene          17          47        0
944      18  ENSG00000075643.5  36187019 gene          18          19        0
708      18 ENSG00000066926.10  57571588 gene          18          31        0
651      19 ENSG00000065268.10    982793 gene          19           2        0
1526     19 ENSG00000099817.11   1032229 gene          19           2        0
976      19  ENSG00000076826.9   7594599 gene          19           8        0
277      19 ENSG00000021488.12  32869435 gene          19          23        0
13395    19  ENSG00000267475.1  32691414 gene          19          23        0
11312    19 ENSG00000197782.14  40070352 gene          19          27        0
4459     19 ENSG00000130529.15  49114644 gene          19          35        0
8552     19 ENSG00000167766.18  52615913 gene          19          36        0
9027     19 ENSG00000170949.17  53087254 gene          19          36        0
11285    20 ENSG00000197586.12  25195384 gene          20          18        0
10660    20  ENSG00000186191.7  33077982 gene          20          20        0
11015    20 ENSG00000196090.12  43187301 gene          20          27        0
12140    20  ENSG00000223891.5  44210535 gene          20          28        0
7376     21 ENSG00000160305.17  46438335 gene          21          24        0
1440     22 ENSG00000093072.15  17197305 gene          22           2        0
1546     22 ENSG00000099957.16  21003838 gene          22           4        0
10551    22  ENSG00000185339.8  30596847 gene          22          10        0
10343    22 ENSG00000183569.17  42552989 gene          22          18        0
10997    22 ENSG00000189306.10  42490805 gene          22          18        0
10747    22 ENSG00000186976.14  43808777 gene          22          19        0
99        2 ENSG00000006607.13 241354886 gene           2         144        0
8665      2 ENSG00000168397.16 241636141 gene           2         144        0
10082     2 ENSG00000180902.17 241732627 gene           2         144        0
8101      6 ENSG00000165097.13  18153891 gene           6          14        0
8296     10  ENSG00000166295.8  72211079 gene          10          49        0
9454     11  ENSG00000174370.9 128809746 gene          11          80        0
6594     12 ENSG00000150977.10 123430286 gene          12          75        0
7162     12  ENSG00000158023.9 121918251 gene          12          75        0
3866     13 ENSG00000123179.13  49691102 gene          13          21        0
7111     14 ENSG00000157379.13  24291558 gene          14           3        0
5639     14 ENSG00000140043.11  73844725 gene          14          34        0
163      14 ENSG00000009830.11  77311261 gene          14          36        0
10088    15 ENSG00000180953.11  79922366 gene          15          37        0
1334     19 ENSG00000089847.12   4183158 gene          19           4        0
7353     21 ENSG00000160200.17  42941853 gene          21          22        0
7358     21  ENSG00000160213.6  43767678 gene          21          22        0
         susie_pip       mu2 region_tag          PVE       genename
8829  4.160727e-02  4.562732        1_3 1.841639e-07           DFFB
7228  1.659371e-01 16.299471       1_14 2.623781e-06            CDA
10362 3.536819e-02  4.661882       1_24 1.599501e-07          BMP8A
11907 1.142417e-02  9.805167       1_69 1.086651e-07          GSTM2
3657  6.710729e-02 12.989658       1_86 8.456251e-07            TNN
13141 3.085086e-02  9.011259      1_131 2.696890e-07 RP11-488L18.10
12651 1.896278e-02  7.677279        2_2 1.412277e-07     AC108488.4
13152 4.407680e-02  6.839611        2_6 2.924502e-07   RP11-254F7.2
6061  8.657371e-02  8.343336       2_57 7.007066e-07         ZNF514
3534  1.411348e-01 14.521299       2_63 1.988153e-06        C2orf40
837   6.084210e-02  5.313185       2_75 3.135953e-07          LIMS2
11022 3.003384e-02 95.300331      2_118 2.776615e-06        SPATS2L
5431  1.360629e-01 16.535574      2_127 2.182576e-06           ATIC
5865  3.482796e-02  7.164635      2_143 2.420653e-07         CAPN10
11971 1.188509e-01 11.239305        3_8 1.295843e-06          TTLL3
4655  4.169085e-02  4.603728        4_5 1.861919e-07        TMEM128
12866 6.874848e-02  7.086021       4_27 4.725806e-07  RP11-734I18.1
12748 5.990045e-02  4.698849       4_59 2.730436e-07    RP11-10L7.1
7927  6.030136e-02  4.990715      4_102 2.919445e-07        TMEM144
12863 6.017045e-02  5.706989      4_120 3.331200e-07  RP11-696F12.1
12845 1.198806e-01 13.958888        5_6 1.623343e-06      LINC02145
7951  5.059846e-02  5.388011        5_9 2.644698e-07           CMBL
4716  4.239101e-02  5.025132       5_46 2.066482e-07          BHMT2
8063  6.077566e-02  5.731451       5_77 3.379128e-07        ALDH7A1
12810 6.464465e-02  7.490826      5_105 4.697564e-07      LINC01411
6251  4.335437e-02  4.532337        6_3 1.906187e-07          MYLK4
6252  5.601621e-02  5.601698       6_12 3.043994e-07         TBC1D7
11690 6.007886e-02 27.777389       6_27 1.618913e-06           MICB
11695 8.047401e-03  9.415482       6_27 7.350360e-08         POU5F1
11944 1.296462e-02 13.811198       6_27 1.737007e-07         GTF2H4
11647 4.710762e-03  4.543353       6_27 2.076243e-08         NOTCH4
11633 1.396179e-02 14.381660       6_27 1.947873e-07        HSD17B8
9023  2.867181e-02  4.543987       6_39 1.263871e-07          PAQR8
2358  4.702895e-02  5.445215       7_16 2.484224e-07         ANKMY2
6768  5.023305e-02  6.151104       7_43 2.997458e-07         ZNF117
12433 8.419199e-02 11.240276       7_43 9.180328e-07         ZNF736
11275 5.200805e-02  8.969464       7_92 4.525301e-07           SSPO
10781 3.396058e-02  8.645892       7_94 2.848363e-07          WDR86
2296  6.147956e-02  8.163856       7_98 4.868963e-07          LMBR1
10215 4.632397e-02  4.535147        8_3 2.038016e-07           CLN8
12980 2.476107e-02  8.773421       8_15 2.107408e-07        C8orf49
5316  6.841080e-02  7.031033       8_48 4.666102e-07            GGH
11294 5.306128e-02  4.855033        9_6 2.499081e-07       PDCD1LG2
2416  3.206349e-02  4.883384       9_12 1.518945e-07           MPDZ
5242  5.319653e-02  4.542096       9_25 2.343959e-07           APTX
11958 2.328524e-02  5.078618       9_27 1.147193e-07           GALT
6423  2.466585e-01 20.836963       9_67 4.985871e-06          HMCN2
8177  8.784738e-02 13.072608       9_73 1.114042e-06        SDCCAG3
5186  5.387607e-02 10.479376      10_14 5.476987e-07           STAM
4791  6.836780e-02  8.431123      10_51 5.591746e-07          SFTPD
5404  1.445696e-01 15.503954      10_59 2.174352e-06           MYOF
8952  6.789768e-02  9.622812      10_81 6.338221e-07           MGMT
9724  3.960285e-02  4.552221       11_1 1.748881e-07         TMEM80
3726  3.951884e-02  4.530694       11_4 1.736918e-07          TRIM6
2696  4.172490e-02  4.834936       11_9 1.957025e-07        GALNT18
764   6.774877e-02  6.210837       12_9 4.081896e-07         CLEC2D
40    4.723176e-02 11.482884      12_16 5.261329e-07          RECQL
11169 5.232829e-02  6.129059      12_39 3.111292e-07         SRGAP1
2808  1.228938e-01 13.104567      12_42 1.562295e-06         NUP107
10495 1.028720e-01 12.642046      12_81 1.261609e-06          NOC4L
13025 4.396147e-02  4.711916      12_81 2.009464e-07  RP13-977J11.2
2018  1.444954e-01 17.024716      15_15 2.386407e-06           EHD4
10624 6.678218e-02  5.884794      15_17 3.812433e-07         TRIM69
1214  4.551849e-02  8.109793       16_1 3.581031e-07         MRPL28
1938  1.796355e-01 21.054359       16_1 3.668973e-06          NPRL3
1980  5.161597e-02  6.782632      16_13 3.396196e-07         CPPED1
10341 9.125280e-02  9.051325      16_19 8.012514e-07          ACSM5
6544  4.402615e-02  5.613910      16_24 2.397654e-07           TBX6
5732  6.711738e-02  7.912542      16_48 5.151830e-07          ADAD2
5743  5.375185e-02  6.618971      16_54 3.451392e-07           GAS8
4344  4.956765e-02  5.281360       17_5 2.539536e-07           USP6
10120 2.682028e-02  4.984821      17_21 1.296950e-07       TMEM132E
13108 1.319177e-05 10.880709      17_27 1.392421e-10          ITGB3
13351 3.910169e-02  4.592795      17_42 1.742140e-07         MYO15B
13289 4.163151e-02  5.107075      17_46 2.062552e-07  RP11-1055B8.4
5796  5.954408e-02  4.816236      17_47 2.781998e-07        SLC16A3
944   4.326557e-02  4.562763      18_19 1.915053e-07          MOCOS
708   5.743433e-02  7.227610      18_31 4.026954e-07           FECH
651   4.324470e-02  7.698742       19_2 3.229707e-07          WDR18
1526  2.684089e-01 25.177110       19_2 6.555612e-06         POLR2E
976   1.960340e-02  4.535871       19_8 8.625865e-08        CAMSAP3
277   1.329474e-01 17.026823      19_23 2.195957e-06         SLC7A9
13395 3.501099e-02  4.556265      19_23 1.547476e-07   CTD-2538C1.2
11312 3.907937e-02  4.680517      19_27 1.774401e-07        ZNF780A
4459  1.348543e-02  5.099611      19_35 6.671329e-08          TRPM4
8552  3.337832e-02  4.565379      19_36 1.478263e-07          ZNF83
9027  3.869678e-02  5.924854      19_36 2.224144e-07         ZNF160
11285 5.941315e-02  8.129045      20_18 4.685247e-07         ENTPD6
10660 5.525131e-02  6.947800      20_20 3.723920e-07         BPIFB4
11015 3.867975e-02  4.572127      20_27 1.715586e-07          PTPRT
12140 2.660360e-02  6.086123      20_28 1.570694e-07      OSER1-AS1
7376  3.970524e-02  6.138704      21_24 2.364476e-07          DIP2A
1440  4.014696e-02  4.625767       22_2 1.801552e-07          CECR1
1546  1.909623e-02  7.224043       22_4 1.338253e-07          P2RX6
10551 1.456716e-01 16.721527      22_10 2.362986e-06           TCN2
10343 4.385792e-02  4.696390      22_18 1.998125e-07         SERHL2
10997 4.472207e-02  4.876429      22_18 2.115603e-07          RRP7A
10747 5.866690e-02  6.643992      22_19 3.781226e-07         EFCAB6
99    3.773174e-02  4.920054      2_144 1.800889e-07          FARP2
8665  3.819760e-02  5.543522      2_144 2.054150e-07          ATG4B
10082 3.665308e-02  4.592015      2_144 1.632767e-07         D2HGDH
8101  3.367752e-02 47.184002       6_14 1.541506e-06          KDM1B
8296  7.630441e-05 11.415357      10_49 8.449861e-10        ANAPC16
9454  2.571903e-02 19.573814      11_80 4.883605e-07       C11orf45
6594  9.310735e-04  4.788344      12_75 4.324937e-09         RILPL2
7162  1.577479e-03  9.393574      12_75 1.437490e-08          WDR66
3866  5.454124e-02 10.259270      13_21 5.428150e-07           EBPL
7111  1.370814e-02  5.168687       14_3 6.873363e-08          DHRS1
5639  1.457521e-02  5.222525      14_34 7.384237e-08          PTGR2
163   2.537794e-02  4.919146      14_36 1.211034e-07          POMT2
10088 2.986045e-02  7.646866      15_37 2.215084e-07           ST20
1334  3.238781e-02  7.550345       19_4 2.372241e-07        ANKRD24
7353  1.054189e-02  5.401357      21_22 5.523721e-08            CBS
7358  1.076657e-02  5.588965      21_22 5.837398e-08           CSTB
           gene_type           z num_eqtl
8829  protein_coding -0.05606883        4
7228  protein_coding -2.11159404        5
10362 protein_coding -0.04886818        4
11907 protein_coding -1.13618389        4
3657  protein_coding -1.56712336        4
13141        lincRNA -1.15241658        4
12651        lincRNA  0.88509724        4
13152        lincRNA -0.65540955        5
6061  protein_coding  1.18030677        5
3534  protein_coding  1.75470100        5
837   protein_coding  0.67333098        4
11022 protein_coding  8.11412691        4
5431  protein_coding -1.95877307        4
5865  protein_coding -1.08230144        4
11971 protein_coding  1.41366703        4
4655  protein_coding  0.35456062        4
12866        lincRNA  0.99806910        4
12748        lincRNA  0.16815663        4
7927  protein_coding -0.36615299        5
12863        lincRNA  0.52938089        4
12845        lincRNA -1.84178478        4
7951  protein_coding  0.57035749        4
4716  protein_coding  0.55240521        4
8063  protein_coding  0.67205955        4
12810        lincRNA -0.87537902        4
6251  protein_coding -0.01538851        4
6252  protein_coding  0.58335172        4
11690 protein_coding  2.73153856        4
11695 protein_coding  1.19707710        4
11944 protein_coding  1.54571781        4
11647 protein_coding -0.04864491        4
11633 protein_coding -1.70351244        4
9023  protein_coding  0.05092472        4
2358  protein_coding  0.59449010        4
6768  protein_coding  0.29410631        4
12433 protein_coding -1.42962198        4
11275 protein_coding  1.17327949        4
10781 protein_coding -1.07155972        4
2296  protein_coding  1.12846744        4
10215 protein_coding  0.14155900        5
12980        lincRNA  2.00933006        4
5316  protein_coding -1.03772803        5
11294 protein_coding -0.13876175        4
2416  protein_coding -0.31465621        4
5242  protein_coding -0.10240511        4
11958 protein_coding  0.54117071        4
6423  protein_coding  2.58282602        6
8177  protein_coding  1.62552220        4
5186  protein_coding -1.21519734        5
4791  protein_coding -1.06876491        4
5404  protein_coding  1.89158295        4
8952  protein_coding -1.23104737        4
9724  protein_coding -0.21243537        4
3726  protein_coding  0.03791292        4
2696  protein_coding -0.25931605        4
764   protein_coding -0.74959674        4
40    protein_coding  1.39889693        4
11169 protein_coding -0.64611366        4
2808  protein_coding  1.57728136        4
10495 protein_coding -1.59211541        6
13025        lincRNA -0.17676741        4
2018  protein_coding -2.18379904        4
10624 protein_coding  0.68260923        4
1214  protein_coding  1.46503375        4
1938  protein_coding -2.38794751        5
1980  protein_coding -0.78395748        4
10341 protein_coding -1.23454259        4
6544  protein_coding  0.23682640        4
5732  protein_coding -1.14113140        4
5743  protein_coding -0.82726355        4
4344  protein_coding -0.28654155        4
10120 protein_coding -0.33913678        4
13108 protein_coding -1.57910576        4
13351 protein_coding  0.07613186        5
13289        lincRNA -0.44622658        4
5796  protein_coding -0.36078965        4
944   protein_coding -0.13469930        4
708   protein_coding  0.79795344        4
651   protein_coding  1.06428772        4
1526  protein_coding -2.83809922        4
976   protein_coding -0.01811536        4
277   protein_coding -2.02040773        4
13395        lincRNA  0.08674841        4
11312 protein_coding  0.34051207        4
4459  protein_coding -0.38500405        4
8552  protein_coding -0.10819463        5
9027  protein_coding -0.58121153        4
11285 protein_coding  1.11446112        4
10660 protein_coding  0.88525055        4
11015 protein_coding  0.06492351        4
12140        lincRNA -0.42101203        5
7376  protein_coding  0.60857479        4
1440  protein_coding -0.16372486        4
1546  protein_coding -0.82364067        4
10551 protein_coding  2.42724084        4
10343 protein_coding  0.22201409        4
10997 protein_coding -0.17607349        6
10747 protein_coding -0.82194388        4
99    protein_coding  0.51764732        4
8665  protein_coding  1.00584632        4
10082 protein_coding  0.02223367        4
8101  protein_coding -7.80377324        4
8296  protein_coding  1.50960211        4
9454  protein_coding -4.26886900        4
6594  protein_coding -0.31787410        4
7162  protein_coding -1.03591335        4
3866  protein_coding -0.71354410        4
7111  protein_coding  0.28913514        4
5639  protein_coding -0.74125560        4
163   protein_coding -0.09069729        5
10088 protein_coding  1.24686579        4
1334  protein_coding -0.95784361        4
7353  protein_coding -0.47878923        4
7358  protein_coding -0.62864698        4</code></pre>
<pre class="r"><code>#distribution of number of eQTL for genes with PIP&gt;0.8
table(ctwas_gene_res$num_eqtl[ctwas_gene_res$susie_pip&gt;0.8])/sum(ctwas_gene_res$susie_pip&gt;0.8)</code></pre>
<pre><code>
    1     2     3 
0.625 0.250 0.125 </code></pre>
<pre class="r"><code>#genes with 2+ eQTL and PIP&gt;0.8
ctwas_gene_res[ctwas_gene_res$num_eqtl&gt;1 &amp; ctwas_gene_res$susie_pip&gt;0.8,]</code></pre>
<pre><code>      chrom                 id       pos type region_tag1 region_tag2 cs_index
3275      1 ENSG00000116132.11 170662622 gene           1          84        4
6114      2 ENSG00000144589.21 219597759 gene           2         130        0
712       2 ENSG00000067066.16 230415508 gene           2         135        0
9691      2  ENSG00000176720.4 241555444 gene           2         144        0
2293      7 ENSG00000105974.11 116519968 gene           7          70        1
9012      8 ENSG00000170873.18 124576649 gene           8          82        1
8992      9  ENSG00000170681.6 100578027 gene           9          50        0
9257     10 ENSG00000172650.13  73650294 gene          10          49        5
2444     10 ENSG00000107651.12 119893621 gene          10          74        1
11818    14 ENSG00000205683.11  72894134 gene          14          34        1
2138     19 ENSG00000104964.14   3056215 gene          19           4        0
6914     21 ENSG00000154721.14  25638800 gene          21           9        1
      susie_pip       mu2 region_tag          PVE genename      gene_type
3275  0.9998808 119.34663       1_84 1.157627e-04    PRRX1 protein_coding
6114  0.8406381  18.96825      2_130 1.546845e-05  STK11IP protein_coding
712   0.8658723  18.73540      2_135 1.573719e-05    SP100 protein_coding
9691  0.8255975  19.18259      2_144 1.536335e-05      BOK protein_coding
2293  1.0000000 622.84815       7_70 6.042165e-04     CAV1 protein_coding
9012  0.8594108  20.87861       8_82 1.740655e-05    MTSS1 protein_coding
8992  0.8478869  23.34736       9_50 1.920376e-05     MURC protein_coding
9257  0.9779202  48.91516      10_49 4.640420e-05    AGAP5 protein_coding
2444  0.9517809  22.38862      10_74 2.067163e-05  SEC23IP protein_coding
11818 0.9574312  33.35512      14_34 3.097993e-05     DPF3 protein_coding
2138  0.9403219  20.20438       19_4 1.843030e-05      AES protein_coding
6914  0.9640543  22.19279       21_9 2.075505e-05     JAM2 protein_coding
              z num_eqtl
3275  14.667578        2
6114  -3.868022        2
712   -3.671335        2
9691   3.910125        3
2293  15.567870        3
9012   4.402634        2
8992   4.911964        2
9257  11.518590        2
2444  -4.565228        2
11818  6.264960        3
2138   4.182804        3
6914   4.563232        2</code></pre>
</div>
<div id="ctwas-genes-in-go-terms-enriched-for-silver-standard-genes" class="section level2">
<h2>cTWAS genes in GO terms enriched for silver standard genes</h2>
<pre class="r"><code>#reload silver standard genes
known_annotations &lt;- read_xlsx(&quot;data/summary_known_genes_annotations.xlsx&quot;, sheet=&quot;LDL&quot;)</code></pre>
<pre><code>New names:
* `` -&gt; ...4
* `` -&gt; ...5</code></pre>
<pre class="r"><code>known_annotations &lt;- unique(known_annotations$`Gene Symbol`)

#GO enrichment analysis for silver standard genes
dbs &lt;- c(&quot;GO_Biological_Process_2021&quot;, &quot;GO_Cellular_Component_2021&quot;, &quot;GO_Molecular_Function_2021&quot;)
genes &lt;- known_annotations
GO_enrichment &lt;- enrichr(genes, dbs)</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.</code></pre>
<pre class="r"><code>for (db in dbs){
  print(db)
  df &lt;- GO_enrichment[[db]]
  df &lt;- df[df$Adjusted.P.value&lt;0.05,c(&quot;Term&quot;, &quot;Overlap&quot;, &quot;Adjusted.P.value&quot;, &quot;Genes&quot;)]
  plotEnrich(GO_enrichment[[db]])
  print(df)
}</code></pre>
<pre><code>[1] &quot;GO_Biological_Process_2021&quot;
                                                                                       Term
1                                                        cholesterol transport (GO:0030301)
2                                                      cholesterol homeostasis (GO:0042632)
3                                                           sterol homeostasis (GO:0055092)
4                                                           cholesterol efflux (GO:0033344)
5                                                             sterol transport (GO:0015918)
6                                                cholesterol metabolic process (GO:0008203)
7                                                     sterol metabolic process (GO:0016125)
8                            triglyceride-rich lipoprotein particle remodeling (GO:0034370)
9                                 high-density lipoprotein particle remodeling (GO:0034375)
10                                               reverse cholesterol transport (GO:0043691)
11                                         secondary alcohol metabolic process (GO:1902652)
12                                   regulation of lipoprotein lipase activity (GO:0051004)
13                                                      phospholipid transport (GO:0015914)
14                                                             lipid transport (GO:0006869)
15                                                    acylglycerol homeostasis (GO:0055090)
16                            very-low-density lipoprotein particle remodeling (GO:0034372)
17                                                    triglyceride homeostasis (GO:0070328)
18                                              triglyceride metabolic process (GO:0006641)
19                                                         phospholipid efflux (GO:0033700)
20                                                      chylomicron remodeling (GO:0034371)
21                                         regulation of cholesterol transport (GO:0032374)
22                                                        chylomicron assembly (GO:0034378)
23                                               chylomicron remnant clearance (GO:0034382)
24                                               lipoprotein metabolic process (GO:0042157)
25                                               diterpenoid metabolic process (GO:0016101)
26                            positive regulation of steroid metabolic process (GO:0045940)
27                                                  retinoid metabolic process (GO:0001523)
28                                                           lipid homeostasis (GO:0055088)
29                                      negative regulation of lipase activity (GO:0060192)
30                           positive regulation of cholesterol esterification (GO:0010873)
31                                           intestinal cholesterol absorption (GO:0030299)
32                                negative regulation of cholesterol transport (GO:0032375)
33                                                 intestinal lipid absorption (GO:0098856)
34                                              acylglycerol metabolic process (GO:0006639)
35                                    regulation of cholesterol esterification (GO:0010872)
36                                                    phospholipid homeostasis (GO:0055091)
37                              very-low-density lipoprotein particle assembly (GO:0034379)
38                              positive regulation of lipid metabolic process (GO:0045834)
39                                  high-density lipoprotein particle assembly (GO:0034380)
40                          negative regulation of lipoprotein lipase activity (GO:0051005)
41                       negative regulation of lipoprotein particle clearance (GO:0010985)
42              regulation of very-low-density lipoprotein particle remodeling (GO:0010901)
43                                         intracellular cholesterol transport (GO:0032367)
44                                positive regulation of cholesterol transport (GO:0032376)
45                             regulation of intestinal cholesterol absorption (GO:0030300)
46                          positive regulation of lipoprotein lipase activity (GO:0051006)
47                                              acylglycerol catabolic process (GO:0046464)
48                           positive regulation of lipid biosynthetic process (GO:0046889)
49                       positive regulation of triglyceride metabolic process (GO:0090208)
50                         positive regulation of triglyceride lipase activity (GO:0061365)
51                                       regulation of lipid catabolic process (GO:0050994)
52                                                   steroid metabolic process (GO:0008202)
53                              positive regulation of lipid catabolic process (GO:0050996)
54                                              triglyceride catabolic process (GO:0019433)
55                                             organophosphate ester transport (GO:0015748)
56                                       phosphatidylcholine metabolic process (GO:0046470)
57                                regulation of triglyceride catabolic process (GO:0010896)
58                                                fatty acid metabolic process (GO:0006631)
59                               regulation of fatty acid biosynthetic process (GO:0042304)
60                                              regulation of sterol transport (GO:0032371)
61              cellular response to low-density lipoprotein particle stimulus (GO:0071404)
62                                 low-density lipoprotein particle remodeling (GO:0034374)
63                  regulation of macrophage derived foam cell differentiation (GO:0010743)
64                                                 organic substance transport (GO:0071702)
65                                            regulation of cholesterol efflux (GO:0010874)
66                                               receptor-mediated endocytosis (GO:0006898)
67                                      secondary alcohol biosynthetic process (GO:1902653)
68                                           regulation of cholesterol storage (GO:0010885)
69                                                          cholesterol import (GO:0070508)
70                                                               sterol import (GO:0035376)
71                                    monocarboxylic acid biosynthetic process (GO:0072330)
72                                            cholesterol biosynthetic process (GO:0006695)
73                           positive regulation of cellular metabolic process (GO:0031325)
74                                                 sterol biosynthetic process (GO:0016126)
75                         positive regulation of fatty acid metabolic process (GO:0045923)
76                                 regulation of receptor-mediated endocytosis (GO:0048259)
77                                             fatty acid biosynthetic process (GO:0006633)
78                             regulation of Cdc42 protein signal transduction (GO:0032489)
79                       positive regulation of triglyceride catabolic process (GO:0010898)
80                                                  lipid biosynthetic process (GO:0008610)
81                                   positive regulation of cholesterol efflux (GO:0010875)
82                        negative regulation of receptor-mediated endocytosis (GO:0048261)
83                                                       lipoprotein transport (GO:0042953)
84                                       regulation of lipid metabolic process (GO:0019216)
85                                       monocarboxylic acid metabolic process (GO:0032787)
86                                                    lipoprotein localization (GO:0044872)
87                                                     lipid catabolic process (GO:0016042)
88                      positive regulation of fatty acid biosynthetic process (GO:0045723)
89                                              intracellular sterol transport (GO:0032366)
90                                                steroid biosynthetic process (GO:0006694)
91                                    regulation of lipid biosynthetic process (GO:0046890)
92                                               regulation of lipase activity (GO:0060191)
93                        positive regulation of cellular biosynthetic process (GO:0031328)
94                                        regulation of amyloid-beta clearance (GO:1900221)
95                                              phospholipid metabolic process (GO:0006644)
96                                   regulation of intestinal lipid absorption (GO:1904729)
97             positive regulation of protein catabolic process in the vacuole (GO:1904352)
98                                 positive regulation of biosynthetic process (GO:0009891)
99                                 regulation of cholesterol metabolic process (GO:0090181)
100                                                  foam cell differentiation (GO:0090077)
101                                 positive regulation of cholesterol storage (GO:0010886)
102                               macrophage derived foam cell differentiation (GO:0010742)
103                              organic hydroxy compound biosynthetic process (GO:1901617)
104                                    regulation of steroid metabolic process (GO:0019218)
105                               organonitrogen compound biosynthetic process (GO:1901566)
106                             negative regulation of lipid metabolic process (GO:0045833)
107                          regulation of lysosomal protein catabolic process (GO:1905165)
108                              positive regulation of amyloid-beta clearance (GO:1900223)
109                                           cellular lipid catabolic process (GO:0044242)
110                                                       chemical homeostasis (GO:0048878)
111                                              cholesterol catabolic process (GO:0006707)
112                                                   sterol catabolic process (GO:0016127)
113                                       steroid hormone biosynthetic process (GO:0120178)
114                   regulation of low-density lipoprotein particle clearance (GO:0010988)
115                                                bile acid metabolic process (GO:0008206)
116                                   phosphatidylcholine biosynthetic process (GO:0006656)
117                                                  alcohol catabolic process (GO:0046164)
118                                          organophosphate catabolic process (GO:0046434)
119                                       regulation of phospholipase activity (GO:0010517)
120                                  positive regulation of lipid localization (GO:1905954)
121                              positive regulation of phospholipid transport (GO:2001140)
122                                      glycerophospholipid metabolic process (GO:0006650)
123                                         organic hydroxy compound transport (GO:0015850)
124        negative regulation of macrophage derived foam cell differentiation (GO:0010745)
125                                     positive regulation of lipid transport (GO:0032370)
126                                   C21-steroid hormone biosynthetic process (GO:0006700)
127                                                      membrane organization (GO:0061024)
128                                         positive regulation of endocytosis (GO:0045807)
129                    positive regulation of multicellular organismal process (GO:0051240)
130                                   negative regulation of catabolic process (GO:0009895)
131                             negative regulation of lipid catabolic process (GO:0050995)
132                                          carbohydrate derivative transport (GO:1901264)
133        positive regulation of macrophage derived foam cell differentiation (GO:0010744)
134                                                          protein transport (GO:0015031)
135                                                       fatty acid transport (GO:0015908)
136                                       positive regulation of lipid storage (GO:0010884)
137                                      C21-steroid hormone metabolic process (GO:0008207)
138                                             phospholipid catabolic process (GO:0009395)
139                                         negative regulation of endocytosis (GO:0045806)
140                                    regulation of primary metabolic process (GO:0080090)
141                    negative regulation of multicellular organismal process (GO:0051241)
142                                             bile acid biosynthetic process (GO:0006699)
143  regulation of low-density lipoprotein particle receptor catabolic process (GO:0032803)
144                              regulation of Rho protein signal transduction (GO:0035023)
145                             regulation of small molecule metabolic process (GO:0062012)
146                          positive regulation of cellular catabolic process (GO:0031331)
147                                                       artery morphogenesis (GO:0048844)
148                     negative regulation of cellular component organization (GO:0051129)
149                                                       glycolipid transport (GO:0046836)
150                      positive regulation of lipoprotein particle clearance (GO:0010986)
151                                    positive regulation of sterol transport (GO:0032373)
152                                            long-chain fatty acid transport (GO:0015909)
153                                                        response to insulin (GO:0032868)
154                                  regulation of bile acid metabolic process (GO:1904251)
155                          positive regulation of receptor catabolic process (GO:2000646)
156                                           positive regulation of transport (GO:0051050)
157                      negative regulation of endothelial cell proliferation (GO:0001937)
158                          negative regulation of endothelial cell migration (GO:0010596)
159                          regulation of nitrogen compound metabolic process (GO:0051171)
160                              negative regulation of amyloid-beta clearance (GO:1900222)
161                          negative regulation of cellular metabolic process (GO:0031324)
162                                   glycerophospholipid biosynthetic process (GO:0046474)
163 negative regulation of production of molecular mediator of immune response (GO:0002701)
164                                unsaturated fatty acid biosynthetic process (GO:0006636)
165                                                            anion transport (GO:0006820)
166                       positive regulation of receptor-mediated endocytosis (GO:0048260)
167                                 negative regulation of cholesterol storage (GO:0010887)
168                               regulation of bile acid biosynthetic process (GO:0070857)
169                                           peptidyl-amino acid modification (GO:0018193)
170                low-density lipoprotein particle receptor catabolic process (GO:0032802)
171                low-density lipoprotein receptor particle metabolic process (GO:0032799)
172                                                          protein oxidation (GO:0018158)
173                               positive regulation by host of viral process (GO:0044794)
174                   positive regulation of triglyceride biosynthetic process (GO:0010867)
175                                                   receptor internalization (GO:0031623)
176                                                        response to glucose (GO:0009749)
177                     positive regulation of cellular component organization (GO:0051130)
178                     negative regulation of fatty acid biosynthetic process (GO:0045717)
179                                          negative regulation of hemostasis (GO:1900047)
180                                           peptidyl-methionine modification (GO:0018206)
181                                                          ethanol oxidation (GO:0006069)
182                            negative regulation of amyloid fibril formation (GO:1905907)
183                           negative regulation of protein metabolic process (GO:0051248)
184                                   unsaturated fatty acid metabolic process (GO:0033559)
185                                     alpha-linolenic acid metabolic process (GO:0036109)
186                                                     platelet degranulation (GO:0002576)
187                                   negative regulation of metabolic process (GO:0009892)
188                                     negative regulation of cell activation (GO:0050866)
189                                         negative regulation of coagulation (GO:0050819)
190                                                    cGMP-mediated signaling (GO:0019934)
191                                                      intestinal absorption (GO:0050892)
192                                                 receptor metabolic process (GO:0043112)
193                                                 regulation of phagocytosis (GO:0050764)
194                                     regulation of amyloid fibril formation (GO:1905906)
195                                 regulation of sequestering of triglyceride (GO:0010889)
196                            regulation of triglyceride biosynthetic process (GO:0010866)
197                                    post-translational protein modification (GO:0043687)
198                                                  regulation of endocytosis (GO:0030100)
199                                                   amyloid fibril formation (GO:1990000)
200                    positive regulation of small molecule metabolic process (GO:0062013)
201                                       cellular response to nutrient levels (GO:0031669)
202     negative regulation of cytokine production involved in immune response (GO:0002719)
203                        negative regulation of fatty acid metabolic process (GO:0045922)
204                             regulation of cholesterol biosynthetic process (GO:0045540)
205                                 regulation of steroid biosynthetic process (GO:0050810)
206                                   positive regulation of catabolic process (GO:0009896)
207                                amyloid precursor protein metabolic process (GO:0042982)
208                                  nitric oxide mediated signal transduction (GO:0007263)
209                      positive regulation of nitric-oxide synthase activity (GO:0051000)
210                                                  ethanol metabolic process (GO:0006067)
211                  positive regulation of cellular protein catabolic process (GO:1903364)
212                                                     response to fatty acid (GO:0070542)
213                                                           long-term memory (GO:0007616)
214                                       negative regulation of lipid storage (GO:0010888)
215                                            linoleic acid metabolic process (GO:0043651)
216                          negative regulation of lipid biosynthetic process (GO:0051055)
217                                                regulation of lipid storage (GO:0010883)
218                                regulation of interleukin-1 beta production (GO:0032651)
219                                    long-chain fatty acid metabolic process (GO:0001676)
220              regulation of cytokine production involved in immune response (GO:0002718)
221                                         cellular protein metabolic process (GO:0044267)
222                                    negative regulation of defense response (GO:0031348)
223                                       transport across blood-brain barrier (GO:0150104)
224                                                 receptor catabolic process (GO:0032801)
225                                                         response to hexose (GO:0009746)
226                                                       regulated exocytosis (GO:0045055)
227                                   regulation of endothelial cell migration (GO:0010594)
228                                   negative regulation of protein transport (GO:0051224)
229                                             positive regulation of binding (GO:0051099)
230                                            regulation of blood coagulation (GO:0030193)
231                              positive regulation of monooxygenase activity (GO:0032770)
232                                       negative regulation of wound healing (GO:0061045)
233                     negative regulation of macromolecule metabolic process (GO:0010605)
234                                 long-chain fatty acid biosynthetic process (GO:0042759)
235                                         regulation of developmental growth (GO:0048638)
236                                                   regulation of cell death (GO:0010941)
237                                                 regulation of angiogenesis (GO:0045765)
238                                        regulation of inflammatory response (GO:0050727)
239                                                   apoptotic cell clearance (GO:0043277)
240                              cellular response to peptide hormone stimulus (GO:0071375)
241             negative regulation of blood vessel endothelial cell migration (GO:0043537)
242                            phosphate-containing compound metabolic process (GO:0006796)
243                           negative regulation of epithelial cell migration (GO:0010633)
244                                          cellular response to amyloid-beta (GO:1904646)
245                                       cyclic-nucleotide-mediated signaling (GO:0019935)
246                                     regulation of receptor internalization (GO:0002090)
247                                                          response to lipid (GO:0033993)
248         regulation of vascular associated smooth muscle cell proliferation (GO:1904705)
249                          regulation of protein-containing complex assembly (GO:0043254)
250                                                              ion transport (GO:0006811)
251                       negative regulation of response to external stimulus (GO:0032102)
252                                              regulation of protein binding (GO:0043393)
253                                regulation of cellular component biogenesis (GO:0044087)
254                                   negative regulation of protein secretion (GO:0050709)
255                                   negative regulation of secretion by cell (GO:1903531)
256                               regulation of nitric-oxide synthase activity (GO:0050999)
257                                   negative regulation of blood coagulation (GO:0030195)
258                                     cellular response to organic substance (GO:0071310)
259                                      cellular response to insulin stimulus (GO:0032869)
260                                                   response to amyloid-beta (GO:1904645)
261              establishment of protein localization to extracellular region (GO:0035592)
262                                   regulation of cellular metabolic process (GO:0031323)
263                                    regulation of protein metabolic process (GO:0051246)
264                                positive regulation of cell differentiation (GO:0045597)
265                        negative regulation of cell projection organization (GO:0031345)
266                            positive regulation of fat cell differentiation (GO:0045600)
267                               negative regulation of BMP signaling pathway (GO:0030514)
268                       negative regulation of cellular biosynthetic process (GO:0031327)
269                                        positive regulation of phagocytosis (GO:0050766)
    Overlap Adjusted.P.value
1     28/51     3.291336e-55
2     29/71     1.134840e-52
3     29/72     1.264418e-52
4     16/24     1.003687e-32
5     15/21     2.203564e-31
6     20/77     5.273224e-31
7     19/70     7.882518e-30
8     12/13     1.520527e-27
9     13/18     2.514128e-27
10    12/17     5.731565e-25
11    15/49     2.711706e-24
12    12/21     2.245426e-23
13    15/59     5.665269e-23
14   17/109     2.748742e-22
15    12/25     3.145271e-22
16      9/9     2.283165e-21
17    12/31     7.414146e-21
18    13/55     1.935295e-19
19     9/12     4.196729e-19
20      8/9     5.374944e-18
21    10/25     1.639871e-17
22     8/10     2.436689e-17
23      7/7     1.679428e-16
24      7/9     5.763376e-15
25    11/64     8.362646e-15
26     7/13     2.508790e-13
27    11/92     5.133855e-13
28    10/64     5.133855e-13
29      6/9     3.296018e-12
30      6/9     3.296018e-12
31      6/9     3.296018e-12
32     6/11     1.693836e-11
33     6/11     1.693836e-11
34     8/41     3.013332e-11
35     6/12     3.013332e-11
36     6/12     3.013332e-11
37     6/12     3.013332e-11
38     7/25     4.654994e-11
39     6/13     5.294950e-11
40      5/6     5.459029e-11
41      5/6     5.459029e-11
42      5/6     5.459029e-11
43     6/15     1.393181e-10
44     7/33     3.496187e-10
45      5/8     4.627510e-10
46      5/8     4.627510e-10
47     7/35     5.017364e-10
48     7/35     5.017364e-10
49     6/19     6.556580e-10
50      5/9     9.553575e-10
51     6/21     1.253116e-09
52    9/104     1.493946e-09
53     6/22     1.653549e-09
54     6/23     2.189811e-09
55     6/25     3.733511e-09
56     8/77     3.733511e-09
57     5/12     5.225810e-09
58    9/124     6.552660e-09
59     6/29     9.279729e-09
60      4/5     9.798195e-09
61     5/14     1.207993e-08
62     5/14     1.207993e-08
63     6/31     1.339781e-08
64    9/136     1.355933e-08
65     6/33     1.942867e-08
66    9/143     2.054367e-08
67     6/34     2.282600e-08
68     5/16     2.390304e-08
69      4/6     2.513038e-08
70      4/6     2.513038e-08
71     7/63     2.556641e-08
72     6/35     2.556641e-08
73    8/105     3.517095e-08
74     6/38     4.196695e-08
75     5/18     4.228478e-08
76     6/39     4.816188e-08
77     7/71     5.609765e-08
78      4/8     1.033779e-07
79      4/8     1.033779e-07
80     7/80     1.258618e-07
81     5/23     1.517283e-07
82     5/26     2.906610e-07
83     4/10     2.936601e-07
84     7/92     3.199662e-07
85    8/143     3.471498e-07
86     4/11     4.442138e-07
87     5/29     4.906437e-07
88     4/13     9.252034e-07
89     4/13     9.252034e-07
90     6/65     9.598111e-07
91     5/35     1.249797e-06
92     4/14     1.249797e-06
93    8/180     1.884951e-06
94     4/16     2.212485e-06
95     6/76     2.336232e-06
96      3/5     3.664395e-06
97      3/5     3.664395e-06
98     5/44     3.827136e-06
99     4/21     6.819051e-06
100     3/6     6.952354e-06
101     3/6     6.952354e-06
102     3/6     6.952354e-06
103    5/50     6.991423e-06
104    4/23     9.554179e-06
105   7/158     1.040987e-05
106    4/24     1.121951e-05
107     3/7     1.146235e-05
108     3/7     1.146235e-05
109    4/27     1.788032e-05
110    5/65     2.452122e-05
111     3/9     2.639634e-05
112     3/9     2.639634e-05
113    4/31     3.060279e-05
114    3/10     3.695601e-05
115    4/33     3.856854e-05
116    4/33     3.856854e-05
117    3/11     4.775659e-05
118    3/11     4.775659e-05
119    3/11     4.775659e-05
120    3/11     4.775659e-05
121    3/11     4.775659e-05
122    5/80     6.182763e-05
123    4/40     7.973389e-05
124    3/13     7.973389e-05
125    3/13     7.973389e-05
126    3/15     1.252218e-04
127   7/242     1.420233e-04
128    4/48     1.598563e-04
129   8/345     1.704405e-04
130    4/49     1.709436e-04
131    3/18     2.111817e-04
132    3/18     2.111817e-04
133    3/18     2.111817e-04
134   8/369     2.646441e-04
135    3/20     2.892290e-04
136    3/21     3.341258e-04
137    3/24     4.974036e-04
138    3/24     4.974036e-04
139    3/25     5.597802e-04
140   5/130     5.616142e-04
141   6/214     6.166972e-04
142    3/27     6.934199e-04
143     2/5     7.406583e-04
144    4/73     7.449454e-04
145    3/28     7.586859e-04
146   5/141     7.898802e-04
147    3/30     9.228897e-04
148    4/80     1.034287e-03
149     2/6     1.049782e-03
150     2/6     1.049782e-03
151     2/6     1.049782e-03
152    3/32     1.085012e-03
153    4/84     1.208000e-03
154     2/7     1.428577e-03
155     2/7     1.428577e-03
156    4/91     1.611630e-03
157    3/37     1.625395e-03
158    3/38     1.749224e-03
159     2/8     1.841132e-03
160     2/8     1.841132e-03
161    3/39     1.855101e-03
162   5/177     2.046657e-03
163     2/9     2.304288e-03
164     2/9     2.304288e-03
165    3/43     2.420344e-03
166    3/44     2.575438e-03
167    2/10     2.756295e-03
168    2/10     2.756295e-03
169    2/10     2.756295e-03
170    2/10     2.756295e-03
171    2/10     2.756295e-03
172    2/11     3.303347e-03
173    2/11     3.303347e-03
174    2/11     3.303347e-03
175    3/49     3.337798e-03
176    3/49     3.337798e-03
177   4/114     3.341630e-03
178    2/12     3.781332e-03
179    2/12     3.781332e-03
180    2/12     3.781332e-03
181    2/12     3.781332e-03
182    2/12     3.781332e-03
183    3/52     3.822266e-03
184    3/54     4.245657e-03
185    2/13     4.386588e-03
186   4/125     4.489058e-03
187    3/56     4.645735e-03
188    2/14     4.945884e-03
189    2/14     4.945884e-03
190    2/14     4.945884e-03
191    2/14     4.945884e-03
192    3/58     4.985945e-03
193    3/58     4.985945e-03
194    2/15     5.548828e-03
195    2/15     5.548828e-03
196    2/15     5.548828e-03
197   6/345     5.596173e-03
198    3/61     5.626994e-03
199    3/63     6.147256e-03
200    2/16     6.200855e-03
201    3/66     6.840974e-03
202    2/17     6.840974e-03
203    2/17     6.840974e-03
204    2/17     6.840974e-03
205    2/17     6.840974e-03
206    3/67     7.093595e-03
207    2/18     7.532008e-03
208    2/18     7.532008e-03
209    2/18     7.532008e-03
210    2/19     8.241667e-03
211    2/19     8.241667e-03
212    2/19     8.241667e-03
213    2/19     8.241667e-03
214    2/20     9.094348e-03
215    2/21     9.982653e-03
216    2/22     1.085553e-02
217    2/22     1.085553e-02
218    3/83     1.230478e-02
219    3/83     1.230478e-02
220    2/24     1.273659e-02
221   6/417     1.291939e-02
222    3/85     1.298477e-02
223    3/86     1.336097e-02
224    2/25     1.350640e-02
225    2/25     1.350640e-02
226   4/180     1.399465e-02
227    3/89     1.440735e-02
228    2/26     1.440735e-02
229    3/90     1.479001e-02
230    2/27     1.539039e-02
231    2/28     1.646589e-02
232    2/29     1.757027e-02
233   4/194     1.771224e-02
234    2/30     1.862299e-02
235    2/31     1.977865e-02
236   3/102     2.036757e-02
237   4/203     2.042828e-02
238   4/206     2.141587e-02
239    2/33     2.198466e-02
240   3/106     2.228428e-02
241    2/34     2.311351e-02
242   4/212     2.328380e-02
243    2/35     2.415927e-02
244    2/35     2.415927e-02
245    2/36     2.531623e-02
246    2/36     2.531623e-02
247   3/114     2.646649e-02
248    2/37     2.648825e-02
249   3/116     2.742785e-02
250   3/116     2.742785e-02
251   3/118     2.842391e-02
252   3/118     2.842391e-02
253    2/39     2.842391e-02
254    2/39     2.842391e-02
255    2/39     2.842391e-02
256    2/39     2.842391e-02
257    2/40     2.973753e-02
258   3/123     3.119488e-02
259   3/129     3.533580e-02
260    2/44     3.533580e-02
261    2/46     3.834204e-02
262    2/47     3.980515e-02
263    2/48     4.128649e-02
264   4/258     4.183273e-02
265    2/49     4.262417e-02
266    2/51     4.583569e-02
267    2/52     4.720898e-02
268    2/52     4.720898e-02
269    2/53     4.877011e-02
                                                                                                                                                                       Genes
1         SCARB1;CETP;LCAT;LIPC;NPC1L1;LIPG;CD36;APOE;LDLRAP1;APOB;LDLR;ABCA1;ABCG8;STARD3;ABCG5;OSBPL5;APOA2;APOA1;APOC3;APOA4;APOA5;NPC1;SOAT1;STAR;NPC2;SOAT2;APOC2;APOC1
2   SCARB1;CETP;MTTP;PCSK9;LPL;LCAT;ABCB11;CYP7A1;LIPC;LIPG;APOE;LDLRAP1;APOB;LDLR;ABCA1;ABCG8;ABCG5;EPHX2;APOA2;APOA1;APOC3;APOA4;APOA5;SOAT1;NPC1;NPC2;SOAT2;APOC2;ANGPTL3
3   SCARB1;CETP;MTTP;PCSK9;LPL;LCAT;ABCB11;CYP7A1;LIPC;LIPG;APOE;LDLRAP1;APOB;LDLR;ABCA1;ABCG8;ABCG5;EPHX2;APOA2;APOA1;APOC3;APOA4;APOA5;SOAT1;NPC1;NPC2;SOAT2;APOC2;ANGPTL3
4                                                                              ABCA1;ABCG8;SCARB1;ABCG5;APOA2;APOA1;APOC3;APOA4;APOA5;NPC1;SOAT1;NPC2;SOAT2;APOC2;APOC1;APOE
5                                                                                    ABCG8;CETP;STARD3;ABCG5;OSBPL5;APOA2;APOA1;LCAT;NPC1;NPC1L1;NPC2;CD36;APOB;LDLRAP1;LDLR
6                                              ABCA1;STARD3;CETP;OSBPL5;APOA2;APOA1;LCAT;APOA4;HMGCR;APOA5;CYP7A1;CYP27A1;SOAT1;SOAT2;NPC1L1;ANGPTL3;APOE;DHCR7;LDLRAP1;APOB
7                                                      ABCA1;STARD3;CETP;OSBPL5;APOA2;APOA1;LCAT;APOA4;HMGCR;LIPA;CYP7A1;CYP27A1;SOAT1;SOAT2;ANGPTL3;APOE;DHCR7;LDLRAP1;APOB
8                                                                                                           CETP;LIPC;APOC2;APOA2;APOA1;APOC3;LCAT;LPL;APOA4;APOE;APOB;APOA5
9                                                                                                   CETP;SCARB1;APOA2;APOA1;APOC3;LCAT;APOA4;LIPC;APOC2;APOC1;LIPG;APOE;PLTP
10                                                                                                       ABCA1;CETP;SCARB1;LIPC;APOC2;LIPG;APOA2;APOA1;APOC3;LCAT;APOA4;APOE
11                                                                             ABCA1;STARD3;CETP;OSBPL5;APOA2;APOA1;LCAT;APOA4;CYP27A1;SOAT1;SOAT2;ANGPTL3;APOE;LDLRAP1;APOB
12                                                                                                   LIPC;SORT1;APOC2;APOH;APOC1;ANGPTL3;APOA1;APOC3;LPL;APOA4;ANGPTL4;APOA5
13                                                                                    ABCA1;SCARB1;OSBPL5;MTTP;APOA2;APOA1;APOC3;APOA4;APOA5;NPC2;APOC2;APOC1;APOE;LDLR;PLTP
14                                                                        ABCA1;SCARB1;ABCG8;CETP;ABCG5;OSBPL5;MTTP;APOA1;APOA4;ABCB11;APOA5;NPC2;NPC1L1;CD36;APOE;LDLR;PLTP
15                                                                                                   CETP;SCARB1;LIPC;APOC2;ANGPTL3;LPL;APOA1;APOC3;APOA4;APOE;ANGPTL4;APOA5
16                                                                                                                           CETP;LIPC;APOC2;APOA1;LCAT;LPL;APOA4;APOE;APOA5
17                                                                                                   CETP;SCARB1;LIPC;APOC2;ANGPTL3;LPL;APOA1;APOC3;APOA4;APOE;ANGPTL4;APOA5
18                                                                                                      CETP;APOA2;LPL;APOC3;APOA5;LIPC;LIPI;APOH;LIPG;APOC1;APOE;APOB;LPIN3
19                                                                                                                      ABCA1;APOC2;APOC1;APOA2;APOA1;APOC3;APOA4;APOE;APOA5
20                                                                                                                               APOC2;APOA2;APOA1;APOC3;LPL;APOA4;APOE;APOB
21                                                                                                                   CETP;LRP1;APOC2;LIPG;APOC1;APOA2;TSPO;APOA1;APOA4;APOA5
22                                                                                                                              APOC2;MTTP;APOA2;APOA1;APOC3;APOA4;APOE;APOB
23                                                                                                                                     LIPC;APOC2;APOC1;APOC3;APOE;APOB;LDLR
24                                                                                                                                  NPC1L1;MTTP;APOA2;APOA1;APOA4;APOE;APOA5
25                                                                                                               LRP1;ADH1B;APOC2;APOA2;APOA1;LPL;APOC3;APOA4;LRP2;APOE;APOB
26                                                                                                                                APOC1;APOA2;APOA1;APOA4;APOE;LDLRAP1;APOA5
27                                                                                                               LRP1;ADH1B;APOC2;APOA2;APOA1;LPL;APOC3;APOA4;LRP2;APOE;APOB
28                                                                                                               ABCA1;CETP;LIPG;ANGPTL3;APOA1;APOA4;PPARG;APOE;ABCB11;APOA5
29                                                                                                                                   SORT1;APOC1;ANGPTL3;APOA2;APOC3;ANGPTL4
30                                                                                                                                        APOC1;APOA2;APOA1;APOA4;APOE;APOA5
31                                                                                                                                        ABCG8;ABCG5;NPC1L1;SOAT2;CD36;LDLR
32                                                                                                                                       ABCG8;ABCG5;APOC2;APOC1;APOA2;APOC3
33                                                                                                                                        ABCG8;ABCG5;NPC1L1;SOAT2;CD36;LDLR
34                                                                                                                                CETP;APOH;APOC1;APOA2;LPL;APOC3;APOE;APOA5
35                                                                                                                                        APOC1;APOA2;APOA1;APOA4;APOE;APOA5
36                                                                                                                                      ABCA1;CETP;LIPG;ANGPTL3;APOA1;ABCB11
37                                                                                                                                         SOAT1;SOAT2;APOC1;MTTP;APOC3;APOB
38                                                                                                                                APOA2;ANGPTL3;APOA1;APOA4;PPARG;APOE;APOA5
39                                                                                                                                        ABCA1;APOA2;APOA1;APOA4;APOE;APOA5
40                                                                                                                                         SORT1;APOC1;ANGPTL3;APOC3;ANGPTL4
41                                                                                                                                            LRPAP1;APOC2;APOC1;APOC3;PCSK9
42                                                                                                                                             APOC2;APOA2;APOA1;APOC3;APOA5
43                                                                                                                                         ABCA1;NPC1;STAR;NPC2;LDLRAP1;LDLR
44                                                                                                                                      CETP;LRP1;LIPG;APOA1;PPARG;APOE;PLTP
45                                                                                                                                             ABCG8;ABCG5;APOA1;APOA4;APOA5
46                                                                                                                                              APOC2;APOH;APOA1;APOA4;APOA5
47                                                                                                                                      LIPC;LIPI;LIPG;APOA2;LPL;APOC3;APOA5
48                                                                                                                                  SCARB1;APOC2;APOA1;APOA4;APOE;LDLR;APOA5
49                                                                                                                                       SCARB1;APOC2;APOA1;APOA4;APOA5;LDLR
50                                                                                                                                              APOC2;APOH;APOA1;APOA4;APOA5
51                                                                                                                                    APOC1;APOA2;ANGPTL3;APOC3;ABCB11;APOA5
52                                                                                                                     CYP27A1;STARD3;NPC1;STAR;TSPO;LRP2;ABCB11;LIPA;CYP7A1
53                                                                                                                                     APOC2;APOA2;ANGPTL3;APOA1;APOA4;APOA5
54                                                                                                                                            LIPC;LIPI;LIPG;APOC3;LPL;APOA5
55                                                                                                                                         SCARB1;OSBPL5;NPC2;MTTP;LDLR;PLTP
56                                                                                                                              CETP;LIPC;APOA2;APOA1;LCAT;APOA4;APOA5;LPIN3
57                                                                                                                                             APOC2;APOA1;APOC3;APOA4;APOA5
58                                                                                                                        LIPC;LIPI;LIPG;ANGPTL3;LPL;PPARG;CD36;ABCB11;LPIN3
59                                                                                                                                       APOC2;APOC1;APOA1;APOC3;APOA4;APOA5
60                                                                                                                                                     LRP1;APOC1;TSPO;APOA4
61                                                                                                                                                 ABCA1;LPL;PPARG;CD36;LDLR
62                                                                                                                                                  CETP;LIPC;APOA2;APOB;LPA
63                                                                                                                                            ABCA1;CETP;LPL;PPARG;CD36;APOB
64                                                                                                                        ABCA1;ABCG8;CETP;ABCG5;APOA1;APOA4;LRP2;APOA5;PLTP
65                                                                                                                                           CETP;LRP1;APOA1;PPARG;APOE;PLTP
66                                                                                                                        SCARB1;LRP1;APOA1;CD36;LRP2;APOE;LDLRAP1;APOB;LDLR
67                                                                                                                                      NPC1L1;APOA1;APOA4;HMGCR;DHCR7;APOA5
68                                                                                                                                               ABCA1;SCARB1;LPL;PPARG;APOB
69                                                                                                                                                    SCARB1;APOA1;CD36;LDLR
70                                                                                                                                                    SCARB1;APOA1;CD36;LDLR
71                                                                                                                                  CYP27A1;LIPC;LIPI;LIPG;LPL;ABCB11;CYP7A1
72                                                                                                                                      NPC1L1;APOA1;APOA4;HMGCR;DHCR7;APOA5
73                                                                                                                            APOC1;APOA2;PCSK9;APOA1;APOA4;PPARG;APOE;APOA5
74                                                                                                                                      NPC1L1;APOA1;APOA4;HMGCR;DHCR7;APOA5
75                                                                                                                                             APOC2;APOA1;APOA4;PPARG;APOA5
76                                                                                                                                    LRPAP1;APOC2;APOC1;APOC3;LDLRAP1;APOA5
77                                                                                                                                      FADS3;LIPC;LIPI;EPHX2;LIPG;LPL;FADS1
78                                                                                                                                                    ABCA1;APOA1;APOC3;APOE
79                                                                                                                                                   APOC2;APOA1;APOA4;APOA5
80                                                                                                                                       LIPC;STAR;LIPI;LIPG;LPL;HMGCR;FADS1
81                                                                                                                                                LRP1;APOA1;PPARG;APOE;PLTP
82                                                                                                                                            LRPAP1;APOC2;APOC1;PCSK9;APOC3
83                                                                                                                                                      LRP1;PPARG;CD36;APOB
84                                                                                                                                  NPC2;APOC2;APOC1;APOC3;PPARG;HMGCR;DHCR7
85                                                                                                                            NPC1;ADH1B;ANGPTL3;LPL;PPARG;VDAC1;CD36;ABCB11
86                                                                                                                                                      LRP1;PPARG;CD36;APOB
87                                                                                                                                                  LIPC;LIPI;LIPG;LPL;APOA4
88                                                                                                                                                   APOC2;APOA1;APOA4;APOA5
89                                                                                                                                                      ABCA1;NPC1;STAR;NPC2
90                                                                                                                                    CYP27A1;STAR;HMGCR;DHCR7;ABCB11;CYP7A1
91                                                                                                                                               STAR;APOA1;APOA4;APOE;APOA5
92                                                                                                                                                    LIPC;APOA2;ANGPTL3;LPL
93                                                                                                                             SCARB1;STAR;APOC2;APOA1;APOA4;CD36;APOA5;LDLR
94                                                                                                                                                    LRPAP1;LRP1;HMGCR;APOE
95                                                                                                                                         LIPG;APOA2;ANGPTL3;LPL;LCAT;FADS1
96                                                                                                                                                         APOA1;APOA4;APOA5
97                                                                                                                                                            LRP1;LRP2;LDLR
98                                                                                                                                               APOA1;APOA4;APOE;CD36;APOA5
99                                                                                                                                                  EPHX2;APOE;LDLRAP1;KPNB1
100                                                                                                                                                        SOAT1;SOAT2;PPARG
101                                                                                                                                                          SCARB1;LPL;APOB
102                                                                                                                                                        SOAT1;SOAT2;PPARG
103                                                                                                                                        CYP27A1;HMGCR;DHCR7;ABCB11;CYP7A1
104                                                                                                                                                   STAR;EPHX2;APOE;ABCB11
105                                                                                                                                    VAPA;VAPB;APOA2;APOA1;LCAT;APOE;LPIN3
106                                                                                                                                                  APOC2;APOC1;APOA2;APOC3
107                                                                                                                                                           LRP1;LRP2;LDLR
108                                                                                                                                                         LRPAP1;LRP1;APOE
109                                                                                                                                                 LIPG;APOA2;ANGPTL3;LPIN3
110                                                                                                                                          CETP;ANGPTL3;APOA4;PPARG;ABCB11
111                                                                                                                                                      CYP27A1;APOE;CYP7A1
112                                                                                                                                                      CYP27A1;APOE;CYP7A1
113                                                                                                                                                   STARD3;STAR;TSPO;DHCR7
114                                                                                                                                                      APOC3;PCSK9;LDLRAP1
115                                                                                                                                               CYP27A1;NPC1;ABCB11;CYP7A1
116                                                                                                                                                   APOA2;LCAT;APOA1;LPIN3
117                                                                                                                                                      CYP27A1;APOE;CYP7A1
118                                                                                                                                                       LIPG;ANGPTL3;APOA2
119                                                                                                                                                       LRP1;APOC2;ANGPTL3
120                                                                                                                                                            LRP1;LPL;APOB
121                                                                                                                                                          CETP;APOA1;APOE
122                                                                                                                                              CETP;APOA1;LCAT;APOA4;APOA5
123                                                                                                                                                  ABCG8;ABCG5;NPC2;ABCB11
124                                                                                                                                                         ABCA1;CETP;PPARG
125                                                                                                                                                           CETP;LRP1;APOE
126                                                                                                                                                         STARD3;STAR;TSPO
127                                                                                                                                    NPC1;VAPA;VAPB;LRP2;LDLRAP1;APOB;LDLR
128                                                                                                                                                  LRP1;APOE;LDLRAP1;APOA5
129                                                                                                                              GHR;ABCA1;LRPAP1;LRP1;APOC2;CD36;APOE;APOA5
130                                                                                                                                                  APOC1;APOA2;APOC3;HMGCR
131                                                                                                                                                        APOC1;APOA2;APOC3
132                                                                                                                                                         SCARB1;NPC2;PLTP
133                                                                                                                                                            LPL;CD36;APOB
134                                                                                                                                ABCA1;LRP1;MTTP;PPARG;CD36;LRP2;APOE;APOB
135                                                                                                                                                          PPARG;APOE;CD36
136                                                                                                                                                          SCARB1;LPL;APOB
137                                                                                                                                                         STARD3;STAR;TSPO
138                                                                                                                                                       LIPG;APOA2;ANGPTL3
139                                                                                                                                                        APOC2;APOC1;APOC3
140                                                                                                                                              PPARG;HMGCR;APOE;DHCR7;LDLR
141                                                                                                                                     LRPAP1;APOA2;APOA1;APOC3;APOA4;HMGCR
142                                                                                                                                                    CYP27A1;ABCB11;CYP7A1
143                                                                                                                                                               PCSK9;APOE
144                                                                                                                                                   ABCA1;APOA1;APOC3;APOE
145                                                                                                                                                        EPHX2;APOE;ABCB11
146                                                                                                                                             APOC2;APOA1;APOA4;APOE;APOA5
147                                                                                                                                                        LRP1;ANGPTL3;LRP2
148                                                                                                                                                  APOA2;APOA1;APOC3;APOA4
149                                                                                                                                                                NPC2;PLTP
150                                                                                                                                                             LIPG;LDLRAP1
151                                                                                                                                                                CETP;LIPG
152                                                                                                                                                          PPARG;APOE;CD36
153                                                                                                                                                  SORT1;PCSK9;PPARG;LPIN3
154                                                                                                                                                            ABCB11;CYP7A1
155                                                                                                                                                               PCSK9;APOE
156                                                                                                                                                    LRP1;APOA2;APOA1;APOE
157                                                                                                                                                          APOH;PPARG;APOE
158                                                                                                                                                          APOH;PPARG;APOE
159                                                                                                                                                                APOE;LDLR
160                                                                                                                                                             LRPAP1;HMGCR
161                                                                                                                                                        LRPAP1;PCSK9;APOE
162                                                                                                                                              LIPI;APOA2;APOA1;LCAT;LPIN3
163                                                                                                                                                              APOA2;APOA1
164                                                                                                                                                              FADS3;FADS1
165                                                                                                                                                         TSPO;VDAC2;VDAC1
166                                                                                                                                                      PCSK9;LDLRAP1;APOA5
167                                                                                                                                                              ABCA1;PPARG
168                                                                                                                                                              STAR;CYP7A1
169                                                                                                                                                              APOA2;APOA1
170                                                                                                                                                              MYLIP;PCSK9
171                                                                                                                                                              MYLIP;PCSK9
172                                                                                                                                                              APOA2;APOA1
173                                                                                                                                                                VAPA;APOE
174                                                                                                                                                              SCARB1;LDLR
175                                                                                                                                                        LRP1;CD36;LDLRAP1
176                                                                                                                                                         APOA2;LPL;CYP7A1
177                                                                                                                                                    LRP1;APOC2;APOE;APOA5
178                                                                                                                                                              APOC1;APOC3
179                                                                                                                                                                APOH;APOE
180                                                                                                                                                              APOA2;APOA1
181                                                                                                                                                              ALDH2;ADH1B
182                                                                                                                                                                APOE;LDLR
183                                                                                                                                                          HMGCR;APOE;LDLR
184                                                                                                                                                        FADS3;FADS2;FADS1
185                                                                                                                                                              FADS2;FADS1
186                                                                                                                                                    ITIH4;APOH;APOA1;CD36
187                                                                                                                                                        APOC2;APOC1;APOC3
188                                                                                                                                                                APOE;LDLR
189                                                                                                                                                                APOH;APOE
190                                                                                                                                                                APOE;CD36
191                                                                                                                                                              NPC1L1;CD36
192                                                                                                                                                        LRP1;CD36;LDLRAP1
193                                                                                                                                                       SCARB1;APOA2;APOA1
194                                                                                                                                                                APOE;LDLR
195                                                                                                                                                                LPL;PPARG
196                                                                                                                                                              SCARB1;LDLR
197                                                                                                                                        APOA2;PCSK9;APOA1;APOE;APOB;APOA5
198                                                                                                                                                         LRPAP1;LRP1;APOE
199                                                                                                                                                         APOA1;APOA4;CD36
200                                                                                                                                                            PPARG;LDLRAP1
201                                                                                                                                                          PCSK9;LPL;FADS1
202                                                                                                                                                              APOA2;APOA1
203                                                                                                                                                              APOC1;APOC3
204                                                                                                                                                               APOE;KPNB1
205                                                                                                                                                              STAR;CYP7A1
206                                                                                                                                                      APOA2;ANGPTL3;APOA5
207                                                                                                                                                             APOE;LDLRAP1
208                                                                                                                                                                APOE;CD36
209                                                                                                                                                              SCARB1;APOE
210                                                                                                                                                              ALDH2;ADH1B
211                                                                                                                                                               PCSK9;APOE
212                                                                                                                                                                 LPL;CD36
213                                                                                                                                                                APOE;LDLR
214                                                                                                                                                              ABCA1;PPARG
215                                                                                                                                                              FADS2;FADS1
216                                                                                                                                                              APOC1;APOC3
217                                                                                                                                                                 LPL;APOB
218                                                                                                                                                           APOA1;LPL;CD36
219                                                                                                                                                        FADS2;EPHX2;FADS1
220                                                                                                                                                              APOA2;APOA1
221                                                                                                                                        APOA2;PCSK9;APOA1;APOE;APOB;APOA5
222                                                                                                                                                         APOA1;PPARG;APOE
223                                                                                                                                                           LRP1;CD36;LRP2
224                                                                                                                                                              MYLIP;PCSK9
225                                                                                                                                                                APOA2;LPL
226                                                                                                                                                    ITIH4;APOH;APOA1;CD36
227                                                                                                                                                         SCARB1;APOH;APOE
228                                                                                                                                                               HMGCR;APOE
229                                                                                                                                                          LRP1;PPARG;APOE
230                                                                                                                                                                APOH;APOE
231                                                                                                                                                              SCARB1;APOE
232                                                                                                                                                                APOH;APOE
233                                                                                                                                                   LRPAP1;PCSK9;APOE;LDLR
234                                                                                                                                                              EPHX2;FADS1
235                                                                                                                                                                 GHR;APOE
236                                                                                                                                                         LRPAP1;LRP1;CD36
237                                                                                                                                               APOH;ANGPTL3;PPARG;ANGPTL4
238                                                                                                                                                     APOA1;LPL;PPARG;APOE
239                                                                                                                                                              SCARB1;LRP1
240                                                                                                                                                        PCSK9;PPARG;LPIN3
241                                                                                                                                                               PPARG;APOE
242                                                                                                                                                   EPHX2;ANGPTL3;LPL;LCAT
243                                                                                                                                                                APOH;APOE
244                                                                                                                                                                LRP1;CD36
245                                                                                                                                                                APOE;CD36
246                                                                                                                                                             LRPAP1;PCSK9
247                                                                                                                                                         APOA4;PPARG;CD36
248                                                                                                                                                            PPARG;LDLRAP1
249                                                                                                                                                          ABCA1;CD36;APOE
250                                                                                                                                                         TSPO;VDAC2;VDAC1
251                                                                                                                                                         APOA1;PPARG;APOE
252                                                                                                                                                      LRPAP1;LRP1;LDLRAP1
253                                                                                                                                                                APOE;CD36
254                                                                                                                                                               HMGCR;APOE
255                                                                                                                                                               HMGCR;APOE
256                                                                                                                                                              SCARB1;APOE
257                                                                                                                                                                APOH;APOE
258                                                                                                                                                         GHR;LRP2;LDLRAP1
259                                                                                                                                                        PCSK9;PPARG;LPIN3
260                                                                                                                                                                LRP1;CD36
261                                                                                                                                                               ABCA1;MTTP
262                                                                                                                                                              NPC2;ABCB11
263                                                                                                                                                                APOE;LDLR
264                                                                                                                                                      LPL;PPARG;CD36;APOB
265                                                                                                                                                               MYLIP;APOE
266                                                                                                                                                                LPL;PPARG
267                                                                                                                                                               PPARG;LRP2
268                                                                                                                                                              APOC1;APOC3
269                                                                                                                                                              APOA2;APOA1
[1] &quot;GO_Cellular_Component_2021&quot;
                                                          Term Overlap
1               high-density lipoprotein particle (GO:0034364)   12/19
2                                     chylomicron (GO:0042627)   10/10
3   triglyceride-rich plasma lipoprotein particle (GO:0034385)   10/15
4           very-low-density lipoprotein particle (GO:0034361)   10/15
5                                  early endosome (GO:0005769)  13/266
6                low-density lipoprotein particle (GO:0034362)     4/7
7     spherical high-density lipoprotein particle (GO:0034366)     4/8
8                     endoplasmic reticulum lumen (GO:0005788)  10/285
9                      endocytic vesicle membrane (GO:0030666)   8/158
10                 endoplasmic reticulum membrane (GO:0005789)  14/712
11                                       lysosome (GO:0005764)  11/477
12                                  lytic vacuole (GO:0000323)   8/219
13                              endocytic vesicle (GO:0030139)   7/189
14     clathrin-coated endocytic vesicle membrane (GO:0030669)    5/69
15              clathrin-coated endocytic vesicle (GO:0045334)    5/85
16               clathrin-coated vesicle membrane (GO:0030665)    5/90
17                             lysosomal membrane (GO:0005765)   8/330
18                  intracellular organelle lumen (GO:0070013)  12/848
19       collagen-containing extracellular matrix (GO:0062023)   8/380
20                        endocytic vesicle lumen (GO:0071682)    3/21
21                       organelle outer membrane (GO:0031968)   5/142
22 ATP-binding cassette (ABC) transporter complex (GO:0043190)     2/6
23                         lytic vacuole membrane (GO:0098852)   6/267
24                              endosome membrane (GO:0010008)   6/325
25                   mitochondrial outer membrane (GO:0005741)   4/126
26                   platelet dense granule lumen (GO:0031089)    2/14
27                                        vesicle (GO:0031982)   5/226
28                          endolysosome membrane (GO:0036020)    2/17
29                    basolateral plasma membrane (GO:0016323)   4/151
30                   cytoplasmic vesicle membrane (GO:0030659)   6/380
31                         platelet dense granule (GO:0042827)    2/21
32                                lysosomal lumen (GO:0043202)    3/86
33                                   endolysosome (GO:0036019)    2/25
34                        secretory granule lumen (GO:0034774)   5/316
35                          brush border membrane (GO:0031526)    2/37
36                         mitochondrial envelope (GO:0005740)   3/127
37       extracellular membrane-bounded organelle (GO:0065010)    2/56
38                          extracellular vesicle (GO:1903561)    2/59
39                                 vacuolar lumen (GO:0005775)   3/161
40                                        caveola (GO:0005901)    2/60
   Adjusted.P.value
1      5.261200e-24
2      6.209923e-24
3      9.203512e-21
4      9.203512e-21
5      1.246888e-10
6      7.731648e-08
7      1.321996e-07
8      6.153361e-07
9      8.075627e-07
10     1.200431e-06
11     6.211359e-06
12     7.353950e-06
13     2.938912e-05
14     2.938912e-05
15     7.671871e-05
16     9.509306e-05
17     1.065748e-04
18     1.698308e-04
19     2.574496e-04
20     2.574496e-04
21     6.432532e-04
22     8.164449e-04
23     1.420908e-03
24     3.827987e-03
25     3.898444e-03
26     4.116966e-03
27     4.199032e-03
28     5.675277e-03
29     6.551600e-03
30     6.795770e-03
31     7.845057e-03
32     1.043474e-02
33     1.043474e-02
34     1.423039e-02
35     2.126720e-02
36     2.763580e-02
37     4.460396e-02
38     4.700422e-02
39     4.700422e-02
40     4.700422e-02
                                                                                Genes
1                  CETP;APOC2;APOH;APOC1;APOA2;APOA1;APOC3;LCAT;APOA4;APOE;APOA5;PLTP
2                            APOC2;APOH;APOC1;APOA2;APOA1;APOC3;APOA4;APOE;APOB;APOA5
3                            APOC2;APOH;APOC1;APOA2;APOA1;APOC3;APOA4;APOE;APOB;APOA5
4                            APOC2;APOH;APOC1;APOA2;APOA1;APOC3;APOA4;APOE;APOB;APOA5
5          LRP1;SORT1;APOA2;PCSK9;APOA1;APOC3;APOA4;APOC2;LIPG;APOE;LDLRAP1;APOB;LDLR
6                                                               APOC2;APOE;APOB;APOA5
7                                                             APOC2;APOA2;APOA1;APOC3
8                            LRPAP1;LIPC;MTTP;APOA2;PCSK9;APOA1;APOA4;APOE;APOB;APOA5
9                                        SCARB1;LRP1;CD36;LRP2;APOE;LDLRAP1;APOB;LDLR
10 ABCA1;STARD3;HMGCR;CYP7A1;FADS2;NCEH1;SOAT1;VAPA;SOAT2;VAPB;DHCR7;APOB;FADS1;LPIN3
11                       SCARB1;STARD3;NPC1;LRP1;NPC2;SORT1;PCSK9;LRP2;APOB;LIPA;LDLR
12                                        SCARB1;NPC1;NPC2;SORT1;PCSK9;LRP2;LIPA;LDLR
13                                             ABCA1;SCARB1;LRP1;APOA1;CD36;APOE;APOB
14                                                        LRP2;APOE;LDLRAP1;APOB;LDLR
15                                                        LRP2;APOE;LDLRAP1;APOB;LDLR
16                                                        LRP2;APOE;LDLRAP1;APOB;LDLR
17                                       SCARB1;STARD3;NPC1;LRP1;VAPA;PCSK9;LRP2;LDLR
18              CYP27A1;LIPC;ALDH2;MTTP;APOA2;PCSK9;APOA1;APOA4;APOE;APOB;APOA5;KPNB1
19                                  ITIH4;APOH;ANGPTL3;APOA1;APOC3;APOA4;ANGPTL4;APOE
20                                                                    APOA1;APOE;APOB
21                                                       VDAC3;TSPO;VDAC2;VDAC1;DHCR7
22                                                                        ABCG8;ABCG5
23                                                 SCARB1;STARD3;NPC1;LRP1;PCSK9;LRP2
24                                                STARD3;SORT1;PCSK9;ABCB11;APOB;LDLR
25                                                             VDAC3;TSPO;VDAC2;VDAC1
26                                                                         ITIH4;APOH
27                                                         ABCA1;CETP;VAPA;APOA1;APOE
28                                                                         PCSK9;LDLR
29                                                              LRP1;MTTP;ABCB11;LDLR
30                                                   SCARB1;LRP1;SORT1;CD36;APOB;LDLR
31                                                                         ITIH4;APOH
32                                                                     NPC2;APOB;LIPA
33                                                                         PCSK9;LDLR
34                                                        ITIH4;NPC2;APOH;APOA1;KPNB1
35                                                                          LRP2;CD36
36                                                                   STAR;VDAC2;VDAC1
37                                                                         APOA1;APOE
38                                                                         APOA1;APOE
39                                                                     NPC2;APOB;LIPA
40                                                                        SCARB1;CD36
[1] &quot;GO_Molecular_Function_2021&quot;
                                                                                                                                                                                Term
1                                                                                                                                                   cholesterol binding (GO:0015485)
2                                                                                                                                                        sterol binding (GO:0032934)
3                                                                                                                                         cholesterol transfer activity (GO:0120020)
4                                                                                                                                              sterol transfer activity (GO:0120015)
5                                                                                                                                 lipoprotein particle receptor binding (GO:0070325)
6                                                                                                       phosphatidylcholine-sterol O-acyltransferase activator activity (GO:0060228)
7                                                                                                                                          lipoprotein particle binding (GO:0071813)
8                                                                                                                              low-density lipoprotein particle binding (GO:0030169)
9                                                                                                                                             lipase inhibitor activity (GO:0055102)
10                                                                                                                    low-density lipoprotein particle receptor binding (GO:0050750)
11                                                                                                                                          lipoprotein lipase activity (GO:0004465)
12                                                                                                                                                 amyloid-beta binding (GO:0001540)
13                                                                                                                                         triglyceride lipase activity (GO:0004806)
14                                                                                                                                                      lipase activity (GO:0016298)
15                                                                                                                                                       lipase binding (GO:0035473)
16                                                                                                                                           apolipoprotein A-I binding (GO:0034186)
17                                                                                                                                      apolipoprotein receptor binding (GO:0034190)
18                                                                                                                                            phospholipase A1 activity (GO:0008970)
19                                                                                                                                            lipase activator activity (GO:0060229)
20                                                                                                                                  carboxylic ester hydrolase activity (GO:0052689)
21                                                                                                                                 voltage-gated anion channel activity (GO:0008308)
22                                                                                                                                   voltage-gated ion channel activity (GO:0005244)
23                                                                                                                             phosphatidylcholine transporter activity (GO:0008525)
24                                                                                                                                  protein heterodimerization activity (GO:0046982)
25                                                                                                                                phosphatidylcholine transfer activity (GO:0120019)
26                                                                                                                                               phospholipase activity (GO:0004620)
27                                                                                                                                           O-acyltransferase activity (GO:0008374)
28                                                                                                                            high-density lipoprotein particle binding (GO:0008035)
29                                                                                                                                           ceramide transfer activity (GO:0120017)
30                                                                                                                                         clathrin heavy chain binding (GO:0032050)
31                                                                                                                                     phospholipase inhibitor activity (GO:0004859)
32                                                                                                                                    protein homodimerization activity (GO:0042803)
33                                                                                                                                               anion channel activity (GO:0005253)
34                                                                                                                                       phospholipid transfer activity (GO:0120014)
35 oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen, NAD(P)H as one donor, and incorporation of one atom of oxygen (GO:0016709)
36                                                                                                                                         steroid hydroxylase activity (GO:0008395)
37                                                                                                                                                         NADP binding (GO:0050661)
38                                                                                                                                         peptidase inhibitor activity (GO:0030414)
39                                                                                                                                     endopeptidase regulator activity (GO:0061135)
   Overlap Adjusted.P.value
1    17/50     2.288174e-28
2    17/60     4.390322e-27
3    11/18     5.744987e-22
4    11/19     1.020647e-21
5    10/28     4.677379e-17
6      6/6     3.484605e-14
7     8/24     2.055320e-13
8     6/17     3.140441e-10
9     5/10     1.805253e-09
10    6/23     2.016360e-09
11     4/5     9.113232e-09
12    7/80     1.430772e-07
13    5/23     1.612041e-07
14    6/49     1.859889e-07
15     3/5     3.788104e-06
16     3/5     3.788104e-06
17     3/6     7.112969e-06
18    3/10     3.991032e-05
19    3/12     6.897596e-05
20    5/96     1.501407e-04
21    3/16     1.501407e-04
22    3/16     1.501407e-04
23    3/18     2.082322e-04
24   6/188     3.016202e-04
25     2/5     7.035280e-04
26    4/73     7.035280e-04
27    3/30     8.567836e-04
28     2/6     9.653527e-04
29     2/8     1.732107e-03
30     2/9     2.147965e-03
31    2/10     2.592555e-03
32   8/636     7.345938e-03
33    3/68     7.879809e-03
34    2/22     1.181407e-02
35    2/36     2.870121e-02
36    2/36     2.870121e-02
37    2/36     2.870121e-02
38    2/40     3.429432e-02
39    2/46     4.375411e-02
                                                                                                Genes
1  ABCA1;STARD3;CETP;OSBPL5;APOA2;APOA1;APOC3;APOA4;APOA5;NPC1;SOAT1;STAR;NPC2;SOAT2;TSPO;VDAC2;VDAC1
2  ABCA1;STARD3;CETP;OSBPL5;APOA2;APOA1;APOC3;APOA4;APOA5;NPC1;SOAT1;STAR;NPC2;SOAT2;TSPO;VDAC2;VDAC1
3                                        ABCA1;ABCG8;CETP;ABCG5;NPC2;MTTP;APOA2;APOA1;APOA4;APOB;PLTP
4                                        ABCA1;ABCG8;CETP;ABCG5;NPC2;MTTP;APOA2;APOA1;APOA4;APOB;PLTP
5                                         LRPAP1;LRP1;APOA2;PCSK9;APOA1;APOC3;APOE;APOB;LDLRAP1;APOA5
6                                                                  APOC1;APOA2;APOA1;APOA4;APOE;APOA5
7                                                          SCARB1;LIPC;APOA2;LPL;PCSK9;CD36;LDLR;PLTP
8                                                                    SCARB1;LIPC;PCSK9;CD36;LDLR;PLTP
9                                                                     APOC2;APOC1;ANGPTL3;APOA2;APOC3
10                                                               LRPAP1;PCSK9;APOE;APOB;LDLRAP1;APOA5
11                                                                                 LIPC;LIPI;LIPG;LPL
12                                                           LRPAP1;LRP1;APOA1;CD36;APOE;LDLRAP1;LDLR
13                                                                            LIPC;LIPI;LIPG;LCAT;LPL
14                                                                       LIPC;LIPI;LIPG;LCAT;LPL;LIPA
15                                                                                  LRPAP1;APOB;APOA5
16                                                                                  ABCA1;SCARB1;LCAT
17                                                                                  APOA2;APOA1;PCSK9
18                                                                                      LIPC;LIPG;LPL
19                                                                                   APOC2;APOH;APOA5
20                                                                            LIPC;LIPG;LPL;LCAT;LIPA
21                                                                                  VDAC3;VDAC2;VDAC1
22                                                                                  VDAC3;VDAC2;VDAC1
23                                                                                    ABCA1;MTTP;PLTP
24                                                                   ABCG8;ABCG5;VAPA;VAPB;MTTP;APOA2
25                                                                                          MTTP;PLTP
26                                                                                 LIPC;LIPI;LIPG;LPL
27                                                                                   SOAT1;SOAT2;LCAT
28                                                                                         APOA2;PLTP
29                                                                                          MTTP;PLTP
30                                                                                          LRP1;LDLR
31                                                                                      APOC1;ANGPTL3
32                                                         GHR;STARD3;VAPB;EPHX2;APOA2;LPL;APOA4;APOE
33                                                                                  VDAC3;VDAC2;VDAC1
34                                                                                          MTTP;PLTP
35                                                                                     CYP27A1;CYP7A1
36                                                                                     CYP27A1;CYP7A1
37                                                                                        HMGCR;DHCR7
38                                                                                          ITIH4;LPA
39                                                                                          ITIH4;LPA</code></pre>
<pre class="r"><code>GO_known_annotations &lt;- do.call(rbind, GO_enrichment)
GO_known_annotations &lt;- GO_known_annotations[GO_known_annotations$Adjusted.P.value&lt;0.05,]

#GO enrichment analysis for cTWAS genes

genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.8]
GO_enrichment &lt;- enrichr(genes, dbs)</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.</code></pre>
<pre class="r"><code>GO_ctwas_genes &lt;- do.call(rbind, GO_enrichment)

#optionally subset to only significant GO terms
#GO_ctwas_genes &lt;- GO_ctwas_genes[GO_ctwas_genes$Adjusted.P.value&lt;0.05,]

#identify cTWAS genes in silver standard enriched GO terms
GO_ctwas_genes &lt;- GO_ctwas_genes[GO_ctwas_genes$Term %in% GO_known_annotations$Term,]

overlap_genes &lt;- lapply(GO_ctwas_genes$Genes, function(x){unlist(strsplit(x, &quot;;&quot;))})
overlap_genes &lt;- -sort(-table(unlist(overlap_genes)))

#ctwas genes in silver standard enriched GO terms, not already in silver standard
overlap_genes[!(names(overlap_genes) %in% known_annotations)]</code></pre>
<pre><code>
   CAV1     BOK STK11IP     PGP   PSMB7   SP100   AKAP6    GNB4 SEC23IP   MTSS1 
     31       9       5       4       4       4       3       3       3       1 
 NKX2-5 
      1 </code></pre>
<pre class="r"><code>save(overlap_genes, file=paste0(results_dir, &quot;/overlap_genes.Rd&quot;))
load(paste0(results_dir, &quot;/overlap_genes.Rd&quot;))

overlap_genes &lt;- overlap_genes[!(names(overlap_genes) %in% known_annotations)]
overlap_genes</code></pre>
<pre><code>
   CAV1     BOK STK11IP     PGP   PSMB7   SP100   AKAP6    GNB4 SEC23IP   MTSS1 
     31       9       5       4       4       4       3       3       3       1 
 NKX2-5 
      1 </code></pre>
<pre class="r"><code>overlap_genes &lt;- names(overlap_genes)
#ctwas_gene_res[ctwas_gene_res$genename %in% overlap_genes, report_cols,]</code></pre>
</div>
<div id="results-for-paper" class="section level2">
<h2>Results for Paper</h2>
<pre class="r"><code>out_table &lt;- ctwas_gene_res

report_cols &lt;- report_cols[!(report_cols %in% c(&quot;mu2&quot;, &quot;PVE&quot;))]
report_cols &lt;- c(report_cols,&quot;silver&quot;,&quot;GO_overlap_silver&quot;, &quot;bystander&quot;)

#reload silver standard genes
known_annotations &lt;- read_xlsx(&quot;data/summary_known_genes_annotations.xlsx&quot;, sheet=&quot;LDL&quot;)</code></pre>
<pre><code>New names:
* `` -&gt; ...4
* `` -&gt; ...5</code></pre>
<pre class="r"><code>known_annotations &lt;- unique(known_annotations$`Gene Symbol`)

out_table$silver &lt;- F
out_table$silver[out_table$genename %in% known_annotations] &lt;- T

library(biomaRt)
library(GenomicRanges)</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    first, rename</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    reduce</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>ensembl &lt;- useEnsembl(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
G_list &lt;- getBM(filters= &quot;chromosome_name&quot;, attributes= c(&quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;end_position&quot;,&quot;gene_biotype&quot;), values=1:22, mart=ensembl)
G_list &lt;- G_list[G_list$hgnc_symbol!=&quot;&quot;,]
G_list &lt;- G_list[G_list$gene_biotype %in% c(&quot;protein_coding&quot;,&quot;lncRNA&quot;),]
G_list$start &lt;- G_list$start_position
G_list$end &lt;- G_list$end_position
G_list_granges &lt;- makeGRangesFromDataFrame(G_list, keep.extra.columns=T)

known_annotations_positions &lt;- G_list[G_list$hgnc_symbol %in% known_annotations,]
half_window &lt;- 1000000
known_annotations_positions$start &lt;- known_annotations_positions$start_position - half_window
known_annotations_positions$end &lt;- known_annotations_positions$end_position + half_window
known_annotations_positions$start[known_annotations_positions$start&lt;1] &lt;- 1
known_annotations_granges &lt;- makeGRangesFromDataFrame(known_annotations_positions, keep.extra.columns=T)

bystanders_extended &lt;- findOverlaps(known_annotations_granges,G_list_granges)
bystanders_extended &lt;- unique(subjectHits(bystanders_extended))
bystanders_extended &lt;- G_list$hgnc_symbol[bystanders_extended]
bystanders_extended &lt;- unique(bystanders_extended[!(bystanders_extended %in% known_annotations)])

save(bystanders_extended, file=paste0(results_dir, &quot;/bystanders_extended.Rd&quot;))

load(paste0(results_dir, &quot;/bystanders_extended.Rd&quot;))

#add extended bystanders list to output
out_table$bystander &lt;- F
out_table$bystander[out_table$genename %in% bystanders_extended] &lt;- T

#reload GO overlaps with silver standard
load(paste0(results_dir, &quot;/overlap_genes.Rd&quot;))

out_table$GO_overlap_silver &lt;- NA
out_table$GO_overlap_silver[out_table$susie_pip&gt;0.8] &lt;- 0

for (i in names(overlap_genes)){
  out_table$GO_overlap_silver[out_table$genename==i] &lt;- overlap_genes[i]
}</code></pre>
<div id="ctwas-identifies-high-confidence-liver-genes-associated-with-ldl-cholesterol" class="section level3">
<h3>cTWAS identifies high-confidence liver genes associated with LDL cholesterol</h3>
<pre class="r"><code>full.gene.pip.summary &lt;- data.frame(gene_name = ctwas_gene_res$genename, 
                                    gene_pip = ctwas_gene_res$susie_pip, 
                                    gene_id = ctwas_gene_res$id, 
                                    chr = as.integer(ctwas_gene_res$chrom),
                                    start = ctwas_gene_res$pos / 1e3,
                                    is_highlight = F, stringsAsFactors = F) %&gt;% as_tibble()
full.gene.pip.summary$is_highlight &lt;- full.gene.pip.summary$gene_pip &gt; 0.80

don &lt;- full.gene.pip.summary %&gt;% 
  
  # Compute chromosome size
  group_by(chr) %&gt;% 
  summarise(chr_len=max(start)) %&gt;% 
  
  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %&gt;%
  dplyr::select(-chr_len) %&gt;%
  
  # Add this info to the initial dataset
  left_join(full.gene.pip.summary, ., by=c(&quot;chr&quot;=&quot;chr&quot;)) %&gt;%
  
  # Add a cumulative position of each SNP
  arrange(chr, start) %&gt;%
  mutate( BPcum=start+tot)

axisdf &lt;- don %&gt;% group_by(chr) %&gt;% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

x_axis_labels &lt;- axisdf$chr
x_axis_labels[seq(1,21,2)] &lt;- &quot;&quot;

ggplot(don, aes(x=BPcum, y=gene_pip)) +
  
  # Show all points
  ggrastr::geom_point_rast(aes(color=as.factor(chr)), size=2) +
  scale_color_manual(values = rep(c(&quot;grey&quot;, &quot;skyblue&quot;), 22 )) +
  
  # custom X axis:
  # scale_x_continuous(label = axisdf$chr, 
  #                    breaks= axisdf$center,
  #                    guide = guide_axis(n.dodge = 2)) +
  scale_x_continuous(label = x_axis_labels,
                     breaks = axisdf$center) +
  
  scale_y_continuous(expand = c(0, 0), limits = c(0,1.25), breaks=(1:5)*0.2, minor_breaks=(1:10)*0.1) + # remove space between plot area and x axis
  
  # Add highlighted points
  ggrastr::geom_point_rast(data=subset(don, is_highlight==T), color=&quot;orange&quot;, size=2) +
  
  # Add label using ggrepel to avoid overlapping
  ggrepel::geom_label_repel(data=subset(don, is_highlight==T), 
                            aes(label=gene_name), 
                            size=4,
                            min.segment.length = 0, 
                            label.size = NA,
                            fill = alpha(c(&quot;white&quot;),0)) +
  
  # Custom the theme:
  theme_bw() +
  theme( 
    text = element_text(size = 14),
    legend.position=&quot;none&quot;,
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  ) +
  xlab(&quot;Chromosome&quot;) + 
  ylab(&quot;cTWAS PIP&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-28-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#number of SNPs at PIP&gt;0.8 threshold
sum(out_table$susie_pip&gt;0.8)</code></pre>
<pre><code>[1] 32</code></pre>
<pre class="r"><code>#number of SNPs at PIP&gt;0.5 threshold
sum(out_table$susie_pip&gt;0.5)</code></pre>
<pre><code>[1] 68</code></pre>
<pre class="r"><code>#genes with PIP&gt;0.8
head(out_table[order(-out_table$susie_pip),report_cols], sum(out_table$susie_pip&gt;0.8))</code></pre>
<pre><code>          genename region_tag susie_pip         z num_eqtl silver
2293          CAV1       7_70 1.0000000 15.567870        3  FALSE
3275         PRRX1       1_84 0.9998808 14.667578        2  FALSE
4009           DEK       6_14 0.9925129 -9.000000        1  FALSE
3527         CCND2       12_4 0.9906174 -5.283784        1  FALSE
1310           PXN      12_75 0.9826498 -5.328302        1  FALSE
12836 RP11-325L7.2       5_82 0.9826020 12.356322        1  FALSE
9257         AGAP5      10_49 0.9779202 11.518590        2  FALSE
3523         KLF12      13_36 0.9770829 -5.072464        1  FALSE
6621         AKAP6       14_8 0.9721549 -9.197368        1  FALSE
6914          JAM2       21_9 0.9640543  4.563232        2  FALSE
9639         DLEU1      13_21 0.9586468  4.697095        1  FALSE
11818         DPF3      14_34 0.9574312  6.264960        3  FALSE
10521       FAM43A      3_120 0.9569176 -5.487179        1  FALSE
2444       SEC23IP      10_74 0.9517809 -4.565228        2  FALSE
13075    LINC01629      14_36 0.9471613 -5.695652        1  FALSE
2138           AES       19_4 0.9403219  4.182804        3  FALSE
4658        POPDC3       6_70 0.9252511 -4.758170        1  FALSE
10290       NKX2-5      5_103 0.9228745 -9.391892        1  FALSE
7515       TNFSF13       17_7 0.9135188 -5.883117        1  FALSE
5185          GYPC       2_74 0.9068050 -6.380531        1  FALSE
8248         CMTM5       14_3 0.9067300 -5.472727        1  FALSE
10548       SCN10A       3_28 0.8867624 -8.814286        1  FALSE
13967 RP5-890E16.5      17_28 0.8660551 -4.761194        1  FALSE
712          SP100      2_135 0.8658723 -3.671335        2  FALSE
9012         MTSS1       8_82 0.8594108  4.402634        2  FALSE
8992          MURC       9_50 0.8478869  4.911964        2  FALSE
5223         PSMB7       9_64 0.8430652 -4.820896        1  FALSE
6114       STK11IP      2_130 0.8406381 -3.868022        2  FALSE
10416          PGP       16_2 0.8298586  5.943820        1  FALSE
9691           BOK      2_144 0.8255975  3.910125        3  FALSE
8420          MARS      12_36 0.8180336 -3.366197        1  FALSE
3088          GNB4      3_110 0.8097696 -5.583333        1  FALSE
      GO_overlap_silver bystander
2293                 31     FALSE
3275                  0     FALSE
4009                  0     FALSE
3527                  0     FALSE
1310                  0     FALSE
12836                 0     FALSE
9257                  0     FALSE
3523                  0     FALSE
6621                  3     FALSE
6914                  0     FALSE
9639                  0     FALSE
11818                 0     FALSE
10521                 0     FALSE
2444                  3     FALSE
13075                 0     FALSE
2138                  0     FALSE
4658                  0     FALSE
10290                 1     FALSE
7515                  0     FALSE
5185                  0     FALSE
8248                  0     FALSE
10548                 0     FALSE
13967                 0     FALSE
712                   4     FALSE
9012                  1     FALSE
8992                  0     FALSE
5223                  4     FALSE
6114                  5      TRUE
10416                 4     FALSE
9691                  9     FALSE
8420                  0     FALSE
3088                  3     FALSE</code></pre>
<pre class="r"><code>head(out_table[order(-out_table$susie_pip),report_cols[-(7:8)]], sum(out_table$susie_pip&gt;0.8))</code></pre>
<pre><code>          genename region_tag susie_pip         z num_eqtl silver
2293          CAV1       7_70 1.0000000 15.567870        3  FALSE
3275         PRRX1       1_84 0.9998808 14.667578        2  FALSE
4009           DEK       6_14 0.9925129 -9.000000        1  FALSE
3527         CCND2       12_4 0.9906174 -5.283784        1  FALSE
1310           PXN      12_75 0.9826498 -5.328302        1  FALSE
12836 RP11-325L7.2       5_82 0.9826020 12.356322        1  FALSE
9257         AGAP5      10_49 0.9779202 11.518590        2  FALSE
3523         KLF12      13_36 0.9770829 -5.072464        1  FALSE
6621         AKAP6       14_8 0.9721549 -9.197368        1  FALSE
6914          JAM2       21_9 0.9640543  4.563232        2  FALSE
9639         DLEU1      13_21 0.9586468  4.697095        1  FALSE
11818         DPF3      14_34 0.9574312  6.264960        3  FALSE
10521       FAM43A      3_120 0.9569176 -5.487179        1  FALSE
2444       SEC23IP      10_74 0.9517809 -4.565228        2  FALSE
13075    LINC01629      14_36 0.9471613 -5.695652        1  FALSE
2138           AES       19_4 0.9403219  4.182804        3  FALSE
4658        POPDC3       6_70 0.9252511 -4.758170        1  FALSE
10290       NKX2-5      5_103 0.9228745 -9.391892        1  FALSE
7515       TNFSF13       17_7 0.9135188 -5.883117        1  FALSE
5185          GYPC       2_74 0.9068050 -6.380531        1  FALSE
8248         CMTM5       14_3 0.9067300 -5.472727        1  FALSE
10548       SCN10A       3_28 0.8867624 -8.814286        1  FALSE
13967 RP5-890E16.5      17_28 0.8660551 -4.761194        1  FALSE
712          SP100      2_135 0.8658723 -3.671335        2  FALSE
9012         MTSS1       8_82 0.8594108  4.402634        2  FALSE
8992          MURC       9_50 0.8478869  4.911964        2  FALSE
5223         PSMB7       9_64 0.8430652 -4.820896        1  FALSE
6114       STK11IP      2_130 0.8406381 -3.868022        2  FALSE
10416          PGP       16_2 0.8298586  5.943820        1  FALSE
9691           BOK      2_144 0.8255975  3.910125        3  FALSE
8420          MARS      12_36 0.8180336 -3.366197        1  FALSE
3088          GNB4      3_110 0.8097696 -5.583333        1  FALSE</code></pre>
<pre class="r"><code>head(out_table[order(-out_table$susie_pip),report_cols[c(1,7:8)]], sum(out_table$susie_pip&gt;0.8))</code></pre>
<pre><code>          genename GO_overlap_silver bystander
2293          CAV1                31     FALSE
3275         PRRX1                 0     FALSE
4009           DEK                 0     FALSE
3527         CCND2                 0     FALSE
1310           PXN                 0     FALSE
12836 RP11-325L7.2                 0     FALSE
9257         AGAP5                 0     FALSE
3523         KLF12                 0     FALSE
6621         AKAP6                 3     FALSE
6914          JAM2                 0     FALSE
9639         DLEU1                 0     FALSE
11818         DPF3                 0     FALSE
10521       FAM43A                 0     FALSE
2444       SEC23IP                 3     FALSE
13075    LINC01629                 0     FALSE
2138           AES                 0     FALSE
4658        POPDC3                 0     FALSE
10290       NKX2-5                 1     FALSE
7515       TNFSF13                 0     FALSE
5185          GYPC                 0     FALSE
8248         CMTM5                 0     FALSE
10548       SCN10A                 0     FALSE
13967 RP5-890E16.5                 0     FALSE
712          SP100                 4     FALSE
9012         MTSS1                 1     FALSE
8992          MURC                 0     FALSE
5223         PSMB7                 4     FALSE
6114       STK11IP                 5      TRUE
10416          PGP                 4     FALSE
9691           BOK                 9     FALSE
8420          MARS                 0     FALSE
3088          GNB4                 3     FALSE</code></pre>
</div>
<div id="high-confidence-ctwas-genes-are-enriched-for-cholesterol-related-go-terms" class="section level3">
<h3>High-confidence cTWAS genes are enriched for cholesterol-related GO terms</h3>
<pre class="r"><code>#GO enrichment analysis
library(enrichR)

dbs &lt;- c(&quot;GO_Biological_Process_2021&quot;, &quot;GO_Cellular_Component_2021&quot;, &quot;GO_Molecular_Function_2021&quot;)
genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.8]

#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 32</code></pre>
<pre class="r"><code>GO_enrichment &lt;- enrichr(genes, dbs)</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.</code></pre>
<pre class="r"><code>db &lt;- dbs[1]
#for (db in dbs){
  print(db)</code></pre>
<pre><code>[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<pre class="r"><code>  df &lt;- GO_enrichment[[db]]
  print(plotEnrich(GO_enrichment[[db]]))</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-30-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  df &lt;- df[df$Adjusted.P.value&lt;0.05,c(&quot;Term&quot;, &quot;Overlap&quot;, &quot;Adjusted.P.value&quot;, &quot;Genes&quot;)]
  print(df)</code></pre>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>#}</code></pre>
</div>
<div id="ctwas-avoids-false-positives-when-multiple-genes-are-in-a-region" class="section level3">
<h3>cTWAS avoids false positives when multiple genes are in a region</h3>
<p>TNKS is a silver standard (assumed true positive gene) that is correctly detected. The bystander gene RP11-115J16.2 is significant using TWAS but has low PIP using cTWAS.</p>
<pre class="r"><code>#TNKS gene
locus_plot4(&quot;8_12&quot;, label=&quot;cTWAS&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-31-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>out_table[out_table$region_tag==&quot;8_12&quot;,report_cols[-(7:8)]]</code></pre>
<pre><code>     genename region_tag  susie_pip          z num_eqtl silver
9342     TNKS       8_12 0.04205266 -0.6212121        1   TRUE</code></pre>
<pre class="r"><code>out_table[out_table$region_tag==&quot;8_12&quot;,report_cols[c(1,7:8)]]</code></pre>
<pre><code>     genename GO_overlap_silver bystander
9342     TNKS                NA     FALSE</code></pre>
<p>FADS1 is a silver standard gene (assumed true positive gene) that is correctly detected. There are 5 significant TWAS genes at this locus, including FADS2, another silver standard gene. FADS2 is not detected due to its high LD with FADS1. The remaining 3 bystander genes at this locus have low PIP using cTWAS.</p>
<pre class="r"><code>#FADS1 gene
locus_plot3(&quot;11_34&quot;, focus=&quot;FADS1&quot;)

out_table[out_table$region_tag==&quot;11_34&quot;,report_cols[-(7:8)]]
out_table[out_table$region_tag==&quot;11_34&quot;,report_cols[c(1,7:8)]]

#number of significant TWAS genes at this locus
sum(abs(out_table$z[out_table$region_tag==&quot;11_34&quot;])&gt;sig_thresh)</code></pre>
</div>
<div id="ctwas-avoids-false-positives-when-snps-in-a-region-are-considerably-more-significant" class="section level3">
<h3>cTWAS avoids false positives when SNPs in a region are (considerably) more significant</h3>
<p>POLK is a gene that is significant using TWAS but not detected using TWAS. cTWAS places a high posterior probability on SNPs are this locus. OpenTargets suggets that the causal gene at this locus is <a href="https://genetics.opentargets.org/study-locus/GCST010245/5_75360714_T_C">HMGCR</a> (note: different GWAS, similar population), which is not imputed in our dataset. cTWAS selected the variants at this locus because the causal gene is not imputed. Note that <a href="https://pubmed.ncbi.nlm.nih.gov/33020666/">MR-JTI</a> claims POLK is causal using their method, and their paper includes a discussion of its potential relevance to LDL.</p>
<pre class="r"><code>locus_plot(&quot;5_45&quot;, label=&quot;TWAS&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-33-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#locus_plot(&quot;5_45&quot;, label=&quot;TWAS&quot;, rerun_ctwas = T)

out_table[out_table$region_tag==&quot;5_45&quot;,report_cols[-(7:8)]]</code></pre>
<pre><code>      genename region_tag  susie_pip           z num_eqtl silver
2987     PDE8B       5_45 0.32802254 -3.50495050        1  FALSE
4717     ZBED3       5_45 0.03451392 -0.18829320        2  FALSE
6215     CRHBP       5_45 0.03798827 -0.53072626        1  FALSE
7949     F2RL2       5_45 0.03349611  0.06301653        3  FALSE
7954     F2RL1       5_45 0.03724985 -0.47727864        2  FALSE
7955     AGGF1       5_45 0.03590199  0.41340782        1  FALSE
7956     WDR41       5_45 0.14016344  2.12999618        3  FALSE
9106      TBCA       5_45 0.03870813 -0.60273973        1  FALSE
10108      F2R       5_45 0.03508658 -0.41198282        3  FALSE</code></pre>
<pre class="r"><code>out_table[out_table$region_tag==&quot;5_45&quot;,report_cols[c(1,7:8)]]</code></pre>
<pre><code>      genename GO_overlap_silver bystander
2987     PDE8B                NA     FALSE
4717     ZBED3                NA     FALSE
6215     CRHBP                NA     FALSE
7949     F2RL2                NA     FALSE
7954     F2RL1                NA     FALSE
7955     AGGF1                NA     FALSE
7956     WDR41                NA     FALSE
9106      TBCA                NA     FALSE
10108      F2R                NA     FALSE</code></pre>
</div>
<div id="ctwas-is-more-precise-than-twas-in-distinguishing-silver-standard-and-bystander-genes" class="section level3">
<h3>cTWAS is more precise than TWAS in distinguishing silver standard and bystander genes</h3>
<pre class="r"><code>load(paste0(results_dir, &quot;/known_annotations.Rd&quot;))
load(paste0(results_dir, &quot;/bystanders.Rd&quot;))

#remove genes without imputed expression from bystander list
unrelated_genes &lt;- unrelated_genes[unrelated_genes %in% ctwas_gene_res$genename]

#subset results to genes in known annotations or bystanders
ctwas_gene_res_subset &lt;- ctwas_gene_res[ctwas_gene_res$genename %in% c(known_annotations, unrelated_genes),]

#assign ctwas and TWAS genes
ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;sig_thresh]

#sensitivity / recall
sensitivity &lt;- rep(NA,2)
names(sensitivity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
sensitivity[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
sensitivity</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#specificity / (1 - False Positive Rate)
specificity &lt;- rep(NA,2)
names(specificity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
specificity[&quot;ctwas&quot;] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity[&quot;TWAS&quot;] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
specificity</code></pre>
<pre><code>    ctwas      TWAS 
0.9983471 0.9966942 </code></pre>
<pre class="r"><code>#precision / PPV / (1 - False Discovery Rate)
precision &lt;- rep(NA,2)
names(precision) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
precision[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(twas_genes)
precision</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#store sensitivity and specificity calculations for plots
sensitivity_plot &lt;- sensitivity
specificity_plot &lt;- specificity

#precision / PPV by PIP threshold
pip_range &lt;- c(0.5, 0.8, 1)
precision_range &lt;- rep(NA, length(pip_range))
number_detected &lt;- rep(NA, length(pip_range))

for (i in 1:length(pip_range)){
  pip_upper &lt;- pip_range[i]

  if (i==1){
    pip_lower &lt;- 0
  } else {
    pip_lower &lt;- pip_range[i-1]
  }
  
  #assign ctwas genes using PIP threshold
  ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;=pip_lower]
  
  number_detected[i] &lt;- length(ctwas_genes)
  precision_range[i] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
}

names(precision_range) &lt;- paste0(&quot;&gt;= &quot;, c(0, pip_range[-length(pip_range)]))

precision_range &lt;- precision_range*100

precision_range &lt;- c(precision_range, precision[&quot;TWAS&quot;]*100)
names(precision_range)[4] &lt;- &quot;TWAS Bonferroni&quot;
number_detected &lt;- c(number_detected, length(twas_genes))

barplot(precision_range, ylim=c(0,100), main=&quot;Precision for Distinguishing Silver Standard and Bystander Genes&quot;, xlab=&quot;PIP Threshold for Detection&quot;, ylab=&quot;% of Detected Genes in Silver Standard&quot;)
abline(h=20, lty=2)
abline(h=40, lty=2)
abline(h=60, lty=2)
abline(h=80, lty=2)
xx &lt;- barplot(precision_range, add=T, col=c(rep(&quot;darkgrey&quot;,3), &quot;white&quot;))
text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, &quot; detected&quot;), pos = 3, cex=0.8)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-34-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#text(x = xx, y = precision_range, label = paste0(round(precision_range,1), &quot;%&quot;), pos = 3, cex=0.8, offset = 1.5)

#false discovery rate by PIP threshold

barplot(100-precision_range, ylim=c(0,100), main=&quot;False Discovery Rate for Distinguishing Silver Standard and Bystander Genes&quot;, xlab=&quot;PIP Threshold for Detection&quot;, ylab=&quot;% Bystanders in Detected Genes&quot;)
abline(h=20, lty=2)
abline(h=40, lty=2)
abline(h=60, lty=2)
abline(h=80, lty=2)
xx &lt;- barplot(100-precision_range, add=T, col=c(rep(&quot;darkgrey&quot;,3), &quot;white&quot;))
text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, &quot; detected&quot;), pos = 3, cex=0.8)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-34-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-2">
Past versions of unnamed-chunk-34-2.png
</button>
</p>
<div id="fig-unnamed-chunk-34-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-34-2.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#text(x = xx, y = precision_range, label = paste0(round(precision_range,1), &quot;%&quot;), pos = 3, cex=0.8, offset = 1.5)</code></pre>
</div>
<div id="undetected-silver-standard-genes-have-low-twas-z-scores-or-stronger-signal-from-nearby-variants" class="section level3">
<h3>Undetected silver standard genes have low TWAS z-scores or stronger signal from nearby variants</h3>
<p>For all 69 silver standard genes, sequentially bin each gene using the following criteria: 1) gene not imputed; 2) gene detected by cTWAS at PIP&gt;0.8; 3) gene insignificant by TWAS; 4) gene nearby a detected silver standard gene; 5) gene nearby a detected bystander gene; 6) gene nearby a detected SNP; 7) inconclusive.</p>
<pre class="r"><code>#reload silver standard genes
known_annotations &lt;- read_xlsx(&quot;data/summary_known_genes_annotations.xlsx&quot;, sheet=&quot;LDL&quot;)</code></pre>
<pre><code>New names:
* `` -&gt; ...4
* `` -&gt; ...5</code></pre>
<pre class="r"><code>known_annotations &lt;- unique(known_annotations$`Gene Symbol`)

#categorize silver standard genes by case
silver_standard_case &lt;- c()
uncertain_regions &lt;- matrix(NA, 0, 2)

for (i in 1:length(known_annotations)){
  current_gene &lt;- known_annotations[i]
  
  if (current_gene %in% ctwas_gene_res$genename) {
    if (ctwas_gene_res$susie_pip[ctwas_gene_res$genename == current_gene] &gt; 0.8){
      silver_standard_case &lt;- c(silver_standard_case, &quot;Detected (PIP &gt; 0.8)&quot;)
    } else {
      if (abs(ctwas_gene_res$z[ctwas_gene_res$genename == current_gene]) &lt; sig_thresh){
        silver_standard_case &lt;- c(silver_standard_case, &quot;Insignificant z-score&quot;)
      } else {
        current_region &lt;- ctwas_gene_res$region_tag[ctwas_gene_res$genename == current_gene]
        current_gene_res &lt;- ctwas_gene_res[ctwas_gene_res$region_tag==current_region,]
        current_snp_res &lt;- ctwas_snp_res[ctwas_snp_res$region_tag==current_region,]
        
        if (any(current_gene_res$susie_pip&gt;0.8)){
          if (any(current_gene_res$genename[current_gene_res$susie_pip&gt;0.8] %in% known_annotations)){
            silver_standard_case &lt;- c(silver_standard_case, &quot;Nearby Silver Standard Gene&quot;)
          } else {
            silver_standard_case &lt;- c(silver_standard_case, &quot;Nearby Bystander Gene&quot;)
          }
        } else {
          #if (any(current_snp_res$susie_pip&gt;0.8)){
          if (sum(current_snp_res$susie_pip)&gt;0.8){
            silver_standard_case &lt;- c(silver_standard_case, &quot;Nearby SNP(s)&quot;)
          } else {
            silver_standard_case &lt;- c(silver_standard_case, &quot;Inconclusive&quot;)
            
            uncertain_regions &lt;- rbind(uncertain_regions, c(current_gene, ctwas_gene_res$region_tag[ctwas_gene_res$genename == current_gene]))
            
            print(c(current_gene, ctwas_gene_res$region_tag[ctwas_gene_res$genename == current_gene]))
          }
        }
      }
    }
  } else {
    silver_standard_case &lt;- c(silver_standard_case, &quot;Not Imputed&quot;)
  }
}
names(silver_standard_case) &lt;- known_annotations

#table of outcomes for silver standard genes
-sort(-table(silver_standard_case))</code></pre>
<pre><code>silver_standard_case
Insignificant z-score           Not Imputed 
                   44                    25 </code></pre>
<pre class="r"><code>#show inconclusive genes
silver_standard_case[silver_standard_case==&quot;Inconclusive&quot;]</code></pre>
<pre><code>named character(0)</code></pre>
<pre class="r"><code># for (i in 1:nrow(uncertain_regions)){
#   locus_plot3(uncertain_regions[i,2], focus=uncertain_regions[i,1])
# }

#pie chart of outcomes for silver standard genes
df &lt;- data.frame(-sort(-table(silver_standard_case)))
names(df) &lt;- c(&quot;Outcome&quot;, &quot;Frequency&quot;)
#df &lt;- df[df$Outcome!=&quot;Not Imputed&quot;,] #exclude genes not imputed
df$Outcome &lt;- droplevels(df$Outcome) #exclude genes not imputed

bp&lt;- ggplot(df, aes(x=Outcome, y=Frequency, fill=Outcome)) + geom_bar(width = 1, stat = &quot;identity&quot;, position=position_dodge()) + 
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + theme(legend.position = &quot;none&quot;)
bp</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pie &lt;- ggplot(df, aes(x=&quot;&quot;, y=Frequency, fill=Outcome)) + geom_bar(width = 1, stat = &quot;identity&quot;)
pie &lt;- pie + coord_polar(&quot;y&quot;, start=0) + theme_minimal() + theme(axis.title.y=element_blank())
pie</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-2">
Past versions of unnamed-chunk-35-2.png
</button>
</p>
<div id="fig-unnamed-chunk-35-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-2.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>locus_plot3(focus=&quot;KPNB1&quot;, region_tag=&quot;17_27&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-3">
Past versions of unnamed-chunk-35-3.png
</button>
</p>
<div id="fig-unnamed-chunk-35-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-3.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>locus_plot3(focus=&quot;LPIN3&quot;, region_tag=&quot;20_25&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-4">
Past versions of unnamed-chunk-35-4.png
</button>
</p>
<div id="fig-unnamed-chunk-35-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-4.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>locus_plot3(focus=&quot;LIPC&quot;, region_tag=&quot;15_26&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-5">
Past versions of unnamed-chunk-35-5.png
</button>
</p>
<div id="fig-unnamed-chunk-35-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-35-5.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="some-ctwas-genes-share-biological-characteristics-with-silver-standard-genes" class="section level3">
<h3>Some cTWAS genes share biological characteristics with silver standard genes</h3>
<p>Perform GO enrichment analysis using silver standard genes. Identify detected cTWAS genes not already in silver standard that are also members of these GO terms.</p>
<pre class="r"><code>#reload silver standard genes
known_annotations &lt;- read_xlsx(&quot;data/summary_known_genes_annotations.xlsx&quot;, sheet=&quot;LDL&quot;)</code></pre>
<pre><code>New names:
* `` -&gt; ...4
* `` -&gt; ...5</code></pre>
<pre class="r"><code>known_annotations &lt;- unique(known_annotations$`Gene Symbol`)

#GO enrichment analysis for silver standard genes
dbs &lt;- c(&quot;GO_Biological_Process_2021&quot;, &quot;GO_Cellular_Component_2021&quot;, &quot;GO_Molecular_Function_2021&quot;)
genes &lt;- known_annotations
GO_enrichment &lt;- enrichr(genes, dbs)</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.</code></pre>
<pre class="r"><code># for (db in dbs){
#   print(db)
#   df &lt;- GO_enrichment[[db]]
#   df &lt;- df[df$Adjusted.P.value&lt;0.05,c(&quot;Term&quot;, &quot;Overlap&quot;, &quot;Adjusted.P.value&quot;, &quot;Genes&quot;)]
#   plotEnrich(GO_enrichment[[db]])
#   print(df)
# }

GO_known_annotations &lt;- do.call(rbind, GO_enrichment)
GO_known_annotations &lt;- GO_known_annotations[GO_known_annotations$Adjusted.P.value&lt;0.05,]

#GO enrichment analysis for cTWAS genes

genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.8]
GO_enrichment &lt;- enrichr(genes, dbs)</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.</code></pre>
<pre class="r"><code>GO_ctwas_genes &lt;- do.call(rbind, GO_enrichment)

#identify cTWAS genes in silver standard enriched GO terms
GO_ctwas_genes &lt;- GO_ctwas_genes[GO_ctwas_genes$Term %in% GO_known_annotations$Term,]

GO_ctwas_genes_byterms &lt;- as.data.frame(matrix(NA, 0, 2))

for (i in 1:nrow(GO_ctwas_genes)){
  for (j in unlist(strsplit(GO_ctwas_genes$Genes[i], split=&quot;;&quot;))){
    GO_ctwas_genes_byterms&lt;- rbind(GO_ctwas_genes_byterms, c(j, GO_ctwas_genes$Term[i]))
  }
  colnames(GO_ctwas_genes_byterms) &lt;- c(&quot;Gene&quot;, &quot;GO_term&quot;)
}</code></pre>
<pre><code>Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated

Warning in `[&lt;-.factor`(`*tmp*`, ri, value = &quot;PGP&quot;): invalid factor level, NA
generated</code></pre>
<pre class="r"><code>GO_ctwas_genes_byterms &lt;- GO_ctwas_genes_byterms[order(GO_ctwas_genes_byterms$Gene),]
GO_ctwas_genes_byterms &lt;- GO_ctwas_genes_byterms[!(GO_ctwas_genes_byterms$Gene %in% known_annotations),]

#detected cTWAS genes (not on silver standard) that overlap with GO terms enriched for silver standard genes

GO_ctwas_genes_byterms</code></pre>
<pre><code>   Gene                                                           GO_term
1  CAV1 negative regulation of cellular biosynthetic process (GO:0031327)
3  CAV1                                                              &lt;NA&gt;
4  CAV1                                                              &lt;NA&gt;
5  CAV1                                                              &lt;NA&gt;
6  CAV1                                                              &lt;NA&gt;
7  CAV1                                                              &lt;NA&gt;
8  CAV1                                                              &lt;NA&gt;
11 CAV1                                                              &lt;NA&gt;
13 CAV1                                                              &lt;NA&gt;
14 CAV1                                                              &lt;NA&gt;
15 CAV1                                                              &lt;NA&gt;
16 CAV1                                                              &lt;NA&gt;
17 CAV1                                                              &lt;NA&gt;
18 CAV1                                                              &lt;NA&gt;
19 CAV1                                                              &lt;NA&gt;
20 CAV1                                                              &lt;NA&gt;
24 CAV1                                                              &lt;NA&gt;
26 CAV1                                                              &lt;NA&gt;
27 CAV1                                                              &lt;NA&gt;
28 CAV1                                                              &lt;NA&gt;
31 CAV1                                                              &lt;NA&gt;
35 CAV1                                                              &lt;NA&gt;
38 CAV1                                                              &lt;NA&gt;
40 CAV1                                                              &lt;NA&gt;
42 CAV1                                                              &lt;NA&gt;
48 CAV1                                                              &lt;NA&gt;
57 CAV1                                                              &lt;NA&gt;
59 CAV1                                                              &lt;NA&gt;
61 CAV1                                                              &lt;NA&gt;
64 CAV1                                                              &lt;NA&gt;
65 CAV1                                                              &lt;NA&gt;
2  &lt;NA&gt; negative regulation of cellular biosynthetic process (GO:0031327)
9  &lt;NA&gt;                                                              &lt;NA&gt;
10 &lt;NA&gt;                                                              &lt;NA&gt;
12 &lt;NA&gt;                                                              &lt;NA&gt;
21 &lt;NA&gt;                                                              &lt;NA&gt;
22 &lt;NA&gt;                                                              &lt;NA&gt;
23 &lt;NA&gt;                                                              &lt;NA&gt;
25 &lt;NA&gt;                                                              &lt;NA&gt;
29 &lt;NA&gt;                                                              &lt;NA&gt;
30 &lt;NA&gt;                                                              &lt;NA&gt;
32 &lt;NA&gt;                                                              &lt;NA&gt;
33 &lt;NA&gt;                                                              &lt;NA&gt;
34 &lt;NA&gt;                                                              &lt;NA&gt;
36 &lt;NA&gt;                                                              &lt;NA&gt;
37 &lt;NA&gt;                                                              &lt;NA&gt;
39 &lt;NA&gt;                                                              &lt;NA&gt;
41 &lt;NA&gt;                                                              &lt;NA&gt;
43 &lt;NA&gt;                                                              &lt;NA&gt;
44 &lt;NA&gt;                                                              &lt;NA&gt;
45 &lt;NA&gt;                                                              &lt;NA&gt;
46 &lt;NA&gt;                                                              &lt;NA&gt;
47 &lt;NA&gt;                                                              &lt;NA&gt;
49 &lt;NA&gt;                                                              &lt;NA&gt;
50 &lt;NA&gt;                                                              &lt;NA&gt;
51 &lt;NA&gt;                                                              &lt;NA&gt;
52 &lt;NA&gt;                                                              &lt;NA&gt;
53 &lt;NA&gt;                                                              &lt;NA&gt;
54 &lt;NA&gt;                                                              &lt;NA&gt;
55 &lt;NA&gt;                                                              &lt;NA&gt;
56 &lt;NA&gt;                                                              &lt;NA&gt;
58 &lt;NA&gt;                                                              &lt;NA&gt;
60 &lt;NA&gt;                                                              &lt;NA&gt;
62 &lt;NA&gt;                                                              &lt;NA&gt;
63 &lt;NA&gt;                                                              &lt;NA&gt;
66 &lt;NA&gt;                                                              &lt;NA&gt;
67 &lt;NA&gt;                                                              &lt;NA&gt;
68 &lt;NA&gt;                                                              &lt;NA&gt;</code></pre>
</div>
<div id="ctwas-detects-some-genes-that-are-not-significant-using-a-stringent-twas-threshold" class="section level3">
<h3>cTWAS detects some genes that are not significant using a stringent TWAS threshold</h3>
<p>TTC39B is a member of the Dyslipidaemia term in the disease_GLAD4U. This gene was not included in our silver standard. This gene is not significant using TWAS but is detected by cTWAS.</p>
<pre class="r"><code>locus_plot3(focus=&quot;TTC39B&quot;, region_tag=&quot;9_13&quot;)</code></pre>
<p><img src="figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f9aca3e2d24354b4f463a69a54776af5196a1a2d/docs/figure/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd/unnamed-chunk-37-1.png" target="_blank">f9aca3e</a>
</td>
<td>
sq-96
</td>
<td>
2021-12-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] GenomicRanges_1.36.1 GenomeInfoDb_1.20.0  IRanges_2.18.1      
 [4] S4Vectors_0.22.1     BiocGenerics_0.30.0  biomaRt_2.40.1      
 [7] ctwas_0.1.31         forcats_0.5.1        stringr_1.4.0       
[10] dplyr_1.0.7          purrr_0.3.4          readr_2.1.1         
[13] tidyr_1.1.4          tidyverse_1.3.1      tibble_3.1.6        
[16] readxl_1.3.1         WebGestaltR_0.4.4    disgenet2r_0.99.2   
[19] enrichR_3.0          cowplot_1.0.0        ggplot2_3.3.5       
[22] workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0       colorspace_2.0-2       rjson_0.2.20          
  [4] ellipsis_0.3.2         rprojroot_2.0.2        XVector_0.24.0        
  [7] fs_1.5.2               rstudioapi_0.13        farver_2.1.0          
 [10] ggrepel_0.8.1          bit64_4.0.5            AnnotationDbi_1.46.0  
 [13] fansi_0.5.0            lubridate_1.8.0        xml2_1.3.3            
 [16] codetools_0.2-16       logging_0.10-108       doParallel_1.0.16     
 [19] cachem_1.0.6           knitr_1.36             jsonlite_1.7.2        
 [22] apcluster_1.4.8        Cairo_1.5-12.2         broom_0.7.10          
 [25] dbplyr_2.1.1           compiler_3.6.1         httr_1.4.2            
 [28] backports_1.4.1        assertthat_0.2.1       Matrix_1.2-18         
 [31] fastmap_1.1.0          cli_3.1.0              later_0.8.0           
 [34] prettyunits_1.1.1      htmltools_0.5.2        tools_3.6.1           
 [37] igraph_1.2.10          GenomeInfoDbData_1.2.1 gtable_0.3.0          
 [40] glue_1.5.1             reshape2_1.4.4         doRNG_1.8.2           
 [43] Rcpp_1.0.7             Biobase_2.44.0         cellranger_1.1.0      
 [46] jquerylib_0.1.4        vctrs_0.3.8            svglite_1.2.2         
 [49] iterators_1.0.13       xfun_0.29              rvest_1.0.2           
 [52] lifecycle_1.0.1        rngtools_1.5.2         XML_3.99-0.3          
 [55] zlibbioc_1.30.0        scales_1.1.1           vroom_1.5.7           
 [58] hms_1.1.1              promises_1.0.1         yaml_2.2.1            
 [61] curl_4.3.2             memoise_2.0.1          ggrastr_1.0.1         
 [64] gdtools_0.1.9          stringi_1.7.6          RSQLite_2.2.8         
 [67] highr_0.9              foreach_1.5.1          rlang_0.4.12          
 [70] pkgconfig_2.0.3        bitops_1.0-7           evaluate_0.14         
 [73] lattice_0.20-38        labeling_0.4.2         bit_4.0.4             
 [76] tidyselect_1.1.1       plyr_1.8.6             magrittr_2.0.1        
 [79] R6_2.5.1               generics_0.1.1         DBI_1.1.1             
 [82] pgenlibr_0.3.1         pillar_1.6.4           haven_2.4.3           
 [85] whisker_0.3-2          withr_2.4.3            RCurl_1.98-1.5        
 [88] modelr_0.1.8           crayon_1.4.2           utf8_1.2.2            
 [91] tzdb_0.2.0             rmarkdown_2.11         progress_1.2.2        
 [94] grid_3.6.1             data.table_1.14.2      blob_1.2.2            
 [97] git2r_0.26.1           reprex_2.0.1           digest_0.6.29         
[100] httpuv_1.5.1           munsell_0.5.0          beeswarm_0.2.3        
[103] vipor_0.4.5           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
