<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Hypothalamus</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Hypothalamus</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeeb13ecf45ad6ec66ccac1b6f91b74c140a7261d6targetblankeb13ecfa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6" target="_blank">eb13ecf</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeeb13ecf45ad6ec66ccac1b6f91b74c140a7261d6targetblankeb13ecfa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6" target="_blank">eb13ecf</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Hypothalamus.Rmd</code>) and HTML (<code>docs/BMI_Brain_Hypothalamus.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6/analysis/BMI_Brain_Hypothalamus.Rmd" target="_blank">eb13ecf</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e6bc16990373312d301ab183ec936aff373d2497/docs/BMI_Brain_Hypothalamus.html" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/87fee8b56a4f94471388b3fc74b464d3b6efdd41/analysis/BMI_Brain_Hypothalamus.Rmd" target="_blank">87fee8b</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre><code>[1] 11083</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1107  757  667  439  549  622  531  429  416  429  660  610  213  357  369  492 
  17   18   19   20   21   22 
 680  168  851  327  134  276 </code></pre>
<pre><code>[1] 8872</code></pre>
<pre><code>[1] 0.8005053</code></pre>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-4-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>        gene          snp 
0.0049608595 0.0002982899 </code></pre>
<pre><code>    gene      snp 
15.52429 17.75596 </code></pre>
<pre><code>[1] 336107</code></pre>
<pre><code>[1]   11083 7535010</code></pre>
<pre><code>       gene         snp 
0.002539502 0.118737825 </code></pre>
<pre><code>[1]  0.01272065 15.93546282</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-5-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>            genename region_tag susie_pip       mu2          PVE         z
7493           PPM1M       3_36 0.9999998 244.75253 7.281981e-04  4.537140
3276           CCND2       12_4 0.9397782  28.25790 7.901103e-05 -5.119990
7598           ZNF12        7_9 0.7814221  26.90490 6.255175e-05  5.089286
7840          ALKBH3      11_27 0.7683120  28.49878 6.514579e-05 -5.127700
13153 RP11-1109F11.3      12_54 0.7651666  30.71378 6.992166e-05  6.456677
8379           CENPX      17_46 0.7105395  23.78657 5.028547e-05  4.110746
8812            RARG      12_33 0.7026347  25.49150 5.329022e-05 -4.106087
241             ISL1       5_30 0.7013735  26.15680 5.458288e-05 -5.009605
7356        SERPINI1      3_103 0.6982908  23.16012 4.811711e-05 -4.064505
4821           DCAF7      17_38 0.6563408  30.15661 5.888902e-05  5.436897
3176          PRRC2C       1_84 0.6252925  28.14482 5.236055e-05 -5.172951
584             NGFR      17_29 0.6249642  28.07281 5.219916e-05 -4.005394
11412        NCKIPSD       3_34 0.6000790  26.29681 4.694983e-05  4.490993
13194   CTC-498M16.4       5_52 0.5916685  52.94952 9.321009e-05  7.705884
5798            ECE2      3_113 0.5826731  28.51559 4.943445e-05 -5.287344
5712          THSD7B       2_81 0.5485336  27.38466 4.469233e-05  5.321314
11611         HRAT92        7_1 0.5483159  24.30683 3.965351e-05 -3.927490
7806         R3HCC1L      10_62 0.5455598  39.54709 6.419177e-05  7.438889
5498           CARM1       19_9 0.5430815  32.91258 5.318013e-05  5.016317
155            CSDE1       1_71 0.5346700  22.41865 3.566298e-05 -4.744544
      num_eqtl
7493         2
3276         1
7598         2
7840         2
13153        1
8379         2
8812         1
241          1
7356         1
4821         1
3176         1
584          2
11412        1
13194        1
5798         1
5712         1
11611        2
7806         1
5498         1
155          1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-6-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>       genename region_tag    susie_pip      mu2          PVE          z
7665    CCDC171       9_13 0.000000e+00 50550.12 0.000000e+00   8.034327
8735      NEGR1       1_46 0.000000e+00 43383.72 0.000000e+00 -10.695227
9420      STX19       3_59 0.000000e+00 31106.49 0.000000e+00  -5.059656
7889       LEO1      15_21 4.127809e-13 27969.54 3.435005e-14   4.602678
5271      MFAP1      15_16 0.000000e+00 23764.59 0.000000e+00   4.302998
13397 LINC02019       3_35 1.110223e-15 22551.06 7.449028e-17  -4.467972
5098      TMOD3      15_21 0.000000e+00 22268.83 0.000000e+00   5.411998
4029      TMOD2      15_21 0.000000e+00 21601.60 0.000000e+00   5.231719
1293      WDR76      15_16 0.000000e+00 21486.56 0.000000e+00   4.740358
11601    CKMT1A      15_16 0.000000e+00 21284.13 0.000000e+00   4.129652
2876       CISH       3_35 0.000000e+00 20260.39 0.000000e+00  -3.798838
3017      PLCL1      2_117 0.000000e+00 18664.08 0.000000e+00  -5.641781
1015      CCNT2       2_80 1.554312e-15 18644.85 8.622231e-17   3.685900
2875      HEMK1       3_35 0.000000e+00 17517.21 0.000000e+00  -3.881751
4998    TUBGCP4      15_16 0.000000e+00 16916.45 0.000000e+00   3.431538
9416      DHFR2       3_59 0.000000e+00 16605.15 0.000000e+00   2.760649
9414      NSUN3       3_59 0.000000e+00 15678.49 0.000000e+00   4.755360
8261       ADAL      15_16 0.000000e+00 14821.09 0.000000e+00  -2.861302
125    CACNA2D2       3_35 0.000000e+00 14198.82 0.000000e+00  -4.008712
5136     CNOT6L       4_52 0.000000e+00 14094.61 0.000000e+00   3.421551
      num_eqtl
7665         2
8735         2
9420         1
7889         1
5271         1
13397        2
5098         1
4029         1
1293         2
11601        1
2876         1
3017         1
1015         1
2875         1
4998         1
9416         2
9414         1
8261         1
125          1
5136         1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>            genename region_tag  susie_pip        mu2          PVE         z
7493           PPM1M       3_36 0.99999983  244.75253 7.281981e-04  4.537140
2953          LANCL1      2_124 0.04150332 4670.09050 5.766743e-04 -3.714167
9392         FAM220A        7_8 0.21633910  445.76487 2.869216e-04 -1.293295
13194   CTC-498M16.4       5_52 0.59166851   52.94952 9.321009e-05  7.705884
4791            RAC1        7_8 0.15841126  178.70493 8.422577e-05 -5.512237
2896           SPCS1       3_36 0.08081493  348.59408 8.381738e-05 -5.066891
3276           CCND2       12_4 0.93977817   28.25790 7.901103e-05 -5.119990
13153 RP11-1109F11.3      12_54 0.76516663   30.71378 6.992166e-05  6.456677
12160         ATP5J2       7_62 0.45795733   50.81152 6.923244e-05 -7.116991
2926           ITGB6       2_96 0.49914478   45.97271 6.827301e-05  5.451232
7840          ALKBH3      11_27 0.76831199   28.49878 6.514579e-05 -5.127700
7806         R3HCC1L      10_62 0.54555981   39.54709 6.419177e-05  7.438889
7598           ZNF12        7_9 0.78142209   26.90490 6.255175e-05  5.089286
13639         DHRS11      17_22 0.34692049   60.57451 6.252336e-05 -8.142012
4821           DCAF7      17_38 0.65634080   30.15661 5.888902e-05  5.436897
241             ISL1       5_30 0.70137346   26.15680 5.458288e-05 -5.009605
8812            RARG      12_33 0.70263472   25.49150 5.329022e-05 -4.106087
5498           CARM1       19_9 0.54308147   32.91258 5.318013e-05  5.016317
3176          PRRC2C       1_84 0.62529253   28.14482 5.236055e-05 -5.172951
584             NGFR      17_29 0.62496421   28.07281 5.219916e-05 -4.005394
      num_eqtl
7493         2
2953         2
9392         1
13194        1
4791         4
2896         1
3276         1
13153        1
12160        1
2926         1
7840         2
7806         1
7598         2
13639        1
4821         1
241          1
8812         1
5498         1
3176         1
584          2</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>      genename region_tag    susie_pip         mu2          PVE          z
7489     MST1R       3_35 1.814174e-03  1050.66826 5.671095e-06 -12.627554
38        RBM6       3_35 3.575409e-04   906.71176 9.645337e-07  12.536042
9046    KCTD13      16_24 4.543363e-02   109.27774 1.477174e-05 -11.490673
9045    ASPHD1      16_24 9.053323e-03   101.20568 2.726060e-06 -11.336675
7484    RNF123       3_35 4.940603e-12   823.15719 1.210000e-14 -10.959165
6178     TAOK2      16_24 1.619349e-02    92.70010 4.466252e-06  10.737701
8735     NEGR1       1_46 0.000000e+00 43383.71612 0.000000e+00 -10.695227
11930   NPIPB7      16_23 5.217439e-02    86.00817 1.335118e-05  10.452595
10430     CLN3      16_23 5.217439e-02    86.00817 1.335118e-05  10.452595
8365    INO80E      16_24 1.515007e-02    78.23179 3.526309e-06  10.102104
8032    ZNF646      16_24 5.107933e-02    75.83587 1.152504e-05 -10.000364
5486      SAE1      19_33 3.540535e-03    97.45885 1.026627e-06   9.848747
7487     CAMKV       3_35 3.330669e-16  1446.63894 1.433554e-18   9.847856
2753  COL4A3BP       5_44 1.986762e-02    68.98511 4.077779e-06  -9.828145
458      PRSS8      16_24 1.089810e-02    70.69376 2.292210e-06  -9.764760
1830      KAT8      16_24 9.420617e-03    68.77957 1.927797e-06  -9.705982
11411      LAT      16_23 1.060996e-01    82.99715 2.619989e-05  -9.552834
8031    ZNF668      16_24 1.101252e-02    70.05297 2.295281e-06   9.549888
2458     MTCH2      11_29 7.571534e-03    81.32046 1.831918e-06  -9.514152
10711  SULT1A2      16_23 4.339290e-02    80.36465 1.037543e-05  -9.448875
      num_eqtl
7489         2
38           1
9046         1
9045         2
7484         1
6178         1
8735         2
11930        1
10430        1
8365         2
8032         1
5486         1
7487         1
2753         1
458          1
1830         1
11411        1
8031         2
2458         1
10711        2</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-9-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-9-2.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.02012091</code></pre>
<pre><code>      genename region_tag    susie_pip         mu2          PVE          z
7489     MST1R       3_35 1.814174e-03  1050.66826 5.671095e-06 -12.627554
38        RBM6       3_35 3.575409e-04   906.71176 9.645337e-07  12.536042
9046    KCTD13      16_24 4.543363e-02   109.27774 1.477174e-05 -11.490673
9045    ASPHD1      16_24 9.053323e-03   101.20568 2.726060e-06 -11.336675
7484    RNF123       3_35 4.940603e-12   823.15719 1.210000e-14 -10.959165
6178     TAOK2      16_24 1.619349e-02    92.70010 4.466252e-06  10.737701
8735     NEGR1       1_46 0.000000e+00 43383.71612 0.000000e+00 -10.695227
11930   NPIPB7      16_23 5.217439e-02    86.00817 1.335118e-05  10.452595
10430     CLN3      16_23 5.217439e-02    86.00817 1.335118e-05  10.452595
8365    INO80E      16_24 1.515007e-02    78.23179 3.526309e-06  10.102104
8032    ZNF646      16_24 5.107933e-02    75.83587 1.152504e-05 -10.000364
5486      SAE1      19_33 3.540535e-03    97.45885 1.026627e-06   9.848747
7487     CAMKV       3_35 3.330669e-16  1446.63894 1.433554e-18   9.847856
2753  COL4A3BP       5_44 1.986762e-02    68.98511 4.077779e-06  -9.828145
458      PRSS8      16_24 1.089810e-02    70.69376 2.292210e-06  -9.764760
1830      KAT8      16_24 9.420617e-03    68.77957 1.927797e-06  -9.705982
11411      LAT      16_23 1.060996e-01    82.99715 2.619989e-05  -9.552834
8031    ZNF668      16_24 1.101252e-02    70.05297 2.295281e-06   9.549888
2458     MTCH2      11_29 7.571534e-03    81.32046 1.831918e-06  -9.514152
10711  SULT1A2      16_23 4.339290e-02    80.36465 1.037543e-05  -9.448875
      num_eqtl
7489         2
38           1
9046         1
9045         2
7484         1
6178         1
8735         2
11930        1
10430        1
8365         2
8032         1
5486         1
7487         1
2753         1
458          1
1830         1
11411        1
8031         2
2458         1
10711        2</code></pre>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre><code>[1] 41</code></pre>
<pre><code>[1] 22</code></pre>
<pre><code>[1] 4.586313</code></pre>
<pre><code>[1] 2</code></pre>
<pre><code>[1] 223</code></pre>
<pre><code>     genename region_tag susie_pip      mu2          PVE       z num_eqtl
7493    PPM1M       3_36 0.9999998 244.7525 0.0007281981 4.53714        2</code></pre>
<pre><code>     ctwas       TWAS 
0.00000000 0.07317073 </code></pre>
<pre><code>    ctwas      TWAS 
0.9998192 0.9801103 </code></pre>
<pre><code>     ctwas       TWAS 
0.00000000 0.01345291 </code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-10-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readxl_1.3.1    cowplot_1.0.0   ggplot2_3.3.5   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1  xfun_0.29         purrr_0.3.4       colorspace_2.0-2 
 [5] vctrs_0.3.8       generics_0.1.1    htmltools_0.5.2   yaml_2.2.1       
 [9] utf8_1.2.2        blob_1.2.2        rlang_0.4.12      jquerylib_0.1.4  
[13] later_0.8.0       pillar_1.6.4      glue_1.5.1        withr_2.4.3      
[17] DBI_1.1.1         bit64_4.0.5       lifecycle_1.0.1   stringr_1.4.0    
[21] cellranger_1.1.0  munsell_0.5.0     gtable_0.3.0      evaluate_0.14    
[25] memoise_2.0.1     labeling_0.4.2    knitr_1.36        fastmap_1.1.0    
[29] httpuv_1.5.1      fansi_0.5.0       highr_0.9         Rcpp_1.0.7       
[33] promises_1.0.1    scales_1.1.1      cachem_1.0.6      farver_2.1.0     
[37] fs_1.5.2          bit_4.0.4         digest_0.6.29     stringi_1.7.6    
[41] dplyr_1.0.7       rprojroot_2.0.2   grid_3.6.1        tools_3.6.1      
[45] magrittr_2.0.1    tibble_3.1.6      RSQLite_2.2.8     crayon_1.4.2     
[49] whisker_0.3-2     pkgconfig_2.0.3   ellipsis_0.3.2    data.table_1.14.2
[53] assertthat_0.2.1  rmarkdown_2.11    R6_2.5.1          git2r_0.26.1     
[57] compiler_3.6.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
