<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Cortex</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Cortex</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Hippocampus_S.Rmd</code>) and HTML (<code>docs/BMI_Brain_Hippocampus_S.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1058990dac15746212ef84c84aa7cc13dc9948bf/analysis/BMI_Brain_Hippocampus_S.Rmd" target="_blank">1058990</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 21015</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1988 1451 1226  814  817 1108 1205  768  844  935 1262 1061  401  722  697  946 
  17   18   19   20   21   22 
1404  275 1490  695  258  648 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 19381</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.9222</code></pre>
<pre class="r"><code>#add z scores to results
load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_gene.Rd&quot;))
ctwas_gene_res$z &lt;- z_gene[ctwas_gene_res$intron_id,]$z

z_snp &lt;- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z &lt;- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$intron_id=NA

ctwas_res &lt;- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#get number of eQTL for geens
num_eqtl &lt;- c()
for (i in 1:22){
  load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, i, &quot;.exprqc.Rd&quot;))
  num_eqtl &lt;- c(num_eqtl, unlist(lapply(wgtlist, nrow)))
}
ctwas_gene_res$num_eqtl &lt;- num_eqtl[ctwas_gene_res$intron_id]

#store columns to report
report_cols &lt;- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c(&quot;type&quot;, &quot;region_tag1&quot;, &quot;region_tag2&quot;, &quot;cs_index&quot;, &quot;gene_type&quot;, &quot;z_flag&quot;, &quot;id&quot;, &quot;chrom&quot;, &quot;pos&quot;))]
first_cols &lt;- c(&quot;genename&quot;, &quot;region_tag&quot;)
report_cols &lt;- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps &lt;- c(&quot;id&quot;, report_cols[-1])
report_cols_snps &lt;- report_cols_snps[!(report_cols_snps %in% &quot;num_eqtl&quot;)]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s1.susieIrss.txt&quot;))
n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
rm(ctwas_res_s1)</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0072760 0.0002872 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
24.48 17.13 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 336107</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   21015 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.01114 0.11031 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1]  0.4344 14.8260</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>             genename region_tag susie_pip      mu2       PVE
20492            NSD1      5_106    1.0000 13952.43 4.151e-02
20657         SLC38A2      12_29    1.0000 15720.07 4.677e-02
20521             DST       6_42    1.0000 15093.71 4.491e-02
20744             MOK      14_53    1.0000  9686.11 2.882e-02
20984           ASCC2      22_10    0.9911  8705.86 2.567e-02
20761 TNFSF12-TNFSF13       17_7    0.9426    29.32 8.224e-05
20638            GBF1      10_65    0.7791   291.14 6.749e-04
8294         FAM160B2       8_23    0.7784    26.68 6.180e-05
1804           SH3YL1        2_1    0.7781    45.46 1.052e-04
20279           MAST2       1_28    0.7748 26644.32 6.142e-02
17334           DOT1L       19_3    0.7711    22.94 5.262e-05
8140           PTPRN2       7_98    0.7659    22.89 5.217e-05
11308          EFEMP2      11_36    0.7597    50.86 1.150e-04
20873   ZNF559-ZNF177       19_9    0.7500 16816.72 3.753e-02
20874   ZNF559-ZNF177       19_9    0.7500 16816.72 3.753e-02
16827           RPTOR      17_45    0.7475    47.73 1.061e-04
10939         BDNF-AS      11_19    0.7427   120.71 2.667e-04
16218             MLX      17_25    0.7425    31.57 6.974e-05
7779            PTCD1       7_61    0.7360    55.22 1.209e-04
8343         SLC25A37       8_24    0.7311    26.64 5.795e-05
                          intron_id       z num_eqtl
20492  intron_5_177235945_177239756   2.957        1
20657   intron_12_46367340_46370512  -2.953        1
20521    intron_6_56468999_56469883   2.984        1
20744 intron_14_102265912_102283478   2.804        1
20984   intron_22_29825257_29825622  -2.882        1
20761     intron_17_7559702_7560049  -4.226        2
20638 intron_10_102260116_102344051  -5.281        1
8294     intron_8_22094518_22096337   4.806        1
1804         intron_2_253115_263211   5.990        1
20279    intron_1_45997799_46000959  -4.325        1
17334     intron_19_2199939_2202700  -4.106        1
8140   intron_7_157682937_157690552  -4.473        1
11308   intron_11_65868629_65869360  -8.201        1
20873     intron_19_9364948_9371610   3.338        1
20874     intron_19_9364948_9376316  -3.338        1
16827   intron_17_80730706_80754010  -6.093        1
10939   intron_11_27658462_27659171 -13.138        1
16218   intron_17_42567655_42568837   5.692        1
7779     intron_7_99452440_99454041   7.169        1
8343     intron_8_23529212_23566108   5.008        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>      genename region_tag susie_pip   mu2     PVE                    intron_id
17747    MAST3      19_14    0.0000 54893 0.00000  intron_19_18107618_18121685
17748    MAST3      19_14    0.0000 46415 0.00000  intron_19_18110741_18121685
20281    MAST2       1_28    0.2252 26647 0.01786   intron_1_46019697_46021950
20279    MAST2       1_28    0.7748 26644 0.06142   intron_1_45997799_46000959
17751    MAST3      19_14    0.0000 26358 0.00000  intron_19_18147017_18147443
17752    MAST3      19_14    0.0000 24335 0.00000  intron_19_18147044_18147443
11418  IGHMBP2      11_38    0.0000 22271 0.00000  intron_11_68929357_68933299
20275     NASP       1_28    0.0000 22177 0.00000   intron_1_45616703_45618061
20274     NASP       1_28    0.0000 22176 0.00000   intron_1_45616393_45616626
20271     NASP       1_28    0.0000 21778 0.00000   intron_1_45591270_45600385
20272     NASP       1_28    0.0000 21778 0.00000   intron_1_45591270_45602255
14243    TMOD3      15_21    0.0000 18818 0.00000  intron_15_51938260_51947292
3684     HEMK1       3_35    0.0000 18621 0.00000   intron_3_50579740_50579844
3683     HEMK1       3_35    0.0000 18620 0.00000   intron_3_50577573_50577826
3682     HEMK1       3_35    0.0000 18619 0.00000   intron_3_50571801_50577052
3681   C3orf18       3_35    0.0000 18615 0.00000   intron_3_50569574_50570931
2843     PLCL1      2_117    0.0000 18554 0.00000 intron_2_197805339_198001952
20467  FAM193A        4_3    0.0000 18231 0.00000     intron_4_2695129_2696363
4200  ARHGEF26       3_95    0.0000 17488 0.00000 intron_3_154152932_154187685
11416   MRPL21      11_38    0.0000 16886 0.00000  intron_11_68898012_68900548
           z num_eqtl
17747  6.803        1
17748 -6.367        2
20281 -4.301        1
20279 -4.325        1
17751 -5.317        2
17752  4.715        3
11418  4.486        1
20275  4.287        1
20274 -4.289        1
20271  3.951        1
20272 -3.951        1
14243 -2.890        2
3684   4.698        1
3683   4.701        1
3682   4.704        1
3681   4.682        1
2843  -5.642        1
20467  5.435        1
4200  -4.115        1
11416 -3.950        2</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>           genename region_tag susie_pip      mu2       PVE
20279         MAST2       1_28  0.774763 26644.32 0.0614180
20657       SLC38A2      12_29  1.000000 15720.07 0.0467710
20521           DST       6_42  1.000000 15093.71 0.0449075
20492          NSD1      5_106  1.000000 13952.43 0.0415119
20873 ZNF559-ZNF177       19_9  0.750000 16816.72 0.0375254
20874 ZNF559-ZNF177       19_9  0.750000 16816.72 0.0375254
20744           MOK      14_53  0.999958  9686.11 0.0288173
20984         ASCC2      22_10  0.991073  8705.86 0.0256708
12984         SPG20      13_13  0.555556 13496.28 0.0223082
12982         SPG20      13_13  0.555556 13496.28 0.0223082
12983         SPG20      13_13  0.555556 13496.28 0.0223082
20281         MAST2       1_28  0.225237 26647.22 0.0178573
20638          GBF1      10_65  0.779128   291.14 0.0006749
10233         TNKS2      10_59  0.395483   401.72 0.0004727
3735          ITIH4       3_36  0.503272   179.62 0.0002690
10939       BDNF-AS      11_19  0.742708   120.71 0.0002667
20716         DUSP6      12_54  0.700419   104.11 0.0002170
18226          GIPR      19_32  0.550913   117.02 0.0001918
20520           DST       6_42  0.003621 15062.10 0.0001623
14369        MAP2K5      15_31  0.546286    93.09 0.0001513
                          intron_id       z num_eqtl
20279    intron_1_45997799_46000959  -4.325        1
20657   intron_12_46367340_46370512  -2.953        1
20521    intron_6_56468999_56469883   2.984        1
20492  intron_5_177235945_177239756   2.957        1
20873     intron_19_9364948_9371610   3.338        1
20874     intron_19_9364948_9376316  -3.338        1
20744 intron_14_102265912_102283478   2.804        1
20984   intron_22_29825257_29825622  -2.882        1
12984   intron_13_36335832_36346582   2.819        1
12982   intron_13_36335832_36336287   2.819        1
12983   intron_13_36335832_36346225  -2.819        1
20281    intron_1_46019697_46021950  -4.301        1
20638 intron_10_102260116_102344051  -5.281        1
10233   intron_10_91817229_91819270   2.504        2
3735     intron_3_52818536_52819393   5.995        2
10939   intron_11_27658462_27659171 -13.138        1
20716   intron_12_89351025_89351640   6.438        1
18226   intron_19_45677779_45677906  10.801        1
20520    intron_6_56464756_56466078   2.892        1
14369   intron_15_67727945_67748231  -9.894        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag susie_pip    mu2       PVE
10938     BDNF-AS      11_19 2.649e-01 119.76 9.439e-05
10939     BDNF-AS      11_19 7.427e-01 120.71 2.667e-04
15153      KCTD13      16_24 5.618e-02 107.69 1.800e-05
1922  DNAJC27-AS1       2_15 7.658e-05 191.49 4.363e-08
1924  DNAJC27-AS1       2_15 7.658e-05 191.49 4.363e-08
1923  DNAJC27-AS1       2_15 7.658e-05 191.49 4.363e-08
15165       DOC2A      16_24 3.116e-02 101.21 9.382e-06
18226        GIPR      19_32 5.509e-01 117.02 1.918e-04
18225        GIPR      19_32 3.890e-01 116.24 1.345e-04
15159      INO80E      16_24 2.218e-02  94.37 6.226e-06
3660         RBM6       3_35 8.382e-14 162.75 4.059e-17
15137      ATXN2L      16_23 2.003e-01  89.70 5.346e-05
15136      ATXN2L      16_23 1.989e-01  89.68 5.308e-05
15126     SULT1A2      16_23 1.625e-01  94.76 4.581e-05
15158       TAOK2      16_24 2.141e-02  93.56 5.958e-06
15134      ATXN2L      16_23 1.144e-01  88.01 2.994e-05
15135      ATXN2L      16_23 1.144e-01  88.01 2.994e-05
1402       SEC16B       1_87 2.992e-02  95.09 8.466e-06
15127     SULT1A2      16_23 7.048e-02  93.17 1.954e-05
15118        CLN3      16_23 4.389e-02  87.81 1.147e-05
                         intron_id      z num_eqtl
10938  intron_11_27640005_27659171  13.36        2
10939  intron_11_27658462_27659171 -13.14        1
15153  intron_16_29923359_29925790 -11.30        2
1922    intron_2_24972232_24972626 -11.29        1
1924    intron_2_24972232_25001415  11.29        1
1923    intron_2_24972232_24979060  11.29        1
15165  intron_16_30007090_30007173 -11.08        2
18226  intron_19_45677779_45677906  10.80        1
18225  intron_19_45677108_45677710  10.77        1
15159  intron_16_30001040_30001212  10.77        1
3660    intron_3_50047326_50048245  10.75        2
15137  intron_16_28835399_28835549 -10.74        1
15136  intron_16_28835399_28835546  10.74        1
15126  intron_16_28592443_28593252 -10.74        2
15158  intron_16_29989805_29990784 -10.73        1
15134  intron_16_28826390_28826862 -10.68        1
15135  intron_16_28826986_28829401 -10.68        1
1402  intron_1_177932569_177932698  10.59        1
15127  intron_16_28592481_28593252 -10.51        1
15118  intron_16_28487541_28487662 -10.48        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.0188</code></pre>
<pre><code>         genename region_tag susie_pip    mu2       PVE
10938     BDNF-AS      11_19 2.649e-01 119.76 9.439e-05
10939     BDNF-AS      11_19 7.427e-01 120.71 2.667e-04
15153      KCTD13      16_24 5.618e-02 107.69 1.800e-05
1922  DNAJC27-AS1       2_15 7.658e-05 191.49 4.363e-08
1924  DNAJC27-AS1       2_15 7.658e-05 191.49 4.363e-08
1923  DNAJC27-AS1       2_15 7.658e-05 191.49 4.363e-08
15165       DOC2A      16_24 3.116e-02 101.21 9.382e-06
18226        GIPR      19_32 5.509e-01 117.02 1.918e-04
18225        GIPR      19_32 3.890e-01 116.24 1.345e-04
15159      INO80E      16_24 2.218e-02  94.37 6.226e-06
3660         RBM6       3_35 8.382e-14 162.75 4.059e-17
15137      ATXN2L      16_23 2.003e-01  89.70 5.346e-05
15136      ATXN2L      16_23 1.989e-01  89.68 5.308e-05
15126     SULT1A2      16_23 1.625e-01  94.76 4.581e-05
15158       TAOK2      16_24 2.141e-02  93.56 5.958e-06
15134      ATXN2L      16_23 1.144e-01  88.01 2.994e-05
15135      ATXN2L      16_23 1.144e-01  88.01 2.994e-05
1402       SEC16B       1_87 2.992e-02  95.09 8.466e-06
15127     SULT1A2      16_23 7.048e-02  93.17 1.954e-05
15118        CLN3      16_23 4.389e-02  87.81 1.147e-05
                         intron_id      z num_eqtl
10938  intron_11_27640005_27659171  13.36        2
10939  intron_11_27658462_27659171 -13.14        1
15153  intron_16_29923359_29925790 -11.30        2
1922    intron_2_24972232_24972626 -11.29        1
1924    intron_2_24972232_25001415  11.29        1
1923    intron_2_24972232_24979060  11.29        1
15165  intron_16_30007090_30007173 -11.08        2
18226  intron_19_45677779_45677906  10.80        1
18225  intron_19_45677108_45677710  10.77        1
15159  intron_16_30001040_30001212  10.77        1
3660    intron_3_50047326_50048245  10.75        2
15137  intron_16_28835399_28835549 -10.74        1
15136  intron_16_28835399_28835546  10.74        1
15126  intron_16_28592443_28593252 -10.74        2
15158  intron_16_29989805_29990784 -10.73        1
15134  intron_16_28826390_28826862 -10.68        1
15135  intron_16_28826986_28829401 -10.68        1
1402  intron_1_177932569_177932698  10.59        1
15127  intron_16_28592481_28593252 -10.51        1
15118  intron_16_28487541_28487662 -10.48        1</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 62</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                              Term Overlap Adjusted.P.value
1 phosphatase binding (GO:0019902)   4/114          0.04581
                       Genes
1 SLC9A3R2;SH3YL1;DLG1;MAST2</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Warning in disease_enrichment(entities = genes, vocabulary = &quot;HGNC&quot;, database =
&quot;CURATED&quot;): Removing duplicates from input list.</code></pre>
<pre><code>                                                 Description     FDR Ratio
7                                                Body Weight 0.02296  2/27
75                              Mucinous carcinoma of breast 0.02296  1/27
81 Familial encephalopathy with neuroserpin inclusion bodies 0.02296  1/27
87                      FOCAL SEGMENTAL GLOMERULOSCLEROSIS 6 0.02296  1/27
88                  CUTIS LAXA, AUTOSOMAL RECESSIVE, TYPE IB 0.02296  1/27
89     NEUROPATHY, HEREDITARY SENSORY AND AUTONOMIC, TYPE VI 0.02296  1/27
91  HYPOGONADOTROPIC HYPOGONADISM 19 WITH OR WITHOUT ANOSMIA 0.02296  1/27
92      EPIDERMOLYSIS BULLOSA SIMPLEX, AUTOSOMAL RECESSIVE 2 0.02296  1/27
94          COMBINED OXIDATIVE PHOSPHORYLATION DEFICIENCY 28 0.02296  1/27
95                            5q35 microduplication syndrome 0.02296  1/27
   BgRatio
7  15/9703
75  1/9703
81  1/9703
87  1/9703
88  1/9703
89  1/9703
91  1/9703
92  1/9703
94  1/9703
95  1/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>Warning: ggrepel: 26 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 41</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.718</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 6</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 395</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>             genename region_tag susie_pip       mu2       PVE
20490            NSD1      5_106 0.000e+00  6811.096 0.000e+00
20491            NSD1      5_106 0.000e+00  5205.305 0.000e+00
20492            NSD1      5_106 1.000e+00 13952.430 4.151e-02
20517             DST       6_42 0.000e+00  2414.058 0.000e+00
20518             DST       6_42 0.000e+00   621.083 0.000e+00
20519             DST       6_42 0.000e+00   453.810 0.000e+00
20520             DST       6_42 3.621e-03 15062.096 1.623e-04
20521             DST       6_42 1.000e+00 15093.715 4.491e-02
20522             DST       6_42 0.000e+00   165.192 0.000e+00
20657         SLC38A2      12_29 1.000e+00 15720.069 4.677e-02
20734             MOK      14_53 2.093e-05  7518.338 4.683e-07
20735             MOK      14_53 1.003e-05    71.089 2.121e-09
20736             MOK      14_53 5.982e-06   195.337 3.477e-09
20737             MOK      14_53 9.597e-06    68.300 1.950e-09
20738             MOK      14_53 5.256e-06  2869.037 4.486e-08
20739             MOK      14_53 5.292e-06  3017.702 4.752e-08
20740             MOK      14_53 5.096e-06  2093.290 3.174e-08
20741             MOK      14_53 1.914e-05  7538.181 4.293e-07
20742             MOK      14_53 5.147e-06  6423.996 9.838e-08
20743             MOK      14_53 5.623e-06  4083.485 6.831e-08
20744             MOK      14_53 1.000e+00  9686.112 2.882e-02
20745             MOK      14_53 7.981e-06  3034.162 7.205e-08
20759 TNFSF12-TNFSF13       17_7 6.706e-03     9.243 1.844e-07
20760 TNFSF12-TNFSF13       17_7 6.706e-03     9.243 1.844e-07
20761 TNFSF12-TNFSF13       17_7 9.426e-01    29.325 8.224e-05
20983           ASCC2      22_10 0.000e+00    35.450 0.000e+00
20984           ASCC2      22_10 9.911e-01  8705.864 2.567e-02
20985           ASCC2      22_10 0.000e+00    53.437 0.000e+00
                          intron_id        z num_eqtl
20490  intron_5_177136030_177136873  0.31367        1
20491  intron_5_177136030_177191884  0.10561        2
20492  intron_5_177235945_177239756  2.95697        1
20517    intron_6_56463156_56463565  1.49128        2
20518    intron_6_56463764_56464685 -0.61897        2
20519    intron_6_56463764_56466078  0.53305        2
20520    intron_6_56464756_56466078  2.89236        1
20521    intron_6_56468999_56469883  2.98420        1
20522    intron_6_56861915_56863071 -1.55200        1
20657   intron_12_46367340_46370512 -2.95342        1
20734 intron_14_102226408_102232535  2.16037        1
20735 intron_14_102229657_102231707 -1.25920        1
20736 intron_14_102229657_102232535  0.57789        2
20737 intron_14_102231821_102232535 -1.22029        2
20738 intron_14_102250987_102251756  0.10090        2
20739 intron_14_102250990_102251756  0.08111        2
20740 intron_14_102251804_102251917  0.19341        1
20741 intron_14_102251995_102263546  2.09729        1
20742 intron_14_102251995_102283478 -0.74917        2
20743 intron_14_102263616_102283478 -0.14125        2
20744 intron_14_102265912_102283478  2.80412        1
20745 intron_14_102283592_102304962  1.04323        1
20759     intron_17_7559297_7559846  1.33178        1
20760     intron_17_7559702_7559846 -1.33178        1
20761     intron_17_7559702_7560049 -4.22617        2
20983   intron_22_29802208_29804638  0.08732        1
20984   intron_22_29825257_29825622 -2.88165        1
20985   intron_22_29825257_29832245 -0.60398        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.09756 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9987 0.9598 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.01013 </code></pre>
<p><img src="figure/BMI_Brain_Hippocampus_S.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1              forcats_0.5.1            
 [3] stringr_1.4.0             dplyr_1.0.7              
 [5] purrr_0.3.4               readr_2.1.1              
 [7] tidyr_1.1.4               tidyverse_1.3.1          
 [9] tibble_3.1.6              WebGestaltR_0.4.4        
[11] disgenet2r_0.99.2         enrichR_3.0              
[13] cowplot_1.0.0             ggplot2_3.3.5            
[15] EnsDb.Hsapiens.v79_2.99.0 ensembldb_2.8.0          
[17] AnnotationFilter_1.8.0    GenomicFeatures_1.36.3   
[19] AnnotationDbi_1.46.0      Biobase_2.44.0           
[21] GenomicRanges_1.36.1      GenomeInfoDb_1.20.0      
[23] IRanges_2.18.1            S4Vectors_0.22.1         
[25] BiocGenerics_0.30.0       workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            colorspace_2.0-2           
  [3] rjson_0.2.20                ellipsis_0.3.2             
  [5] rprojroot_2.0.2             XVector_0.24.0             
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                ggrepel_0.9.1              
 [11] bit64_4.0.5                 lubridate_1.8.0            
 [13] fansi_0.5.0                 xml2_1.3.3                 
 [15] codetools_0.2-16            doParallel_1.0.16          
 [17] cachem_1.0.6                knitr_1.36                 
 [19] jsonlite_1.7.2              apcluster_1.4.8            
 [21] Cairo_1.5-12.2              Rsamtools_2.0.0            
 [23] broom_0.7.10                dbplyr_2.1.1               
 [25] compiler_3.6.1              httr_1.4.2                 
 [27] backports_1.4.1             assertthat_0.2.1           
 [29] Matrix_1.2-18               fastmap_1.1.0              
 [31] lazyeval_0.2.2              cli_3.1.0                  
 [33] later_0.8.0                 htmltools_0.5.2            
 [35] prettyunits_1.1.1           tools_3.6.1                
 [37] igraph_1.2.10               gtable_0.3.0               
 [39] glue_1.5.1                  GenomeInfoDbData_1.2.1     
 [41] reshape2_1.4.4              doRNG_1.8.2                
 [43] Rcpp_1.0.7                  cellranger_1.1.0           
 [45] jquerylib_0.1.4             vctrs_0.3.8                
 [47] Biostrings_2.52.0           svglite_1.2.2              
 [49] rtracklayer_1.44.4          iterators_1.0.13           
 [51] xfun_0.29                   rvest_1.0.2                
 [53] lifecycle_1.0.1             rngtools_1.5.2             
 [55] XML_3.99-0.3                zlibbioc_1.30.0            
 [57] scales_1.1.1                vroom_1.5.7                
 [59] hms_1.1.1                   promises_1.0.1             
 [61] ProtGenerics_1.16.0         SummarizedExperiment_1.14.1
 [63] yaml_2.2.1                  curl_4.3.2                 
 [65] memoise_2.0.1               ggrastr_1.0.1              
 [67] gdtools_0.1.9               biomaRt_2.40.1             
 [69] stringi_1.7.6               RSQLite_2.2.8              
 [71] highr_0.9                   foreach_1.5.1              
 [73] BiocParallel_1.18.0         rlang_0.4.12               
 [75] pkgconfig_2.0.3             matrixStats_0.57.0         
 [77] bitops_1.0-7                evaluate_0.14              
 [79] lattice_0.20-38             GenomicAlignments_1.20.1   
 [81] labeling_0.4.2              bit_4.0.4                  
 [83] tidyselect_1.1.1            plyr_1.8.6                 
 [85] magrittr_2.0.1              R6_2.5.1                   
 [87] generics_0.1.1              DelayedArray_0.10.0        
 [89] DBI_1.1.1                   haven_2.4.3                
 [91] pillar_1.6.4                whisker_0.3-2              
 [93] withr_2.4.3                 RCurl_1.98-1.5             
 [95] modelr_0.1.8                crayon_1.4.2               
 [97] utf8_1.2.2                  tzdb_0.2.0                 
 [99] rmarkdown_2.11              progress_1.2.2             
[101] grid_3.6.1                  data.table_1.14.2          
[103] blob_1.2.2                  git2r_0.26.1               
[105] reprex_2.0.1                digest_0.6.29              
[107] httpuv_1.5.1                munsell_0.5.0              
[109] beeswarm_0.2.3              vipor_0.4.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
