<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Cortex</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Cortex</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Hypothalamus_S.Rmd</code>) and HTML (<code>docs/BMI_Brain_Hypothalamus_S.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1058990dac15746212ef84c84aa7cc13dc9948bf/analysis/BMI_Brain_Hypothalamus_S.Rmd" target="_blank">1058990</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 23356</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
2187 1624 1391  915  964 1243 1339  822  921 1040 1364 1154  441  791  748 1054 
  17   18   19   20   21   22 
1559  319 1682  757  294  747 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 21465</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.919</code></pre>
<pre class="r"><code>#add z scores to results
load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_gene.Rd&quot;))
ctwas_gene_res$z &lt;- z_gene[ctwas_gene_res$intron_id,]$z

z_snp &lt;- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z &lt;- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$intron_id=NA

ctwas_res &lt;- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#get number of eQTL for geens
num_eqtl &lt;- c()
for (i in 1:22){
  load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, i, &quot;.exprqc.Rd&quot;))
  num_eqtl &lt;- c(num_eqtl, unlist(lapply(wgtlist, nrow)))
}
ctwas_gene_res$num_eqtl &lt;- num_eqtl[ctwas_gene_res$intron_id]

#store columns to report
report_cols &lt;- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c(&quot;type&quot;, &quot;region_tag1&quot;, &quot;region_tag2&quot;, &quot;cs_index&quot;, &quot;gene_type&quot;, &quot;z_flag&quot;, &quot;id&quot;, &quot;chrom&quot;, &quot;pos&quot;))]
first_cols &lt;- c(&quot;genename&quot;, &quot;region_tag&quot;)
report_cols &lt;- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps &lt;- c(&quot;id&quot;, report_cols[-1])
report_cols_snps &lt;- report_cols_snps[!(report_cols_snps %in% &quot;num_eqtl&quot;)]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s1.susieIrss.txt&quot;))
n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
rm(ctwas_res_s1)</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0054948 0.0002901 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
22.86 17.36 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 336107</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   23356 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.00873 0.11289 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1]  0.4274 14.4758</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>             genename region_tag susie_pip      mu2       PVE
22750             DST       6_42    1.0000 14599.16 4.344e-02
22844         SLC38A2      12_29    1.0000 14762.51 4.392e-02
23098           TMOD3      15_21    1.0000 34316.74 1.021e-01
22721            NSD1      5_106    1.0000 13458.30 4.004e-02
23083             MOK      14_53    0.9813  9526.17 2.781e-02
23152 TNFSF12-TNFSF13       17_7    0.9167    28.79 7.851e-05
22630        SERPINI1      3_103    0.8119    25.28 6.107e-05
23270   ZNF559-ZNF177       19_9    0.7500 15880.52 3.544e-02
23271   ZNF559-ZNF177       19_9    0.7500 15880.52 3.544e-02
19459          ZBTB7A       19_4    0.7130   169.08 3.587e-04
19392           DOT1L       19_3    0.7077    24.87 5.236e-05
9639         SLC25A37       8_24    0.6929    27.12 5.590e-05
284       RP1-37C10.3       1_12    0.6895    28.65 5.877e-05
5144             DLG1      3_121    0.6889    23.82 4.883e-05
9030            PTCD1       7_61    0.6705    55.04 1.098e-04
22529          MAPK11      22_24    0.6664    24.75 4.907e-05
19870           CRTC1      19_15    0.6417    49.72 9.493e-05
50               MIB2        1_1    0.6334    51.32 9.672e-05
6253          C5orf64       5_37    0.6316    27.27 5.125e-05
22836         DENND1A       9_63    0.6217    30.06 5.560e-05
                          intron_id      z num_eqtl
22750    intron_6_56468999_56469883  2.985        1
22844   intron_12_46367340_46370512  2.953        1
23098   intron_15_51938260_51947292 -4.591        1
22721  intron_5_177235945_177239756 -2.957        1
23083 intron_14_102265912_102283478  2.804        1
23152     intron_17_7559702_7560049 -4.225        1
22630  intron_3_167735823_167789111 -4.409        2
23270     intron_19_9364948_9371610  3.338        1
23271     intron_19_9364948_9376316 -3.338        1
19459     intron_19_4055247_4065430 -7.758        1
19392     intron_19_2199939_2202700  4.106        1
9639     intron_8_23529212_23566108  5.008        1
284      intron_1_16986358_16986805  4.915        1
5144   intron_3_197149796_197194425 -4.462        1
9030     intron_7_99429806_99433278  7.174        1
22529   intron_22_50267949_50270177  4.182        2
19870   intron_19_18743026_18744108 -6.969        1
50         intron_1_1628722_1629133  6.613        1
6253     intron_5_61704108_61730046 -3.939        1
22836  intron_9_123757822_123769514  4.837        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>           genename region_tag susie_pip   mu2       PVE
19848         MAST3      19_14 0.000e+00 55987 0.000e+00
19849         MAST3      19_14 0.000e+00 51343 0.000e+00
10335       CCDC171       9_13 0.000e+00 48050 0.000e+00
19850         MAST3      19_14 0.000e+00 44887 0.000e+00
10336       CCDC171       9_13 0.000e+00 42064 0.000e+00
19852         MAST3      19_14 0.000e+00 41441 0.000e+00
19853         MAST3      19_14 0.000e+00 41441 0.000e+00
23098         TMOD3      15_21 1.000e+00 34317 1.021e-01
13009        MRPL21      11_38 2.188e-11 23136 1.506e-12
13011       IGHMBP2      11_38 1.443e-15 22844 9.809e-17
15926    CATSPER2P1      15_16 0.000e+00 22773 0.000e+00
4326          HEMK1       3_35 0.000e+00 19014 0.000e+00
4325          HEMK1       3_35 0.000e+00 19014 0.000e+00
4324          HEMK1       3_35 0.000e+00 19012 0.000e+00
4323        C3orf18       3_35 0.000e+00 19009 0.000e+00
13010        MRPL21      11_38 0.000e+00 18514 0.000e+00
15919          ADAL      15_16 0.000e+00 16855 0.000e+00
3374          RFTN2      2_117 0.000e+00 16713 0.000e+00
23270 ZNF559-ZNF177       19_9 7.500e-01 15881 3.544e-02
23271 ZNF559-ZNF177       19_9 7.500e-01 15881 3.544e-02
                         intron_id      z num_eqtl
19848  intron_19_18107618_18121685  6.803        1
19849  intron_19_18110741_18121685 -7.147        1
10335   intron_9_15888088_15888983 -8.107        2
19850  intron_19_18118254_18121685 -6.233        1
10336   intron_9_15889101_15920270 -7.233        1
19852  intron_19_18147017_18147443 -6.726        1
19853  intron_19_18147044_18147443  6.726        1
23098  intron_15_51938260_51947292 -4.591        1
13009  intron_11_68898012_68900535  4.267        2
13011  intron_11_68929357_68933299  4.486        1
15926  intron_15_43744219_43744395 -4.508        1
4326    intron_3_50579740_50579844  4.698        1
4325    intron_3_50577573_50577826 -4.701        1
4324    intron_3_50572208_50577052 -4.704        1
4323    intron_3_50569574_50570931  4.682        1
13010  intron_11_68898012_68900548 -4.056        2
15919  intron_15_43351041_43351751  3.360        1
3374  intron_2_197646666_197675320 -6.263        1
23270    intron_19_9364948_9371610  3.338        1
23271    intron_19_9364948_9376316 -3.338        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>           genename region_tag susie_pip     mu2       PVE
23098         TMOD3      15_21   1.00000 34316.7 0.1021006
22844       SLC38A2      12_29   1.00000 14762.5 0.0439221
22750           DST       6_42   1.00000 14599.2 0.0434360
22721          NSD1      5_106   1.00000 13458.3 0.0400417
23270 ZNF559-ZNF177       19_9   0.75000 15880.5 0.0354363
23271 ZNF559-ZNF177       19_9   0.75000 15880.5 0.0354363
23083           MOK      14_53   0.98132  9526.2 0.0278134
14627         SPG20      13_13   0.55556 12691.6 0.0209782
14625         SPG20      13_13   0.55556 12691.6 0.0209782
14626         SPG20      13_13   0.55556 12691.6 0.0209782
22873 RP11-386G11.5      12_31   0.57927  1617.3 0.0027874
1911         TATDN3      1_108   0.13827  3325.6 0.0013681
1909         TATDN3      1_108   0.12687  3324.5 0.0012549
1910         TATDN3      1_108   0.12687  3324.5 0.0012549
1908         TATDN3      1_108   0.12687  3324.5 0.0012549
1906         TATDN3      1_108   0.11321  3323.5 0.0011195
1907         TATDN3      1_108   0.10786  3323.3 0.0010665
16896        TNRC6A      16_21   0.41245   481.6 0.0005910
1904         TATDN3      1_108   0.05456  3323.0 0.0005395
1905         TATDN3      1_108   0.05454  3323.0 0.0005392
                          intron_id      z num_eqtl
23098   intron_15_51938260_51947292 -4.591        1
22844   intron_12_46367340_46370512  2.953        1
22750    intron_6_56468999_56469883  2.985        1
22721  intron_5_177235945_177239756 -2.957        1
23270     intron_19_9364948_9371610  3.338        1
23271     intron_19_9364948_9376316 -3.338        1
23083 intron_14_102265912_102283478  2.804        1
14627   intron_13_36335832_36346582  2.819        1
14625   intron_13_36335832_36336287  2.819        1
14626   intron_13_36335832_36346240 -2.819        1
22873   intron_12_49005543_49005743  4.788        1
1911   intron_1_212812328_212815013  3.296        1
1909   intron_1_212807848_212812248  3.301        1
1910   intron_1_212807848_212812251 -3.301        1
1908   intron_1_212807848_212812227 -3.301        1
1906   intron_1_212804429_212804596  3.300        1
1907   intron_1_212804651_212807736  3.301        1
16896   intron_16_24805151_24805605 -5.878        1
1904   intron_1_212797196_212802701  3.275        1
1905   intron_1_212797196_212804320  3.275        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag susie_pip    mu2       PVE
1694       SEC16B       1_87 2.117e-02 157.12 9.896e-06
12527     BDNF-AS      11_19 5.029e-01 119.38 1.786e-04
12528     BDNF-AS      11_19 5.029e-01 119.38 1.786e-04
4297         RBM6       3_35 1.865e-03 908.78 5.042e-06
7387     C6orf106       6_28 3.514e-01 178.54 1.866e-04
2314  DNAJC27-AS1       2_15 4.919e-05 190.73 2.791e-08
7389     UHRF1BP1       6_28 5.651e-02 157.08 2.641e-05
4288       RNF123       3_35 1.812e-11 823.70 4.441e-14
4286       RNF123       3_35 1.686e-11 823.28 4.131e-14
4289       RNF123       3_35 1.501e-11 822.69 3.674e-14
1770         IPO9      1_102 3.373e-01 114.59 1.150e-04
6318      ANKDD1B       5_44 1.001e-02  86.95 2.589e-06
16971       DOC2A      16_24 1.782e-02  94.41 5.006e-06
16938      ATXN2L      16_23 2.740e-01  89.63 7.307e-05
16939      ATXN2L      16_23 2.679e-01  89.57 7.139e-05
16966       TAOK2      16_24 1.723e-02  93.64 4.800e-06
16965       TAOK2      16_24 1.723e-02  93.64 4.800e-06
16961     TMEM219      16_24 1.677e-02  93.16 4.649e-06
16978       MAPK3      16_24 2.028e-02  91.64 5.529e-06
16930     SULT1A2      16_23 4.978e-02  91.04 1.348e-05
                         intron_id      z num_eqtl
1694  intron_1_177932569_177932698  13.65        2
12527  intron_11_27640005_27659171  13.14        1
12528  intron_11_27658462_27659171 -13.14        1
4297    intron_3_49999513_50046716  12.54        1
7387    intron_6_34654779_34696446 -11.51        2
2314    intron_2_24972232_24972626 -11.29        1
7389    intron_6_34859571_34864002  11.10        1
4288    intron_3_49712656_49713513  10.96        1
4286    intron_3_49698822_49698980 -10.96        1
4289    intron_3_49714174_49715575 -10.95        1
1770  intron_1_201874372_201874832 -10.91        1
6318    intron_5_75669383_75670979  10.85        2
16971  intron_16_30007090_30007173 -10.76        1
16938  intron_16_28835399_28835546  10.74        1
16939  intron_16_28835399_28835549 -10.74        1
16966  intron_16_29989805_29990784 -10.73        1
16965  intron_16_29989805_29990543  10.73        1
16961  intron_16_29971578_29972950  10.72        1
16978  intron_16_30114708_30115595  10.48        1
16930  intron_16_28592443_28593252 -10.45        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.01768</code></pre>
<pre><code>         genename region_tag susie_pip    mu2       PVE
1694       SEC16B       1_87 2.117e-02 157.12 9.896e-06
12527     BDNF-AS      11_19 5.029e-01 119.38 1.786e-04
12528     BDNF-AS      11_19 5.029e-01 119.38 1.786e-04
4297         RBM6       3_35 1.865e-03 908.78 5.042e-06
7387     C6orf106       6_28 3.514e-01 178.54 1.866e-04
2314  DNAJC27-AS1       2_15 4.919e-05 190.73 2.791e-08
7389     UHRF1BP1       6_28 5.651e-02 157.08 2.641e-05
4288       RNF123       3_35 1.812e-11 823.70 4.441e-14
4286       RNF123       3_35 1.686e-11 823.28 4.131e-14
4289       RNF123       3_35 1.501e-11 822.69 3.674e-14
1770         IPO9      1_102 3.373e-01 114.59 1.150e-04
6318      ANKDD1B       5_44 1.001e-02  86.95 2.589e-06
16971       DOC2A      16_24 1.782e-02  94.41 5.006e-06
16938      ATXN2L      16_23 2.740e-01  89.63 7.307e-05
16939      ATXN2L      16_23 2.679e-01  89.57 7.139e-05
16966       TAOK2      16_24 1.723e-02  93.64 4.800e-06
16965       TAOK2      16_24 1.723e-02  93.64 4.800e-06
16961     TMEM219      16_24 1.677e-02  93.16 4.649e-06
16978       MAPK3      16_24 2.028e-02  91.64 5.529e-06
16930     SULT1A2      16_23 4.978e-02  91.04 1.348e-05
                         intron_id      z num_eqtl
1694  intron_1_177932569_177932698  13.65        2
12527  intron_11_27640005_27659171  13.14        1
12528  intron_11_27658462_27659171 -13.14        1
4297    intron_3_49999513_50046716  12.54        1
7387    intron_6_34654779_34696446 -11.51        2
2314    intron_2_24972232_24972626 -11.29        1
7389    intron_6_34859571_34864002  11.10        1
4288    intron_3_49712656_49713513  10.96        1
4286    intron_3_49698822_49698980 -10.96        1
4289    intron_3_49714174_49715575 -10.95        1
1770  intron_1_201874372_201874832 -10.91        1
6318    intron_5_75669383_75670979  10.85        2
16971  intron_16_30007090_30007173 -10.76        1
16938  intron_16_28835399_28835546  10.74        1
16939  intron_16_28835399_28835549 -10.74        1
16966  intron_16_29989805_29990784 -10.73        1
16965  intron_16_29989805_29990543  10.73        1
16961  intron_16_29971578_29972950  10.72        1
16978  intron_16_30114708_30115595  10.48        1
16930  intron_16_28592443_28593252 -10.45        1</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 39</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                                    Term
1 regulation of transcription regulatory region DNA binding (GO:2000677)
2                             regulation of p38MAPK cascade (GO:1900744)
  Overlap Adjusted.P.value             Genes
1    3/27         0.007827 NSD1;DOT1L;ZBTB7A
2    3/37         0.010272   DLG1;AGER;VEGFA
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Warning in disease_enrichment(entities = genes, vocabulary = &quot;HGNC&quot;, database =
&quot;CURATED&quot;): Removing duplicates from input list.</code></pre>
<pre><code>                                                               Description
39                                                 Epilepsy, Temporal Lobe
40                                                       Uncinate Epilepsy
153                                Epilepsy, Benign Psychomotor, Childhood
154                                             Epilepsy, Lateral Temporal
165                                           Choroidal Neovascularization
225                                             Caudal Duplication Anomaly
227              Familial encephalopathy with neuroserpin inclusion bodies
237 MICROVASCULAR COMPLICATIONS OF DIABETES, SUSCEPTIBILITY TO, 1(finding)
243                  NEUROPATHY, HEREDITARY SENSORY AND AUTONOMIC, TYPE VI
247                   EPIDERMOLYSIS BULLOSA SIMPLEX, AUTOSOMAL RECESSIVE 2
        FDR Ratio BgRatio
39  0.03841  2/19 23/9703
40  0.03841  2/19 23/9703
153 0.03841  2/19 23/9703
154 0.03841  2/19 23/9703
165 0.03841  1/19  1/9703
225 0.03841  1/19  1/9703
227 0.03841  1/19  1/9703
237 0.03841  1/19  1/9703
243 0.03841  1/19  1/9703
247 0.03841  1/19  1/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>Warning: ggrepel: 12 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 41</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 18</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.74</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 7</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 413</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>             genename region_tag susie_pip      mu2       PVE
22630        SERPINI1      3_103 8.119e-01    25.28 6.107e-05
22631        SERPINI1      3_103 1.019e-02    13.37 4.054e-07
22719            NSD1      5_106 0.000e+00  6568.48 0.000e+00
22720            NSD1      5_106 0.000e+00  6568.48 0.000e+00
22721            NSD1      5_106 1.000e+00 13458.30 4.004e-02
22745             DST       6_42 0.000e+00   246.65 0.000e+00
22746             DST       6_42 0.000e+00  3436.28 0.000e+00
22747             DST       6_42 0.000e+00  1527.08 0.000e+00
22748             DST       6_42 0.000e+00   708.73 0.000e+00
22749             DST       6_42 9.124e-07 14456.84 3.925e-08
22750             DST       6_42 1.000e+00 14599.16 4.344e-02
22751             DST       6_42 0.000e+00   449.94 0.000e+00
22844         SLC38A2      12_29 1.000e+00 14762.51 4.392e-02
23073             MOK      14_53 7.512e-05  9329.69 2.085e-06
23074             MOK      14_53 3.636e-06   492.78 5.330e-09
23075             MOK      14_53 3.451e-06  2715.99 2.789e-08
23076             MOK      14_53 3.443e-06  2632.98 2.697e-08
23077             MOK      14_53 3.389e-06  2057.19 2.075e-08
23078             MOK      14_53 3.389e-06  2057.19 2.075e-08
23079             MOK      14_53 1.728e-05  7136.13 3.668e-07
23080             MOK      14_53 3.534e-06  6182.00 6.499e-08
23081             MOK      14_53 3.679e-06  4006.23 4.386e-08
23082             MOK      14_53 3.672e-06  4008.64 4.379e-08
23083             MOK      14_53 9.813e-01  9526.17 2.781e-02
23084             MOK      14_53 5.715e-06  2984.67 5.075e-08
23098           TMOD3      15_21 1.000e+00 34316.74 1.021e-01
23152 TNFSF12-TNFSF13       17_7 9.167e-01    28.79 7.851e-05
                          intron_id       z num_eqtl
22630  intron_3_167735823_167789111 -4.4091        2
22631  intron_3_167790602_167794620 -2.3445        1
22719  intron_5_177136030_177136873  0.3137        1
22720  intron_5_177136030_177191884 -0.3137        1
22721  intron_5_177235945_177239756 -2.9570        1
22745    intron_6_56460254_56463046 -0.9143        1
22746    intron_6_56463156_56463565  1.7242        2
22747    intron_6_56463764_56464685 -0.9696        2
22748    intron_6_56463764_56466078  0.6698        2
22749    intron_6_56464756_56466078  2.8806        1
22750    intron_6_56468999_56469883  2.9850        1
22751    intron_6_56598006_56598476  3.7368        1
22844   intron_12_46367340_46370512  2.9534        1
23073 intron_14_102226408_102232535  2.9326        1
23074 intron_14_102229657_102232535  0.3552        2
23075 intron_14_102250987_102251444 -0.1144        2
23076 intron_14_102250987_102251756  0.1250        2
23077 intron_14_102250990_102251756  0.1934        1
23078 intron_14_102251804_102251917  0.1934        1
23079 intron_14_102251995_102263546  2.1267        1
23080 intron_14_102251995_102283478 -0.6866        2
23081 intron_14_102263616_102265823 -0.1576        1
23082 intron_14_102263616_102283478 -0.1530        2
23083 intron_14_102265912_102283478  2.8041        1
23084 intron_14_102283592_102304962 -1.0432        1
23098   intron_15_51938260_51947292 -4.5908        1
23152     intron_17_7559702_7560049 -4.2247        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code> ctwas   TWAS 
0.0000 0.1463 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9988 0.9612 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.01453 </code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus_S.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1              forcats_0.5.1            
 [3] stringr_1.4.0             dplyr_1.0.7              
 [5] purrr_0.3.4               readr_2.1.1              
 [7] tidyr_1.1.4               tidyverse_1.3.1          
 [9] tibble_3.1.6              WebGestaltR_0.4.4        
[11] disgenet2r_0.99.2         enrichR_3.0              
[13] cowplot_1.0.0             ggplot2_3.3.5            
[15] EnsDb.Hsapiens.v79_2.99.0 ensembldb_2.8.0          
[17] AnnotationFilter_1.8.0    GenomicFeatures_1.36.3   
[19] AnnotationDbi_1.46.0      Biobase_2.44.0           
[21] GenomicRanges_1.36.1      GenomeInfoDb_1.20.0      
[23] IRanges_2.18.1            S4Vectors_0.22.1         
[25] BiocGenerics_0.30.0       workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            colorspace_2.0-2           
  [3] rjson_0.2.20                ellipsis_0.3.2             
  [5] rprojroot_2.0.2             XVector_0.24.0             
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                ggrepel_0.9.1              
 [11] bit64_4.0.5                 lubridate_1.8.0            
 [13] fansi_0.5.0                 xml2_1.3.3                 
 [15] codetools_0.2-16            doParallel_1.0.16          
 [17] cachem_1.0.6                knitr_1.36                 
 [19] jsonlite_1.7.2              apcluster_1.4.8            
 [21] Cairo_1.5-12.2              Rsamtools_2.0.0            
 [23] broom_0.7.10                dbplyr_2.1.1               
 [25] compiler_3.6.1              httr_1.4.2                 
 [27] backports_1.4.1             assertthat_0.2.1           
 [29] Matrix_1.2-18               fastmap_1.1.0              
 [31] lazyeval_0.2.2              cli_3.1.0                  
 [33] later_0.8.0                 htmltools_0.5.2            
 [35] prettyunits_1.1.1           tools_3.6.1                
 [37] igraph_1.2.10               gtable_0.3.0               
 [39] glue_1.5.1                  GenomeInfoDbData_1.2.1     
 [41] reshape2_1.4.4              doRNG_1.8.2                
 [43] Rcpp_1.0.7                  cellranger_1.1.0           
 [45] jquerylib_0.1.4             vctrs_0.3.8                
 [47] Biostrings_2.52.0           svglite_1.2.2              
 [49] rtracklayer_1.44.4          iterators_1.0.13           
 [51] xfun_0.29                   rvest_1.0.2                
 [53] lifecycle_1.0.1             rngtools_1.5.2             
 [55] XML_3.99-0.3                zlibbioc_1.30.0            
 [57] scales_1.1.1                vroom_1.5.7                
 [59] hms_1.1.1                   promises_1.0.1             
 [61] ProtGenerics_1.16.0         SummarizedExperiment_1.14.1
 [63] yaml_2.2.1                  curl_4.3.2                 
 [65] memoise_2.0.1               ggrastr_1.0.1              
 [67] gdtools_0.1.9               biomaRt_2.40.1             
 [69] stringi_1.7.6               RSQLite_2.2.8              
 [71] highr_0.9                   foreach_1.5.1              
 [73] BiocParallel_1.18.0         rlang_0.4.12               
 [75] pkgconfig_2.0.3             matrixStats_0.57.0         
 [77] bitops_1.0-7                evaluate_0.14              
 [79] lattice_0.20-38             GenomicAlignments_1.20.1   
 [81] labeling_0.4.2              bit_4.0.4                  
 [83] tidyselect_1.1.1            plyr_1.8.6                 
 [85] magrittr_2.0.1              R6_2.5.1                   
 [87] generics_0.1.1              DelayedArray_0.10.0        
 [89] DBI_1.1.1                   haven_2.4.3                
 [91] pillar_1.6.4                whisker_0.3-2              
 [93] withr_2.4.3                 RCurl_1.98-1.5             
 [95] modelr_0.1.8                crayon_1.4.2               
 [97] utf8_1.2.2                  tzdb_0.2.0                 
 [99] rmarkdown_2.11              progress_1.2.2             
[101] grid_3.6.1                  data.table_1.14.2          
[103] blob_1.2.2                  git2r_0.26.1               
[105] reprex_2.0.1                digest_0.6.29              
[107] httpuv_1.5.1                munsell_0.5.0              
[109] beeswarm_0.2.3              vipor_0.4.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
