<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Cortex</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Cortex</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd</code>) and HTML (<code>docs/BMI_Brain_Spinal_cord_cervical_c-1_S.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1058990dac15746212ef84c84aa7cc13dc9948bf/analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd" target="_blank">1058990</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 22198</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
2051 1538 1269  920  873 1191 1238  775  894 1019 1359 1094  456  760  792  963 
  17   18   19   20   21   22 
1518  325 1538  689  312  624 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 20471</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.9222</code></pre>
<pre class="r"><code>#add z scores to results
load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_gene.Rd&quot;))
ctwas_gene_res$z &lt;- z_gene[ctwas_gene_res$intron_id,]$z

z_snp &lt;- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z &lt;- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$intron_id=NA

ctwas_res &lt;- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#get number of eQTL for geens
num_eqtl &lt;- c()
for (i in 1:22){
  load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, i, &quot;.exprqc.Rd&quot;))
  num_eqtl &lt;- c(num_eqtl, unlist(lapply(wgtlist, nrow)))
}
ctwas_gene_res$num_eqtl &lt;- num_eqtl[ctwas_gene_res$intron_id]

#store columns to report
report_cols &lt;- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c(&quot;type&quot;, &quot;region_tag1&quot;, &quot;region_tag2&quot;, &quot;cs_index&quot;, &quot;gene_type&quot;, &quot;z_flag&quot;, &quot;id&quot;, &quot;chrom&quot;, &quot;pos&quot;))]
first_cols &lt;- c(&quot;genename&quot;, &quot;region_tag&quot;)
report_cols &lt;- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps &lt;- c(&quot;id&quot;, report_cols[-1])
report_cols_snps &lt;- report_cols_snps[!(report_cols_snps %in% &quot;num_eqtl&quot;)]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s1.susieIrss.txt&quot;))
n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
rm(ctwas_res_s1)</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0057692 0.0002894 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
25.83 17.24 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 336107</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   22198 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>    gene      snp 
0.009843 0.111851 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1]  0.3821 16.2553</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>        genename region_tag susie_pip      mu2       PVE
21963    IGHMBP2      11_38    1.0000 27703.01 8.242e-02
21975    SLC38A2      12_29    1.0000 16754.50 4.985e-02
21738    CCDC127        5_1    0.9991 12476.13 3.709e-02
21903     ZNF37A      10_28    0.9988 10226.25 3.039e-02
22168       GIPR      19_32    0.9314  5642.01 1.564e-02
21870    SLC37A3       7_86    0.8773    24.65 6.435e-05
18736      CARM1       19_9    0.7999    31.83 7.575e-05
6714    HLA-DQB1       6_26    0.7895   596.93 1.402e-03
4474    SERPINI1      3_103    0.7757    23.98 5.534e-05
15718   SLC9A3R2       16_2    0.7646    28.03 6.376e-05
2844       ITGB6       2_96    0.7628    59.94 1.360e-04
3996    SLC25A26       3_45    0.7587    24.87 5.613e-05
5104       NUP54       4_51    0.7548    30.81 6.918e-05
11456       CARS       11_2    0.7546    25.64 5.757e-05
3114      LANCL1      2_124    0.7500  4675.56 1.043e-02
18766 ZNF561-AS1       19_9    0.7493  8544.87 1.905e-02
18767 ZNF561-AS1       19_9    0.7493  8544.87 1.905e-02
9273     TBC1D31       8_80    0.7382    25.36 5.571e-05
21923       GBF1      10_65    0.7340   295.14 6.445e-04
17767      EXOC7      17_42    0.7334    24.09 5.257e-05
                          intron_id      z num_eqtl
21963   intron_11_68911603_68914823  4.420        1
21975   intron_12_46366945_46367076  2.953        1
21738        intron_5_216859_218093  2.900        1
21903   intron_10_38094490_38095538  2.474        1
22168   intron_19_45677108_45677710 10.766        1
21870  intron_7_140348767_140351273 -4.575        2
18736   intron_19_10919680_10919877  5.161        1
6714     intron_6_32660249_32661347  3.643        1
4474   intron_3_167735823_167789111 -4.409        2
15718     intron_16_2037600_2037859 -4.967        1
2844   intron_2_160258046_160270202  5.596        1
3996     intron_3_66263379_66346364  4.492        1
5104     intron_4_76132719_76134175  4.259        1
11456     intron_11_3039931_3040896  4.623        2
3114   intron_2_210476412_210476620  3.497        1
18766     intron_19_9632417_9633460  2.621        1
18767     intron_19_9632417_9645074 -2.621        1
9273   intron_8_123130333_123134114 -4.335        1
21923 intron_10_102260116_102344051 -5.281        1
17767   intron_17_76101863_76103361  4.151        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>         genename region_tag susie_pip   mu2     PVE
21948     BDNF-AS      11_19         0 65329 0.00000
18968       MAST3      19_14         0 51168 0.00000
9562      CCDC171       9_13         0 45231 0.00000
9563      CCDC171       9_13         0 41910 0.00000
21558        NADK        1_1         0 38644 0.00000
21559        NADK        1_1         0 38644 0.00000
18970       MAST3      19_14         0 35618 0.00000
18971       MAST3      19_14         0 35278 0.00000
21963     IGHMBP2      11_38         1 27703 0.08242
21961      MRPL21      11_38         0 27137 0.00000
21947     BDNF-AS      11_19         0 26088 0.00000
14997  CATSPER2P1      15_16         0 22684 0.00000
21962      MRPL21      11_38         0 21890 0.00000
21944     BDNF-AS      11_19         0 21513 0.00000
21687       UBXN4       2_80         0 18980 0.00000
3884        HEMK1       3_35         0 18932 0.00000
3883      C3orf18       3_35         0 18927 0.00000
21557 RP1-283E3.8        1_1         0 18711 0.00000
4435     ARHGEF26       3_95         0 17761 0.00000
21567        GNB1        1_1         0 16877 0.00000
                         intron_id      z num_eqtl
21948  intron_11_27676140_27676982 12.379        1
18968  intron_19_18110741_18121685 -7.147        1
9562    intron_9_15888088_15888983 -7.903        1
9563    intron_9_15889101_15920270 -7.233        1
21558     intron_1_1753649_1754051  5.047        1
21559     intron_1_1753649_1754284 -5.047        1
18970  intron_19_18147017_18147443 -5.650        2
18971  intron_19_18147044_18147443  5.609        2
21963  intron_11_68911603_68914823  4.420        1
21961  intron_11_68898012_68900535  4.267        2
21947  intron_11_27659228_27676982 -1.202        1
14997  intron_15_43744219_43744395 -4.508        1
21962  intron_11_68898012_68900548 -4.068        2
21944  intron_11_27640005_27658241 -5.845        1
21687 intron_2_135754277_135755517 -3.871        1
3884    intron_3_50577573_50577826  4.701        1
3883    intron_3_50569574_50570931  4.682        1
21557     intron_1_1732523_1734690 -2.833        2
4435  intron_3_154152932_154187685 -4.115        1
21567     intron_1_1839238_1879787 -5.351        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>        genename region_tag susie_pip     mu2       PVE
21963    IGHMBP2      11_38   1.00000 27703.0 0.0824232
21975    SLC38A2      12_29   1.00000 16754.5 0.0498487
21738    CCDC127        5_1   0.99913 12476.1 0.0370873
21903     ZNF37A      10_28   0.99882 10226.3 0.0303897
13694      SPG20      13_13   0.55556 14365.0 0.0237441
13695      SPG20      13_13   0.55556 14365.0 0.0237441
13696      SPG20      13_13   0.55556 14365.0 0.0237441
18766 ZNF561-AS1       19_9   0.74929  8544.9 0.0190493
18767 ZNF561-AS1       19_9   0.74929  8544.9 0.0190493
22168       GIPR      19_32   0.93143  5642.0 0.0156352
3114      LANCL1      2_124   0.74998  4675.6 0.0104330
17181      KRT10      17_24   0.50001  3924.3 0.0058379
17182      KRT10      17_24   0.50001  3924.3 0.0058379
1674      TATDN3      1_108   0.25575  3455.9 0.0026296
1673      TATDN3      1_108   0.25575  3455.9 0.0026296
1671      TATDN3      1_108   0.21165  3455.0 0.0021757
6714    HLA-DQB1       6_26   0.78947   596.9 0.0014021
1669        NSL1      1_108   0.09212  3452.8 0.0009463
1668        NSL1      1_108   0.09186  3452.8 0.0009437
1667        NSL1      1_108   0.08566  3452.4 0.0008799
                         intron_id      z num_eqtl
21963  intron_11_68911603_68914823  4.420        1
21975  intron_12_46366945_46367076  2.953        1
21738       intron_5_216859_218093  2.900        1
21903  intron_10_38094490_38095538  2.474        1
13694  intron_13_36335832_36336287  2.819        1
13695  intron_13_36335832_36346240 -2.819        1
13696  intron_13_36335832_36346582  2.819        1
18766    intron_19_9632417_9633460  2.621        1
18767    intron_19_9632417_9645074 -2.621        1
22168  intron_19_45677108_45677710 10.766        1
3114  intron_2_210476412_210476620  3.497        1
17181  intron_17_40818482_40818767 -1.980        1
17182  intron_17_40818482_40818820  1.980        1
1674  intron_1_212807848_212812248  3.301        1
1673  intron_1_212807848_212812227 -3.301        1
1671  intron_1_212804429_212804596  3.299        1
6714    intron_6_32660249_32661347  3.643        1
1669  intron_1_212787637_212791530  3.258        1
1668  intron_1_212782426_212784363  3.258        1
1667  intron_1_212739601_212766189 -3.259        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag susie_pip      mu2       PVE
21945     BDNF-AS      11_19 0.000e+00  9768.90 0.000e+00
21946     BDNF-AS      11_19 0.000e+00  9768.90 0.000e+00
21948     BDNF-AS      11_19 0.000e+00 65329.18 0.000e+00
2061  DNAJC27-AS1       2_15 6.231e-05   192.41 3.567e-08
2062  DNAJC27-AS1       2_15 6.231e-05   192.41 3.567e-08
1544       SHISA4      1_102 1.551e-01   112.96 5.211e-05
16056      INO80E      16_24 1.750e-02    95.94 4.994e-06
22168        GIPR      19_32 9.314e-01  5642.01 1.564e-02
16033       SH2B1      16_23 1.924e-01    90.98 5.209e-05
16030      ATXN2L      16_23 1.775e-01    90.74 4.791e-05
16031      ATXN2L      16_23 1.722e-01    90.66 4.645e-05
16028      ATXN2L      16_23 1.654e-01    90.80 4.468e-05
16054       TAOK2      16_24 1.690e-02    95.13 4.783e-06
16055       TAOK2      16_24 1.690e-02    95.13 4.783e-06
16034       SH2B1      16_23 8.572e-02    88.76 2.264e-05
16013        CLN3      16_23 3.292e-02    90.82 8.897e-06
16067       MAPK3      16_24 1.972e-02    92.82 5.447e-06
16024     SULT1A1      16_23 3.546e-02    92.35 9.742e-06
16010        CLN3      16_23 2.753e-02    89.84 7.359e-06
16022     SULT1A2      16_23 6.075e-02    97.07 1.755e-05
                         intron_id      z num_eqtl
21945  intron_11_27640005_27659171  13.14        1
21946  intron_11_27658462_27659171 -13.14        1
21948  intron_11_27676140_27676982  12.38        1
2061    intron_2_24972232_24972626 -11.29        1
2062    intron_2_24972232_24979060  11.29        1
1544  intron_1_201890587_201891401  10.77        1
16056  intron_16_30001040_30005221 -10.77        1
22168  intron_19_45677108_45677710  10.77        1
16033  intron_16_28870811_28871780 -10.75        1
16030  intron_16_28835399_28835546  10.74        1
16031  intron_16_28835399_28835549 -10.74        1
16028  intron_16_28826390_28826862 -10.74        1
16054  intron_16_29989805_29990543  10.73        1
16055  intron_16_29989805_29990784 -10.73        1
16034  intron_16_28872758_28873170  10.65        1
16013  intron_16_28487541_28487662 -10.48        1
16067  intron_16_30114708_30115595  10.48        1
16024  intron_16_28607077_28608291  10.45        1
16010  intron_16_28477359_28477464 -10.42        1
16022  intron_16_28592481_28593252 -10.42        2</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.0173</code></pre>
<pre><code>         genename region_tag susie_pip      mu2       PVE
21945     BDNF-AS      11_19 0.000e+00  9768.90 0.000e+00
21946     BDNF-AS      11_19 0.000e+00  9768.90 0.000e+00
21948     BDNF-AS      11_19 0.000e+00 65329.18 0.000e+00
2061  DNAJC27-AS1       2_15 6.231e-05   192.41 3.567e-08
2062  DNAJC27-AS1       2_15 6.231e-05   192.41 3.567e-08
1544       SHISA4      1_102 1.551e-01   112.96 5.211e-05
16056      INO80E      16_24 1.750e-02    95.94 4.994e-06
22168        GIPR      19_32 9.314e-01  5642.01 1.564e-02
16033       SH2B1      16_23 1.924e-01    90.98 5.209e-05
16030      ATXN2L      16_23 1.775e-01    90.74 4.791e-05
16031      ATXN2L      16_23 1.722e-01    90.66 4.645e-05
16028      ATXN2L      16_23 1.654e-01    90.80 4.468e-05
16054       TAOK2      16_24 1.690e-02    95.13 4.783e-06
16055       TAOK2      16_24 1.690e-02    95.13 4.783e-06
16034       SH2B1      16_23 8.572e-02    88.76 2.264e-05
16013        CLN3      16_23 3.292e-02    90.82 8.897e-06
16067       MAPK3      16_24 1.972e-02    92.82 5.447e-06
16024     SULT1A1      16_23 3.546e-02    92.35 9.742e-06
16010        CLN3      16_23 2.753e-02    89.84 7.359e-06
16022     SULT1A2      16_23 6.075e-02    97.07 1.755e-05
                         intron_id      z num_eqtl
21945  intron_11_27640005_27659171  13.14        1
21946  intron_11_27658462_27659171 -13.14        1
21948  intron_11_27676140_27676982  12.38        1
2061    intron_2_24972232_24972626 -11.29        1
2062    intron_2_24972232_24979060  11.29        1
1544  intron_1_201890587_201891401  10.77        1
16056  intron_16_30001040_30005221 -10.77        1
22168  intron_19_45677108_45677710  10.77        1
16033  intron_16_28870811_28871780 -10.75        1
16030  intron_16_28835399_28835546  10.74        1
16031  intron_16_28835399_28835549 -10.74        1
16028  intron_16_28826390_28826862 -10.74        1
16054  intron_16_29989805_29990543  10.73        1
16055  intron_16_29989805_29990784 -10.73        1
16034  intron_16_28872758_28873170  10.65        1
16013  intron_16_28487541_28487662 -10.48        1
16067  intron_16_30114708_30115595  10.48        1
16024  intron_16_28607077_28608291  10.45        1
16010  intron_16_28477359_28477464 -10.42        1
16022  intron_16_28592481_28593252 -10.42        2</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 50</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                              Term Overlap
1                 histone H2A-K119 monoubiquitination (GO:0036353)     2/8
2 regulation of gene expression by genetic imprinting (GO:0006349)    2/11
3 dosage compensation by inactivation of X chromosome (GO:0009048)    2/12
  Adjusted.P.value       Genes
1          0.03983 PCGF5;PCGF3
2          0.03983 PCGF5;PCGF3
3          0.03983 PCGF5;PCGF3
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                           Term Overlap Adjusted.P.value
1                   sex chromosome (GO:0000803)     2/6         0.003075
2                     X chromosome (GO:0000805)     2/7         0.003075
3                     PRC1 complex (GO:0035102)    2/15         0.010129
4 clathrin-coated vesicle membrane (GO:0030665)    3/90         0.016572
                    Genes
1             PCGF5;PCGF3
2             PCGF5;PCGF3
3             PCGF5;PCGF3
4 NECAP2;DENND1A;HLA-DQB1
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                        Term Overlap
1 histone-arginine N-methyltransferase activity (GO:0008469)     2/7
2         arginine N-methyltransferase activity (GO:0016273)     2/8
3 protein-arginine N-methyltransferase activity (GO:0016274)    2/11
  Adjusted.P.value       Genes
1         0.006751 PRMT2;CARM1
2         0.006751 PRMT2;CARM1
3         0.008801 PRMT2;CARM1</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Warning in disease_enrichment(entities = genes, vocabulary = &quot;HGNC&quot;, database =
&quot;CURATED&quot;): Removing duplicates from input list.</code></pre>
<pre><code>                                                  Description     FDR Ratio
7                                                 Body Weight 0.03135  2/23
52                                      Kleine-Levin Syndrome 0.03135  1/23
66                              Progressive cerebellar ataxia 0.03135  1/23
95        SPINAL MUSCULAR ATROPHY WITH RESPIRATORY DISTRESS 1 0.03135  1/23
96  Familial encephalopathy with neuroserpin inclusion bodies 0.03135  1/23
107              CHARCOT-MARIE-TOOTH DISEASE, AXONAL, TYPE 2S 0.03135  1/23
108                          AMELOGENESIS IMPERFECTA, TYPE IH 0.03135  1/23
109          COMBINED OXIDATIVE PHOSPHORYLATION DEFICIENCY 28 0.03135  1/23
119        MITOCHONDRIAL COMPLEX I DEFICIENCY, NUCLEAR TYPE 9 0.03135  1/23
31                                              Megaesophagus 0.03757  1/23
    BgRatio
7   15/9703
52   1/9703
66   1/9703
95   1/9703
96   1/9703
107  1/9703
108  1/9703
109  1/9703
119  1/9703
31   2/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>Warning: ggrepel: 10 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 41</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 16</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.729</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 6</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 384</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>      genename region_tag susie_pip       mu2       PVE
21737  CCDC127        5_1  0.000000  5842.199 0.000e+00
21738  CCDC127        5_1  0.999131 12476.133 3.709e-02
21869  SLC37A3       7_86  0.016379    14.730 7.178e-07
21870  SLC37A3       7_86  0.877306    24.655 6.435e-05
21871  SLC37A3       7_86  0.007393    10.178 2.239e-07
21872  SLC37A3       7_86  0.006355     8.456 1.599e-07
21873  SLC37A3       7_86  0.005140     5.592 8.552e-08
21874  SLC37A3       7_86  0.006931    10.054 2.073e-07
21903   ZNF37A      10_28  0.998819 10226.250 3.039e-02
21904   ZNF37A      10_28  0.000000   674.300 0.000e+00
21905   ZNF37A      10_28  0.000000   221.583 0.000e+00
21906   ZNF37A      10_28  0.000000   672.802 0.000e+00
21907   ZNF37A      10_28  0.000000   490.625 0.000e+00
21908   ZNF37A      10_28  0.000000   490.625 0.000e+00
21963  IGHMBP2      11_38  1.000000 27703.006 8.242e-02
21964  IGHMBP2      11_38  0.000000   189.421 0.000e+00
21965  IGHMBP2      11_38  0.000000  2113.315 0.000e+00
21975  SLC38A2      12_29  1.000000 16754.503 4.985e-02
                         intron_id       z num_eqtl
21737       intron_5_205958_216729 -0.7549        1
21738       intron_5_216859_218093  2.9002        1
21869 intron_7_140337349_140348626  0.9305        1
21870 intron_7_140348767_140351273 -4.5748        2
21871 intron_7_140351451_140352062 -1.6921        2
21872 intron_7_140351969_140352062  1.4735        2
21873 intron_7_140352146_140355668  0.7598        1
21874 intron_7_140369682_140380282  1.7377        1
21903  intron_10_38094490_38095538  2.4741        1
21904  intron_10_38095224_38095358 -0.3934        1
21905  intron_10_38095629_38095744  0.3492        2
21906  intron_10_38095633_38095744 -0.2644        1
21907  intron_10_38115290_38117390 -1.6427        1
21908  intron_10_38115290_38146732  1.6427        1
21963  intron_11_68911603_68914823  4.4200        1
21964  intron_11_68929357_68933299  2.3995        1
21965  intron_11_68938354_68939534  0.9482        1
21975  intron_12_46366945_46367076  2.9534        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>  ctwas    TWAS 
0.02439 0.17073 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9992 0.9629 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.16667 0.01823 </code></pre>
<p><img src="figure/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1              forcats_0.5.1            
 [3] stringr_1.4.0             dplyr_1.0.7              
 [5] purrr_0.3.4               readr_2.1.1              
 [7] tidyr_1.1.4               tidyverse_1.3.1          
 [9] tibble_3.1.6              WebGestaltR_0.4.4        
[11] disgenet2r_0.99.2         enrichR_3.0              
[13] cowplot_1.0.0             ggplot2_3.3.5            
[15] EnsDb.Hsapiens.v79_2.99.0 ensembldb_2.8.0          
[17] AnnotationFilter_1.8.0    GenomicFeatures_1.36.3   
[19] AnnotationDbi_1.46.0      Biobase_2.44.0           
[21] GenomicRanges_1.36.1      GenomeInfoDb_1.20.0      
[23] IRanges_2.18.1            S4Vectors_0.22.1         
[25] BiocGenerics_0.30.0       workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            colorspace_2.0-2           
  [3] rjson_0.2.20                ellipsis_0.3.2             
  [5] rprojroot_2.0.2             XVector_0.24.0             
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                ggrepel_0.9.1              
 [11] bit64_4.0.5                 lubridate_1.8.0            
 [13] fansi_0.5.0                 xml2_1.3.3                 
 [15] codetools_0.2-16            doParallel_1.0.16          
 [17] cachem_1.0.6                knitr_1.36                 
 [19] jsonlite_1.7.2              apcluster_1.4.8            
 [21] Cairo_1.5-12.2              Rsamtools_2.0.0            
 [23] broom_0.7.10                dbplyr_2.1.1               
 [25] compiler_3.6.1              httr_1.4.2                 
 [27] backports_1.4.1             assertthat_0.2.1           
 [29] Matrix_1.2-18               fastmap_1.1.0              
 [31] lazyeval_0.2.2              cli_3.1.0                  
 [33] later_0.8.0                 htmltools_0.5.2            
 [35] prettyunits_1.1.1           tools_3.6.1                
 [37] igraph_1.2.10               gtable_0.3.0               
 [39] glue_1.5.1                  GenomeInfoDbData_1.2.1     
 [41] reshape2_1.4.4              doRNG_1.8.2                
 [43] Rcpp_1.0.7                  cellranger_1.1.0           
 [45] jquerylib_0.1.4             vctrs_0.3.8                
 [47] Biostrings_2.52.0           svglite_1.2.2              
 [49] rtracklayer_1.44.4          iterators_1.0.13           
 [51] xfun_0.29                   rvest_1.0.2                
 [53] lifecycle_1.0.1             rngtools_1.5.2             
 [55] XML_3.99-0.3                zlibbioc_1.30.0            
 [57] scales_1.1.1                vroom_1.5.7                
 [59] hms_1.1.1                   promises_1.0.1             
 [61] ProtGenerics_1.16.0         SummarizedExperiment_1.14.1
 [63] yaml_2.2.1                  curl_4.3.2                 
 [65] memoise_2.0.1               ggrastr_1.0.1              
 [67] gdtools_0.1.9               biomaRt_2.40.1             
 [69] stringi_1.7.6               RSQLite_2.2.8              
 [71] highr_0.9                   foreach_1.5.1              
 [73] BiocParallel_1.18.0         rlang_0.4.12               
 [75] pkgconfig_2.0.3             matrixStats_0.57.0         
 [77] bitops_1.0-7                evaluate_0.14              
 [79] lattice_0.20-38             GenomicAlignments_1.20.1   
 [81] labeling_0.4.2              bit_4.0.4                  
 [83] tidyselect_1.1.1            plyr_1.8.6                 
 [85] magrittr_2.0.1              R6_2.5.1                   
 [87] generics_0.1.1              DelayedArray_0.10.0        
 [89] DBI_1.1.1                   haven_2.4.3                
 [91] pillar_1.6.4                whisker_0.3-2              
 [93] withr_2.4.3                 RCurl_1.98-1.5             
 [95] modelr_0.1.8                crayon_1.4.2               
 [97] utf8_1.2.2                  tzdb_0.2.0                 
 [99] rmarkdown_2.11              progress_1.2.2             
[101] grid_3.6.1                  data.table_1.14.2          
[103] blob_1.2.2                  git2r_0.26.1               
[105] reprex_2.0.1                digest_0.6.29              
[107] httpuv_1.5.1                munsell_0.5.0              
[109] beeswarm_0.2.3              vipor_0.4.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
