<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Hypothalamus</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Hypothalamus</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreebbf6737bebe3c390e6f7b8e1ce35f818fda82fe3targetblankbbf6737a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/bbf6737bebe3c390e6f7b8e1ce35f818fda82fe3" target="_blank">bbf6737</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreebbf6737bebe3c390e6f7b8e1ce35f818fda82fe3targetblankbbf6737a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/bbf6737bebe3c390e6f7b8e1ce35f818fda82fe3" target="_blank">bbf6737</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Hypothalamus.Rmd</code>) and HTML (<code>docs/BMI_Brain_Hypothalamus.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bbf6737bebe3c390e6f7b8e1ce35f818fda82fe3/analysis/BMI_Brain_Hypothalamus.Rmd" target="_blank">bbf6737</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-21
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/91f38fa8b991bcb34c11575638dc25fc81457a5f/docs/BMI_Brain_Hypothalamus.html" target="_blank">91f38fa</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6/analysis/BMI_Brain_Hypothalamus.Rmd" target="_blank">eb13ecf</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e6bc16990373312d301ab183ec936aff373d2497/docs/BMI_Brain_Hypothalamus.html" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/87fee8b56a4f94471388b3fc74b464d3b6efdd41/analysis/BMI_Brain_Hypothalamus.Rmd" target="_blank">87fee8b</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 11083</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1107  757  667  439  549  622  531  429  416  429  660  610  213  357  369  492 
  17   18   19   20   21   22 
 680  168  851  327  134  276 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 8872</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.8005</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-5-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0049609 0.0002983 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
15.52 17.76 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 336107</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   11083 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.00254 0.11874 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1]  0.01272 15.93546</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>            genename region_tag susie_pip    mu2       PVE      z num_eqtl
7493           PPM1M       3_36    1.0000 244.75 7.282e-04  4.537        2
3276           CCND2       12_4    0.9398  28.26 7.901e-05 -5.120        1
7598           ZNF12        7_9    0.7814  26.90 6.255e-05  5.089        2
7840          ALKBH3      11_27    0.7683  28.50 6.515e-05 -5.128        2
13153 RP11-1109F11.3      12_54    0.7652  30.71 6.992e-05  6.457        1
8379           CENPX      17_46    0.7105  23.79 5.029e-05  4.111        2
8812            RARG      12_33    0.7026  25.49 5.329e-05 -4.106        1
241             ISL1       5_30    0.7014  26.16 5.458e-05 -5.010        1
7356        SERPINI1      3_103    0.6983  23.16 4.812e-05 -4.065        1
4821           DCAF7      17_38    0.6563  30.16 5.889e-05  5.437        1
3176          PRRC2C       1_84    0.6253  28.14 5.236e-05 -5.173        1
584             NGFR      17_29    0.6250  28.07 5.220e-05 -4.005        2
11412        NCKIPSD       3_34    0.6001  26.30 4.695e-05  4.491        1
13194   CTC-498M16.4       5_52    0.5917  52.95 9.321e-05  7.706        1
5798            ECE2      3_113    0.5827  28.52 4.943e-05 -5.287        1
5712          THSD7B       2_81    0.5485  27.38 4.469e-05  5.321        1
11611         HRAT92        7_1    0.5483  24.31 3.965e-05 -3.927        2
7806         R3HCC1L      10_62    0.5456  39.55 6.419e-05  7.439        1
5498           CARM1       19_9    0.5431  32.91 5.318e-05  5.016        1
155            CSDE1       1_71    0.5347  22.42 3.566e-05 -4.745        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>       genename region_tag susie_pip   mu2       PVE       z num_eqtl
7665    CCDC171       9_13 0.000e+00 50550 0.000e+00   8.034        2
8735      NEGR1       1_46 0.000e+00 43384 0.000e+00 -10.695        2
9420      STX19       3_59 0.000e+00 31106 0.000e+00  -5.060        1
7889       LEO1      15_21 4.128e-13 27970 3.435e-14   4.603        1
5271      MFAP1      15_16 0.000e+00 23765 0.000e+00   4.303        1
13397 LINC02019       3_35 1.110e-15 22551 7.449e-17  -4.468        2
5098      TMOD3      15_21 0.000e+00 22269 0.000e+00   5.412        1
4029      TMOD2      15_21 0.000e+00 21602 0.000e+00   5.232        1
1293      WDR76      15_16 0.000e+00 21487 0.000e+00   4.740        2
11601    CKMT1A      15_16 0.000e+00 21284 0.000e+00   4.130        1
2876       CISH       3_35 0.000e+00 20260 0.000e+00  -3.799        1
3017      PLCL1      2_117 0.000e+00 18664 0.000e+00  -5.642        1
1015      CCNT2       2_80 1.554e-15 18645 8.622e-17   3.686        1
2875      HEMK1       3_35 0.000e+00 17517 0.000e+00  -3.882        1
4998    TUBGCP4      15_16 0.000e+00 16916 0.000e+00   3.432        1
9416      DHFR2       3_59 0.000e+00 16605 0.000e+00   2.761        2
9414      NSUN3       3_59 0.000e+00 15678 0.000e+00   4.755        1
8261       ADAL      15_16 0.000e+00 14821 0.000e+00  -2.861        1
125    CACNA2D2       3_35 0.000e+00 14199 0.000e+00  -4.009        1
5136     CNOT6L       4_52 0.000e+00 14095 0.000e+00   3.422        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>            genename region_tag susie_pip     mu2       PVE      z num_eqtl
7493           PPM1M       3_36   1.00000  244.75 7.282e-04  4.537        2
2953          LANCL1      2_124   0.04150 4670.09 5.767e-04 -3.714        2
9392         FAM220A        7_8   0.21634  445.76 2.869e-04 -1.293        1
13194   CTC-498M16.4       5_52   0.59167   52.95 9.321e-05  7.706        1
4791            RAC1        7_8   0.15841  178.70 8.423e-05 -5.512        4
2896           SPCS1       3_36   0.08081  348.59 8.382e-05 -5.067        1
3276           CCND2       12_4   0.93978   28.26 7.901e-05 -5.120        1
13153 RP11-1109F11.3      12_54   0.76517   30.71 6.992e-05  6.457        1
12160         ATP5J2       7_62   0.45796   50.81 6.923e-05 -7.117        1
2926           ITGB6       2_96   0.49914   45.97 6.827e-05  5.451        1
7840          ALKBH3      11_27   0.76831   28.50 6.515e-05 -5.128        2
7806         R3HCC1L      10_62   0.54556   39.55 6.419e-05  7.439        1
7598           ZNF12        7_9   0.78142   26.90 6.255e-05  5.089        2
13639         DHRS11      17_22   0.34692   60.57 6.252e-05 -8.142        1
4821           DCAF7      17_38   0.65634   30.16 5.889e-05  5.437        1
241             ISL1       5_30   0.70137   26.16 5.458e-05 -5.010        1
8812            RARG      12_33   0.70263   25.49 5.329e-05 -4.106        1
5498           CARM1       19_9   0.54308   32.91 5.318e-05  5.016        1
3176          PRRC2C       1_84   0.62529   28.14 5.236e-05 -5.173        1
584             NGFR      17_29   0.62496   28.07 5.220e-05 -4.005        2</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>      genename region_tag susie_pip      mu2       PVE       z num_eqtl
7489     MST1R       3_35 1.814e-03  1050.67 5.671e-06 -12.628        2
38        RBM6       3_35 3.575e-04   906.71 9.645e-07  12.536        1
9046    KCTD13      16_24 4.543e-02   109.28 1.477e-05 -11.491        1
9045    ASPHD1      16_24 9.053e-03   101.21 2.726e-06 -11.337        2
7484    RNF123       3_35 4.941e-12   823.16 1.210e-14 -10.959        1
6178     TAOK2      16_24 1.619e-02    92.70 4.466e-06  10.738        1
8735     NEGR1       1_46 0.000e+00 43383.72 0.000e+00 -10.695        2
11930   NPIPB7      16_23 5.217e-02    86.01 1.335e-05  10.453        1
10430     CLN3      16_23 5.217e-02    86.01 1.335e-05  10.453        1
8365    INO80E      16_24 1.515e-02    78.23 3.526e-06  10.102        2
8032    ZNF646      16_24 5.108e-02    75.84 1.153e-05 -10.000        1
5486      SAE1      19_33 3.541e-03    97.46 1.027e-06   9.849        1
7487     CAMKV       3_35 3.331e-16  1446.64 1.434e-18   9.848        1
2753  COL4A3BP       5_44 1.987e-02    68.99 4.078e-06  -9.828        1
458      PRSS8      16_24 1.090e-02    70.69 2.292e-06  -9.765        1
1830      KAT8      16_24 9.421e-03    68.78 1.928e-06  -9.706        1
11411      LAT      16_23 1.061e-01    83.00 2.620e-05  -9.553        1
8031    ZNF668      16_24 1.101e-02    70.05 2.295e-06   9.550        2
2458     MTCH2      11_29 7.572e-03    81.32 1.832e-06  -9.514        1
10711  SULT1A2      16_23 4.339e-02    80.36 1.038e-05  -9.449        2</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.02012</code></pre>
<pre><code>      genename region_tag susie_pip      mu2       PVE       z num_eqtl
7489     MST1R       3_35 1.814e-03  1050.67 5.671e-06 -12.628        2
38        RBM6       3_35 3.575e-04   906.71 9.645e-07  12.536        1
9046    KCTD13      16_24 4.543e-02   109.28 1.477e-05 -11.491        1
9045    ASPHD1      16_24 9.053e-03   101.21 2.726e-06 -11.337        2
7484    RNF123       3_35 4.941e-12   823.16 1.210e-14 -10.959        1
6178     TAOK2      16_24 1.619e-02    92.70 4.466e-06  10.738        1
8735     NEGR1       1_46 0.000e+00 43383.72 0.000e+00 -10.695        2
11930   NPIPB7      16_23 5.217e-02    86.01 1.335e-05  10.453        1
10430     CLN3      16_23 5.217e-02    86.01 1.335e-05  10.453        1
8365    INO80E      16_24 1.515e-02    78.23 3.526e-06  10.102        2
8032    ZNF646      16_24 5.108e-02    75.84 1.153e-05 -10.000        1
5486      SAE1      19_33 3.541e-03    97.46 1.027e-06   9.849        1
7487     CAMKV       3_35 3.331e-16  1446.64 1.434e-18   9.848        1
2753  COL4A3BP       5_44 1.987e-02    68.99 4.078e-06  -9.828        1
458      PRSS8      16_24 1.090e-02    70.69 2.292e-06  -9.765        1
1830      KAT8      16_24 9.421e-03    68.78 1.928e-06  -9.706        1
11411      LAT      16_23 1.061e-01    83.00 2.620e-05  -9.553        1
8031    ZNF668      16_24 1.101e-02    70.05 2.295e-06   9.550        2
2458     MTCH2      11_29 7.572e-03    81.32 1.832e-06  -9.514        1
10711  SULT1A2      16_23 4.339e-02    80.36 1.038e-05  -9.449        2</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 23</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                                           Description     FDR
31                                          Neoplasm Recurrence, Local 0.03848
94           Familial encephalopathy with neuroserpin inclusion bodies 0.03848
106 MEGALENCEPHALY-POLYMICROGYRIA-POLYDACTYLY-HYDROCEPHALUS SYNDROME 3 0.03848
5                                                    Bladder Exstrophy 0.04184
15                                         Communicating Hydrocephalus 0.04184
23                                            Congenital Hydrocephalus 0.04184
41                                Congenital musculoskeletal anomalies 0.04184
48                                              POLYDACTYLY, POSTAXIAL 0.04184
54                                              Hydrocephalus Ex-Vacuo 0.04184
65                                        Post-Traumatic Hydrocephalus 0.04184
    Ratio BgRatio
31   2/10 39/9703
94   1/10  1/9703
106  1/10  1/9703
5    1/10  2/9703
15   1/10  7/9703
23   1/10  8/9703
41   1/10  6/9703
48   1/10  4/9703
54   1/10  7/9703
65   1/10  7/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 41</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 22</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.586</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 223</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>     genename region_tag susie_pip   mu2       PVE     z num_eqtl
7493    PPM1M       3_36         1 244.8 0.0007282 4.537        2</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.07317 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9998 0.9801 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.01345 </code></pre>
<p><img src="figure/BMI_Brain_Hypothalamus.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7      
 [5] purrr_0.3.4       readr_2.1.1       tidyr_1.1.4       tidyverse_1.3.1  
 [9] tibble_3.1.6      WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0      
[13] cowplot_1.0.0     ggplot2_3.3.5     workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       doParallel_1.0.16
 [5] httr_1.4.2        rprojroot_2.0.2   tools_3.6.1       backports_1.4.1  
 [9] doRNG_1.8.2       utf8_1.2.2        R6_2.5.1          vipor_0.4.5      
[13] DBI_1.1.1         colorspace_2.0-2  withr_2.4.3       ggrastr_1.0.1    
[17] tidyselect_1.1.1  bit_4.0.4         curl_4.3.2        compiler_3.6.1   
[21] git2r_0.26.1      cli_3.1.0         rvest_1.0.2       Cairo_1.5-12.2   
[25] xml2_1.3.3        labeling_0.4.2    scales_1.1.1      apcluster_1.4.8  
[29] digest_0.6.29     rmarkdown_2.11    svglite_1.2.2     pkgconfig_2.0.3  
[33] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[37] rlang_0.4.12      rstudioapi_0.13   RSQLite_2.2.8     jquerylib_0.1.4  
[41] farver_2.1.0      generics_0.1.1    jsonlite_1.7.2    vroom_1.5.7      
[45] magrittr_2.0.1    Matrix_1.2-18     ggbeeswarm_0.6.0  Rcpp_1.0.7       
[49] munsell_0.5.0     fansi_0.5.0       gdtools_0.1.9     lifecycle_1.0.1  
[53] stringi_1.7.6     whisker_0.3-2     yaml_2.2.1        plyr_1.8.6       
[57] grid_3.6.1        blob_1.2.2        ggrepel_0.9.1     parallel_3.6.1   
[61] promises_1.0.1    crayon_1.4.2      lattice_0.20-38   haven_2.4.3      
[65] hms_1.1.1         knitr_1.36        pillar_1.6.4      igraph_1.2.10    
[69] rjson_0.2.20      rngtools_1.5.2    reshape2_1.4.4    codetools_0.2-16 
[73] reprex_2.0.1      glue_1.5.1        evaluate_0.14     data.table_1.14.2
[77] modelr_0.1.8      vctrs_0.3.8       tzdb_0.2.0        httpuv_1.5.1     
[81] foreach_1.5.1     cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[85] cachem_1.0.6      xfun_0.29         broom_0.7.10      later_0.8.0      
[89] iterators_1.0.13  beeswarm_0.2.3    memoise_2.0.1     ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
