<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Anterior cingulate cortex BA24</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Anterior cingulate cortex BA24</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeeb13ecf45ad6ec66ccac1b6f91b74c140a7261d6targetblankeb13ecfa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6" target="_blank">eb13ecf</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeeb13ecf45ad6ec66ccac1b6f91b74c140a7261d6targetblankeb13ecfa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6" target="_blank">eb13ecf</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd</code>) and HTML (<code>docs/BMI_Brain_Anterior_cingulate_cortex_BA24.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6/analysis/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd" target="_blank">eb13ecf</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e6bc16990373312d301ab183ec936aff373d2497/docs/BMI_Brain_Anterior_cingulate_cortex_BA24.html" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/87fee8b56a4f94471388b3fc74b464d3b6efdd41/analysis/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd" target="_blank">87fee8b</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre><code>[1] 10914</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1077  759  640  423  520  617  523  418  401  429  656  600  225  367  374  510 
  17   18   19   20   21   22 
 639  170  840  329  119  278 </code></pre>
<pre><code>[1] 8792</code></pre>
<pre><code>[1] 0.8055708</code></pre>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<p><img src="figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-4-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>        gene          snp 
0.0075151663 0.0002940872 </code></pre>
<pre><code>    gene      snp 
20.13648 17.22258 </code></pre>
<pre><code>[1] 336107</code></pre>
<pre><code>[1]   10914 7535010</code></pre>
<pre><code>       gene         snp 
0.004913926 0.113548318 </code></pre>
<pre><code>[1]  0.1774827 15.9888151</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-5-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>       genename region_tag susie_pip         mu2          PVE         z
8261      MRPL1       4_52 1.0000000 18758.92578 5.581236e-02  3.486435
12013   CCDC169      13_13 1.0000000 11401.82993 3.392321e-02 -2.791355
2884   ARHGEF26       3_95 0.9999995 20272.03741 6.031421e-02 -4.170865
925      PIK3C3      18_23 0.9333936    52.03619 1.445083e-04  6.867052
12058      ETV5      3_114 0.8936853    95.62289 2.542547e-04  9.862284
9400    SHARPIN       8_94 0.8589433    25.30546 6.466976e-05 -4.430963
2457      STIM2       4_22 0.7910960    23.29647 5.483298e-05 -4.483915
4656      HTR1B       6_53 0.7782866    26.47483 6.130489e-05 -4.637032
3541      ZMIZ2       7_33 0.7564988    66.00229 1.485558e-04 -8.105339
9452         TH       11_2 0.7320834    27.22928 5.930881e-05  4.798775
9668      KCNB2       8_53 0.7286468    65.37647 1.417297e-04 -8.225507
279        CPS1      2_124 0.7268233  4573.07473 9.889165e-03 -3.534889
5792       ECE2      3_113 0.7217494    29.16568 6.262979e-05 -5.315245
9221     PNPLA2       11_1 0.7209733    26.40319 5.663671e-05  5.117727
318        VRK2       2_38 0.7085776    24.09783 5.080282e-05  3.526868
12683 LINC01977      17_45 0.6965391    28.35031 5.875243e-05  5.229978
7428      PPM1M       3_37 0.6865923   185.89444 3.797413e-04  4.213260
10559    SRGAP1      12_39 0.6839117    23.32861 4.746914e-05 -4.065215
7744    R3HCC1L      10_62 0.6724263    39.81237 7.964988e-05  7.438889
10515   ZKSCAN5       7_61 0.6438389    53.55690 1.025924e-04  7.133466
      num_eqtl
8261         1
12013        1
2884         1
925          2
12058        1
9400         1
2457         2
4656         1
3541         1
9452         1
9668         1
279          1
5792         1
9221         1
318          2
12683        1
7428         2
10559        2
7744         1
10515        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-6-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>          genename region_tag    susie_pip      mu2          PVE          z
10292      SLC38A3       3_35 0.000000e+00 65689.74 0.000000e+00   6.725828
7423         CAMKV       3_35 0.000000e+00 51451.22 0.000000e+00  -9.847856
7600       CCDC171       9_13 0.000000e+00 49126.82 0.000000e+00   8.020412
38            RBM6       3_35 0.000000e+00 39742.84 0.000000e+00  12.536042
9           SEMA3F       3_35 0.000000e+00 38958.32 0.000000e+00  12.510562
11183      DEFB134       8_15 0.000000e+00 37946.52 0.000000e+00  -6.778423
7425         MST1R       3_35 0.000000e+00 33940.06 0.000000e+00 -12.621262
4697          HEY2       6_84 0.000000e+00 32081.18 0.000000e+00   4.922936
9326         STX19       3_59 0.000000e+00 30090.45 0.000000e+00  -5.059656
12530 RP11-490G2.2       1_60 8.154418e-04 29860.63 7.244599e-05   5.044019
7824          LEO1      15_21 1.925138e-11 26586.28 1.522797e-12   4.647326
5275        LYSMD2      15_21 0.000000e+00 25404.36 0.000000e+00  -4.402599
5271         MFAP1      15_16 0.000000e+00 22989.16 0.000000e+00   4.302998
11993         HYPK      15_16 0.000000e+00 22890.18 0.000000e+00   4.322039
9321         DHFR2       3_59 0.000000e+00 22806.25 0.000000e+00   3.750765
7420        RNF123       3_35 0.000000e+00 22473.32 0.000000e+00 -10.959165
11448       CKMT1A      15_16 0.000000e+00 20589.76 0.000000e+00   4.129652
2884      ARHGEF26       3_95 9.999995e-01 20272.04 6.031421e-02  -4.170865
8261         MRPL1       4_52 1.000000e+00 18758.93 5.581236e-02   3.486435
5134        CNOT6L       4_52 0.000000e+00 16965.72 0.000000e+00   3.412987
      num_eqtl
10292        1
7423         1
7600         2
38           1
9            1
11183        1
7425         2
4697         1
9326         1
12530        1
7824         1
5275         1
5271         1
11993        1
9321         2
7420         1
11448        1
2884         1
8261         1
5134         1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>          genename region_tag    susie_pip         mu2          PVE          z
2884      ARHGEF26       3_95 0.9999995207 20272.03741 6.031421e-02  -4.170865
8261         MRPL1       4_52 1.0000000000 18758.92578 5.581236e-02   3.486435
12013      CCDC169      13_13 0.9999999966 11401.82993 3.392321e-02  -2.791355
279           CPS1      2_124 0.7268233484  4573.07473 9.889165e-03  -3.534889
9576       C3orf58       3_88 0.0832453487  8119.53387 2.011007e-03   2.513826
7428         PPM1M       3_37 0.6865922953   185.89444 3.797413e-04   4.213260
7455      TMEM161B       5_52 0.0251787538  4397.98701 3.294660e-04  -8.071918
12058         ETV5      3_114 0.8936853101    95.62289 2.542547e-04   9.862284
12992 CTC-498M16.4       5_52 0.0095883283  5230.50810 1.492139e-04   7.699937
3541         ZMIZ2       7_33 0.7564988371    66.00229 1.485558e-04  -8.105339
925         PIK3C3      18_23 0.9333936461    52.03619 1.445083e-04   6.867052
9668         KCNB2       8_53 0.7286467567    65.37647 1.417297e-04  -8.225507
6613         GPR61       1_67 0.5486703588    79.16926 1.292381e-04   8.755235
5135          USO1       4_51 0.3440095019   118.68613 1.214767e-04  -2.133639
10515      ZKSCAN5       7_61 0.6438389275    53.55690 1.025924e-04   7.133466
7849          MC4R      18_33 0.2648612509   128.45959 1.012296e-04  13.311794
9057         NUPR1      16_23 0.4846290022    68.57940 9.888389e-05 -10.531969
13435       DHRS11      17_22 0.5080182943    61.80695 9.341983e-05  -8.142012
7744       R3HCC1L      10_62 0.6724262976    39.81237 7.964988e-05   7.438889
12530 RP11-490G2.2       1_60 0.0008154418 29860.62951 7.244599e-05   5.044019
      num_eqtl
2884         1
8261         1
12013        1
279          1
9576         1
7428         2
7455         2
12058        1
12992        2
3541         1
925          2
9668         1
6613         1
5135         1
10515        1
7849         1
9057         2
13435        1
7744         1
12530        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>      genename region_tag   susie_pip         mu2          PVE          z
7849      MC4R      18_33 0.264861251   128.45959 1.012296e-04  13.311794
7425     MST1R       3_35 0.000000000 33940.05530 0.000000e+00 -12.621262
38        RBM6       3_35 0.000000000 39742.83773 0.000000e+00  12.536042
9       SEMA3F       3_35 0.000000000 38958.32298 0.000000e+00  12.510562
7420    RNF123       3_35 0.000000000 22473.32267 0.000000e+00 -10.959165
1778     MAPK3      16_24 0.020311269    98.48382 5.951472e-06  10.880016
6171     TAOK2      16_24 0.023429174    94.27317 6.571545e-06  10.737701
8291    INO80E      16_24 0.023393272    94.16156 6.553707e-06  10.733559
9057     NUPR1      16_23 0.484629002    68.57940 9.888389e-05 -10.531969
10490  SULT1A1      16_23 0.048304398    62.62825 9.000764e-06  10.415275
11757   NPIPB7      16_23 0.024446107    63.77065 4.638238e-06  10.037986
7963    ZNF668      16_24 0.100918865    77.12944 2.315874e-05  10.000364
7964    ZNF646      16_24 0.100918865    77.12944 2.315874e-05 -10.000364
10597  SULT1A2      16_23 0.013171246    56.46886 2.212882e-06  -9.958386
12058     ETV5      3_114 0.893685310    95.62289 2.542547e-04   9.862284
5471      SAE1      19_33 0.004569095    99.74287 1.355921e-06   9.848747
7423     CAMKV       3_35 0.000000000 51451.21504 0.000000e+00  -9.847856
457      PRSS8      16_24 0.017222207    72.05362 3.692045e-06  -9.764760
11270      LAT      16_23 0.220743354    56.62442 3.718894e-05  -9.552834
2477     MTCH2      11_29 0.010422084    83.55427 2.590870e-06  -9.551496
      num_eqtl
7849         1
7425         2
38           1
9            1
7420         1
1778         1
6171         1
8291         1
9057         2
10490        1
11757        1
7963         1
7964         1
10597        2
12058        1
5471         1
7423         1
457          1
11270        1
2477         1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-9-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-9-2.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.02272311</code></pre>
<pre><code>      genename region_tag   susie_pip         mu2          PVE          z
7849      MC4R      18_33 0.264861251   128.45959 1.012296e-04  13.311794
7425     MST1R       3_35 0.000000000 33940.05530 0.000000e+00 -12.621262
38        RBM6       3_35 0.000000000 39742.83773 0.000000e+00  12.536042
9       SEMA3F       3_35 0.000000000 38958.32298 0.000000e+00  12.510562
7420    RNF123       3_35 0.000000000 22473.32267 0.000000e+00 -10.959165
1778     MAPK3      16_24 0.020311269    98.48382 5.951472e-06  10.880016
6171     TAOK2      16_24 0.023429174    94.27317 6.571545e-06  10.737701
8291    INO80E      16_24 0.023393272    94.16156 6.553707e-06  10.733559
9057     NUPR1      16_23 0.484629002    68.57940 9.888389e-05 -10.531969
10490  SULT1A1      16_23 0.048304398    62.62825 9.000764e-06  10.415275
11757   NPIPB7      16_23 0.024446107    63.77065 4.638238e-06  10.037986
7963    ZNF668      16_24 0.100918865    77.12944 2.315874e-05  10.000364
7964    ZNF646      16_24 0.100918865    77.12944 2.315874e-05 -10.000364
10597  SULT1A2      16_23 0.013171246    56.46886 2.212882e-06  -9.958386
12058     ETV5      3_114 0.893685310    95.62289 2.542547e-04   9.862284
5471      SAE1      19_33 0.004569095    99.74287 1.355921e-06   9.848747
7423     CAMKV       3_35 0.000000000 51451.21504 0.000000e+00  -9.847856
457      PRSS8      16_24 0.017222207    72.05362 3.692045e-06  -9.764760
11270      LAT      16_23 0.220743354    56.62442 3.718894e-05  -9.552834
2477     MTCH2      11_29 0.010422084    83.55427 2.590870e-06  -9.551496
      num_eqtl
7849         1
7425         2
38           1
9            1
7420         1
1778         1
6171         1
8291         1
9057         2
10490        1
11757        1
7963         1
7964         1
10597        2
12058        1
5471         1
7423         1
457          1
11270        1
2477         1</code></pre>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre><code>[1] 41</code></pre>
<pre><code>[1] 24</code></pre>
<pre><code>[1] 4.583103</code></pre>
<pre><code>[1] 6</code></pre>
<pre><code>[1] 248</code></pre>
<pre><code>      genename region_tag susie_pip         mu2          PVE         z num_eqtl
2884  ARHGEF26       3_95 0.9999995 20272.03741 6.031421e-02 -4.170865        1
8261     MRPL1       4_52 1.0000000 18758.92578 5.581236e-02  3.486435        1
9400   SHARPIN       8_94 0.8589433    25.30546 6.466976e-05 -4.430963        1
12013  CCDC169      13_13 1.0000000 11401.82993 3.392321e-02 -2.791355        1</code></pre>
<pre><code>     ctwas       TWAS 
0.00000000 0.07317073 </code></pre>
<pre><code>    ctwas      TWAS 
0.9994490 0.9775023 </code></pre>
<pre><code>     ctwas       TWAS 
0.00000000 0.01209677 </code></pre>
<p><img src="figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd/unnamed-chunk-10-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readxl_1.3.1    cowplot_1.0.0   ggplot2_3.3.5   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1  xfun_0.29         purrr_0.3.4       colorspace_2.0-2 
 [5] vctrs_0.3.8       generics_0.1.1    htmltools_0.5.2   yaml_2.2.1       
 [9] utf8_1.2.2        blob_1.2.2        rlang_0.4.12      jquerylib_0.1.4  
[13] later_0.8.0       pillar_1.6.4      glue_1.5.1        withr_2.4.3      
[17] DBI_1.1.1         bit64_4.0.5       lifecycle_1.0.1   stringr_1.4.0    
[21] cellranger_1.1.0  munsell_0.5.0     gtable_0.3.0      evaluate_0.14    
[25] memoise_2.0.1     labeling_0.4.2    knitr_1.36        fastmap_1.1.0    
[29] httpuv_1.5.1      fansi_0.5.0       highr_0.9         Rcpp_1.0.7       
[33] promises_1.0.1    scales_1.1.1      cachem_1.0.6      farver_2.1.0     
[37] fs_1.5.2          bit_4.0.4         digest_0.6.29     stringi_1.7.6    
[41] dplyr_1.0.7       rprojroot_2.0.2   grid_3.6.1        tools_3.6.1      
[45] magrittr_2.0.1    tibble_3.1.6      RSQLite_2.2.8     crayon_1.4.2     
[49] whisker_0.3-2     pkgconfig_2.0.3   ellipsis_0.3.2    data.table_1.14.2
[53] assertthat_0.2.1  rmarkdown_2.11    R6_2.5.1          git2r_0.26.1     
[57] compiler_3.6.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
