<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Cortex</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Cortex</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Substantia_nigra_S.Rmd</code>) and HTML (<code>docs/BMI_Brain_Substantia_nigra_S.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1058990dac15746212ef84c84aa7cc13dc9948bf/analysis/BMI_Brain_Substantia_nigra_S.Rmd" target="_blank">1058990</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 17941</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1660 1259 1029  722  677  898 1022  619  742  824 1064  869  349  624  606  843 
  17   18   19   20   21   22 
1196  235 1302  618  253  530 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 16639</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.9274</code></pre>
<pre class="r"><code>#add z scores to results
load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_gene.Rd&quot;))
ctwas_gene_res$z &lt;- z_gene[ctwas_gene_res$intron_id,]$z

z_snp &lt;- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z &lt;- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$intron_id=NA

ctwas_res &lt;- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#get number of eQTL for geens
num_eqtl &lt;- c()
for (i in 1:22){
  load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, i, &quot;.exprqc.Rd&quot;))
  num_eqtl &lt;- c(num_eqtl, unlist(lapply(wgtlist, nrow)))
}
ctwas_gene_res$num_eqtl &lt;- num_eqtl[ctwas_gene_res$intron_id]

#store columns to report
report_cols &lt;- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c(&quot;type&quot;, &quot;region_tag1&quot;, &quot;region_tag2&quot;, &quot;cs_index&quot;, &quot;gene_type&quot;, &quot;z_flag&quot;, &quot;id&quot;, &quot;chrom&quot;, &quot;pos&quot;))]
first_cols &lt;- c(&quot;genename&quot;, &quot;region_tag&quot;)
report_cols &lt;- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps &lt;- c(&quot;id&quot;, report_cols[-1])
report_cols_snps &lt;- report_cols_snps[!(report_cols_snps %in% &quot;num_eqtl&quot;)]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s1.susieIrss.txt&quot;))
n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
rm(ctwas_res_s1)</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0077201 0.0002893 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
26.53 16.47 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 336107</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   17941 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.01093 0.10681 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1]  0.155 16.092</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>             genename region_tag susie_pip      mu2       PVE
17781             MOK      14_53    1.0000  9581.22 2.851e-02
17576             DST       6_42    1.0000 11325.01 3.369e-02
17702           SPG20      13_13    0.9961 15543.69 4.606e-02
17790 TNFSF12-TNFSF13       17_7    0.9399    29.11 8.139e-05
17498          EIF2B5      3_113    0.9073    38.06 1.027e-04
17599          PMS2P3       7_48    0.8810    86.89 2.278e-04
17540           RGS12        4_4    0.8723    37.67 9.778e-05
3625         SERPINI1      3_103    0.7968    23.94 5.676e-05
3805            PCGF3        4_2    0.7888    24.68 5.792e-05
7128         SLC25A37       8_24    0.7886    27.02 6.339e-05
14809           DOT1L       19_3    0.7618    23.25 5.271e-05
16862             BID       22_2    0.7175    27.78 5.930e-05
524              NASP       1_28    0.7128    27.17 5.762e-05
12854          TNRC6A      16_21    0.7071   446.99 9.403e-04
17659   RP11-386G11.5      12_31    0.7017  1484.82 3.100e-03
8195           PRRC2B       9_69    0.6941    22.21 4.586e-05
3263         SLC25A26       3_45    0.6891    24.24 4.970e-05
10318         SLCO1A2      12_16    0.6849    25.49 5.194e-05
5985         FLJ44511        7_1    0.6822    22.46 4.558e-05
6911            KCNH2       7_93    0.6671    39.68 7.875e-05
                          intron_id      z num_eqtl
17781 intron_14_102265912_102283478  2.804        1
17576    intron_6_56464756_56466078  2.881        1
17702   intron_13_36312477_36313968  2.776        1
17790     intron_17_7559702_7560049 -4.225        1
17498  intron_3_184278313_184283785  6.068        2
17599    intron_7_75514779_75515718  9.959        1
17540      intron_4_3293099_3316070  6.796        1
3625   intron_3_167735823_167789111 -4.405        2
3805         intron_4_731110_732400 -4.645        2
7128     intron_8_23529212_23566108  5.008        1
14809     intron_19_2199939_2202700  4.106        1
16862   intron_22_17739488_17773610  5.130        1
524      intron_1_45600457_45602255  5.388        1
12854   intron_16_24730300_24750726  5.887        1
17659   intron_12_49005543_49005743 -4.788        1
8195   intron_9_131488096_131491422  3.949        1
3263     intron_3_66263379_66304453 -4.417        1
10318   intron_12_21319579_21324594 -4.572        1
5985         intron_7_524280_524854  3.984        2
6911   intron_7_150951120_150951448 -6.291        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>      genename region_tag susie_pip   mu2       PVE
17371   MMP23B        1_1 0.000e+00 28198 0.000e+00
17375     NADK        1_1 0.000e+00 27440 0.000e+00
15171    MAST3      19_14 0.000e+00 24187 0.000e+00
15169    MAST3      19_14 0.000e+00 23926 0.000e+00
15170    MAST3      19_14 0.000e+00 23868 0.000e+00
17642  BDNF-AS      11_19 0.000e+00 23839 0.000e+00
15172    MAST3      19_14 0.000e+00 23484 0.000e+00
9798    MRPL21      11_38 4.880e-10 20928 3.039e-11
9802   IGHMBP2      11_38 1.722e-07 20831 1.068e-08
9801   IGHMBP2      11_38 6.743e-07 20789 4.171e-08
2449     PLCL1      2_117 1.001e-08 17642 5.257e-10
3165     HEMK1       3_35 0.000e+00 17319 0.000e+00
3164     HEMK1       3_35 0.000e+00 17318 0.000e+00
17473   WDSUB1       2_96 0.000e+00 16665 0.000e+00
17702    SPG20      13_13 9.961e-01 15544 4.606e-02
17703    SPG20      13_13 5.563e-04 15516 2.568e-05
17704    SPG20      13_13 4.241e-02 15514 1.958e-03
17705    SPG20      13_13 4.241e-02 15514 1.958e-03
17706    SPG20      13_13 4.241e-02 15514 1.958e-03
17380     GNB1        1_1 0.000e+00 15433 0.000e+00
                         intron_id      z num_eqtl
17371     intron_1_1676846_1692449  4.905        1
17375     intron_1_1755476_1756258 -4.601        1
15171  intron_19_18147017_18147443 -5.270        2
15169  intron_19_18107618_18121685  5.673        1
15170  intron_19_18110366_18110652  5.647        1
17642  intron_11_27659228_27676982 -1.202        1
15172  intron_19_18147044_18147443  5.187        2
9798   intron_11_68898012_68900535  4.240        2
9802   intron_11_68930434_68933299 -4.493        1
9801   intron_11_68929357_68933299  4.575        2
2449  intron_2_197805339_198001952 -5.642        1
3165    intron_3_50577573_50577826  4.701        1
3164    intron_3_50572208_50577052 -4.704        1
17473 intron_2_159257985_159259810  4.164        1
17702  intron_13_36312477_36313968  2.776        1
17703  intron_13_36312477_36314227 -2.722        2
17704  intron_13_36335832_36346225 -2.819        1
17705  intron_13_36335832_36346240 -2.819        1
17706  intron_13_36335832_36346582  2.819        1
17380     intron_1_1839238_1890820  5.351        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>           genename region_tag susie_pip      mu2       PVE
17702         SPG20      13_13   0.99607 15543.69 0.0460647
17576           DST       6_42   1.00000 11325.01 0.0336945
17781           MOK      14_53   1.00000  9581.22 0.0285065
1369         TATDN3      1_108   0.34685  3387.57 0.0034958
17659 RP11-386G11.5      12_31   0.70166  1484.82 0.0030997
1366         TATDN3      1_108   0.22233  3386.80 0.0022403
1368         TATDN3      1_108   0.22233  3386.80 0.0022403
17705         SPG20      13_13   0.04241 15513.67 0.0019577
17704         SPG20      13_13   0.04241 15513.67 0.0019577
17706         SPG20      13_13   0.04241 15513.67 0.0019577
1364         TATDN3      1_108   0.17705  3385.74 0.0017834
1365         TATDN3      1_108   0.16655  3385.60 0.0016776
17555  TMEM161B-AS1       5_52   0.12335  4407.43 0.0016175
1359           NSL1      1_108   0.11578  3381.75 0.0011649
12854        TNRC6A      16_21   0.70705   446.99 0.0009403
5247          PGBD1       6_22   0.01893 12742.78 0.0007175
1367         TATDN3      1_108   0.06123  3370.79 0.0006141
8749          BTAF1      10_59   0.33147   267.47 0.0002638
17599        PMS2P3       7_48   0.88105    86.89 0.0002278
3208          ITIH4       3_36   0.43989   173.74 0.0002274
                          intron_id      z num_eqtl
17702   intron_13_36312477_36313968  2.776        1
17576    intron_6_56464756_56466078  2.881        1
17781 intron_14_102265912_102283478  2.804        1
1369   intron_1_212812328_212815013  3.295        1
17659   intron_12_49005543_49005743 -4.788        1
1366   intron_1_212807848_212812227 -3.301        1
1368   intron_1_212807848_212812251 -3.301        1
17705   intron_13_36335832_36346240 -2.819        1
17704   intron_13_36335832_36346225 -2.819        1
17706   intron_13_36335832_36346582  2.819        1
1364   intron_1_212804429_212804596  3.300        1
1365   intron_1_212804651_212807736  3.301        1
17555    intron_5_88270585_88282042 -8.048        1
1359   intron_1_212738686_212739534 -3.259        1
12854   intron_16_24730300_24750726  5.887        1
5247     intron_6_28281918_28283776 -3.377        1
1367   intron_1_212807848_212812248  3.365        2
8749    intron_10_91996570_91997080 -2.915        1
17599    intron_7_75514779_75515718  9.959        1
3208     intron_3_52818536_52819393  6.000        2</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>          genename region_tag susie_pip     mu2       PVE
17641      BDNF-AS      11_19 0.000e+00 9001.51 0.000e+00
3145          RBM6       3_35 4.921e-03  843.54 1.235e-05
17640      BDNF-AS      11_19 0.000e+00 9677.57 0.000e+00
1678   DNAJC27-AS1       2_15 1.074e-04  191.78 6.129e-08
1679   DNAJC27-AS1       2_15 1.074e-04  191.78 6.129e-08
1680   DNAJC27-AS1       2_15 1.074e-04  191.78 6.129e-08
3147          RBM6       3_35 2.016e-13  156.43 9.383e-17
15610         GIPR      19_32 5.626e-01  116.75 1.954e-04
12923       INO80E      16_24 2.299e-02   96.99 6.635e-06
15609         GIPR      19_32 3.937e-01  115.96 1.358e-04
12906        SH2B1      16_23 2.396e-01   87.23 6.217e-05
12905       ATXN2L      16_23 2.090e-01   86.83 5.400e-05
12904       ATXN2L      16_23 2.076e-01   86.81 5.361e-05
12902 RP11-57A19.4      16_23 1.828e-01   86.79 4.720e-05
1182        SEC16B       1_87 3.122e-02   95.87 8.907e-06
12896      SULT1A2      16_23 9.591e-02   94.94 2.709e-05
12887         CLN3      16_23 4.117e-02   86.92 1.065e-05
12897      SULT1A2      16_23 7.622e-02   92.01 2.086e-05
5457      C6orf106       6_28 2.666e-05  121.52 9.637e-09
12901      SULT1A1      16_23 2.944e-02   81.25 7.117e-06
                         intron_id      z num_eqtl
17641  intron_11_27658462_27659171 -13.14        1
3145    intron_3_49999513_50046716  12.54        1
17640  intron_11_27640005_27659171  12.11        1
1678    intron_2_24972232_24972626 -11.29        1
1679    intron_2_24972232_24979060  11.29        1
1680    intron_2_24972232_25001415  11.29        1
3147    intron_3_50047326_50048245  10.81        2
15610  intron_19_45677779_45677906  10.80        1
12923  intron_16_30001040_30005221 -10.77        1
15609  intron_19_45677108_45677710  10.77        1
12906  intron_16_28869383_28871780  10.76        1
12905  intron_16_28835399_28835549 -10.74        1
12904  intron_16_28835399_28835546  10.74        1
12902  intron_16_28715434_28723164 -10.72        1
1182  intron_1_177932569_177932698 -10.59        1
12896  intron_16_28592443_28593252 -10.57        1
12887  intron_16_28487541_28487662 -10.48        1
12897  intron_16_28592481_28593252 -10.42        2
5457    intron_6_34654779_34696446 -10.24        1
12901  intron_16_28620133_28623131  10.07        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.01728</code></pre>
<pre><code>          genename region_tag susie_pip     mu2       PVE
17641      BDNF-AS      11_19 0.000e+00 9001.51 0.000e+00
3145          RBM6       3_35 4.921e-03  843.54 1.235e-05
17640      BDNF-AS      11_19 0.000e+00 9677.57 0.000e+00
1678   DNAJC27-AS1       2_15 1.074e-04  191.78 6.129e-08
1679   DNAJC27-AS1       2_15 1.074e-04  191.78 6.129e-08
1680   DNAJC27-AS1       2_15 1.074e-04  191.78 6.129e-08
3147          RBM6       3_35 2.016e-13  156.43 9.383e-17
15610         GIPR      19_32 5.626e-01  116.75 1.954e-04
12923       INO80E      16_24 2.299e-02   96.99 6.635e-06
15609         GIPR      19_32 3.937e-01  115.96 1.358e-04
12906        SH2B1      16_23 2.396e-01   87.23 6.217e-05
12905       ATXN2L      16_23 2.090e-01   86.83 5.400e-05
12904       ATXN2L      16_23 2.076e-01   86.81 5.361e-05
12902 RP11-57A19.4      16_23 1.828e-01   86.79 4.720e-05
1182        SEC16B       1_87 3.122e-02   95.87 8.907e-06
12896      SULT1A2      16_23 9.591e-02   94.94 2.709e-05
12887         CLN3      16_23 4.117e-02   86.92 1.065e-05
12897      SULT1A2      16_23 7.622e-02   92.01 2.086e-05
5457      C6orf106       6_28 2.666e-05  121.52 9.637e-09
12901      SULT1A1      16_23 2.944e-02   81.25 7.117e-06
                         intron_id      z num_eqtl
17641  intron_11_27658462_27659171 -13.14        1
3145    intron_3_49999513_50046716  12.54        1
17640  intron_11_27640005_27659171  12.11        1
1678    intron_2_24972232_24972626 -11.29        1
1679    intron_2_24972232_24979060  11.29        1
1680    intron_2_24972232_25001415  11.29        1
3147    intron_3_50047326_50048245  10.81        2
15610  intron_19_45677779_45677906  10.80        1
12923  intron_16_30001040_30005221 -10.77        1
15609  intron_19_45677108_45677710  10.77        1
12906  intron_16_28869383_28871780  10.76        1
12905  intron_16_28835399_28835549 -10.74        1
12904  intron_16_28835399_28835546  10.74        1
12902  intron_16_28715434_28723164 -10.72        1
1182  intron_1_177932569_177932698 -10.59        1
12896  intron_16_28592443_28593252 -10.57        1
12887  intron_16_28487541_28487662 -10.48        1
12897  intron_16_28592481_28593252 -10.42        2
5457    intron_6_34654779_34696446 -10.24        1
12901  intron_16_28620133_28623131  10.07        1</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 41</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                                    Term
1 regulation of potassium ion export across plasma membrane (GO:1903764)
2             signal transduction in response to DNA damage (GO:0042770)
  Overlap Adjusted.P.value           Genes
1     2/7          0.03345      KCNH2;DLG1
2    3/52          0.03345 DOT1L;BID;HINFP
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                                 Description     FDR Ratio
5                                                Body Weight 0.03264  2/20
59                             Progressive cerebellar ataxia 0.03264  1/20
82 Familial encephalopathy with neuroserpin inclusion bodies 0.03264  1/20
89     NEUROPATHY, HEREDITARY SENSORY AND AUTONOMIC, TYPE VI 0.03264  1/20
92      EPIDERMOLYSIS BULLOSA SIMPLEX, AUTOSOMAL RECESSIVE 2 0.03264  1/20
93          COMBINED OXIDATIVE PHOSPHORYLATION DEFICIENCY 28 0.03264  1/20
19                                            Cardiac Arrest 0.05763  1/20
27                                         Muscle Spasticity 0.05763  1/20
35                                             Schizophrenia 0.05763  6/20
38                                       Torsades de Pointes 0.05763  1/20
    BgRatio
5   15/9703
59   1/9703
82   1/9703
89   1/9703
92   1/9703
93   1/9703
19   5/9703
27   5/9703
35 883/9703
38   3/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>Warning: ggrepel: 8 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 41</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 16</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.686</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 7</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 310</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>             genename region_tag susie_pip       mu2       PVE
11838             MOK      14_54 3.877e-03     7.674 8.853e-08
17573             DST       6_42 0.000e+00   195.435 0.000e+00
17574             DST       6_42 0.000e+00  1543.673 0.000e+00
17575             DST       6_42 0.000e+00  1552.793 0.000e+00
17576             DST       6_42 1.000e+00 11325.007 3.369e-02
17577             DST       6_42 0.000e+00   368.171 0.000e+00
17702           SPG20      13_13 9.961e-01 15543.691 4.606e-02
17703           SPG20      13_13 5.563e-04 15515.542 2.568e-05
17704           SPG20      13_13 4.241e-02 15513.667 1.958e-03
17705           SPG20      13_13 4.241e-02 15513.667 1.958e-03
17706           SPG20      13_13 4.241e-02 15513.667 1.958e-03
17772             MOK      14_53 6.280e-05  9383.727 1.753e-06
17773             MOK      14_53 1.009e-05    70.695 2.123e-09
17774             MOK      14_53 9.967e-06    70.396 2.088e-09
17775             MOK      14_53 5.234e-06  2072.466 3.228e-08
17776             MOK      14_53 5.234e-06  2072.466 3.228e-08
17777             MOK      14_53 5.234e-06  2072.466 3.228e-08
17778             MOK      14_53 5.040e-06  6505.599 9.755e-08
17779             MOK      14_53 6.032e-06  4031.677 7.236e-08
17780             MOK      14_53 5.941e-06  4039.599 7.140e-08
17781             MOK      14_53 1.000e+00  9581.221 2.851e-02
17790 TNFSF12-TNFSF13       17_7 9.399e-01    29.105 8.139e-05
                          intron_id       z num_eqtl
11838 intron_14_102233789_102250812 -1.2728        1
17573    intron_6_56463156_56463565  0.7310        1
17574    intron_6_56463764_56464685 -1.0970        2
17575    intron_6_56463764_56466078  1.1002        2
17576    intron_6_56464756_56466078  2.8806        1
17577    intron_6_56598006_56600069 -3.7368        1
17702   intron_13_36312477_36313968  2.7757        1
17703   intron_13_36312477_36314227 -2.7224        2
17704   intron_13_36335832_36346225 -2.8188        1
17705   intron_13_36335832_36346240 -2.8188        1
17706   intron_13_36335832_36346582  2.8188        1
17772 intron_14_102226408_102232535  2.9326        1
17773 intron_14_102229657_102231707 -1.2704        1
17774 intron_14_102229657_102232535  1.2592        1
17775 intron_14_102250987_102251756  0.1934        1
17776 intron_14_102250990_102251756  0.1934        1
17777 intron_14_102251804_102251917  0.1934        1
17778 intron_14_102251995_102263546  0.7714        2
17779 intron_14_102251995_102283478  0.1647        1
17780 intron_14_102263616_102283478 -0.1437        2
17781 intron_14_102265912_102283478  2.8041        1
17790     intron_17_7559702_7560049 -4.2247        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>ctwas  TWAS 
0.000 0.122 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9974 0.9643 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.01613 </code></pre>
<p><img src="figure/BMI_Brain_Substantia_nigra_S.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1              forcats_0.5.1            
 [3] stringr_1.4.0             dplyr_1.0.7              
 [5] purrr_0.3.4               readr_2.1.1              
 [7] tidyr_1.1.4               tidyverse_1.3.1          
 [9] tibble_3.1.6              WebGestaltR_0.4.4        
[11] disgenet2r_0.99.2         enrichR_3.0              
[13] cowplot_1.0.0             ggplot2_3.3.5            
[15] EnsDb.Hsapiens.v79_2.99.0 ensembldb_2.8.0          
[17] AnnotationFilter_1.8.0    GenomicFeatures_1.36.3   
[19] AnnotationDbi_1.46.0      Biobase_2.44.0           
[21] GenomicRanges_1.36.1      GenomeInfoDb_1.20.0      
[23] IRanges_2.18.1            S4Vectors_0.22.1         
[25] BiocGenerics_0.30.0       workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            colorspace_2.0-2           
  [3] rjson_0.2.20                ellipsis_0.3.2             
  [5] rprojroot_2.0.2             XVector_0.24.0             
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                ggrepel_0.9.1              
 [11] bit64_4.0.5                 lubridate_1.8.0            
 [13] fansi_0.5.0                 xml2_1.3.3                 
 [15] codetools_0.2-16            doParallel_1.0.16          
 [17] cachem_1.0.6                knitr_1.36                 
 [19] jsonlite_1.7.2              apcluster_1.4.8            
 [21] Cairo_1.5-12.2              Rsamtools_2.0.0            
 [23] broom_0.7.10                dbplyr_2.1.1               
 [25] compiler_3.6.1              httr_1.4.2                 
 [27] backports_1.4.1             assertthat_0.2.1           
 [29] Matrix_1.2-18               fastmap_1.1.0              
 [31] lazyeval_0.2.2              cli_3.1.0                  
 [33] later_0.8.0                 htmltools_0.5.2            
 [35] prettyunits_1.1.1           tools_3.6.1                
 [37] igraph_1.2.10               gtable_0.3.0               
 [39] glue_1.5.1                  GenomeInfoDbData_1.2.1     
 [41] reshape2_1.4.4              doRNG_1.8.2                
 [43] Rcpp_1.0.7                  cellranger_1.1.0           
 [45] jquerylib_0.1.4             vctrs_0.3.8                
 [47] Biostrings_2.52.0           svglite_1.2.2              
 [49] rtracklayer_1.44.4          iterators_1.0.13           
 [51] xfun_0.29                   rvest_1.0.2                
 [53] lifecycle_1.0.1             rngtools_1.5.2             
 [55] XML_3.99-0.3                zlibbioc_1.30.0            
 [57] scales_1.1.1                vroom_1.5.7                
 [59] hms_1.1.1                   promises_1.0.1             
 [61] ProtGenerics_1.16.0         SummarizedExperiment_1.14.1
 [63] yaml_2.2.1                  curl_4.3.2                 
 [65] memoise_2.0.1               ggrastr_1.0.1              
 [67] gdtools_0.1.9               biomaRt_2.40.1             
 [69] stringi_1.7.6               RSQLite_2.2.8              
 [71] highr_0.9                   foreach_1.5.1              
 [73] BiocParallel_1.18.0         rlang_0.4.12               
 [75] pkgconfig_2.0.3             matrixStats_0.57.0         
 [77] bitops_1.0-7                evaluate_0.14              
 [79] lattice_0.20-38             GenomicAlignments_1.20.1   
 [81] labeling_0.4.2              bit_4.0.4                  
 [83] tidyselect_1.1.1            plyr_1.8.6                 
 [85] magrittr_2.0.1              R6_2.5.1                   
 [87] generics_0.1.1              DelayedArray_0.10.0        
 [89] DBI_1.1.1                   haven_2.4.3                
 [91] pillar_1.6.4                whisker_0.3-2              
 [93] withr_2.4.3                 RCurl_1.98-1.5             
 [95] modelr_0.1.8                crayon_1.4.2               
 [97] utf8_1.2.2                  tzdb_0.2.0                 
 [99] rmarkdown_2.11              progress_1.2.2             
[101] grid_3.6.1                  data.table_1.14.2          
[103] blob_1.2.2                  git2r_0.26.1               
[105] reprex_2.0.1                digest_0.6.29              
[107] httpuv_1.5.1                munsell_0.5.0              
[109] beeswarm_0.2.3              vipor_0.4.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
