<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>BMI - Brain Cortex</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">BMI - Brain Cortex</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1058990dac15746212ef84c84aa7cc13dc9948bftargetblank1058990a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/1058990dac15746212ef84c84aa7cc13dc9948bf" target="_blank">1058990</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/BMI_Brain_Cerebellum_S.Rmd</code>) and HTML (<code>docs/BMI_Brain_Cerebellum_S.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1058990dac15746212ef84c84aa7cc13dc9948bf/analysis/BMI_Brain_Cerebellum_S.Rmd" target="_blank">1058990</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 31854</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
2966 2106 1909 1122 1289 1627 1747 1117 1313 1326 1918 1617  589 1091 1116 1582 
  17   18   19   20   21   22 
2254  398 2296 1033  414 1024 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 28977</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.9097</code></pre>
<pre class="r"><code>#add z scores to results
load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_gene.Rd&quot;))
ctwas_gene_res$z &lt;- z_gene[ctwas_gene_res$intron_id,]$z

z_snp &lt;- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z &lt;- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$intron_id=NA

ctwas_res &lt;- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#get number of eQTL for geens
num_eqtl &lt;- c()
for (i in 1:22){
  load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, i, &quot;.exprqc.Rd&quot;))
  num_eqtl &lt;- c(num_eqtl, unlist(lapply(wgtlist, nrow)))
}
ctwas_gene_res$num_eqtl &lt;- num_eqtl[ctwas_gene_res$intron_id]

#store columns to report
report_cols &lt;- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c(&quot;type&quot;, &quot;region_tag1&quot;, &quot;region_tag2&quot;, &quot;cs_index&quot;, &quot;gene_type&quot;, &quot;z_flag&quot;, &quot;id&quot;, &quot;chrom&quot;, &quot;pos&quot;))]
first_cols &lt;- c(&quot;genename&quot;, &quot;region_tag&quot;)
report_cols &lt;- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps &lt;- c(&quot;id&quot;, report_cols[-1])
report_cols_snps &lt;- report_cols_snps[!(report_cols_snps %in% &quot;num_eqtl&quot;)]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s1.susieIrss.txt&quot;))
n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
rm(ctwas_res_s1)</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0072010 0.0002783 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
18.26 17.28 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 336107</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   31854 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.01246 0.10784 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1]  0.5037 16.7477</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>             genename region_tag susie_pip      mu2       PVE
30707         ANKRD28       3_11    1.0000 37497.55 1.116e-01
30814             DST       6_42    1.0000 12290.08 3.657e-02
31486            LEO1      15_21    0.9999 28041.27 8.342e-02
31385           SPG20      13_13    0.9841 10509.37 3.077e-02
30809         PHACTR1       6_11    0.9620    41.98 1.202e-04
31645 TNFSF12-TNFSF13       17_7    0.9349    28.18 7.839e-05
30840         SLC37A3       7_86    0.9055    24.04 6.476e-05
31207          EFEMP2      11_36    0.9007    52.47 1.406e-04
27934          PPP6R1      19_38    0.7975    23.40 5.551e-05
12641        FAM160B2       8_23    0.7838    28.91 6.743e-05
12715        SLC25A37       8_24    0.7813    25.55 5.938e-05
16323          TRIM66       11_7    0.7776    41.75 9.658e-05
30982           CTBP2      10_78    0.7490    31.62 7.046e-05
29527         BCL2L13       22_2    0.7464    28.04 6.228e-05
7162           ANAPC4       4_21    0.7441   106.81 2.365e-04
3169            BIRC6       2_20    0.7407  1159.91 2.556e-03
18013         SLC38A2      12_29    0.7321 11262.14 2.453e-02
8892            SFXN1      5_105    0.7321   645.50 1.406e-03
18125            CBX5      12_33    0.7316    25.44 5.537e-05
10951          ANKMY2       7_16    0.7292    23.06 5.003e-05
                          intron_id        z num_eqtl
30707    intron_3_15713641_15714571   5.3854        1
30814    intron_6_56468999_56469883   2.9850        1
31486   intron_15_51954575_51958742  -4.6095        1
31385   intron_13_36312477_36313968   2.7757        1
30809    intron_6_13182686_13205815  -6.4898        2
31645     intron_17_7559702_7560049  -4.2247        1
30840  intron_7_140348767_140351273  -4.5689        1
31207   intron_11_65872018_65872244   8.2696        1
27934   intron_19_55247109_55258435  -4.1644        1
12641    intron_8_22098310_22098423  -5.0631        1
12715    intron_8_23529212_23566108   5.0076        1
16323     intron_11_8620572_8621032   6.7162        2
30982 intron_10_125039155_125160319  -5.4234        1
29527   intron_22_17655832_17683214   5.1962        2
7162     intron_4_25407253_25409698 -10.7729        2
3169     intron_2_32575366_32590806  -3.2457        1
18013   intron_12_46367340_46370512   2.9534        1
8892   intron_5_175512196_175513463   0.2299        1
18125   intron_12_54257692_54280008  -4.7317        1
10951    intron_7_16602509_16604721   3.8750        2</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>           genename region_tag susie_pip   mu2       PVE
13637       CCDC171       9_13 0.0000000 42556 0.000e+00
13639       CCDC171       9_13 0.0000000 40838 0.000e+00
13636       CCDC171       9_13 0.0000000 40403 0.000e+00
13638       CCDC171       9_13 0.0000000 38366 0.000e+00
30707       ANKRD28       3_11 0.9999999 37498 1.116e-01
30709       ANKRD28       3_11 0.0000000 36948 0.000e+00
30710       ANKRD28       3_11 0.0000000 36919 0.000e+00
26803        PIK3R2      19_14 0.0000000 35388 0.000e+00
30923         NT5C2      10_66 0.0000000 33060 0.000e+00
30920         NT5C2      10_66 0.0000000 33060 0.000e+00
30903 C10orf32-ASMT      10_66 0.0000000 28606 0.000e+00
30917         NT5C2      10_66 0.0000000 28568 0.000e+00
31486          LEO1      15_21 0.9999230 28041 8.342e-02
31484         TMOD3      15_21 0.0003981 27971 3.313e-05
30783       ANAPC10       4_94 0.0000000 27177 0.000e+00
30784       ANAPC10       4_94 0.0000000 27177 0.000e+00
30579        PIK3R3       1_29 0.7037875 26246 5.496e-02
30580        PIK3R3       1_29 0.7037875 26246 5.496e-02
30581        PIK3R3       1_29 0.1689523 26236 1.319e-02
30575         MAST2       1_29 0.0000000 25968 0.000e+00
                          intron_id      z num_eqtl
13637    intron_9_15883191_15885514  8.979        1
13639    intron_9_15889101_15920270 -7.233        1
13636    intron_9_15874663_15920270 -8.163        1
13638    intron_9_15888088_15888983 -7.315        1
30707    intron_3_15713641_15714571  5.385        1
30709    intron_3_15714656_15720915 -5.253        1
30710    intron_3_15737233_15751750  5.255        1
26803   intron_19_18163388_18166160 -6.025        1
30923 intron_10_103181328_103193236  5.042        1
30920 intron_10_103174982_103181185  5.037        1
30903 intron_10_102870211_102872448 -4.742        1
30917 intron_10_103111805_103139406 -4.364        2
31486   intron_15_51954575_51958742 -4.610        1
31484   intron_15_51934354_51946990  4.604        1
30783  intron_4_144995603_145064572  4.514        1
30784  intron_4_144995603_145081660 -4.514        1
30579    intron_1_46046625_46055795 -4.435        1
30580    intron_1_46046625_46061929 -4.435        1
30581    intron_1_46055971_46061929  4.419        1
30575    intron_1_46010939_46021950  4.309        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>        genename region_tag susie_pip     mu2      PVE
30707    ANKRD28       3_11    1.0000 37497.6 0.111564
31486       LEO1      15_21    0.9999 28041.3 0.083423
30579     PIK3R3       1_29    0.7038 26245.8 0.054957
30580     PIK3R3       1_29    0.7038 26245.8 0.054957
30814        DST       6_42    1.0000 12290.1 0.036566
31385      SPG20      13_13    0.9841 10509.4 0.030771
18013    SLC38A2      12_29    0.7321 11262.1 0.024531
30581     PIK3R3       1_29    0.1690 26236.0 0.013188
26499     ZNF562       19_9    0.5344  6887.6 0.010950
7841     CCDC127        5_1    0.2443  9575.7 0.006961
7630       MFSD8       4_84    0.2688  6310.1 0.005047
7629       MFSD8       4_84    0.2688  6310.1 0.005047
31387      SPG20      13_13    0.1192 10490.6 0.003721
31386      SPG20      13_13    0.1192 10490.6 0.003721
26497 ZNF561-AS1       19_9    0.1757  6885.2 0.003599
26498 ZNF561-AS1       19_9    0.1757  6885.2 0.003599
3169       BIRC6       2_20    0.7407  1159.9 0.002556
7632     C4orf29       4_84    0.1291  6306.1 0.002422
7631     C4orf29       4_84    0.1203  6305.8 0.002257
8892       SFXN1      5_105    0.7321   645.5 0.001406
                         intron_id       z num_eqtl
30707   intron_3_15713641_15714571  5.3854        1
31486  intron_15_51954575_51958742 -4.6095        1
30579   intron_1_46046625_46055795 -4.4353        1
30580   intron_1_46046625_46061929 -4.4353        1
30814   intron_6_56468999_56469883  2.9850        1
31385  intron_13_36312477_36313968  2.7757        1
18013  intron_12_46367340_46370512  2.9534        1
30581   intron_1_46055971_46061929  4.4189        1
26499    intron_19_9656653_9658009 -2.6445        1
7841        intron_5_216859_218093  2.9002        1
7630  intron_4_127957592_127965897  2.5077        1
7629  intron_4_127957592_127965072 -2.5077        1
31387  intron_13_36335832_36346582  2.8188        1
31386  intron_13_36335832_36346240 -2.8188        1
26497    intron_19_9632417_9633460  2.6214        1
26498    intron_19_9632417_9645074 -2.6214        1
3169    intron_2_32575366_32590806 -3.2457        1
7632  intron_4_127984423_128008920  2.5104        1
7631  intron_4_127984423_127989721  2.5099        1
8892  intron_5_175512196_175513463  0.2299        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag susie_pip      mu2       PVE
5612         RBM6       3_35 1.240e-03   883.21 3.258e-06
5616         RBM6       3_35 1.201e-03   883.67 3.157e-06
5615         RBM6       3_35 1.201e-03   883.67 3.157e-06
31008     BDNF-AS      11_19 0.000e+00 10361.36 0.000e+00
9699     C6orf106       6_28 3.448e-01   174.17 1.787e-04
9700     C6orf106       6_28 1.741e-01   165.06 8.551e-05
5605        MST1R       3_35 1.761e-11   180.79 9.471e-15
3028  DNAJC27-AS1       2_15 4.842e-05   186.70 2.690e-08
8202         POC5       5_44 6.189e-01    92.55 1.704e-04
3026  DNAJC27-AS1       2_15 5.217e-05   182.88 2.839e-08
22656      KCTD13      16_24 1.666e-02   101.63 5.037e-06
9703     UHRF1BP1       6_28 3.333e-02   154.47 1.532e-05
5596       RNF123       3_35 1.462e-11   800.39 3.482e-14
5597       RNF123       3_35 1.462e-11   800.39 3.482e-14
5598       RNF123       3_35 1.214e-11   799.40 2.886e-14
5599       RNF123       3_35 1.214e-11   799.40 2.886e-14
9701     C6orf106       6_28 3.543e-03   139.75 1.473e-06
22631       SH2B1      16_23 1.279e-01    62.66 2.384e-05
22673       DOC2A      16_24 2.674e-02    92.65 7.370e-06
7162       ANAPC4       4_21 7.441e-01   106.81 2.365e-04
                        intron_id      z num_eqtl
5612   intron_3_49999513_50048245 -12.54        1
5616   intron_3_50059746_50060956  12.54        1
5615   intron_3_50059746_50060801 -12.54        1
31008 intron_11_27640005_27659171  11.99        1
9699   intron_6_34654779_34688841  11.54        2
9700   intron_6_34654779_34696446 -11.49        2
5605   intron_3_49887562_49889924  11.31        2
3028   intron_2_24972232_25001415  11.29        1
8202   intron_5_75690562_75692396  11.29        1
3026   intron_2_24972232_24972626 -11.13        2
22656 intron_16_29923359_29925539  11.12        2
9703   intron_6_34836348_34855617  11.06        1
5596   intron_3_49712656_49713513  10.96        1
5597   intron_3_49712892_49713513 -10.96        1
5598   intron_3_49714174_49715492  10.95        1
5599   intron_3_49714174_49715575 -10.95        1
9701   intron_6_34688933_34696446  10.90        1
22631 intron_16_28869383_28870709 -10.81        2
22673 intron_16_30007090_30007173 -10.79        2
7162   intron_4_25407253_25409698 -10.77        2</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.02088</code></pre>
<pre><code>         genename region_tag susie_pip      mu2       PVE
5612         RBM6       3_35 1.240e-03   883.21 3.258e-06
5616         RBM6       3_35 1.201e-03   883.67 3.157e-06
5615         RBM6       3_35 1.201e-03   883.67 3.157e-06
31008     BDNF-AS      11_19 0.000e+00 10361.36 0.000e+00
9699     C6orf106       6_28 3.448e-01   174.17 1.787e-04
9700     C6orf106       6_28 1.741e-01   165.06 8.551e-05
5605        MST1R       3_35 1.761e-11   180.79 9.471e-15
3028  DNAJC27-AS1       2_15 4.842e-05   186.70 2.690e-08
8202         POC5       5_44 6.189e-01    92.55 1.704e-04
3026  DNAJC27-AS1       2_15 5.217e-05   182.88 2.839e-08
22656      KCTD13      16_24 1.666e-02   101.63 5.037e-06
9703     UHRF1BP1       6_28 3.333e-02   154.47 1.532e-05
5596       RNF123       3_35 1.462e-11   800.39 3.482e-14
5597       RNF123       3_35 1.462e-11   800.39 3.482e-14
5598       RNF123       3_35 1.214e-11   799.40 2.886e-14
5599       RNF123       3_35 1.214e-11   799.40 2.886e-14
9701     C6orf106       6_28 3.543e-03   139.75 1.473e-06
22631       SH2B1      16_23 1.279e-01    62.66 2.384e-05
22673       DOC2A      16_24 2.674e-02    92.65 7.370e-06
7162       ANAPC4       4_21 7.441e-01   106.81 2.365e-04
                        intron_id      z num_eqtl
5612   intron_3_49999513_50048245 -12.54        1
5616   intron_3_50059746_50060956  12.54        1
5615   intron_3_50059746_50060801 -12.54        1
31008 intron_11_27640005_27659171  11.99        1
9699   intron_6_34654779_34688841  11.54        2
9700   intron_6_34654779_34696446 -11.49        2
5605   intron_3_49887562_49889924  11.31        2
3028   intron_2_24972232_25001415  11.29        1
8202   intron_5_75690562_75692396  11.29        1
3026   intron_2_24972232_24972626 -11.13        2
22656 intron_16_29923359_29925539  11.12        2
9703   intron_6_34836348_34855617  11.06        1
5596   intron_3_49712656_49713513  10.96        1
5597   intron_3_49712892_49713513 -10.96        1
5598   intron_3_49714174_49715492  10.95        1
5599   intron_3_49714174_49715575 -10.95        1
9701   intron_6_34688933_34696446  10.90        1
22631 intron_16_28869383_28870709 -10.81        2
22673 intron_16_30007090_30007173 -10.79        2
7162   intron_4_25407253_25409698 -10.77        2</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 71</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                                    Term
1                  protein import into mitochondrial matrix (GO:0030150)
2 protein transmembrane import into intracellular organelle (GO:0044743)
3                                          serine transport (GO:0032329)
  Overlap Adjusted.P.value                 Genes
1    3/19          0.01920 TOMM40;TIMM17A;TIMM23
2    3/32          0.04321 TOMM40;TIMM17A;TIMM23
3     2/7          0.04321         SFXN1;SLC38A2
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                                        Term
1                  integral component of mitochondrial membrane (GO:0032592)
2           intrinsic component of mitochondrial inner membrane (GO:0031304)
3            integral component of mitochondrial inner membrane (GO:0031305)
4 TIM23 mitochondrial import inner membrane translocase complex (GO:0005744)
  Overlap Adjusted.P.value                       Genes
1    4/66         0.006913 TOMM40;TIMM17A;TIMM23;SFXN1
2    3/39         0.014477        TIMM17A;TIMM23;SFXN1
3    3/51         0.021360        TIMM17A;TIMM23;SFXN1
4    2/14         0.023219              TIMM17A;TIMM23
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-16-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                      Term Overlap
1 L-serine transmembrane transporter activity (GO:0015194)     2/6
2   serine transmembrane transporter activity (GO:0022889)     2/8
  Adjusted.P.value         Genes
1          0.01967 SFXN1;SLC38A2
2          0.01967 SFXN1;SLC38A2</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Warning in disease_enrichment(entities = genes, vocabulary = &quot;HGNC&quot;, database =
&quot;CURATED&quot;): Removing duplicates from input list.</code></pre>
<pre><code>                                                  Description     FDR Ratio
8                                            Anemia, Neonatal 0.04104  1/35
93                                 Sulfite oxidase deficiency 0.04104  1/35
129                                         Lewy Body Disease 0.04104  2/35
146 Familial encephalopathy with neuroserpin inclusion bodies 0.04104  1/35
148                                          ELLIPTOCYTOSIS 3 0.04104  1/35
156                              SPHEROCYTOSIS, HEREDITARY, 2 0.04104  1/35
157                                   RETINITIS PIGMENTOSA 42 0.04104  1/35
162                                          Sulfocysteinuria 0.04104  1/35
166                  CUTIS LAXA, AUTOSOMAL RECESSIVE, TYPE IB 0.04104  1/35
168     NEUROPATHY, HEREDITARY SENSORY AND AUTONOMIC, TYPE VI 0.04104  1/35
    BgRatio
8    1/9703
93   1/9703
129 22/9703
146  1/9703
148  1/9703
156  1/9703
157  1/9703
162  1/9703
166  1/9703
168  1/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>Warning: ggrepel: 33 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 41</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.802</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 665</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>             genename region_tag susie_pip       mu2       PVE
30810             DST       6_42 1.725e-04 12264.281 6.296e-06
30811             DST       6_42 0.000e+00  6635.362 0.000e+00
30812             DST       6_42 0.000e+00  6726.554 0.000e+00
30813             DST       6_42 7.097e-05 12263.018 2.589e-06
30814             DST       6_42 1.000e+00 12290.078 3.657e-02
30815             DST       6_42 0.000e+00   393.267 0.000e+00
30816             DST       6_42 0.000e+00   133.519 0.000e+00
30838         SLC37A3       7_86 2.413e-02    14.326 1.029e-06
30839         SLC37A3       7_86 8.301e-03     6.512 1.608e-07
30840         SLC37A3       7_86 9.055e-01    24.039 6.476e-05
30841         SLC37A3       7_86 8.734e-03     7.509 1.951e-07
30842         SLC37A3       7_86 9.838e-03     8.912 2.609e-07
30843         SLC37A3       7_86 8.244e-03     6.456 1.583e-07
30844         SLC37A3       7_86 1.153e-02     9.935 3.408e-07
30845         SLC37A3       7_86 1.029e-02     9.892 3.029e-07
30846         SLC37A3       7_86 1.029e-02     9.892 3.029e-07
30847         SLC37A3       7_86 1.211e-02     9.931 3.579e-07
31385           SPG20      13_13 9.841e-01 10509.371 3.077e-02
31386           SPG20      13_13 1.192e-01 10490.574 3.721e-03
31387           SPG20      13_13 1.192e-01 10490.574 3.721e-03
31486            LEO1      15_21 9.999e-01 28041.266 8.342e-02
31642 TNFSF12-TNFSF13       17_7 6.791e-03     9.367 1.893e-07
31643 TNFSF12-TNFSF13       17_7 6.791e-03     9.367 1.893e-07
31644 TNFSF12-TNFSF13       17_7 6.791e-03     9.367 1.893e-07
31645 TNFSF12-TNFSF13       17_7 9.349e-01    28.181 7.839e-05
                         intron_id       z num_eqtl
30810   intron_6_56463156_56463565  2.9051        1
30811   intron_6_56463764_56464685  2.1176        2
30812   intron_6_56463764_56466078 -2.1324        2
30813   intron_6_56464756_56466078  2.8924        1
30814   intron_6_56468999_56469883  2.9850        1
30815   intron_6_56598006_56598476  3.7368        1
30816   intron_6_56851604_56900421  1.5511        1
30838 intron_7_140337349_140348626  0.9305        1
30839 intron_7_140345263_140345869 -1.0554        1
30840 intron_7_140348767_140351273 -4.5689        1
30841 intron_7_140351451_140352062 -1.4265        3
30842 intron_7_140351969_140352062  1.6409        3
30843 intron_7_140352181_140355668  1.0749        2
30844 intron_7_140355764_140364408  1.4247        1
30845 intron_7_140369682_140379214  1.7377        1
30846 intron_7_140369682_140380282  1.7377        1
30847 intron_7_140380390_140382438 -1.2641        1
31385  intron_13_36312477_36313968  2.7757        1
31386  intron_13_36335832_36346240 -2.8188        1
31387  intron_13_36335832_36346582  2.8188        1
31486  intron_15_51954575_51958742 -4.6095        1
31642    intron_17_7559297_7559846  1.3318        1
31643    intron_17_7559297_7559851  1.3318        1
31644    intron_17_7559702_7559846 -1.3318        1
31645    intron_17_7559702_7560049 -4.2247        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code> ctwas   TWAS 
0.0000 0.2683 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9988 0.9532 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.01654 </code></pre>
<p><img src="figure/BMI_Brain_Cerebellum_S.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1              forcats_0.5.1            
 [3] stringr_1.4.0             dplyr_1.0.7              
 [5] purrr_0.3.4               readr_2.1.1              
 [7] tidyr_1.1.4               tidyverse_1.3.1          
 [9] tibble_3.1.6              WebGestaltR_0.4.4        
[11] disgenet2r_0.99.2         enrichR_3.0              
[13] cowplot_1.0.0             ggplot2_3.3.5            
[15] EnsDb.Hsapiens.v79_2.99.0 ensembldb_2.8.0          
[17] AnnotationFilter_1.8.0    GenomicFeatures_1.36.3   
[19] AnnotationDbi_1.46.0      Biobase_2.44.0           
[21] GenomicRanges_1.36.1      GenomeInfoDb_1.20.0      
[23] IRanges_2.18.1            S4Vectors_0.22.1         
[25] BiocGenerics_0.30.0       workflowr_1.6.2          

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0            colorspace_2.0-2           
  [3] rjson_0.2.20                ellipsis_0.3.2             
  [5] rprojroot_2.0.2             XVector_0.24.0             
  [7] fs_1.5.2                    rstudioapi_0.13            
  [9] farver_2.1.0                ggrepel_0.9.1              
 [11] bit64_4.0.5                 lubridate_1.8.0            
 [13] fansi_0.5.0                 xml2_1.3.3                 
 [15] codetools_0.2-16            doParallel_1.0.16          
 [17] cachem_1.0.6                knitr_1.36                 
 [19] jsonlite_1.7.2              apcluster_1.4.8            
 [21] Cairo_1.5-12.2              Rsamtools_2.0.0            
 [23] broom_0.7.10                dbplyr_2.1.1               
 [25] compiler_3.6.1              httr_1.4.2                 
 [27] backports_1.4.1             assertthat_0.2.1           
 [29] Matrix_1.2-18               fastmap_1.1.0              
 [31] lazyeval_0.2.2              cli_3.1.0                  
 [33] later_0.8.0                 htmltools_0.5.2            
 [35] prettyunits_1.1.1           tools_3.6.1                
 [37] igraph_1.2.10               gtable_0.3.0               
 [39] glue_1.5.1                  GenomeInfoDbData_1.2.1     
 [41] reshape2_1.4.4              doRNG_1.8.2                
 [43] Rcpp_1.0.7                  cellranger_1.1.0           
 [45] jquerylib_0.1.4             vctrs_0.3.8                
 [47] Biostrings_2.52.0           svglite_1.2.2              
 [49] rtracklayer_1.44.4          iterators_1.0.13           
 [51] xfun_0.29                   rvest_1.0.2                
 [53] lifecycle_1.0.1             rngtools_1.5.2             
 [55] XML_3.99-0.3                zlibbioc_1.30.0            
 [57] scales_1.1.1                vroom_1.5.7                
 [59] hms_1.1.1                   promises_1.0.1             
 [61] ProtGenerics_1.16.0         SummarizedExperiment_1.14.1
 [63] yaml_2.2.1                  curl_4.3.2                 
 [65] memoise_2.0.1               ggrastr_1.0.1              
 [67] gdtools_0.1.9               biomaRt_2.40.1             
 [69] stringi_1.7.6               RSQLite_2.2.8              
 [71] highr_0.9                   foreach_1.5.1              
 [73] BiocParallel_1.18.0         rlang_0.4.12               
 [75] pkgconfig_2.0.3             matrixStats_0.57.0         
 [77] bitops_1.0-7                evaluate_0.14              
 [79] lattice_0.20-38             GenomicAlignments_1.20.1   
 [81] labeling_0.4.2              bit_4.0.4                  
 [83] tidyselect_1.1.1            plyr_1.8.6                 
 [85] magrittr_2.0.1              R6_2.5.1                   
 [87] generics_0.1.1              DelayedArray_0.10.0        
 [89] DBI_1.1.1                   haven_2.4.3                
 [91] pillar_1.6.4                whisker_0.3-2              
 [93] withr_2.4.3                 RCurl_1.98-1.5             
 [95] modelr_0.1.8                crayon_1.4.2               
 [97] utf8_1.2.2                  tzdb_0.2.0                 
 [99] rmarkdown_2.11              progress_1.2.2             
[101] grid_3.6.1                  data.table_1.14.2          
[103] blob_1.2.2                  git2r_0.26.1               
[105] reprex_2.0.1                digest_0.6.29              
[107] httpuv_1.5.1                munsell_0.5.0              
[109] beeswarm_0.2.3              vipor_0.4.5                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
