<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />

<meta name="date" content="2022-10-18" />

<title>LDL - Liver</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">LDL - Liver</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2022-10-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config_b38.R
</td>
<td style="text-align:left;">
code/ctwas_config_b38.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree78f86642877269d0798ae5e6a4d071adefe74d8ftargetblank78f8664a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/78f86642877269d0798ae5e6a4d071adefe74d8f" target="_blank">78f8664</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree78f86642877269d0798ae5e6a4d071adefe74d8ftargetblank78f8664a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/78f86642877269d0798ae5e6a4d071adefe74d8f" target="_blank">78f8664</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Proposal plots.R
    Untracked:  RGS14.pdf
    Untracked:  RNF186.pdf
    Untracked:  SCZ_annotation.xlsx
    Untracked:  SLC8B1.pdf
    Untracked:  UKB_analysis_allweights_scz/
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  cache/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/IBD_ME_3kb_0.05_out/
    Untracked:  code/LDL_S_out/LDL_Liver.err
    Untracked:  code/LDL_S_out/LDL_Liver.out
    Untracked:  code/LDL_out/
    Untracked:  code/run_IBD_analysis_ME_3kb_0.05.sbatch
    Untracked:  code/run_IBD_analysis_ME_3kb_0.05.sh
    Untracked:  code/run_IBD_ctwas_rss_LDR_ME_3kb_0.05.R
    Untracked:  code/run_LDL_analysis.sbatch
    Untracked:  code/run_LDL_analysis.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/FUMA_output/
    Untracked:  data/GO_Terms/
    Untracked:  data/IBD_ME/
    Untracked:  data/LDL/
    Untracked:  data/LDL_S/
    Untracked:  data/PGC3_SCZ_wave3_public.v2.tsv
    Untracked:  data/SCZ/
    Untracked:  data/SCZ_2014_EUR/
    Untracked:  data/SCZ_2014_EUR_ME/
    Untracked:  data/SCZ_2018/
    Untracked:  data/SCZ_2018_ME/
    Untracked:  data/SCZ_2018_S/
    Untracked:  data/SCZ_2020/
    Untracked:  data/SCZ_S/
    Untracked:  data/Supplementary Table 15 - MAGMA.xlsx
    Untracked:  data/Supplementary Table 20 - Prioritised Genes.xlsx
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/gwas_sumstats/
    Untracked:  data/magma.genes.out
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/notes.txt
    Untracked:  data/scz_2018.RDS
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  temp_LDR/
    Untracked:  top_genes_32.txt
    Untracked:  top_genes_37.txt
    Untracked:  top_genes_43.txt
    Untracked:  top_genes_54.txt
    Untracked:  top_genes_81.txt
    Untracked:  z_snp_pos_SCZ.RData
    Untracked:  z_snp_pos_SCZ_2014_EUR.RData
    Untracked:  z_snp_pos_SCZ_2018.RData
    Untracked:  z_snp_pos_SCZ_2020.RData

Unstaged changes:
    Deleted:    analysis/BMI_S_results.Rmd
    Modified:   analysis/LDL_Liver.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Deleted:    code/LDL_S_out/T2D_Liver.err
    Deleted:    code/LDL_S_out/T2D_Liver.out
    Modified:   code/SCZ_out/SCZ_Brain_Amygdala.err
    Modified:   code/SCZ_out/SCZ_Brain_Amygdala.out
    Modified:   code/SCZ_out/SCZ_Brain_Anterior_cingulate_cortex_BA24.err
    Modified:   code/SCZ_out/SCZ_Brain_Anterior_cingulate_cortex_BA24.out
    Modified:   code/SCZ_out/SCZ_Brain_Caudate_basal_ganglia.err
    Modified:   code/SCZ_out/SCZ_Brain_Caudate_basal_ganglia.out
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellar_Hemisphere.err
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellar_Hemisphere.out
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellum.err
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellum.out
    Modified:   code/SCZ_out/SCZ_Brain_Cortex.err
    Modified:   code/SCZ_out/SCZ_Brain_Cortex.out
    Modified:   code/SCZ_out/SCZ_Brain_Frontal_Cortex_BA9.err
    Modified:   code/SCZ_out/SCZ_Brain_Frontal_Cortex_BA9.out
    Modified:   code/SCZ_out/SCZ_Brain_Hippocampus.err
    Modified:   code/SCZ_out/SCZ_Brain_Hippocampus.out
    Modified:   code/SCZ_out/SCZ_Brain_Hypothalamus.err
    Modified:   code/SCZ_out/SCZ_Brain_Hypothalamus.out
    Modified:   code/SCZ_out/SCZ_Brain_Nucleus_accumbens_basal_ganglia.err
    Modified:   code/SCZ_out/SCZ_Brain_Nucleus_accumbens_basal_ganglia.out
    Modified:   code/SCZ_out/SCZ_Brain_Putamen_basal_ganglia.err
    Modified:   code/SCZ_out/SCZ_Brain_Putamen_basal_ganglia.out
    Modified:   code/SCZ_out/SCZ_Brain_Spinal_cord_cervical_c-1.err
    Modified:   code/SCZ_out/SCZ_Brain_Spinal_cord_cervical_c-1.out
    Modified:   code/SCZ_out/SCZ_Brain_Substantia_nigra.err
    Modified:   code/SCZ_out/SCZ_Brain_Substantia_nigra.out
    Deleted:    code/run_IBD_ctwas_rss_LDR_ME.R
    Modified:   code/run_LDL_analysis_S.sbatch
    Modified:   code/run_LDL_analysis_S.sh
    Modified:   code/run_LDL_ctwas_rss_LDR_S.R
    Modified:   code/run_SCZ_analysis.sbatch
    Modified:   code/run_SCZ_analysis.sh
    Modified:   code/run_SCZ_ctwas_rss_LDR.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/LDL_Liver_S.Rmd</code>) and HTML (<code>docs/LDL_Liver_S.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a4ec7aae4149e6604627a21b3c7d4b6e7abed08/analysis/LDL_Liver_S.Rmd" target="_blank">6a4ec7a</a>
</td>
<td>
sq-96
</td>
<td>
2022-10-18
</td>
<td>
updtae
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/6a4ec7aae4149e6604627a21b3c7d4b6e7abed08/docs/LDL_Liver_S.html" target="_blank">6a4ec7a</a>
</td>
<td>
sq-96
</td>
<td>
2022-10-18
</td>
<td>
updtae
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/601071b4ceffb4733ad1e5af771d2ef502e28b04/analysis/LDL_Liver_S.Rmd" target="_blank">601071b</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-01
</td>
<td>
adadwd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/143efdd7801ff4c6396ff7b3e862ac02f88fffc8/docs/LDL_Liver_S.html" target="_blank">143efdd</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/fe3e6a49a7686c1a64077e8fc33be1e1f9afad9f/analysis/LDL_Liver_S.Rmd" target="_blank">fe3e6a4</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 21553</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
2079 1523 1276  769  878 1116 1176  701  854  931 1344 1116  391  772  677 1085 
  17   18   19   20   21   22 
1462  258 1524  659  262  700 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 19937</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.925</code></pre>
<pre class="r"><code>library(reticulate)
use_python(&quot;/scratch/midway2/shengqian/miniconda3/envs/PythonForR/bin/python&quot;,required=T)</code></pre>
<pre><code>finish</code></pre>
<pre class="r"><code>#add z scores to results
load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_gene.Rd&quot;))
ctwas_gene_res &lt;- py$ctwas_gene_res_df 
ctwas_gene_res$z &lt;- z_gene[ctwas_gene_res$id,]$z

z_snp &lt;- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z &lt;- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
#ctwas_gene_res$genename=NA
#ctwas_gene_res$gene_type=NA
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_gene_res = ctwas_gene_res[,!(colnames(ctwas_gene_res) %in% c(&quot;intron_start&quot;, &quot;intron_end&quot;,&quot;intron_id&quot;))]


ctwas_res &lt;- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#get number of sQTL for geens
num_sqtl &lt;- c()
for (i in 1:22){
  load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_chr&quot;, i, &quot;.exprqc.Rd&quot;))
  num_sqtl &lt;- c(num_sqtl, unlist(lapply(wgtlist, nrow)))
}
ctwas_gene_res$num_sqtl &lt;- num_sqtl[ctwas_gene_res$id]

#store columns to report
report_cols &lt;- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c(&quot;type&quot;, &quot;region_tag1&quot;, &quot;region_tag2&quot;, &quot;cs_index&quot;, &quot;gene_type&quot;, &quot;z_flag&quot;))]
first_cols &lt;- c(&quot;genename&quot;, &quot;region_tag&quot;)
report_cols &lt;- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps &lt;- c(&quot;id&quot;, report_cols[-1])
report_cols_snps &lt;- report_cols_snps[!(report_cols_snps %in% &quot;num_sqtl&quot;)]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s1.susieIrss.txt&quot;))
n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
rm(ctwas_res_s1)</code></pre>
<pre class="r"><code>library(ggplot2)
library(cowplot)

load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s2.susieIrssres.Rd&quot;))

group_size &lt;- c(nrow(ctwas_gene_res), n_snps)

#estimated group prior (all iterations)
estimated_group_prior_all &lt;- group_prior_rec
rownames(estimated_group_prior_all) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior_all[&quot;snp&quot;,] &lt;- estimated_group_prior_all[&quot;snp&quot;,]*thin #adjust parameter to account for thin argument

#estimated group prior variance (all iterations)
estimated_group_prior_var_all &lt;- group_prior_var_rec
rownames(estimated_group_prior_var_all) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)

#estimated group PVE (all iterations)
estimated_group_pve_all &lt;- estimated_group_prior_var_all*estimated_group_prior_all*group_size/sample_size #check PVE calculation
rownames(estimated_group_pve_all) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)

#estimated enrichment of genes (all iterations)
estimated_enrichment_all &lt;- estimated_group_prior_all[&quot;gene&quot;,]/estimated_group_prior_all[&quot;snp&quot;,]

title_size &lt;- 12

df &lt;- data.frame(niter = rep(1:ncol(estimated_group_prior_all), 2),
                 value = c(estimated_group_prior_all[&quot;gene&quot;,], estimated_group_prior_all[&quot;snp&quot;,]),
                 group = rep(c(&quot;Gene&quot;, &quot;SNP&quot;), each = ncol(estimated_group_prior_all)))
df$group &lt;- as.factor(df$group)

p_pi &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(pi)) +
  ggtitle(&quot;Proportion Causal&quot;) +
  theme_cowplot()

p_pi &lt;- p_pi + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = &quot;Group&quot;)) + theme (legend.title = element_text(size=12, face=&quot;bold&quot;))

df &lt;- data.frame(niter = rep(1:ncol(estimated_group_prior_var_all ), 2),
                 value = c(estimated_group_prior_var_all[&quot;gene&quot;,], estimated_group_prior_var_all[&quot;snp&quot;,]),
                 group = rep(c(&quot;Gene&quot;, &quot;SNP&quot;), each = ncol(estimated_group_prior_var_all)))
df$group &lt;- as.factor(df$group)
p_sigma2 &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(sigma^2)) +
  ggtitle(&quot;Effect Size&quot;) +
  theme_cowplot()

p_sigma2 &lt;- p_sigma2 + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = &quot;Group&quot;)) + theme (legend.title = element_text(size=12, face=&quot;bold&quot;))

df &lt;- data.frame(niter = rep(1:ncol(estimated_group_pve_all ), 2),
                 value = c(estimated_group_pve_all[&quot;gene&quot;,], estimated_group_pve_all[&quot;snp&quot;,]),
                 group = rep(c(&quot;Gene&quot;, &quot;SNP&quot;), each = ncol(estimated_group_pve_all)))
df$group &lt;- as.factor(df$group)
p_pve &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(h^2[G])) +
  ggtitle(&quot;PVE&quot;) +
  theme_cowplot()

p_pve &lt;- p_pve + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = &quot;Group&quot;)) + theme (legend.title = element_text(size=12, face=&quot;bold&quot;))

df &lt;- data.frame(niter = 1:length(estimated_enrichment_all),
                 value = estimated_enrichment_all,
                 group = rep(&quot;Gene&quot;, length(estimated_enrichment_all)))
df$group &lt;- as.factor(df$group)
p_enrich &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(pi[G]/pi[S])) +
  ggtitle(&quot;Enrichment&quot;) +
  theme_cowplot()

p_enrich &lt;- p_enrich + theme(plot.title=element_text(size=title_size)) + 
  expand_limits(y=0) + 
  guides(color = guide_legend(title = &quot;Group&quot;)) + theme (legend.title = element_text(size=12, face=&quot;bold&quot;))

plot_grid(p_pi, p_sigma2, p_enrich, p_pve)</code></pre>
<p><img src="figure/LDL_Liver_S.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a4ec7aae4149e6604627a21b3c7d4b6e7abed08/docs/figure/LDL_Liver_S.Rmd/unnamed-chunk-9-1.png" target="_blank">6a4ec7a</a>
</td>
<td>
sq-96
</td>
<td>
2022-10-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/143efdd7801ff4c6396ff7b3e862ac02f88fffc8/docs/figure/LDL_Liver_S.Rmd/unnamed-chunk-9-1.png" target="_blank">143efdd</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- estimated_group_prior_all[,ncol(group_prior_rec)]
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
1.289e-02 6.246e-05 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- estimated_group_prior_var_all[,ncol(group_prior_var_rec)]
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
13.29 35.66 </code></pre>
<pre class="r"><code>#estimated enrichment
estimated_enrichment &lt;- estimated_enrichment_all[ncol(group_prior_var_rec)]
print(estimated_enrichment)</code></pre>
<pre><code>[1] 206.3</code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 343621</code></pre>
<pre class="r"><code>#report group size
print(group_size)</code></pre>
<pre><code>[1]   21553 8696600</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_pve_all[,ncol(group_prior_rec)] #check PVE calculation
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.01074 0.05638 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
#c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))

#total PVE
sum(estimated_group_pve)</code></pre>
<pre><code>[1] 0.06712</code></pre>
<pre class="r"><code>#PVE attributable to gene expression
estimated_group_pve[&quot;gene&quot;]/sum(estimated_group_pve)</code></pre>
<pre><code>gene 
0.16 </code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<pre class="r"><code>#distribution of PIPs
hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main=&quot;Distribution of Gene PIPs&quot;)</code></pre>
<p><img src="figure/LDL_Liver_S.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a4ec7aae4149e6604627a21b3c7d4b6e7abed08/docs/figure/LDL_Liver_S.Rmd/unnamed-chunk-11-1.png" target="_blank">6a4ec7a</a>
</td>
<td>
sq-96
</td>
<td>
2022-10-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#genes with PIP&gt;0.8 or 20 highest PIPs
head(ctwas_gene_res[order(-ctwas_gene_res$susie_pip),report_cols], max(sum(ctwas_gene_res$susie_pip&gt;0.8), 20))</code></pre>
<pre><code>      genename region_tag chrom                            id       pos
21159     LDLR       19_9    19   intron_19_11120522_11123174  11120205
20916       HP      16_38    16   intron_16_72057466_72058254  72072066
20403    HLA-B       6_25     6    intron_6_31270085_31355317  31272369
20643    LRCH4       7_61     7  intron_7_100575304_100575705 100574936
20876    ITGAL      16_24    16   intron_16_30505462_30506715  30506720
21422    BCAT2      19_34    19   intron_19_48800297_48810984  48800483
20697   KIF13B       8_28     8    intron_8_29092878_29099133  29092792
20738  ST3GAL4      11_77    11 intron_11_126356302_126406096 126355981
20310    USP53       4_77     4  intron_4_119217673_119220116 119220165
21414    SPHK2      19_33    19   intron_19_48624430_48625891  48624233
20748     PHC1       12_9    12     intron_12_8921750_8922633   8922098
20245   SPRED2       2_42     2    intron_2_65314169_65316734  65314461
21287  CYP4F12      19_13    19   intron_19_15685197_15696430  15685032
20986    ABCA8      17_39    17   intron_17_68883882_68884331  68883786
20267   UGT1A1      2_137     2  intron_2_233761151_233767034 233761240
20776   R3HDM2      12_36    12   intron_12_57395818_57430720  57395576
410      KDM1A       1_16     1    intron_1_23030634_23050387  23030789
6609   ZSCAN31       6_22     6    intron_6_28327533_28329303  28327111
18354   ZNF227      19_30    19   intron_19_44213244_44217791  44213516
20954    FLOT2      17_17    17   intron_17_28889026_28897526  28889154
      susie_pip    mu2       PVE       z num_sqtl
21159    1.0000 751.97 2.188e-03  26.898        2
20916    0.9933 272.19 7.868e-04  21.869        2
20403    0.9914  62.55 1.805e-04   8.790        4
20643    0.9895  31.67 9.119e-05   5.294        2
20876    0.9618  21.82 6.107e-05  -4.428        1
21422    0.9583  27.94 7.792e-05  -5.197        1
20697    0.9476  23.56 6.496e-05  -4.718        1
20738    0.9427  76.58 2.101e-04 -12.154        2
20310    0.9392  26.01 7.110e-05   4.888        1
21414    0.9138  42.12 1.120e-04   8.721        1
20748    0.9054  36.98 9.744e-05   6.156        1
20245    0.8639  22.11 5.558e-05  -4.438        1
21287    0.8606  36.25 9.079e-05  -5.868        2
20986    0.8284  27.86 6.717e-05  -4.775        2
20267    0.8104  30.83 7.271e-05   5.450        1
20776    0.8029  36.25 8.469e-05  -5.818        1
410      0.7924  20.69 4.772e-05   4.158        1
6609     0.7913  24.62 5.670e-05   2.711        1
18354    0.7903  30.19 6.943e-05   4.698        1
20954    0.7864  26.93 6.162e-05  -3.738        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<pre class="r"><code>#set nominal signifiance threshold for z scores
alpha &lt;- 0.05

#bonferroni adjusted threshold for z scores
sig_thresh &lt;- qnorm(1-(alpha/nrow(ctwas_gene_res)/2), lower=T)</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre class="r"><code>a &lt;- locus_plot_final_pub(region_tag=&quot;19_9&quot;, return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel=&quot;both&quot;,
                      legend_side=&quot;left&quot;,
                      legend_panel=&quot;cTWAS&quot;)</code></pre>
<p><img src="figure/LDL_Liver_S.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a4ec7aae4149e6604627a21b3c7d4b6e7abed08/docs/figure/LDL_Liver_S.Rmd/unnamed-chunk-16-1.png" target="_blank">6a4ec7a</a>
</td>
<td>
sq-96
</td>
<td>
2022-10-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/143efdd7801ff4c6396ff7b3e862ac02f88fffc8/docs/figure/LDL_Liver_S.Rmd/unnamed-chunk-16-1.png" target="_blank">143efdd</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Gviz_1.38.4          GenomicRanges_1.46.0 GenomeInfoDb_1.26.7 
 [4] IRanges_2.24.1       S4Vectors_0.28.1     BiocGenerics_0.40.0 
 [7] cowplot_1.1.1        ggplot2_3.3.6        reticulate_1.26     
[10] workflowr_1.7.0     

loaded via a namespace (and not attached):
  [1] backports_1.2.1             Hmisc_4.5-0                
  [3] BiocFileCache_2.2.0         lazyeval_0.2.2             
  [5] splines_4.1.0               BiocParallel_1.28.0        
  [7] digest_0.6.29               ensembldb_2.18.4           
  [9] htmltools_0.5.3             fansi_1.0.3                
 [11] magrittr_2.0.3              checkmate_2.0.0            
 [13] memoise_2.0.1               BSgenome_1.62.0            
 [15] cluster_2.1.2               Biostrings_2.62.0          
 [17] matrixStats_0.62.0          prettyunits_1.1.1          
 [19] jpeg_0.1-8.1                colorspace_2.0-3           
 [21] blob_1.2.3                  rappdirs_0.3.3             
 [23] xfun_0.24                   dplyr_1.0.9                
 [25] callr_3.7.0                 crayon_1.5.1               
 [27] RCurl_1.98-1.6              jsonlite_1.8.0             
 [29] survival_3.2-11             VariantAnnotation_1.40.0   
 [31] glue_1.6.2                  gtable_0.3.0               
 [33] zlibbioc_1.40.0             XVector_0.34.0             
 [35] DelayedArray_0.20.0         scales_1.2.0               
 [37] DBI_1.1.2                   Rcpp_1.0.9                 
 [39] progress_1.2.2              htmlTable_2.2.1            
 [41] foreign_0.8-81              bit_4.0.4                  
 [43] Formula_1.2-4               htmlwidgets_1.5.3          
 [45] httr_1.4.3                  RColorBrewer_1.1-3         
 [47] ellipsis_0.3.2              pkgconfig_2.0.3            
 [49] XML_3.99-0.6                farver_2.1.0               
 [51] nnet_7.3-16                 sass_0.4.0                 
 [53] dbplyr_2.1.1                utf8_1.2.2                 
 [55] here_1.0.1                  tidyselect_1.1.2           
 [57] labeling_0.4.2              rlang_1.0.4                
 [59] later_1.2.0                 AnnotationDbi_1.56.1       
 [61] munsell_0.5.0               tools_4.1.0                
 [63] cachem_1.0.6                cli_3.3.0                  
 [65] generics_0.1.2              RSQLite_2.2.14             
 [67] evaluate_0.15               stringr_1.4.0              
 [69] fastmap_1.1.0               yaml_2.2.1                 
 [71] processx_3.5.3              knitr_1.33                 
 [73] bit64_4.0.5                 fs_1.5.2                   
 [75] purrr_0.3.4                 KEGGREST_1.34.0            
 [77] AnnotationFilter_1.18.0     whisker_0.4                
 [79] xml2_1.3.2                  biomaRt_2.50.0             
 [81] compiler_4.1.0              rstudioapi_0.13            
 [83] filelock_1.0.2              curl_4.3.2                 
 [85] png_0.1-7                   tibble_3.1.7               
 [87] bslib_0.4.0                 stringi_1.7.6              
 [89] highr_0.9                   ps_1.7.0                   
 [91] GenomicFeatures_1.46.1      lattice_0.20-44            
 [93] ProtGenerics_1.26.0         Matrix_1.3-3               
 [95] vctrs_0.4.1                 pillar_1.7.0               
 [97] lifecycle_1.0.1             jquerylib_0.1.4            
 [99] data.table_1.14.2           bitops_1.0-7               
[101] httpuv_1.6.1                rtracklayer_1.54.0         
[103] R6_2.5.1                    BiocIO_1.4.0               
[105] latticeExtra_0.6-29         promises_1.2.0.1           
[107] gridExtra_2.3               dichromat_2.0-0.1          
[109] assertthat_0.2.1            SummarizedExperiment_1.24.0
[111] rprojroot_2.0.3             rjson_0.2.20               
[113] withr_2.5.0                 GenomicAlignments_1.30.0   
[115] Rsamtools_2.10.0            GenomeInfoDbData_1.2.7     
[117] parallel_4.1.0              hms_1.1.1                  
[119] rpart_4.1-15                rmarkdown_2.9              
[121] MatrixGenerics_1.6.0        git2r_0.28.0               
[123] biovizBase_1.42.0           getPass_0.2-2              
[125] Biobase_2.54.0              base64enc_0.1-3            
[127] restfulr_0.0.13            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
