<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />

<meta name="date" content="2022-10-18" />

<title>LDL - Liver</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">LDL - Liver</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2022-10-18</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config_b38.R
</td>
<td style="text-align:left;">
code/ctwas_config_b38.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1585ea7fc75ac2c19d9d0293655a0f88436b6d35targetblank1585ea7a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/1585ea7fc75ac2c19d9d0293655a0f88436b6d35" target="_blank">1585ea7</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1585ea7fc75ac2c19d9d0293655a0f88436b6d35targetblank1585ea7a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/1585ea7fc75ac2c19d9d0293655a0f88436b6d35" target="_blank">1585ea7</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  Proposal plots.R
    Untracked:  RGS14.pdf
    Untracked:  RNF186.pdf
    Untracked:  SCZ_annotation.xlsx
    Untracked:  SLC8B1.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/LDL_Liver_S_gene_level.Rmd
    Untracked:  cache/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/IBD_ME_3kb_0.05_out/
    Untracked:  code/LDL_S_out/LDL_Liver.err
    Untracked:  code/LDL_S_out/LDL_Liver.out
    Untracked:  code/LDL_out/
    Untracked:  code/run_IBD_analysis_ME_3kb_0.05.sbatch
    Untracked:  code/run_IBD_analysis_ME_3kb_0.05.sh
    Untracked:  code/run_IBD_ctwas_rss_LDR_ME_3kb_0.05.R
    Untracked:  code/run_LDL_analysis.sbatch
    Untracked:  code/run_LDL_analysis.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/FUMA_output/
    Untracked:  data/GO_Terms/
    Untracked:  data/IBD_ME/
    Untracked:  data/LDL/
    Untracked:  data/LDL_S/
    Untracked:  data/PGC3_SCZ_wave3_public.v2.tsv
    Untracked:  data/SCZ/
    Untracked:  data/SCZ_2014_EUR/
    Untracked:  data/SCZ_2014_EUR_ME/
    Untracked:  data/SCZ_2018/
    Untracked:  data/SCZ_2018_ME/
    Untracked:  data/SCZ_2018_S/
    Untracked:  data/SCZ_2020/
    Untracked:  data/SCZ_S/
    Untracked:  data/Supplementary Table 15 - MAGMA.xlsx
    Untracked:  data/Supplementary Table 20 - Prioritised Genes.xlsx
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/gwas_sumstats/
    Untracked:  data/magma.genes.out
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/notes.txt
    Untracked:  data/scz_2018.RDS
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  temp_LDR/
    Untracked:  top_genes_32.txt
    Untracked:  top_genes_37.txt
    Untracked:  top_genes_43.txt
    Untracked:  top_genes_54.txt
    Untracked:  top_genes_81.txt
    Untracked:  z_snp_pos_SCZ.RData
    Untracked:  z_snp_pos_SCZ_2014_EUR.RData
    Untracked:  z_snp_pos_SCZ_2018.RData
    Untracked:  z_snp_pos_SCZ_2020.RData

Unstaged changes:
    Deleted:    analysis/BMI_S_results.Rmd
    Modified:   analysis/LDL_Liver.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Deleted:    code/LDL_S_out/T2D_Liver.err
    Deleted:    code/LDL_S_out/T2D_Liver.out
    Modified:   code/SCZ_out/SCZ_Brain_Amygdala.err
    Modified:   code/SCZ_out/SCZ_Brain_Amygdala.out
    Modified:   code/SCZ_out/SCZ_Brain_Anterior_cingulate_cortex_BA24.err
    Modified:   code/SCZ_out/SCZ_Brain_Anterior_cingulate_cortex_BA24.out
    Modified:   code/SCZ_out/SCZ_Brain_Caudate_basal_ganglia.err
    Modified:   code/SCZ_out/SCZ_Brain_Caudate_basal_ganglia.out
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellar_Hemisphere.err
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellar_Hemisphere.out
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellum.err
    Modified:   code/SCZ_out/SCZ_Brain_Cerebellum.out
    Modified:   code/SCZ_out/SCZ_Brain_Cortex.err
    Modified:   code/SCZ_out/SCZ_Brain_Cortex.out
    Modified:   code/SCZ_out/SCZ_Brain_Frontal_Cortex_BA9.err
    Modified:   code/SCZ_out/SCZ_Brain_Frontal_Cortex_BA9.out
    Modified:   code/SCZ_out/SCZ_Brain_Hippocampus.err
    Modified:   code/SCZ_out/SCZ_Brain_Hippocampus.out
    Modified:   code/SCZ_out/SCZ_Brain_Hypothalamus.err
    Modified:   code/SCZ_out/SCZ_Brain_Hypothalamus.out
    Modified:   code/SCZ_out/SCZ_Brain_Nucleus_accumbens_basal_ganglia.err
    Modified:   code/SCZ_out/SCZ_Brain_Nucleus_accumbens_basal_ganglia.out
    Modified:   code/SCZ_out/SCZ_Brain_Putamen_basal_ganglia.err
    Modified:   code/SCZ_out/SCZ_Brain_Putamen_basal_ganglia.out
    Modified:   code/SCZ_out/SCZ_Brain_Spinal_cord_cervical_c-1.err
    Modified:   code/SCZ_out/SCZ_Brain_Spinal_cord_cervical_c-1.out
    Modified:   code/SCZ_out/SCZ_Brain_Substantia_nigra.err
    Modified:   code/SCZ_out/SCZ_Brain_Substantia_nigra.out
    Deleted:    code/run_IBD_ctwas_rss_LDR_ME.R
    Modified:   code/run_LDL_analysis_S.sbatch
    Modified:   code/run_LDL_analysis_S.sh
    Modified:   code/run_LDL_ctwas_rss_LDR_S.R
    Modified:   code/run_SCZ_analysis.sbatch
    Modified:   code/run_SCZ_analysis.sh
    Modified:   code/run_SCZ_ctwas_rss_LDR.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 21553</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
2079 1523 1276  769  878 1116 1176  701  854  931 1344 1116  391  772  677 1085 
  17   18   19   20   21   22 
1462  258 1524  659  262  700 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 19937</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.925</code></pre>
<pre class="r"><code>library(reticulate)
use_python(&quot;/scratch/midway2/shengqian/miniconda3/envs/PythonForR/bin/python&quot;,required=T)</code></pre>
<pre><code>INFO:numexpr.utils:Note: NumExpr detected 28 cores but &quot;NUMEXPR_MAX_THREADS&quot; not set, so enforcing safe limit of 8.</code></pre>
<pre><code>finish</code></pre>
<pre class="r"><code>#add z scores to results
load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_expr_z_gene.Rd&quot;))
ctwas_gene_res &lt;- py$ctwas_gene_res_df
ctwas_gene_res$z &lt;- z_gene[ctwas_gene_res$intron_id,]$z

z_snp &lt;- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z &lt;- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$intron_id=NA
ctwas_snp_res$intron_pos=NA

saveRDS(ctwas_gene_res, file = paste0(results_dir,&quot;/&quot;,analysis_id,&quot;_ctwas_intron_res.RDS&quot;))
saveRDS(ctwas_snp_res, file =  paste0(results_dir,&quot;/&quot;,analysis_id,&quot;_ctwas_snp_res.RDS&quot;))</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>ctwas_res &lt;- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#store columns to report
report_cols &lt;- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c(&quot;type&quot;, &quot;region_tag1&quot;, &quot;region_tag2&quot;, &quot;cs_index&quot;, &quot;gene_type&quot;, &quot;z_flag&quot;, &quot;id&quot;, &quot;chrom&quot;, &quot;pos&quot;))]
first_cols &lt;- c(&quot;genename&quot;, &quot;region_tag&quot;)
report_cols &lt;- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps &lt;- c(&quot;id&quot;, report_cols[-1])
report_cols_snps &lt;- report_cols_snps[!(report_cols_snps %in% &quot;num_sqtl&quot;)]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 &lt;- data.table::fread(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s1.susieIrss.txt&quot;))
n_snps &lt;- sum(ctwas_res_s1$type==&quot;SNP&quot;)/thin
rm(ctwas_res_s1)</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<pre class="r"><code>#distribution of PIPs
hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main=&quot;Distribution of Gene PIPs&quot;)</code></pre>
<p><img src="figure/LDL_Liver_S_gene_level.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#genes with PIP&gt;0.8 or 20 highest PIPs
head(ctwas_gene_res[order(-ctwas_gene_res$susie_pip),report_cols], max(sum(ctwas_gene_res$susie_pip&gt;0.8), 20))</code></pre>
<pre><code>        genename region_tag susie_pip    mu2       PVE       z num_intron
1353       CNIH4      1_114    1.8027  35.42 2.610e-04   5.273          6
2343        FKRP      19_33    1.7010  18.81 1.495e-04  -3.018          5
1053      CCDC57      17_47    1.4400  20.38 1.053e-04  -4.344         19
4796         PXK       3_40    1.2808  53.31 1.553e-04   6.852          6
190       ADHFE1       8_50    1.2508  18.20 6.146e-05  -3.584         14
5488    SLC22A18       11_2    1.1989  19.89 7.752e-05   4.096         13
1623     CYP4F12      19_13    1.1917  36.25 1.315e-04  -5.868         14
3812      MYO15B      17_42    1.1814  35.52 8.339e-05  -4.053         26
3848     NADSYN1      11_40    1.1678  11.24 2.297e-05   1.456         20
119         ACP6       1_73    1.1614  18.88 7.057e-05   3.994          5
4247       PARP9       3_76    1.1591  40.89 1.482e-04   6.409          8
1530 CTC-360G5.8      19_26    1.1535  21.58 7.356e-05   4.428         11
4637      PPP6R2      22_24    1.1406  25.23 8.252e-05  -4.832         14
5313      SEC16B       1_87    1.1343  22.62 6.981e-05  -3.767         15
2919      IL17RC        3_8    1.1248  21.95 5.256e-05   2.782         19
2089      ERGIC3      20_21    1.1191  51.67 1.667e-04  -7.267          5
3301       LRCH4       7_61    1.0948  31.67 1.087e-04   5.294          6
6059       THOP1       19_3    1.0915  27.51 8.892e-05   5.087          7
4378      PHLDB1      11_71    1.0895  23.26 6.229e-05   3.789          8
101         ACCS      11_27    1.0606  12.34 2.970e-05  -1.952         15
2820          HP      16_38    1.0598 273.62 8.764e-04  21.869          6
21         ABCA8      17_39    1.0506  27.86 8.623e-05  -4.775         12
3018       ITGAL      16_24    1.0361  21.82 6.601e-05  -4.428          3
4047        NQO2        6_3    1.0351  13.31 2.705e-05  -1.687         22
534        ASGR1       17_6    1.0344  83.95 2.614e-04   9.645          2
5736      SPRED2       2_42    1.0246  32.05 7.230e-05  -4.438          2
5788     ST3GAL4      11_77    1.0244  76.58 2.204e-04 -12.154          5
2349       FLOT2      17_17    1.0193  31.16 8.363e-05  -3.738          3
3847       NADK2       5_24    1.0159  20.93 6.200e-05   4.321          3
3981        NINL      20_19    1.0000  25.08 6.795e-05  -4.733          7
3226        LDLR       19_9    1.0000 751.97 2.188e-03  26.898          2
6593       USP53       4_77    0.9986  26.01 7.209e-05   4.888          4
3200       LAMA5      20_36    0.9982  14.84 2.480e-05   1.946         16
883      C2orf42       2_46    0.9963  22.94 5.154e-05  -3.356         12
2771       HLA-B       6_25    0.9916  62.55 1.790e-04   8.790         11
6591       USP47       11_9    0.9913  19.00 4.980e-05   3.654          7
6135    TMEM150A       2_54    0.9866  19.99 5.486e-05   4.079          4
4626     PPP2R5C      14_53    0.9859  20.51 5.247e-05  -3.976          6
813    C14orf159      14_46    0.9817  18.47 2.745e-05   2.224         17
4413      PIH1D1      19_34    0.9810  23.13 6.245e-05   3.988          7
45        ABHD12      20_18    0.9713  26.14 6.203e-05  -4.696          7
681        BCAT2      19_34    0.9682  27.94 7.568e-05  -5.197          2
4870      RALGDS       9_70    0.9665  17.53 4.588e-05   3.277          4
5408      SH3BP2        4_3    0.9603  21.48 4.788e-05   2.887         10
3121      KIF13B       8_28    0.9476  23.56 6.155e-05  -4.718          1
4569         POR       7_48    0.9384  33.80 7.885e-05  -5.646          5
5910       TACC1       8_34    0.9325  18.93 4.616e-05   3.936          4
1040     CCDC159      19_10    0.9318  40.93 7.676e-05  -3.958          7
6115       TMED4       7_32    0.9194  28.69 6.984e-05  -4.609          3
4448      PLA2G6      22_15    0.9176  15.78 3.232e-05  -2.419         14
5721       SPHK2      19_33    0.9153  42.12 1.026e-04   8.721          2
31         ABCC3      17_29    0.9118  13.25 2.257e-05  -1.908         16
3778      MTSS1L      16_37    0.9096  17.51 4.056e-05  -2.815          7
6524      UGT1A1      2_137    0.9090  30.83 6.881e-05   5.450          4
898           C5       9_62    0.9073  24.46 4.818e-05  -4.725          8
4365        PHC1       12_9    0.9054  36.98 8.822e-05   6.156          1
3083       KDM1A       1_16    0.8913  22.63 4.809e-05   4.158          3
4965       REPS1       6_92    0.8869  34.02 6.208e-05   3.381          8
1609      CYP2C8      10_61    0.8844  17.74 3.879e-05   3.549          8
2552        GNMT       6_33    0.8810  24.77 5.145e-05  -4.897          4
2358        FMO3       1_84    0.8770  23.11 3.982e-05   4.221          9
6240        TNK2      3_120    0.8753  23.32 3.771e-05   2.814         10
1392     COL18A1      21_23    0.8702  18.80 2.432e-05  -2.110         15
763         BRD9        5_1    0.8564  19.17 3.832e-05   3.964         12
5790     ST3GAL6       3_62    0.8559  27.99 4.872e-05   3.387          9
4818      R3HDM2      12_36    0.8555  36.25 7.375e-05  -5.818          3
525        ASAP3       1_16    0.8532  25.69 5.020e-05  -4.936          7
3619        MMAB      12_66    0.8529  18.44 3.180e-05   3.833         11
282      ALDH1A2      15_26    0.8444  62.53 1.146e-04  -7.783          9
1724     DENND5A       11_8    0.8428  22.43 3.220e-05   3.050          9
4580       PPCDC      15_35    0.8401  18.78 3.588e-05   4.030          5
1630      CYSTM1       5_83    0.8389  18.90 3.795e-05  -4.021          4
5814       STAU1      20_30    0.8386  21.35 4.004e-05   4.296          6
4361        PGS1      17_45    0.8372  48.60 9.077e-05  -7.141          5
1948       EFNA1       1_76    0.8364  28.21 5.706e-05   3.331          3
5926      TANGO2       22_4    0.8325  19.01 3.475e-05  -3.713          9
5365    SERPINF2       17_2    0.8273  20.50 3.732e-05   4.092          5
5352       SEPT2      2_144    0.8273  19.09 3.419e-05   3.767          9
2963      INPPL1      11_40    0.8260  12.15 1.504e-05  -1.424         12
3821       MYO9B      19_14    0.8226  20.72 3.883e-05  -4.344          7
1480      CRELD1        3_8    0.8185  19.42 2.475e-05   2.483         14
5440       SIPA1      11_36    0.8120  20.64 3.793e-05  -4.440          5
4132       NUP88       17_5    0.8105  19.78 2.973e-05   3.087         11
6328       TRIM4       7_61    0.8100  24.15 4.431e-05  -4.329          3
6693       WDR27      6_111    0.8061  16.95 1.531e-05   1.828         22
5255       SBNO2       19_2    0.8007  31.61 4.431e-05   3.473         12
     num_sqtl
1353        7
2343        5
1053       26
4796        7
190        14
5488       20
1623       16
3812       38
3848       27
119         6
4247       10
1530       13
4637       15
5313       16
2919       24
2089        6
3301        7
6059        7
4378        8
101        19
2820        9
21         15
3018        3
4047       39
534         2
5736        2
5788       10
2349        3
3847        3
3981        7
3226        3
6593        5
3200       25
883        14
2771       28
6591        7
6135        4
4626        7
813        21
4413        7
45          9
681         2
4870        4
5408       11
3121        1
4569        6
5910        4
1040        8
6115        4
4448       22
5721        2
31         20
3778        7
6524        4
898         9
4365        1
3083        3
4965       10
1609        8
2552        4
2358        9
6240       11
1392       16
763        15
5790       10
4818        3
525         7
3619       19
282         9
1724       10
4580        7
1630        6
5814        6
4361        5
1948        4
5926        9
5365        7
5352       10
2963       13
3821        7
1480       16
5440        6
4132       15
6328        5
6693       33
5255       14</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<pre class="r"><code>#set nominal signifiance threshold for z scores
alpha &lt;- 0.05

#bonferroni adjusted threshold for z scores
sig_thresh &lt;- qnorm(1-(alpha/nrow(ctwas_gene_res)/2), lower=T)</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>library(&quot;readxl&quot;)

known_annotations &lt;- read_xlsx(&quot;data/summary_known_genes_annotations.xlsx&quot;, sheet=&quot;LDL&quot;)</code></pre>
<pre><code>New names:</code></pre>
<pre class="r"><code>known_annotations &lt;- unique(known_annotations$`Gene Symbol`)

unrelated_genes &lt;- ctwas_gene_res$genename[!(ctwas_gene_res$genename %in% known_annotations)]

#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 69</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 38</code></pre>
<pre class="r"><code>#assign ctwas, TWAS, and bystander genes
ctwas_genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.9]
twas_genes &lt;- ctwas_gene_res$genename[abs(ctwas_gene_res$z)&gt;sig_thresh]
novel_genes &lt;- ctwas_genes[!(ctwas_genes %in% twas_genes)]

#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.49</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 56</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 219</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>        genename region_tag susie_pip   mu2       PVE      z num_intron
31         ABCC3      17_29    0.9118 13.25 2.257e-05 -1.908         16
101         ACCS      11_27    1.0606 12.34 2.970e-05 -1.952         15
119         ACP6       1_73    1.1614 18.88 7.057e-05  3.994          5
190       ADHFE1       8_50    1.2508 18.20 6.146e-05 -3.584         14
813    C14orf159      14_46    0.9817 18.47 2.745e-05  2.224         17
883      C2orf42       2_46    0.9963 22.94 5.154e-05 -3.356         12
1040     CCDC159      19_10    0.9318 40.93 7.676e-05 -3.958          7
1053      CCDC57      17_47    1.4400 20.38 1.053e-04 -4.344         19
1530 CTC-360G5.8      19_26    1.1535 21.58 7.356e-05  4.428         11
2343        FKRP      19_33    1.7010 18.81 1.495e-04 -3.018          5
2349       FLOT2      17_17    1.0193 31.16 8.363e-05 -3.738          3
2919      IL17RC        3_8    1.1248 21.95 5.256e-05  2.782         19
3018       ITGAL      16_24    1.0361 21.82 6.601e-05 -4.428          3
3200       LAMA5      20_36    0.9982 14.84 2.480e-05  1.946         16
3778      MTSS1L      16_37    0.9096 17.51 4.056e-05 -2.815          7
3812      MYO15B      17_42    1.1814 35.52 8.339e-05 -4.053         26
3847       NADK2       5_24    1.0159 20.93 6.200e-05  4.321          3
3848     NADSYN1      11_40    1.1678 11.24 2.297e-05  1.456         20
4047        NQO2        6_3    1.0351 13.31 2.705e-05 -1.687         22
4378      PHLDB1      11_71    1.0895 23.26 6.229e-05  3.789          8
4413      PIH1D1      19_34    0.9810 23.13 6.245e-05  3.988          7
4448      PLA2G6      22_15    0.9176 15.78 3.232e-05 -2.419         14
4626     PPP2R5C      14_53    0.9859 20.51 5.247e-05 -3.976          6
4870      RALGDS       9_70    0.9665 17.53 4.588e-05  3.277          4
5313      SEC16B       1_87    1.1343 22.62 6.981e-05 -3.767         15
5408      SH3BP2        4_3    0.9603 21.48 4.788e-05  2.887         10
5488    SLC22A18       11_2    1.1989 19.89 7.752e-05  4.096         13
5736      SPRED2       2_42    1.0246 32.05 7.230e-05 -4.438          2
5910       TACC1       8_34    0.9325 18.93 4.616e-05  3.936          4
6135    TMEM150A       2_54    0.9866 19.99 5.486e-05  4.079          4
6591       USP47       11_9    0.9913 19.00 4.980e-05  3.654          7
     num_sqtl
31         20
101        19
119         6
190        14
813        21
883        14
1040        8
1053       26
1530       13
2343        5
2349        3
2919       24
3018        3
3200       25
3778        7
3812       38
3847        3
3848       27
4047       39
4378        8
4413        7
4448       22
4626        7
4870        4
5313       16
5408       11
5488       20
5736        2
5910        4
6135        4
6591        7</code></pre>
<pre class="r"><code>#sensitivity / recall
sensitivity &lt;- rep(NA,2)
names(sensitivity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
sensitivity[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
sensitivity</code></pre>
<pre><code>  ctwas    TWAS 
0.01449 0.05797 </code></pre>
<pre class="r"><code>#specificity
specificity &lt;- rep(NA,2)
names(specificity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
specificity[&quot;ctwas&quot;] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity[&quot;TWAS&quot;] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
specificity</code></pre>
<pre><code> ctwas   TWAS 
0.9921 0.9692 </code></pre>
<pre class="r"><code>#precision / PPV
precision &lt;- rep(NA,2)
names(precision) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
precision[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(twas_genes)
precision</code></pre>
<pre><code>  ctwas    TWAS 
0.01786 0.01826 </code></pre>
<pre class="r"><code>#ROC curves

pip_range &lt;- (0:1000)/1000
sensitivity &lt;- rep(NA, length(pip_range))
specificity &lt;- rep(NA, length(pip_range))

for (index in 1:length(pip_range)){
  pip &lt;- pip_range[index]
  ctwas_genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;=pip]
  sensitivity[index] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
}

plot(1-specificity, sensitivity, type=&quot;l&quot;, xlim=c(0,1), ylim=c(0,1))

sig_thresh_range &lt;- seq(from=0, to=max(abs(ctwas_gene_res$z)), length.out=length(pip_range))

for (index in 1:length(sig_thresh_range)){
  sig_thresh_plot &lt;- sig_thresh_range[index]
  twas_genes &lt;- ctwas_gene_res$genename[abs(ctwas_gene_res$z)&gt;=sig_thresh_plot]
  sensitivity[index] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
}

lines(1-specificity, sensitivity, xlim=c(0,1), ylim=c(0,1), col=&quot;red&quot;, lty=2)</code></pre>
<p><img src="figure/LDL_Liver_S_gene_level.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readxl_1.4.0    dplyr_1.0.9     reticulate_1.26 workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.2  xfun_0.24         bslib_0.4.0       purrr_0.3.4      
 [5] lattice_0.20-44   vctrs_0.4.1       generics_0.1.2    htmltools_0.5.3  
 [9] yaml_2.2.1        utf8_1.2.2        rlang_1.0.4       later_1.2.0      
[13] pillar_1.7.0      jquerylib_0.1.4   DBI_1.1.2         glue_1.6.2       
[17] lifecycle_1.0.1   stringr_1.4.0     cellranger_1.1.0  evaluate_0.15    
[21] knitr_1.33        callr_3.7.0       fastmap_1.1.0     httpuv_1.6.1     
[25] ps_1.7.0          fansi_1.0.3       highr_0.9         Rcpp_1.0.9       
[29] promises_1.2.0.1  cachem_1.0.6      jsonlite_1.8.0    fs_1.5.2         
[33] png_0.1-7         digest_0.6.29     stringi_1.7.6     processx_3.5.3   
[37] getPass_0.2-2     rprojroot_2.0.3   grid_4.1.0        here_1.0.1       
[41] cli_3.3.0         tools_4.1.0       magrittr_2.0.3    sass_0.4.0       
[45] tibble_3.1.7      crayon_1.5.1      whisker_0.4       pkgconfig_2.0.3  
[49] ellipsis_0.3.2    Matrix_1.3-3      data.table_1.14.2 assertthat_0.2.1 
[53] rmarkdown_2.9     httr_1.4.3        rstudioapi_0.13   R6_2.5.1         
[57] git2r_0.28.0      compiler_4.1.0   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
