<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />

<meta name="date" content="2023-10-14" />

<title>OxHb</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">OxHb</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2023-10-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/
</td>
<td style="text-align:left;">
data/OxHb
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree3fd82cfb292c1eef311031acdda0bad73eb2fea8targetblank3fd82cfa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/3fd82cfb292c1eef311031acdda0bad73eb2fea8" target="_blank">3fd82cf</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree3fd82cfb292c1eef311031acdda0bad73eb2fea8targetblank3fd82cfa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/3fd82cfb292c1eef311031acdda0bad73eb2fea8" target="_blank">3fd82cf</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  LDL_LDLR_locus1.pdf
    Untracked:  LDL_TEME199_genetrack.pdf
    Untracked:  LDL_TEME199_locus.pdf
    Untracked:  Proposal plots.R
    Untracked:  RGS14.pdf
    Untracked:  RNF186.pdf
    Untracked:  Rplots.pdf
    Untracked:  SCZ_annotation.xlsx
    Untracked:  SLC8B1.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  cache/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/FUMA_output/
    Untracked:  data/GO_Terms/
    Untracked:  data/GTEx_Analysis_v8_eQTL.tar
    Untracked:  data/G_list.RData
    Untracked:  data/IBD_ME/
    Untracked:  data/LB/
    Untracked:  data/LDL/
    Untracked:  data/LDL_E_S/
    Untracked:  data/LDL_E_S_M/
    Untracked:  data/LDL_M/
    Untracked:  data/LDL_S/
    Untracked:  data/OxHb/
    Untracked:  data/PGC3_SCZ_wave3_public.v2.tsv
    Untracked:  data/Predictive_Models/
    Untracked:  data/Supplementary Table 15 - MAGMA.xlsx
    Untracked:  data/Supplementary Table 20 - Prioritised Genes.xlsx
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/WhiteBlood_E/
    Untracked:  data/WhiteBlood_E_M/
    Untracked:  data/WhiteBlood_E_S_M/
    Untracked:  data/WhiteBlood_E_S_M_PC/
    Untracked:  data/WhiteBlood_M/
    Untracked:  data/WhiteBlood_M_compare/
    Untracked:  data/WhiteBlood_M_enet/
    Untracked:  data/cpg_annot.RData
    Untracked:  data/eqtl/
    Untracked:  data/gencode.v26.GRCh38.genes.gtf
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/gwas_sumstats/
    Untracked:  data/magma.genes.out
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/mqtl/
    Untracked:  data/notes.txt
    Untracked:  data/scz_2018.RDS
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/test/
    Untracked:  hist.pdf
    Untracked:  ld_pip.pdf
    Untracked:  submit.sh
    Untracked:  temp_LDR/
    Untracked:  test-B1.snpgwas.txt

Unstaged changes:
    Deleted:    analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd
    Deleted:    analysis/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd
    Deleted:    analysis/Autism_Brain_Amygdala.Rmd
    Deleted:    analysis/Autism_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Deleted:    analysis/Autism_Brain_Caudate_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Cerebellar_Hemisphere.Rmd
    Deleted:    analysis/Autism_Brain_Cerebellum.Rmd
    Deleted:    analysis/Autism_Brain_Cortex.Rmd
    Deleted:    analysis/Autism_Brain_Frontal_Cortex_BA9.Rmd
    Deleted:    analysis/Autism_Brain_Hippocampus.Rmd
    Deleted:    analysis/Autism_Brain_Hypothalamus.Rmd
    Deleted:    analysis/Autism_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Putamen_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Spinal_cord_cervical_c-1.Rmd
    Deleted:    analysis/Autism_Brain_Substantia_nigra.Rmd
    Deleted:    analysis/BMI_Brain_Amygdala.Rmd
    Deleted:    analysis/BMI_Brain_Amygdala_S.Rmd
    Deleted:    analysis/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Deleted:    analysis/BMI_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Deleted:    analysis/BMI_Brain_Caudate_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Caudate_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellar_Hemisphere.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellar_Hemisphere_S.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellum.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellum_S.Rmd
    Deleted:    analysis/BMI_Brain_Cortex.Rmd
    Deleted:    analysis/BMI_Brain_Cortex_S.Rmd
    Deleted:    analysis/BMI_Brain_Frontal_Cortex_BA9.Rmd
    Deleted:    analysis/BMI_Brain_Frontal_Cortex_BA9_S.Rmd
    Deleted:    analysis/BMI_Brain_Hippocampus.Rmd
    Deleted:    analysis/BMI_Brain_Hippocampus_S.Rmd
    Deleted:    analysis/BMI_Brain_Hypothalamus.Rmd
    Deleted:    analysis/BMI_Brain_Hypothalamus_S.Rmd
    Deleted:    analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Putamen_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Putamen_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Spinal_cord_cervical_c-1.Rmd
    Deleted:    analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Deleted:    analysis/BMI_Brain_Substantia_nigra.Rmd
    Deleted:    analysis/BMI_Brain_Substantia_nigra_S.Rmd
    Deleted:    analysis/BMI_S_results.Rmd
    Deleted:    analysis/Glucose_Adipose_Subcutaneous.Rmd
    Deleted:    analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Modified:   analysis/OxHB.Rmd
    Deleted:    code/WhiteBlood_M_ener_out/WhiteBlood_WholeBlood.out
    Deleted:    code/White_Blood_M_out/White_Blood_BreastMammary.err
    Deleted:    code/White_Blood_M_out/White_Blood_BreastMammary.out
    Deleted:    code/White_Blood_M_out/White_Blood_ColonTransverse.err
    Deleted:    code/White_Blood_M_out/White_Blood_ColonTransverse.out
    Deleted:    code/White_Blood_M_out/White_Blood_KidneyCortex.err
    Deleted:    code/White_Blood_M_out/White_Blood_KidneyCortex.out
    Deleted:    code/White_Blood_M_out/White_Blood_Lung.err
    Deleted:    code/White_Blood_M_out/White_Blood_Lung.out
    Deleted:    code/White_Blood_M_out/White_Blood_MuscleSkeletal.err
    Deleted:    code/White_Blood_M_out/White_Blood_MuscleSkeletal.out
    Deleted:    code/White_Blood_M_out/White_Blood_Ovary.err
    Deleted:    code/White_Blood_M_out/White_Blood_Ovary.out
    Deleted:    code/White_Blood_M_out/White_Blood_Prostate.err
    Deleted:    code/White_Blood_M_out/White_Blood_Prostate.out
    Deleted:    code/White_Blood_M_out/White_Blood_Testis.err
    Deleted:    code/White_Blood_M_out/White_Blood_Testis.out
    Deleted:    code/White_Blood_M_out/White_Blood_WholeBlood.err
    Deleted:    code/White_Blood_M_out/White_Blood_WholeBlood.out
    Deleted:    code/run_IBD_ctwas_rss_LDR_ME.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/OxHB.Rmd</code>) and HTML (<code>docs/OxHB.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3fd82cfb292c1eef311031acdda0bad73eb2fea8/analysis/OxHB.Rmd" target="_blank">3fd82cf</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/3fd82cfb292c1eef311031acdda0bad73eb2fea8/docs/OxHB.html" target="_blank">3fd82cf</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/70d8ee87defd07dd0f388013a4ebc80690fe33fb/analysis/OxHB.Rmd" target="_blank">70d8ee8</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/70d8ee87defd07dd0f388013a4ebc80690fe33fb/docs/OxHB.html" target="_blank">70d8ee8</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/d04dec98d50beda9d49a049d09f3607c2e4ea5cb/analysis/OxHB.Rmd" target="_blank">d04dec9</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/d04dec98d50beda9d49a049d09f3607c2e4ea5cb/docs/OxHB.html" target="_blank">d04dec9</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/f442f0e5746534ded6194155b45d49ea86f8bf9f/docs/OxHB.html" target="_blank">f442f0e</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/analysis/OxHB.Rmd" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/OxHB.html" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e0d796ff985efb017a25dc1941b312a624f0576d/analysis/OxHB.Rmd" target="_blank">e0d796f</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="artery-coronary" class="section level3">
<h3>Artery Coronary</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003789661 0.0048196252 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.6727008 0.5860888 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
12.71783 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>       SNP       gene 
0.71103798 0.01352419 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7245622</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.98133467 0.01866533 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-3-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#print top genes
head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
164440     9 ENSG00000136868.10 113209149 gene           9          58        0
263152    20 ENSG00000089177.18  16581461 gene          20          12        0
272677    22 ENSG00000184381.18  38145259 gene          22          15        0
261678    20  ENSG00000275491.1   3807832 gene          20           3        0
90495      5 ENSG00000132356.11  40798520 gene           5          27        0
117969     6 ENSG00000178425.13 116099964 gene           6          77        0
        susie_pip      mu2  genename         z
164440 0.11803156 9.216227   SLC31A1 -3.933203
263152 0.09730827 8.050336    KIF16B -3.085439
272677 0.08515657 7.548930    PLA2G6  3.053686
261678 0.08398874 7.466874 LINC01730 -2.945771
90495  0.08384722 7.723628    PRKAA1 -3.439407
117969 0.08334959 7.383582    NT5DC1  3.118395</code></pre>
</div>
<div id="artery-tibial" class="section level3">
<h3>Artery Tibial</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003778214 0.0040115252 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.6684344 0.5565171 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
10.61752 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>       SNP       gene 
0.70439414 0.01309908 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7174932</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.98174327 0.01825673 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-7-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
114696     6 ENSG00000005700.14  82247064 gene           6          56        0
272419    22 ENSG00000159496.14  23679302 gene          22           6        0
179945    10 ENSG00000151923.17 119601669 gene          10          74        0
151888     8  ENSG00000164841.4 108787887 gene           8          74        0
241123    16 ENSG00000040199.18  71804965 gene          16          38        0
243700    17 ENSG00000174292.12   7380407 gene          17           6        0
        susie_pip      mu2 genename         z
114696 0.11468988 9.126035     IBTK -3.787795
272419 0.08483873 7.971765     RGL4  3.286735
179945 0.08104538 7.839875    TIAL1  3.480318
151888 0.07962949 7.989882   TMEM74 -3.623924
241123 0.07790796 7.513540   PHLPP2 -3.126064
243700 0.07591772 7.535001     TNK1  3.186510</code></pre>
</div>
<div id="heart-atrial-appendage" class="section level3">
<h3>Heart Atrial Appendage</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003779586 0.0050855938 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.6799803 0.8693670 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
13.45542 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>       SNP       gene 
0.71682145 0.02367152 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.740493</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.96803275 0.03196725 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-11-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
165901     9 ENSG00000136944.17 126565356 gene           9          65        0
57759      3 ENSG00000144908.13 126016332 gene           3          78        0
95338      5  ENSG00000250049.5  92296991 gene           5          54        0
95337      5  ENSG00000248864.1  92329276 gene           5          54        0
65473      4 ENSG00000159733.13   2395131 gene           4           3        0
224367    14 ENSG00000027075.14  61235011 gene          14          28        0
       susie_pip      mu2      genename         z
165901 0.1584757 12.62261         LMX1B  3.258848
57759  0.1535566 13.15896       ALDH1L1  3.817762
95338  0.1511946 12.76268 RP11-348J24.2  3.951225
95337  0.1505937 12.74289 RP11-348J24.1  3.943968
65473  0.1463878 12.17379       ZFYVE28  3.257767
224367 0.1338625 11.63597         PRKCH -3.100649</code></pre>
</div>
<div id="heart-left-ventricle" class="section level3">
<h3>Heart Left Ventricle</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003788274 0.0054372428 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.6450365 2.3567095 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
14.35282 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>      SNP      gene 
0.6815476 0.0633139 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7448615</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.91499909 0.08500091 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-15-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
232652    15 ENSG00000137819.13  69299268 gene          15          32        0
95174      5  ENSG00000248864.1  92329276 gene           5          54        0
256443    19 ENSG00000171970.12   2897135 gene          19           3        0
176862    10 ENSG00000176273.14  93893644 gene          10          60        0
100805     5 ENSG00000070814.17 150356800 gene           5          88        0
271544    22  ENSG00000272733.1  23686225 gene          22           6        0
       susie_pip      mu2      genename         z
232652 0.3830957 28.46332         PAQR5  4.277398
95174  0.2857001 25.74061 RP11-348J24.1  3.943968
256443 0.2260197 23.52581         ZNF57 -3.403225
176862 0.2147015 23.56849       SLC35G1  3.538506
100805 0.2105803 23.08782         TCOF1  3.426022
271544 0.2065853 23.00182    KB-208E9.1  3.247406</code></pre>
</div>
<div id="kidney-cortex" class="section level3">
<h3>Kidney Cortex</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>        SNP        gene 
0.000379666 0.004579254 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>     SNP     gene 
0.681686 2.502479 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
12.06127 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>       SNP       gene 
0.72186587 0.03975231 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7616182</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.94780546 0.05219454 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-19-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
231596    15 ENSG00000137819.13  69299268 gene          15          32        0
36106      2  ENSG00000115145.9 152176276 gene           2          91        0
221967    14 ENSG00000073712.14  52952836 gene          14          21        0
91273      5 ENSG00000177058.11  55773348 gene           5          32        0
117508     6 ENSG00000178425.13 116099964 gene           6          77        0
260478    20  ENSG00000275491.1   3807832 gene          20           3        0
       susie_pip      mu2  genename         z
231596 0.3563809 29.67982     PAQR5 -4.277398
36106  0.2477434 26.97233     STAM2  3.836622
221967 0.1606260 22.27244    FERMT2 -2.990795
91273  0.1594552 22.35682   SLC38A9 -3.307919
117508 0.1553302 22.18781    NT5DC1 -3.118395
260478 0.1502463 22.10096 LINC01730 -2.945771</code></pre>
</div>
<div id="liver" class="section level3">
<h3>Liver</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003814401 0.0041596106 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.7027292 1.2651829 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
10.90502 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>      SNP      gene 
0.7476266 0.0231917 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7708182</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.96991289 0.03008711 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-23-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
232268    15 ENSG00000137819.13  69299268 gene          15          32        0
95276      5  ENSG00000272021.1  95849392 gene           5          56        0
270005    21 ENSG00000160201.11  43031887 gene          21          21        0
176557    10 ENSG00000176273.14  93893644 gene          10          60        0
191545    11  ENSG00000137673.8 102530930 gene          11          59        0
103734     6 ENSG00000112699.10   2236692 gene           6           2        0
       susie_pip      mu2   genename         z
232268 0.2710718 20.70799      PAQR5 -4.277398
95276  0.1672517 17.25707 AC008592.8 -3.348117
270005 0.1395812 16.08112      U2AF1 -3.187321
176557 0.1365257 16.35490    SLC35G1 -3.538506
191545 0.1237505 17.10431       MMP7  3.217326
103734 0.1186136 15.80775       GMDS -3.273856</code></pre>
</div>
<div id="lung" class="section level3">
<h3>Lung</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003797659 0.0033581097 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.6790112 0.5315708 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
8.842579 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>       SNP       gene 
0.71922254 0.01041293 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7296355</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.98572859 0.01427141 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-27-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
233304    15 ENSG00000137819.13  69299268 gene          15          32        0
151315     8  ENSG00000164935.6 104341570 gene           8          71        0
271364    21 ENSG00000160201.11  43031887 gene          21          21        0
135028     7 ENSG00000160963.13 101362454 gene           7          62        0
206165    12 ENSG00000136048.13 101902056 gene          12          61        0
192366    11  ENSG00000137673.8 102530930 gene          11          59        0
        susie_pip      mu2 genename         z
233304 0.11355832 9.606008    PAQR5 -4.277398
151315 0.08171085 8.866802  DCSTAMP  3.944082
271364 0.06675872 7.344053    U2AF1 -3.187321
135028 0.06281241 7.414368  COL26A1 -3.615133
206165 0.05941452 7.182225    DRAM1  3.223839
192366 0.05497479 7.729388     MMP7  3.217326</code></pre>
</div>
<div id="pancreas" class="section level3">
<h3>Pancreas</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003791387 0.0039006092 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.6969396 0.4391335 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
10.28808 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>        SNP        gene 
0.736993487 0.008557348 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7455508</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.98852211 0.01147789 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-31-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id      pos type region_tag1 region_tag2 cs_index
90503      5 ENSG00000145592.13 40793056 gene           5          27        0
261658    20 ENSG00000125818.17  1106481 gene          20           2        0
49270      3 ENSG00000114812.12 42490447 gene           3          30        0
247840    17  ENSG00000226101.1 72071669 gene          17          41        0
270029    21 ENSG00000157557.11 38806138 gene          21          18        0
222953    14 ENSG00000073712.14 52952836 gene          14          21        0
        susie_pip      mu2  genename         z
90503  0.08901000 7.071124     RPL37  3.953186
261658 0.07566466 6.306397     PSMF1 -3.324110
49270  0.07178122 6.091420     VIPR1  3.378381
247840 0.06966362 5.895581 LINC02097  3.190044
270029 0.06813460 5.936244      ETS2 -3.200392
222953 0.06115363 5.446101    FERMT2 -2.990795</code></pre>
</div>
<div id="whole_blood" class="section level3">
<h3>Whole_Blood</h3>
<pre class="r"><code>#estimated prior inclusion probability
ctwas_parameters$group_prior</code></pre>
<pre><code>         SNP         gene 
0.0003797145 0.0041089502 </code></pre>
<pre class="r"><code>#estimated prior effect size
ctwas_parameters$group_prior_var</code></pre>
<pre><code>      SNP      gene 
0.6782780 0.4515726 </code></pre>
<pre class="r"><code>#estimated enrichment of genes over variants
ctwas_parameters$enrichment</code></pre>
<pre><code>    gene 
10.82116 </code></pre>
<pre class="r"><code>#PVE explained by genes and variants
ctwas_parameters$group_pve</code></pre>
<pre><code>        SNP        gene 
0.718348783 0.009142991 </code></pre>
<pre class="r"><code>#total heritability (sum of PVE)
ctwas_parameters$total_pve</code></pre>
<pre><code>[1] 0.7274918</code></pre>
<pre class="r"><code>#attributable heritability
ctwas_parameters$attributable_pve</code></pre>
<pre><code>       SNP       gene 
0.98743217 0.01256783 </code></pre>
<pre class="r"><code>#plot convergence
ctwas_parameters$convergence_plot</code></pre>
<p><img src="figure/OxHB.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-35-1">
Past versions of unnamed-chunk-35-1.png
</button>
</p>
<div id="fig-unnamed-chunk-35-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6a15ad1385ddbd5f0eff3bf96fba289afc4f929d/docs/figure/OxHB.Rmd/unnamed-chunk-35-1.png" target="_blank">6a15ad1</a>
</td>
<td>
sq-96
</td>
<td>
2023-10-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(ctwas_res[ctwas_res$type==&quot;gene&quot;,])</code></pre>
<pre><code>       chrom                 id       pos type region_tag1 region_tag2 cs_index
177030    10 ENSG00000119943.12  98387340 gene          10          62        0
95379      5  ENSG00000236882.7  95851444 gene           5          56        0
95380      5  ENSG00000272021.1  95849392 gene           5          56        0
194215    11 ENSG00000198331.10 125883330 gene          11          77        0
245150    17  ENSG00000126368.5  40092101 gene          17          23        0
100768     5 ENSG00000070814.17 150356800 gene           5          88        0
        susie_pip      mu2   genename         z
177030 0.08194366 6.590506    PYROXD2 -3.420306
95379  0.07885068 6.485879  LINC01554 -3.375656
95380  0.07679971 6.401168 AC008592.8 -3.348117
194215 0.07624593 6.758820      HYLS1  4.095459
245150 0.07585300 6.116364      NR1D1  3.146562
100768 0.07280063 6.239463      TCOF1  3.426022</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cowplot_1.1.1   ggplot2_3.4.0   ctwas_0.1.40    workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9        lattice_0.20-44   getPass_0.2-2     ps_1.7.2         
 [5] assertthat_0.2.1  rprojroot_2.0.3   digest_0.6.31     foreach_1.5.2    
 [9] utf8_1.2.2        R6_2.5.1          RSQLite_2.2.19    evaluate_0.19    
[13] httr_1.4.4        highr_0.9         pillar_1.8.1      rlang_1.1.1      
[17] rstudioapi_0.14   data.table_1.14.6 blob_1.2.3        whisker_0.4.1    
[21] callr_3.7.3       jquerylib_0.1.4   Matrix_1.3-3      rmarkdown_2.19   
[25] labeling_0.4.2    stringr_1.5.0     bit_4.0.5         munsell_0.5.0    
[29] compiler_4.1.0    httpuv_1.6.7      xfun_0.35         pkgconfig_2.0.3  
[33] htmltools_0.5.4   tidyselect_1.2.0  tibble_3.1.8      logging_0.10-108 
[37] codetools_0.2-18  fansi_1.0.3       dplyr_1.0.10      withr_2.5.0      
[41] later_1.3.0       grid_4.1.0        jsonlite_1.8.4    gtable_0.3.1     
[45] lifecycle_1.0.3   DBI_1.1.3         git2r_0.30.1      magrittr_2.0.3   
[49] scales_1.2.1      cli_3.6.1         stringi_1.7.8     cachem_1.0.6     
[53] farver_2.1.0      fs_1.5.2          promises_1.2.0.1  pgenlibr_0.3.2   
[57] bslib_0.4.1       vctrs_0.6.3       generics_0.1.3    iterators_1.0.14 
[61] tools_4.1.0       bit64_4.0.5       glue_1.6.2        processx_3.8.0   
[65] fastmap_1.1.0     yaml_2.3.6        colorspace_2.0-3  memoise_2.0.1    
[69] knitr_1.41        sass_0.4.4       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
