<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>SCZ 2018 - Brain_Amygdala</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SCZ 2018 - Brain_Amygdala</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-19
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreebcaadf34e5a5412397e69dcade41613a634040a5targetblankbcaadf3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/bcaadf34e5a5412397e69dcade41613a634040a5" target="_blank">bcaadf3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreebcaadf34e5a5412397e69dcade41613a634040a5targetblankbcaadf3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/bcaadf34e5a5412397e69dcade41613a634040a5" target="_blank">bcaadf3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  G_list.RData
    Untracked:  Rplot.png
    Untracked:  SCZ_annotation.xlsx
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/Autism_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/SCZ_2014_EUR_out/
    Untracked:  code/SCZ_2018_S_out/
    Untracked:  code/SCZ_2018_out/
    Untracked:  code/SCZ_2020_Single_out/
    Untracked:  code/SCZ_2020_out/
    Untracked:  code/SCZ_S_out/
    Untracked:  code/SCZ_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/process_scz_2018_snps.R
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_Autism_analysis.sbatch
    Untracked:  code/run_Autism_analysis.sh
    Untracked:  code/run_Autism_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_SCZ_2014_EUR_analysis.sbatch
    Untracked:  code/run_SCZ_2014_EUR_analysis.sh
    Untracked:  code/run_SCZ_2014_EUR_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_2018_analysis.sbatch
    Untracked:  code/run_SCZ_2018_analysis.sh
    Untracked:  code/run_SCZ_2018_analysis_S.sbatch
    Untracked:  code/run_SCZ_2018_analysis_S.sh
    Untracked:  code/run_SCZ_2018_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_2018_ctwas_rss_LDR_S.R
    Untracked:  code/run_SCZ_2020_Single_analysis.sbatch
    Untracked:  code/run_SCZ_2020_Single_analysis.sh
    Untracked:  code/run_SCZ_2020_Single_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_2020_analysis.sbatch
    Untracked:  code/run_SCZ_2020_analysis.sh
    Untracked:  code/run_SCZ_2020_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_analysis.sbatch
    Untracked:  code/run_SCZ_analysis.sh
    Untracked:  code/run_SCZ_analysis_S.sbatch
    Untracked:  code/run_SCZ_analysis_S.sh
    Untracked:  code/run_SCZ_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  code/wflow_build.R
    Untracked:  code/wflow_build.sbatch
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/GO_Terms/
    Untracked:  data/PGC3_SCZ_wave3_public.v2.tsv
    Untracked:  data/SCZ/
    Untracked:  data/SCZ_2014_EUR/
    Untracked:  data/SCZ_2018/
    Untracked:  data/SCZ_2018_S/
    Untracked:  data/SCZ_2020/
    Untracked:  data/SCZ_S/
    Untracked:  data/Supplementary Table 15 - MAGMA.xlsx
    Untracked:  data/Supplementary Table 20 - Prioritised Genes.xlsx
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/scz_2018.RDS
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt
    Untracked:  top_genes_32.txt
    Untracked:  top_genes_37.txt
    Untracked:  top_genes_43.txt
    Untracked:  top_genes_54.txt
    Untracked:  top_genes_81.txt
    Untracked:  z_snp_pos_SCZ.RData
    Untracked:  z_snp_pos_SCZ_2014_EUR.RData
    Untracked:  z_snp_pos_SCZ_2018.RData
    Untracked:  z_snp_pos_SCZ_2020.RData

Unstaged changes:
    Deleted:    analysis/BMI_S_results.Rmd
    Modified:   analysis/SCZ_2018_Brain_Amygdala_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Caudate_basal_ganglia_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Cerebellar_Hemisphere_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Cerebellum_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Cortex_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Frontal_Cortex_BA9_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Hippocampus_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Hypothalamus_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Putamen_basal_ganglia_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Modified:   analysis/SCZ_2018_Brain_Substantia_nigra_S.Rmd
    Modified:   analysis/ttt.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SCZ_2018_Brain_Amygdala_S.Rmd</code>) and HTML (<code>docs/SCZ_2018_Brain_Amygdala_S.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bcaadf34e5a5412397e69dcade41613a634040a5/analysis/SCZ_2018_Brain_Amygdala_S.Rmd" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/bcaadf34e5a5412397e69dcade41613a634040a5/docs/SCZ_2018_Brain_Amygdala_S.html" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/analysis/SCZ_2018_Brain_Amygdala_S.Rmd" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/SCZ_2018_Brain_Amygdala_S.html" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/7d08c9ba1c682ee3e1fbb5c8dea3fc6dace6d9c5/analysis/SCZ_2018_Brain_Amygdala_S.Rmd" target="_blank">7d08c9b</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/7d08c9ba1c682ee3e1fbb5c8dea3fc6dace6d9c5/docs/SCZ_2018_Brain_Amygdala_S.html" target="_blank">7d08c9b</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-18
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/analysis/SCZ_2018_Brain_Amygdala_S.Rmd" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/SCZ_2018_Brain_Amygdala_S.html" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/011327d95727f257c7807d62218d5256ace3f7ac/docs/SCZ_2018_Brain_Amygdala_S.html" target="_blank">011327d</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6c6abbde38c4b39636b2eb5c8f5679b8dc8f3768/analysis/SCZ_2018_Brain_Amygdala_S.Rmd" target="_blank">6c6abbd</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(reticulate)
use_python(&quot;/scratch/midway2/shengqian/miniconda3/envs/PythonForR/bin/python&quot;,required=T)</code></pre>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 15774</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1480 1078  891  631  654  819  931  556  642  735  948  858  322  583  532  618 
  17   18   19   20   21   22 
1087  198 1146  543   34  488 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 14087</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.8931</code></pre>
<pre><code>INFO:numexpr.utils:Note: NumExpr detected 56 cores but &quot;NUMEXPR_MAX_THREADS&quot; not set, so enforcing safe limit of 8.</code></pre>
<pre><code>finish</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-12-1.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>     gene       snp 
0.0048321 0.0003214 </code></pre>
<pre><code> gene   snp 
 9.84 10.49 </code></pre>
<pre><code>[1] 105318</code></pre>
<pre><code>[1]    6393 6309950</code></pre>
<pre><code>    gene      snp 
0.002886 0.202078 </code></pre>
<pre><code>[1] 0.007093 1.090103</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bcaadf34e5a5412397e69dcade41613a634040a5/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-14-1.png" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-14-1.png" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-14-1.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>      genename region_tag susie_pip   mu2       PVE      z num_intron num_sqtl
2889 LINC00320       21_6    1.0610 28.32 2.926e-04  5.336          6        6
2984      LRP8       1_33    0.9257 24.92 1.962e-04 -4.820          3        3
371     APOPT1      14_54    0.8221 43.94 2.808e-04 -7.429          5        6
6193   ZDHHC20       13_2    0.7562 24.63 1.313e-04 -4.784          3        4
2836     LAMA5      20_37    0.7084 30.04 1.381e-04 -4.371         10       14
1662    DPYSL3       5_86    0.7025 25.41 1.190e-04 -4.157          1        1
611       BDNF      11_19    0.6948 23.25 1.066e-04 -4.348          1        1
4020     PLCB2      15_14    0.6903 25.17 9.585e-05 -4.470          5        5
1554      DGKZ      11_28    0.6889 46.65 2.102e-04  7.216          2        2
552     B3GAT1      11_84    0.6636 25.37 9.490e-05  4.345          7       11
4326   PYROXD2      10_62    0.6365 24.28 8.561e-05  3.755         10       10
122     ACTR1B       2_57    0.6179 21.59 7.826e-05  3.978          3        3
1482     DBF4B      17_26    0.5949 20.61 6.733e-05 -3.890          4        4
3695     NTRK3      15_41    0.5946 22.72 7.628e-05 -4.457          1        1
5579     TMED4       7_32    0.5724 22.25 6.781e-05 -4.862          3        3
781    C2orf80      2_123    0.5393 25.65 5.402e-05 -3.011         10       12
235       AKT3      1_128    0.5277 34.14 8.384e-05  6.266          5        5
6256    ZNF211      19_39    0.5206 22.95 5.737e-05 -3.624          4        5
510     ATP2B2        3_8    0.4886 31.83 6.736e-05  4.229          3        3
1213     CNIH3      1_114    0.4872 21.03 4.196e-05 -3.852          7        9</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>      genename region_tag susie_pip   mu2       PVE      z num_intron num_sqtl
2889 LINC00320       21_6    1.0610 28.32 2.926e-04  5.336          6        6
371     APOPT1      14_54    0.8221 43.94 2.808e-04 -7.429          5        6
1554      DGKZ      11_28    0.6889 46.65 2.102e-04  7.216          2        2
2984      LRP8       1_33    0.9257 24.92 1.962e-04 -4.820          3        3
2836     LAMA5      20_37    0.7084 30.04 1.381e-04 -4.371         10       14
6193   ZDHHC20       13_2    0.7562 24.63 1.313e-04 -4.784          3        4
1662    DPYSL3       5_86    0.7025 25.41 1.190e-04 -4.157          1        1
611       BDNF      11_19    0.6948 23.25 1.066e-04 -4.348          1        1
4020     PLCB2      15_14    0.6903 25.17 9.585e-05 -4.470          5        5
552     B3GAT1      11_84    0.6636 25.37 9.490e-05  4.345          7       11
4326   PYROXD2      10_62    0.6365 24.28 8.561e-05  3.755         10       10
3686     NT5C2      10_66    0.4472 46.77 8.490e-05 -8.511          7        9
235       AKT3      1_128    0.5277 34.14 8.384e-05  6.266          5        5
4906     SF3B1      2_117    0.4467 43.83 8.117e-05  7.002          2        2
122     ACTR1B       2_57    0.6179 21.59 7.826e-05  3.978          3        3
3695     NTRK3      15_41    0.5946 22.72 7.628e-05 -4.457          1        1
5579     TMED4       7_32    0.5724 22.25 6.781e-05 -4.862          3        3
510     ATP2B2        3_8    0.4886 31.83 6.736e-05  4.229          3        3
1482     DBF4B      17_26    0.5949 20.61 6.733e-05 -3.890          4        4
3636    NPEPL1      20_34    0.4865 34.73 6.584e-05  3.996         13       15</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-16-1.png" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-16-1.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-2">
Past versions of unnamed-chunk-16-2.png
</button>
</p>
<div id="fig-unnamed-chunk-16-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bcaadf34e5a5412397e69dcade41613a634040a5/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-16-2.png" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-16-2.png" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-16-2.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.01611</code></pre>
<pre><code>         genename region_tag susie_pip    mu2       PVE       z num_intron
3933        PGBD1       6_22 1.867e-02 157.22 2.841e-07 -13.087          2
370          APOM       6_26 2.029e-01 123.81 4.840e-05  11.541          2
1506         DDR1       6_26 1.524e-02 119.73 2.641e-07 -11.175          2
480         ATAT1       6_24 2.265e-02  79.41 3.868e-07  11.039          1
795      C6orf136       6_24 4.367e-02  79.12 1.433e-06 -11.031          2
2082        FLOT1       6_24 1.053e-01  77.80 8.177e-06 -10.981          6
568          BAG6       6_26 5.880e-04 108.10 3.549e-10 -10.247          4
4565         RNF5       6_26 6.016e-05  96.37 3.312e-12  -9.714          1
968        CCHCR1       6_26 8.126e-10  89.72 5.578e-22  -9.376          8
3686        NT5C2      10_66 4.472e-01  46.77 8.490e-05  -8.511          7
3043       MAD1L1        7_3 2.348e-01  63.35 2.501e-05  -8.215          3
2468        HLA-F       6_23 3.791e-02  61.03 6.413e-07  -8.066          2
2209       GIGYF2      2_137 3.745e-01  50.80 5.226e-05  -7.841          4
4685 RP5-874C20.8       6_22 2.206e-02  37.39 1.296e-07   7.631          4
6384      ZSCAN16       6_22 1.933e-02  52.88 1.201e-07   7.468          3
371        APOPT1      14_54 8.221e-01  43.94 2.808e-04  -7.429          5
1554         DGKZ      11_28 6.889e-01  46.65 2.102e-04   7.216          2
4968       SKIV2L       6_26 2.813e-08  77.27 5.804e-19   7.101          4
4906        SF3B1      2_117 4.467e-01  43.83 8.117e-05   7.002          2
485         ATG13      11_28 1.646e-01  43.02 1.107e-05   6.977          2
     num_sqtl
3933        3
370         2
1506        2
480         1
795         2
2082        6
568         6
4565        1
968        12
3686        9
3043        3
2468        3
2209        4
4685        4
6384        3
371         6
1554        2
4968        5
4906        2
485         2</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 18</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bcaadf34e5a5412397e69dcade41613a634040a5/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-1.png" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-1.png" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-1.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                                                  Term Overlap
1    positive regulation of neuron projection development (GO:0010976)    4/88
2         positive regulation of phospholipase C activity (GO:0010863)    3/43
3            modulation of chemical synaptic transmission (GO:0050804)   3/109
4     positive regulation of cell projection organization (GO:0031346)   3/117
5       cellular response to nerve growth factor stimulus (GO:1990090)    2/22
6                  activation of phospholipase C activity (GO:0007202)    2/32
7             regulation of neuron projection development (GO:0010975)   3/165
8                  regulation of trans-synaptic signaling (GO:0099177)    2/35
9 positive regulation of protein tyrosine kinase activity (GO:0061098)    2/42
  Adjusted.P.value                  Genes
1        0.0002400 BDNF;NTRK3;DPYSL3;LRP8
2        0.0008678       BDNF;NTRK3;PLCB2
3        0.0082176         BDNF;DGKZ;LRP8
4        0.0082176      BDNF;NTRK3;DPYSL3
5        0.0082176             BDNF;NTRK3
6        0.0131378             BDNF;PLCB2
7        0.0131378      BDNF;NTRK3;DPYSL3
8        0.0131378              BDNF;LRP8
9        0.0168359              BDNF;LRP8
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-2">
Past versions of unnamed-chunk-19-2.png
</button>
</p>
<div id="fig-unnamed-chunk-19-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bcaadf34e5a5412397e69dcade41613a634040a5/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-2.png" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-2.png" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-2.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-3">
Past versions of unnamed-chunk-19-3.png
</button>
</p>
<div id="fig-unnamed-chunk-19-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bcaadf34e5a5412397e69dcade41613a634040a5/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-3.png" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-3.png" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-19-3.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                                           Description     FDR
49                                                  Status Epilepticus 0.01491
53                                                 Unipolar Depression 0.01491
75                                                    Petit mal status 0.01491
82                                        Grand Mal Status Epilepticus 0.01491
87                                  Complex Partial Status Epilepticus 0.01491
119                                    Status Epilepticus, Subclinical 0.01491
120                                  Non-Convulsive Status Epilepticus 0.01491
121                                  Simple Partial Status Epilepticus 0.01491
124                                          Major Depressive Disorder 0.01491
142 MEGALENCEPHALY-POLYMICROGYRIA-POLYDACTYLY-HYDROCEPHALUS SYNDROME 2 0.01491
    Ratio  BgRatio
49    2/7  68/9703
53    3/7 259/9703
75    2/7  67/9703
82    2/7  67/9703
87    2/7  67/9703
119   2/7  67/9703
120   2/7  67/9703
121   2/7  67/9703
124   3/7 243/9703
142   1/7   1/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Warning: replacing previous import &#39;lifecycle::last_warnings&#39; by
&#39;rlang::last_warnings&#39; when loading &#39;hms&#39;</code></pre>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<p><img src="figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/bcaadf34e5a5412397e69dcade41613a634040a5/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-22-1.png" target="_blank">bcaadf3</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/be614ede37e6ed9f95b0d359f083e32f53defaad/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-22-1.png" target="_blank">be614ed</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/2749be95e34fe08835249dbb967fdd153d69cfa4/docs/figure/SCZ_2018_Brain_Amygdala_S.Rmd/unnamed-chunk-22-1.png" target="_blank">2749be9</a>
</td>
<td>
sq-96
</td>
<td>
2022-05-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 130</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 49</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.47</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 3</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 103</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>[1] genename   region_tag susie_pip  mu2        PVE        z          num_intron
[8] num_sqtl  
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>   ctwas     TWAS 
0.007692 0.100000 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9997 0.9858 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code> ctwas   TWAS 
0.3333 0.1262 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.4.0      forcats_0.5.1     stringr_1.4.0     purrr_0.3.4      
 [5] readr_1.4.0       tidyr_1.1.3       tidyverse_1.3.1   tibble_3.1.7     
 [9] WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0       cowplot_1.1.1    
[13] ggplot2_3.3.5     dplyr_1.0.7       reticulate_1.25   workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] fs_1.5.0          lubridate_1.7.10  doParallel_1.0.16 httr_1.4.2       
 [5] rprojroot_2.0.2   tools_4.1.0       backports_1.2.1   doRNG_1.8.2      
 [9] bslib_0.2.5.1     utf8_1.2.1        R6_2.5.0          vipor_0.4.5      
[13] DBI_1.1.1         colorspace_2.0-2  withr_2.4.2       ggrastr_1.0.1    
[17] tidyselect_1.1.1  processx_3.5.2    curl_4.3.2        compiler_4.1.0   
[21] git2r_0.28.0      rvest_1.0.0       cli_3.0.0         Cairo_1.5-15     
[25] xml2_1.3.2        labeling_0.4.2    sass_0.4.0        scales_1.1.1     
[29] callr_3.7.0       systemfonts_1.0.4 apcluster_1.4.9   digest_0.6.27    
[33] rmarkdown_2.9     svglite_2.0.0     pkgconfig_2.0.3   htmltools_0.5.1.1
[37] dbplyr_2.1.1      highr_0.9         rlang_1.0.2       rstudioapi_0.13  
[41] jquerylib_0.1.4   farver_2.1.0      generics_0.1.0    jsonlite_1.7.2   
[45] magrittr_2.0.1    Matrix_1.3-3      ggbeeswarm_0.6.0  Rcpp_1.0.7       
[49] munsell_0.5.0     fansi_0.5.0       lifecycle_1.0.0   stringi_1.6.2    
[53] whisker_0.4       yaml_2.2.1        plyr_1.8.6        grid_4.1.0       
[57] ggrepel_0.9.1     parallel_4.1.0    promises_1.2.0.1  crayon_1.4.1     
[61] lattice_0.20-44   haven_2.4.1       hms_1.1.0         knitr_1.33       
[65] ps_1.6.0          pillar_1.7.0      igraph_1.2.6      rjson_0.2.20     
[69] rngtools_1.5      reshape2_1.4.4    codetools_0.2-18  reprex_2.0.0     
[73] glue_1.4.2        evaluate_0.14     getPass_0.2-2     modelr_0.1.8     
[77] data.table_1.14.0 png_0.1-7         vctrs_0.3.8       httpuv_1.6.1     
[81] foreach_1.5.1     cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[85] xfun_0.24         broom_0.7.8       later_1.2.0       iterators_1.0.13 
[89] beeswarm_0.4.0    ellipsis_0.3.2    here_1.0.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
