<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>Autism - Brain Amygdala</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Autism - Brain Amygdala</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree3dd5b4c06fd9ac2be9b0a0519067bba809374ee8targetblank3dd5b4ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/3dd5b4c06fd9ac2be9b0a0519067bba809374ee8" target="_blank">3dd5b4c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree3dd5b4c06fd9ac2be9b0a0519067bba809374ee8targetblank3dd5b4ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/3dd5b4c06fd9ac2be9b0a0519067bba809374ee8" target="_blank">3dd5b4c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/figure/
    Ignored:    data/AF/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Autism_Brain_Amygdala.Rmd
    Untracked:  analysis/Autism_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Untracked:  analysis/Autism_Brain_Caudate_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Cerebellar_Hemisphere.Rmd
    Untracked:  analysis/Autism_Brain_Cerebellum.Rmd
    Untracked:  analysis/Autism_Brain_Cortex.Rmd
    Untracked:  analysis/Autism_Brain_Frontal_Cortex_BA9.Rmd
    Untracked:  analysis/Autism_Brain_Hippocampus.Rmd
    Untracked:  analysis/Autism_Brain_Hypothalamus.Rmd
    Untracked:  analysis/Autism_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Putamen_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Spinal_cord_cervical_c-1.Rmd
    Untracked:  analysis/Autism_Brain_Substantia_nigra.Rmd
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/Autism_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/SCZ_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_Autism_analysis.sbatch
    Untracked:  code/run_Autism_analysis.sh
    Untracked:  code/run_Autism_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_SCZ_analysis.sbatch
    Untracked:  code/run_SCZ_analysis.sh
    Untracked:  code/run_SCZ_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/Autism/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/SCZ/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Amygdala_S.Rmd
    Modified:   analysis/BMI_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Modified:   analysis/BMI_Brain_Caudate_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellar_Hemisphere_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellum_S.Rmd
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/BMI_Brain_Cortex_S.Rmd
    Modified:   analysis/BMI_Brain_Frontal_Cortex_BA9_S.Rmd
    Modified:   analysis/BMI_Brain_Hippocampus_S.Rmd
    Modified:   analysis/BMI_Brain_Hypothalamus_S.Rmd
    Modified:   analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Putamen_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Modified:   analysis/BMI_Brain_Substantia_nigra_S.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SCZ_Brain_Amygdala.Rmd</code>) and HTML (<code>docs/SCZ_Brain_Amygdala.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3dd5b4c06fd9ac2be9b0a0519067bba809374ee8/analysis/SCZ_Brain_Amygdala.Rmd" target="_blank">3dd5b4c</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 10292</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1005  740  588  392  518  542  481  374  407  397  632  597  222  335  340  460 
  17   18   19   20   21   22 
 598  163  797  309  120  275 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 8408</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.8169</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0072071 0.0002653 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code>  gene    snp 
14.431  8.448 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 82315</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   10292 7573890</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>  gene    snp 
0.0130 0.2062 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.04663 1.55649</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>          genename region_tag susie_pip    mu2       PVE      z num_eqtl
10169       ZNF823      19_10    0.9717  30.47 0.0003597  5.455        1
11179    HIST1H2BN       6_21    0.8621 100.93 0.0010571 10.773        1
11222   AC012074.2       2_15    0.8529  23.33 0.0002418  4.648        2
2362        B3GAT1      11_84    0.7950  23.70 0.0002289 -4.459        2
242          VSIG2      11_77    0.7863  26.81 0.0002560 -3.818        1
5997        TMEM56       1_58    0.7396  21.89 0.0001967 -3.918        1
3251        CYSTM1       5_83    0.7299  23.06 0.0002045 -4.025        1
10971    LINC00390      13_17    0.6968  22.04 0.0001866 -4.220        1
5798       ARFGAP2      11_29    0.6909  25.35 0.0002128  4.740        1
5920         PLBD2      12_68    0.6614  22.50 0.0001808  3.986        1
3172         HSDL2       9_57    0.6477  25.20 0.0001983 -4.322        1
10503        ZFP57       6_23    0.6316  73.92 0.0005672  7.267        1
10520        TCTN1      12_67    0.6306  24.15 0.0001850  4.840        1
5344          RIT1       1_76    0.5471  23.38 0.0001554 -3.496        1
377         CTNNA1       5_82    0.5417  25.60 0.0001685  5.064        1
9014        FAM83H       8_94    0.5412  24.57 0.0001616  4.317        2
677        PPP2R5B      11_36    0.5310  25.56 0.0001649 -4.623        1
11888 RP11-220I1.5       9_28    0.5132  23.80 0.0001484 -4.450        1
9316         ACOT1      14_34    0.4857  28.81 0.0001700  3.967        2
2406           MDK      11_28    0.4793  39.90 0.0002323 -6.357        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>           genename region_tag susie_pip    mu2       PVE       z num_eqtl
8899       HLA-DQB1       6_26 5.651e-14 882.98 6.062e-16  4.2352        1
9980       HLA-DQA1       6_26 7.838e-14 755.62 7.195e-16  4.0876        1
9870       HLA-DRB1       6_26 8.471e-14 504.68 5.194e-16  4.3158        1
10465          MSH5       6_26 8.843e-13 381.27 4.096e-15  8.8122        1
9057          ACBD4      17_27 0.000e+00 196.81 0.000e+00  1.1059        1
10458       SLC44A4       6_26 5.934e-12 170.69 1.230e-14  6.2502        1
9377          FMNL1      17_27 0.000e+00 123.32 0.000e+00 -0.6638        1
11179     HIST1H2BN       6_21 8.621e-01 100.93 1.057e-03 10.7729        1
10706         CLIC1       6_26 2.934e-13  86.02 3.066e-16  0.4634        1
1218           PUS7       7_65 0.000e+00  77.22 0.000e+00 -3.2022        1
10503         ZFP57       6_23 6.316e-01  73.92 5.672e-04  7.2673        1
9573         BTN3A2       6_20 1.147e-02  68.67 9.569e-06  9.0494        3
2240          GOSR2      17_27 0.000e+00  68.45 0.000e+00 -2.5096        1
4578           NMT1      17_27 0.000e+00  66.09 0.000e+00  2.2782        1
8918          RPRML      17_27 0.000e+00  63.52 0.000e+00  1.5727        1
8984      HIST1H2BC       6_20 1.263e-02  55.02 8.443e-06 -8.0277        1
437        MPHOSPH9      12_75 1.291e-01  48.60 7.624e-05  7.1580        1
4868          PRDM5       4_78 0.000e+00  45.06 0.000e+00  0.3272        2
6704          LEMD2       6_28 1.496e-01  42.65 7.752e-05  4.2792        2
12183 RP11-247A12.7       9_66 3.131e-01  41.87 1.593e-04  4.3022        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>        genename region_tag susie_pip    mu2       PVE      z num_eqtl
11179  HIST1H2BN       6_21    0.8621 100.93 0.0010571 10.773        1
10503      ZFP57       6_23    0.6316  73.92 0.0005672  7.267        1
10169     ZNF823      19_10    0.9717  30.47 0.0003597  5.455        1
242        VSIG2      11_77    0.7863  26.81 0.0002560 -3.818        1
11222 AC012074.2       2_15    0.8529  23.33 0.0002418  4.648        2
2406         MDK      11_28    0.4793  39.90 0.0002323 -6.357        1
2362      B3GAT1      11_84    0.7950  23.70 0.0002289 -4.459        2
5798     ARFGAP2      11_29    0.6909  25.35 0.0002128  4.740        1
3251      CYSTM1       5_83    0.7299  23.06 0.0002045 -4.025        1
3172       HSDL2       9_57    0.6477  25.20 0.0001983 -4.322        1
5997      TMEM56       1_58    0.7396  21.89 0.0001967 -3.918        1
10971  LINC00390      13_17    0.6968  22.04 0.0001866 -4.220        1
10520      TCTN1      12_67    0.6306  24.15 0.0001850  4.840        1
5920       PLBD2      12_68    0.6614  22.50 0.0001808  3.986        1
413       ARID1B      6_102    0.3600  39.03 0.0001707  3.907        1
9316       ACOT1      14_34    0.4857  28.81 0.0001700  3.967        2
5866       TAOK2      16_24    0.3554  39.04 0.0001685  6.189        1
377       CTNNA1       5_82    0.5417  25.60 0.0001685  5.064        1
677      PPP2R5B      11_36    0.5310  25.56 0.0001649 -4.623        1
9014      FAM83H       8_94    0.5412  24.57 0.0001616  4.317        2</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>       genename region_tag susie_pip    mu2       PVE      z num_eqtl
11179 HIST1H2BN       6_21 8.621e-01 100.93 1.057e-03 10.773        1
9573     BTN3A2       6_20 1.147e-02  68.67 9.569e-06  9.049        3
10465      MSH5       6_26 8.843e-13 381.27 4.096e-15  8.812        1
8984  HIST1H2BC       6_20 1.263e-02  55.02 8.443e-06 -8.028        1
5778      CNNM2      10_66 1.144e-01  38.59 5.363e-05 -7.691        1
10503     ZFP57       6_23 6.316e-01  73.92 5.672e-04  7.267        1
437    MPHOSPH9      12_75 1.291e-01  48.60 7.624e-05  7.158        1
5903      ABCB9      12_75 4.331e-03  40.53 2.133e-06  6.404        1
2406        MDK      11_28 4.793e-01  39.90 2.323e-04 -6.357        1
11226   ZSCAN31       6_22 9.442e-03  28.74 3.296e-06 -6.270        2
10458   SLC44A4       6_26 5.934e-12 170.69 1.230e-14  6.250        1
9777       DPYD       1_60 6.071e-03  37.20 2.744e-06 -6.222        1
5866      TAOK2      16_24 3.554e-01  39.04 1.685e-04  6.189        1
8969     HARBI1      11_28 1.658e-01  37.21 7.496e-05  6.169        1
10617   DNAJC19      3_111 2.294e-01  39.04 1.088e-04  6.158        1
2590     TRIM38       6_20 9.834e-03  30.60 3.656e-06  5.841        2
10231   ZKSCAN8       6_22 6.487e-03  37.40 2.948e-06  5.829        1
3264      SNX19      11_81 4.623e-02  37.69 2.117e-05  5.761        3
7375        CKB      14_54 9.933e-03  29.53 3.563e-06 -5.704        1
9987    ZSCAN16       6_22 7.220e-03  35.90 3.149e-06  5.677        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.005733</code></pre>
<pre><code>       genename region_tag susie_pip    mu2       PVE      z num_eqtl
11179 HIST1H2BN       6_21 8.621e-01 100.93 1.057e-03 10.773        1
9573     BTN3A2       6_20 1.147e-02  68.67 9.569e-06  9.049        3
10465      MSH5       6_26 8.843e-13 381.27 4.096e-15  8.812        1
8984  HIST1H2BC       6_20 1.263e-02  55.02 8.443e-06 -8.028        1
5778      CNNM2      10_66 1.144e-01  38.59 5.363e-05 -7.691        1
10503     ZFP57       6_23 6.316e-01  73.92 5.672e-04  7.267        1
437    MPHOSPH9      12_75 1.291e-01  48.60 7.624e-05  7.158        1
5903      ABCB9      12_75 4.331e-03  40.53 2.133e-06  6.404        1
2406        MDK      11_28 4.793e-01  39.90 2.323e-04 -6.357        1
11226   ZSCAN31       6_22 9.442e-03  28.74 3.296e-06 -6.270        2
10458   SLC44A4       6_26 5.934e-12 170.69 1.230e-14  6.250        1
9777       DPYD       1_60 6.071e-03  37.20 2.744e-06 -6.222        1
5866      TAOK2      16_24 3.554e-01  39.04 1.685e-04  6.189        1
8969     HARBI1      11_28 1.658e-01  37.21 7.496e-05  6.169        1
10617   DNAJC19      3_111 2.294e-01  39.04 1.088e-04  6.158        1
2590     TRIM38       6_20 9.834e-03  30.60 3.656e-06  5.841        2
10231   ZKSCAN8       6_22 6.487e-03  37.40 2.948e-06  5.829        1
3264      SNX19      11_81 4.623e-02  37.69 2.117e-05  5.761        3
7375        CKB      14_54 9.933e-03  29.53 3.563e-06 -5.704        1
9987    ZSCAN16       6_22 7.220e-03  35.90 3.149e-06  5.677        1</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 18</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                         Description     FDR Ratio BgRatio
45                               JOUBERT SYNDROME 13 0.02597   1/9  1/9703
50                                 NOONAN SYNDROME 8 0.02597   1/9  1/9703
36 Macular Dystrophy, Butterfly-Shaped Pigmentary, 2 0.03114   1/9  3/9703
38 Patterned dystrophy of retinal pigment epithelium 0.03114   1/9  3/9703
54     Butterfly-shaped pigmentary macular dystrophy 0.03114   1/9  3/9703
22                 Amelogenesis Imperfecta, Type III 0.03459   1/9  4/9703
35          Diabetes Mellitus, Transient Neonatal, 1 0.03704   1/9  5/9703
1                            Amelogenesis Imperfecta 0.06312   1/9 12/9703
2       Hereditary Nonpolyposis Colorectal Neoplasms 0.06312   1/9 26/9703
3                                  Diabetes Mellitus 0.06312   1/9 32/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 41</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.571</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 3</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 59</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>[1] genename   region_tag susie_pip  mu2        PVE        z          num_eqtl  
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9997 0.9943 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<p><img src="figure/SCZ_Brain_Amygdala.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7      
 [5] purrr_0.3.4       readr_2.1.1       tidyr_1.1.4       tidyverse_1.3.1  
 [9] tibble_3.1.6      WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0      
[13] cowplot_1.0.0     ggplot2_3.3.5     workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       doParallel_1.0.17
 [5] httr_1.4.2        rprojroot_2.0.2   tools_3.6.1       backports_1.4.1  
 [9] doRNG_1.8.2       utf8_1.2.2        R6_2.5.1          vipor_0.4.5      
[13] DBI_1.1.2         colorspace_2.0-2  withr_2.4.3       ggrastr_1.0.1    
[17] tidyselect_1.1.1  bit_4.0.4         curl_4.3.2        compiler_3.6.1   
[21] git2r_0.26.1      rvest_1.0.2       cli_3.1.0         Cairo_1.5-12.2   
[25] xml2_1.3.3        labeling_0.4.2    scales_1.1.1      apcluster_1.4.8  
[29] digest_0.6.29     rmarkdown_2.11    svglite_1.2.2     pkgconfig_2.0.3  
[33] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[37] rlang_1.0.1       rstudioapi_0.13   RSQLite_2.2.8     jquerylib_0.1.4  
[41] farver_2.1.0      generics_0.1.1    jsonlite_1.7.2    vroom_1.5.7      
[45] magrittr_2.0.2    Matrix_1.2-18     ggbeeswarm_0.6.0  Rcpp_1.0.8       
[49] munsell_0.5.0     fansi_1.0.2       gdtools_0.1.9     lifecycle_1.0.1  
[53] stringi_1.7.6     whisker_0.3-2     yaml_2.2.1        plyr_1.8.6       
[57] grid_3.6.1        blob_1.2.2        ggrepel_0.9.1     parallel_3.6.1   
[61] promises_1.0.1    crayon_1.5.0      lattice_0.20-38   haven_2.4.3      
[65] hms_1.1.1         knitr_1.36        pillar_1.6.4      igraph_1.2.10    
[69] rjson_0.2.20      rngtools_1.5.2    reshape2_1.4.4    codetools_0.2-16 
[73] reprex_2.0.1      glue_1.6.2        evaluate_0.14     data.table_1.14.2
[77] modelr_0.1.8      vctrs_0.3.8       tzdb_0.2.0        httpuv_1.5.1     
[81] foreach_1.5.2     cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[85] cachem_1.0.6      xfun_0.29         broom_0.7.10      later_0.8.0      
[89] iterators_1.0.14  beeswarm_0.2.3    memoise_2.0.1     ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
