<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>SCZ - Brain Frontal Cortex BA9</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SCZ - Brain Frontal Cortex BA9</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree4dd47fbc430a3f74482856839b723b4de2ca92dbtargetblank4dd47fba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/4dd47fbc430a3f74482856839b723b4de2ca92db" target="_blank">4dd47fb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree4dd47fbc430a3f74482856839b723b4de2ca92dbtargetblank4dd47fba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/4dd47fbc430a3f74482856839b723b4de2ca92db" target="_blank">4dd47fb</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    data/AF/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Autism_Brain_Amygdala.Rmd
    Untracked:  analysis/Autism_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Untracked:  analysis/Autism_Brain_Caudate_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Cerebellar_Hemisphere.Rmd
    Untracked:  analysis/Autism_Brain_Cerebellum.Rmd
    Untracked:  analysis/Autism_Brain_Cortex.Rmd
    Untracked:  analysis/Autism_Brain_Frontal_Cortex_BA9.Rmd
    Untracked:  analysis/Autism_Brain_Hippocampus.Rmd
    Untracked:  analysis/Autism_Brain_Hypothalamus.Rmd
    Untracked:  analysis/Autism_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Putamen_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Spinal_cord_cervical_c-1.Rmd
    Untracked:  analysis/Autism_Brain_Substantia_nigra.Rmd
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/Autism_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/SCZ_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_Autism_analysis.sbatch
    Untracked:  code/run_Autism_analysis.sh
    Untracked:  code/run_Autism_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_SCZ_analysis.sbatch
    Untracked:  code/run_SCZ_analysis.sh
    Untracked:  code/run_SCZ_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/Autism/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/SCZ/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Amygdala_S.Rmd
    Modified:   analysis/BMI_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Modified:   analysis/BMI_Brain_Caudate_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellar_Hemisphere_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellum_S.Rmd
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/BMI_Brain_Cortex_S.Rmd
    Modified:   analysis/BMI_Brain_Frontal_Cortex_BA9_S.Rmd
    Modified:   analysis/BMI_Brain_Hippocampus_S.Rmd
    Modified:   analysis/BMI_Brain_Hypothalamus_S.Rmd
    Modified:   analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Putamen_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Modified:   analysis/BMI_Brain_Substantia_nigra_S.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SCZ_Brain_Frontal_Cortex_BA9.Rmd</code>) and HTML (<code>docs/SCZ_Brain_Frontal_Cortex_BA9.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/4dd47fbc430a3f74482856839b723b4de2ca92db/analysis/SCZ_Brain_Frontal_Cortex_BA9.Rmd" target="_blank">4dd47fb</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-28
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/75a146615b51c4871f234c7d5cc78179e55ef29c/docs/SCZ_Brain_Frontal_Cortex_BA9.html" target="_blank">75a1466</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1c69dd24cb48d62e1c884f260adbee91383bbe92/analysis/SCZ_Brain_Frontal_Cortex_BA9.Rmd" target="_blank">1c69dd2</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/SCZ_Brain_Frontal_Cortex_BA9.html" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3dd5b4c06fd9ac2be9b0a0519067bba809374ee8/analysis/SCZ_Brain_Frontal_Cortex_BA9.Rmd" target="_blank">3dd5b4c</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 11507</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1142  839  648  452  570  593  533  445  426  459  695  674  240  377  369  524 
  17   18   19   20   21   22 
 705  179  882  347  120  288 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 9036</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.7853</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-5-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0101748 0.0002529 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
8.988 8.832 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 82315</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   11507 7573890</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.01278 0.20549 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.07922 1.52765</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-7-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>          genename region_tag susie_pip   mu2       PVE      z num_eqtl
13483 RP11-230C9.4      6_102    0.9873 24.01 0.0002879 -4.866        2
7629         THOC7       3_43    0.9813 34.05 0.0004059 -6.066        2
11134       ZNF823      19_10    0.9749 29.06 0.0003441  5.468        2
12304   AC012074.2       2_15    0.8746 21.96 0.0002333  4.623        1
10221        ACOT1      14_34    0.8412 22.58 0.0002308  4.284        3
9133       MAP3K11      11_36    0.8338 23.52 0.0002382 -4.544        1
108          ELAC2      17_11    0.8056 21.71 0.0002124  4.542        1
6584         TADA1       1_82    0.7488 23.41 0.0002130 -4.174        2
3758       BHLHE41      12_18    0.7421 22.88 0.0002063  4.024        1
6336       ARFGAP2      11_29    0.7316 23.92 0.0002126  4.740        1
6470         PLBD2      12_68    0.7284 20.64 0.0001827  3.986        1
9457        LPCAT4      15_10    0.7113 20.24 0.0001749 -4.205        2
14019        ERICD       8_92    0.7082 21.16 0.0001821 -4.157        1
6317         CNNM2      10_66    0.6998 48.44 0.0004117 -8.902        2
9024          FUT9       6_65    0.6687 29.04 0.0002360  5.427        1
12293   AC073283.4       2_30    0.6190 20.75 0.0001561 -3.969        2
491        TRAPPC3       1_22    0.6176 23.44 0.0001759  4.907        1
733        PPP2R5B      11_36    0.6117 24.40 0.0001813 -4.623        1
4755          SOX5      12_17    0.6076 25.53 0.0001884  3.966        1
7965        GTF2A1      14_39    0.5965 20.91 0.0001515 -4.352        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-8-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>          genename region_tag susie_pip     mu2       PVE        z num_eqtl
3530         CRHR1      17_27 2.970e-01 3095.21 1.117e-02 -3.36232        1
7121      ARHGAP27      17_27 0.000e+00 2291.37 0.000e+00 -2.08012        1
11430      HLA-DOA       6_26 6.350e-14  565.12 4.360e-16  6.84691        1
10942     HLA-DQA1       6_26 1.136e-13  486.81 6.717e-16  1.95455        1
10825     HLA-DRB1       6_26 1.346e-13  364.90 5.965e-16 -1.49219        1
11728        CLIC1       6_26 7.577e-13  363.20 3.343e-15  8.81238        2
11464         MSH5       6_26 6.363e-13  260.67 2.015e-15  7.40963        2
12571          C4A       6_26 1.336e-12  154.49 2.508e-15  5.29092        1
5338         PRDM5       4_78 0.000e+00  134.51 0.000e+00  0.06252        1
10287        FMNL1      17_27 0.000e+00  123.57 0.000e+00  0.66376        1
9925         ACBD4      17_27 0.000e+00  108.05 0.000e+00  0.26990        2
5014          NMT1      17_27 0.000e+00  100.35 0.000e+00  2.52018        2
10493       BTN3A2       6_20 1.836e-02   62.84 1.401e-05  8.94434        2
9090         DCAKD      17_27 0.000e+00   58.82 0.000e+00 -0.72756        1
2463         GOSR2      17_27 0.000e+00   56.03 0.000e+00 -3.44243        2
8482          TNXB       6_26 1.119e-13   55.82 7.589e-17  3.42145        1
6317         CNNM2      10_66 6.998e-01   48.44 4.117e-04 -8.90156        2
2871        PRSS16       6_21 5.667e-02   47.78 3.290e-05 -7.60149        1
13323    LINC01415      18_30 1.919e-01   46.63 1.087e-04 -5.32426        1
13051 RP11-490G2.2       1_60 1.276e-02   46.39 7.190e-06  7.32158        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>          genename region_tag susie_pip     mu2       PVE      z num_eqtl
3530         CRHR1      17_27    0.2970 3095.21 0.0111688 -3.362        1
6317         CNNM2      10_66    0.6998   48.44 0.0004117 -8.902        2
7629         THOC7       3_43    0.9813   34.05 0.0004059 -6.066        2
11134       ZNF823      19_10    0.9749   29.06 0.0003441  5.468        2
13483 RP11-230C9.4      6_102    0.9873   24.01 0.0002879 -4.866        2
9133       MAP3K11      11_36    0.8338   23.52 0.0002382 -4.544        1
9024          FUT9       6_65    0.6687   29.04 0.0002360  5.427        1
12304   AC012074.2       2_15    0.8746   21.96 0.0002333  4.623        1
10221        ACOT1      14_34    0.8412   22.58 0.0002308  4.284        3
1619        ZC3H7B      22_17    0.3965   45.66 0.0002200  5.015        3
6584         TADA1       1_82    0.7488   23.41 0.0002130 -4.174        2
6336       ARFGAP2      11_29    0.7316   23.92 0.0002126  4.740        1
108          ELAC2      17_11    0.8056   21.71 0.0002124  4.542        1
3758       BHLHE41      12_18    0.7421   22.88 0.0002063  4.024        1
4755          SOX5      12_17    0.6076   25.53 0.0001884  3.966        1
6470         PLBD2      12_68    0.7284   20.64 0.0001827  3.986        1
14019        ERICD       8_92    0.7082   21.16 0.0001821 -4.157        1
733        PPP2R5B      11_36    0.6117   24.40 0.0001813 -4.623        1
491        TRAPPC3       1_22    0.6176   23.44 0.0001759  4.907        1
748         ATP1B3       3_87    0.5394   26.72 0.0001751  3.663        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>          genename region_tag susie_pip    mu2       PVE      z num_eqtl
10493       BTN3A2       6_20 1.836e-02  62.84 1.401e-05  8.944        2
6317         CNNM2      10_66 6.998e-01  48.44 4.117e-04 -8.902        2
11728        CLIC1       6_26 7.577e-13 363.20 3.343e-15  8.812        2
7067       ZSCAN12       6_22 1.489e-02  41.30 7.471e-06 -8.008        1
939          NT5C2      10_66 2.700e-01  37.11 1.217e-04  7.804        1
2871        PRSS16       6_21 5.667e-02  47.78 3.290e-05 -7.601        1
11464         MSH5       6_26 6.363e-13 260.67 2.015e-15  7.410        2
13051 RP11-490G2.2       1_60 1.276e-02  46.39 7.190e-06  7.322        1
11430      HLA-DOA       6_26 6.350e-14 565.12 4.360e-16  6.847        1
10634      ZSCAN23       6_22 8.161e-02  45.53 4.514e-05 -6.793        1
9986       ARL6IP4      12_75 7.424e-03  38.54 3.476e-06  6.491        1
12308      ZSCAN31       6_22 2.258e-02  29.34 8.050e-06 -6.446        2
6452         ABCB9      12_75 6.069e-03  37.31 2.751e-06  6.404        1
10988      ZSCAN26       6_22 1.391e-02  33.86 5.721e-06  6.349        3
6407         TAOK2      16_24 3.620e-01  37.85 1.665e-04  6.300        1
9343         ATG13      11_28 2.963e-01  35.09 1.263e-04 -6.169        1
11633      DNAJC19      3_111 2.203e-01  36.38 9.736e-05  6.158        1
11089          NMB      15_39 1.795e-01  40.21 8.768e-05  6.132        1
7629         THOC7       3_43 9.813e-01  34.05 4.059e-04 -6.066        2
8634        INO80E      16_24 1.278e-01  36.26 5.630e-05  6.051        2</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-11-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-11-2.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.006431</code></pre>
<pre><code>          genename region_tag susie_pip    mu2       PVE      z num_eqtl
10493       BTN3A2       6_20 1.836e-02  62.84 1.401e-05  8.944        2
6317         CNNM2      10_66 6.998e-01  48.44 4.117e-04 -8.902        2
11728        CLIC1       6_26 7.577e-13 363.20 3.343e-15  8.812        2
7067       ZSCAN12       6_22 1.489e-02  41.30 7.471e-06 -8.008        1
939          NT5C2      10_66 2.700e-01  37.11 1.217e-04  7.804        1
2871        PRSS16       6_21 5.667e-02  47.78 3.290e-05 -7.601        1
11464         MSH5       6_26 6.363e-13 260.67 2.015e-15  7.410        2
13051 RP11-490G2.2       1_60 1.276e-02  46.39 7.190e-06  7.322        1
11430      HLA-DOA       6_26 6.350e-14 565.12 4.360e-16  6.847        1
10634      ZSCAN23       6_22 8.161e-02  45.53 4.514e-05 -6.793        1
9986       ARL6IP4      12_75 7.424e-03  38.54 3.476e-06  6.491        1
12308      ZSCAN31       6_22 2.258e-02  29.34 8.050e-06 -6.446        2
6452         ABCB9      12_75 6.069e-03  37.31 2.751e-06  6.404        1
10988      ZSCAN26       6_22 1.391e-02  33.86 5.721e-06  6.349        3
6407         TAOK2      16_24 3.620e-01  37.85 1.665e-04  6.300        1
9343         ATG13      11_28 2.963e-01  35.09 1.263e-04 -6.169        1
11633      DNAJC19      3_111 2.203e-01  36.38 9.736e-05  6.158        1
11089          NMB      15_39 1.795e-01  40.21 8.768e-05  6.132        1
7629         THOC7       3_43 9.813e-01  34.05 4.059e-04 -6.066        2
8634        INO80E      16_24 1.278e-01  36.26 5.630e-05  6.051        2</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 28</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-14-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-14-2.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-14-3.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                        Description    FDR Ratio BgRatio
21                                     Spasmophilia 0.0055   1/9  1/9703
24                                           Tetany 0.0055   1/9  1/9703
31                                 Tetany, Neonatal 0.0055   1/9  1/9703
56                                        Tetanilla 0.0055   1/9  1/9703
63                          SENIOR-LOKEN SYNDROME 7 0.0055   1/9  1/9703
64                          HYPOMAGNESEMIA 6, RENAL 0.0055   1/9  1/9703
67                   PROSTATE CANCER, HEREDITARY, 2 0.0055   1/9  1/9703
68       SPASTIC PARAPLEGIA 53, AUTOSOMAL RECESSIVE 0.0055   1/9  1/9703
70 COMBINED OXIDATIVE PHOSPHORYLATION DEFICIENCY 17 0.0055   1/9  1/9703
71                         BARDET-BIEDL SYNDROME 16 0.0055   1/9  1/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-17-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 130</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 64</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.594</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 7</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 74</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>      genename region_tag susie_pip   mu2       PVE      z num_eqtl
9133   MAP3K11      11_36    0.8338 23.52 0.0002382 -4.544        1
10221    ACOT1      14_34    0.8412 22.58 0.0002308  4.284        3
108      ELAC2      17_11    0.8056 21.71 0.0002124  4.542        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>  ctwas    TWAS 
0.02308 0.06154 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9997 0.9942 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code> ctwas   TWAS 
0.4286 0.1081 </code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/75a146615b51c4871f234c7d5cc78179e55ef29c/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-20-1.png" target="_blank">75a1466</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-20-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes---bystanders-only" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes - bystanders only</h2>
<pre class="r"><code>library(biomaRt)
library(GenomicRanges)</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    first, rename</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    reduce</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code># 
ensembl &lt;- useEnsembl(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
G_list &lt;- getBM(filters= &quot;chromosome_name&quot;, attributes = c(&quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;end_position&quot;,&quot;gene_biotype&quot;), values=1:22, mart=ensembl)
G_list &lt;- G_list[G_list$hgnc_symbol!=&quot;&quot;,]
G_list &lt;- G_list[G_list$gene_biotype %in% c(&quot;protein_coding&quot;,&quot;lncRNA&quot;),]
G_list$start &lt;- G_list$start_position
G_list$end &lt;- G_list$end_position
G_list_granges &lt;- makeGRangesFromDataFrame(G_list, keep.extra.columns=T)
# 
# #remove genes without imputed expression from gene lists
known_annotations &lt;- known_annotations[known_annotations %in% ctwas_gene_res$genename]
# 
known_annotations_positions &lt;- G_list[G_list$hgnc_symbol %in% known_annotations,]
half_window &lt;- 1000000
known_annotations_positions$start &lt;- known_annotations_positions$start_position - half_window
known_annotations_positions$end &lt;- known_annotations_positions$end_position + half_window
known_annotations_positions$start[known_annotations_positions$start&lt;1] &lt;- 1
known_annotations_granges &lt;- makeGRangesFromDataFrame(known_annotations_positions, keep.extra.columns=T)
# 
bystanders &lt;- findOverlaps(known_annotations_granges,G_list_granges)
bystanders &lt;- unique(subjectHits(bystanders))
bystanders &lt;- G_list$hgnc_symbol[bystanders]
bystanders &lt;- unique(bystanders[!(bystanders %in% known_annotations)])
unrelated_genes &lt;- bystanders
# 
# #save gene lists
save(known_annotations, file=paste0(results_dir, &quot;/known_annotations.Rd&quot;))
save(unrelated_genes, file=paste0(results_dir, &quot;/bystanders.Rd&quot;))


load(paste0(results_dir, &quot;/known_annotations.Rd&quot;))
load(paste0(results_dir, &quot;/bystanders.Rd&quot;))


#remove genes without imputed expression from bystander list
unrelated_genes &lt;- unrelated_genes[unrelated_genes %in% ctwas_gene_res$genename]

#number of genes in known annotations (with imputed expression)
print(length(known_annotations))</code></pre>
<pre><code>[1] 64</code></pre>
<pre class="r"><code>#subset results to genes in known annotations or bystanders
ctwas_gene_res_subset &lt;- ctwas_gene_res[ctwas_gene_res$genename %in% c(known_annotations, unrelated_genes),]

#assign ctwas and TWAS genes
ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;sig_thresh]

#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.594</code></pre>
<pre class="r"><code>#number of ctwas genes (in known annotations or bystanders)
length(ctwas_genes)</code></pre>
<pre><code>[1] 4</code></pre>
<pre class="r"><code>#number of TWAS genes (in known annotations or bystanders)
length(twas_genes)</code></pre>
<pre><code>[1] 25</code></pre>
<pre class="r"><code>#sensitivity / recall
sensitivity &lt;- rep(NA,2)
names(sensitivity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
sensitivity[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
sensitivity</code></pre>
<pre><code>  ctwas    TWAS 
0.04688 0.12500 </code></pre>
<pre class="r"><code>#specificity / (1 - False Positive Rate)
specificity &lt;- rep(NA,2)
names(specificity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
specificity[&quot;ctwas&quot;] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity[&quot;TWAS&quot;] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
specificity</code></pre>
<pre><code> ctwas   TWAS 
0.9989 0.9807 </code></pre>
<pre class="r"><code>#precision / PPV / (1 - False Discovery Rate)
precision &lt;- rep(NA,2)
names(precision) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
precision[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(twas_genes)
precision</code></pre>
<pre><code>ctwas  TWAS 
 0.75  0.32 </code></pre>
<pre class="r"><code>#store sensitivity and specificity calculations for plots
sensitivity_plot &lt;- sensitivity
specificity_plot &lt;- specificity

#precision / PPV by PIP bin
pip_range &lt;- c(0.2, 0.4, 0.6, 0.8, 1)
precision_range &lt;- rep(NA, length(pip_range))

for (i in 1:length(pip_range)){
  pip_upper &lt;- pip_range[i]

  if (i==1){
    pip_lower &lt;- 0
  } else {
    pip_lower &lt;- pip_range[i-1]
  }
  
  #assign ctwas genes in PIP bin
  ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;=pip_lower &amp; ctwas_gene_res_subset$susie_pip&lt;pip_upper]
  
  precision_range[i] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
}

names(precision_range) &lt;- paste(c(0, pip_range[-length(pip_range)]), pip_range,sep=&quot; - &quot;)

barplot(precision_range, ylim=c(0,1), main=&quot;Precision by PIP Range&quot;, xlab=&quot;PIP Range&quot;, ylab=&quot;Precision&quot;)
abline(h=0.2, lty=2)
abline(h=0.4, lty=2)
abline(h=0.6, lty=2)
abline(h=0.8, lty=2)
barplot(precision_range, add=T, col=&quot;darkgrey&quot;)</code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/75a146615b51c4871f234c7d5cc78179e55ef29c/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-21-1.png" target="_blank">75a1466</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-21-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#precision / PPV by PIP threshold
#pip_range &lt;- c(0.2, 0.4, 0.6, 0.8, 1)
pip_range &lt;- c(0.5, 0.8, 1)
precision_range &lt;- rep(NA, length(pip_range))
number_detected &lt;- rep(NA, length(pip_range))

for (i in 1:length(pip_range)){
  pip_upper &lt;- pip_range[i]

  if (i==1){
    pip_lower &lt;- 0
  } else {
    pip_lower &lt;- pip_range[i-1]
  }
  
  #assign ctwas genes using PIP threshold
  ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;=pip_lower]
  
  number_detected[i] &lt;- length(ctwas_genes)
  precision_range[i] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
}

names(precision_range) &lt;- paste0(&quot;&gt;= &quot;, c(0, pip_range[-length(pip_range)]))

precision_range &lt;- precision_range*100

precision_range &lt;- c(precision_range, precision[&quot;TWAS&quot;]*100)
names(precision_range)[4] &lt;- &quot;TWAS Bonferroni&quot;
number_detected &lt;- c(number_detected, length(twas_genes))

barplot(precision_range, ylim=c(0,100), main=&quot;Precision for Distinguishing Silver Standard and Bystander Genes&quot;, xlab=&quot;PIP Threshold for Detection&quot;, ylab=&quot;% of Detected Genes in Silver Standard&quot;)
abline(h=20, lty=2)
abline(h=40, lty=2)
abline(h=60, lty=2)
abline(h=80, lty=2)
xx &lt;- barplot(precision_range, add=T, col=c(rep(&quot;darkgrey&quot;,3), &quot;white&quot;))
text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, &quot; detected&quot;), pos = 3, cex=0.8)</code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#text(x = xx, y = precision_range, label = paste0(round(precision_range,1), &quot;%&quot;), pos = 3, cex=0.8, offset = 1.5)

#false discovery rate by PIP threshold

barplot(100-precision_range, ylim=c(0,100), main=&quot;False Discovery Rate for Distinguishing Silver Standard and Bystander Genes&quot;, xlab=&quot;PIP Threshold for Detection&quot;, ylab=&quot;% Bystanders in Detected Genes&quot;)
abline(h=20, lty=2)
abline(h=40, lty=2)
abline(h=60, lty=2)
abline(h=80, lty=2)
xx &lt;- barplot(100-precision_range, add=T, col=c(rep(&quot;darkgrey&quot;,3), &quot;white&quot;))
text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, &quot; detected&quot;), pos = 3, cex=0.8)</code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#text(x = xx, y = precision_range, label = paste0(round(precision_range,1), &quot;%&quot;), pos = 3, cex=0.8, offset = 1.5)

#ROC curves

pip_range &lt;- (0:1000)/1000
sensitivity &lt;- rep(NA, length(pip_range))
specificity &lt;- rep(NA, length(pip_range))

for (index in 1:length(pip_range)){
  pip &lt;- pip_range[index]
  ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;=pip]
  sensitivity[index] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
}

plot(1-specificity, sensitivity, type=&quot;l&quot;, xlim=c(0,1), ylim=c(0,1), main=&quot;&quot;, xlab=&quot;1 - Specificity&quot;, ylab=&quot;Sensitivity&quot;)
title(expression(&quot;ROC Curve for cTWAS (black) and TWAS (&quot; * phantom(&quot;red&quot;) * &quot;)&quot;))
title(expression(phantom(&quot;ROC Curve for cTWAS (black) and TWAS (&quot;) * &quot;red&quot; * phantom(&quot;)&quot;)), col.main=&quot;red&quot;)

sig_thresh_range &lt;- seq(from=0, to=max(abs(ctwas_gene_res_subset$z)), length.out=length(pip_range))

for (index in 1:length(sig_thresh_range)){
  sig_thresh_plot &lt;- sig_thresh_range[index]
  twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;=sig_thresh_plot]
  sensitivity[index] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
}

lines(1-specificity, sensitivity, xlim=c(0,1), ylim=c(0,1), col=&quot;red&quot;, lty=1)

abline(a=0,b=1,lty=3)

#add previously computed points from the analysis
ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;sig_thresh]

points(1-specificity_plot[&quot;ctwas&quot;], sensitivity_plot[&quot;ctwas&quot;], pch=21, bg=&quot;black&quot;)
points(1-specificity_plot[&quot;TWAS&quot;], sensitivity_plot[&quot;TWAS&quot;], pch=21, bg=&quot;red&quot;)</code></pre>
<p><img src="figure/SCZ_Brain_Frontal_Cortex_BA9.Rmd/unnamed-chunk-21-4.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] GenomicRanges_1.36.1 GenomeInfoDb_1.20.0  IRanges_2.18.1      
 [4] S4Vectors_0.22.1     BiocGenerics_0.30.0  biomaRt_2.40.1      
 [7] readxl_1.3.1         forcats_0.5.1        stringr_1.4.0       
[10] dplyr_1.0.7          purrr_0.3.4          readr_2.1.1         
[13] tidyr_1.1.4          tidyverse_1.3.1      tibble_3.1.6        
[16] WebGestaltR_0.4.4    disgenet2r_0.99.2    enrichR_3.0         
[19] cowplot_1.0.0        ggplot2_3.3.5        workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] ggbeeswarm_0.6.0       colorspace_2.0-2       rjson_0.2.20          
  [4] ellipsis_0.3.2         rprojroot_2.0.2        XVector_0.24.0        
  [7] fs_1.5.2               rstudioapi_0.13        farver_2.1.0          
 [10] ggrepel_0.9.1          bit64_4.0.5            AnnotationDbi_1.46.0  
 [13] fansi_1.0.2            lubridate_1.8.0        xml2_1.3.3            
 [16] codetools_0.2-16       doParallel_1.0.17      cachem_1.0.6          
 [19] knitr_1.36             jsonlite_1.7.2         apcluster_1.4.8       
 [22] Cairo_1.5-12.2         broom_0.7.10           dbplyr_2.1.1          
 [25] compiler_3.6.1         httr_1.4.2             backports_1.4.1       
 [28] assertthat_0.2.1       Matrix_1.2-18          fastmap_1.1.0         
 [31] cli_3.1.0              later_0.8.0            prettyunits_1.1.1     
 [34] htmltools_0.5.2        tools_3.6.1            igraph_1.2.10         
 [37] GenomeInfoDbData_1.2.1 gtable_0.3.0           glue_1.6.2            
 [40] reshape2_1.4.4         doRNG_1.8.2            Rcpp_1.0.8            
 [43] Biobase_2.44.0         cellranger_1.1.0       jquerylib_0.1.4       
 [46] vctrs_0.3.8            svglite_1.2.2          iterators_1.0.14      
 [49] xfun_0.29              rvest_1.0.2            lifecycle_1.0.1       
 [52] rngtools_1.5.2         XML_3.99-0.3           zlibbioc_1.30.0       
 [55] scales_1.1.1           vroom_1.5.7            hms_1.1.1             
 [58] promises_1.0.1         yaml_2.2.1             curl_4.3.2            
 [61] memoise_2.0.1          ggrastr_1.0.1          gdtools_0.1.9         
 [64] stringi_1.7.6          RSQLite_2.2.8          highr_0.9             
 [67] foreach_1.5.2          rlang_1.0.1            pkgconfig_2.0.3       
 [70] bitops_1.0-7           evaluate_0.14          lattice_0.20-38       
 [73] labeling_0.4.2         bit_4.0.4              tidyselect_1.1.1      
 [76] plyr_1.8.6             magrittr_2.0.2         R6_2.5.1              
 [79] generics_0.1.1         DBI_1.1.2              pillar_1.6.4          
 [82] haven_2.4.3            whisker_0.3-2          withr_2.4.3           
 [85] RCurl_1.98-1.5         modelr_0.1.8           crayon_1.5.0          
 [88] utf8_1.2.2             tzdb_0.2.0             rmarkdown_2.11        
 [91] progress_1.2.2         grid_3.6.1             data.table_1.14.2     
 [94] blob_1.2.2             git2r_0.26.1           reprex_2.0.1          
 [97] digest_0.6.29          httpuv_1.5.1           munsell_0.5.0         
[100] beeswarm_0.2.3         vipor_0.4.5           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
