<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>SCZ - Brain Cerebellum</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SCZ - Brain Cerebellum</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1c69dd24cb48d62e1c884f260adbee91383bbe92targetblank1c69dd2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/1c69dd24cb48d62e1c884f260adbee91383bbe92" target="_blank">1c69dd2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree1c69dd24cb48d62e1c884f260adbee91383bbe92targetblank1c69dd2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/1c69dd24cb48d62e1c884f260adbee91383bbe92" target="_blank">1c69dd2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    data/AF/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Autism_Brain_Amygdala.Rmd
    Untracked:  analysis/Autism_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Untracked:  analysis/Autism_Brain_Caudate_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Cerebellar_Hemisphere.Rmd
    Untracked:  analysis/Autism_Brain_Cerebellum.Rmd
    Untracked:  analysis/Autism_Brain_Cortex.Rmd
    Untracked:  analysis/Autism_Brain_Frontal_Cortex_BA9.Rmd
    Untracked:  analysis/Autism_Brain_Hippocampus.Rmd
    Untracked:  analysis/Autism_Brain_Hypothalamus.Rmd
    Untracked:  analysis/Autism_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Putamen_basal_ganglia.Rmd
    Untracked:  analysis/Autism_Brain_Spinal_cord_cervical_c-1.Rmd
    Untracked:  analysis/Autism_Brain_Substantia_nigra.Rmd
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/Autism_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/SCZ_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_Autism_analysis.sbatch
    Untracked:  code/run_Autism_analysis.sh
    Untracked:  code/run_Autism_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_SCZ_analysis.sbatch
    Untracked:  code/run_SCZ_analysis.sh
    Untracked:  code/run_SCZ_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/Autism/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/SCZ/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Amygdala_S.Rmd
    Modified:   analysis/BMI_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Modified:   analysis/BMI_Brain_Caudate_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellar_Hemisphere_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellum_S.Rmd
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/BMI_Brain_Cortex_S.Rmd
    Modified:   analysis/BMI_Brain_Frontal_Cortex_BA9_S.Rmd
    Modified:   analysis/BMI_Brain_Hippocampus_S.Rmd
    Modified:   analysis/BMI_Brain_Hypothalamus_S.Rmd
    Modified:   analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Putamen_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Modified:   analysis/BMI_Brain_Substantia_nigra_S.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SCZ_Brain_Cerebellum.Rmd</code>) and HTML (<code>docs/SCZ_Brain_Cerebellum.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1c69dd24cb48d62e1c884f260adbee91383bbe92/analysis/SCZ_Brain_Cerebellum.Rmd" target="_blank">1c69dd2</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/SCZ_Brain_Cerebellum.html" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3dd5b4c06fd9ac2be9b0a0519067bba809374ee8/analysis/SCZ_Brain_Cerebellum.Rmd" target="_blank">3dd5b4c</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 11540</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1112  817  667  419  566  572  571  434  457  464  693  659  232  385  381  551 
  17   18   19   20   21   22 
 700  174  910  343  128  305 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 8891</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.7705</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-5-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0147028 0.0002445 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
8.286 8.849 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 82315</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   11540 7573890</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.01708 0.19908 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.08822 1.53847</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-7-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>        genename region_tag susie_pip   mu2       PVE      z num_eqtl
11129     ZNF823      19_10    0.9840 28.84 0.0003447  5.501        2
4195       FEZF1       7_74    0.9784 27.54 0.0003273 -5.314        1
5873      GALNT2      1_117    0.9207 22.69 0.0002538  4.705        2
12285 AC012074.2       2_15    0.9057 21.29 0.0002343  4.623        1
1532     PIK3IP1      22_11    0.8790 21.10 0.0002253  4.340        1
6351     ARFGAP2      11_29    0.7986 23.89 0.0002318  4.740        1
3127       SF3B1      2_117    0.7879 42.04 0.0004024  6.725        1
7609    SERPINI1      3_103    0.7858 19.74 0.0001885 -4.085        2
11503      DISP3        1_9    0.7784 20.82 0.0001969  3.912        1
3914       CNOT1      16_31    0.7776 26.36 0.0002490  5.341        2
1753        PTK6      20_37    0.7546 29.21 0.0002677 -5.380        2
3758        SSPN      12_18    0.7528 21.07 0.0001927  4.024        1
450       ARID1B      6_102    0.7328 21.01 0.0001871 -3.907        1
6495    FAM177A1       14_9    0.6873 22.12 0.0001847 -4.480        3
12147    ANKRD63      15_14    0.6838 28.66 0.0002381  5.452        1
3267      MAP7D1       1_22    0.6838 23.90 0.0001985 -4.907        1
10100     NPIPA1      16_15    0.6712 21.50 0.0001753  4.072        1
9461      DIRAS1       19_3    0.6677 22.93 0.0001860 -4.658        1
13946     EBLN3P       9_28    0.6670 21.43 0.0001737 -4.450        1
753       ATP1B3       3_87    0.6637 22.08 0.0001781  3.663        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-8-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>         genename region_tag susie_pip     mu2       PVE       z num_eqtl
11951     PLEKHM1      17_27 2.769e-05 2961.56 9.961e-07  3.2051        1
12200     FAM215B      17_27 0.000e+00 2370.93 0.000e+00 -3.3160        1
9739     HLA-DQB1       6_26 9.426e-14  859.83 9.846e-16  4.2352        1
10933    HLA-DQA1       6_26 1.703e-13  480.05 9.932e-16  1.9545        1
10811    HLA-DRB1       6_26 1.069e-13  446.81 5.803e-16  3.7636        2
3541       KANSL1      17_27 0.000e+00  317.95 0.000e+00  3.0672        1
10394      ARL17A      17_27 0.000e+00  317.95 0.000e+00  3.0672        1
12087      ARL17B      17_27 0.000e+00  317.95 0.000e+00 -3.0672        1
12366    HLA-DQA2       6_26 1.492e-13  293.87 5.327e-16  0.8547        1
9906        ACBD4      17_27 0.000e+00  156.57 0.000e+00  1.8587        2
5027         NMT1      17_27 0.000e+00  126.28 0.000e+00  2.7209        1
2468         WNT3      17_27 0.000e+00  123.13 0.000e+00  0.2418        1
10265       FMNL1      17_27 0.000e+00  120.89 0.000e+00  0.6638        1
9085        DCAKD      17_27 0.000e+00  108.98 0.000e+00 -1.8291        2
12183     CYP21A2       6_26 2.681e-12   96.41 3.140e-15  0.1375        1
11460      HSPA1L       6_26 1.806e-13   92.93 2.039e-16  0.9130        1
11924         C4B       6_26 1.782e-13   91.62 1.983e-16 -2.5606        2
7137     ARHGAP27      17_27 0.000e+00   84.67 0.000e+00  0.9681        2
13535 RP1-86C11.7       6_21 1.812e-01   66.58 1.465e-04  9.0332        1
4427        C1QL1      17_27 0.000e+00   63.90 0.000e+00  2.1524        2</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>        genename region_tag susie_pip   mu2       PVE      z num_eqtl
3127       SF3B1      2_117    0.7879 42.04 0.0004024  6.725        1
11129     ZNF823      19_10    0.9840 28.84 0.0003447  5.501        2
4195       FEZF1       7_74    0.9784 27.54 0.0003273 -5.314        1
13918  LINC02033       3_28    0.6267 41.50 0.0003160 -6.688        1
1753        PTK6      20_37    0.7546 29.21 0.0002677 -5.380        2
5873      GALNT2      1_117    0.9207 22.69 0.0002538  4.705        2
3914       CNOT1      16_31    0.7776 26.36 0.0002490  5.341        2
12147    ANKRD63      15_14    0.6838 28.66 0.0002381  5.452        1
12285 AC012074.2       2_15    0.9057 21.29 0.0002343  4.623        1
6351     ARFGAP2      11_29    0.7986 23.89 0.0002318  4.740        1
1532     PIK3IP1      22_11    0.8790 21.10 0.0002253  4.340        1
3267      MAP7D1       1_22    0.6838 23.90 0.0001985 -4.907        1
11381  LINC00222       6_73    0.6071 26.74 0.0001972 -5.182        1
11503      DISP3        1_9    0.7784 20.82 0.0001969  3.912        1
3758        SSPN      12_18    0.7528 21.07 0.0001927  4.024        1
7609    SERPINI1      3_103    0.7858 19.74 0.0001885 -4.085        2
450       ARID1B      6_102    0.7328 21.01 0.0001871 -3.907        1
9461      DIRAS1       19_3    0.6677 22.93 0.0001860 -4.658        1
412       CTNNA1       5_82    0.6500 23.43 0.0001850  5.046        1
6495    FAM177A1       14_9    0.6873 22.12 0.0001847 -4.480        3</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag susie_pip   mu2       PVE      z num_eqtl
12064       HCG11       6_20  0.027923 63.10 2.140e-05  9.082        1
13097  CTA-14H9.5       6_20  0.027923 63.10 2.140e-05  9.082        1
13535 RP1-86C11.7       6_21  0.181176 66.58 1.465e-04  9.033        1
2890       PRSS16       6_21  0.039048 57.48 2.727e-05 -9.032        2
10943     ZSCAN16       6_22  0.020120 63.87 1.561e-05 -8.509        1
9834    HIST1H2BC       6_20  0.029117 49.53 1.752e-05 -8.028        1
10473      BTN3A2       6_20  0.021513 45.47 1.188e-05  7.711        2
10608    HIST1H1C       6_20  0.021800 41.34 1.095e-05 -7.382        2
5150        PGBD1       6_22  0.020856 44.61 1.130e-05 -7.166        3
7085      ZSCAN12       6_22  0.036348 34.55 1.526e-05 -6.844        1
10787     ZKSCAN3       6_22  0.029919 32.32 1.175e-05  6.777        2
3127        SF3B1      2_117  0.787861 42.04 4.024e-04  6.725        1
6323      CYP17A1      10_66  0.009517 29.77 3.442e-06  6.720        1
10627     ZSCAN23       6_22  0.115011 43.04 6.014e-05 -6.711        2
13918   LINC02033       3_28  0.626738 41.50 3.160e-04 -6.688        1
10984     ZSCAN26       6_22  0.021741 35.25 9.309e-06  6.656        3
2725       OGFOD2      12_75  0.006418 38.94 3.036e-06  6.518        1
9965      ARL6IP4      12_75  0.005859 38.62 2.749e-06 -6.491        1
2655          MDK      11_28  0.386935 37.17 1.747e-04 -6.357        1
3131        HSPE1      2_117  0.072993 35.44 3.143e-05  6.243        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-11-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-11-2.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.007539</code></pre>
<pre><code>         genename region_tag susie_pip   mu2       PVE      z num_eqtl
12064       HCG11       6_20  0.027923 63.10 2.140e-05  9.082        1
13097  CTA-14H9.5       6_20  0.027923 63.10 2.140e-05  9.082        1
13535 RP1-86C11.7       6_21  0.181176 66.58 1.465e-04  9.033        1
2890       PRSS16       6_21  0.039048 57.48 2.727e-05 -9.032        2
10943     ZSCAN16       6_22  0.020120 63.87 1.561e-05 -8.509        1
9834    HIST1H2BC       6_20  0.029117 49.53 1.752e-05 -8.028        1
10473      BTN3A2       6_20  0.021513 45.47 1.188e-05  7.711        2
10608    HIST1H1C       6_20  0.021800 41.34 1.095e-05 -7.382        2
5150        PGBD1       6_22  0.020856 44.61 1.130e-05 -7.166        3
7085      ZSCAN12       6_22  0.036348 34.55 1.526e-05 -6.844        1
10787     ZKSCAN3       6_22  0.029919 32.32 1.175e-05  6.777        2
3127        SF3B1      2_117  0.787861 42.04 4.024e-04  6.725        1
6323      CYP17A1      10_66  0.009517 29.77 3.442e-06  6.720        1
10627     ZSCAN23       6_22  0.115011 43.04 6.014e-05 -6.711        2
13918   LINC02033       3_28  0.626738 41.50 3.160e-04 -6.688        1
10984     ZSCAN26       6_22  0.021741 35.25 9.309e-06  6.656        3
2725       OGFOD2      12_75  0.006418 38.94 3.036e-06  6.518        1
9965      ARL6IP4      12_75  0.005859 38.62 2.749e-06 -6.491        1
2655          MDK      11_28  0.386935 37.17 1.747e-04 -6.357        1
3131        HSPE1      2_117  0.072993 35.44 3.143e-05  6.243        1</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 32</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-14-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                                               Term Overlap
1         negative regulation of lipid kinase activity (GO:0090219)     2/8
2        positive regulation of mRNA metabolic process (GO:1903313)    2/13
3         regulation of neuron projection arborization (GO:0150011)    2/15
4           post-transcriptional gene silencing by RNA (GO:0035194)    2/20
5  positive regulation of cell projection organization (GO:0031346)   3/117
6             positive regulation of dephosphorylation (GO:0035306)    2/29
7                              gene silencing by miRNA (GO:0035195)    2/32
8     positive regulation of protein dephosphorylation (GO:0035307)    2/34
9            regulation of protein autophosphorylation (GO:0031952)    2/37
10                                mRNA destabilization (GO:0061157)    2/38
11      regulation of protein tyrosine kinase activity (GO:0061097)    2/39
12             regulation of protein dephosphorylation (GO:0035304)    2/41
   Adjusted.P.value              Genes
1           0.02002    PIK3IP1;PPP2R5A
2           0.02485        MOV10;CNOT1
3           0.02485         MOV10;DLG4
4           0.03355        MOV10;CNOT1
5           0.04669 DLG4;PTK6;SERPINI1
6           0.04669       PTPA;PPP2R5A
7           0.04669        MOV10;CNOT1
8           0.04669       PTPA;PPP2R5A
9           0.04669    PPP2R5B;PPP2R5A
10          0.04669        MOV10;CNOT1
11          0.04669          DLG4;PTK6
12          0.04727       PTPA;PPP2R5A
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-14-2.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                              Term Overlap Adjusted.P.value
1 protein phosphatase type 2A complex (GO:0000159)    3/17         0.000152
                 Genes
1 PTPA;PPP2R5B;PPP2R5A
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-14-3.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                                 Term Overlap Adjusted.P.value
1 protein phosphatase activator activity (GO:0072542)    3/13        6.315e-05
2 protein phosphatase regulator activity (GO:0019888)    3/57        3.080e-03
3         phosphatase activator activity (GO:0019211)    2/14        4.460e-03
                 Genes
1 PTPA;PPP2R5B;PPP2R5A
2 PTPA;PPP2R5B;PPP2R5A
3      PPP2R5B;PPP2R5A</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                                 Description     FDR Ratio
62                             Disproportionate tall stature 0.02646  1/13
65 Familial encephalopathy with neuroserpin inclusion bodies 0.02646  1/13
70                               Hematopoetic Myelodysplasia 0.02646  2/13
78  HYPOGONADOTROPIC HYPOGONADISM 22 WITH OR WITHOUT ANOSMIA 0.02646  1/13
54                Refractory anemia with ringed sideroblasts 0.03172  1/13
63         Macular Dystrophy, Butterfly-Shaped Pigmentary, 2 0.03172  1/13
66         Patterned dystrophy of retinal pigment epithelium 0.03172  1/13
72                     CHROMOSOME 6q24-q25 DELETION SYNDROME 0.03172  1/13
75                                  MYELODYSPLASTIC SYNDROME 0.03172  2/13
79             Butterfly-shaped pigmentary macular dystrophy 0.03172  1/13
   BgRatio
62  1/9703
65  1/9703
70 29/9703
78  1/9703
54  2/9703
63  3/9703
66  3/9703
72  2/9703
75 67/9703
79  3/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-17-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 130</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 67</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.595</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 5</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 87</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>     genename region_tag susie_pip  mu2       PVE    z num_eqtl
1532  PIK3IP1      22_11     0.879 21.1 0.0002253 4.34        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>   ctwas     TWAS 
0.007692 0.053846 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9997 0.9930 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.20000 0.08046 </code></pre>
<p><img src="figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Cerebellum.Rmd/unnamed-chunk-20-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7      
 [5] purrr_0.3.4       readr_2.1.1       tidyr_1.1.4       tidyverse_1.3.1  
 [9] tibble_3.1.6      WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0      
[13] cowplot_1.0.0     ggplot2_3.3.5     workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       doParallel_1.0.17
 [5] httr_1.4.2        rprojroot_2.0.2   tools_3.6.1       backports_1.4.1  
 [9] doRNG_1.8.2       utf8_1.2.2        R6_2.5.1          vipor_0.4.5      
[13] DBI_1.1.2         colorspace_2.0-2  withr_2.4.3       ggrastr_1.0.1    
[17] tidyselect_1.1.1  bit_4.0.4         curl_4.3.2        compiler_3.6.1   
[21] git2r_0.26.1      rvest_1.0.2       cli_3.1.0         Cairo_1.5-12.2   
[25] xml2_1.3.3        labeling_0.4.2    scales_1.1.1      apcluster_1.4.8  
[29] digest_0.6.29     rmarkdown_2.11    svglite_1.2.2     pkgconfig_2.0.3  
[33] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[37] rlang_1.0.1       rstudioapi_0.13   RSQLite_2.2.8     jquerylib_0.1.4  
[41] farver_2.1.0      generics_0.1.1    jsonlite_1.7.2    vroom_1.5.7      
[45] magrittr_2.0.2    Matrix_1.2-18     ggbeeswarm_0.6.0  Rcpp_1.0.8       
[49] munsell_0.5.0     fansi_1.0.2       gdtools_0.1.9     lifecycle_1.0.1  
[53] stringi_1.7.6     whisker_0.3-2     yaml_2.2.1        plyr_1.8.6       
[57] grid_3.6.1        blob_1.2.2        ggrepel_0.9.1     parallel_3.6.1   
[61] promises_1.0.1    crayon_1.5.0      lattice_0.20-38   haven_2.4.3      
[65] hms_1.1.1         knitr_1.36        pillar_1.6.4      igraph_1.2.10    
[69] rjson_0.2.20      rngtools_1.5.2    reshape2_1.4.4    codetools_0.2-16 
[73] reprex_2.0.1      glue_1.6.2        evaluate_0.14     data.table_1.14.2
[77] modelr_0.1.8      vctrs_0.3.8       tzdb_0.2.0        httpuv_1.5.1     
[81] foreach_1.5.2     cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[85] cachem_1.0.6      xfun_0.29         broom_0.7.10      later_0.8.0      
[89] iterators_1.0.14  beeswarm_0.2.3    memoise_2.0.1     ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
