<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>SCZ - Brain Putamen basal ganglia</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SCZ - Brain Putamen basal ganglia</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree35c93eaa46bcce24c334bfd1b736b8b46c2b193etargetblank35c93eaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/35c93eaa46bcce24c334bfd1b736b8b46c2b193e" target="_blank">35c93ea</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree35c93eaa46bcce24c334bfd1b736b8b46c2b193etargetblank35c93eaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/35c93eaa46bcce24c334bfd1b736b8b46c2b193e" target="_blank">35c93ea</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    data/AF/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/Autism_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/SCZ_S_out/
    Untracked:  code/SCZ_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_Autism_analysis.sbatch
    Untracked:  code/run_Autism_analysis.sh
    Untracked:  code/run_Autism_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_SCZ_analysis.sbatch
    Untracked:  code/run_SCZ_analysis.sh
    Untracked:  code/run_SCZ_analysis_S.sbatch
    Untracked:  code/run_SCZ_analysis_S.sh
    Untracked:  code/run_SCZ_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  code/wflow_build-18162038.err
    Untracked:  code/wflow_build-18162038.out
    Untracked:  code/wflow_build.R
    Untracked:  code/wflow_build.sbatch
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/Autism/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/SCZ/
    Untracked:  data/SCZ_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd</code>) and HTML (<code>docs/SCZ_Brain_Putamen_basal_ganglia.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ed11e40f7ab37217222130a7af2cae055f908c2d/analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd" target="_blank">ed11e40</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/ed11e40f7ab37217222130a7af2cae055f908c2d/docs/SCZ_Brain_Putamen_basal_ganglia.html" target="_blank">ed11e40</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f93f4b793ca8f036244ef550efe46166587d1353/analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd" target="_blank">f93f4b7</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/f93f4b793ca8f036244ef550efe46166587d1353/docs/SCZ_Brain_Putamen_basal_ganglia.html" target="_blank">f93f4b7</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/08b7e6f23c05ffa11524bad21d72b426eb9550ed/analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd" target="_blank">08b7e6f</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/08b7e6f23c05ffa11524bad21d72b426eb9550ed/docs/SCZ_Brain_Putamen_basal_ganglia.html" target="_blank">08b7e6f</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/3f6d4107d70179d8f092dcd1001c81e26fee346f/docs/SCZ_Brain_Putamen_basal_ganglia.html" target="_blank">3f6d410</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/18287f5a3f8a0a297286a406f7d3f6f7105f89a4/analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd" target="_blank">18287f5</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/75a146615b51c4871f234c7d5cc78179e55ef29c/docs/SCZ_Brain_Putamen_basal_ganglia.html" target="_blank">75a1466</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/1c69dd24cb48d62e1c884f260adbee91383bbe92/analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd" target="_blank">1c69dd2</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/SCZ_Brain_Putamen_basal_ganglia.html" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3dd5b4c06fd9ac2be9b0a0519067bba809374ee8/analysis/SCZ_Brain_Putamen_basal_ganglia.Rmd" target="_blank">3dd5b4c</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 11274</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1123  795  659  440  534  593  558  410  421  464  669  651  228  384  383  516 
  17   18   19   20   21   22 
 678  179  848  340  123  278 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 8904</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.7898</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-5-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0123138 0.0002506 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code>  gene    snp 
10.662  8.503 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 82315</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   11274 7573890</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.01798 0.19610 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.1204 1.4388</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-7-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>        genename region_tag susie_pip     mu2       PVE      z num_eqtl
3448       CRHR1      17_27    0.9977 3537.44 0.0428738  3.362        1
10867     ZNF823      19_10    0.9813   29.49 0.0003516  5.455        1
4092       FEZF1       7_74    0.9787   28.51 0.0003390 -5.314        1
11990 AC012074.2       2_15    0.9012   21.92 0.0002399  4.623        1
8791       GNG12       1_42    0.8876   22.45 0.0002421  4.526        2
3043       SF3B1      2_117    0.8595   43.81 0.0004574  6.725        1
11945  HIST1H2BN       6_21    0.7887   91.05 0.0008723 10.773        1
6321       PLBD2      12_68    0.7749   20.26 0.0001907  3.986        1
8798        FUT9       6_65    0.7434   29.72 0.0002684  5.427        1
7435    SERPINI1      3_103    0.6967   20.40 0.0001726 -4.038        1
433       ARID1B      6_102    0.6827   22.81 0.0001892 -3.907        1
13621  LINC02033       3_27    0.6750   42.14 0.0003455 -6.688        1
11497      AS3MT      10_66    0.6629   47.24 0.0003805  8.510        2
4444       REEP2       5_82    0.6544   27.96 0.0002223  5.204        1
10737      PCBP2      12_33    0.6333   22.13 0.0001703  4.202        1
11110 LIN28B-AS1       6_70    0.6276   23.11 0.0001762 -4.630        2
3935        KLC1      14_54    0.6130   41.31 0.0003076  7.069        1
11329      ITSN1      21_14    0.6071   24.37 0.0001798  3.885        1
5721      CEP170      1_128    0.5858   24.35 0.0001733 -4.678        1
905        NT5C2      10_66    0.5494   40.33 0.0002691 -8.066        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-8-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>         genename region_tag susie_pip     mu2       PVE        z num_eqtl
3448        CRHR1      17_27 9.977e-01 3537.44 4.287e-02  3.36232        1
10167      ARL17A      17_27 0.000e+00 3039.45 0.000e+00 -2.78513        2
6964     ARHGAP27      17_27 0.000e+00  338.09 0.000e+00  0.34014        1
12064    HLA-DQA2       6_26 1.865e-14  251.01 5.688e-17  0.21639        1
10691    HLA-DQA1       6_26 2.209e-14  201.77 5.415e-17  3.44601        1
12119      LY6G5B       6_26 7.245e-09  185.79 1.635e-11 -7.00014        1
10035       FMNL1      17_27 0.000e+00  140.29 0.000e+00 -0.66376        1
11190        MSH5       6_26 1.045e-04  127.66 1.621e-07  7.59028        2
4897         NMT1      17_27 0.000e+00  119.81 0.000e+00  2.85333        1
8857        DCAKD      17_27 0.000e+00  115.79 0.000e+00 -2.99967        1
9689        ACBD4      17_27 0.000e+00  105.48 0.000e+00  0.12846        2
11639         C4B       6_26 5.842e-13  102.32 7.262e-16 -4.92818        1
11945   HIST1H2BN       6_21 7.887e-01   91.05 8.723e-04 10.77288        1
9507     HLA-DQB1       6_26 6.661e-15   85.64 6.930e-18  4.33946        1
10958    HLA-DRB5       6_26 4.996e-15   75.21 4.565e-18  2.07566        1
10276      HEXIM1      17_27 0.000e+00   73.25 0.000e+00 -2.84515        1
2412        GOSR2      17_27 0.000e+00   72.14 0.000e+00 -2.50963        1
13230 RP1-86C11.7       6_21 1.175e-01   71.63 1.022e-04  9.03322        1
11877     CYP21A2       6_26 1.862e-13   63.59 1.438e-16 -0.01504        1
10244      BTN3A2       6_20 1.530e-02   56.53 1.051e-05  8.19734        3</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>        genename region_tag susie_pip     mu2       PVE      z num_eqtl
3448       CRHR1      17_27    0.9977 3537.44 0.0428738  3.362        1
11945  HIST1H2BN       6_21    0.7887   91.05 0.0008723 10.773        1
3043       SF3B1      2_117    0.8595   43.81 0.0004574  6.725        1
11497      AS3MT      10_66    0.6629   47.24 0.0003805  8.510        2
10867     ZNF823      19_10    0.9813   29.49 0.0003516  5.455        1
13621  LINC02033       3_27    0.6750   42.14 0.0003455 -6.688        1
4092       FEZF1       7_74    0.9787   28.51 0.0003390 -5.314        1
3935        KLC1      14_54    0.6130   41.31 0.0003076  7.069        1
905        NT5C2      10_66    0.5494   40.33 0.0002691 -8.066        1
8798        FUT9       6_65    0.7434   29.72 0.0002684  5.427        1
2590         MDK      11_28    0.5312   38.43 0.0002480 -6.357        1
8791       GNG12       1_42    0.8876   22.45 0.0002421  4.526        2
11990 AC012074.2       2_15    0.9012   21.92 0.0002399  4.623        1
4444       REEP2       5_82    0.6544   27.96 0.0002223  5.204        1
6321       PLBD2      12_68    0.7749   20.26 0.0001907  3.986        1
433       ARID1B      6_102    0.6827   22.81 0.0001892 -3.907        1
11329      ITSN1      21_14    0.6071   24.37 0.0001798  3.885        1
11110 LIN28B-AS1       6_70    0.6276   23.11 0.0001762 -4.630        2
5721      CEP170      1_128    0.5858   24.35 0.0001733 -4.678        1
7435    SERPINI1      3_103    0.6967   20.40 0.0001726 -4.038        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag susie_pip    mu2       PVE      z num_eqtl
11945   HIST1H2BN       6_21 7.887e-01  91.05 8.723e-04 10.773        1
13230 RP1-86C11.7       6_21 1.175e-01  71.63 1.022e-04  9.033        1
11497       AS3MT      10_66 6.629e-01  47.24 3.805e-04  8.510        2
10244      BTN3A2       6_20 1.530e-02  56.53 1.051e-05  8.197        3
905         NT5C2      10_66 5.494e-01  40.33 2.691e-04 -8.066        1
6164        CNNM2      10_66 7.538e-02  34.34 3.145e-05 -7.691        1
11190        MSH5       6_26 1.045e-04 127.66 1.621e-07  7.590        2
3935         KLC1      14_54 6.130e-01  41.31 3.076e-04  7.069        1
12119      LY6G5B       6_26 7.245e-09 185.79 1.635e-11 -7.000        1
10392     ZSCAN23       6_22 1.162e-01  47.98 6.775e-05 -6.789        2
3043        SF3B1      2_117 8.595e-01  43.81 4.574e-04  6.725        1
10545     ZKSCAN3       6_22 2.269e-02  33.66 9.278e-06  6.709        1
13621   LINC02033       3_27 6.750e-01  42.14 3.455e-04 -6.688        1
10732     ZSCAN26       6_22 1.608e-02  37.56 7.336e-06  6.658        3
6302        ABCB9      12_75 7.651e-03  38.72 3.599e-06  6.404        1
2590          MDK      11_28 5.312e-01  38.43 2.480e-04 -6.357        1
5872       CCDC39      3_111 2.934e-01  38.48 1.372e-04 -6.338        1
2929         FXR1      3_111 1.977e-01  37.68 9.050e-05  6.308        1
1212     PPP1R13B      14_54 9.198e-02  42.90 4.793e-05  6.297        1
13228   U91328.19       6_20 5.982e-02  41.81 3.038e-05 -6.254        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-11-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-2">
Past versions of unnamed-chunk-11-2.png
</button>
</p>
<div id="fig-unnamed-chunk-11-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-11-2.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.006741</code></pre>
<pre><code>         genename region_tag susie_pip    mu2       PVE      z num_eqtl
11945   HIST1H2BN       6_21 7.887e-01  91.05 8.723e-04 10.773        1
13230 RP1-86C11.7       6_21 1.175e-01  71.63 1.022e-04  9.033        1
11497       AS3MT      10_66 6.629e-01  47.24 3.805e-04  8.510        2
10244      BTN3A2       6_20 1.530e-02  56.53 1.051e-05  8.197        3
905         NT5C2      10_66 5.494e-01  40.33 2.691e-04 -8.066        1
6164        CNNM2      10_66 7.538e-02  34.34 3.145e-05 -7.691        1
11190        MSH5       6_26 1.045e-04 127.66 1.621e-07  7.590        2
3935         KLC1      14_54 6.130e-01  41.31 3.076e-04  7.069        1
12119      LY6G5B       6_26 7.245e-09 185.79 1.635e-11 -7.000        1
10392     ZSCAN23       6_22 1.162e-01  47.98 6.775e-05 -6.789        2
3043        SF3B1      2_117 8.595e-01  43.81 4.574e-04  6.725        1
10545     ZKSCAN3       6_22 2.269e-02  33.66 9.278e-06  6.709        1
13621   LINC02033       3_27 6.750e-01  42.14 3.455e-04 -6.688        1
10732     ZSCAN26       6_22 1.608e-02  37.56 7.336e-06  6.658        3
6302        ABCB9      12_75 7.651e-03  38.72 3.599e-06  6.404        1
2590          MDK      11_28 5.312e-01  38.43 2.480e-04 -6.357        1
5872       CCDC39      3_111 2.934e-01  38.48 1.372e-04 -6.338        1
2929         FXR1      3_111 1.977e-01  37.68 9.050e-05  6.308        1
1212     PPP1R13B      14_54 9.198e-02  42.90 4.793e-05  6.297        1
13228   U91328.19       6_20 5.982e-02  41.81 3.038e-05 -6.254        1</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 27</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-14-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                                                        Term
1                    regulation of leukocyte cell-cell adhesion (GO:1903037)
2 regulation of leukocyte adhesion to vascular endothelial cell (GO:1904994)
3          positive regulation of neuron projection development (GO:0010976)
4           positive regulation of cell projection organization (GO:0031346)
  Overlap Adjusted.P.value             Genes
1    2/12          0.01811          FUT9;MDK
2    2/13          0.01811          FUT9;MDK
3    3/88          0.01985 FUT9;MDK;SERPINI1
4   3/117          0.03438 FUT9;MDK;SERPINI1
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-14-2.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                              Term Overlap Adjusted.P.value
1 protein phosphatase type 2A complex (GO:0000159)    2/17         0.008722
         Genes
1 PTPA;PPP2R5B
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-3">
Past versions of unnamed-chunk-14-3.png
</button>
</p>
<div id="fig-unnamed-chunk-14-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-14-3.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                                 Term Overlap Adjusted.P.value
1 protein phosphatase activator activity (GO:0072542)    2/13         0.007189
         Genes
1 PTPA;PPP2R5B</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                                  Description     FDR Ratio
3                                           Anxiety Disorders 0.02009  2/13
60                                   Anxiety States, Neurotic 0.02009  2/13
94                                 Anxiety neurosis (finding) 0.02009  2/13
103 Familial encephalopathy with neuroserpin inclusion bodies 0.02009  1/13
113                SPASTIC PARAPLEGIA 72, AUTOSOMAL RECESSIVE 0.02009  1/13
114                 SPASTIC PARAPLEGIA 72, AUTOSOMAL DOMINANT 0.02009  1/13
116                SPASTIC PARAPLEGIA 45, AUTOSOMAL RECESSIVE 0.02009  1/13
117                                     CONE-ROD DYSTROPHY 20 0.02009  1/13
118  HYPOGONADOTROPIC HYPOGONADISM 22 WITH OR WITHOUT ANOSMIA 0.02009  1/13
28                        Neoplasms, Glandular and Epithelial 0.02125  1/13
    BgRatio
3   44/9703
60  44/9703
94  44/9703
103  1/9703
113  1/9703
114  1/9703
116  1/9703
117  1/9703
118  1/9703
28   2/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<pre><code>Warning: ggrepel: 6 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-17-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 130</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 63</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.59</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 6</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 76</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>     genename region_tag susie_pip     mu2       PVE     z num_eqtl
8791    GNG12       1_42    0.8876   22.45 0.0002421 4.526        2
3448    CRHR1      17_27    0.9977 3537.44 0.0428738 3.362        1</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>  ctwas    TWAS 
0.02308 0.06923 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9997 0.9940 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code> ctwas   TWAS 
0.5000 0.1184 </code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/75a146615b51c4871f234c7d5cc78179e55ef29c/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-20-1.png" target="_blank">75a1466</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ff6403af6ee51518b5688c45524c9108825b8e3e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-20-1.png" target="_blank">ff6403a</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ctwas-is-more-precise-than-twas-in-distinguishing-silver-standard-and-bystander-genes" class="section level2">
<h2>cTWAS is more precise than TWAS in distinguishing silver standard and bystander genes</h2>
<pre class="r"><code>#number of genes in known annotations (with imputed expression)
print(length(known_annotations))</code></pre>
<pre><code>[1] 63</code></pre>
<pre class="r"><code>#number of bystander genes (with imputed expression)
print(length(unrelated_genes))</code></pre>
<pre><code>[1] 830</code></pre>
<pre class="r"><code>#subset results to genes in known annotations or bystanders
ctwas_gene_res_subset &lt;- ctwas_gene_res[ctwas_gene_res$genename %in% c(known_annotations, unrelated_genes),]

#assign ctwas and TWAS genes
ctwas_genes &lt;- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)&gt;sig_thresh]

#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.59</code></pre>
<pre class="r"><code>#number of ctwas genes (in known annotations or bystanders)
length(ctwas_genes)</code></pre>
<pre><code>[1] 3</code></pre>
<pre class="r"><code>#number of TWAS genes (in known annotations or bystanders)
length(twas_genes)</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>#sensitivity / recall
sensitivity</code></pre>
<pre><code>  ctwas    TWAS 
0.04762 0.14286 </code></pre>
<pre class="r"><code>#specificity / (1 - False Positive Rate)
specificity</code></pre>
<pre><code> ctwas   TWAS 
1.0000 0.9867 </code></pre>
<pre class="r"><code>#precision / PPV / (1 - False Discovery Rate)
precision</code></pre>
<pre><code>ctwas  TWAS 
 1.00  0.45 </code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3f6d4107d70179d8f092dcd1001c81e26fee346f/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-24-1.png" target="_blank">3f6d410</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3f6d4107d70179d8f092dcd1001c81e26fee346f/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-25-1.png" target="_blank">3f6d410</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="undetected-silver-standard-genes-have-low-twas-z-scores-or-stronger-signal-from-nearby-variants" class="section level2">
<h2>Undetected silver standard genes have low TWAS z-scores or stronger signal from nearby variants</h2>
<pre class="r"><code>#table of outcomes for silver standard genes
-sort(-table(silver_standard_case))</code></pre>
<pre><code>silver_standard_case
          Not Imputed Insignificant z-score         Nearby SNP(s) 
                   67                    53                     7 
 Detected (PIP &gt; 0.8) 
                    3 </code></pre>
<pre class="r"><code>#show inconclusive genes
silver_standard_case[silver_standard_case==&quot;Inconclusive&quot;]</code></pre>
<pre><code>named character(0)</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f93f4b793ca8f036244ef550efe46166587d1353/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-29-1.png" target="_blank">f93f4b7</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3f6d4107d70179d8f092dcd1001c81e26fee346f/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-29-1.png" target="_blank">3f6d410</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-plot-for-three-silver-standard-genes-that-ctwas-identifies" class="section level2">
<h2>Locus plot for three silver standard genes that cTWAS identifies</h2>
<div id="locus-2_117-the-output-of-ctwas-is-very-clear.-only-gene-sf3b1-has-signal." class="section level4">
<h4>Locus 2_117: The output of cTWAS is very clear. Only gene SF3B1 has signal.</h4>
<pre class="r"><code>locus_plot(&quot;2_117&quot;, label=&quot;TWAS&quot;)</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/35c93eaa46bcce24c334bfd1b736b8b46c2b193e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-30-1.png" target="_blank">35c93ea</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f93f4b793ca8f036244ef550efe46166587d1353/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-30-1.png" target="_blank">f93f4b7</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3f6d4107d70179d8f092dcd1001c81e26fee346f/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-30-1.png" target="_blank">3f6d410</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-17_27-in-twas-no-snp-or-gene-passes-the-threshold.-but-ctwas-is-able-to-identify-gene-crhr1-with-high-pip." class="section level4">
<h4>Locus 17_27: In TWAS, no SNP or gene passes the threshold. But cTWAS is able to identify gene CRHR1 with high PIP.</h4>
<pre class="r"><code>locus_plot(&quot;17_27&quot;, label=&quot;TWAS&quot;)</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-31-1">
Past versions of unnamed-chunk-31-1.png
</button>
</p>
<div id="fig-unnamed-chunk-31-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/35c93eaa46bcce24c334bfd1b736b8b46c2b193e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-31-1.png" target="_blank">35c93ea</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-19_10-both-twas-and-ctwas-work-well-in-this-locus." class="section level4">
<h4>Locus 19_10: Both TWAS and cTWAS work well in this locus.</h4>
<pre class="r"><code>locus_plot(&quot;19_10&quot;, label=&quot;TWAS&quot;)</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/35c93eaa46bcce24c334bfd1b736b8b46c2b193e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-32-1.png" target="_blank">35c93ea</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="some-know-scz-risk-genes-that-ctwas-does-not-find" class="section level2">
<h2>Some know SCZ risk genes that cTWAS does not find</h2>
<div id="locus-15_42-ctwas-assign-higher-pip-to-the-snp-rs8032315-rather-than-the-gene-furin.-this-snp-is-both-eqtl-and-sqtl-of-fes-in-nerve-tibial-and-sqtl-of-furin-in-nerve-tibial" class="section level4">
<h4>Locus 15_42: cTWAS assign higher PIP to the SNP rs8032315 rather than the gene FURIN. This SNP is both eQTL and sQTL of FES in Nerve-Tibial and sQTL of FURIN in Nerve-Tibial</h4>
<pre class="r"><code>locus_plot(&quot;15_42&quot;, label=&quot;TWAS&quot;)</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/35c93eaa46bcce24c334bfd1b736b8b46c2b193e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-33-1.png" target="_blank">35c93ea</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="locus-19_34-gene-irf3-is-around-the-threshold-in-twas.-there-are-other-two-genes-of-similar-value.-in-ctwas-all-of-them-get-intermediate-pips." class="section level4">
<h4>Locus 19_34: Gene IRF3 is around the threshold in TWAS. There are other two genes of similar value. In ctwas, all of them get intermediate PIPs.</h4>
<pre class="r"><code>locus_plot5(&quot;19_34&quot;, focus=&quot;IRF3&quot;)</code></pre>
<p><img src="figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/35c93eaa46bcce24c334bfd1b736b8b46c2b193e/docs/figure/SCZ_Brain_Putamen_basal_ganglia.Rmd/unnamed-chunk-34-1.png" target="_blank">35c93ea</a>
</td>
<td>
sq-96
</td>
<td>
2022-03-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7      
 [5] purrr_0.3.4       readr_2.1.1       tidyr_1.1.4       tidyverse_1.3.1  
 [9] tibble_3.1.6      WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0      
[13] cowplot_1.0.0     ggplot2_3.3.5     workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       doParallel_1.0.17
 [5] httr_1.4.2        rprojroot_2.0.2   tools_3.6.1       backports_1.4.1  
 [9] doRNG_1.8.2       utf8_1.2.2        R6_2.5.1          vipor_0.4.5      
[13] DBI_1.1.2         colorspace_2.0-2  withr_2.4.3       ggrastr_1.0.1    
[17] tidyselect_1.1.1  processx_3.5.2    bit_4.0.4         curl_4.3.2       
[21] compiler_3.6.1    git2r_0.26.1      rvest_1.0.2       cli_3.1.0        
[25] Cairo_1.5-12.2    xml2_1.3.3        labeling_0.4.2    scales_1.1.1     
[29] callr_3.7.0       apcluster_1.4.8   digest_0.6.29     rmarkdown_2.11   
[33] svglite_1.2.2     pkgconfig_2.0.3   htmltools_0.5.2   dbplyr_2.1.1     
[37] fastmap_1.1.0     highr_0.9         rlang_1.0.1       rstudioapi_0.13  
[41] RSQLite_2.2.8     jquerylib_0.1.4   farver_2.1.0      generics_0.1.1   
[45] jsonlite_1.7.2    vroom_1.5.7       magrittr_2.0.2    Matrix_1.2-18    
[49] ggbeeswarm_0.6.0  Rcpp_1.0.8        munsell_0.5.0     fansi_1.0.2      
[53] gdtools_0.1.9     lifecycle_1.0.1   stringi_1.7.6     whisker_0.3-2    
[57] yaml_2.2.1        plyr_1.8.6        grid_3.6.1        blob_1.2.2       
[61] ggrepel_0.9.1     parallel_3.6.1    promises_1.0.1    crayon_1.5.0     
[65] lattice_0.20-38   haven_2.4.3       hms_1.1.1         knitr_1.36       
[69] ps_1.6.0          pillar_1.6.4      igraph_1.2.10     rjson_0.2.20     
[73] rngtools_1.5.2    reshape2_1.4.4    codetools_0.2-16  reprex_2.0.1     
[77] glue_1.6.2        evaluate_0.14     getPass_0.2-2     modelr_0.1.8     
[81] data.table_1.14.2 vctrs_0.3.8       tzdb_0.2.0        httpuv_1.5.1     
[85] foreach_1.5.2     cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[89] cachem_1.0.6      xfun_0.29         broom_0.7.10      later_0.8.0      
[93] iterators_1.0.14  beeswarm_0.2.3    memoise_2.0.1     ellipsis_0.3.2   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
