<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>SCZ - Brain Hypothalamus</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SCZ - Brain Hypothalamus</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree8b00ee6c948b09e1fa8984748374f513a6bdb5a8targetblank8b00ee6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/8b00ee6c948b09e1fa8984748374f513a6bdb5a8" target="_blank">8b00ee6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree8b00ee6c948b09e1fa8984748374f513a6bdb5a8targetblank8b00ee6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/8b00ee6c948b09e1fa8984748374f513a6bdb5a8" target="_blank">8b00ee6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    data/AF/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/SCZ_2018_Brain_Amygdala.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Caudate_basal_ganglia.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Cerebellar_Hemisphere.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Cerebellum.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Cortex.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Frontal_Cortex_BA9.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Hippocampus.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Hypothalamus.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Putamen_basal_ganglia.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Spinal_cord_cervical_c-1.Rmd
    Untracked:  analysis/SCZ_2018_Brain_Substantia_nigra.Rmd
    Untracked:  analysis/SCZ_2020_Single_Brain_Cortex.Rmd
    Untracked:  analysis/SCZ_2020_Single_Brain_Hippocampus.Rmd
    Untracked:  analysis/SCZ_Annotation_Analysis.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/Autism_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/SCZ_2014_EUR_out/
    Untracked:  code/SCZ_2018_out/
    Untracked:  code/SCZ_2020_Single_out/
    Untracked:  code/SCZ_2020_out/
    Untracked:  code/SCZ_S_out/
    Untracked:  code/SCZ_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/process_scz_2018_snps.R
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_Autism_analysis.sbatch
    Untracked:  code/run_Autism_analysis.sh
    Untracked:  code/run_Autism_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_SCZ_2014_EUR_analysis.sbatch
    Untracked:  code/run_SCZ_2014_EUR_analysis.sh
    Untracked:  code/run_SCZ_2014_EUR_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_2018_analysis.sbatch
    Untracked:  code/run_SCZ_2018_analysis.sh
    Untracked:  code/run_SCZ_2018_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_2020_Single_analysis.sbatch
    Untracked:  code/run_SCZ_2020_Single_analysis.sh
    Untracked:  code/run_SCZ_2020_Single_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_2020_analysis.sbatch
    Untracked:  code/run_SCZ_2020_analysis.sh
    Untracked:  code/run_SCZ_2020_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_analysis.sbatch
    Untracked:  code/run_SCZ_analysis.sh
    Untracked:  code/run_SCZ_analysis_S.sbatch
    Untracked:  code/run_SCZ_analysis_S.sh
    Untracked:  code/run_SCZ_ctwas_rss_LDR.R
    Untracked:  code/run_SCZ_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  code/wflow_build.R
    Untracked:  code/wflow_build.sbatch
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/BMI/
    Untracked:  data/PGC3_SCZ_wave3_public.v2.tsv
    Untracked:  data/SCZ/
    Untracked:  data/SCZ_2014_EUR/
    Untracked:  data/SCZ_2018/
    Untracked:  data/SCZ_2020/
    Untracked:  data/SCZ_2020_Single/
    Untracked:  data/SCZ_S/
    Untracked:  data/Supplementary Table 15 - MAGMA.xlsx
    Untracked:  data/Supplementary Table 20 - Prioritised Genes.xlsx
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/scz_2018.RDS
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt
    Untracked:  top_genes_32.txt
    Untracked:  top_genes_37.txt
    Untracked:  top_genes_81.txt

Unstaged changes:
    Modified:   analysis/SCZ_2020_Brain_Cortex.Rmd
    Modified:   analysis/SCZ_Cross_Tissue_Analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 9567</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
943 663 573 390 482 553 471 366 357 395 569 552 184 321 332 385 578 154 739 282 
 21  22 
 30 248 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 6811</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.7119</code></pre>
<!-- ## Check convergence of parameters -->
<!-- ```{r echo=F,message=F} -->
<!-- library(ggplot2) -->
<!-- library(cowplot) -->
<!-- load(paste0(results_dir, "/", analysis_id, "_ctwas.s2.susieIrssres.Rd")) -->
<!-- df <- data.frame(niter = rep(1:ncol(group_prior_rec), 2), -->
<!--                  value = c(group_prior_rec[1,], group_prior_rec[2,]), -->
<!--                  group = rep(c("Gene", "SNP"), each = ncol(group_prior_rec))) -->
<!-- df$group <- as.factor(df$group) -->
<!-- df$value[df$group=="SNP"] <- df$value[df$group=="SNP"]*thin #adjust parameter to account for thin argument -->
<!-- p_pi <- ggplot(df, aes(x=niter, y=value, group=group)) + -->
<!--   geom_line(aes(color=group)) + -->
<!--   geom_point(aes(color=group)) + -->
<!--   xlab("Iteration") + ylab(bquote(pi)) + -->
<!--   ggtitle("Prior mean") + -->
<!--   theme_cowplot() -->
<!-- df <- data.frame(niter = rep(1:ncol(group_prior_var_rec), 2), -->
<!--                  value = c(group_prior_var_rec[1,], group_prior_var_rec[2,]), -->
<!--                  group = rep(c("Gene", "SNP"), each = ncol(group_prior_var_rec))) -->
<!-- df$group <- as.factor(df$group) -->
<!-- p_sigma2 <- ggplot(df, aes(x=niter, y=value, group=group)) + -->
<!--   geom_line(aes(color=group)) + -->
<!--   geom_point(aes(color=group)) + -->
<!--   xlab("Iteration") + ylab(bquote(sigma^2)) + -->
<!--   ggtitle("Prior variance") + -->
<!--   theme_cowplot() -->
<!-- plot_grid(p_pi, p_sigma2) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #estimated group prior -->
<!-- estimated_group_prior <- group_prior_rec[,ncol(group_prior_rec)] -->
<!-- names(estimated_group_prior) <- c("gene", "snp") -->
<!-- estimated_group_prior["snp"] <- estimated_group_prior["snp"]*thin #adjust parameter to account for thin argument -->
<!-- print(estimated_group_prior) -->
<!-- #estimated group prior variance -->
<!-- estimated_group_prior_var <- group_prior_var_rec[,ncol(group_prior_var_rec)] -->
<!-- names(estimated_group_prior_var) <- c("gene", "snp") -->
<!-- print(estimated_group_prior_var) -->
<!-- #report sample size -->
<!-- print(sample_size) -->
<!-- #report group size -->
<!-- group_size <- c(nrow(ctwas_gene_res), n_snps) -->
<!-- print(group_size) -->
<!-- #estimated group PVE -->
<!-- estimated_group_pve <- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation -->
<!-- names(estimated_group_pve) <- c("gene", "snp") -->
<!-- print(estimated_group_pve) -->
<!-- #compare sum(PIP*mu2/sample_size) with above PVE calculation -->
<!-- c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE)) -->
<!-- ``` -->
<!-- ## Genes with highest PIPs -->
<!-- ```{r echo=F} -->
<!-- #distribution of PIPs -->
<!-- hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main="Distribution of Gene PIPs") -->
<!-- #genes with PIP>0.8 or 20 highest PIPs -->
<!-- head(ctwas_gene_res[order(-ctwas_gene_res$susie_pip),report_cols], max(sum(ctwas_gene_res$susie_pip>0.8), 20)) -->
<!-- ``` -->
<!-- ## Genes with largest effect sizes -->
<!-- ```{r echo=F} -->
<!-- #plot PIP vs effect size -->
<!-- plot(ctwas_gene_res$susie_pip, ctwas_gene_res$mu2, xlab="PIP", ylab="mu^2", main="Gene PIPs vs Effect Size") -->
<!-- #genes with 20 largest effect sizes -->
<!-- head(ctwas_gene_res[order(-ctwas_gene_res$mu2),report_cols],20) -->
<!-- ``` -->
<!-- ## Genes with highest PVE -->
<!-- ```{r echo=F} -->
<!-- #genes with 20 highest pve -->
<!-- head(ctwas_gene_res[order(-ctwas_gene_res$PVE),report_cols],20) -->
<!-- ``` -->
<!-- ## Genes with largest z scores -->
<!-- ```{r echo=F} -->
<!-- #genes with 20 largest z scores -->
<!-- head(ctwas_gene_res[order(-abs(ctwas_gene_res$z)),report_cols],20) -->
<!-- ``` -->
<!-- ## Comparing z scores and PIPs -->
<!-- ```{r echo=F} -->
<!-- #set nominal signifiance threshold for z scores -->
<!-- alpha <- 0.05 -->
<!-- #bonferroni adjusted threshold for z scores -->
<!-- sig_thresh <- qnorm(1-(alpha/nrow(ctwas_gene_res)/2), lower=T) -->
<!-- #Q-Q plot for z scores -->
<!-- obs_z <- ctwas_gene_res$z[order(ctwas_gene_res$z)] -->
<!-- exp_z <- qnorm((1:nrow(ctwas_gene_res))/nrow(ctwas_gene_res)) -->
<!-- plot(exp_z, obs_z, xlab="Expected z", ylab="Observed z", main="Gene z score Q-Q plot") -->
<!-- abline(a=0,b=1) -->
<!-- #plot z score vs PIP -->
<!-- plot(abs(ctwas_gene_res$z), ctwas_gene_res$susie_pip, xlab="abs(z)", ylab="PIP") -->
<!-- abline(v=sig_thresh, col="red", lty=2) -->
<!-- #proportion of significant z scores -->
<!-- mean(abs(ctwas_gene_res$z) > sig_thresh) -->
<!-- ``` -->
<!-- ## GO enrichment analysis for genes with PIP>0.5 -->
<!-- ```{r echo=FALSE, message=FALSE} -->
<!-- #GO enrichment analysis -->
<!-- library(enrichR) -->
<!-- dbs <- c("GO_Biological_Process_2021", "GO_Cellular_Component_2021", "GO_Molecular_Function_2021") -->
<!-- genes <- ctwas_gene_res$genename[ctwas_gene_res$susie_pip>0.5] -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #number of genes for gene set enrichment -->
<!-- length(genes) -->
<!-- ``` -->
<!-- ```{r echo=F, message=FALSE} -->
<!-- if (length(genes)>0){ -->
<!--   GO_enrichment <- enrichr(genes, dbs) -->
<!--   for (db in dbs){ -->
<!--     print(db) -->
<!--     df <- GO_enrichment[[db]] -->
<!--     print(plotEnrich(GO_enrichment[[db]])) -->
<!--     df <- df[df$Adjusted.P.value<0.05,c("Term", "Overlap", "Adjusted.P.value", "Genes")] -->
<!--     print(df) -->
<!--   } -->
<!-- } -->
<!-- ``` -->
<!-- ## DisGeNET enrichment analysis for genes with PIP>0.5 -->
<!-- ```{r echo=F, message=FALSE} -->
<!-- if (length(genes)>0){ -->
<!--   library(disgenet2r) -->
<!--   disgenet_api_key <- get_disgenet_api_key( -->
<!--                     email = "wesleycrouse@gmail.com", -->
<!--                     password = "uchicago1" ) -->
<!--   Sys.setenv(DISGENET_API_KEY= disgenet_api_key) -->
<!--   res_enrich <-disease_enrichment(entities=genes, vocabulary = "HGNC", -->
<!--                                database = "CURATED" ) -->
<!--   df <- res_enrich@qresult[1:10, c("Description", "FDR", "Ratio",  "BgRatio")] -->
<!--   print(df) -->
<!-- } -->
<!-- ``` -->
<!-- ## WebGestalt enrichment analysis for genes with PIP>0.5 -->
<!-- ```{r echo=F, message=FALSE} -->
<!-- if (length(genes)>0){ -->
<!--   #WebGestalt enrichment -->
<!--   library(WebGestaltR) -->
<!--   background <- ctwas_gene_res$genename -->
<!--   #listGeneSet() -->
<!--   databases <- c("pathway_KEGG", "disease_GLAD4U", "disease_OMIM") -->
<!--   enrichResult <- WebGestaltR(enrichMethod="ORA", organism="hsapiens", -->
<!--                               interestGene=genes, referenceGene=background, -->
<!--                               enrichDatabase=databases, interestGeneType="genesymbol", -->
<!--                               referenceGeneType="genesymbol", isOutput=F) -->
<!--   print(enrichResult[,c("description", "size", "overlap", "FDR", "database", "userId")]) -->
<!-- } -->
<!-- ``` -->
<!-- ## PIP Manhattan Plot -->
<!-- ```{r echo=F, message=F} -->
<!-- library(tibble) -->
<!-- library(tidyverse) -->
<!-- full.gene.pip.summary <- data.frame(gene_name = ctwas_gene_res$genename, -->
<!--                                     gene_pip = ctwas_gene_res$susie_pip, -->
<!--                                     gene_id = ctwas_gene_res$id, -->
<!--                                     chr = as.integer(ctwas_gene_res$chrom), -->
<!--                                     start = ctwas_gene_res$pos / 1e3, -->
<!--                                     is_highlight = F, stringsAsFactors = F) %>% as_tibble() -->
<!-- full.gene.pip.summary$is_highlight <- full.gene.pip.summary$gene_pip > 0.50 -->
<!-- don <- full.gene.pip.summary %>% -->
<!--   # Compute chromosome size -->
<!--   group_by(chr) %>% -->
<!--   summarise(chr_len=max(start)) %>% -->
<!--   # Calculate cumulative position of each chromosome -->
<!--   mutate(tot=cumsum(chr_len)-chr_len) %>% -->
<!--   dplyr::select(-chr_len) %>% -->
<!--   # Add this info to the initial dataset -->
<!--   left_join(full.gene.pip.summary, ., by=c("chr"="chr")) %>% -->
<!--   # Add a cumulative position of each SNP -->
<!--   arrange(chr, start) %>% -->
<!--   mutate( BPcum=start+tot) -->
<!-- axisdf <- don %>% group_by(chr) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 ) -->
<!-- x_axis_labels <- axisdf$chr -->
<!-- x_axis_labels[seq(1,21,2)] <- "" -->
<!-- ggplot(don, aes(x=BPcum, y=gene_pip)) + -->
<!--   # Show all points -->
<!--   ggrastr::geom_point_rast(aes(color=as.factor(chr)), size=2) + -->
<!--   scale_color_manual(values = rep(c("grey", "skyblue"), 22 )) + -->
<!--   # custom X axis: -->
<!--   # scale_x_continuous(label = axisdf$chr, -->
<!--   #                    breaks= axisdf$center, -->
<!--   #                    guide = guide_axis(n.dodge = 2)) + -->
<!--   scale_x_continuous(label = x_axis_labels, -->
<!--                      breaks = axisdf$center) + -->
<!--   scale_y_continuous(expand = c(0, 0), limits = c(0,1.25), breaks=(1:5)*0.2, minor_breaks=(1:10)*0.1) + # remove space between plot area and x axis -->
<!--   # Add highlighted points -->
<!--   ggrastr::geom_point_rast(data=subset(don, is_highlight==T), color="orange", size=2) + -->
<!--   # Add label using ggrepel to avoid overlapping -->
<!--   ggrepel::geom_label_repel(data=subset(don, is_highlight==T), -->
<!--                             aes(label=gene_name), -->
<!--                             size=4, -->
<!--                             min.segment.length = 0, -->
<!--                             label.size = NA, -->
<!--                             fill = alpha(c("white"),0)) + -->
<!--   # Custom the theme: -->
<!--   theme_bw() + -->
<!--   theme( -->
<!--     text = element_text(size = 14), -->
<!--     legend.position="none", -->
<!--     panel.border = element_blank(), -->
<!--     panel.grid.major.x = element_blank(), -->
<!--     panel.grid.minor.x = element_blank() -->
<!--   ) + -->
<!--   xlab("Chromosome") + -->
<!--   ylab("cTWAS PIP") -->
<!-- ``` -->
<!-- ## Sensitivity, specificity and precision for silver standard genes -->
<!-- ```{r echo=F} -->
<!-- library("readxl") -->
<!-- known_annotations <- read_xlsx("data/summary_known_genes_annotations.xlsx", sheet="SCZ") -->
<!-- known_annotations <- unique(known_annotations$`Gene Symbol`) -->
<!-- unrelated_genes <- ctwas_gene_res$genename[!(ctwas_gene_res$genename %in% known_annotations)] -->
<!-- #assign ctwas, TWAS, and bystander genes -->
<!-- ctwas_genes <- ctwas_gene_res$genename[ctwas_gene_res$susie_pip>0.8] -->
<!-- twas_genes <- ctwas_gene_res$genename[abs(ctwas_gene_res$z)>sig_thresh] -->
<!-- novel_genes <- ctwas_genes[!(ctwas_genes %in% twas_genes)] -->
<!-- #sensitivity / recall -->
<!-- sensitivity <- rep(NA,2) -->
<!-- names(sensitivity) <- c("ctwas", "TWAS") -->
<!-- sensitivity["ctwas"] <- sum(ctwas_genes %in% known_annotations)/length(known_annotations) -->
<!-- sensitivity["TWAS"] <- sum(twas_genes %in% known_annotations)/length(known_annotations) -->
<!-- #sensitivity -->
<!-- #specificity -->
<!-- specificity <- rep(NA,2) -->
<!-- names(specificity) <- c("ctwas", "TWAS") -->
<!-- specificity["ctwas"] <- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes) -->
<!-- specificity["TWAS"] <- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes) -->
<!-- #specificity -->
<!-- #precision / PPV -->
<!-- precision <- rep(NA,2) -->
<!-- names(precision) <- c("ctwas", "TWAS") -->
<!-- precision["ctwas"] <- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes) -->
<!-- precision["TWAS"] <- sum(twas_genes %in% known_annotations)/length(twas_genes) -->
<!-- #precision -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #number of genes in known annotations -->
<!-- print(length(known_annotations)) -->
<!-- #number of genes in known annotations with imputed expression -->
<!-- print(sum(known_annotations %in% ctwas_gene_res$genename)) -->
<!-- #significance threshold for TWAS -->
<!-- print(sig_thresh) -->
<!-- #number of ctwas genes -->
<!-- length(ctwas_genes) -->
<!-- #number of TWAS genes -->
<!-- length(twas_genes) -->
<!-- #show novel genes (ctwas genes with not in TWAS genes) -->
<!-- ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols] -->
<!-- #sensitivity / recall -->
<!-- print(sensitivity) -->
<!-- #specificity -->
<!-- print(specificity) -->
<!-- #precision / PPV -->
<!-- print(precision) -->
<!-- ``` -->
<!-- ```{r,echo=FALSE} -->
<!-- #ROC curves -->
<!-- pip_range <- (0:1000)/1000 -->
<!-- sensitivity <- rep(NA, length(pip_range)) -->
<!-- specificity <- rep(NA, length(pip_range)) -->
<!-- for (index in 1:length(pip_range)){ -->
<!--   pip <- pip_range[index] -->
<!--   ctwas_genes <- ctwas_gene_res$genename[ctwas_gene_res$susie_pip>=pip] -->
<!--   sensitivity[index] <- sum(ctwas_genes %in% known_annotations)/length(known_annotations) -->
<!--   specificity[index] <- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes) -->
<!-- } -->
<!-- plot(1-specificity, sensitivity, type="l", xlim=c(0,1), ylim=c(0,1)) -->
<!-- sig_thresh_range <- seq(from=0, to=max(abs(ctwas_gene_res$z)), length.out=length(pip_range)) -->
<!-- for (index in 1:length(sig_thresh_range)){ -->
<!--   sig_thresh_plot <- sig_thresh_range[index] -->
<!--   twas_genes <- ctwas_gene_res$genename[abs(ctwas_gene_res$z)>=sig_thresh_plot] -->
<!--   sensitivity[index] <- sum(twas_genes %in% known_annotations)/length(known_annotations) -->
<!--   specificity[index] <- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes) -->
<!-- } -->
<!-- lines(1-specificity, sensitivity, xlim=c(0,1), ylim=c(0,1), col="red", lty=2) -->
<!-- ``` -->
<!-- ## cTWAS is more precise than TWAS in distinguishing silver standard and bystander genes -->
<!-- ```{r echo=FALSE, message=FALSE} -->
<!-- library(biomaRt) -->
<!-- library(GenomicRanges) -->
<!-- ensembl <- useEnsembl(biomart="ENSEMBL_MART_ENSEMBL", dataset="hsapiens_gene_ensembl") -->
<!-- G_list <- getBM(filters= "chromosome_name", attributes= c("hgnc_symbol","chromosome_name","start_position","end_position","gene_biotype"), values=1:22, mart=ensembl) -->
<!-- G_list <- G_list[G_list$hgnc_symbol!="",] -->
<!-- G_list <- G_list[G_list$gene_biotype %in% c("protein_coding","lncRNA"),] -->
<!-- G_list$start <- G_list$start_position -->
<!-- G_list$end <- G_list$end_position -->
<!-- G_list_granges <- makeGRangesFromDataFrame(G_list, keep.extra.columns=T) -->
<!-- #remove genes without imputed expression from gene lists -->
<!-- known_annotations <- known_annotations[known_annotations %in% ctwas_gene_res$genename] -->
<!-- known_annotations_positions <- G_list[G_list$hgnc_symbol %in% known_annotations,] -->
<!-- half_window <- 1000000 -->
<!-- known_annotations_positions$start <- known_annotations_positions$start_position - half_window -->
<!-- known_annotations_positions$end <- known_annotations_positions$end_position + half_window -->
<!-- known_annotations_positions$start[known_annotations_positions$start<1] <- 1 -->
<!-- known_annotations_granges <- makeGRangesFromDataFrame(known_annotations_positions, keep.extra.columns=T) -->
<!-- bystanders <- findOverlaps(known_annotations_granges,G_list_granges) -->
<!-- bystanders <- unique(subjectHits(bystanders)) -->
<!-- bystanders <- G_list$hgnc_symbol[bystanders] -->
<!-- bystanders <- unique(bystanders[!(bystanders %in% known_annotations)]) -->
<!-- unrelated_genes <- bystanders -->
<!-- #save gene lists -->
<!-- save(known_annotations, file=paste0(results_dir, "/known_annotations.Rd")) -->
<!-- save(unrelated_genes, file=paste0(results_dir, "/bystanders.Rd")) -->
<!-- ``` -->
<!-- ```{r echo=FALSE} -->
<!-- load(paste0(results_dir, "/known_annotations.Rd")) -->
<!-- load(paste0(results_dir, "/bystanders.Rd")) -->
<!-- #remove genes without imputed expression from bystander list -->
<!-- unrelated_genes <- unrelated_genes[unrelated_genes %in% ctwas_gene_res$genename] -->
<!-- #subset results to genes in known annotations or bystanders -->
<!-- ctwas_gene_res_subset <- ctwas_gene_res[ctwas_gene_res$genename %in% c(known_annotations, unrelated_genes),] -->
<!-- #assign ctwas and TWAS genes -->
<!-- ctwas_genes <- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip>0.8] -->
<!-- twas_genes <- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)>sig_thresh] -->
<!-- #sensitivity / recall -->
<!-- sensitivity <- rep(NA,2) -->
<!-- names(sensitivity) <- c("ctwas", "TWAS") -->
<!-- sensitivity["ctwas"] <- sum(ctwas_genes %in% known_annotations)/length(known_annotations) -->
<!-- sensitivity["TWAS"] <- sum(twas_genes %in% known_annotations)/length(known_annotations) -->
<!-- #specificity / (1 - False Positive Rate) -->
<!-- specificity <- rep(NA,2) -->
<!-- names(specificity) <- c("ctwas", "TWAS") -->
<!-- specificity["ctwas"] <- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes) -->
<!-- specificity["TWAS"] <- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes) -->
<!-- #precision / PPV / (1 - False Discovery Rate) -->
<!-- precision <- rep(NA,2) -->
<!-- names(precision) <- c("ctwas", "TWAS") -->
<!-- precision["ctwas"] <- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes) -->
<!-- precision["TWAS"] <- sum(twas_genes %in% known_annotations)/length(twas_genes) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #number of genes in known annotations (with imputed expression) -->
<!-- print(length(known_annotations)) -->
<!-- #number of bystander genes (with imputed expression) -->
<!-- print(length(unrelated_genes)) -->
<!-- #subset results to genes in known annotations or bystanders -->
<!-- ctwas_gene_res_subset <- ctwas_gene_res[ctwas_gene_res$genename %in% c(known_annotations, unrelated_genes),] -->
<!-- #assign ctwas and TWAS genes -->
<!-- ctwas_genes <- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip>0.8] -->
<!-- twas_genes <- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)>sig_thresh] -->
<!-- #significance threshold for TWAS -->
<!-- print(sig_thresh) -->
<!-- #number of ctwas genes (in known annotations or bystanders) -->
<!-- length(ctwas_genes) -->
<!-- #number of TWAS genes (in known annotations or bystanders) -->
<!-- length(twas_genes) -->
<!-- #sensitivity / recall -->
<!-- sensitivity -->
<!-- #specificity / (1 - False Positive Rate) -->
<!-- specificity -->
<!-- #precision / PPV / (1 - False Discovery Rate) -->
<!-- precision -->
<!-- ``` -->
<!-- ```{r echo=FALSE} -->
<!-- #store sensitivity and specificity calculations for plots -->
<!-- sensitivity_plot <- sensitivity -->
<!-- specificity_plot <- specificity -->
<!-- #precision / PPV by PIP threshold -->
<!-- pip_range <- c(0.5, 0.8, 1) -->
<!-- precision_range <- rep(NA, length(pip_range)) -->
<!-- number_detected <- rep(NA, length(pip_range)) -->
<!-- for (i in 1:length(pip_range)){ -->
<!--   pip_upper <- pip_range[i] -->
<!--   if (i==1){ -->
<!--     pip_lower <- 0 -->
<!--   } else { -->
<!--     pip_lower <- pip_range[i-1] -->
<!--   } -->
<!--   #assign ctwas genes using PIP threshold -->
<!--   ctwas_genes <- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip>=pip_lower] -->
<!--   number_detected[i] <- length(ctwas_genes) -->
<!--   precision_range[i] <- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes) -->
<!-- } -->
<!-- names(precision_range) <- paste0(">= ", c(0, pip_range[-length(pip_range)])) -->
<!-- precision_range <- precision_range*100 -->
<!-- precision_range <- c(precision_range, precision["TWAS"]*100) -->
<!-- names(precision_range)[4] <- "TWAS Bonferroni" -->
<!-- number_detected <- c(number_detected, length(twas_genes)) -->
<!-- barplot(precision_range, ylim=c(0,100), main="Precision for Distinguishing Silver Standard and Bystander Genes", xlab="PIP Threshold for Detection", ylab="% of Detected Genes in Silver Standard") -->
<!-- abline(h=20, lty=2) -->
<!-- abline(h=40, lty=2) -->
<!-- abline(h=60, lty=2) -->
<!-- abline(h=80, lty=2) -->
<!-- xx <- barplot(precision_range, add=T, col=c(rep("darkgrey",3), "white")) -->
<!-- text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, " detected"), pos = 3, cex=0.8) -->
<!-- ``` -->
<!-- ```{r echo=FALSE} -->
<!-- barplot(100-precision_range, ylim=c(0,100), main="False Discovery Rate for Distinguishing Silver Standard and Bystander Genes", xlab="PIP Threshold for Detection", ylab="% Bystanders in Detected Genes") -->
<!-- abline(h=20, lty=2) -->
<!-- abline(h=40, lty=2) -->
<!-- abline(h=60, lty=2) -->
<!-- abline(h=80, lty=2) -->
<!-- xx <- barplot(100-precision_range, add=T, col=c(rep("darkgrey",3), "white")) -->
<!-- text(x = xx, y = rep(0, length(number_detected)), label = paste0(number_detected, " detected"), pos = 3, cex=0.8) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- pip_range <- (0:1000)/1000 -->
<!-- sensitivity <- rep(NA, length(pip_range)) -->
<!-- specificity <- rep(NA, length(pip_range)) -->
<!-- for (index in 1:length(pip_range)){ -->
<!--   pip <- pip_range[index] -->
<!--   ctwas_genes <- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip>=pip] -->
<!--   sensitivity[index] <- sum(ctwas_genes %in% known_annotations)/length(known_annotations) -->
<!--   specificity[index] <- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes) -->
<!-- } -->
<!-- plot(1-specificity, sensitivity, type="l", xlim=c(0,1), ylim=c(0,1), main="", xlab="1 - Specificity", ylab="Sensitivity") -->
<!-- title(expression("ROC Curve for cTWAS (black) and TWAS (" * phantom("red") * ")")) -->
<!-- title(expression(phantom("ROC Curve for cTWAS (black) and TWAS (") * "red" * phantom(")")), col.main="red") -->
<!-- sig_thresh_range <- seq(from=0, to=max(abs(ctwas_gene_res_subset$z)), length.out=length(pip_range)) -->
<!-- for (index in 1:length(sig_thresh_range)){ -->
<!--   sig_thresh_plot <- sig_thresh_range[index] -->
<!--   twas_genes <- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)>=sig_thresh_plot] -->
<!--   sensitivity[index] <- sum(twas_genes %in% known_annotations)/length(known_annotations) -->
<!--   specificity[index] <- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes) -->
<!-- } -->
<!-- lines(1-specificity, sensitivity, xlim=c(0,1), ylim=c(0,1), col="red", lty=1) -->
<!-- abline(a=0,b=1,lty=3) -->
<!-- #add previously computed points from the analysis -->
<!-- ctwas_genes <- ctwas_gene_res_subset$genename[ctwas_gene_res_subset$susie_pip>0.8] -->
<!-- twas_genes <- ctwas_gene_res_subset$genename[abs(ctwas_gene_res_subset$z)>sig_thresh] -->
<!-- points(1-specificity_plot["ctwas"], sensitivity_plot["ctwas"], pch=21, bg="black") -->
<!-- points(1-specificity_plot["TWAS"], sensitivity_plot["TWAS"], pch=21, bg="red") -->
<!-- ``` -->
<!-- ## Undetected silver standard genes have low TWAS z-scores or stronger signal from nearby variants -->
<!-- ```{r echo=FALSE,message=FALSE} -->
<!-- known_annotations <- read_xlsx("data/summary_known_genes_annotations.xlsx", sheet="SCZ") -->
<!-- known_annotations <- unique(known_annotations$`Gene Symbol`) -->
<!-- #categorize silver standard genes by case -->
<!-- silver_standard_case <- c() -->
<!-- uncertain_regions <- matrix(NA, 0, 2) -->
<!-- for (i in 1:length(known_annotations)){ -->
<!--   current_gene <- known_annotations[i] -->
<!--   if (current_gene %in% ctwas_gene_res$genename) { -->
<!--     if (ctwas_gene_res$susie_pip[ctwas_gene_res$genename == current_gene] > 0.8){ -->
<!--       silver_standard_case <- c(silver_standard_case, "Detected (PIP > 0.8)") -->
<!--     } else { -->
<!--       if (abs(ctwas_gene_res$z[ctwas_gene_res$genename == current_gene]) < sig_thresh){ -->
<!--         silver_standard_case <- c(silver_standard_case, "Insignificant z-score") -->
<!--       } else { -->
<!--         current_region <- ctwas_gene_res$region_tag[ctwas_gene_res$genename == current_gene] -->
<!--         current_gene_res <- ctwas_gene_res[ctwas_gene_res$region_tag==current_region,] -->
<!--         current_snp_res <- ctwas_snp_res[ctwas_snp_res$region_tag==current_region,] -->
<!--         if (any(current_gene_res$susie_pip>0.8)){ -->
<!--           if (any(current_gene_res$genename[current_gene_res$susie_pip>0.8] %in% known_annotations)){ -->
<!--             silver_standard_case <- c(silver_standard_case, "Nearby Silver Standard Gene") -->
<!--           } else { -->
<!--             silver_standard_case <- c(silver_standard_case, "Nearby Bystander Gene") -->
<!--           } -->
<!--         } else { -->
<!--           #if (any(current_snp_res$susie_pip>0.8)){ -->
<!--           if (sum(current_snp_res$susie_pip)>0.8){ -->
<!--             silver_standard_case <- c(silver_standard_case, "Nearby SNP(s)") -->
<!--           } else { -->
<!--             silver_standard_case <- c(silver_standard_case, "Inconclusive") -->
<!--             uncertain_regions <- rbind(uncertain_regions, c(current_gene, ctwas_gene_res$region_tag[ctwas_gene_res$genename == current_gene])) -->
<!--             print(c(current_gene, ctwas_gene_res$region_tag[ctwas_gene_res$genename == current_gene])) -->
<!--           } -->
<!--         } -->
<!--       } -->
<!--     } -->
<!--   } else { -->
<!--     silver_standard_case <- c(silver_standard_case, "Not Imputed") -->
<!--   } -->
<!-- } -->
<!-- names(silver_standard_case) <- known_annotations -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #table of outcomes for silver standard genes -->
<!-- -sort(-table(silver_standard_case)) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- #show inconclusive genes -->
<!-- silver_standard_case[silver_standard_case=="Inconclusive"] -->
<!-- ``` -->
<!-- ```{r echo=FALSE} -->
<!-- #pie chart of outcomes for silver standard genes -->
<!-- df <- data.frame(-sort(-table(silver_standard_case))) -->
<!-- names(df) <- c("Outcome", "Frequency") -->
<!-- #df <- df[df$Outcome!="Not Imputed",] #exclude genes not imputed -->
<!-- df$Outcome <- droplevels(df$Outcome) #exclude genes not imputed -->
<!-- pie <- ggplot(df, aes(x="", y=Frequency, fill=Outcome)) + geom_bar(width = 1, stat = "identity") -->
<!-- pie <- pie + coord_polar("y", start=0) + theme_minimal() + theme(axis.title.y=element_blank()) -->
<!-- pie -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8        highr_0.9         jquerylib_0.1.4   compiler_3.6.1   
 [5] pillar_1.6.4      later_0.8.0       git2r_0.26.1      tools_3.6.1      
 [9] getPass_0.2-2     bit_4.0.4         digest_0.6.29     memoise_2.0.1    
[13] RSQLite_2.2.8     evaluate_0.14     tibble_3.1.6      lifecycle_1.0.1  
[17] pkgconfig_2.0.3   rlang_1.0.1       DBI_1.1.2         cli_3.1.0        
[21] rstudioapi_0.13   yaml_2.2.1        xfun_0.29         fastmap_1.1.0    
[25] httr_1.4.2        stringr_1.4.0     knitr_1.36        fs_1.5.2         
[29] vctrs_0.3.8       bit64_4.0.5       rprojroot_2.0.2   data.table_1.14.2
[33] glue_1.6.2        R6_2.5.1          processx_3.5.2    fansi_1.0.2      
[37] rmarkdown_2.11    blob_1.2.2        callr_3.7.0       magrittr_2.0.2   
[41] whisker_0.3-2     ps_1.6.0          promises_1.0.1    htmltools_0.5.2  
[45] ellipsis_0.3.2    httpuv_1.5.1      utf8_1.2.2        stringi_1.7.6    
[49] cachem_1.0.6      crayon_1.5.0     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
