<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>T2D - Adipose Visceral Omentum</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">T2D - Adipose Visceral Omentum</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree87fee8b56a4f94471388b3fc74b464d3b6efdd41targetblank87fee8ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/87fee8b56a4f94471388b3fc74b464d3b6efdd41" target="_blank">87fee8b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree87fee8b56a4f94471388b3fc74b464d3b6efdd41targetblank87fee8ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/87fee8b56a4f94471388b3fc74b464d3b6efdd41" target="_blank">87fee8b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/T2D_Adipose_Visceral_Omentum.Rmd</code>) and HTML (<code>docs/T2D_Adipose_Visceral_Omentum.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/87fee8b56a4f94471388b3fc74b464d3b6efdd41/analysis/T2D_Adipose_Visceral_Omentum.Rmd" target="_blank">87fee8b</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>qclist_all &lt;- list()

qc_files &lt;- paste0(results_dir, &quot;/&quot;, list.files(results_dir, pattern=&quot;exprqc.Rd&quot;))

for (i in 1:length(qc_files)){
  load(qc_files[i])
  chr &lt;- unlist(strsplit(rev(unlist(strsplit(qc_files[i], &quot;_&quot;)))[1], &quot;[.]&quot;))[1]
  qclist_all[[chr]] &lt;- cbind(do.call(rbind, lapply(qclist,unlist)), as.numeric(substring(chr,4)))
}

qclist_all &lt;- data.frame(do.call(rbind, qclist_all))
colnames(qclist_all)[ncol(qclist_all)] &lt;- &quot;chr&quot;

rm(qclist, wgtlist, z_gene_chr)

#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 12185</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1194  835  712  475  563  699  615  440  443  493  756  683  238  409  386  560 
  17   18   19   20   21   22 
 761  189  944  358  140  292 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 8998</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.7384489</code></pre>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<pre class="r"><code>library(ggplot2)
library(cowplot)</code></pre>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<pre class="r"><code>load(paste0(results_dir, &quot;/&quot;, analysis_id, &quot;_ctwas.s2.susieIrssres.Rd&quot;))
  
df &lt;- data.frame(niter = rep(1:ncol(group_prior_rec), 2),
                 value = c(group_prior_rec[1,], group_prior_rec[2,]),
                 group = rep(c(&quot;Gene&quot;, &quot;SNP&quot;), each = ncol(group_prior_rec)))
df$group &lt;- as.factor(df$group)

df$value[df$group==&quot;SNP&quot;] &lt;- df$value[df$group==&quot;SNP&quot;]*thin #adjust parameter to account for thin argument

p_pi &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(pi)) +
  ggtitle(&quot;Prior mean&quot;) +
  theme_cowplot()

df &lt;- data.frame(niter = rep(1:ncol(group_prior_var_rec), 2),
                 value = c(group_prior_var_rec[1,], group_prior_var_rec[2,]),
                 group = rep(c(&quot;Gene&quot;, &quot;SNP&quot;), each = ncol(group_prior_var_rec)))
df$group &lt;- as.factor(df$group)
p_sigma2 &lt;- ggplot(df, aes(x=niter, y=value, group=group)) +
  geom_line(aes(color=group)) +
  geom_point(aes(color=group)) +
  xlab(&quot;Iteration&quot;) + ylab(bquote(sigma^2)) +
  ggtitle(&quot;Prior variance&quot;) +
  theme_cowplot()

plot_grid(p_pi, p_sigma2)</code></pre>
<p><img src="figure/T2D_Adipose_Visceral_Omentum.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>        gene          snp 
0.0025614131 0.0001731345 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code>    gene      snp 
4.847456 1.566893 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 337159</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   12185 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>        gene          snp 
0.0004487292 0.0060627844 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.00382862 0.10984297</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<pre class="r"><code>#distribution of PIPs
hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main=&quot;Distribution of Gene PIPs&quot;)</code></pre>
<p><img src="figure/T2D_Adipose_Visceral_Omentum.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#genes with PIP&gt;0.8 or 20 highest PIPs
head(ctwas_gene_res[order(-ctwas_gene_res$susie_pip),report_cols], max(sum(ctwas_gene_res$susie_pip&gt;0.8), 20))</code></pre>
<pre><code>          genename region_tag susie_pip      mu2          PVE         z
3645         CCND2       12_4 0.9946137 28.50621 8.409285e-05  5.657050
4051          DAW1      2_134 0.2557105 38.53316 2.922459e-05  3.995461
3945         KLHL7       7_20 0.2114448 35.56377 2.230335e-05  3.787363
7075          NUS1       6_78 0.1835593 34.68276 1.888232e-05  3.716370
14409  RP6-65G23.5      14_33 0.1742867 32.72766 1.691782e-05  3.369949
2121        NIPAL2       8_67 0.1635638 35.32732 1.713812e-05 -3.401950
8191         AGGF1       5_45 0.1619237 32.70607 1.570739e-05 -3.452706
13545  RP3-473L9.4      12_67 0.1553368 31.93831 1.471471e-05 -3.298735
8858       CCDC88B      11_36 0.1460633 31.69211 1.372959e-05 -3.360511
14565    LINC01126       2_27 0.1455878 33.69123 1.454812e-05  4.291921
6837         NPAS3       14_8 0.1353112 34.30489 1.376749e-05 -3.716549
9597          ARV1      1_118 0.1341710 31.39587 1.249385e-05  3.273664
12323       KCTD11       17_6 0.1333512 30.95776 1.224424e-05  3.073744
8090        YEATS2      3_112 0.1302892 31.31955 1.210289e-05 -3.202339
32           MTMR7       8_18 0.1291267 31.32164 1.199570e-05 -3.429825
6807         ABCB9      12_75 0.1274801 31.46364 1.189643e-05  3.159926
13809        IKBKE      1_105 0.1256552 30.94571 1.153310e-05  3.103061
14290 RP11-535A5.1      18_11 0.1253370 30.32443 1.127294e-05 -2.997627
11993      PHACTR4       1_19 0.1200516 31.70775 1.129012e-05 -3.507332
2524          HPS1      10_62 0.1167168 30.48916 1.055465e-05  3.123886
      num_eqtl
3645         1
4051         2
3945         3
7075         1
14409        1
2121         3
8191         1
13545        1
8858         2
14565        2
6837         1
9597         2
12323        1
8090         1
32           1
6807         1
13809        2
14290        1
11993        4
2524         3</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<pre class="r"><code>#plot PIP vs effect size
plot(ctwas_gene_res$susie_pip, ctwas_gene_res$mu2, xlab=&quot;PIP&quot;, ylab=&quot;mu^2&quot;, main=&quot;Gene PIPs vs Effect Size&quot;)</code></pre>
<p><img src="figure/T2D_Adipose_Visceral_Omentum.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#genes with 20 largest effect sizes
head(ctwas_gene_res[order(-ctwas_gene_res$mu2),report_cols],20)</code></pre>
<pre><code>         genename region_tag  susie_pip      mu2          PVE         z
4051         DAW1      2_134 0.25571048 38.53316 2.922459e-05  3.995461
3945        KLHL7       7_20 0.21144483 35.56377 2.230335e-05  3.787363
2121       NIPAL2       8_67 0.16356382 35.32732 1.713812e-05 -3.401950
7075         NUS1       6_78 0.18355933 34.68276 1.888232e-05  3.716370
6837        NPAS3       14_8 0.13531118 34.30489 1.376749e-05 -3.716549
14565   LINC01126       2_27 0.14558776 33.69123 1.454812e-05  4.291921
14409 RP6-65G23.5      14_33 0.17428667 32.72766 1.691782e-05  3.369949
8191        AGGF1       5_45 0.16192374 32.70607 1.570739e-05 -3.452706
13545 RP3-473L9.4      12_67 0.15533683 31.93831 1.471471e-05 -3.298735
11993     PHACTR4       1_19 0.12005158 31.70775 1.129012e-05 -3.507332
8858      CCDC88B      11_36 0.14606327 31.69211 1.372959e-05 -3.360511
6807        ABCB9      12_75 0.12748008 31.46364 1.189643e-05  3.159926
9597         ARV1      1_118 0.13417100 31.39587 1.249385e-05  3.273664
32          MTMR7       8_18 0.12912666 31.32164 1.199570e-05 -3.429825
8090       YEATS2      3_112 0.13028920 31.31955 1.210289e-05 -3.202339
13443  AP001257.1      11_34 0.09367297 31.20250 8.668999e-06  3.366387
12319      ZNF888      19_36 0.10353254 31.03883 9.531198e-06 -3.029676
12323      KCTD11       17_6 0.13335121 30.95776 1.224424e-05  3.073744
13809       IKBKE      1_105 0.12565524 30.94571 1.153310e-05  3.103061
2524         HPS1      10_62 0.11671675 30.48916 1.055465e-05  3.123886
      num_eqtl
4051         2
3945         3
2121         3
7075         1
6837         1
14565        2
14409        1
8191         1
13545        1
11993        4
8858         2
6807         1
9597         2
32           1
8090         1
13443        2
12319        1
12323        1
13809        2
2524         3</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre class="r"><code>#genes with 20 highest pve
head(ctwas_gene_res[order(-ctwas_gene_res$PVE),report_cols],20)</code></pre>
<pre><code>          genename region_tag susie_pip      mu2          PVE         z
3645         CCND2       12_4 0.9946137 28.50621 8.409285e-05  5.657050
4051          DAW1      2_134 0.2557105 38.53316 2.922459e-05  3.995461
3945         KLHL7       7_20 0.2114448 35.56377 2.230335e-05  3.787363
7075          NUS1       6_78 0.1835593 34.68276 1.888232e-05  3.716370
2121        NIPAL2       8_67 0.1635638 35.32732 1.713812e-05 -3.401950
14409  RP6-65G23.5      14_33 0.1742867 32.72766 1.691782e-05  3.369949
8191         AGGF1       5_45 0.1619237 32.70607 1.570739e-05 -3.452706
13545  RP3-473L9.4      12_67 0.1553368 31.93831 1.471471e-05 -3.298735
14565    LINC01126       2_27 0.1455878 33.69123 1.454812e-05  4.291921
6837         NPAS3       14_8 0.1353112 34.30489 1.376749e-05 -3.716549
8858       CCDC88B      11_36 0.1460633 31.69211 1.372959e-05 -3.360511
9597          ARV1      1_118 0.1341710 31.39587 1.249385e-05  3.273664
12323       KCTD11       17_6 0.1333512 30.95776 1.224424e-05  3.073744
8090        YEATS2      3_112 0.1302892 31.31955 1.210289e-05 -3.202339
32           MTMR7       8_18 0.1291267 31.32164 1.199570e-05 -3.429825
6807         ABCB9      12_75 0.1274801 31.46364 1.189643e-05  3.159926
13809        IKBKE      1_105 0.1256552 30.94571 1.153310e-05  3.103061
11993      PHACTR4       1_19 0.1200516 31.70775 1.129012e-05 -3.507332
14290 RP11-535A5.1      18_11 0.1253370 30.32443 1.127294e-05 -2.997627
2524          HPS1      10_62 0.1167168 30.48916 1.055465e-05  3.123886
      num_eqtl
3645         1
4051         2
3945         3
7075         1
2121         3
14409        1
8191         1
13545        1
14565        2
6837         1
8858         2
9597         2
12323        1
8090         1
32           1
6807         1
13809        2
11993        4
14290        1
2524         3</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre class="r"><code>#genes with 20 largest z scores
head(ctwas_gene_res[order(-abs(ctwas_gene_res$z)),report_cols],20)</code></pre>
<pre><code>         genename region_tag  susie_pip      mu2          PVE         z
3645        CCND2       12_4 0.99461370 28.50621 8.409285e-05  5.657050
14565   LINC01126       2_27 0.14558776 33.69123 1.454812e-05  4.291921
4051         DAW1      2_134 0.25571048 38.53316 2.922459e-05  3.995461
3945        KLHL7       7_20 0.21144483 35.56377 2.230335e-05  3.787363
6837        NPAS3       14_8 0.13531118 34.30489 1.376749e-05 -3.716549
7075         NUS1       6_78 0.18355933 34.68276 1.888232e-05  3.716370
7378        AP3S2      15_41 0.10383473 28.94874 8.915332e-06 -3.581700
1715         RBX1      22_17 0.08266243 26.75251 6.559005e-06 -3.521311
11993     PHACTR4       1_19 0.12005158 31.70775 1.129012e-05 -3.507332
12306      PPP1CB       2_17 0.07525356 26.87607 5.998712e-06  3.490303
9476       DNAJB7      22_17 0.07386308 25.78841 5.649595e-06  3.462008
5668       CNOT6L       4_52 0.10237070 29.01722 8.810423e-06  3.460483
8191        AGGF1       5_45 0.16192374 32.70607 1.570739e-05 -3.452706
13095       ARPIN      15_41 0.10007519 28.63082 8.498172e-06 -3.432049
32          MTMR7       8_18 0.12912666 31.32164 1.199570e-05 -3.429825
2121       NIPAL2       8_67 0.16356382 35.32732 1.713812e-05 -3.401950
14409 RP6-65G23.5      14_33 0.17428667 32.72766 1.691782e-05  3.369949
13443  AP001257.1      11_34 0.09367297 31.20250 8.668999e-06  3.366387
8858      CCDC88B      11_36 0.14606327 31.69211 1.372959e-05 -3.360511
10661       SH2D7      15_36 0.07646632 26.65208 6.044585e-06 -3.348970
      num_eqtl
3645         1
14565        2
4051         2
3945         3
6837         1
7075         1
7378         1
1715         1
11993        4
12306        1
9476         1
5668         1
8191         1
13095        2
32           1
2121         3
14409        1
13443        2
8858         2
10661        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<pre class="r"><code>#set nominal signifiance threshold for z scores
alpha &lt;- 0.05

#bonferroni adjusted threshold for z scores
sig_thresh &lt;- qnorm(1-(alpha/nrow(ctwas_gene_res)/2), lower=T)

#Q-Q plot for z scores
obs_z &lt;- ctwas_gene_res$z[order(ctwas_gene_res$z)]
exp_z &lt;- qnorm((1:nrow(ctwas_gene_res))/nrow(ctwas_gene_res))

plot(exp_z, obs_z, xlab=&quot;Expected z&quot;, ylab=&quot;Observed z&quot;, main=&quot;Gene z score Q-Q plot&quot;)
abline(a=0,b=1)</code></pre>
<p><img src="figure/T2D_Adipose_Visceral_Omentum.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#plot z score vs PIP
plot(abs(ctwas_gene_res$z), ctwas_gene_res$susie_pip, xlab=&quot;abs(z)&quot;, ylab=&quot;PIP&quot;)
abline(v=sig_thresh, col=&quot;red&quot;, lty=2)</code></pre>
<p><img src="figure/T2D_Adipose_Visceral_Omentum.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#proportion of significant z scores
mean(abs(ctwas_gene_res$z) &gt; sig_thresh)</code></pre>
<pre><code>[1] 8.206812e-05</code></pre>
<pre class="r"><code>#genes with most significant z scores
head(ctwas_gene_res[order(-abs(ctwas_gene_res$z)),report_cols],20)</code></pre>
<pre><code>         genename region_tag  susie_pip      mu2          PVE         z
3645        CCND2       12_4 0.99461370 28.50621 8.409285e-05  5.657050
14565   LINC01126       2_27 0.14558776 33.69123 1.454812e-05  4.291921
4051         DAW1      2_134 0.25571048 38.53316 2.922459e-05  3.995461
3945        KLHL7       7_20 0.21144483 35.56377 2.230335e-05  3.787363
6837        NPAS3       14_8 0.13531118 34.30489 1.376749e-05 -3.716549
7075         NUS1       6_78 0.18355933 34.68276 1.888232e-05  3.716370
7378        AP3S2      15_41 0.10383473 28.94874 8.915332e-06 -3.581700
1715         RBX1      22_17 0.08266243 26.75251 6.559005e-06 -3.521311
11993     PHACTR4       1_19 0.12005158 31.70775 1.129012e-05 -3.507332
12306      PPP1CB       2_17 0.07525356 26.87607 5.998712e-06  3.490303
9476       DNAJB7      22_17 0.07386308 25.78841 5.649595e-06  3.462008
5668       CNOT6L       4_52 0.10237070 29.01722 8.810423e-06  3.460483
8191        AGGF1       5_45 0.16192374 32.70607 1.570739e-05 -3.452706
13095       ARPIN      15_41 0.10007519 28.63082 8.498172e-06 -3.432049
32          MTMR7       8_18 0.12912666 31.32164 1.199570e-05 -3.429825
2121       NIPAL2       8_67 0.16356382 35.32732 1.713812e-05 -3.401950
14409 RP6-65G23.5      14_33 0.17428667 32.72766 1.691782e-05  3.369949
13443  AP001257.1      11_34 0.09367297 31.20250 8.668999e-06  3.366387
8858      CCDC88B      11_36 0.14606327 31.69211 1.372959e-05 -3.360511
10661       SH2D7      15_36 0.07646632 26.65208 6.044585e-06 -3.348970
      num_eqtl
3645         1
14565        2
4051         2
3945         3
6837         1
7075         1
7378         1
1715         1
11993        4
12306        1
9476         1
5668         1
8191         1
13095        2
32           1
2121         3
14409        1
13443        2
8858         2
10661        1</code></pre>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>library(&quot;readxl&quot;)

known_annotations &lt;- read_xlsx(&quot;data/summary_known_genes_annotations.xlsx&quot;, sheet=&quot;T2D&quot;)
known_annotations &lt;- unique(known_annotations$`Gene Symbol`)

unrelated_genes &lt;- ctwas_gene_res$genename[!(ctwas_gene_res$genename %in% known_annotations)]

#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 72</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>#assign ctwas, TWAS, and bystander genes
ctwas_genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res$genename[abs(ctwas_gene_res$z)&gt;sig_thresh]
novel_genes &lt;- ctwas_genes[!(ctwas_genes %in% twas_genes)]

#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.606075</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>[1] genename   region_tag susie_pip  mu2        PVE        z          num_eqtl  
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>#sensitivity / recall
sensitivity &lt;- rep(NA,2)
names(sensitivity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
sensitivity[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
sensitivity</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#specificity
specificity &lt;- rep(NA,2)
names(specificity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
specificity[&quot;ctwas&quot;] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity[&quot;TWAS&quot;] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
specificity</code></pre>
<pre><code>    ctwas      TWAS 
0.9999177 0.9999177 </code></pre>
<pre class="r"><code>#precision / PPV
precision &lt;- rep(NA,2)
names(precision) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
precision[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(twas_genes)
precision</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#ROC curves

pip_range &lt;- (0:1000)/1000
sensitivity &lt;- rep(NA, length(pip_range))
specificity &lt;- rep(NA, length(pip_range))

for (index in 1:length(pip_range)){
  pip &lt;- pip_range[index]
  ctwas_genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;=pip]
  sensitivity[index] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
}

plot(1-specificity, sensitivity, type=&quot;l&quot;, xlim=c(0,1), ylim=c(0,1))

sig_thresh_range &lt;- seq(from=0, to=max(abs(ctwas_gene_res$z)), length.out=length(pip_range))

for (index in 1:length(sig_thresh_range)){
  sig_thresh_plot &lt;- sig_thresh_range[index]
  twas_genes &lt;- ctwas_gene_res$genename[abs(ctwas_gene_res$z)&gt;=sig_thresh_plot]
  sensitivity[index] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
  specificity[index] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
}

lines(1-specificity, sensitivity, xlim=c(0,1), ylim=c(0,1), col=&quot;red&quot;, lty=2)</code></pre>
<p><img src="figure/T2D_Adipose_Visceral_Omentum.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes---bystanders-only" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes - bystanders only</h2>
<p>This section first uses all silver standard genes to identify bystander genes within 1Mb. The silver standard and bystander gene lists are then subset to only genes with imputed expression in this analysis. Then, the ctwas and TWAS gene lists from this analysis are subset to only genes that are in the (subset) silver standard and bystander genes. These gene lists are then used to compute sensitivity, specificity and precision for ctwas and TWAS.</p>
<pre class="r"><code>library(biomaRt)
library(GenomicRanges)</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>ensembl &lt;- useEnsembl(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
G_list &lt;- getBM(filters= &quot;chromosome_name&quot;, attributes= c(&quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;end_position&quot;,&quot;gene_biotype&quot;), values=1:22, mart=ensembl)
G_list &lt;- G_list[G_list$hgnc_symbol!=&quot;&quot;,]
G_list &lt;- G_list[G_list$gene_biotype %in% c(&quot;protein_coding&quot;,&quot;lncRNA&quot;),]
G_list$start &lt;- G_list$start_position
G_list$end &lt;- G_list$end_position
G_list_granges &lt;- makeGRangesFromDataFrame(G_list, keep.extra.columns=T)

known_annotations_positions &lt;- G_list[G_list$hgnc_symbol %in% known_annotations,]
half_window &lt;- 1000000
known_annotations_positions$start &lt;- known_annotations_positions$start_position - half_window
known_annotations_positions$end &lt;- known_annotations_positions$end_position + half_window
known_annotations_positions$start[known_annotations_positions$start&lt;1] &lt;- 1
known_annotations_granges &lt;- makeGRangesFromDataFrame(known_annotations_positions, keep.extra.columns=T)

bystanders &lt;- findOverlaps(known_annotations_granges,G_list_granges)
bystanders &lt;- unique(subjectHits(bystanders))
bystanders &lt;- G_list$hgnc_symbol[bystanders]
bystanders &lt;- bystanders[!(bystanders %in% known_annotations)]
unrelated_genes &lt;- bystanders

#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 72</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>#number of bystander genes
print(length(unrelated_genes))</code></pre>
<pre><code>[1] 1847</code></pre>
<pre class="r"><code>#number of bystander genes with imputed expression
print(sum(unrelated_genes %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 949</code></pre>
<pre class="r"><code>#remove genes without imputed expression from gene lists
known_annotations &lt;- known_annotations[known_annotations %in% ctwas_gene_res$genename]
unrelated_genes &lt;- unrelated_genes[unrelated_genes %in% ctwas_gene_res$genename]

#assign ctwas and TWAS genes
ctwas_genes &lt;- ctwas_gene_res$genename[ctwas_gene_res$susie_pip&gt;0.8]
twas_genes &lt;- ctwas_gene_res$genename[abs(ctwas_gene_res$z)&gt;sig_thresh]

#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.606075</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#number of ctwas genes in known annotations or bystanders
sum(ctwas_genes %in% c(known_annotations, unrelated_genes))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>#number of ctwas genes
length(twas_genes)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#number of TWAS genes
sum(twas_genes %in% c(known_annotations, unrelated_genes))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>#remove genes not in known or bystander lists from results
ctwas_genes &lt;- ctwas_genes[ctwas_genes %in% c(known_annotations, unrelated_genes)]
twas_genes &lt;- twas_genes[twas_genes %in% c(known_annotations, unrelated_genes)]

#sensitivity / recall
sensitivity &lt;- rep(NA,2)
names(sensitivity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
sensitivity[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(known_annotations)
sensitivity</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#specificity
specificity &lt;- rep(NA,2)
names(specificity) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
specificity[&quot;ctwas&quot;] &lt;- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity[&quot;TWAS&quot;] &lt;- sum(!(unrelated_genes %in% twas_genes))/length(unrelated_genes)
specificity</code></pre>
<pre><code>ctwas  TWAS 
    1     1 </code></pre>
<pre class="r"><code>#precision / PPV
precision &lt;- rep(NA,2)
names(precision) &lt;- c(&quot;ctwas&quot;, &quot;TWAS&quot;)
precision[&quot;ctwas&quot;] &lt;- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision[&quot;TWAS&quot;] &lt;- sum(twas_genes %in% known_annotations)/length(twas_genes)
precision</code></pre>
<pre><code>ctwas  TWAS 
  NaN   NaN </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] GenomicRanges_1.36.1 GenomeInfoDb_1.20.0  IRanges_2.18.1      
 [4] S4Vectors_0.22.1     BiocGenerics_0.30.0  biomaRt_2.40.1      
 [7] readxl_1.3.1         cowplot_1.0.0        ggplot2_3.3.5       
[10] workflowr_1.6.2     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7             prettyunits_1.1.1      assertthat_0.2.1      
 [4] rprojroot_2.0.2        digest_0.6.29          utf8_1.2.2            
 [7] R6_2.5.1               cellranger_1.1.0       RSQLite_2.2.8         
[10] evaluate_0.14          httr_1.4.2             highr_0.9             
[13] pillar_1.6.4           zlibbioc_1.30.0        progress_1.2.2        
[16] rlang_0.4.12           curl_4.3.2             data.table_1.14.2     
[19] whisker_0.3-2          jquerylib_0.1.4        blob_1.2.2            
[22] rmarkdown_2.11         labeling_0.4.2         stringr_1.4.0         
[25] RCurl_1.98-1.5         bit_4.0.4              munsell_0.5.0         
[28] compiler_3.6.1         httpuv_1.5.1           xfun_0.29             
[31] pkgconfig_2.0.3        htmltools_0.5.2        tidyselect_1.1.1      
[34] GenomeInfoDbData_1.2.1 tibble_3.1.6           XML_3.99-0.3          
[37] fansi_0.5.0            crayon_1.4.2           dplyr_1.0.7           
[40] withr_2.4.3            later_0.8.0            bitops_1.0-7          
[43] grid_3.6.1             gtable_0.3.0           lifecycle_1.0.1       
[46] DBI_1.1.1              git2r_0.26.1           magrittr_2.0.1        
[49] scales_1.1.1           stringi_1.7.6          cachem_1.0.6          
[52] XVector_0.24.0         farver_2.1.0           fs_1.5.2              
[55] promises_1.0.1         ellipsis_0.3.2         generics_0.1.1        
[58] vctrs_0.3.8            tools_3.6.1            bit64_4.0.5           
[61] Biobase_2.44.0         glue_1.5.1             purrr_0.3.4           
[64] hms_1.1.1              fastmap_1.1.0          yaml_2.2.1            
[67] AnnotationDbi_1.46.0   colorspace_2.0-2       memoise_2.0.1         
[70] knitr_1.36            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
