<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>T2D - Liver</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">T2D - Liver</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeeb13ecf45ad6ec66ccac1b6f91b74c140a7261d6targetblankeb13ecfa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6" target="_blank">eb13ecf</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeeb13ecf45ad6ec66ccac1b6f91b74c140a7261d6targetblankeb13ecfa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6" target="_blank">eb13ecf</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/T2D_Liver.Rmd</code>) and HTML (<code>docs/T2D_Liver.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6/analysis/T2D_Liver.Rmd" target="_blank">eb13ecf</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e6bc16990373312d301ab183ec936aff373d2497/docs/T2D_Liver.html" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/87fee8b56a4f94471388b3fc74b464d3b6efdd41/analysis/T2D_Liver.Rmd" target="_blank">87fee8b</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre><code>[1] 10290</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1015  728  627  403  457  577  509  392  388  394  607  572  190  353  340  483 
  17   18   19   20   21   22 
 629  151  808  290  108  269 </code></pre>
<pre><code>[1] 8215</code></pre>
<pre><code>[1] 0.7983479</code></pre>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<pre><code>
********************************************************</code></pre>
<pre><code>Note: As of version 1.0.0, cowplot does not change the</code></pre>
<pre><code>  default ggplot2 theme anymore. To recover the previous</code></pre>
<pre><code>  behavior, execute:
  theme_set(theme_cowplot())</code></pre>
<pre><code>********************************************************</code></pre>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Liver.Rmd/unnamed-chunk-4-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>        gene          snp 
0.0041007085 0.0001802713 </code></pre>
<pre><code>    gene      snp 
4.388896 1.534856 </code></pre>
<pre><code>[1] 337159</code></pre>
<pre><code>[1]   10290 7535010</code></pre>
<pre><code>        gene          snp 
0.0005492813 0.0061836269 </code></pre>
<pre><code>[1] 0.004834584 0.111688709</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Liver.Rmd/unnamed-chunk-5-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>          genename region_tag susie_pip      mu2          PVE         z
3212         CCND2       12_4 0.9959793 27.97577 8.264138e-05  5.657050
2240       SEC23IP      10_74 0.6381754 61.38049 1.161811e-04 -3.610724
12661    LINC01126       2_27 0.4142141 28.18774 3.462982e-05  4.620415
10283        MCMBP      10_74 0.3685673 60.32290 6.594232e-05  3.522402
703         GUCY2C      12_12 0.2541535 35.09246 2.645301e-05  3.878767
6307          NUS1       6_78 0.2350043 32.14897 2.240826e-05  3.716370
12541  RP6-65G23.5      14_33 0.2244479 30.24036 2.013111e-05  3.369949
5911          CIZ1       9_66 0.2067132 30.11029 1.846071e-05 -3.513905
6558         AP3S2      15_41 0.1990701 30.35974 1.792543e-05 -3.745658
7641         NDEL1       17_8 0.1907755 28.90161 1.635346e-05 -3.136833
10118        RABL6       9_74 0.1894969 30.14016 1.693998e-05  3.491221
2577        GNPTAB      12_61 0.1868989 29.62473 1.642201e-05  3.600816
4089         UBAC1       9_72 0.1742949 28.55651 1.476233e-05  3.438703
9318          LIPF      10_56 0.1709291 29.37002 1.488969e-05 -2.991710
12431 RP11-535A5.1      18_11 0.1656953 28.08816 1.380381e-05 -2.997627
12123       UPK3BL       7_63 0.1607226 28.40836 1.354217e-05  3.134982
1624       TPD52L2      20_38 0.1531083 28.20869 1.280993e-05 -3.090905
1483          RPL3      22_16 0.1466952 27.86306 1.212300e-05  3.284537
4539         ISCA1       9_44 0.1449422 27.76422 1.193563e-05  3.269765
3541       ARHGAP9      12_36 0.1430809 26.51095 1.125051e-05  2.925673
      num_eqtl
3212         1
2240         1
12661        1
10283        1
703          1
6307         1
12541        1
5911         2
6558         1
7641         1
10118        1
2577         1
4089         1
9318         1
12431        1
12123        1
1624         1
1483         1
4539         1
3541         1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Liver.Rmd/unnamed-chunk-6-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>          genename region_tag susie_pip      mu2          PVE         z
2240       SEC23IP      10_74 0.6381754 61.38049 1.161811e-04 -3.610724
10283        MCMBP      10_74 0.3685673 60.32290 6.594232e-05  3.522402
703         GUCY2C      12_12 0.2541535 35.09246 2.645301e-05  3.878767
8349          GPHN      14_32 0.1374525 33.29520 1.357374e-05 -3.426575
6307          NUS1       6_78 0.2350043 32.14897 2.240826e-05  3.716370
6558         AP3S2      15_41 0.1990701 30.35974 1.792543e-05 -3.745658
12541  RP6-65G23.5      14_33 0.2244479 30.24036 2.013111e-05  3.369949
10118        RABL6       9_74 0.1894969 30.14016 1.693998e-05  3.491221
5911          CIZ1       9_66 0.2067132 30.11029 1.846071e-05 -3.513905
5073         ETNK1      12_16 0.1407134 29.98608 1.251470e-05  3.169725
7288         AGGF1       5_45 0.1050999 29.89108 9.317706e-06 -3.154473
2577        GNPTAB      12_61 0.1868989 29.62473 1.642201e-05  3.600816
9318          LIPF      10_56 0.1709291 29.37002 1.488969e-05 -2.991710
7641         NDEL1       17_8 0.1907755 28.90161 1.635346e-05 -3.136833
4089         UBAC1       9_72 0.1742949 28.55651 1.476233e-05  3.438703
12123       UPK3BL       7_63 0.1607226 28.40836 1.354217e-05  3.134982
1624       TPD52L2      20_38 0.1531083 28.20869 1.280993e-05 -3.090905
12661    LINC01126       2_27 0.4142141 28.18774 3.462982e-05  4.620415
12431 RP11-535A5.1      18_11 0.1656953 28.08816 1.380381e-05 -2.997627
1460        PPP6R2      22_24 0.1361427 27.99436 1.130395e-05 -3.283527
      num_eqtl
2240         1
10283        1
703          1
8349         2
6307         1
6558         1
12541        1
10118        1
5911         2
5073         1
7288         2
2577         1
9318         1
7641         1
4089         1
12123        1
1624         1
12661        1
12431        1
1460         1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>          genename region_tag susie_pip      mu2          PVE         z
2240       SEC23IP      10_74 0.6381754 61.38049 1.161811e-04 -3.610724
3212         CCND2       12_4 0.9959793 27.97577 8.264138e-05  5.657050
10283        MCMBP      10_74 0.3685673 60.32290 6.594232e-05  3.522402
12661    LINC01126       2_27 0.4142141 28.18774 3.462982e-05  4.620415
703         GUCY2C      12_12 0.2541535 35.09246 2.645301e-05  3.878767
6307          NUS1       6_78 0.2350043 32.14897 2.240826e-05  3.716370
12541  RP6-65G23.5      14_33 0.2244479 30.24036 2.013111e-05  3.369949
5911          CIZ1       9_66 0.2067132 30.11029 1.846071e-05 -3.513905
6558         AP3S2      15_41 0.1990701 30.35974 1.792543e-05 -3.745658
10118        RABL6       9_74 0.1894969 30.14016 1.693998e-05  3.491221
2577        GNPTAB      12_61 0.1868989 29.62473 1.642201e-05  3.600816
7641         NDEL1       17_8 0.1907755 28.90161 1.635346e-05 -3.136833
9318          LIPF      10_56 0.1709291 29.37002 1.488969e-05 -2.991710
4089         UBAC1       9_72 0.1742949 28.55651 1.476233e-05  3.438703
12431 RP11-535A5.1      18_11 0.1656953 28.08816 1.380381e-05 -2.997627
8349          GPHN      14_32 0.1374525 33.29520 1.357374e-05 -3.426575
12123       UPK3BL       7_63 0.1607226 28.40836 1.354217e-05  3.134982
1624       TPD52L2      20_38 0.1531083 28.20869 1.280993e-05 -3.090905
5073         ETNK1      12_16 0.1407134 29.98608 1.251470e-05  3.169725
1483          RPL3      22_16 0.1466952 27.86306 1.212300e-05  3.284537
      num_eqtl
2240         1
3212         1
10283        1
12661        1
703          1
6307         1
12541        1
5911         2
6558         1
10118        1
2577         1
7641         1
9318         1
4089         1
12431        1
8349         2
12123        1
1624         1
5073         1
1483         1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag  susie_pip      mu2          PVE         z
3212        CCND2       12_4 0.99597929 27.97577 8.264138e-05  5.657050
12661   LINC01126       2_27 0.41421408 28.18774 3.462982e-05  4.620415
703        GUCY2C      12_12 0.25415347 35.09246 2.645301e-05  3.878767
6558        AP3S2      15_41 0.19907014 30.35974 1.792543e-05 -3.745658
6307         NUS1       6_78 0.23500434 32.14897 2.240826e-05  3.716370
2240      SEC23IP      10_74 0.63817542 61.38049 1.161811e-04 -3.610724
2577       GNPTAB      12_61 0.18689890 29.62473 1.642201e-05  3.600816
10283       MCMBP      10_74 0.36856732 60.32290 6.594232e-05  3.522402
1505         RBX1      22_17 0.13906601 26.28028 1.083967e-05 -3.521311
5911         CIZ1       9_66 0.20671321 30.11029 1.846071e-05 -3.513905
10118       RABL6       9_74 0.18949686 30.14016 1.693998e-05  3.491221
4089        UBAC1       9_72 0.17429489 28.55651 1.476233e-05  3.438703
10840      PPP1CB       2_17 0.08612930 23.42382 5.983755e-06  3.433773
7172        SPDYA       2_17 0.08537626 23.34966 5.912661e-06 -3.429510
8349         GPHN      14_32 0.13745250 33.29520 1.357374e-05 -3.426575
5040       CNOT6L       4_52 0.13254813 27.11893 1.066133e-05  3.423769
12541 RP6-65G23.5      14_33 0.22444790 30.24036 2.013111e-05  3.369949
1483         RPL3      22_16 0.14669520 27.86306 1.212300e-05  3.284537
1460       PPP6R2      22_24 0.13614270 27.99436 1.130395e-05 -3.283527
2417         GLRB      4_101 0.13567594 27.33830 1.100119e-05  3.269896
      num_eqtl
3212         1
12661        1
703          1
6558         1
6307         1
2240         1
2577         1
10283        1
1505         1
5911         2
10118        1
4089         1
10840        3
7172         2
8349         2
5040         1
12541        1
1483         1
1460         1
2417         1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Liver.Rmd/unnamed-chunk-9-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Liver.Rmd/unnamed-chunk-9-2.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] 0.0001943635</code></pre>
<pre><code>         genename region_tag  susie_pip      mu2          PVE         z
3212        CCND2       12_4 0.99597929 27.97577 8.264138e-05  5.657050
12661   LINC01126       2_27 0.41421408 28.18774 3.462982e-05  4.620415
703        GUCY2C      12_12 0.25415347 35.09246 2.645301e-05  3.878767
6558        AP3S2      15_41 0.19907014 30.35974 1.792543e-05 -3.745658
6307         NUS1       6_78 0.23500434 32.14897 2.240826e-05  3.716370
2240      SEC23IP      10_74 0.63817542 61.38049 1.161811e-04 -3.610724
2577       GNPTAB      12_61 0.18689890 29.62473 1.642201e-05  3.600816
10283       MCMBP      10_74 0.36856732 60.32290 6.594232e-05  3.522402
1505         RBX1      22_17 0.13906601 26.28028 1.083967e-05 -3.521311
5911         CIZ1       9_66 0.20671321 30.11029 1.846071e-05 -3.513905
10118       RABL6       9_74 0.18949686 30.14016 1.693998e-05  3.491221
4089        UBAC1       9_72 0.17429489 28.55651 1.476233e-05  3.438703
10840      PPP1CB       2_17 0.08612930 23.42382 5.983755e-06  3.433773
7172        SPDYA       2_17 0.08537626 23.34966 5.912661e-06 -3.429510
8349         GPHN      14_32 0.13745250 33.29520 1.357374e-05 -3.426575
5040       CNOT6L       4_52 0.13254813 27.11893 1.066133e-05  3.423769
12541 RP6-65G23.5      14_33 0.22444790 30.24036 2.013111e-05  3.369949
1483         RPL3      22_16 0.14669520 27.86306 1.212300e-05  3.284537
1460       PPP6R2      22_24 0.13614270 27.99436 1.130395e-05 -3.283527
2417         GLRB      4_101 0.13567594 27.33830 1.100119e-05  3.269896
      num_eqtl
3212         1
12661        1
703          1
6558         1
6307         1
2240         1
2577         1
10283        1
1505         1
5911         2
10118        1
4089         1
10840        3
7172         2
8349         2
5040         1
12541        1
1483         1
1460         1
2417         1</code></pre>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre><code>[1] 72</code></pre>
<pre><code>[1] 33</code></pre>
<pre><code>[1] 4.570782</code></pre>
<pre><code>[1] 1</code></pre>
<pre><code>[1] 2</code></pre>
<pre><code>[1] genename   region_tag susie_pip  mu2        PVE        z          num_eqtl  
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre><code>    ctwas      TWAS 
0.9999025 0.9998050 </code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Liver.Rmd/unnamed-chunk-10-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readxl_1.3.1    cowplot_1.0.0   ggplot2_3.3.5   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1  xfun_0.29         purrr_0.3.4       colorspace_2.0-2 
 [5] vctrs_0.3.8       generics_0.1.1    htmltools_0.5.2   yaml_2.2.1       
 [9] utf8_1.2.2        blob_1.2.2        rlang_0.4.12      jquerylib_0.1.4  
[13] later_0.8.0       pillar_1.6.4      glue_1.5.1        withr_2.4.3      
[17] DBI_1.1.1         bit64_4.0.5       lifecycle_1.0.1   stringr_1.4.0    
[21] cellranger_1.1.0  munsell_0.5.0     gtable_0.3.0      evaluate_0.14    
[25] memoise_2.0.1     labeling_0.4.2    knitr_1.36        fastmap_1.1.0    
[29] httpuv_1.5.1      fansi_0.5.0       highr_0.9         Rcpp_1.0.7       
[33] promises_1.0.1    scales_1.1.1      cachem_1.0.6      farver_2.1.0     
[37] fs_1.5.2          bit_4.0.4         digest_0.6.29     stringi_1.7.6    
[41] dplyr_1.0.7       rprojroot_2.0.2   grid_3.6.1        tools_3.6.1      
[45] magrittr_2.0.1    tibble_3.1.6      RSQLite_2.2.8     crayon_1.4.2     
[49] whisker_0.3-2     pkgconfig_2.0.3   ellipsis_0.3.2    data.table_1.14.2
[53] assertthat_0.2.1  rmarkdown_2.11    R6_2.5.1          git2r_0.26.1     
[57] compiler_3.6.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
