<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>T2D - Adipose Subcutaneous</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">T2D - Adipose Subcutaneous</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree5c37a5d91b99c4299f94f486608c8735de02fd62targetblank5c37a5da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/5c37a5d91b99c4299f94f486608c8735de02fd62" target="_blank">5c37a5d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree5c37a5d91b99c4299f94f486608c8735de02fd62targetblank5c37a5da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/5c37a5d91b99c4299f94f486608c8735de02fd62" target="_blank">5c37a5d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    data/AF/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/Glucose_Adipose_Subcutaneous.Rmd
    Untracked:  analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Untracked:  analysis/Splicing_Test.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_S_out/
    Untracked:  code/BMI_out/
    Untracked:  code/Glucose_out/
    Untracked:  code/LDL_S_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_analysis_S.sbatch
    Untracked:  code/run_BMI_analysis_S.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_ctwas_rss_LDR_S.R
    Untracked:  code/run_Glucose_analysis.sbatch
    Untracked:  code/run_Glucose_analysis.sh
    Untracked:  code/run_Glucose_ctwas_rss_LDR.R
    Untracked:  code/run_LDL_analysis_S.sbatch
    Untracked:  code/run_LDL_analysis_S.sh
    Untracked:  code/run_LDL_ctwas_rss_LDR_S.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/BMI/
    Untracked:  data/BMI_S/
    Untracked:  data/Glucose/
    Untracked:  data/LDL_S/
    Untracked:  data/T2D/
    Untracked:  data/TEST/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

Unstaged changes:
    Modified:   analysis/BMI_Brain_Amygdala_S.Rmd
    Modified:   analysis/BMI_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Modified:   analysis/BMI_Brain_Caudate_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellar_Hemisphere_S.Rmd
    Modified:   analysis/BMI_Brain_Cerebellum_S.Rmd
    Modified:   analysis/BMI_Brain_Cortex.Rmd
    Modified:   analysis/BMI_Brain_Cortex_S.Rmd
    Modified:   analysis/BMI_Brain_Frontal_Cortex_BA9_S.Rmd
    Modified:   analysis/BMI_Brain_Hippocampus_S.Rmd
    Modified:   analysis/BMI_Brain_Hypothalamus_S.Rmd
    Modified:   analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Putamen_basal_ganglia_S.Rmd
    Modified:   analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Modified:   analysis/BMI_Brain_Substantia_nigra_S.Rmd
    Modified:   analysis/LDL_Liver_S.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/T2D_Adipose_Subcutaneous.Rmd</code>) and HTML (<code>docs/T2D_Adipose_Subcutaneous.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/5c37a5d91b99c4299f94f486608c8735de02fd62/analysis/T2D_Adipose_Subcutaneous.Rmd" target="_blank">5c37a5d</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-26
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/3fa3a64422585dc9f94588111e57ecd6aaa3aa5e/docs/T2D_Adipose_Subcutaneous.html" target="_blank">3fa3a64</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/91f38fa8b991bcb34c11575638dc25fc81457a5f/docs/T2D_Adipose_Subcutaneous.html" target="_blank">91f38fa</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6/analysis/T2D_Adipose_Subcutaneous.Rmd" target="_blank">eb13ecf</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e6bc16990373312d301ab183ec936aff373d2497/docs/T2D_Adipose_Subcutaneous.html" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/87fee8b56a4f94471388b3fc74b464d3b6efdd41/analysis/T2D_Adipose_Subcutaneous.Rmd" target="_blank">87fee8b</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 8591</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
874 679 536 364 458 498 421 323 333 362 534 523 182 297 322 342 434 142 429 246 
 21  22 
102 190 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 4545</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.529</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/3fa3a64422585dc9f94588111e57ecd6aaa3aa5e/docs/figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-5-1.png" target="_blank">3fa3a64</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-5-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0164053 0.0003491 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
9.542 8.857 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 62892</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]    8591 5017200</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>   gene     snp 
0.02138 0.24667 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.09756 1.39401</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>      genename region_tag susie_pip   mu2       PVE      z num_eqtl
7109    LONRF1       8_15    0.9929 29.21 0.0004612 -5.425        2
5112     P2RX4      12_74    0.9615 25.94 0.0003966  5.081        1
250     ANGEL1      14_36    0.9577 22.68 0.0003453  4.529        2
7272     KCNS2       8_68    0.9085 26.21 0.0003787 -5.148        1
7654     PTH1R       3_33    0.8980 30.11 0.0004300 -5.646        1
6918   ARL14EP      11_21    0.8715 23.28 0.0003226 -4.878        2
13239  N4BP2L2      13_11    0.8441 24.34 0.0003267 -4.430        1
10030   FAM89B      11_36    0.7961 24.33 0.0003080  5.011        1
8836   FAM234A       16_1    0.7888 31.19 0.0003912  5.621        1
11712    PARVA       11_9    0.7846 20.35 0.0002539 -3.862        1
7267      GDF6       8_67    0.7821 21.36 0.0002656 -4.372        1
3924     SEPT7       7_26    0.7751 20.71 0.0002553 -4.167        1
3988    ZC3H13      13_19    0.7732 20.56 0.0002527  4.310        1
2323    DNASE2      19_10    0.7718 18.79 0.0002306 -3.744        1
4009    KBTBD4      11_29    0.7474 26.46 0.0003145 -5.098        1
3282     GRB14      2_100    0.7262 21.32 0.0002461  4.745        1
11149    SLIT1      10_62    0.7054 24.45 0.0002742  4.797        2
7490     UBE2Z      17_28    0.7027 47.08 0.0005260 -7.392        1
6245     MRPS5       2_57    0.6981 19.67 0.0002184 -3.810        2
10469 TMEM132E      17_20    0.6954 21.33 0.0002358 -3.693        2</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>           genename region_tag susie_pip    mu2       PVE       z num_eqtl
7048          JAZF1       7_23  0.010490 138.57 2.311e-05 -12.731        1
14285  RP11-395N3.2      2_133  0.035389 109.24 6.147e-05 -11.370        1
356            ANK1       8_36  0.040526  59.39 3.827e-05   8.667        1
6954        ZFP36L2       2_27  0.039417  55.29 3.465e-05   8.377        1
2723           WFS1        4_7  0.029567  53.23 2.502e-05  10.642        1
8537             C2       6_26  0.219641  52.12 1.820e-04   7.149        1
11174        KCNJ11      11_12  0.022353  49.77 1.769e-05   7.231        1
11270       NCR3LG1      11_12  0.022725  49.31 1.782e-05  -7.169        2
14659     LINC01126       2_27  0.049134  47.85 3.738e-05  -7.520        2
1417         PABPC4       1_24  0.208389  47.85 1.585e-04  -7.054        1
7490          UBE2Z      17_28  0.702743  47.08 5.260e-04  -7.392        1
3368          THADA       2_27  0.037436  45.75 2.723e-05   7.450        1
943           ZZEF1       17_4  0.090687  43.96 6.339e-05   6.917        1
10282        ZNF664      12_75  0.315661  43.32 2.174e-04  -6.452        1
7491           SNF8      17_28  0.064662  43.17 4.438e-05   6.300        1
10726         BMP8A       1_24  0.090841  42.59 6.152e-05   6.868        1
6449         CDKAL1       6_15  0.003752  40.70 2.428e-06  -8.192        1
10555        UBE2E2       3_17  0.498581  39.49 3.131e-04   6.081        1
3647         CCDC92      12_75  0.108518  38.03 6.562e-05  -5.499        5
13467 RP11-419C23.1       8_33  0.506567  37.39 3.012e-04  -6.317        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>           genename region_tag susie_pip   mu2       PVE      z num_eqtl
7490          UBE2Z      17_28    0.7027 47.08 0.0005260 -7.392        1
7109         LONRF1       8_15    0.9929 29.21 0.0004612 -5.425        2
7654          PTH1R       3_33    0.8980 30.11 0.0004300 -5.646        1
5112          P2RX4      12_74    0.9615 25.94 0.0003966  5.081        1
8836        FAM234A       16_1    0.7888 31.19 0.0003912  5.621        1
7272          KCNS2       8_68    0.9085 26.21 0.0003787 -5.148        1
250          ANGEL1      14_36    0.9577 22.68 0.0003453  4.529        2
13239       N4BP2L2      13_11    0.8441 24.34 0.0003267 -4.430        1
6918        ARL14EP      11_21    0.8715 23.28 0.0003226 -4.878        2
4009         KBTBD4      11_29    0.7474 26.46 0.0003145 -5.098        1
10555        UBE2E2       3_17    0.4986 39.49 0.0003131  6.081        1
10030        FAM89B      11_36    0.7961 24.33 0.0003080  5.011        1
13467 RP11-419C23.1       8_33    0.5066 37.39 0.0003012 -6.317        1
6645         CAMK2G      10_49    0.5495 32.01 0.0002796  5.013        1
11149         SLIT1      10_62    0.7054 24.45 0.0002742  4.797        2
7267           GDF6       8_67    0.7821 21.36 0.0002656 -4.372        1
3924          SEPT7       7_26    0.7751 20.71 0.0002553 -4.167        1
11712         PARVA       11_9    0.7846 20.35 0.0002539 -3.862        1
3988         ZC3H13      13_19    0.7732 20.56 0.0002527  4.310        1
3282          GRB14      2_100    0.7262 21.32 0.0002461  4.745        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>           genename region_tag susie_pip    mu2       PVE       z num_eqtl
7048          JAZF1       7_23  0.010490 138.57 2.311e-05 -12.731        1
14285  RP11-395N3.2      2_133  0.035389 109.24 6.147e-05 -11.370        1
2723           WFS1        4_7  0.029567  53.23 2.502e-05  10.642        1
356            ANK1       8_36  0.040526  59.39 3.827e-05   8.667        1
6954        ZFP36L2       2_27  0.039417  55.29 3.465e-05   8.377        1
6449         CDKAL1       6_15  0.003752  40.70 2.428e-06  -8.192        1
14659     LINC01126       2_27  0.049134  47.85 3.738e-05  -7.520        2
3368          THADA       2_27  0.037436  45.75 2.723e-05   7.450        1
7490          UBE2Z      17_28  0.702743  47.08 5.260e-04  -7.392        1
11174        KCNJ11      11_12  0.022353  49.77 1.769e-05   7.231        1
11270       NCR3LG1      11_12  0.022725  49.31 1.782e-05  -7.169        2
8537             C2       6_26  0.219641  52.12 1.820e-04   7.149        1
1417         PABPC4       1_24  0.208389  47.85 1.585e-04  -7.054        1
943           ZZEF1       17_4  0.090687  43.96 6.339e-05   6.917        1
10726         BMP8A       1_24  0.090841  42.59 6.152e-05   6.868        1
10282        ZNF664      12_75  0.315661  43.32 2.174e-04  -6.452        1
12119          MICB       6_25  0.483138  29.84 2.292e-04   6.443        3
13467 RP11-419C23.1       8_33  0.506567  37.39 3.012e-04  -6.317        1
7491           SNF8      17_28  0.064662  43.17 4.438e-05   6.300        1
7368          AP3S2      15_41  0.392664  36.08 2.252e-04   6.273        2</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.009079</code></pre>
<pre><code>           genename region_tag susie_pip    mu2       PVE       z num_eqtl
7048          JAZF1       7_23  0.010490 138.57 2.311e-05 -12.731        1
14285  RP11-395N3.2      2_133  0.035389 109.24 6.147e-05 -11.370        1
2723           WFS1        4_7  0.029567  53.23 2.502e-05  10.642        1
356            ANK1       8_36  0.040526  59.39 3.827e-05   8.667        1
6954        ZFP36L2       2_27  0.039417  55.29 3.465e-05   8.377        1
6449         CDKAL1       6_15  0.003752  40.70 2.428e-06  -8.192        1
14659     LINC01126       2_27  0.049134  47.85 3.738e-05  -7.520        2
3368          THADA       2_27  0.037436  45.75 2.723e-05   7.450        1
7490          UBE2Z      17_28  0.702743  47.08 5.260e-04  -7.392        1
11174        KCNJ11      11_12  0.022353  49.77 1.769e-05   7.231        1
11270       NCR3LG1      11_12  0.022725  49.31 1.782e-05  -7.169        2
8537             C2       6_26  0.219641  52.12 1.820e-04   7.149        1
1417         PABPC4       1_24  0.208389  47.85 1.585e-04  -7.054        1
943           ZZEF1       17_4  0.090687  43.96 6.339e-05   6.917        1
10726         BMP8A       1_24  0.090841  42.59 6.152e-05   6.868        1
10282        ZNF664      12_75  0.315661  43.32 2.174e-04  -6.452        1
12119          MICB       6_25  0.483138  29.84 2.292e-04   6.443        3
13467 RP11-419C23.1       8_33  0.506567  37.39 3.012e-04  -6.317        1
7491           SNF8      17_28  0.064662  43.17 4.438e-05   6.300        1
7368          AP3S2      15_41  0.392664  36.08 2.252e-04   6.273        2</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 34</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                                 Description     FDR Ratio BgRatio
47  Jansen type metaphyseal chondrodysplasia 0.02097  1/16  1/9703
69                  Eiken Skeletal Dysplasia 0.02097  1/16  1/9703
71        Failure of Tooth Eruption, Primary 0.02097  1/16  1/9703
72         Chondrodysplasia, blomstrand type 0.02097  1/16  1/9703
76     MICROPHTHALMIA, ISOLATED 4 (disorder) 0.02097  1/16  1/9703
83             LEBER CONGENITAL AMAUROSIS 17 0.02097  1/16  1/9703
89            MULTIPLE SYNOSTOSES SYNDROME 4 0.02097  1/16  1/9703
48                          Hyperargininemia 0.02933  1/16  2/9703
73 KLIPPEL-FEIL SYNDROME, AUTOSOMAL DOMINANT 0.02933  1/16  2/9703
79 MICROPHTHALMIA, ISOLATED, WITH COLOBOMA 6 0.02933  1/16  2/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<p><img src="figure/T2D_Adipose_Subcutaneous.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 72</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 25</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.533</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 34</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 78</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>       genename region_tag susie_pip   mu2       PVE      z num_eqtl
6245      MRPS5       2_57    0.6981 19.67 0.0002184 -3.810        2
11514   PRPF40A       2_91    0.5173 20.30 0.0001670 -3.928        1
11477     CPNE4       3_81    0.5763 21.37 0.0001958  4.312        1
10066    ZBTB38       3_86    0.5397 17.85 0.0001532 -3.085        1
7947     DCAF16       4_17    0.6680 21.59 0.0002293 -4.110        2
13258 LINC01184       5_78    0.6725 19.24 0.0002058  3.793        1
3924      SEPT7       7_26    0.7751 20.71 0.0002553 -4.167        1
11034     ZBTB6       9_63    0.5209 21.63 0.0001792  4.264        1
11712     PARVA       11_9    0.7846 20.35 0.0002539 -3.862        1
6672      EIF3M      11_22    0.5005 18.64 0.0001483 -3.890        1
3988     ZC3H13      13_19    0.7732 20.56 0.0002527  4.310        1
10469  TMEM132E      17_20    0.6954 21.33 0.0002358 -3.693        2
4665     ZNF236      18_45    0.6700 20.05 0.0002136 -4.227        2
2323     DNASE2      19_10    0.7718 18.79 0.0002306 -3.744        1
1860       CTSZ      20_34    0.6903 19.10 0.0002097 -3.496        2
7267       GDF6       8_67    0.7821 21.36 0.0002656 -4.372        1
4388       RNF6       13_6    0.6475 23.06 0.0002374 -4.341        1
13239   N4BP2L2      13_11    0.8441 24.34 0.0003267 -4.430        1
250      ANGEL1      14_36    0.9577 22.68 0.0003453  4.529        2</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.02778 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9960 0.9911 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>  ctwas    TWAS 
0.00000 0.02564 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7      
 [5] purrr_0.3.4       readr_2.1.1       tidyr_1.1.4       tidyverse_1.3.1  
 [9] tibble_3.1.6      WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0      
[13] cowplot_1.0.0     ggplot2_3.3.5     workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       doParallel_1.0.17
 [5] httr_1.4.2        rprojroot_2.0.2   tools_3.6.1       backports_1.4.1  
 [9] doRNG_1.8.2       utf8_1.2.2        R6_2.5.1          vipor_0.4.5      
[13] DBI_1.1.2         colorspace_2.0-2  withr_2.4.3       ggrastr_1.0.1    
[17] tidyselect_1.1.1  bit_4.0.4         curl_4.3.2        compiler_3.6.1   
[21] git2r_0.26.1      rvest_1.0.2       cli_3.1.0         Cairo_1.5-12.2   
[25] xml2_1.3.3        labeling_0.4.2    scales_1.1.1      apcluster_1.4.8  
[29] digest_0.6.29     rmarkdown_2.11    svglite_1.2.2     pkgconfig_2.0.3  
[33] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[37] rlang_1.0.1       rstudioapi_0.13   RSQLite_2.2.8     jquerylib_0.1.4  
[41] farver_2.1.0      generics_0.1.1    jsonlite_1.7.2    vroom_1.5.7      
[45] magrittr_2.0.2    Matrix_1.2-18     ggbeeswarm_0.6.0  Rcpp_1.0.8       
[49] munsell_0.5.0     fansi_1.0.2       gdtools_0.1.9     lifecycle_1.0.1  
[53] stringi_1.7.6     whisker_0.3-2     yaml_2.2.1        plyr_1.8.6       
[57] grid_3.6.1        blob_1.2.2        ggrepel_0.9.1     parallel_3.6.1   
[61] promises_1.0.1    crayon_1.5.0      lattice_0.20-38   haven_2.4.3      
[65] hms_1.1.1         knitr_1.36        pillar_1.6.4      igraph_1.2.10    
[69] rjson_0.2.20      rngtools_1.5.2    reshape2_1.4.4    codetools_0.2-16 
[73] reprex_2.0.1      glue_1.6.2        evaluate_0.14     data.table_1.14.2
[77] modelr_0.1.8      vctrs_0.3.8       tzdb_0.2.0        httpuv_1.5.1     
[81] foreach_1.5.2     cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[85] cachem_1.0.6      xfun_0.29         broom_0.7.10      later_0.8.0      
[89] iterators_1.0.14  beeswarm_0.2.3    memoise_2.0.1     ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
