<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>T2D - Liver</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">T2D - Liver</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2021-2-6</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config.R
</td>
<td style="text-align:left;">
code/ctwas_config.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreecf57e2f83da97cfcf5e6a5ce5d569945347ad730targetblankcf57e2fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/cf57e2f83da97cfcf5e6a5ce5d569945347ad730" target="_blank">cf57e2f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreecf57e2f83da97cfcf5e6a5ce5d569945347ad730targetblankcf57e2fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/cf57e2f83da97cfcf5e6a5ce5d569945347ad730" target="_blank">cf57e2f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/AF_out/
    Untracked:  code/BMI_out/
    Untracked:  code/T2D_out/
    Untracked:  code/ctwas_config.R
    Untracked:  code/mapping.R
    Untracked:  code/out/
    Untracked:  code/run_AF_analysis.sbatch
    Untracked:  code/run_AF_analysis.sh
    Untracked:  code/run_AF_ctwas_rss_LDR.R
    Untracked:  code/run_BMI_analysis.sbatch
    Untracked:  code/run_BMI_analysis.sh
    Untracked:  code/run_BMI_ctwas_rss_LDR.R
    Untracked:  code/run_T2D_analysis.sbatch
    Untracked:  code/run_T2D_analysis.sh
    Untracked:  code/run_T2D_ctwas_rss_LDR.R
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/AF/
    Untracked:  data/BMI/
    Untracked:  data/T2D/
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  data/untitled.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/T2D_Liver.Rmd</code>) and HTML (<code>docs/T2D_Liver.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/cf57e2f83da97cfcf5e6a5ce5d569945347ad730/analysis/T2D_Liver.Rmd" target="_blank">cf57e2f</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/721f7e8f754a609efb3d852301ad86f42238b730/analysis/T2D_Liver.Rmd" target="_blank">721f7e8</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-14
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/91f38fa8b991bcb34c11575638dc25fc81457a5f/docs/T2D_Liver.html" target="_blank">91f38fa</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/eb13ecf45ad6ec66ccac1b6f91b74c140a7261d6/analysis/T2D_Liver.Rmd" target="_blank">eb13ecf</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/e6bc16990373312d301ab183ec936aff373d2497/docs/T2D_Liver.html" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/87fee8b56a4f94471388b3fc74b464d3b6efdd41/analysis/T2D_Liver.Rmd" target="_blank">87fee8b</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre class="r"><code>#number of imputed weights
nrow(qclist_all)</code></pre>
<pre><code>[1] 10290</code></pre>
<pre class="r"><code>#number of imputed weights by chromosome
table(qclist_all$chr)</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1015  728  627  403  457  577  509  392  388  394  607  572  190  353  340  483 
  17   18   19   20   21   22 
 629  151  808  290  108  269 </code></pre>
<pre class="r"><code>#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 8215</code></pre>
<pre class="r"><code>#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)</code></pre>
<pre><code>[1] 0.7983</code></pre>
</div>
<div id="check-convergence-of-parameters" class="section level2">
<h2>Check convergence of parameters</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/e6bc16990373312d301ab183ec936aff373d2497/docs/figure/T2D_Liver.Rmd/unnamed-chunk-5-1.png" target="_blank">e6bc169</a>
</td>
<td>
sq-96
</td>
<td>
2022-02-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- group_prior_rec[,ncol(group_prior_rec)]
names(estimated_group_prior) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
estimated_group_prior[&quot;snp&quot;] &lt;- estimated_group_prior[&quot;snp&quot;]*thin #adjust parameter to account for thin argument
print(estimated_group_prior)</code></pre>
<pre><code>     gene       snp 
0.0041007 0.0001803 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- group_prior_var_rec[,ncol(group_prior_var_rec)]
names(estimated_group_prior_var) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_prior_var)</code></pre>
<pre><code> gene   snp 
4.389 1.535 </code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 337159</code></pre>
<pre class="r"><code>#report group size
group_size &lt;- c(nrow(ctwas_gene_res), n_snps)
print(group_size)</code></pre>
<pre><code>[1]   10290 7535010</code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_prior_var*estimated_group_prior*group_size/sample_size #check PVE calculation
names(estimated_group_pve) &lt;- c(&quot;gene&quot;, &quot;snp&quot;)
print(estimated_group_pve)</code></pre>
<pre><code>     gene       snp 
0.0005493 0.0061836 </code></pre>
<pre class="r"><code>#compare sum(PIP*mu2/sample_size) with above PVE calculation
c(sum(ctwas_gene_res$PVE),sum(ctwas_snp_res$PVE))</code></pre>
<pre><code>[1] 0.004835 0.111689</code></pre>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>          genename region_tag susie_pip   mu2       PVE      z num_eqtl
3212         CCND2       12_4    0.9960 27.98 8.264e-05  5.657        1
2240       SEC23IP      10_74    0.6382 61.38 1.162e-04 -3.611        1
12661    LINC01126       2_27    0.4142 28.19 3.463e-05  4.620        1
10283        MCMBP      10_74    0.3686 60.32 6.594e-05  3.522        1
703         GUCY2C      12_12    0.2542 35.09 2.645e-05  3.879        1
6307          NUS1       6_78    0.2350 32.15 2.241e-05  3.716        1
12541  RP6-65G23.5      14_33    0.2244 30.24 2.013e-05  3.370        1
5911          CIZ1       9_66    0.2067 30.11 1.846e-05 -3.514        2
6558         AP3S2      15_41    0.1991 30.36 1.793e-05 -3.746        1
7641         NDEL1       17_8    0.1908 28.90 1.635e-05 -3.137        1
10118        RABL6       9_74    0.1895 30.14 1.694e-05  3.491        1
2577        GNPTAB      12_61    0.1869 29.62 1.642e-05  3.601        1
4089         UBAC1       9_72    0.1743 28.56 1.476e-05  3.439        1
9318          LIPF      10_56    0.1709 29.37 1.489e-05 -2.992        1
12431 RP11-535A5.1      18_11    0.1657 28.09 1.380e-05 -2.998        1
12123       UPK3BL       7_63    0.1607 28.41 1.354e-05  3.135        1
1624       TPD52L2      20_38    0.1531 28.21 1.281e-05 -3.091        1
1483          RPL3      22_16    0.1467 27.86 1.212e-05  3.285        1
4539         ISCA1       9_44    0.1449 27.76 1.194e-05  3.270        1
3541       ARHGAP9      12_36    0.1431 26.51 1.125e-05  2.926        1</code></pre>
</div>
<div id="genes-with-largest-effect-sizes" class="section level2">
<h2>Genes with largest effect sizes</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>          genename region_tag susie_pip   mu2       PVE      z num_eqtl
2240       SEC23IP      10_74    0.6382 61.38 1.162e-04 -3.611        1
10283        MCMBP      10_74    0.3686 60.32 6.594e-05  3.522        1
703         GUCY2C      12_12    0.2542 35.09 2.645e-05  3.879        1
8349          GPHN      14_32    0.1375 33.30 1.357e-05 -3.427        2
6307          NUS1       6_78    0.2350 32.15 2.241e-05  3.716        1
6558         AP3S2      15_41    0.1991 30.36 1.793e-05 -3.746        1
12541  RP6-65G23.5      14_33    0.2244 30.24 2.013e-05  3.370        1
10118        RABL6       9_74    0.1895 30.14 1.694e-05  3.491        1
5911          CIZ1       9_66    0.2067 30.11 1.846e-05 -3.514        2
5073         ETNK1      12_16    0.1407 29.99 1.251e-05  3.170        1
7288         AGGF1       5_45    0.1051 29.89 9.318e-06 -3.154        2
2577        GNPTAB      12_61    0.1869 29.62 1.642e-05  3.601        1
9318          LIPF      10_56    0.1709 29.37 1.489e-05 -2.992        1
7641         NDEL1       17_8    0.1908 28.90 1.635e-05 -3.137        1
4089         UBAC1       9_72    0.1743 28.56 1.476e-05  3.439        1
12123       UPK3BL       7_63    0.1607 28.41 1.354e-05  3.135        1
1624       TPD52L2      20_38    0.1531 28.21 1.281e-05 -3.091        1
12661    LINC01126       2_27    0.4142 28.19 3.463e-05  4.620        1
12431 RP11-535A5.1      18_11    0.1657 28.09 1.380e-05 -2.998        1
1460        PPP6R2      22_24    0.1361 27.99 1.130e-05 -3.284        1</code></pre>
</div>
<div id="genes-with-highest-pve" class="section level2">
<h2>Genes with highest PVE</h2>
<pre><code>          genename region_tag susie_pip   mu2       PVE      z num_eqtl
2240       SEC23IP      10_74    0.6382 61.38 1.162e-04 -3.611        1
3212         CCND2       12_4    0.9960 27.98 8.264e-05  5.657        1
10283        MCMBP      10_74    0.3686 60.32 6.594e-05  3.522        1
12661    LINC01126       2_27    0.4142 28.19 3.463e-05  4.620        1
703         GUCY2C      12_12    0.2542 35.09 2.645e-05  3.879        1
6307          NUS1       6_78    0.2350 32.15 2.241e-05  3.716        1
12541  RP6-65G23.5      14_33    0.2244 30.24 2.013e-05  3.370        1
5911          CIZ1       9_66    0.2067 30.11 1.846e-05 -3.514        2
6558         AP3S2      15_41    0.1991 30.36 1.793e-05 -3.746        1
10118        RABL6       9_74    0.1895 30.14 1.694e-05  3.491        1
2577        GNPTAB      12_61    0.1869 29.62 1.642e-05  3.601        1
7641         NDEL1       17_8    0.1908 28.90 1.635e-05 -3.137        1
9318          LIPF      10_56    0.1709 29.37 1.489e-05 -2.992        1
4089         UBAC1       9_72    0.1743 28.56 1.476e-05  3.439        1
12431 RP11-535A5.1      18_11    0.1657 28.09 1.380e-05 -2.998        1
8349          GPHN      14_32    0.1375 33.30 1.357e-05 -3.427        2
12123       UPK3BL       7_63    0.1607 28.41 1.354e-05  3.135        1
1624       TPD52L2      20_38    0.1531 28.21 1.281e-05 -3.091        1
5073         ETNK1      12_16    0.1407 29.99 1.251e-05  3.170        1
1483          RPL3      22_16    0.1467 27.86 1.212e-05  3.285        1</code></pre>
</div>
<div id="genes-with-largest-z-scores" class="section level2">
<h2>Genes with largest z scores</h2>
<pre><code>         genename region_tag susie_pip   mu2       PVE      z num_eqtl
3212        CCND2       12_4   0.99598 27.98 8.264e-05  5.657        1
12661   LINC01126       2_27   0.41421 28.19 3.463e-05  4.620        1
703        GUCY2C      12_12   0.25415 35.09 2.645e-05  3.879        1
6558        AP3S2      15_41   0.19907 30.36 1.793e-05 -3.746        1
6307         NUS1       6_78   0.23500 32.15 2.241e-05  3.716        1
2240      SEC23IP      10_74   0.63818 61.38 1.162e-04 -3.611        1
2577       GNPTAB      12_61   0.18690 29.62 1.642e-05  3.601        1
10283       MCMBP      10_74   0.36857 60.32 6.594e-05  3.522        1
1505         RBX1      22_17   0.13907 26.28 1.084e-05 -3.521        1
5911         CIZ1       9_66   0.20671 30.11 1.846e-05 -3.514        2
10118       RABL6       9_74   0.18950 30.14 1.694e-05  3.491        1
4089        UBAC1       9_72   0.17429 28.56 1.476e-05  3.439        1
10840      PPP1CB       2_17   0.08613 23.42 5.984e-06  3.434        3
7172        SPDYA       2_17   0.08538 23.35 5.913e-06 -3.430        2
8349         GPHN      14_32   0.13745 33.30 1.357e-05 -3.427        2
5040       CNOT6L       4_52   0.13255 27.12 1.066e-05  3.424        1
12541 RP6-65G23.5      14_33   0.22445 30.24 2.013e-05  3.370        1
1483         RPL3      22_16   0.14670 27.86 1.212e-05  3.285        1
1460       PPP6R2      22_24   0.13614 27.99 1.130e-05 -3.284        1
2417         GLRB      4_101   0.13568 27.34 1.100e-05  3.270        1</code></pre>
</div>
<div id="comparing-z-scores-and-pips" class="section level2">
<h2>Comparing z scores and PIPs</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/T2D_Liver.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] 0.0001944</code></pre>
<pre><code>         genename region_tag susie_pip   mu2       PVE      z num_eqtl
3212        CCND2       12_4   0.99598 27.98 8.264e-05  5.657        1
12661   LINC01126       2_27   0.41421 28.19 3.463e-05  4.620        1
703        GUCY2C      12_12   0.25415 35.09 2.645e-05  3.879        1
6558        AP3S2      15_41   0.19907 30.36 1.793e-05 -3.746        1
6307         NUS1       6_78   0.23500 32.15 2.241e-05  3.716        1
2240      SEC23IP      10_74   0.63818 61.38 1.162e-04 -3.611        1
2577       GNPTAB      12_61   0.18690 29.62 1.642e-05  3.601        1
10283       MCMBP      10_74   0.36857 60.32 6.594e-05  3.522        1
1505         RBX1      22_17   0.13907 26.28 1.084e-05 -3.521        1
5911         CIZ1       9_66   0.20671 30.11 1.846e-05 -3.514        2
10118       RABL6       9_74   0.18950 30.14 1.694e-05  3.491        1
4089        UBAC1       9_72   0.17429 28.56 1.476e-05  3.439        1
10840      PPP1CB       2_17   0.08613 23.42 5.984e-06  3.434        3
7172        SPDYA       2_17   0.08538 23.35 5.913e-06 -3.430        2
8349         GPHN      14_32   0.13745 33.30 1.357e-05 -3.427        2
5040       CNOT6L       4_52   0.13255 27.12 1.066e-05  3.424        1
12541 RP6-65G23.5      14_33   0.22445 30.24 2.013e-05  3.370        1
1483         RPL3      22_16   0.14670 27.86 1.212e-05  3.285        1
1460       PPP6R2      22_24   0.13614 27.99 1.130e-05 -3.284        1
2417         GLRB      4_101   0.13568 27.34 1.100e-05  3.270        1</code></pre>
</div>
<div id="gene-set-enrichment-for-genes-with-pip0.5" class="section level2">
<h2>Gene set enrichment for genes with PIP&gt;0.5</h2>
<pre><code>[1] 2</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                                                            Term
1  positive regulation of cyclin-dependent protein serine/threonine kinase activity (GO:0045737)
2                   positive regulation of cyclin-dependent protein kinase activity (GO:1904031)
3                      positive regulation of G1/S transition of mitotic cell cycle (GO:1900087)
4                           positive regulation of cell cycle G1/S phase transition (GO:1902808)
5                            regulation of cyclin-dependent protein kinase activity (GO:1904029)
6                        positive regulation of mitotic cell cycle phase transition (GO:1901992)
7                                                             COPII vesicle coating (GO:0048208)
8                                                                   vesicle coating (GO:0006901)
9                                          vesicle targeting, rough ER to cis-Golgi (GO:0048207)
10                                                positive regulation of cell cycle (GO:0045787)
11                                                     COPII-coated vesicle budding (GO:0090114)
12                              regulation of G1/S transition of mitotic cell cycle (GO:2000045)
13          regulation of cyclin-dependent protein serine/threonine kinase activity (GO:0000079)
14                  positive regulation of protein serine/threonine kinase activity (GO:0071902)
15                           regulation of protein serine/threonine kinase activity (GO:0071900)
16                                                               Golgi organization (GO:0007030)
17                                                 endomembrane system organization (GO:0010256)
18                                              mitotic cell cycle phase transition (GO:0044772)
19                              positive regulation of protein modification process (GO:0031401)
20                                           positive regulation of phosphorylation (GO:0042327)
21                                            regulation of protein phosphorylation (GO:0001932)
22                                              protein-containing complex assembly (GO:0065003)
23                                                    cellular protein localization (GO:0034613)
24                                                  intracellular protein transport (GO:0006886)
25                                                                protein transport (GO:0015031)
26                                   positive regulation of protein phosphorylation (GO:0001934)
27                                     negative regulation of programmed cell death (GO:0043069)
28                                                           organelle organization (GO:0006996)
   Overlap Adjusted.P.value   Genes
1     1/17          0.01949   CCND2
2     1/20          0.01949   CCND2
3     1/26          0.01949   CCND2
4     1/35          0.01949   CCND2
5     1/54          0.01949   CCND2
6     1/58          0.01949   CCND2
7     1/63          0.01949 SEC23IP
8     1/63          0.01949 SEC23IP
9     1/63          0.01949 SEC23IP
10    1/66          0.01949   CCND2
11    1/70          0.01949 SEC23IP
12    1/71          0.01949   CCND2
13    1/82          0.02077   CCND2
14   1/106          0.02435   CCND2
15   1/111          0.02435   CCND2
16   1/130          0.02673 SEC23IP
17   1/199          0.03697 SEC23IP
18   1/209          0.03697   CCND2
19   1/214          0.03697   CCND2
20   1/253          0.03978   CCND2
21   1/266          0.03978   CCND2
22   1/267          0.03978 SEC23IP
23   1/329          0.04581 SEC23IP
24   1/336          0.04581 SEC23IP
25   1/369          0.04612 SEC23IP
26   1/371          0.04612   CCND2
27   1/381          0.04612   CCND2
28   1/420          0.04898 SEC23IP
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>                                                             Term Overlap
1 cyclin-dependent protein kinase holoenzyme complex (GO:0000307)    1/30
2            serine/threonine protein kinase complex (GO:1902554)    1/37
3         COPII-coated ER to Golgi transport vesicle (GO:0030134)    1/79
4                                     coated vesicle (GO:0030135)    1/84
5                                   nuclear membrane (GO:0031965)   1/204
  Adjusted.P.value   Genes
1          0.01848   CCND2
2          0.01848   CCND2
3          0.02096 SEC23IP
4          0.02096 SEC23IP
5          0.04059   CCND2
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<pre><code>                                                                              Term
1 cyclin-dependent protein serine/threonine kinase regulator activity (GO:0016538)
2                                                     lipase activity (GO:0016298)
3                                              phospholipase activity (GO:0004620)
4                                   protein kinase regulator activity (GO:0019887)
  Overlap Adjusted.P.value   Genes
1    1/44          0.01700   CCND2
2    1/49          0.01700 SEC23IP
3    1/73          0.01700 SEC23IP
4    1/98          0.01711   CCND2
                                                         Description     FDR
6                                        Communicating Hydrocephalus 0.00202
19                                            POLYDACTYLY, POSTAXIAL 0.00202
22                                            Hydrocephalus Ex-Vacuo 0.00202
24                                      Post-Traumatic Hydrocephalus 0.00202
25                                         Obstructive Hydrocephalus 0.00202
30                                         Cerebral ventriculomegaly 0.00202
32                                              Perisylvian syndrome 0.00202
33 Megalanecephaly Polymicrogyria-Polydactyly Hydrocephalus Syndrome 0.00202
34                                     POSTAXIAL POLYDACTYLY, TYPE B 0.00202
36                                                  Alcohol Toxicity 0.00202
   Ratio BgRatio
6    1/1  7/9703
19   1/1  4/9703
22   1/1  7/9703
24   1/1  7/9703
25   1/1  7/9703
30   1/1  7/9703
32   1/1  4/9703
33   1/1  4/9703
34   1/1  3/9703
36   1/1  2/9703
Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>NULL</code></pre>
</div>
<div id="pip-manhattan-plot" class="section level2">
<h2>PIP Manhattan Plot</h2>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sensitivity-specificity-and-precision-for-silver-standard-genes" class="section level2">
<h2>Sensitivity, specificity and precision for silver standard genes</h2>
<pre class="r"><code>#number of genes in known annotations
print(length(known_annotations))</code></pre>
<pre><code>[1] 72</code></pre>
<pre class="r"><code>#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>#significance threshold for TWAS
print(sig_thresh)</code></pre>
<pre><code>[1] 4.571</code></pre>
<pre class="r"><code>#number of ctwas genes
length(ctwas_genes)</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#number of TWAS genes
length(twas_genes)</code></pre>
<pre><code>[1] 2</code></pre>
<pre class="r"><code>#show novel genes (ctwas genes with not in TWAS genes)
ctwas_gene_res[ctwas_gene_res$genename %in% novel_genes,report_cols]</code></pre>
<pre><code>[1] genename   region_tag susie_pip  mu2        PVE        z          num_eqtl  
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>#sensitivity / recall
print(sensitivity)</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<pre class="r"><code>#specificity
print(specificity)</code></pre>
<pre><code> ctwas   TWAS 
0.9999 0.9998 </code></pre>
<pre class="r"><code>#precision / PPV
print(precision)</code></pre>
<pre><code>ctwas  TWAS 
    0     0 </code></pre>
<p><img src="figure/T2D_Liver.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7      
 [5] purrr_0.3.4       readr_2.1.1       tidyr_1.1.4       tidyverse_1.3.1  
 [9] tibble_3.1.6      WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.0      
[13] cowplot_1.0.0     ggplot2_3.3.5     workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   bit64_4.0.5       doParallel_1.0.16
 [5] httr_1.4.2        rprojroot_2.0.2   tools_3.6.1       backports_1.4.1  
 [9] doRNG_1.8.2       utf8_1.2.2        R6_2.5.1          vipor_0.4.5      
[13] DBI_1.1.1         colorspace_2.0-2  withr_2.4.3       ggrastr_1.0.1    
[17] tidyselect_1.1.1  bit_4.0.4         curl_4.3.2        compiler_3.6.1   
[21] git2r_0.26.1      cli_3.1.0         rvest_1.0.2       Cairo_1.5-12.2   
[25] xml2_1.3.3        labeling_0.4.2    scales_1.1.1      apcluster_1.4.8  
[29] digest_0.6.29     rmarkdown_2.11    svglite_1.2.2     pkgconfig_2.0.3  
[33] htmltools_0.5.2   dbplyr_2.1.1      fastmap_1.1.0     highr_0.9        
[37] rlang_0.4.12      rstudioapi_0.13   RSQLite_2.2.8     jquerylib_0.1.4  
[41] farver_2.1.0      generics_0.1.1    jsonlite_1.7.2    vroom_1.5.7      
[45] magrittr_2.0.1    Matrix_1.2-18     ggbeeswarm_0.6.0  Rcpp_1.0.7       
[49] munsell_0.5.0     fansi_0.5.0       gdtools_0.1.9     lifecycle_1.0.1  
[53] stringi_1.7.6     whisker_0.3-2     yaml_2.2.1        plyr_1.8.6       
[57] grid_3.6.1        blob_1.2.2        ggrepel_0.9.1     parallel_3.6.1   
[61] promises_1.0.1    crayon_1.4.2      lattice_0.20-38   haven_2.4.3      
[65] hms_1.1.1         knitr_1.36        pillar_1.6.4      igraph_1.2.10    
[69] rjson_0.2.20      rngtools_1.5.2    reshape2_1.4.4    codetools_0.2-16 
[73] reprex_2.0.1      glue_1.5.1        evaluate_0.14     data.table_1.14.2
[77] modelr_0.1.8      vctrs_0.3.8       tzdb_0.2.0        httpuv_1.5.1     
[81] foreach_1.5.1     cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[85] cachem_1.0.6      xfun_0.29         broom_0.7.10      later_0.8.0      
[89] iterators_1.0.13  beeswarm_0.2.3    memoise_2.0.1     ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
