<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sheng Qian" />


<title>WhiteBlood - WholeBlood Traits Jointly</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">WhiteBlood - WholeBlood Traits Jointly</h1>
<h4 class="author">Sheng Qian</h4>
<h4 class="date">2023-2-1</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-02-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config_b38.R
</td>
<td style="text-align:left;">
code/ctwas_config_b38.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/mashr_sqtl/sqtl/mashr/mashr_Liver_Splicing_mapping.RData
</td>
<td style="text-align:left;">
data/mashr_sqtl/sqtl/mashr/mashr_Liver_Splicing_mapping.RData
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/mashr_sqtl/sqtl/mashr/mashr_Whole_Blood_Splicing_mapping.RData
</td>
<td style="text-align:left;">
data/mashr_sqtl/sqtl/mashr/mashr_Whole_Blood_Splicing_mapping.RData
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/mqtl/WholeBlood.db
</td>
<td style="text-align:left;">
data/mqtl/WholeBlood.db
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/G_list.RData
</td>
<td style="text-align:left;">
data/G_list.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeda7a0f70b9796e45b45ac2e60d2b3e3386af7a7atargetblankda7a0f7a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/da7a0f70b9796e45b45ac2e60d2b3e3386af7a7a" target="_blank">da7a0f7</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistreeda7a0f70b9796e45b45ac2e60d2b3e3386af7a7atargetblankda7a0f7a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/da7a0f70b9796e45b45ac2e60d2b3e3386af7a7a" target="_blank">da7a0f7</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Proposal plots.R
    Untracked:  RGS14.pdf
    Untracked:  RNF186.pdf
    Untracked:  SCZ_annotation.xlsx
    Untracked:  SLC8B1.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  cache/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/FUMA_output/
    Untracked:  data/GO_Terms/
    Untracked:  data/GTEx_Analysis_v8_eQTL.tar
    Untracked:  data/G_list.RData
    Untracked:  data/IBD_ME/
    Untracked:  data/LDL/
    Untracked:  data/LDL_E_S/
    Untracked:  data/LDL_M/
    Untracked:  data/LDL_S/
    Untracked:  data/LDL_multi/
    Untracked:  data/PGC3_SCZ_wave3_public.v2.tsv
    Untracked:  data/RedBlood_M/
    Untracked:  data/SCZ/
    Untracked:  data/SCZ_2014_EUR/
    Untracked:  data/SCZ_2014_EUR_ME/
    Untracked:  data/SCZ_2018/
    Untracked:  data/SCZ_2018_ME/
    Untracked:  data/SCZ_2018_S/
    Untracked:  data/SCZ_2020/
    Untracked:  data/SCZ_S/
    Untracked:  data/Supplementary Table 15 - MAGMA.xlsx
    Untracked:  data/Supplementary Table 20 - Prioritised Genes.xlsx
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/WhiteBlood_E_S_M/
    Untracked:  data/White_Blood_M/
    Untracked:  data/eqtl/
    Untracked:  data/gencode.v26.GRCh38.genes.gtf
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/gwas_sumstats/
    Untracked:  data/magma.genes.out
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/mqtl/
    Untracked:  data/multigroup/
    Untracked:  data/notes.txt
    Untracked:  data/scz_2018.RDS
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  temp_LDR/
    Untracked:  top_genes_32.txt
    Untracked:  top_genes_37.txt
    Untracked:  top_genes_43.txt
    Untracked:  top_genes_54.txt
    Untracked:  top_genes_81.txt
    Untracked:  z_snp_pos_SCZ.RData
    Untracked:  z_snp_pos_SCZ_2014_EUR.RData
    Untracked:  z_snp_pos_SCZ_2018.RData
    Untracked:  z_snp_pos_SCZ_2020.RData

Unstaged changes:
    Deleted:    analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd
    Deleted:    analysis/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd
    Deleted:    analysis/Autism_Brain_Amygdala.Rmd
    Deleted:    analysis/Autism_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Deleted:    analysis/Autism_Brain_Caudate_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Cerebellar_Hemisphere.Rmd
    Deleted:    analysis/Autism_Brain_Cerebellum.Rmd
    Deleted:    analysis/Autism_Brain_Cortex.Rmd
    Deleted:    analysis/Autism_Brain_Frontal_Cortex_BA9.Rmd
    Deleted:    analysis/Autism_Brain_Hippocampus.Rmd
    Deleted:    analysis/Autism_Brain_Hypothalamus.Rmd
    Deleted:    analysis/Autism_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Putamen_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Spinal_cord_cervical_c-1.Rmd
    Deleted:    analysis/Autism_Brain_Substantia_nigra.Rmd
    Deleted:    analysis/BMI_Brain_Amygdala.Rmd
    Deleted:    analysis/BMI_Brain_Amygdala_S.Rmd
    Deleted:    analysis/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Deleted:    analysis/BMI_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Deleted:    analysis/BMI_Brain_Caudate_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Caudate_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellar_Hemisphere.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellar_Hemisphere_S.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellum.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellum_S.Rmd
    Deleted:    analysis/BMI_Brain_Cortex.Rmd
    Deleted:    analysis/BMI_Brain_Cortex_S.Rmd
    Deleted:    analysis/BMI_Brain_Frontal_Cortex_BA9.Rmd
    Deleted:    analysis/BMI_Brain_Frontal_Cortex_BA9_S.Rmd
    Deleted:    analysis/BMI_Brain_Hippocampus.Rmd
    Deleted:    analysis/BMI_Brain_Hippocampus_S.Rmd
    Deleted:    analysis/BMI_Brain_Hypothalamus.Rmd
    Deleted:    analysis/BMI_Brain_Hypothalamus_S.Rmd
    Deleted:    analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Putamen_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Putamen_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Spinal_cord_cervical_c-1.Rmd
    Deleted:    analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Deleted:    analysis/BMI_Brain_Substantia_nigra.Rmd
    Deleted:    analysis/BMI_Brain_Substantia_nigra_S.Rmd
    Deleted:    analysis/BMI_S_results.Rmd
    Deleted:    analysis/Glucose_Adipose_Subcutaneous.Rmd
    Deleted:    analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Modified:   analysis/WhiteBlood_WholeBlood_E.Rmd
    Modified:   analysis/WhiteBlood_WholeBlood_E_S_M.Rmd
    Deleted:    code/run_IBD_ctwas_rss_LDR_ME.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/WhiteBlood_WholeBlood_E_S_M.Rmd</code>) and HTML (<code>docs/WhiteBlood_WholeBlood_E_S_M.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/da7a0f70b9796e45b45ac2e60d2b3e3386af7a7a/analysis/WhiteBlood_WholeBlood_E_S_M.Rmd" target="_blank">da7a0f7</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/da7a0f70b9796e45b45ac2e60d2b3e3386af7a7a/docs/WhiteBlood_WholeBlood_E_S_M.html" target="_blank">da7a0f7</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ba657aa5f36f76495363a8cd40fdb1de04ac9264/analysis/WhiteBlood_WholeBlood_E_S_M.Rmd" target="_blank">ba657aa</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/ba657aa5f36f76495363a8cd40fdb1de04ac9264/docs/WhiteBlood_WholeBlood_E_S_M.html" target="_blank">ba657aa</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/fa89b375daf79a7a1843d31ae9532f0f5797b29f/analysis/WhiteBlood_WholeBlood_E_S_M.Rmd" target="_blank">fa89b37</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/fa89b375daf79a7a1843d31ae9532f0f5797b29f/docs/WhiteBlood_WholeBlood_E_S_M.html" target="_blank">fa89b37</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6286c345d7d7de0c6a56bd29a0792988cb78f287/analysis/WhiteBlood_WholeBlood_E_S_M.Rmd" target="_blank">6286c34</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/6286c345d7d7de0c6a56bd29a0792988cb78f287/docs/WhiteBlood_WholeBlood_E_S_M.html" target="_blank">6286c34</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a76256830927919471e32c34075dde86f20eb5ff/analysis/WhiteBlood_WholeBlood_E_S_M.Rmd" target="_blank">a762568</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/a76256830927919471e32c34075dde86f20eb5ff/docs/WhiteBlood_WholeBlood_E_S_M.html" target="_blank">a762568</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/c5aaf9684518301926e8167b9fcf9276e3b3fc5c/analysis/WhiteBlood_WholeBlood_E_S_M.Rmd" target="_blank">c5aaf96</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/c5aaf9684518301926e8167b9fcf9276e3b3fc5c/docs/WhiteBlood_WholeBlood_E_S_M.html" target="_blank">c5aaf96</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/9b01dadf23cff76bebe62e9c24e4a0996e8c5d84/analysis/WhiteBlood_WholeBlood_E_S_M.Rmd" target="_blank">9b01dad</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-01
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>analysis_id &lt;- params$analysis_id
trait_id &lt;- params$trait_id
weight &lt;- params$weight

results_dir &lt;- paste0(&quot;/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/&quot;, trait_id, &quot;/&quot;, weight)

source(&quot;/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config_b38.R&quot;)
options(digits = 4)</code></pre>
<div id="load-ctwas-results" class="section level3">
<h3>Load ctwas results</h3>
</div>
<div id="check-convergence-of-parameters" class="section level3">
<h3>Check convergence of parameters</h3>
<p><img src="figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-3-1.png" width="70%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6286c345d7d7de0c6a56bd29a0792988cb78f287/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-3-1.png" target="_blank">6286c34</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/a76256830927919471e32c34075dde86f20eb5ff/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-3-1.png" target="_blank">a762568</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/c5aaf9684518301926e8167b9fcf9276e3b3fc5c/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-3-1.png" target="_blank">c5aaf96</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- estimated_group_prior_all[,ncol(group_prior_rec)]
print(estimated_group_prior)</code></pre>
<pre><code>                    SNP             Whole_Blood    Whole_Blood_Splicing 
              0.0001751               0.0225055               0.0052580 
Whole_Blood_Methylation 
              0.0108737 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- estimated_group_prior_var_all[,ncol(group_prior_var_rec)]
print(estimated_group_prior_var)</code></pre>
<pre><code>                    SNP             Whole_Blood    Whole_Blood_Splicing 
                  15.55                   19.40                   29.53 
Whole_Blood_Methylation 
                  14.35 </code></pre>
<pre class="r"><code>#estimated enrichment
estimated_enrichment &lt;- estimated_enrichment_all[ncol(group_prior_var_rec)]
print(estimated_enrichment)</code></pre>
<pre><code>[1] 61.79</code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 350470</code></pre>
<pre class="r"><code>#report group size
print(group_size)</code></pre>
<pre><code>                    SNP             Whole_Blood    Whole_Blood_Splicing 
                8696600                   11198                   20134 
Whole_Blood_Methylation 
                  11858 </code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_pve_all[,ncol(group_prior_rec)]
print(estimated_group_pve)</code></pre>
<pre><code>                    SNP             Whole_Blood    Whole_Blood_Splicing 
               0.067547                0.013950                0.008919 
Whole_Blood_Methylation 
               0.005278 </code></pre>
<pre class="r"><code>#total PVE
sum(estimated_group_pve)</code></pre>
<pre><code>[1] 0.09569</code></pre>
<pre class="r"><code>#attributable PVE
estimated_group_pve/sum(estimated_group_pve)</code></pre>
<pre><code>                    SNP             Whole_Blood    Whole_Blood_Splicing 
                0.70586                 0.14578                 0.09320 
Whole_Blood_Methylation 
                0.05515 </code></pre>
<pre class="r"><code>#load information for all genes
sqlite &lt;- RSQLite::dbDriver(&quot;SQLite&quot;)
db = RSQLite::dbConnect(sqlite, &quot;/project2/compbio/predictdb/mashr_models/mashr_Whole_Blood.db&quot;)
query &lt;- function(...) RSQLite::dbGetQuery(db, ...)
gene_info &lt;- query(&quot;select gene, genename from extra&quot;)
RSQLite::dbDisconnect(db)
ctwas_gene_E_res$genename &lt;- gene_info[sapply(ctwas_gene_E_res$gene_id, match, gene_info$gene),&quot;genename&quot;]

load(&quot;/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/mashr_sqtl/sqtl/mashr/mashr_Whole_Blood_Splicing_mapping.RData&quot;)
ctwas_gene_S_res$genename &lt;- intron_info[sapply(ctwas_gene_S_res$gene_id, match, intron_info$gene), &quot;genename&quot;]

sqlite &lt;- RSQLite::dbDriver(&quot;SQLite&quot;)
db = RSQLite::dbConnect(sqlite, &quot;/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/mqtl/WholeBlood.db&quot;)
query &lt;- function(...) RSQLite::dbGetQuery(db, ...)
gene_info &lt;- query(&quot;select gene, genename from extra&quot;)
RSQLite::dbDisconnect(db)
ctwas_gene_M_res$genename &lt;- gene_info[sapply(ctwas_gene_M_res$gene_id, match, gene_info$gene),&quot;genename&quot;]
ctwas_gene_M_res$genename &lt;- sapply(ctwas_gene_M_res$genename, function(x){unlist(strsplit(x, split=&quot;[;]&quot;))[1]})</code></pre>
</div>
<div id="top-expressionintroncpg-units" class="section level3">
<h3>Top expression/intron/CpG units</h3>
<pre><code>      genename                      gene_id susie_pip       group
39936      FES           ENSG00000182511.11    1.0000  Expression
43125     YBEY  intron_21_46287123_46296162    1.0000    Splicing
38047    TAGAP           ENSG00000164691.16    1.0000  Expression
36651 ARHGAP15           ENSG00000075884.13    1.0000  Expression
41903     CNN2    intron_19_1032696_1036130    1.0000    Splicing
36665    BAZ2B           ENSG00000123636.17    0.9999  Expression
38535    VLDLR           ENSG00000147852.15    0.9999  Expression
38156    CREB5           ENSG00000146592.16    0.9991  Expression
37171   MED12L           ENSG00000144893.12    0.9981  Expression
35946   LAPTM5            ENSG00000162511.7    0.9981  Expression
37527  SLC22A4            ENSG00000197208.5    0.9966  Expression
37106    ATXN7           ENSG00000163635.17    0.9965  Expression
39549     FLT3  intron_13_28024943_28027088    0.9949    Splicing
36279    CNIH4           ENSG00000143771.11    0.9948  Expression
39447    ALDH2           ENSG00000111275.12    0.9941  Expression
39604    LAMP1           ENSG00000185896.10    0.9931  Expression
36089 SLC25A24           ENSG00000085491.15    0.9924  Expression
43166    HDHD5           ENSG00000069998.12    0.9912  Expression
41827   TTC39C           ENSG00000168234.12    0.9872  Expression
41159    YPEL2            ENSG00000175155.8    0.9846  Expression
38704     LIPA           ENSG00000107798.17    0.9839  Expression
41100   KIF18B           ENSG00000186185.13    0.9837  Expression
39339   ACVRL1                   cg21236262    0.9824 Methylation
38973    PTPRJ           ENSG00000149177.12    0.9824  Expression
37291    ARAP2           ENSG00000047365.11    0.9783  Expression
37629    CPEB4           ENSG00000113742.12    0.9741  Expression
40700    MYO1C                   cg02622416    0.9736 Methylation
40445    ZFPM1            ENSG00000179588.8    0.9711  Expression
38575    OSTF1           ENSG00000134996.11    0.9691  Expression
39588    KLF12           ENSG00000118922.17    0.9685  Expression
36440    ITSN2   intron_2_24209221_24209818    0.9663    Splicing
37693    ATXN1           ENSG00000124788.17    0.9616  Expression
38633     NEK6           ENSG00000119408.16    0.9567  Expression
36934   UQCRC1           ENSG00000010256.10    0.9555  Expression
40367     GLG1           ENSG00000090863.11    0.9553  Expression
37233     ADD1           ENSG00000087274.16    0.9550  Expression
37348     TET2           ENSG00000168769.13    0.9535  Expression
36222     SELL            ENSG00000188404.8    0.9505  Expression
36005   CITED4            ENSG00000179862.6    0.9505  Expression
37276  TBC1D14           ENSG00000132405.18    0.9442  Expression
37664     CANX intron_5_179699102_179705679    0.9434    Splicing
36622   HS6ST1            ENSG00000136720.6    0.9385  Expression
36670    CD302            ENSG00000241399.6    0.9303  Expression
39154      CD6                   cg27098804    0.9266 Methylation
36067   CDC14A           ENSG00000079335.18    0.9259  Expression
37398   DDX60L           ENSG00000181381.13    0.9238  Expression
42503  ATP13A1           ENSG00000105726.16    0.9228  Expression
42362    MYO9B           ENSG00000099331.13    0.9165  Expression
40361    ITGAL                   cg24033122    0.9158 Methylation
42233    AP1M2           ENSG00000129354.11    0.9144  Expression
36881     CCR8            ENSG00000179934.6    0.9073  Expression
37134    PAQR9            ENSG00000188582.8    0.9002  Expression
37511  TNFAIP8            ENSG00000145779.7    0.8984  Expression
37439    RAI14           ENSG00000039560.13    0.8925  Expression
38269    CPSF4           ENSG00000160917.14    0.8870  Expression
37608   ADAM19           ENSG00000135074.15    0.8862  Expression
36265     NCF2 intron_1_183570839_183573185    0.8821    Splicing
41207 SLC9A3R1           ENSG00000109062.11    0.8776  Expression
37185    ABCC5           ENSG00000114770.16    0.8707  Expression
41423    UBE2O           ENSG00000175931.12    0.8706  Expression
37312     REST           ENSG00000084093.16    0.8678  Expression
39271    CERS5  intron_12_50134068_50134546    0.8677    Splicing
42844    UBOX5           ENSG00000185019.16    0.8644  Expression
36462     DTNB           ENSG00000138101.18    0.8556  Expression
37475    ZBED3            ENSG00000132846.5    0.8554  Expression
38383     ACHE           ENSG00000087085.13    0.8461  Expression
38011  L3MBTL3            ENSG00000198945.7    0.8426  Expression
37418     NPR3           ENSG00000113389.15    0.8420  Expression
37369     LRBA intron_4_150278004_150282450    0.8388    Splicing
37420   MTMR12           ENSG00000150712.10    0.8381  Expression
36185    RCSD1           ENSG00000198771.10    0.8329  Expression
41409    JMJD6           ENSG00000070495.14    0.8306  Expression
37118   PXYLP1           ENSG00000155893.12    0.8284  Expression
39582     KLF5           ENSG00000102554.13    0.8242  Expression
41886    ELANE            ENSG00000197561.6    0.8125  Expression
38127    JAZF1           ENSG00000153814.11    0.8109  Expression
37994     HSF2           ENSG00000025156.12    0.8108  Expression
39810  ZFP36L1            ENSG00000185650.9    0.8106  Expression
42600     LSM4                   cg15796753    0.8058 Methylation
38758     RBP4           ENSG00000138207.13    0.8035  Expression</code></pre>
</div>
<div id="top-genes-by-expressionsplicingmethylation-pip" class="section level3">
<h3>Top genes by expression/splicing/methylation pip</h3>
<pre class="r"><code>ctwas_gene_E_res &lt;- ctwas_gene_res[ctwas_gene_res$group==&quot;Expression&quot;,]
ctwas_gene_S_res &lt;- ctwas_gene_res[ctwas_gene_res$group==&quot;Splicing&quot;,]
ctwas_gene_M_res &lt;- ctwas_gene_res[ctwas_gene_res$group==&quot;Methylation&quot;,]

df_gene_E &lt;- aggregate(ctwas_gene_E_res$susie_pip,by=list(ctwas_gene_E_res$genename), FUN=sum)
colnames(df_gene_E) &lt;- c(&quot;genename&quot;, &quot;susie_pip&quot;)
df_gene_E$group &lt;- &quot;Expression&quot;

df_gene_S &lt;- aggregate(ctwas_gene_S_res$susie_pip,by=list(ctwas_gene_S_res$genename), FUN=sum)
colnames(df_gene_S) &lt;- c(&quot;genename&quot;, &quot;susie_pip&quot;)
df_gene_S$group &lt;- &quot;Splicing&quot;

df_gene_M &lt;- aggregate(ctwas_gene_M_res$susie_pip,by=list(ctwas_gene_M_res$genename), FUN=sum)
colnames(df_gene_M) &lt;- c(&quot;genename&quot;, &quot;susie_pip&quot;)
df_gene_M$group &lt;- &quot;Methylation&quot;

df_gene &lt;- rbind(df_gene_E,df_gene_S,df_gene_M)
head(df_gene[order(-df_gene$susie_pip),], max(sum(df_gene$susie_pip&gt;0.8), 20))</code></pre>
<pre><code>      genename susie_pip       group
13036    MYO1G    1.3514    Splicing
18333    NINJ2    1.1701 Methylation
15394  TSPAN32    1.1599    Splicing
16152     AMZ1    1.0639 Methylation
12354    ITSN2    1.0254    Splicing
14458   SIRPB1    1.0210    Splicing
18816   PTPRN2    1.0060 Methylation
12655      LYZ    1.0001    Splicing
11001     CNN2    1.0000    Splicing
2988       FES    1.0000  Expression
15750     YBEY    1.0000    Splicing
8211     TAGAP    1.0000  Expression
536   ARHGAP15    1.0000  Expression
812      BAZ2B    0.9999  Expression
9260     VLDLR    0.9999  Expression
13952    RAB34    0.9995    Splicing
1935     CREB5    0.9991  Expression
17144     ELK3    0.9989 Methylation
4774    MED12L    0.9981  Expression
4334    LAPTM5    0.9981  Expression
7574   SLC22A4    0.9966  Expression
750      ATXN7    0.9965  Expression
11772     FLT3    0.9955    Splicing
1761     CNIH4    0.9948  Expression
315      ALDH2    0.9941  Expression
10662     CANX    0.9939    Splicing
4324     LAMP1    0.9931  Expression
7591  SLC25A24    0.9924  Expression
3601     HDHD5    0.9912  Expression
19344   SPTLC2    0.9895 Methylation
8964    TTC39C    0.9872  Expression
9441     YPEL2    0.9846  Expression
4431      LIPA    0.9839  Expression
4188    KIF18B    0.9837  Expression
16046   ACVRL1    0.9824 Methylation
6614     PTPRJ    0.9824  Expression
15239 TNFRSF1A    0.9794    Splicing
517      ARAP2    0.9783  Expression
1896     CPEB4    0.9741  Expression
18244    MYO1C    0.9736 Methylation
9547     ZFPM1    0.9711  Expression
19006    RRBP1    0.9698 Methylation
5750     OSTF1    0.9691  Expression
4219     KLF12    0.9685  Expression
744      ATXN1    0.9616  Expression
5376      NEK6    0.9567  Expression
9163    UQCRC1    0.9555  Expression
3293      GLG1    0.9553  Expression
181       ADD1    0.9550  Expression
8342      TET2    0.9535  Expression
19040    SBNO2    0.9509 Methylation
7349      SELL    0.9505  Expression
1658    CITED4    0.9505  Expression
8245   TBC1D14    0.9442  Expression
13092     NCF2    0.9422    Splicing
3753    HS6ST1    0.9385  Expression
15137    TMCC2    0.9337    Splicing
1386     CD302    0.9303  Expression
16612      CD6    0.9266 Methylation
1422    CDC14A    0.9259  Expression
2212    DDX60L    0.9238  Expression
706    ATP13A1    0.9228  Expression
5206     MYO9B    0.9165  Expression
17736    ITGAL    0.9161 Methylation
452      AP1M2    0.9144  Expression
1345      CCR8    0.9073  Expression
12807  MFSD13A    0.9073    Splicing
13846     PSD4    0.9016    Splicing
5828     PAQR9    0.9002  Expression
8661   TNFAIP8    0.8984  Expression
19844    ZBTB2    0.8967 Methylation
6744     RAI14    0.8925  Expression
1912     CPSF4    0.8870  Expression
154     ADAM19    0.8862  Expression
15682    WDFY2    0.8849    Splicing
7716  SLC9A3R1    0.8776  Expression
36       ABCC5    0.8707  Expression
9069     UBE2O    0.8706  Expression
6880      REST    0.8678  Expression
10886    CERS5    0.8677    Splicing
9089     UBOX5    0.8644  Expression
13614    PLCB2    0.8623    Splicing
17282   FBRSL1    0.8596 Methylation
2424      DTNB    0.8556  Expression
9457     ZBED3    0.8554  Expression
91        ACHE    0.8461  Expression
4310   L3MBTL3    0.8426  Expression
5534      NPR3    0.8420  Expression
19723    UBE2I    0.8420 Methylation
12594     LRBA    0.8388    Splicing
5123    MTMR12    0.8381  Expression
6854     RCSD1    0.8329  Expression
4077     JMJD6    0.8306  Expression
6643    PXYLP1    0.8284  Expression
4223      KLF5    0.8242  Expression
10712  CCDC125    0.8207    Splicing
2581     ELANE    0.8125  Expression
4074     JAZF1    0.8109  Expression
3770      HSF2    0.8108  Expression
9537   ZFP36L1    0.8106  Expression
17992     LSM4    0.8058 Methylation
16212   APOLD1    0.8042 Methylation
6832      RBP4    0.8035  Expression</code></pre>
</div>
<div id="top-genes-by-combined-pip" class="section level3">
<h3>Top genes by combined PIP</h3>
<pre><code>      genename combined_pip expression_pip splicing_pip methylation_pip
11761    YPEL2       1.7531          0.985        0.000           0.768
5450    LAPTM5       1.7444          0.998        0.002           0.744
665   ARHGAP15       1.7228          1.000        0.680           0.042
6495     MYO1G       1.4333          0.082        1.351           0.000
9593   SLC45A4       1.3131          0.786        0.509           0.018
2770    DDX60L       1.2453          0.924        0.322           0.000
10814  TNFAIP8       1.2375          0.898        0.120           0.219
4508     HDHD5       1.2168          0.991        0.226           0.000
6779     NINJ2       1.2094          0.033        0.006           1.170
5772       LYZ       1.1825          0.182        1.000           0.000
9946    SPTLC2       1.1762          0.187        0.000           0.989
11126  TSPAN32       1.1634          0.004        1.160           0.000
8908     RRBP1       1.1285          0.066        0.093           0.970
9412   SLC12A7       1.0927          0.416        0.011           0.666
921      ATXN1       1.0859          0.962        0.058           0.067
98       ACAP1       1.0827          0.349        0.015           0.719
5067     ITSN2       1.0710          0.033        1.025           0.013
452       AMZ1       1.0639          0.000        0.000           1.064
9375    SIRPB1       1.0583          0.037        1.021           0.000
4984    IQGAP2       1.0558          0.443        0.589           0.024
228       ADD1       1.0546          0.955        0.100           0.000
7999     PPP5C       1.0500          0.436        0.004           0.610
11848  ZDHHC18       1.0485          0.009        0.590           0.450
6808     NLRC5       1.0472          0.676        0.074           0.297
3336     EOMES       1.0430          0.645        0.009           0.389
3258      ELK3       1.0411          0.031        0.012           0.999
927      ATXN7       1.0395          0.997        0.043           0.000
3663    FBRSL1       1.0389          0.063        0.117           0.860
10828 TNFRSF1A       1.0375          0.016        0.979           0.042
8263     PTPRJ       1.0347          0.982        0.015           0.038
10315  TBC1D14       1.0311          0.944        0.062           0.025
1699     CD101       1.0281          0.702        0.006           0.320
2203     CNIH4       1.0177          0.995        0.023           0.000
5984    MED12L       1.0147          0.998        0.000           0.017
384      ALDH2       1.0137          0.994        0.020           0.000
3755       FES       1.0119          1.000        0.012           0.000
5296     KLF12       1.0068          0.968        0.038           0.000
8267    PTPRN2       1.0060          0.000        0.000           1.006
3398    ERICH1       1.0036          0.046        0.203           0.754
11531    VLDLR       1.0036          1.000        0.004           0.000
2424     CREB5       1.0014          0.999        0.002           0.000
10273    TAGAP       1.0007          1.000        0.001           0.000
8363     RAB34       1.0006          0.001        1.000           0.000
1010     BAZ2B       1.0005          1.000        0.001           0.000
2206      CNN2       1.0001          0.000        1.000           0.000
11741     YBEY       1.0000          0.000        1.000           0.000
6491     MYO1C       0.9998          0.026        0.000           0.974
2377     CPEB4       0.9996          0.974        0.000           0.025
639      ARAP2       0.9992          0.978        0.021           0.000
9455   SLC22A4       0.9982          0.997        0.002           0.000
9476  SLC25A24       0.9967          0.992        0.004           0.000
3829      FLT3       0.9955          0.000        0.995           0.000
1445      CANX       0.9939          0.000        0.994           0.000
5437     LAMP1       0.9931          0.993        0.000           0.000
7197     OSTF1       0.9924          0.969        0.023           0.000
8810      RORC       0.9913          0.776        0.000           0.216
5567      LIPA       0.9908          0.984        0.006           0.001
1732      CD33       0.9902          0.039        0.767           0.184
11173   TTC39C       0.9878          0.987        0.001           0.000
7704     PLCB2       0.9837          0.121        0.862           0.000
5253    KIF18B       0.9837          0.984        0.000           0.000
175     ACVRL1       0.9824          0.000        0.000           0.982
10956 TRAF3IP3       0.9776          0.686        0.292           0.000
10605    TMCC2       0.9776          0.044        0.934           0.000
11888    ZFPM1       0.9726          0.971        0.001           0.000
11295    UBE2I       0.9675          0.067        0.059           0.842
6709      NEK6       0.9663          0.957        0.004           0.006
7054     NUP88       0.9649          0.172        0.767           0.026
3055      DTNB       0.9621          0.856        0.005           0.102
4139      GLG1       0.9580          0.955        0.003           0.000
9609   SLC5A11       0.9579          0.302        0.000           0.656
11326    UBOX5       0.9568          0.864        0.092           0.000
10429     TET2       0.9561          0.953        0.003           0.000
11413   UQCRC1       0.9555          0.956        0.000           0.000
9172      SELL       0.9551          0.951        0.005           0.000
4698    HS6ST1       0.9546          0.939        0.000           0.016
9046     SBNO2       0.9517          0.000        0.001           0.951
2078    CITED4       0.9505          0.951        0.000           0.000
6162       MLX       0.9491          0.735        0.214           0.000
6611      NCF2       0.9422          0.000        0.942           0.000
11789    ZBTB2       0.9364          0.040        0.000           0.897
4375      GSAP       0.9319          0.630        0.262           0.040
1730     CD302       0.9303          0.930        0.000           0.000
1750       CD6       0.9275          0.000        0.001           0.927
1771    CDC14A       0.9259          0.926        0.000           0.000
867    ATP13A1       0.9248          0.923        0.002           0.000
1299  C20orf96       0.9196          0.721        0.060           0.138
11614    WDFY2       0.9185          0.000        0.885           0.034
46       ABCC5       0.9167          0.871        0.019           0.027
6504     MYO9B       0.9166          0.917        0.000           0.000
5036     ITGAL       0.9161          0.000        0.000           0.916
562      AP1M2       0.9144          0.914        0.000           0.000
5219  KIAA0040       0.9139          0.067        0.085           0.762
8422     RAI14       0.9131          0.892        0.000           0.021
1684      CCR8       0.9073          0.907        0.000           0.000
6077   MFSD13A       0.9073          0.000        0.907           0.000
6401    MTMR12       0.9019          0.838        0.064           0.000
8152      PSD4       0.9016          0.000        0.902           0.000
7296     PAQR9       0.9002          0.900        0.000           0.000
186     ADAM19       0.8995          0.886        0.005           0.008
9005      SAE1       0.8945          0.622        0.272           0.000
2396     CPSF4       0.8870          0.887        0.000           0.000
5841    MAP2K5       0.8800          0.070        0.030           0.780
9639  SLC9A3R1       0.8776          0.878        0.000           0.000
1733      CD36       0.8767          0.563        0.314           0.000
5089     JAZF1       0.8744          0.811        0.063           0.000
5661    LRRC25       0.8725          0.735        0.137           0.000
11302    UBE2O       0.8711          0.871        0.001           0.000
7015    NUDT14       0.8711          0.216        0.557           0.098
5420   L3MBTL3       0.8697          0.843        0.000           0.027
1945     CERS5       0.8680          0.000        0.868           0.000
8596      REST       0.8678          0.868        0.000           0.000
5632      LRBA       0.8612          0.000        0.839           0.022
1626     CCDC9       0.8583          0.667        0.191           0.000
2958    DNASE1       0.8580          0.551        0.007           0.300
11782    ZBED3       0.8554          0.855        0.000           0.000
1547   CCDC125       0.8550          0.034        0.821           0.000
114       ACHE       0.8510          0.846        0.005           0.000
8234     PTK2B       0.8493          0.739        0.053           0.057
8566     RCSD1       0.8459          0.833        0.008           0.005
8145    PRUNE2       0.8427          0.799        0.006           0.038
6906      NPR3       0.8420          0.842        0.000           0.000
5094     JMJD6       0.8357          0.831        0.005           0.000
8300    PXYLP1       0.8357          0.828        0.007           0.000
759      ARRB2       0.8356          0.281        0.554           0.000
3263     ELMO1       0.8347          0.702        0.046           0.086
624     APOLD1       0.8331          0.029        0.000           0.804
5301      KLF5       0.8242          0.824        0.000           0.000
5712      LSM4       0.8177          0.011        0.001           0.806
1502  CATSPER2       0.8145          0.792        0.000           0.023
3251     ELANE       0.8130          0.813        0.000           0.000
8935      RSU1       0.8115          0.037        0.270           0.505
4716      HSF2       0.8108          0.811        0.000           0.000
11876  ZFP36L1       0.8106          0.811        0.000           0.000
674   ARHGAP26       0.8096          0.000        0.022           0.788
694   ARHGEF12       0.8087          0.762        0.046           0.000
5893     MARK3       0.8072          0.696        0.109           0.002
12115   ZNF516       0.8067          0.792        0.004           0.011
182     ADAM10       0.8067          0.798        0.009           0.000
8541      RBP4       0.8035          0.804        0.000           0.000</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.5</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 140</code></pre>
<pre><code>Uploading data to Enrichr... Done.
  Querying GO_Biological_Process_2021... Done.
  Querying GO_Cellular_Component_2021... Done.
  Querying GO_Molecular_Function_2021... Done.
Parsing results... Done.
[1] &quot;GO_Biological_Process_2021&quot;</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/fa89b375daf79a7a1843d31ae9532f0f5797b29f/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-1.png" target="_blank">fa89b37</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6286c345d7d7de0c6a56bd29a0792988cb78f287/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-1.png" target="_blank">6286c34</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                                            Term Overlap
1 neutrophil activation involved in immune response (GO:0002283)  15/485
2       amyloid precursor protein metabolic process (GO:0042982)    4/18
3                          neutrophil degranulation (GO:0043312)  14/481
4                      neutrophil mediated immunity (GO:0002446)  14/488
5                        adipose tissue development (GO:0060612)    3/12
  Adjusted.P.value
1         0.002091
2         0.002795
3         0.002795
4         0.002795
5         0.018207
                                                                                     Genes
1 PTPRN2;ADAM10;PTPRJ;IQGAP2;LYZ;ITGAL;SIRPB1;CNN2;LAMP1;SELL;OSTF1;CD36;DNASE1;ELANE;CD33
2                                                                 ADAM19;ACHE;ADAM10;TMCC2
3        PTPRN2;ADAM10;PTPRJ;IQGAP2;LYZ;ITGAL;SIRPB1;CNN2;LAMP1;SELL;OSTF1;CD36;ELANE;CD33
4        PTPRN2;ADAM10;PTPRJ;IQGAP2;LYZ;ITGAL;SIRPB1;CNN2;LAMP1;SELL;OSTF1;CD36;ELANE;CD33
5                                                                       SPTLC2;ZNF516;RORC
[1] &quot;GO_Cellular_Component_2021&quot;</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-2">
Past versions of unnamed-chunk-13-2.png
</button>
</p>
<div id="fig-unnamed-chunk-13-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/fa89b375daf79a7a1843d31ae9532f0f5797b29f/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-2.png" target="_blank">fa89b37</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6286c345d7d7de0c6a56bd29a0792988cb78f287/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-2.png" target="_blank">6286c34</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>                                      Term Overlap Adjusted.P.value
1            specific granule (GO:0042581)   8/160         0.001436
2  secretory granule membrane (GO:0030667)  10/274         0.001436
3 actin-based cell projection (GO:0098858)    5/83         0.011822
4   specific granule membrane (GO:0035579)    5/91         0.013581
5            tertiary granule (GO:0070820)   6/164         0.025494
                                                         Genes
1                  CNN2;ADAM10;PTPRJ;CD36;LYZ;ITGAL;ELANE;CD33
2 PTPRN2;LAMP1;SELL;ADAM10;PTPRJ;CD36;IQGAP2;ITGAL;SIRPB1;CD33
3                            SLC9A3R1;MYO1C;CD302;IQGAP2;MYO1G
4                                 ADAM10;PTPRJ;CD36;ITGAL;CD33
5                            CNN2;PTPRN2;LAMP1;ADAM10;LYZ;CD33
[1] &quot;GO_Molecular_Function_2021&quot;</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-3">
Past versions of unnamed-chunk-13-3.png
</button>
</p>
<div id="fig-unnamed-chunk-13-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/fa89b375daf79a7a1843d31ae9532f0f5797b29f/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-3.png" target="_blank">fa89b37</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/6286c345d7d7de0c6a56bd29a0792988cb78f287/docs/figure/WhiteBlood_WholeBlood_E_S_M.Rmd/unnamed-chunk-13-3.png" target="_blank">6286c34</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>[1] Term             Overlap          Adjusted.P.value Genes           
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                              Description     FDR Ratio  BgRatio
14  Refractory anaemia with excess blasts 0.09276  1/83   1/9703
45      Cholesterol Ester Storage Disease 0.09276  1/83   1/9703
71                              Epistaxis 0.09276  1/83   1/9703
77                               Freckles 0.09276  1/83   1/9703
108          Acute Promyelocytic Leukemia 0.09276  3/83  46/9703
111         Liver Cirrhosis, Experimental 0.09276 15/83 774/9703
121                             Melanosis 0.09276  1/83   1/9703
122                              Chloasma 0.09276  1/83   1/9703
175                        Telangiectasis 0.09276  1/83   1/9703
186                        Wolman Disease 0.09276  1/83   1/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] WebGestaltR_0.4.4 disgenet2r_0.99.2 enrichR_3.1       cowplot_1.1.1    
[5] ggplot2_3.4.0     workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] httr_1.4.4        sass_0.4.4        bit64_4.0.5       vroom_1.6.0      
 [5] jsonlite_1.8.4    foreach_1.5.2     bslib_0.4.1       assertthat_0.2.1 
 [9] getPass_0.2-2     highr_0.9         doRNG_1.8.2       yaml_2.3.6       
[13] pillar_1.8.1      lattice_0.20-44   glue_1.6.2        digest_0.6.31    
[17] promises_1.2.0.1  colorspace_2.0-3  htmltools_0.5.4   httpuv_1.6.7     
[21] Matrix_1.3-3      plyr_1.8.8        pkgconfig_2.0.3   scales_1.2.1     
[25] processx_3.8.0    svglite_2.1.0     whisker_0.4.1     later_1.3.0      
[29] tzdb_0.3.0        git2r_0.30.1      tibble_3.1.8      generics_0.1.3   
[33] farver_2.1.0      ellipsis_0.3.2    cachem_1.0.6      withr_2.5.0      
[37] cli_3.4.1         crayon_1.5.2      magrittr_2.0.3    evaluate_0.19    
[41] ps_1.7.2          apcluster_1.4.10  fs_1.5.2          fansi_1.0.3      
[45] doParallel_1.0.17 tools_4.1.0       data.table_1.14.6 hms_1.1.2        
[49] lifecycle_1.0.3   stringr_1.5.0     munsell_0.5.0     rngtools_1.5.2   
[53] callr_3.7.3       compiler_4.1.0    jquerylib_0.1.4   systemfonts_1.0.4
[57] rlang_1.0.6       grid_4.1.0        iterators_1.0.14  rstudioapi_0.14  
[61] rjson_0.2.21      igraph_1.3.5      labeling_0.4.2    rmarkdown_2.19   
[65] gtable_0.3.1      codetools_0.2-18  DBI_1.1.3         curl_4.3.2       
[69] reshape2_1.4.4    R6_2.5.1          knitr_1.41        dplyr_1.0.10     
[73] bit_4.0.5         fastmap_1.1.0     utf8_1.2.2        rprojroot_2.0.3  
[77] readr_2.1.3       stringi_1.7.8     parallel_4.1.0    Rcpp_1.0.9       
[81] vctrs_0.5.1       tidyselect_1.2.0  xfun_0.35        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
