<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="sheng Qian" />


<title>WhiteBlood - WholeBlood gene expression</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">cTWAS_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/cTWAS_analysis">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">WhiteBlood - WholeBlood gene expression</h1>
<h4 class="author">sheng Qian</h4>
<h4 class="date">2023-2-1</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-02-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>cTWAS_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/
</td>
<td style="text-align:left;">
data
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config_b38.R
</td>
<td style="text-align:left;">
code/ctwas_config_b38.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/G_list.RData
</td>
<td style="text-align:left;">
data/G_list.RData
</td>
</tr>
<tr>
<td style="text-align:left;">
/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/locus_plot.R
</td>
<td style="text-align:left;">
code/locus_plot.R
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree097dbbcfab7aceda6ae774198bae7627d23e6bb2targetblank097dbbca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/cTWAS_analysis/tree/097dbbcfab7aceda6ae774198bae7627d23e6bb2" target="_blank">097dbbc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96cTWASanalysistree097dbbcfab7aceda6ae774198bae7627d23e6bb2targetblank097dbbca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/cTWAS_analysis/tree/097dbbcfab7aceda6ae774198bae7627d23e6bb2" target="_blank">097dbbc</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .ipynb_checkpoints/

Untracked files:
    Untracked:  Proposal plots.R
    Untracked:  RGS14.pdf
    Untracked:  RNF186.pdf
    Untracked:  SCZ_annotation.xlsx
    Untracked:  SLC8B1.pdf
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  cache/
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  data/.ipynb_checkpoints/
    Untracked:  data/FUMA_output/
    Untracked:  data/GO_Terms/
    Untracked:  data/GTEx_Analysis_v8_eQTL.tar
    Untracked:  data/G_list.RData
    Untracked:  data/IBD_ME/
    Untracked:  data/LDL/
    Untracked:  data/LDL_E_S/
    Untracked:  data/LDL_M/
    Untracked:  data/LDL_S/
    Untracked:  data/PGC3_SCZ_wave3_public.v2.tsv
    Untracked:  data/SCZ/
    Untracked:  data/SCZ_2018/
    Untracked:  data/SCZ_2018_S/
    Untracked:  data/SCZ_2020/
    Untracked:  data/SCZ_S/
    Untracked:  data/Supplementary Table 15 - MAGMA.xlsx
    Untracked:  data/Supplementary Table 20 - Prioritised Genes.xlsx
    Untracked:  data/UKBB/
    Untracked:  data/UKBB_SNPs_Info.text
    Untracked:  data/WhiteBlood_E/
    Untracked:  data/WhiteBlood_E_M/
    Untracked:  data/WhiteBlood_E_S_M/
    Untracked:  data/WhiteBlood_M/
    Untracked:  data/cpg_annot.RData
    Untracked:  data/eqtl/
    Untracked:  data/gencode.v26.GRCh38.genes.gtf
    Untracked:  data/gene_OMIM.txt
    Untracked:  data/gene_pip_0.8.txt
    Untracked:  data/gwas_sumstats/
    Untracked:  data/magma.genes.out
    Untracked:  data/mashr_Heart_Atrial_Appendage.db
    Untracked:  data/mashr_sqtl/
    Untracked:  data/mqtl/
    Untracked:  data/notes.txt
    Untracked:  data/scz_2018.RDS
    Untracked:  data/summary_known_genes_annotations.xlsx
    Untracked:  temp_LDR/

Unstaged changes:
    Deleted:    analysis/Atrial_Fibrillation_Heart_Atrial_Appendage.Rmd
    Deleted:    analysis/Atrial_Fibrillation_Heart_Left_Ventricle.Rmd
    Deleted:    analysis/Autism_Brain_Amygdala.Rmd
    Deleted:    analysis/Autism_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Deleted:    analysis/Autism_Brain_Caudate_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Cerebellar_Hemisphere.Rmd
    Deleted:    analysis/Autism_Brain_Cerebellum.Rmd
    Deleted:    analysis/Autism_Brain_Cortex.Rmd
    Deleted:    analysis/Autism_Brain_Frontal_Cortex_BA9.Rmd
    Deleted:    analysis/Autism_Brain_Hippocampus.Rmd
    Deleted:    analysis/Autism_Brain_Hypothalamus.Rmd
    Deleted:    analysis/Autism_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Putamen_basal_ganglia.Rmd
    Deleted:    analysis/Autism_Brain_Spinal_cord_cervical_c-1.Rmd
    Deleted:    analysis/Autism_Brain_Substantia_nigra.Rmd
    Deleted:    analysis/BMI_Brain_Amygdala.Rmd
    Deleted:    analysis/BMI_Brain_Amygdala_S.Rmd
    Deleted:    analysis/BMI_Brain_Anterior_cingulate_cortex_BA24.Rmd
    Deleted:    analysis/BMI_Brain_Anterior_cingulate_cortex_BA24_S.Rmd
    Deleted:    analysis/BMI_Brain_Caudate_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Caudate_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellar_Hemisphere.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellar_Hemisphere_S.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellum.Rmd
    Deleted:    analysis/BMI_Brain_Cerebellum_S.Rmd
    Deleted:    analysis/BMI_Brain_Cortex.Rmd
    Deleted:    analysis/BMI_Brain_Cortex_S.Rmd
    Deleted:    analysis/BMI_Brain_Frontal_Cortex_BA9.Rmd
    Deleted:    analysis/BMI_Brain_Frontal_Cortex_BA9_S.Rmd
    Deleted:    analysis/BMI_Brain_Hippocampus.Rmd
    Deleted:    analysis/BMI_Brain_Hippocampus_S.Rmd
    Deleted:    analysis/BMI_Brain_Hypothalamus.Rmd
    Deleted:    analysis/BMI_Brain_Hypothalamus_S.Rmd
    Deleted:    analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Nucleus_accumbens_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Putamen_basal_ganglia.Rmd
    Deleted:    analysis/BMI_Brain_Putamen_basal_ganglia_S.Rmd
    Deleted:    analysis/BMI_Brain_Spinal_cord_cervical_c-1.Rmd
    Deleted:    analysis/BMI_Brain_Spinal_cord_cervical_c-1_S.Rmd
    Deleted:    analysis/BMI_Brain_Substantia_nigra.Rmd
    Deleted:    analysis/BMI_Brain_Substantia_nigra_S.Rmd
    Deleted:    analysis/BMI_S_results.Rmd
    Deleted:    analysis/Glucose_Adipose_Subcutaneous.Rmd
    Deleted:    analysis/Glucose_Adipose_Visceral_Omentum.Rmd
    Modified:   analysis/LDL_Liver_GTEX.Rmd
    Modified:   analysis/LDL_Liver_GTEX_mashr.Rmd
    Modified:   analysis/WhiteBlood_WholeBlood_E.Rmd
    Modified:   analysis/WhiteBlood_WholeBlood_E_M.Rmd
    Modified:   analysis/WhiteBlood_WholeBlood_E_S_M.Rmd
    Modified:   analysis/WhiteBlood_WholeBlood_M.Rmd
    Modified:   code/LDL_out/LDL_Liver_GTEx_mashr.err
    Modified:   code/LDL_out/LDL_Liver_GTEx_mashr.out
    Deleted:    code/White_Blood_M_out/White_Blood_BreastMammary.err
    Deleted:    code/White_Blood_M_out/White_Blood_BreastMammary.out
    Deleted:    code/White_Blood_M_out/White_Blood_ColonTransverse.err
    Deleted:    code/White_Blood_M_out/White_Blood_ColonTransverse.out
    Deleted:    code/White_Blood_M_out/White_Blood_KidneyCortex.err
    Deleted:    code/White_Blood_M_out/White_Blood_KidneyCortex.out
    Deleted:    code/White_Blood_M_out/White_Blood_Lung.err
    Deleted:    code/White_Blood_M_out/White_Blood_Lung.out
    Deleted:    code/White_Blood_M_out/White_Blood_MuscleSkeletal.err
    Deleted:    code/White_Blood_M_out/White_Blood_MuscleSkeletal.out
    Deleted:    code/White_Blood_M_out/White_Blood_Ovary.err
    Deleted:    code/White_Blood_M_out/White_Blood_Ovary.out
    Deleted:    code/White_Blood_M_out/White_Blood_Prostate.err
    Deleted:    code/White_Blood_M_out/White_Blood_Prostate.out
    Deleted:    code/White_Blood_M_out/White_Blood_Testis.err
    Deleted:    code/White_Blood_M_out/White_Blood_Testis.out
    Deleted:    code/White_Blood_M_out/White_Blood_WholeBlood.err
    Deleted:    code/White_Blood_M_out/White_Blood_WholeBlood.out
    Deleted:    code/run_IBD_ctwas_rss_LDR_ME.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/WhiteBlood_WholeBlood_E.Rmd</code>) and HTML (<code>docs/WhiteBlood_WholeBlood_E.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/ada18289f58faa1f9f2d114fa14f06e3381e0038/analysis/WhiteBlood_WholeBlood_E.Rmd" target="_blank">ada1828</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-12
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/f49450b80a513f9fbf46e75e90d0537027e02495/analysis/WhiteBlood_WholeBlood_E.Rmd" target="_blank">f49450b</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-04
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/f49450b80a513f9fbf46e75e90d0537027e02495/docs/WhiteBlood_WholeBlood_E.html" target="_blank">f49450b</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-04
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/da7a0f70b9796e45b45ac2e60d2b3e3386af7a7a/analysis/WhiteBlood_WholeBlood_E.Rmd" target="_blank">da7a0f7</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/da7a0f70b9796e45b45ac2e60d2b3e3386af7a7a/docs/WhiteBlood_WholeBlood_E.html" target="_blank">da7a0f7</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/66590cbc75359c3989119974c69a4d376d0caa57/analysis/WhiteBlood_WholeBlood_E.Rmd" target="_blank">66590cb</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/66590cbc75359c3989119974c69a4d376d0caa57/docs/WhiteBlood_WholeBlood_E.html" target="_blank">66590cb</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/9b01dadf23cff76bebe62e9c24e4a0996e8c5d84/analysis/WhiteBlood_WholeBlood_E.Rmd" target="_blank">9b01dad</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-01
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/cTWAS_analysis/9b01dadf23cff76bebe62e9c24e4a0996e8c5d84/docs/WhiteBlood_WholeBlood_E.html" target="_blank">9b01dad</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-01
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="weight-qc" class="section level2">
<h2>Weight QC</h2>
<pre><code>[1] 12623</code></pre>
<pre><code>[1] 11198</code></pre>
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
1137  750  626  401  487  626  562  394  412  432  686  659  194  368  335  553 
  17   18   19   20   21   22 
 731  163  925  315  131  311 </code></pre>
<pre><code>[1] 0.7637</code></pre>
</div>
<div id="load-ctwas-results" class="section level2">
<h2>Load ctwas results</h2>
<div id="check-convergence-of-parameters" class="section level3">
<h3>Check convergence of parameters</h3>
<p><img src="figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#estimated group prior
estimated_group_prior &lt;- estimated_group_prior_all[,ncol(group_prior_rec)]
print(estimated_group_prior)</code></pre>
<pre><code>      SNP      gene 
0.0002005 0.0230977 </code></pre>
<pre class="r"><code>#estimated group prior variance
estimated_group_prior_var &lt;- estimated_group_prior_var_all[,ncol(group_prior_var_rec)]
print(estimated_group_prior_var)</code></pre>
<pre><code>  SNP  gene 
17.23 20.04 </code></pre>
<pre class="r"><code>#estimated enrichment
estimated_enrichment &lt;- estimated_enrichment_all[ncol(group_prior_var_rec)]
print(estimated_enrichment)</code></pre>
<pre><code>[1] 115.2</code></pre>
<pre class="r"><code>#report sample size
print(sample_size)</code></pre>
<pre><code>[1] 350470</code></pre>
<pre class="r"><code>#report group size
print(group_size)</code></pre>
<pre><code>    SNP    gene 
8696600   11198 </code></pre>
<pre class="r"><code>#estimated group PVE
estimated_group_pve &lt;- estimated_group_pve_all[,ncol(group_prior_rec)]
print(estimated_group_pve)</code></pre>
<pre><code>    SNP    gene 
0.08573 0.01479 </code></pre>
<pre class="r"><code>#total PVE
sum(estimated_group_pve)</code></pre>
<pre><code>[1] 0.1005</code></pre>
<pre class="r"><code>#attributable PVE
estimated_group_pve/sum(estimated_group_pve)</code></pre>
<pre><code>   SNP   gene 
0.8529 0.1471 </code></pre>
</div>
</div>
<div id="genes-with-highest-pips" class="section level2">
<h2>Genes with highest PIPs</h2>
<pre class="r"><code>#distribution of PIPs
hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main=&quot;Distribution of Gene PIPs&quot;)</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/9b01dadf23cff76bebe62e9c24e4a0996e8c5d84/docs/figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-6-1.png" target="_blank">9b01dad</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#genes with PIP&gt;0.8 or 20 highest PIPs
head(ctwas_gene_res[order(-ctwas_gene_res$susie_pip),report_cols], max(sum(ctwas_gene_res$susie_pip&gt;0.8), 20))</code></pre>
<pre><code>        genename region_tag susie_pip    mu2       PVE       z num_eqtl
9507         FES      15_43    1.0000  75.21 2.146e-04  -8.837        3
7481       TAGAP      6_103    1.0000  71.50 2.040e-04  -8.435        2
893     ARHGAP15       2_85    1.0000  51.47 1.469e-04   9.193        3
3646       BAZ2B       2_96    0.9999  73.40 2.094e-04  11.102        2
5966       VLDLR        9_3    0.9999  54.07 1.543e-04   7.817        4
1640    KIAA0391       14_9    0.9997  47.50 1.355e-04   7.370        2
7070      LAPTM5       1_20    0.9991  69.47 1.980e-04   9.117        3
10398    SLC22A4       5_79    0.9990 144.87 4.129e-04  13.754        2
5767      MED12L       3_93    0.9979  32.98 9.392e-05  -5.427        2
5908       CREB5       7_24    0.9978 368.79 1.050e-03 -20.722        1
5665       CNIH4      1_114    0.9977 102.42 2.915e-04  -9.422        2
7272       ATXN7       3_43    0.9966  48.09 1.367e-04  -3.630        3
4571       CD101       1_72    0.9959  39.88 1.133e-04   6.274        3
2611       ALDH2      12_67    0.9940 140.78 3.993e-04 -15.815        3
2818     SLC12A7        5_2    0.9933  41.51 1.177e-04   6.027        4
2131     ATP13A1      19_15    0.9932  41.79 1.184e-04   6.167        2
9863       LAMP1      13_62    0.9919  39.52 1.119e-04  -6.303        1
1102    SLC25A24       1_67    0.9914  35.48 1.004e-04   5.941        3
736        HDHD5       22_1    0.9910  25.04 7.079e-05   4.132        3
2312        LIPA      10_57    0.9886  41.77 1.178e-04   6.386        4
10100       SELL       1_83    0.9875  25.36 7.145e-05   3.904        3
5360       NLRC5      16_31    0.9859  44.70 1.258e-04   6.576        2
8044      TTC39C      18_12    0.9851  40.30 1.133e-04   5.211        1
9899      KIF18B      17_26    0.9822  26.92 7.545e-05   5.374        1
1603      SPTLC2      14_36    0.9803  23.87 6.677e-05  -4.039        2
6064       PTPRJ      11_29    0.9791  67.95 1.898e-04  -9.818        2
8108        TET2       4_69    0.9753  24.58 6.839e-05  -5.284        2
412        ARAP2       4_30    0.9741  66.74 1.855e-04  -8.262        2
6686   HIST1H2BD       6_20    0.9706  62.56 1.732e-04   9.575        1
4658       OSTF1       9_35    0.9700  21.78 6.028e-05   4.248        3
9410      DDX60L      4_109    0.9694  21.89 6.056e-05   4.461        5
9272       ZFPM1      16_54    0.9655  36.82 1.014e-04  -4.645        1
171       UQCRC1       3_34    0.9654  29.53 8.135e-05  -5.030        1
2844       CPEB4      5_104    0.9652 123.69 3.407e-04  12.452        2
3293       KLF12      13_36    0.9636  39.69 1.091e-04  -6.340        1
811        ACAP1       17_6    0.9635  62.98 1.732e-04   7.733        2
1426      POLR2E       19_2    0.9625  34.89 9.582e-05  -5.383        5
3323        NEK6       9_64    0.9573  25.83 7.057e-05   5.706        2
9755       UBOX5       20_5    0.9563  27.79 7.582e-05  -4.863        1
1160        ADD1        4_4    0.9540  33.19 9.035e-05  -7.073        1
3758       ATXN1       6_13    0.9531  65.48 1.781e-04   8.173        1
1273        GLG1      16_40    0.9503  24.89 6.748e-05   4.683        2
9287      CITED4       1_25    0.9451  27.11 7.311e-05  -4.750        2
2410         MLX      17_25    0.9418  56.88 1.529e-04   7.850        2
4385     TBC1D14        4_8    0.9394  28.71 7.694e-05   6.255        1
4883      HS6ST1       2_75    0.9379  20.23 5.413e-05  -4.140        1
10114      PAQR9       3_87    0.9354  21.31 5.689e-05  -4.082        2
982       CDC14A       1_61    0.9274  19.52 5.166e-05   3.825        2
10454      ELANE       19_2    0.9255  24.67 6.516e-05  -4.552        2
11564      CD302       2_96    0.9255  33.18 8.761e-05  -6.789        4
4103       AP1M2       19_9    0.9234  39.13 1.031e-04   5.099        4
1408       MYO9B      19_14    0.9084  28.49 7.385e-05   5.238        1
1145        ACHE       7_62    0.9074  36.67 9.493e-05  -3.852        1
8131      RNF181       2_54    0.9056  35.98 9.296e-05  -5.029        1
574         CA11      19_33    0.8993  33.13 8.502e-05  -5.574        2
380        RAI14       5_23    0.8939  19.21 4.899e-05   3.788        1
2053       CCDC9      19_33    0.8920  38.30 9.748e-05   6.874        3
9299        CCR8       3_28    0.8914  21.87 5.563e-05  -2.931        1
1386       ITPR3       6_28    0.8913  40.33 1.026e-04   6.228        5
11657 RNF139-AS1       8_82    0.8843  22.67 5.719e-05   4.450        2
5598        RORC       1_74    0.8837  20.27 5.111e-05   4.101        1
162     TRAF3IP3      1_106    0.8836  24.47 6.170e-05   4.756        2
4670      ADAM19       5_93    0.8827  22.67 5.708e-05   4.089        2
6935       CPSF4       7_61    0.8817  52.18 1.313e-04  -7.253        2
208        PPP5C      19_32    0.8812  25.24 6.346e-05  -4.940        2
5834     TNFAIP8       5_72    0.8764  54.49 1.363e-04   7.624        1
2437    SLC9A3R1      17_42    0.8750  47.02 1.174e-04  -7.630        1
2447       RAB34      17_18    0.8747  23.97 5.983e-05  -4.508        2
5078        DTNB       2_15    0.8737  22.06 5.498e-05  -4.590        2
7233       EOMES       3_20    0.8598  55.96 1.373e-04   7.596        1
8952       UBE2O      17_43    0.8587  27.63 6.769e-05  -5.502        2
755        JMJD6      17_43    0.8561  25.17 6.147e-05   4.742        1
6143      MTMR12       5_22    0.8390  20.75 4.966e-05  -4.003        1
12096  LINC01970      17_47    0.8361  31.57 7.533e-05  -5.271        1
1074        REST       4_41    0.8315  96.36 2.286e-04   9.019        1
9085        GPR4      19_32    0.8299  20.77 4.919e-05   4.252        1
8907      LRRC25      19_15    0.8298  27.13 6.425e-05  -4.768        1
2813        NPR3       5_22    0.8297  21.29 5.041e-05   4.146        1
3430       SMAD9      13_13    0.8271  22.52 5.315e-05  -4.407        2
11105       MEG3      14_52    0.8256  33.96 8.000e-05   5.342        1
4448       ZBED3       5_45    0.8219  19.79 4.641e-05   3.802        2
10656      RCSD1       1_82    0.8196  22.26 5.205e-05   4.345        3
323       RABEP1       17_5    0.8177  61.95 1.445e-04   8.751        2
10280   C20orf96       20_1    0.8123  19.78 4.584e-05  -3.889        2
6513      PXYLP1       3_86    0.8073  31.62 7.283e-05   7.219        2
1768        KLF5      13_35    0.8028  23.54 5.392e-05  -4.513        1</code></pre>
</div>
<div id="go-enrichment-analysis-for-genes-with-pip0.8" class="section level2">
<h2>GO enrichment analysis for genes with PIP&gt;0.8</h2>
<pre class="r"><code>#number of genes for gene set enrichment
length(genes)</code></pre>
<pre><code>[1] 86</code></pre>
</div>
<div id="disgenet-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>DisGeNET enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>                              Description     FDR Ratio  BgRatio
11  Refractory anaemia with excess blasts 0.05522  1/45   1/9703
19  Malignant neoplasm of urinary bladder 0.05522  4/45 141/9703
20                       Bladder Neoplasm 0.05522  4/45 140/9703
28      Cholesterol Ester Storage Disease 0.05522  1/45   1/9703
49                               Freckles 0.05522  1/45   1/9703
70                              Melanosis 0.05522  1/45   1/9703
71                               Chloasma 0.05522  1/45   1/9703
112                        Wolman Disease 0.05522  1/45   1/9703
133                    Cyclic neutropenia 0.05522  1/45   1/9703
134                Cerebellar Gait Ataxia 0.05522  1/45   1/9703</code></pre>
</div>
<div id="webgestalt-enrichment-analysis-for-genes-with-pip0.5" class="section level2">
<h2>WebGestalt enrichment analysis for genes with PIP&gt;0.5</h2>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...
Performing the enrichment analysis...</code></pre>
<pre><code>Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
<pre><code>NULL</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre class="r"><code>a &lt;- locus_plot(region_tag=&quot;17_34&quot;, return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel=&quot;both&quot;,
                      legend_side=&quot;left&quot;,
                      legend_panel=&quot;&quot;)</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>a &lt;- locus_plot(region_tag=&quot;19_32&quot;, return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel=&quot;both&quot;,
                      legend_side=&quot;left&quot;,
                      legend_panel=&quot;&quot;)</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>a &lt;- locus_plot(region_tag=&quot;12_1&quot;, return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel=&quot;both&quot;,
                      legend_side=&quot;left&quot;,
                      legend_panel=&quot;&quot;)</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/cTWAS_analysis/blob/66590cbc75359c3989119974c69a4d376d0caa57/docs/figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-15-1.png" target="_blank">66590cb</a>
</td>
<td>
sq-96
</td>
<td>
2023-02-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>a &lt;- locus_plot(region_tag=&quot;2_14&quot;, return_table=T,
                      focus=NULL,
                      label_genes=NULL,
                      rerun_ctwas=F,
                      rerun_load_only=F,
                      label_panel=&quot;both&quot;,
                      legend_side=&quot;left&quot;,
                      legend_panel=&quot;&quot;)</code></pre>
<p><img src="figure/WhiteBlood_WholeBlood_E.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] Gviz_1.38.4          GenomicRanges_1.46.1 GenomeInfoDb_1.30.1 
 [4] IRanges_2.28.0       S4Vectors_0.32.4     BiocGenerics_0.40.0 
 [7] WebGestaltR_0.4.4    disgenet2r_0.99.2    enrichR_3.1         
[10] cowplot_1.1.1        ggplot2_3.4.0        workflowr_1.7.0     

loaded via a namespace (and not attached):
  [1] backports_1.2.1             Hmisc_4.7-2                
  [3] BiocFileCache_2.2.1         systemfonts_1.0.4          
  [5] plyr_1.8.8                  igraph_1.3.5               
  [7] lazyeval_0.2.2              splines_4.1.0              
  [9] BiocParallel_1.28.3         digest_0.6.31              
 [11] ensembldb_2.18.4            foreach_1.5.2              
 [13] htmltools_0.5.4             fansi_1.0.3                
 [15] checkmate_2.1.0             magrittr_2.0.3             
 [17] memoise_2.0.1               BSgenome_1.62.0            
 [19] cluster_2.1.2               doParallel_1.0.17          
 [21] tzdb_0.3.0                  Biostrings_2.62.0          
 [23] readr_2.1.3                 matrixStats_0.63.0         
 [25] vroom_1.6.0                 svglite_2.1.0              
 [27] prettyunits_1.1.1           jpeg_0.1-10                
 [29] colorspace_2.0-3            blob_1.2.3                 
 [31] rappdirs_0.3.3              xfun_0.35                  
 [33] dplyr_1.0.10                callr_3.7.3                
 [35] crayon_1.5.2                RCurl_1.98-1.9             
 [37] jsonlite_1.8.4              VariantAnnotation_1.40.0   
 [39] survival_3.2-11             iterators_1.0.14           
 [41] glue_1.6.2                  gtable_0.3.1               
 [43] zlibbioc_1.40.0             XVector_0.34.0             
 [45] DelayedArray_0.20.0         apcluster_1.4.10           
 [47] scales_1.2.1                DBI_1.1.3                  
 [49] rngtools_1.5.2              Rcpp_1.0.9                 
 [51] htmlTable_2.4.1             progress_1.2.2             
 [53] foreign_0.8-81              bit_4.0.5                  
 [55] Formula_1.2-4               htmlwidgets_1.6.0          
 [57] httr_1.4.4                  RColorBrewer_1.1-3         
 [59] ellipsis_0.3.2              pkgconfig_2.0.3            
 [61] XML_3.99-0.13               farver_2.1.0               
 [63] nnet_7.3-16                 sass_0.4.4                 
 [65] dbplyr_2.2.1                deldir_1.0-6               
 [67] utf8_1.2.2                  tidyselect_1.2.0           
 [69] labeling_0.4.2              rlang_1.0.6                
 [71] reshape2_1.4.4              later_1.3.0                
 [73] AnnotationDbi_1.56.2        munsell_0.5.0              
 [75] tools_4.1.0                 cachem_1.0.6               
 [77] cli_3.4.1                   generics_0.1.3             
 [79] RSQLite_2.2.19              evaluate_0.19              
 [81] stringr_1.5.0               fastmap_1.1.0              
 [83] yaml_2.3.6                  processx_3.8.0             
 [85] knitr_1.41                  bit64_4.0.5                
 [87] fs_1.5.2                    AnnotationFilter_1.18.0    
 [89] KEGGREST_1.34.0             doRNG_1.8.2                
 [91] whisker_0.4.1               xml2_1.3.3                 
 [93] biomaRt_2.50.3              compiler_4.1.0             
 [95] rstudioapi_0.14             filelock_1.0.2             
 [97] curl_4.3.2                  png_0.1-8                  
 [99] tibble_3.1.8                bslib_0.4.1                
[101] stringi_1.7.8               highr_0.9                  
[103] ps_1.7.2                    GenomicFeatures_1.46.5     
[105] lattice_0.20-44             ProtGenerics_1.26.0        
[107] Matrix_1.3-3                vctrs_0.5.1                
[109] pillar_1.8.1                lifecycle_1.0.3            
[111] jquerylib_0.1.4             data.table_1.14.6          
[113] bitops_1.0-7                httpuv_1.6.7               
[115] rtracklayer_1.54.0          R6_2.5.1                   
[117] BiocIO_1.4.0                latticeExtra_0.6-30        
[119] promises_1.2.0.1            gridExtra_2.3              
[121] codetools_0.2-18            dichromat_2.0-0.1          
[123] assertthat_0.2.1            SummarizedExperiment_1.24.0
[125] rprojroot_2.0.3             rjson_0.2.21               
[127] withr_2.5.0                 GenomicAlignments_1.30.0   
[129] Rsamtools_2.10.0            GenomeInfoDbData_1.2.7     
[131] parallel_4.1.0              hms_1.1.2                  
[133] rpart_4.1-15                rmarkdown_2.19             
[135] MatrixGenerics_1.6.0        git2r_0.30.1               
[137] biovizBase_1.42.0           getPass_0.2-2              
[139] Biobase_2.54.0              base64enc_0.1-3            
[141] interp_1.1-3                restfulr_0.0.15            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
