---
title: "LDL - Liver"
author: "sheng Qian"
date: "2022-10-18"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
params:
  analysis_id: "LDL_Liver"
  trait_id: "LDL_S"
  weight: "Liver"
---

```{r echo=F}
analysis_id <- params$analysis_id
trait_id <- params$trait_id
weight <- params$weight

results_dir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/", trait_id, "/", weight)

source("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/ctwas_config_b38.R")
options(digits = 4)
```

## Weight QC

```{r echo=F}
qclist_all <- list()

qc_files <- paste0(results_dir, "/", list.files(results_dir, pattern="exprqc.Rd"))

for (i in 1:length(qc_files)){
  load(qc_files[i])
  chr <- unlist(strsplit(rev(unlist(strsplit(qc_files[i], "_")))[1], "[.]"))[1]
  qclist_all[[chr]] <- cbind(do.call(rbind, lapply(qclist,unlist)), as.numeric(substring(chr,4)))
}

qclist_all <- data.frame(do.call(rbind, qclist_all))
colnames(qclist_all)[ncol(qclist_all)] <- "chr"

rm(qclist, wgtlist, z_gene_chr)
```

```{r}
#number of imputed weights
nrow(qclist_all)

#number of imputed weights by chromosome
table(qclist_all$chr)

#number of imputed weights without missing variants
sum(qclist_all$nmiss==0)

#proportion of imputed weights without missing variants
mean(qclist_all$nmiss==0)
```

```{r echo=F}
#load ctwas results
ctwas_res <- data.table::fread(paste0(results_dir, "/", analysis_id, "_ctwas.susieIrss.txt"))

#make unique identifier for regions
ctwas_res$region_tag <- paste(ctwas_res$region_tag1, ctwas_res$region_tag2, sep="_")

#load z scores for SNPs and collect sample size
load(paste0(results_dir, "/", analysis_id, "_expr_z_snp.Rd"))

sample_size <- z_snp$ss
sample_size <- as.numeric(names(which.max(table(sample_size))))

#compute PVE for each gene/SNP
ctwas_res$PVE = ctwas_res$susie_pip*ctwas_res$mu2/sample_size

#separate gene and SNP results
ctwas_gene_res <- ctwas_res[ctwas_res$type == "gene", ]
ctwas_gene_res <- data.frame(ctwas_gene_res)
ctwas_snp_res <- ctwas_res[ctwas_res$type == "SNP", ]
ctwas_snp_res <- data.frame(ctwas_snp_res)
```

```{r}
library(reticulate)
use_python("/scratch/midway2/shengqian/miniconda3/envs/PythonForR/bin/python",required=T)
```

```{python echo=F}
from pyensembl import EnsemblRelease #repl_python()
import math
import time
import numpy as np
import pandas as pd

data = EnsemblRelease(88)
ctwas_gene_res_df = r.ctwas_gene_res
ctwas_gene_res_df['intron_start'] = ctwas_gene_res_df['id'].apply(lambda x: int(x.split("_")[2]))
ctwas_gene_res_df['intron_end'] = ctwas_gene_res_df['id'].apply(lambda x: int(x.split("_")[3]))
ctwas_gene_res_df['genename'] = np.nan
ctwas_gene_res_df['intron_id'] = ctwas_gene_res_df['id']
```

```{python echo=F}
for index, row in ctwas_gene_res_df.iterrows():
  gene_names = data.gene_names_at_locus(contig=int(row["chrom"]), position=int(row["intron_start"]), end=int(row["intron_end"]))
  if len(gene_names)!=0:
    for i in gene_names:
      if i != '':
        gene_info = data.genes_by_name(i)[0]
        #if (gene_info.biotype=="lincRNA") | (gene_info.biotype=="protein_coding"):
        if gene_info.biotype=="protein_coding":
          ctwas_gene_res_df.loc[index,'genename'] = gene_info.gene_name
          ctwas_gene_res_df.loc[index,'id'] = gene_info.gene_id
          ctwas_gene_res_df.loc[index,'type'] = gene_info.biotype
          break
      
ctwas_gene_res_df = ctwas_gene_res_df.dropna(subset=['genename'])
print("finish")
```

```{r}
#add z scores to results
load(paste0(results_dir, "/", analysis_id, "_expr_z_gene.Rd"))
ctwas_gene_res <- py$ctwas_gene_res_df
ctwas_gene_res$z <- z_gene[ctwas_gene_res$intron_id,]$z

z_snp <- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z <- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

#merge gene and snp results with added information
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$intron_id=NA
ctwas_snp_res$intron_pos=NA

saveRDS(ctwas_gene_res, file = paste0(results_dir,"/",analysis_id,"_ctwas_intron_res.RDS"))
saveRDS(ctwas_snp_res, file =  paste0(results_dir,"/",analysis_id,"_ctwas_snp_res.RDS"))
```

```{r echo=F}
ctwas_intron_res <- readRDS(paste0(results_dir,"/",analysis_id,"_ctwas_intron_res.RDS"))
ctwas_intron_res <- ctwas_intron_res[!is.na(ctwas_intron_res$genename),]
ctwas_snp_res <- readRDS(paste0(results_dir,"/",analysis_id,"_ctwas_snp_res.RDS"))

#get number of sQTL for inrtons
num_sqtl <- c()
for (i in 1:22){
  load(paste0(results_dir, "/", analysis_id, "_expr_chr", i, ".exprqc.Rd"))
  num_sqtl <- c(num_sqtl, unlist(lapply(wgtlist, nrow)))
}
ctwas_intron_res$num_sqtl <- num_sqtl[ctwas_intron_res$intron_id]
```

```{r echo=F}
library(dplyr)
by_genename <- ctwas_intron_res %>% group_by(genename)
ctwas_gene_res <- data.frame(by_genename %>% summarise(
  chrom = chrom[1],
  id = id[1],
  pos = min(pos),
  type = type[1],
  region_tag1 = region_tag1[1],
  region_tag2 = names(sort(table(region_tag2),decreasing = TRUE)[1]),
  cs_index = cs_index[1],
  susie_pip = sum(susie_pip),
  #susie_pip = susie_pip[which.max(abs(susie_pip))],
  mu2 = mu2[which.max(abs(mu2))],
  region_tag = names(sort(table(region_tag),decreasing = TRUE)[1]),
  PVE = sum(susie_pip*PVE),
  z = z[which.max(abs(z))],
  gene_type = type[1],
  num_intron = length(intron_id),
  num_sqtl = sum(num_sqtl)))

ctwas_snp_res$num_intron <- NA
ctwas_snp_res$num_sqtl <- NA

saveRDS(ctwas_gene_res, file = paste0(results_dir,"/",analysis_id,"_ctwas_gene_res.RDS"))
```


```{r}
ctwas_res <- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#store columns to report
report_cols <- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c("type", "region_tag1", "region_tag2", "cs_index", "gene_type", "z_flag", "id", "chrom", "pos"))]
first_cols <- c("genename", "region_tag")
report_cols <- c(first_cols, report_cols[!(report_cols %in% first_cols)])

report_cols_snps <- c("id", report_cols[-1])
report_cols_snps <- report_cols_snps[!(report_cols_snps %in% "num_sqtl")]

#get number of SNPs from s1 results; adjust for thin argument
ctwas_res_s1 <- data.table::fread(paste0(results_dir, "/", analysis_id, "_ctwas.s1.susieIrss.txt"))
n_snps <- sum(ctwas_res_s1$type=="SNP")/thin
rm(ctwas_res_s1)
```

## Genes with highest PIPs 

```{r}
#distribution of PIPs
hist(ctwas_gene_res$susie_pip, xlim=c(0,1), main="Distribution of Gene PIPs")

#genes with PIP>0.8 or 20 highest PIPs
head(ctwas_gene_res[order(-ctwas_gene_res$susie_pip),report_cols], max(sum(ctwas_gene_res$susie_pip>0.8), 20))
```

## Comparing z scores and PIPs

```{r}
#set nominal signifiance threshold for z scores
alpha <- 0.05

#bonferroni adjusted threshold for z scores
sig_thresh <- qnorm(1-(alpha/nrow(ctwas_gene_res)/2), lower=T)
```

## PIP Manhattan Plot

```{r echo=F, message=F, eval=FALSE}
library(tibble)
library(tidyverse)

full.gene.pip.summary <- data.frame(gene_name = ctwas_gene_res$genename,
                                    gene_pip = ctwas_gene_res$susie_pip,
                                    gene_id = ctwas_gene_res$id,
                                    chr = as.integer(ctwas_gene_res$chrom),
                                    start = ctwas_gene_res$pos / 1e3,
                                    is_highlight = F, stringsAsFactors = F) %>% as_tibble()
full.gene.pip.summary$is_highlight <- full.gene.pip.summary$gene_pip > 0.8

don <- full.gene.pip.summary %>%

  # Compute chromosome size
  group_by(chr) %>%
  summarise(chr_len=max(start)) %>%

  # Calculate cumulative position of each chromosome
  mutate(tot=cumsum(chr_len)-chr_len) %>%
  dplyr::select(-chr_len) %>%

  # Add this info to the initial dataset
  left_join(full.gene.pip.summary, ., by=c("chr"="chr")) %>%

  # Add a cumulative position of each SNP
  arrange(chr, start) %>%
  mutate( BPcum=start+tot)

axisdf <- don %>% group_by(chr) %>% summarize(center=( max(BPcum) + min(BPcum) ) / 2 )

x_axis_labels <- axisdf$chr
x_axis_labels[seq(1,21,2)] <- ""

ggplot(don, aes(x=BPcum, y=gene_pip)) +

  # Show all points
  ggrastr::geom_point_rast(aes(color=as.factor(chr)), size=2) +
  scale_color_manual(values = rep(c("grey", "skyblue"), 22 )) +

  # custom X axis:
  # scale_x_continuous(label = axisdf$chr,
  #                    breaks= axisdf$center,
  #                    guide = guide_axis(n.dodge = 2)) +
  scale_x_continuous(label = x_axis_labels,
                     breaks = axisdf$center) +

  scale_y_continuous(expand = c(0, 0), limits = c(0,1.25), breaks=(1:5)*0.2, minor_breaks=(1:10)*0.1) + # remove space between plot area and x axis

  # Add highlighted points
  ggrastr::geom_point_rast(data=subset(don, is_highlight==T), color="orange", size=2) +

  # Add label using ggrepel to avoid overlapping
  ggrepel::geom_label_repel(data=subset(don, is_highlight==T),
                            aes(label=gene_name),
                            size=2,
                            min.segment.length = 0,
                            label.size = NA,
                            fill = alpha(c("white"),0)) +

  # Custom the theme:
  theme_bw() +
  theme(
    text = element_text(size = 14),
    legend.position="none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  ) +
  xlab("Chromosome") +
  ylab("cTWAS PIP")

```

## Sensitivity, specificity and precision for silver standard genes

```{r}
library("readxl")

known_annotations <- read_xlsx("data/summary_known_genes_annotations.xlsx", sheet="LDL")
known_annotations <- unique(known_annotations$`Gene Symbol`)

unrelated_genes <- ctwas_gene_res$genename[!(ctwas_gene_res$genename %in% known_annotations)]

#number of genes in known annotations
print(length(known_annotations))

#number of genes in known annotations with imputed expression
print(sum(known_annotations %in% ctwas_gene_res$genename))

#assign ctwas, TWAS, and bystander genes
ctwas_genes <- ctwas_gene_res$genename[ctwas_gene_res$susie_pip>0.9]

#significance threshold for TWAS
print(sig_thresh)

#number of ctwas genes
length(ctwas_genes)


#sensitivity / recall
sensitivity <- rep(NA,1)
names(sensitivity) <- c("ctwas")
sensitivity["ctwas"] <- sum(ctwas_genes %in% known_annotations)/length(known_annotations)
sensitivity

#specificity
specificity <- rep(NA,1)
names(specificity) <- c("ctwas")
specificity["ctwas"] <- sum(!(unrelated_genes %in% ctwas_genes))/length(unrelated_genes)
specificity

#precision / PPV
precision <- rep(NA,1)
names(precision) <- c("ctwas")
precision["ctwas"] <- sum(ctwas_genes %in% known_annotations)/length(ctwas_genes)
precision

#ROC curves

pip_range <- (0:1000)/1000
sensitivity <- rep(NA, length(pip_range))
specificity <- rep(NA, length(pip_range))
```