---
title: "OxHb"
author: "sheng Qian"
date: "2023-10-14"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

### Artery Coronary

```{r echo=FALSE, eval=FALSE}
ld_exprfs <- paste0(outputdir, outname,"_chr", 1:22, ".exprqc.Rd")
z_gene <- list()
for (j in 1:length(ld_exprfs)){
  load(ld_exprfs[j])
  z_gene[[j]] <- z_gene_chr
}
z_gene <- do.call(rbind, z_gene)
save(z_gene, file = paste0(outputdir, outname, "_z_gene.Rd"))
```


```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
library(ctwas)
weight <- "Artery_Coronary"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1
```


```{r}
ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
#print top genes
head(ctwas_res[ctwas_res$type=="gene",])
```


### Artery Tibial
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```


### Heart Atrial Appendage
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```


### Heart Left Ventricle
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```

### Kidney Cortex
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```

### Liver
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```

### Lung
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```

### Pancreas
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```

### Whole_Blood
```{r echo=FALSE}
#Artery_Coronary Artery_Tibial Heart_Atrial_Appendage Heart_Left_Ventricle Kidney_Cortex Liver Lung Pancreas Whole_Blood
weight <- "Artery_Tibial"
outputdir <- paste0("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/OxHb/",weight,"/")
outname <- paste0("OxHb_",weight,"_ctwas")
gwas_n <- 966
thin <- 0.1

ctwas_parameters <- ctwas_summarize_parameters(outputdir = outputdir, 
                                               outname = outname, 
                                               gwas_n = gwas_n, 
                                               thin = thin)
```


```{r}
#number of variants in the analysis
ctwas_parameters$n_snps

#number of genes in the analysis
ctwas_parameters$n_genes

#estimated prior inclusion probability
ctwas_parameters$group_prior

#estimated prior effect size
ctwas_parameters$group_prior_var

#estimated enrichment of genes over variants
ctwas_parameters$enrichment

#PVE explained by genes and variants
ctwas_parameters$group_pve

#total heritability (sum of PVE)
ctwas_parameters$total_pve

#attributable heritability
ctwas_parameters$attributable_pve

#plot convergence
ctwas_parameters$convergence_plot
```



```{r echo=FALSE}
#load cTWAS results
ctwas_res <- read.table(paste0(outputdir, outname, ".susieIrss.txt"), header=T)
load(paste0(outputdir, "/", outname, "_z_gene.Rd"))
load(paste0(outputdir, "/", outname, "_z_snp.Rd"))
#load gene information from PredictDB
sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, paste0("/project2/compbio/predictdb/mashr_models/mashr_", weight, ".db"))
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename, gene_type from extra")
RSQLite::dbDisconnect(db)
#add gene names to cTWAS results
ctwas_res$genename[ctwas_res$type=="gene"] <- gene_info$genename[match(ctwas_res$id[ctwas_res$type=="gene"], gene_info$gene)]
#add z-scores to cTWAS results
ctwas_res$z[ctwas_res$type=="SNP"] <- z_snp$z[match(ctwas_res$id[ctwas_res$type=="SNP"], z_snp$id)]
ctwas_res$z[ctwas_res$type=="gene"] <- z_gene$z[match(ctwas_res$id[ctwas_res$type=="gene"], z_gene$id)]
#display the genes with PIP > 0.8
ctwas_res <- ctwas_res[order(-ctwas_res$susie_pip),]
```

```{r}
head(ctwas_res[ctwas_res$type=="gene",])
```