---
title: "Simulation Results using three Tissues (uncorrelated)"
author: "shengqian"
date: "2023-10-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r}
source("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/analysis/simulation_help_functions.R")
```

## Simulation 1: Three causal tissues with equal high PVE (3% each)

### Separate effect size parameters

30% PVE and 2.5e-4 prior inclusion for SNPs, 3% PVE and 0.009 prior inclusion for Liver expression, 3% PVE and 0.009 prior inclusion for Lung expression, 3% PVE and 0.009 prior inclusion for Brain Hippocampus expression. For the cTWAS analysis, each tissue had its own prior inclusion parameter and effect size parameter.

```{r echo=FALSE}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(1, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r echo=FALSE}
results_df <- get_sim_joint_res_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using PIP threshold (gene+tissue)
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]

#mean percent causal using PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)

#results using combined PIP threshold
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]

#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)

#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)=="simutag"), setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df))))]

colMeans(results_df[,setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df)))])

#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)=="simutag"), grep("prior_var", names(results_df)))]

colMeans(results_df[,grep("prior_var", names(results_df))])

#PVE and mean PVE
results_df[,c(which(colnames(results_df)=="simutag"), grep("pve", names(results_df)))]

colMeans(results_df[,grep("pve", names(results_df))])

#TWAS results
results_df[,c(which(colnames(results_df)=="simutag"), grep("twas", names(results_df)))]

sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)
```

```{r echo=FALSE}
plot_df <- data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip-results_df$n_detected_comb_pip_in_causal,
                      ifcausal=F)

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip_in_causal,
                      ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas-results_df$n_detected_comb_twas_in_causal,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas_in_causal,
                            ifcausal=T))
```

```{r echo=FALSE}
#pdf(file = "./prior_set1.pdf", width = 5, height = 5)
y1 <- results_df$prior_weight1
y2 <- results_df$prior_weight2
y3 <- results_df$prior_weight3

truth <- rbind(c(1,0.009),c(2,0.009),c(3,0.009))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Brain"),ylim=c(0,0.025),ylab="Prior inclusion")
#dev.off()
```

```{r echo=FALSE}
y1 <- results_df$prior_weight1/results_df$prior_snp
y2 <- results_df$prior_weight2/results_df$prior_snp
y3 <- results_df$prior_weight3/results_df$prior_snp

truth <- rbind(c(1,36),c(2,36),c(3,36))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,70),ylab="Enrichment")
```

```{r echo=FALSE}
#pdf(file = "./variance_set1.pdf", width = 5, height = 5)
y1 <- results_df$pve_weight1
y2 <- results_df$pve_weight2
y3 <- results_df$pve_weight3

truth <- rbind(c(1,0.03),c(2,0.03),c(3,0.03))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.07),ylab="PVE")
#dev.off()
```

### Individual tissue analyses

```{r echo=FALSE}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(1, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r echo=FALSE}
results_df <- get_sim_ind_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using weight1
results_df[,c("simutag", colnames(results_df)[grep("weight1", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight1)/sum(results_df$n_detected_weight1)

#results using weight2
results_df[,c("simutag", colnames(results_df)[grep("weight2", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight2)/sum(results_df$n_detected_weight2)

#results using weight3
results_df[,c("simutag", colnames(results_df)[grep("weight3", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_weight3)/sum(results_df$n_detected_weight3)

#results using combined analysis
results_df[,c("simutag", colnames(results_df)[grep("combined", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_combined)/sum(results_df$n_detected_combined)
```

```{r echo=FALSE}
#store results for figure
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_weight1-results_df$n_detected_in_causal_weight1,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_in_causal_weight1,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_weight2-results_df$n_detected_in_causal_weight2,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_in_causal_weight2,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_weight3-results_df$n_detected_in_causal_weight3,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_in_causal_weight3,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_combined-results_df$n_detected_in_causal_combined,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_in_causal_combined,
                            ifcausal=T))
```

### Figure

```{r echo=FALSE}
#pdf(file = "./ncausal_set2.pdf", width = 5, height = 5)

plot_df$method <- factor(plot_df$method, levels=c("TWAS", "Multi-tissue cTWAS", "cTWAS-Cerebellum", "cTWAS-Hippocampus", "cTWAS-Caudate", "cTWAS Union"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

#dev.off()
```

```{r echo=FALSE}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(1, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

pip_df <- get_pip_distr_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
data <- data.frame(
  tissue=c(rep("Cerebellum",length(pip_df[[1]])), rep("Hippocampus",length(pip_df[[2]])), rep("Caudate",length(pip_df[[3]]))),
  attribution=c(pip_df[[1]],pip_df[[2]],pip_df[[3]])
)

data$tissue <- factor(data$tissue, levels=unique(data$tissue))

ggbarplot(data, 
          x = "tissue", 
          y = "attribution", 
          add = "mean_se", 
          fill = "tissue", 
          legend = "none", 
          ylab="Attribution", 
          xlab="",
          palette = c("#7fc97f", "#beaed4", "#fdc086")) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

## Simulation 2: Three causal tissues with equal low PVE (1% each)

### Separate effect size parameters

30% PVE and 2.5e-4 prior inclusion for SNPs, 1% PVE and 0.003 prior inclusion for Liver expression, 1% PVE and 0.003 prior inclusion for Lung expression, 1% PVE and 0.003 prior inclusion for Brain Hippocampus expression. Each tissue had its own prior inclusion parameter and effect size parameter. 

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(2, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_joint_res_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using PIP threshold (gene+tissue)
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]

#mean percent causal using PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)

#results using combined PIP threshold
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]

#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)

#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)=="simutag"), setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df))))]

colMeans(results_df[,setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df)))])

#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)=="simutag"), grep("prior_var", names(results_df)))]

colMeans(results_df[,grep("prior_var", names(results_df))])

#PVE and mean PVE
results_df[,c(which(colnames(results_df)=="simutag"), grep("pve", names(results_df)))]

colMeans(results_df[,grep("pve", names(results_df))])

#TWAS results
results_df[,c(which(colnames(results_df)=="simutag"), grep("twas", names(results_df)))]

sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)
```

```{r echo=FALSE}
plot_df <- data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip-results_df$n_detected_comb_pip_in_causal,
                      ifcausal=F)

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip_in_causal,
                      ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas-results_df$n_detected_comb_twas_in_causal,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas_in_causal,
                            ifcausal=T))
```

```{r}
y1 <- results_df$prior_weight1
y2 <- results_df$prior_weight2
y3 <- results_df$prior_weight3

truth <- rbind(c(1,0.003),c(2,0.003),c(3,0.003))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.025),ylab="Prior inclusion")
```

```{r}
y1 <- results_df$prior_weight1/results_df$prior_snp
y2 <- results_df$prior_weight2/results_df$prior_snp
y3 <- results_df$prior_weight3/results_df$prior_snp

truth <- rbind(c(1,12),c(2,12),c(3,12))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,60),ylab="Enrichment")
```

```{r}
y1 <- results_df$pve_weight1
y2 <- results_df$pve_weight2
y3 <- results_df$pve_weight3

truth <- rbind(c(1,0.01),c(2,0.01),c(3,0.01))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.05),ylab="PVE")
```

### Individual tissue analyses

For the cTWAS analysis, each tissue was analyzed individually and the results were combined.

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(2, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_ind_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using weight1
results_df[,c("simutag", colnames(results_df)[grep("weight1", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight1)/sum(results_df$n_detected_weight1)

#results using weight2
results_df[,c("simutag", colnames(results_df)[grep("weight2", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight2)/sum(results_df$n_detected_weight2)

#results using weight3
results_df[,c("simutag", colnames(results_df)[grep("weight3", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_weight3)/sum(results_df$n_detected_weight3)

#results using combined analysis
results_df[,c("simutag", colnames(results_df)[grep("combined", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_combined)/sum(results_df$n_detected_combined)
```

```{r echo=FALSE}
#store results for figure
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_weight1-results_df$n_detected_in_causal_weight1,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_in_causal_weight1,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_weight2-results_df$n_detected_in_causal_weight2,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_in_causal_weight2,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_weight3-results_df$n_detected_in_causal_weight3,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_in_causal_weight3,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_combined-results_df$n_detected_in_causal_combined,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_in_causal_combined,
                            ifcausal=T))
```

### Figure

```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "Multi-tissue cTWAS", "cTWAS-Cerebellum", "cTWAS-Hippocampus", "cTWAS-Caudate", "cTWAS Union"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

```

```{r echo=FALSE}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(2, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

pip_df <- get_pip_distr_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
data <- data.frame(
  tissue=c(rep("Liver",length(pip_df[[1]])), rep("Lung",length(pip_df[[2]])), rep("Brain",length(pip_df[[3]]))),
  attribution=c(pip_df[[1]],pip_df[[2]],pip_df[[3]])
)

data$tissue <- factor(data$tissue, levels=unique(data$tissue))

ggbarplot(data, 
          x = "tissue", 
          y = "attribution", 
          add = "mean_se", 
          fill = "tissue", 
          legend = "none", 
          ylab="Attribution", 
          xlab="",
          palette = c("#7fc97f", "#beaed4", "#fdc086")) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

## Simulation 3: Three causal tissues with unequal PVE

### Separate effect size parameters

30% PVE and 2.5e-4 prior inclusion for SNPs, 3% PVE and 0.009 prior inclusion for Liver expression, 2% PVE and 0.006 prior inclusion for Lung expression, 1% PVE and 0.003 prior inclusion for Brain Hippocampus expression. Each tissue had its own prior inclusion parameter and effect size parameter. 

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(3, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_joint_res_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using PIP threshold (gene+tissue)
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]

#mean percent causal using PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)

#results using combined PIP threshold
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]

#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)

#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)=="simutag"), setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df))))]

colMeans(results_df[,setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df)))])

#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)=="simutag"), grep("prior_var", names(results_df)))]

colMeans(results_df[,grep("prior_var", names(results_df))])

#PVE and mean PVE
results_df[,c(which(colnames(results_df)=="simutag"), grep("pve", names(results_df)))]

colMeans(results_df[,grep("pve", names(results_df))])

#TWAS results
results_df[,c(which(colnames(results_df)=="simutag"), grep("twas", names(results_df)))]

sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)
```

```{r echo=FALSE}
#store results for figure
plot_df <- data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip-results_df$n_detected_comb_pip_in_causal,
                      ifcausal=F)

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip_in_causal,
                      ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas-results_df$n_detected_comb_twas_in_causal,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas_in_causal,
                            ifcausal=T))
```

```{r}
y1 <- results_df$prior_weight1
y2 <- results_df$prior_weight2
y3 <- results_df$prior_weight3

truth <- rbind(c(1,0.009),c(2,0.006),c(3,0.003))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.025),ylab="Prior inclusion")
```

```{r}
y1 <- results_df$prior_weight1/results_df$prior_snp
y2 <- results_df$prior_weight2/results_df$prior_snp
y3 <- results_df$prior_weight3/results_df$prior_snp

truth <- rbind(c(1,36),c(2,24),c(3,12))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,60),ylab="Enrichment")
```

```{r}
y1 <- results_df$pve_weight1
y2 <- results_df$pve_weight2
y3 <- results_df$pve_weight3

truth <- rbind(c(1,0.03),c(2,0.02),c(3,0.01))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.05),ylab="PVE")
```

### Individual tissue analyses

For the cTWAS analysis, each tissue was analyzed individually and the results were combined.

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(3, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_ind_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using weight1
results_df[,c("simutag", colnames(results_df)[grep("weight1", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight1)/sum(results_df$n_detected_weight1)

#results using weight2
results_df[,c("simutag", colnames(results_df)[grep("weight2", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight2)/sum(results_df$n_detected_weight2)

#results using weight3
results_df[,c("simutag", colnames(results_df)[grep("weight3", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_weight3)/sum(results_df$n_detected_weight3)

#results using combined analysis
results_df[,c("simutag", colnames(results_df)[grep("combined", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_combined)/sum(results_df$n_detected_combined)
```

```{r echo=FALSE}
#store results for figure
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_weight1-results_df$n_detected_in_causal_weight1,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_in_causal_weight1,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_weight2-results_df$n_detected_in_causal_weight2,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_in_causal_weight2,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_weight3-results_df$n_detected_in_causal_weight3,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_in_causal_weight3,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_combined-results_df$n_detected_in_causal_combined,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_in_causal_combined,
                            ifcausal=T))
```

### Figure

For the cTWAS analysis, each tissue was analyzed individually and the results were combined.
```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "Multi-tissue cTWAS", "cTWAS-Cerebellum", "cTWAS-Hippocampus", "cTWAS-Caudate", "cTWAS Union"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

```

```{r echo=FALSE}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(3, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

pip_df <- get_pip_distr_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
data <- data.frame(
  tissue=c(rep("Liver",length(pip_df[[1]])), rep("Lung",length(pip_df[[2]])), rep("Brain",length(pip_df[[3]]))),
  attribution=c(pip_df[[1]],pip_df[[2]],pip_df[[3]])
)

data$tissue <- factor(data$tissue, levels=unique(data$tissue))

ggbarplot(data, 
          x = "tissue", 
          y = "attribution", 
          add = "mean_se", 
          fill = "tissue", 
          legend = "none", 
          ylab="Attribution", 
          xlab="",
          palette = c("#7fc97f", "#beaed4", "#fdc086")) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```
## Simulation 4: Two causal tissues with unequal PVE and one non-causal tissue

### Separate effect size parameters

30% PVE and 2.5e-4 prior inclusion for SNPs, 3% PVE and 0.009 prior inclusion for Liver expression, 1% PVE and 0.003 prior inclusion for Lung expression, Brain Hippocampus expression has no effects. Each tissue had its own prior inclusion parameter and effect size parameter. 

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(4, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_joint_res_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using PIP threshold (gene+tissue)
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]

#mean percent causal using PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)

#results using combined PIP threshold
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]

#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)

#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)=="simutag"), setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df))))]

colMeans(results_df[,setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df)))])

#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)=="simutag"), grep("prior_var", names(results_df)))]

colMeans(results_df[,grep("prior_var", names(results_df))])

#PVE and mean PVE
results_df[,c(which(colnames(results_df)=="simutag"), grep("pve", names(results_df)))]

colMeans(results_df[,grep("pve", names(results_df))])

#TWAS results
results_df[,c(which(colnames(results_df)=="simutag"), grep("twas", names(results_df)))]

sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)
```

```{r echo=FALSE}
#store results for figure
plot_df <- data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip-results_df$n_detected_comb_pip_in_causal,
                      ifcausal=F)

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip_in_causal,
                      ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas-results_df$n_detected_comb_twas_in_causal,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas_in_causal,
                            ifcausal=T))
```

```{r}
y1 <- results_df$prior_weight1
y2 <- results_df$prior_weight2
y3 <- results_df$prior_weight3

truth <- rbind(c(1,0.009),c(2,0.003),c(3,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.025),ylab="Prior inclusion")
```

```{r}
y1 <- results_df$prior_weight1/results_df$prior_snp
y2 <- results_df$prior_weight2/results_df$prior_snp
y3 <- results_df$prior_weight3/results_df$prior_snp

truth <- rbind(c(1,36),c(2,12),c(3,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,60),ylab="Enrichment")
```

```{r}
y1 <- results_df$pve_weight1
y2 <- results_df$pve_weight2
y3 <- results_df$pve_weight3

truth <- rbind(c(1,0.03),c(2,0.01),c(3,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.05),ylab="PVE")
```

### Individual tissue analyses

For the cTWAS analysis, each tissue was analyzed individually and the results were combined.

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(4, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_ind_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using weight1
results_df[,c("simutag", colnames(results_df)[grep("weight1", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight1)/sum(results_df$n_detected_weight1)

#results using weight2
results_df[,c("simutag", colnames(results_df)[grep("weight2", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight2)/sum(results_df$n_detected_weight2)

#results using weight3
results_df[,c("simutag", colnames(results_df)[grep("weight3", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_weight3)/sum(results_df$n_detected_weight3)

#results using combined analysis
results_df[,c("simutag", colnames(results_df)[grep("combined", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_combined)/sum(results_df$n_detected_combined)
```

```{r echo=FALSE}
#store results for figure
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_weight1-results_df$n_detected_in_causal_weight1,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_in_causal_weight1,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_weight2-results_df$n_detected_in_causal_weight2,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_in_causal_weight2,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_weight3-results_df$n_detected_in_causal_weight3,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_in_causal_weight3,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_combined-results_df$n_detected_in_causal_combined,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_in_causal_combined,
                            ifcausal=T))
```

### Figure

For the cTWAS analysis, each tissue was analyzed individually and the results were combined.
```{r echo=FALSE}
plot_df$method <- factor(plot_df$method, levels=c("TWAS", "Multi-tissue cTWAS", "cTWAS-Cerebellum", "cTWAS-Hippocampus", "cTWAS-Caudate", "cTWAS Union"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

```

```{r echo=FALSE}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(4, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

pip_df <- get_pip_distr_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
data <- data.frame(
  tissue=c(rep("Liver",length(pip_df[[1]])), rep("Lung",length(pip_df[[2]])), rep("Brain",length(pip_df[[3]]))),
  attribution=c(pip_df[[1]],pip_df[[2]],pip_df[[3]])
)

data$tissue <- factor(data$tissue, levels=unique(data$tissue))

ggbarplot(data, 
          x = "tissue", 
          y = "attribution", 
          add = "mean_se", 
          fill = "tissue", 
          legend = "none", 
          ylab="Attribution", 
          xlab="",
          palette = c("#7fc97f", "#beaed4", "#fdc086")) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
```

## Simulation 5: one causal tissues and two non-causal tissues

### Separate effect size parameters

30% PVE and 2.5e-4 prior inclusion for SNPs, 3% PVE and 0.009 prior inclusion for Liver expression, Lung expression and Brain Hippocampus expression has no effects. Each tissue had its own prior inclusion parameter and effect size parameter. 

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(5, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_joint_res_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r}
#results using PIP threshold (gene+tissue)
results_df[,c("simutag", "n_causal", "n_detected_pip", "n_detected_pip_in_causal")]

#mean percent causal using PIP > 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)

#results using combined PIP threshold
results_df[,c("simutag", "n_causal_combined", "n_detected_comb_pip", "n_detected_comb_pip_in_causal")]

#mean percent causal using combined PIP > 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)

#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)=="simutag"), setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df))))]

colMeans(results_df[,setdiff(grep("prior", names(results_df)), grep("prior_var", names(results_df)))])

#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)=="simutag"), grep("prior_var", names(results_df)))]

colMeans(results_df[,grep("prior_var", names(results_df))])

#PVE and mean PVE
results_df[,c(which(colnames(results_df)=="simutag"), grep("pve", names(results_df)))]

colMeans(results_df[,grep("pve", names(results_df))])

#TWAS results
results_df[,c(which(colnames(results_df)=="simutag"), grep("twas", names(results_df)))]

sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)
```

```{r echo=FALSE}
#store results for figure
plot_df <- data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip-results_df$n_detected_comb_pip_in_causal,
                      ifcausal=F)

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                      method="Multi-tissue cTWAS",
                      count=results_df$n_detected_comb_pip_in_causal,
                      ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas-results_df$n_detected_comb_twas_in_causal,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="TWAS",
                            count=results_df$n_detected_comb_twas_in_causal,
                            ifcausal=T))
```

```{r}
#pdf(file = "./prior_set2.pdf", width = 5, height = 5)
y1 <- results_df$prior_weight1
y2 <- results_df$prior_weight2
y3 <- results_df$prior_weight3

truth <- rbind(c(1,0.009),c(2,0),c(3,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.025),ylab="Prior inclusion")
#dev.off()
```

```{r}
#pdf(file = "./enrichment_set2.pdf", width = 5, height = 5)
y1 <- results_df$prior_weight1/results_df$prior_snp
y2 <- results_df$prior_weight2/results_df$prior_snp
y3 <- results_df$prior_weight3/results_df$prior_snp

truth <- rbind(c(1,36),c(2,0),c(3,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,60),ylab="Enrichment")
#dev.off()
```

```{r}
pdf(file = "./pev_set2.pdf", width = 5, height = 5)
y1 <- results_df$pve_weight1
y2 <- results_df$pve_weight2
y3 <- results_df$pve_weight3

truth <- rbind(c(1,0.03),c(2,0),c(3,0))
est <- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3))
plot_par(truth,est,xlabels = c("Liver","Lung","Hippocampus"),ylim=c(0,0.05),ylab="PVE")
dev.off()
```

### Individual tissue analyses

For the cTWAS analysis, each tissue was analyzed individually and the results were combined.

```{r}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(5, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8
```

```{r}
results_df <- get_sim_ind_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
```

```{r echo=FALSE}
#results using weight1
results_df[,c("simutag", colnames(results_df)[grep("weight1", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight1)/sum(results_df$n_detected_weight1)

#results using weight2
results_df[,c("simutag", colnames(results_df)[grep("weight2", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight1
sum(results_df$n_detected_in_causal_weight2)/sum(results_df$n_detected_weight2)

#results using weight3
results_df[,c("simutag", colnames(results_df)[grep("weight3", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_weight3)/sum(results_df$n_detected_weight3)

#results using combined analysis
results_df[,c("simutag", colnames(results_df)[grep("combined", colnames(results_df))])]

#mean percent causal using PIP > 0.8 for weight3
sum(results_df$n_detected_in_causal_combined)/sum(results_df$n_detected_combined)
```

```{r echo=FALSE}
#store results for figure
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_weight1-results_df$n_detected_in_causal_weight1,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Cerebellum",
                            count=results_df$n_detected_in_causal_weight1,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_weight2-results_df$n_detected_in_causal_weight2,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Hippocampus",
                            count=results_df$n_detected_in_causal_weight2,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_weight3-results_df$n_detected_in_causal_weight3,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS-Caudate",
                            count=results_df$n_detected_in_causal_weight3,
                            ifcausal=T))

plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_combined-results_df$n_detected_in_causal_combined,
                            ifcausal=F))
plot_df <- rbind(plot_df,
                 data.frame(simutag=results_df$simutag,
                            method="cTWAS Union",
                            count=results_df$n_detected_in_causal_combined,
                            ifcausal=T))
```

### Figure

For the cTWAS analysis, each tissue was analyzed individually and the results were combined.
```{r echo=FALSE}
#pdf(file = "./ncausal_set2_simple.pdf", width = 2, height = 5)

plot_df$method <- factor(plot_df$method, levels=c("TWAS", "Multi-tissue cTWAS", "cTWAS-Cerebellum", "cTWAS-Hippocampus", "cTWAS-Caudate", "cTWAS Union"))

library(ggpubr)

colset = c("#ebebeb", "#fb8072")

ggbarplot(plot_df, 
          x = "method", 
          y = "count", 
          add = "mean_se", 
          fill = "ifcausal", 
          legend = "none", 
          ylab="Count", 
          xlab="",
          palette = colset) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))

#dev.off()

```

```{r echo=FALSE}
results_dir <- "/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_correlated_drop_merge/"
runtag = "ukb-s80.45-3_corr"
configtag <- 2

simutags <- paste(5, 1:5, sep = "-")
thin <- 0.1

sample_size <- 45000
PIP_threshold <- 0.8

pip_df <- get_pip_distr_corr(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)
data <- data.frame(
  tissue=c(rep("Liver",length(pip_df[[1]])), rep("Lung",length(pip_df[[2]])), rep("Brain",length(pip_df[[3]]))),
  attribution=c(pip_df[[1]],pip_df[[2]],pip_df[[3]])
)

data$tissue <- factor(data$tissue, levels=unique(data$tissue))

#pdf(file = "./primary_set2.pdf", width = 5, height = 5)
ggbarplot(data, 
          x = "tissue", 
          y = "attribution", 
          add = "mean_se", 
          fill = "tissue", 
          legend = "none", 
          ylab="Proportion", 
          xlab="",
          palette = c("#7fc97f", "#beaed4", "#fdc086")) + grids(linetype = "dashed") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()
```


```{r}
#load cTWAS results
ctwas_res <- data.table::fread("/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_uncorrelated_drop_merge/ukb-s80.45-3_uncorr_simu5-2_config2_LDR.drop.merge.susieIrss.txt")
ctwas_res$region_tag <- paste(ctwas_res$region_tag1, ctwas_res$region_tag2, sep="_")
ctwas_gene_res <- ctwas_res[ctwas_res$type != "SNP", ]
ctwas_gene_res <- data.frame(ctwas_gene_res)
ctwas_snp_res <- ctwas_res[ctwas_res$type == "SNP", ]
ctwas_snp_res <- data.frame(ctwas_snp_res)
load("/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_uncorrelated_drop_merge/ukb-s80.45-3_uncorr_simu5-2_config2_LDR.drop.merge_z_snp.Rd")
z_snp <- z_snp[z_snp$id %in% ctwas_snp_res$id,]
ctwas_snp_res$z <- z_snp$z[match(ctwas_snp_res$id, z_snp$id)]

load("/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_uncorrelated_drop_merge/ukb-s80.45-3_uncorr_simu5-2_config2_LDR.drop.merge_z_gene.Rd")
z_gene$gene_id <- sapply(z_gene$id, function(x){unlist(strsplit(x, split="[|]"))[1]})
ctwas_gene_res$gene_id <- sapply(ctwas_gene_res$id, function(x){unlist(strsplit(x, split="[|]"))[1]})
ctwas_gene_res$group <- sapply(ctwas_gene_res$id, function(x){paste(unlist(strsplit(unlist(strsplit(x, split="[|]"))[2], "_"))[-1], collapse="_")})

ctwas_gene_res$group[ctwas_gene_res$group=="Liver_harmonized"] <- "Liver"
ctwas_gene_res$group[ctwas_gene_res$group=="Lung_harmonized"] <- "Lung"
ctwas_gene_res$group[ctwas_gene_res$group=="Brain_Hippocampus_harmonized"] <- "Brain"
ctwas_gene_Liver_res <- ctwas_gene_res[ctwas_gene_res$group=="Liver",]
ctwas_gene_Lung_res <- ctwas_gene_res[ctwas_gene_res$group=="Lung",]
ctwas_gene_Brain_res <- ctwas_gene_res[ctwas_gene_res$group=="Brain",]

sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, "/project2/compbio/predictdb/mashr_models/mashr_Liver.db")
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename from extra")
RSQLite::dbDisconnect(db)
ctwas_gene_Liver_res$genename <- gene_info[sapply(ctwas_gene_Liver_res$gene_id, match, gene_info$gene),"genename"]

sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, "/project2/compbio/predictdb/mashr_models/mashr_Lung.db")
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename from extra")
RSQLite::dbDisconnect(db)
ctwas_gene_Lung_res$genename <- gene_info[sapply(ctwas_gene_Lung_res$gene_id, match, gene_info$gene),"genename"]

sqlite <- RSQLite::dbDriver("SQLite")
db = RSQLite::dbConnect(sqlite, "/project2/compbio/predictdb/mashr_models/mashr_Brain_Hippocampus.db")
query <- function(...) RSQLite::dbGetQuery(db, ...)
gene_info <- query("select gene, genename from extra")
RSQLite::dbDisconnect(db)
ctwas_gene_Brain_res$genename <- gene_info[sapply(ctwas_gene_Brain_res$gene_id, match, gene_info$gene),"genename"]

ctwas_gene_res <- rbind(ctwas_gene_Liver_res,ctwas_gene_Lung_res,ctwas_gene_Brain_res)
ctwas_gene_res$type <- "gene"
ctwas_gene_res$z <- z_gene$z[match(ctwas_gene_res$id, z_gene$id)]
ctwas_gene_res$plot_name <- paste(ctwas_gene_res$genename, ctwas_gene_res$group, sep="-")

#merge gene and snp results with added information
ctwas_snp_res$gene_id=NA
ctwas_snp_res$group="SNP"
#ctwas_snp_res$alt_name=NA
ctwas_snp_res$genename=NA
ctwas_snp_res$gene_type=NA
ctwas_snp_res$plot_name=NA

ctwas_res <- rbind(ctwas_gene_res,
                   ctwas_snp_res[,colnames(ctwas_gene_res)])

#store columns to report
report_cols <- colnames(ctwas_gene_res)[!(colnames(ctwas_gene_res) %in% c("type", "region_tag1", "region_tag2", "cs_index", "gene_type", "z_flag", "id", "chrom", "pos", "alt_name", "gene_id"))]
first_cols <- c("genename", "gene_type", "gene_id", "group", "region_tag")
report_cols <- c(first_cols, report_cols[!(report_cols %in% first_cols)])
```

```{r echo=FALSE}
load("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/data/G_list.RData")
G_list <- G_list[G_list$gene_biotype %in% c("protein_coding"),]
G_list$hgnc_symbol[G_list$hgnc_symbol==""] <- "-"
G_list$tss <- G_list[,c("end_position", "start_position")][cbind(1:nrow(G_list),G_list$strand/2+1.5)]
alpha <- 0.05
source("/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/code/locus_plot.R")
```


```{r echo=FALSE, eval=FALSE}
region_tag = "16_2"
return_table=F
focus='ENSG00000127561.14|mashr_Liver_harmonized'
label_panel="both"
label_genes=NULL
label_pos=NULL
plot_eqtl=NULL
rerun_ctwas=F
rerun_load_only=F
legend_side="left"
legend_panel="cTWAS"
twas_ymax=NULL
alpha <- 0.05
#xlim=NULL
xlim=c(1.95,2.05)
```

```{r echo=FALSE, eval=FALSE}
pdf(file = "./simulation_locus.pdf", width = 5, height = 3.5)
region_tag1 <- unlist(strsplit(region_tag, "_"))[1]
region_tag2 <- unlist(strsplit(region_tag, "_"))[2]
  
a <- ctwas_res[ctwas_res$region_tag==region_tag,]
regionlist <- readRDS("/project2/xinhe/shengqian/cTWAS/cTWAS_simulation/simulation_uncorrelated_drop_merge/ukb-s80.45-3_uncorr_simu5-2_config2_LDR.drop.merge.regionlist.RDS")
region <- regionlist[[as.numeric(region_tag1)]][[region_tag2]]
  
R_snp_info <- do.call(rbind, lapply(region$regRDS, function(x){data.table::fread(paste0(tools::file_path_sans_ext(x), ".Rvar"))}))
  
a_pos_bkup <- a$pos
a$pos[a$type=="gene"] <- G_list$tss[match(sapply(a$genename[a$type=="gene"], function(x){unlist(strsplit(x, "[.]"))[1]}) ,G_list$hgnc_symbol)]
a$pos[is.na(a$pos)] <- a_pos_bkup[is.na(a$pos)]
a$pos <- a$pos/1000000
  
if (!is.null(xlim)){
    if (is.na(xlim[1])){
        xlim[1] <- min(a$pos)
    }
    
    if (is.na(xlim[2])){
        xlim[2] <- max(a$pos)
    }
    a <- a[a$pos>=xlim[1] & a$pos<=xlim[2],,drop=F]
}
  
if (is.null(focus)){
    focus <- a$id[which.max(abs(a$z)[a$type=="gene"])]
}
  
if (is.null(label_genes)){
    label_genes <- focus
}
  
if (is.null(label_pos)){
    label_pos <- rep(3, length(label_genes))
}
  
if (is.null(plot_eqtl)){
    plot_eqtl <- focus
}
  
focus <- a$id[which(a$id==focus)]
a$focus <- 0
a$focus <- as.numeric(a$id==focus)
    
a$PVALUE <- (-log(2) - pnorm(abs(a$z), lower.tail=F, log.p=T))/log(10)
  
R_gene <- readRDS(region$R_g_file)
R_snp_gene <- readRDS(region$R_sg_file)
R_snp <- as.matrix(Matrix::bdiag(lapply(region$regRDS, readRDS)))
  
rownames(R_gene) <- region$gid
colnames(R_gene) <- region$gid
rownames(R_snp_gene) <- R_snp_info$id
colnames(R_snp_gene) <- region$gid
rownames(R_snp) <- R_snp_info$id
colnames(R_snp) <- R_snp_info$id
  
a$r2max <- NA
a$r2max[a$type=="gene"] <- R_gene[focus,a$id[a$type=="gene"]]
a$r2max[a$type=="SNP"] <- R_snp_gene[a$id[a$type=="SNP"],focus]
  
r2cut <- 0.4
colorsall <- c("#7fc97f", "#beaed4", "#fdc086")
  
start <- min(a$pos)
end <- max(a$pos)
  
layout(matrix(1:3, ncol = 1), widths = 1, heights = c(1.5,1.75,0.75), respect = FALSE)
  
par(mar = c(0, 4.1, 0, 2.1))
  
if (is.null(twas_ymax)){
    twas_ymax <- max(a$PVALUE)*1.1
}
  
plot(a$pos[a$type=="SNP"], a$PVALUE[a$type == "SNP"], pch = 21, xlab=paste0("Chromosome ", region_tag1, " position (Mb)"), frame.plot=FALSE, bg = colorsall[1], ylab = "-log10(p value)", panel.first = grid(), ylim =c(0, twas_ymax), xaxt = 'n', xlim=c(start, end))
  
abline(h=-log10(alpha/nrow(ctwas_gene_res)), col ="red", lty = 2)
points(a$pos[a$type=="SNP" & a$r2max > r2cut], a$PVALUE[a$type == "SNP"  & a$r2max > r2cut], pch = 21, bg = "purple")
points(a$pos[a$type=="SNP" & a$focus == 1], a$PVALUE[a$type == "SNP" & a$focus == 1], pch = 21, bg = "salmon")
points(a$pos[a$type=="gene" & a$group=="Liver"], a$PVALUE[a$type == "gene" & a$group=="Liver"], pch = 22, bg = colorsall[1], cex = 2)
points(a$pos[a$type=="gene" & a$group=="Lung"], a$PVALUE[a$type == "gene" & a$group=="Lung"], pch = 23, bg = colorsall[1], cex = 2)
points(a$pos[a$type=="gene" & a$group=="Brain"], a$PVALUE[a$type == "gene" & a$group=="Brain"], pch = 24, bg = colorsall[1], cex = 2)

points(a$pos[a$type=="gene" & a$r2max > r2cut & a$group=="Liver"], a$PVALUE[a$type == "gene"  & a$r2max > r2cut & a$group=="Liver"], pch = 22, bg = "purple", cex = 2)
points(a$pos[a$type=="gene" & a$r2max > r2cut & a$group=="Lung"], a$PVALUE[a$type == "gene"  & a$r2max > r2cut & a$group=="Lung"], pch = 23, bg = "purple", cex = 2)
points(a$pos[a$type=="gene" & a$r2max > r2cut & a$group=="Brain"], a$PVALUE[a$type == "gene"  & a$r2max > r2cut & a$group=="Brain"], pch = 24, bg = "purple", cex = 2)

points(a$pos[a$type=="gene" & a$focus == 1 & a$group=="Liver"], a$PVALUE[a$type == "gene" & a$focus == 1 & a$group=="Liver"], pch = 22, bg = "salmon", cex = 2)
points(a$pos[a$type=="gene" & a$focus == 1 & a$group=="Lung"], a$PVALUE[a$type == "gene" & a$focus == 1 & a$group=="Lung"], pch = 23, bg = "salmon", cex = 2)
points(a$pos[a$type=="gene" & a$focus == 1 & a$group=="Brain"], a$PVALUE[a$type == "gene" & a$focus == 1 & a$group=="Brain"], pch = 24, bg = "salmon", cex = 2)
  
if (legend_panel=="TWAS"){
    x_pos <- ifelse(legend_side=="right", max(a$pos)-0.2*(max(a$pos)-min(a$pos)), min(a$pos))
    legend(x_pos, y= twas_ymax*0.95, c("Liver","Lung","Brain","SNP","Lead TWAS Gene", "R2 > 0.4", "R2 <= 0.4"), pch = c(22,23,24,21,19,19,19), col = c("black","black","black","black", "salmon", "purple", colorsall[1]), cex=0.7, title.adj = 0)
}
  

label_genes <- a[a$id==label_genes,]$genename


if (label_panel=="TWAS" | label_panel=="both"){
    for (i in 1:length(label_genes)){
        text(a$pos[a$genename==label_genes[i]], a$PVALUE[a$genename==label_genes[i]], labels=a$plot_name[a$genename==label_genes[i]], pos=label_pos[i], cex=0.7)
    }
}
  

par(mar = c(4.1, 4.1, 0, 2.1))
  
plot(a$pos[a$type=="SNP"], a$susie_pip[a$type == "SNP"], pch = 19, xlab=paste0("Chromosome ", region_tag1, " position (Mb)"),frame.plot=FALSE, col = "white", ylim= c(0,1.1), ylab = "cTWAS PIP", xlim = c(start, end))
  
grid()
points(a$pos[a$type=="SNP"], a$susie_pip[a$type == "SNP"], pch = 21, xlab="Genomic position", bg = colorsall[1])
points(a$pos[a$type=="SNP" & a$r2max > r2cut], a$susie_pip[a$type == "SNP"  & a$r2max >r2cut], pch = 21, bg = "purple")
points(a$pos[a$type=="SNP" & a$focus == 1], a$susie_pip[a$type == "SNP" & a$focus == 1], pch = 21, bg = "salmon")
points(a$pos[a$type=="gene" & a$group=="Liver"], a$susie_pip[a$type == "gene" & a$group=="Liver"], pch = 22, bg = colorsall[1], cex = 2)
points(a$pos[a$type=="gene" & a$group=="Lung"], a$susie_pip[a$type == "gene" & a$group=="Lung"], pch = 23, bg = colorsall[1], cex = 2)
points(a$pos[a$type=="gene" & a$group=="Brain"], a$susie_pip[a$type == "gene" & a$group=="Brain"], pch = 24, bg = colorsall[1], cex = 2)
points(a$pos[a$type=="gene" & a$r2max > r2cut & a$group=="Liver"], a$susie_pip[a$type == "gene"  & a$r2max > r2cut & a$group=="Liver"], pch = 22, bg = "purple", cex = 2)
points(a$pos[a$type=="gene" & a$r2max > r2cut & a$group=="Lung"], a$susie_pip[a$type == "gene"  & a$r2max > r2cut & a$group=="Lung"], pch = 23, bg = "purple", cex = 2)
points(a$pos[a$type=="gene" & a$r2max > r2cut & a$group=="Brain"], a$susie_pip[a$type == "gene"  & a$r2max > r2cut & a$group=="Brain"], pch = 24, bg = "purple", cex = 2)

points(a$pos[a$type=="gene" & a$focus == 1 & a$group=="Liver"], a$susie_pip[a$type == "gene" & a$focus == 1 & a$group=="Liver"], pch = 22, bg = "salmon", cex = 2)
points(a$pos[a$type=="gene" & a$focus == 1 & a$group=="Lung"], a$susie_pip[a$type == "gene" & a$focus == 1 & a$group=="Lung"], pch = 23, bg = "salmon", cex = 2)
points(a$pos[a$type=="gene" & a$focus == 1 & a$group=="Brain"], a$susie_pip[a$type == "gene" & a$focus == 1 & a$group=="Brain"], pch = 24, bg = "salmon", cex = 2)
  
if (legend_panel=="cTWAS"){
    x_pos <- ifelse(legend_side=="right", max(a$pos)-0.2*(max(a$pos)-min(a$pos)), min(a$pos))
    legend(x_pos, y= 1 ,c("Liver","Lung","Brain","SNP","Lead TWAS Gene", "R2 > 0.4", "R2 <= 0.4"), pch = c(22,23,24,21,19,19,19), col = c("black","black","black", "black", "salmon", "purple", colorsall[1]), cex=0.7, title.adj = 0)
}

  
if (label_panel=="cTWAS" | label_panel=="both"){
    for (i in 1:length(label_genes)){
    text(a$pos[a$genename==label_genes[i]], a$susie_pip[a$genename==label_genes[i]], labels=a$plot_name[a$genename==label_genes[i]], pos=label_pos[i], cex=0.7)
    }
}
  
if (return_table){
    return(a)
}
dev.off()
```

```{r}
pdf(file = "./simulation_genetrack.pdf", width = 3.86, height = 0.2)
locus_plot_gene_track_pub(16, 1.1e+07, 11169507, label_pos="above")
dev.off()
```